

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Preprocessing &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../../index.html"/>
        <link rel="up" title="Transcriptomics" href="../index.html"/>
        <link rel="next" title="Init" href="../INIT/index.html"/>
        <link rel="prev" title="Moomin" href="../MOOMIN/index.html"/> 

  
  <script src="../../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../../_static/../" + item.website_url + "' ><img src='../../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../index.html">Functions</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../../analysis/index.html">Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../base/index.html">Base</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../index.html">Dataintegration</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../index.html">Transcriptomics</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="../SWIFTCORE/index.html">Swiftcore</a></li>
<li class="toctree-l4"><a class="reference internal" href="../MOOMIN/index.html">Moomin</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="#">Preprocessing</a></li>
<li class="toctree-l4"><a class="reference internal" href="../INIT/index.html">Init</a></li>
<li class="toctree-l4"><a class="reference internal" href="../FASTCORE/index.html">Fastcore</a></li>
<li class="toctree-l4"><a class="reference internal" href="../GIMME/index.html">Gimme</a></li>
<li class="toctree-l4"><a class="reference internal" href="../MBA/index.html">Mba</a></li>
<li class="toctree-l4"><a class="reference internal" href="../iMAT/index.html">Imat</a></li>
<li class="toctree-l4"><a class="reference internal" href="../eFlux/index.html">Eflux</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../metaboAnnotator/index.html">Metaboannotator</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../metanetxMapper/index.html">Metanetxmapper</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../metabotools/index.html">Metabotools</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../chemoInformatics/index.html">Chemoinformatics</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../XomicsToModel/index.html">Xomicstomodel</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../fluxomics/index.html">Fluxomics</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citations.html">Citations</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Dataintegration</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Transcriptomics</a></li>
      <li class="breadcrumb-item active">Preprocessing</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="preprocessing">
<span id="id1"></span><h1>Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.dataIntegration.transcriptomics.preprocessing"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.transcriptomics.preprocessing.GPRparser">
<span class="sig-name descname"><span class="pre">GPRparser</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">getCNFSets</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/dataIntegration/transcriptomics/preprocessing/GPRparser.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.dataIntegration.transcriptomics.preprocessing.GPRparser" title="Permalink to this definition">¶</a></dt>
<dd><p>Maps the GPR rules of the model to a specified format that is used by
the model extraction methods</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>parsedGPR = GPRparser</strong> (<em>model</em>)</p>
</dd>
<dt class="field-even">INPUT<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>model</strong> – cobra model structure</p>
</dd>
<dt class="field-odd">OPTIONAL INPUT<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>getCNFSets</strong> – whether to get the CNF sets (true) or DNF sets (false).
DNF sets represent functional enzyme complexes, while
CNF sets represent the possible subunits of a complex.
(default: false , i.e. DNF sets)</p>
</dd>
<dt class="field-even">OUTPUT<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>parsedGPR</strong> – cell matrix containing parsed GPR rule</p>
</dd>
</dl>
<p>AUTHORS: Thomas Pfau &amp; Anne Richelle, May 2017</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.transcriptomics.preprocessing.findUsedGenesLevels">
<span class="sig-name descname"><span class="pre">findUsedGenesLevels</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">exprData</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/dataIntegration/transcriptomics/preprocessing/findUsedGenesLevels.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.dataIntegration.transcriptomics.preprocessing.findUsedGenesLevels" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns vectors of gene identifiers and corresponding gene expression
levels for each gene present in the model (‘model.genes’).</p>
<dl class="field-list">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>[gene_id, gene_expr] = findUsedGenesLevels</strong> (<em>model, exprData</em>)</p></li>
<li><p><strong>[gene_id, gene_expr, gene_sig] = findUsedGenesLevels</strong> (<em>model, exprData</em>)</p></li>
</ul>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul>
<li><p><strong>model</strong> – input model (COBRA model structure)</p></li>
<li><p><strong>exprData</strong> – mRNA expression data structure
.gene                cell array containing GeneIDs in the same</p>
<blockquote>
<div><p>format as model.genes</p>
</div></blockquote>
<p>.value               Vector containing corresponding expression value (FPKM)
.sig:                [optional field] Vector containing significance values of</p>
<blockquote>
<div><p>expression corresponding to expression values in exprData.value (ex. p-values)</p>
</div></blockquote>
</li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>printLevel</strong> – Printlevel for output (default 0);</p>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>gene_id</strong> – vector of gene identifiers present in the model
that are associated with expression data</p></li>
<li><p><strong>gene_expr</strong> – vector of expression values associated to each
‘gene_id’</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>OPTIONAL OUTPUTS:</dt><dd><dl class="simple">
<dt>gene_sig:             vector of significance values associated to each</dt><dd><p>‘gene_id’</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.transcriptomics.preprocessing.mapExpressionToReactions">
<span class="sig-name descname"><span class="pre">mapExpressionToReactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">expressionData</span></em>, <em class="sig-param"><span class="pre">minSum</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/dataIntegration/transcriptomics/preprocessing/mapExpressionToReactions.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.dataIntegration.transcriptomics.preprocessing.mapExpressionToReactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Determines the expression data associated to each reaction present in
the model</p>
<dl class="field-list">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>[expressionRxns parsedGPR, gene_used] = mapExpressionToReactions</strong> (<em>model, expressionData</em>)</p></li>
<li><p><strong>[expressionRxns, parsedGPR, gene_used, signifRxns] =  mapExpressionToReactions</strong> (<em>model, expressionData, minSum</em>)</p></li>
</ul>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><dl class="simple">
<dt><strong>model                   model strusture</strong></dt><dd><ul>
<li><p><strong>expressionData          mRNA expression data structure</strong></p></li>
</ul>
</dd>
</dl>
</li>
</ul>
<dl class="simple">
<dt>.gene                    cell array containing GeneIDs in the same</dt><dd><p>format as model.genes</p>
</dd>
<dt>.value                  Vector containing corresponding expression</dt><dd><p>value (FPKM/RPKM)</p>
</dd>
<dt>.sig:               [optional field] Vector containing significance values of</dt><dd><p>expression corresponding to expression values in
expressionData.value (ex. p-values)</p>
</dd>
</dl>
</dd>
<dt class="field-odd">OPTIONAL INPUT<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>minSum</strong> – instead of using min and max, use min for AND and Sum
for OR (default: false, i.e. use min)</p>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>expressionRxns</strong> – n x 1 non-negative value for reaction expression, corresponding to model.rxns.
expressionRxns(j) is NaN when there is no expression data for the genes corresponding to reaction j.</p></li>
<li><p><strong>parsedGPR</strong> – cell matrix containing parsed GPR rule</p></li>
<li><p><strong>gene_used</strong> – gene identifier, corresponding to model.rxns, from GPRs
whose value (expression and/or significance) was chosen for that
reaction</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>OPTIONAL OUTPUTS:</dt><dd><p>signifRxns:              significance of reaction expression, corresponding to model.rxns.</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.dataIntegration.transcriptomics.preprocessing.selectGeneFromGPR">
<span class="sig-name descname"><span class="pre">selectGeneFromGPR</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">gene_names</span></em>, <em class="sig-param"><span class="pre">gene_exp</span></em>, <em class="sig-param"><span class="pre">parsedGPR</span></em>, <em class="sig-param"><span class="pre">minSum</span></em>, <em class="sig-param"><span class="pre">gene_sig</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/dataIntegration/transcriptomics/preprocessing/selectGeneFromGPR.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.dataIntegration.transcriptomics.preprocessing.selectGeneFromGPR" title="Permalink to this definition">¶</a></dt>
<dd><p>Map gene expression to reaction expression using the GPR rules. An AND
will be replaced by MIN and an OR will be replaced by MAX.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>[expressionCol,  gene_used] = selectGeneFromGPR</strong> (<em>model, gene_names, gene_exp, parsedGPR, minSum</em>)</p></li>
<li><p><strong>[expressionCol,  gene_used, signifCol] = selectGeneFromGPR</strong> (<em>model, gene_names, gene_exp, parsedGPR, minSum, gene_sig</em>)</p></li>
</ul>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model</strong> – COBRA model struct</p></li>
<li><p><strong>gene_names</strong> – gene identifiers corresponding to gene_exp. Names must
be in the same format as model.genes (column vector)
(as returned by “findUsedGeneLevels.m”)</p></li>
<li><p><strong>gene_exp</strong> – gene FPKM/expression values, corresponding to names (column vector)
(as returned by “findUsedGeneLevels.m”)</p></li>
<li><p><strong>parsedGPR</strong> – GPR matrix as returned by “GPRparser.m”</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>minSum</strong> – instead of using min and max, use min for AND and Sum
for OR</p></li>
<li><p><strong>gene_sig</strong> – vector of significance values associated to each
‘gene_id’ (as returned by “findUsedGeneLevels.m”)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>expressionCol</strong> – reaction expression, corresponding to model.rxns.
No gene-expression data and orphan reactions will
be given a value of NaN.</p></li>
<li><p><strong>gene_used</strong> – gene identifier, corresponding to model.rxns, from GPRs
whose value (expression and/or significance) was chosen for that
reaction</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>OPTIONAL OUTPUTS:</dt><dd><dl class="simple">
<dt>signifCol:      reaction significance, corresponding to model.rxns.</dt><dd><p>No gene-expression data and orphan reactions will
be given a value of 0.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../INIT/index.html" class="btn btn-default float-right" title="Init" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../MOOMIN/index.html" class="btn btn-default" title="Moomin" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>