

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Gmcs &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Analysis" href="../index.html"/>
        <link rel="next" title="Subspaces" href="../subspaces/index.html"/>
        <link rel="prev" title="Room" href="../ROOM/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../QFCA/index.html">Qfca</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rMTA/index.html">Rmta</a></li>
<li class="toctree-l3"><a class="reference internal" href="../MOMA/index.html">Moma</a></li>
<li class="toctree-l3"><a class="reference internal" href="../percentileCalc/index.html">Percentilecalc</a></li>
<li class="toctree-l3"><a class="reference internal" href="../geometricFBA/index.html">Geometricfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../multiSpecies/index.html">Multispecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="../robustnessAnalysis/index.html">Robustnessanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../KLdistance/index.html">Kldistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ICONGEMs/index.html">Icongems</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rFBA/index.html">Rfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../nutritionAlgorithm/index.html">Nutritionalgorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../parsimoniousFBA/index.html">Parsimoniousfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FVA/index.html">Fva</a></li>
<li class="toctree-l3"><a class="reference internal" href="../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxEnrichmentAnalysis/index.html">Fluxenrichmentanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rumba/index.html">Rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../igemRNA/index.html">Igemrna</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fastForest/index.html">Fastforest</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FBA/index.html">Fba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../findMIIS/index.html">Findmiis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../persephone/index.html">Persephone</a></li>
<li class="toctree-l3"><a class="reference internal" href="../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../wholeBody/index.html">Wholebody</a></li>
<li class="toctree-l3"><a class="reference internal" href="../mdFBA/index.html">Mdfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../dynamicFBA/index.html">Dynamicfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../uFBA/index.html">Ufba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sparseFBA/index.html">Sparsefba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ROOM/index.html">Room</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Gmcs</a></li>
<li class="toctree-l3"><a class="reference internal" href="../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../relaxedFBA/index.html">Relaxedfba</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../dataIntegration/index.html">Dataintegration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../citations.html">Citations</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Analysis</a></li>
      <li class="breadcrumb-item active">Gmcs</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="gmcs">
<span id="id1"></span><h1>Gmcs<a class="headerlink" href="#gmcs" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.analysis.gMCS"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.GPR2models">
<span class="sig-name descname"><span class="pre">GPR2models</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">metabolic_model</span></em>, <em class="sig-param"><span class="pre">selected_rxns</span></em>, <em class="sig-param"><span class="pre">separate_transcript</span></em>, <em class="sig-param"><span class="pre">numWorkers</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/GPR2models.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.GPR2models" title="Permalink to this definition">¶</a></dt>
<dd><p>Each GPR rule is converted into a network where the reaction and genes
involved are interconnected.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[networks, rxnNumGenes] = GPR2models</strong> (<em>metabolic_model, selected_rxns, separate_transcript, numWorkers, printLevel</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>metabolic_model</strong> – Metabolic model structure (COBRA Toolbox format)</p></li>
<li><p><strong>selected_rxns</strong> – Index array which indicates selected reactions to
calculate the network.</p></li>
<li><p><strong>separate_transcript</strong> – Character used to discriminate
different isoforms of a gene. Default ‘’.</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>numWorkers</strong> – Maximum number of workers
* 0 - maximum provided by the system (automatic)
(default). If parallel pool is active, numWorkers
is defined by the system, otherwise is 1.
* 1 - sequential
* 2+ - parallel</p></li>
<li><p><strong>printLevel</strong> – show the reactions created in models.
* 0 - shows nothing
* 1 - shows progress by reactions (default)
* 2+ - shows everything (reaction and network generation)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUT<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>networks</strong> – Cell array which contains all the networks.</p></li>
<li><p><strong>rxnNumGenes</strong> – Array with the number of genes present in each GPR
rule for each reaction.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.buildGmatrix">
<span class="sig-name descname"><span class="pre">buildGmatrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_name</span></em>, <em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">separate_isoform</span></em>, <em class="sig-param"><span class="pre">numWorkers</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/buildGmatrix.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.buildGmatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Build the G matrix required for the calculation of genetic Minimal Cut
Sets (gMCSs).</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[G, G_ind, related, n_genes_KO, G_time] = buildGmatrix</strong> (<em>model_name, model_struct, separate_isoform, numWorkers, printLevel</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model_name</strong> – Name of the metabolic model under study.</p></li>
<li><p><strong>model_struct</strong> – Metabolic model structure (COBRA Toolbox format).</p></li>
<li><p><strong>separate_isoform</strong> – Character used to discriminate different isoforms of a gene.</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>numWorkers</strong> – Maximum number of workers
* 0 - maximum provided by the system (automatic)
* 1 - sequential
* 2+ - parallel</p></li>
<li><p><strong>printLevel</strong> – show the reactions created in models.
* 0 - shows nothing
* 1 - shows progress by reactions (default)
* 2+ - shows everything (reaction and network generation)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>G</strong> – G matrix.</p></li>
<li><p><strong>G_ind</strong> – Gene knockouts associated with each row in the G matrix.</p></li>
<li><p><strong>related</strong> – Relationships among rows of the G matrix.</p></li>
<li><p><strong>n_genes_KO</strong> – Number of genes whose knockout is added by each row of the G matrix.</p></li>
<li><p><strong>G_time</strong> – Calculation times for each of the steps.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Example</p>
<p>[G, G_ind, related, n_genes_KO, G_time] = buildGmatrix(‘Recon2.v04’, modelR204, ‘.’)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.calculateGeneMCS">
<span class="sig-name descname"><span class="pre">calculateGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_name</span></em>, <em class="sig-param"><span class="pre">model_struct</span></em>, <em class="sig-param"><span class="pre">n_gmcs</span></em>, <em class="sig-param"><span class="pre">max_len_gmcs</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/calculateGeneMCS.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.calculateGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate genetic Minimal Cut Sets (gMCSs) using the warm-start strategy
available in CPLEX, namely cplex.populate(), with or without selecting a
given knockout, among all the genes included in the model or a given
subset of them. Apaolaza et al., 2017 (Nature Communications).</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[gmcs, gmcs_time] = calculateGeneMCS</strong> (<em>model_name, model_struct, n_gmcs, max_len_gmcs, varargin</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model_name</strong> – Name of the metabolic model under study (in order to
identify the G matrix).</p></li>
<li><p><strong>model_struct</strong> – Metabolic model structure (COBRA Toolbox format).</p></li>
<li><p><strong>n_gmcs</strong> – Number of gMCSs to calculate.</p></li>
<li><p><strong>max_len_gmcs</strong> – Number of genes in the largest gMCS to be calculated.</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>KO</strong> – Selected gene knockout. (default = [])</p></li>
<li><p><strong>gene_set</strong> – Cell array containing the set of genes among which
the gMCSs are wanted to be calculated.
(default = [], all genes)</p></li>
<li><p><strong>target_b</strong> – Desired activity level of the metabolic task to be
disrupted. (default = 1e-3)</p></li>
<li><p><strong>nutrientGMCS</strong> – Boolean variable.  0 to calculate GeneMCS, 1 to
calculate MCS containing genes and nutrients,
known as ngMCS. (default = false)</p></li>
<li><p><strong>exchangeRxns</strong> – Cell array containing the set of reactions to be
included as inputs of nutrients from the cell
environment / culture medium. (default = [], which
are all reactions with only one 1 metabolite
consiedered as input for the model)</p></li>
<li><p><strong>onlyNutrients</strong> – Boolean variable.  1 to calculate MCS only using
selected KO and nutrients, 0 to use everything.
If there is no KO selected, it is set to false.
(default = false)</p></li>
<li><p><strong>separate_transcript</strong> – Character used to separate
different transcripts of a gene. (default = ‘’)
.. rubric:: Examples</p>
<ul>
<li><dl class="simple">
<dt>separate_transcript = ‘’</dt><dd><ul>
<li><p>gene 10005.1    ==&gt;    gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005.2</p></li>
<li><p>gene 10005.3    ==&gt;    gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>separate_transcript = ‘.’</dt><dd><ul>
<li><p>gene 10005.1</p></li>
<li><p>gene 10005.2    ==&gt;    gene 10005</p></li>
<li><p>gene 10005.3</p></li>
</ul>
</dd>
</dl>
</li>
</ul>
</li>
<li><p><strong>forceLength</strong> – 1 if the constraint limiting the length of the
gMCSs is to be active (recommended for
enumerating low order gMCSs), 0 otherwise
(default = 1)</p></li>
<li><p><strong>timelimit</strong> – Time limit for the calculation of gMCSs each time
the solver is called. (default = 1e75)</p></li>
<li><p><strong>numWorkers</strong> – Integer: is the maximun number of workers used
by Cplex and GPR2models. 0 = automatic,
1 = sequential, &gt; 1 = parallel. (default = 0)</p></li>
<li><p><strong>printLevel</strong> – Integer. 1 if the process is wanted to be shown
on the screen, 0 otherwise. (default = 1)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>gmcs</strong> – Cell array containing the calculated gMCSs.</p></li>
<li><p><strong>gmcs_time</strong> – Calculation times of the different processes in
the algorithm.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Example</p>
<p>%With optional values
[gmcs, gmcs_time] = calculateGeneMCS(‘Recon2.v04’, modelR204, 100, 10, …</p>
<blockquote>
<div><p>‘KO’ = ‘6240’, …
‘gene_set’ = {‘2987’; ‘6241’},  …
‘timelimit’ = 300,  …
‘target_b’ = 1e-4,
‘separate_transcript’ = ‘.’,  …
‘forceLength’ = 0, …
‘printLevel’ = 0)</p>
</div></blockquote>
<p>%Without optional values
[gmcs, gmcs_time] = calculateGeneMCS(‘ecoli_core_model’, model, 100, 10)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.calculateMCS">
<span class="sig-name descname"><span class="pre">calculateMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model_struct</span></em>, <em class="sig-param"><span class="pre">n_mcs</span></em>, <em class="sig-param"><span class="pre">max_len_mcs</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/calculateMCS.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.calculateMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate Minimal Cut Sets (MCSs) using the warm-start strategy available
in CPLEX, namely cplex.populate(), with or without selecting a given
knockout, among all the reactions included in the model or a given subset
of them. Tobalina et al., 2016 (Bioinformatics); von Kamp and Klamt, 2014
(PLoS Computational Biology).</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[mcs, mcs_time] = populateMCS</strong> (<em>model_struct, n_mcs, max_len_mcs, options</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model_struct</strong> – Metabolic model structure (COBRA Toolbox format).</p></li>
<li><p><strong>n_mcs</strong> – Number of MCSs to calculate.</p></li>
<li><p><strong>max_len_mcs</strong> – Number of reactions in the largest MCS to be calculated.</p></li>
</ul>
</dd>
</dl>
<p>OPTIONAL INPUT:
:OPTIONAL INPUTS: * <strong>KO</strong> – Selected reaction knockout. (default = [])</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>rxn_set</strong> – Cell array containing the set of reactions among which
the MCSs are wanted to be calculated.
taken from model.rxns
(default = [], all reactions)</p></li>
<li><p><strong>target_b</strong> – Desired activity level of the metabolic task to be
disrupted. (default = 1e-3)</p></li>
<li><p><strong>forceLength</strong> – 1 if the constraint limiting the length of the
MCSs is to be active (recommended for
enumerating low order MCSs), 0 otherwise
(default = 1)</p></li>
<li><p><strong>timelimit</strong> – Time limit for the calculation of MCSs each time
the solver is called. (default = 1e75)</p></li>
<li><p><strong>numWorkers</strong> – Integer: is the maximun number of workers used
by Cplex and GPR2models. 0 = automatic,
1 = sequential, &gt; 1 = parallel. (default = 0)</p></li>
<li><p><strong>printLevel</strong> – Integer. 1 if the process is wanted to be shown
on the screen, 0 otherwise. (default = 1)</p></li>
</ul>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mcs</strong> – Cell array containing the calculated MCSs.</p></li>
<li><p><strong>mcs_time</strong> – Calculation times of the different processes in
the algorithm.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Example</p>
<p>%With optional values
[mcs, mcs_time] = calculateMCS(modelR204, 100, 10, …</p>
<blockquote>
<div><p>‘KO’, ‘r1651’, …
‘rxn_set’, {‘r1652’; ‘r1653’; ‘r1654’}, …
‘timelimit’, 300, …
‘target_b’, 1e-4, …
‘forceLength’, 0, …
‘printLevel’, 0, …</p>
</div></blockquote>
<p>%Without optional values
[mcs, mcs_time] = calculateMCS(model, 100, 10)</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.checkGeneMCS">
<span class="sig-name descname"><span class="pre">checkGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">gMCSs</span></em>, <em class="sig-param"><span class="pre">isoform_separator</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/checkGeneMCS.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.checkGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Check if the gMCS is Minimal and Cut Set
gMCS = gene Minimal Cut Set
Minimal = the knock-off of all the genes of the gMCS is neccesary to
eliminate the objective function
Cut Set = The elimination of all the genes ensures the elimination of the
objective reaction</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[IsCutSet, IsMinimal, geneNotMinimal] = checkGeneMCS</strong> (<em>model, gMCSs, isoform_separator</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model</strong> – Metabolic model in COBRA format.</p></li>
<li><p><strong>gMCSs</strong> – cell array which stores all the gMCSs previously
calculated. Each position contains a gMCS.</p></li>
<li><p><strong>isoform_separator</strong> – genes in the model have different isoformes,
which are identificated by a separator from the
gene name. (Default = ‘’).</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUT<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>IsCutSet</strong> – array which indicates the biomass producion if
all genes of the gMCSs are deleted</p></li>
<li><p><strong>IsMinimal</strong> – array which indicates the maximum of biomass if
any gene is deleted</p></li>
<li><p><strong>geneNotMinimal</strong> – cell array which stores all the genes which
make the gMCS not minimal (this errors are
produced by the restriction of coputation
time).</p></li>
</ul>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The isoform separator is neccesary to consider some genes as the same
gene. Example: genes 555.1, 555.2 and 555.3 are considered as gene 555</p>
</div>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.gMCS.prepareModelNutrientGeneMCS">
<span class="sig-name descname"><span class="pre">prepareModelNutrientGeneMCS</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">exchangeRxns</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/gMCS/prepareModelNutrientGeneMCS.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.gMCS.prepareModelNutrientGeneMCS" title="Permalink to this definition">¶</a></dt>
<dd><p>Add artificial genes for the exchange reactions and prepare model for the
gMCS implementation which includes nutrients from the culture medium.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>model = prepareModelNutrientGeneMCS</strong> (<em>model, exchangeRxns</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>model</strong> – Metabolic model structure (COBRA Toolbox format).</p>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>exchangeRxns</strong> – Exchange reactions to be included (default = all
reactions which start by ‘<a href="#id2"><span class="problematic" id="id3">EX_</span></a>’, ‘<a href="#id4"><span class="problematic" id="id5">DM_</span></a>’ or ‘<a href="#id6"><span class="problematic" id="id7">sink_</span></a>’
and only have one metabolite involved.</p>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>model</strong> – Metabolic model structure with genes for selected
exchanges (COBRA Toolbox format).</p>
</dd>
</dl>
<p class="rubric">Example</p>
<p>model = prepareModelNutrientGeneMCS(model, exchangeRxns);</p>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../subspaces/index.html" class="btn btn-default float-right" title="Subspaces" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ROOM/index.html" class="btn btn-default" title="Room" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>