

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Citations &mdash; The COBRA Toolbox</title>
  

  
  
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="_static/img/favicon-16x16.png">
  
  
  
    <link rel="manifest" href="_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/css/cobra.css" type="text/css" />
  
    <link rel="stylesheet" href="_static/tabs.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="index.html"/>
        <link rel="next" title="Support" href="support.html"/>
        <link rel="prev" title="How to cite" href="cite.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="/cobratoolbox/stable/_static/img/logo.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},
	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='_static/../" + item.website_url + "' >" + value + "</a>" + "<a href='_static/../" + item.website_url + "' ><img src='_static/link.svg'/></a></span>";
		    }
	    },
		
		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="index.html">Home</a></li>
              </ul>
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Using the Toolbox</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/index.html">Functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="submodules/index.html">Submodules</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">FAQ</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="plan.html">Development Plan</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Project Information</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="cite.html">How to cite</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Citations</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#publications-that-cited-cobra-toolbox">Publications that cited COBRA Toolbox</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="funding.html">Funding</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="contents.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Citations</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="citations">
<h1>Citations<a class="headerlink" href="#citations" title="Permalink to this heading">¶</a></h1>
<br><section id="publications-that-cited-cobra-toolbox">
<h2>Publications that cited COBRA Toolbox<a class="headerlink" href="#publications-that-cited-cobra-toolbox" title="Permalink to this heading">¶</a></h2>
<div class="sphinx-tabs docutils container">
<div aria-label="Tabbed content" class="closeable" role="tablist"><button aria-controls="panel-0-0-0" aria-selected="true" class="sphinx-tabs-tab" id="tab-0-0-0" name="0-0" role="tab" tabindex="0">2025</button><button aria-controls="panel-0-0-1" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-1" name="0-1" role="tab" tabindex="-1">2024</button><button aria-controls="panel-0-0-2" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-2" name="0-2" role="tab" tabindex="-1">2023</button><button aria-controls="panel-0-0-3" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-3" name="0-3" role="tab" tabindex="-1">2022</button><button aria-controls="panel-0-0-4" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-4" name="0-4" role="tab" tabindex="-1">2021</button><button aria-controls="panel-0-0-5" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-5" name="0-5" role="tab" tabindex="-1">2020</button><button aria-controls="panel-0-0-6" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-6" name="0-6" role="tab" tabindex="-1">2019</button><button aria-controls="panel-0-0-7" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-7" name="0-7" role="tab" tabindex="-1">2018</button><button aria-controls="panel-0-0-8" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-8" name="0-8" role="tab" tabindex="-1">2017</button><button aria-controls="panel-0-0-9" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-9" name="0-9" role="tab" tabindex="-1">2016</button><button aria-controls="panel-0-0-10" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-10" name="0-10" role="tab" tabindex="-1">2015</button><button aria-controls="panel-0-0-11" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-11" name="0-11" role="tab" tabindex="-1">2014</button><button aria-controls="panel-0-0-12" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-12" name="0-12" role="tab" tabindex="-1">2013</button><button aria-controls="panel-0-0-13" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-13" name="0-13" role="tab" tabindex="-1">2012</button><button aria-controls="panel-0-0-14" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-14" name="0-14" role="tab" tabindex="-1">2011</button><button aria-controls="panel-0-0-15" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-15" name="0-15" role="tab" tabindex="-1">2010</button><button aria-controls="panel-0-0-16" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-16" name="0-16" role="tab" tabindex="-1">2009</button><button aria-controls="panel-0-0-17" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-17" name="0-17" role="tab" tabindex="-1">2008</button><button aria-controls="panel-0-0-18" aria-selected="false" class="sphinx-tabs-tab" id="tab-0-0-18" name="0-18" role="tab" tabindex="-1">2007</button></div><div aria-labelledby="tab-0-0-0" class="sphinx-tabs-panel" id="panel-0-0-0" name="0-0" role="tabpanel" tabindex="0"><div class="docutils container" id="id1">
<ol class="arabic simple" start="1">
<li id="id343"><p>Deepanwita Banerjee, Javier Menasalvas, Yan Chen, Jennifer Gin, Edward E. K. Baidoo, Christopher J. Petzold, Thomas Eng, and Aindrila Mukhopadhyay. Addressing genome scale design tradeoffs in pseudomonas putida for bioconversion of an aromatic carbon source. 11(1):8–8, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00480-z">[DOI]</a>.</p></li>
<li id="id346"><p>Nick Quinn-Bohmann, Alex V. Carr, Christian Diener, and Sean M. Gibbons. Moving from genome-scale to community-scale metabolic models for the human gut microbiome. 10(5):1055–1066, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-025-01972-2">[DOI]</a>.</p></li>
<li id="id450"><p>Sílvia Àvila-Cabré, Joan Albiol, and Pau Ferrer. Metabolic engineering of komagataella phaffii for enhanced 3-hydroxypropionic acid (3-hp) production from methanol. 19(1):19–19, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13036-025-00488-x">[DOI]</a>.</p></li>
<li id="id452"><p>Iqra Mariam, Ulrika Rova, Paul Christakopoulos, Λεωνίδας Μάτσακας, and Alok Patel. Data-driven synthetic microbes for sustainable future. 11(1):74–74, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-025-00556-4">[DOI]</a>.</p></li>
<li id="id551"><p>Zahra Negahban, Valerie C. A. Ward, Anne Richelle, Chris McCready, and Hector Budman. Hybrid dynamic flux balance modeling approach for bioprocesses: an e. coli case study. 48(5):841–856, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-025-03147-z">[DOI]</a>.</p></li>
<li id="id775"><p>Mariam Mohagheghi, Ali Navid, Thomas Mossington, Congwang Ye, Matthew A. Coleman, and Steven Hoang-Phou. Developing a media formulation to sustain ex vivo chloroplast function. 13():1560200–1560200, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2025.1560200">[DOI]</a>.</p></li>
<li id="id776"><p>Daniel Alejandro Caballero Cerbon and Dirk Weuster‐Botz. Exchange of the l-cysteine exporter after in-vivo metabolic control analysis improved the l-cysteine production process with engineered escherichia coli. 24(1):95–95, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-025-02715-y">[DOI]</a>.</p></li>
<li id="id777"><p>Xavier Benedicto, Åsmund Flobak, Miguel Ponce-de-León, and Alfonso Valencia. Using constraint-based metabolic modeling to elucidate drug-induced metabolic changes in a cancer cell line. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.05.08.652850">[DOI]</a>.</p></li>
<li id="id778"><p>Alexis L. Marsh, Myra B. Cohen, and Robert W. Cottingham. A tale from the trenches: applying metamorphic and differential testing to bioinformatics software. ():553–564, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/icst62969.2025.10988959">[DOI]</a>.</p></li>
<li id="id779"><p>Delphine Nègre, Larhlimi Abdelhalim, and Samuel Bertrand. Systematic sensitivity analyses of growth modelling to evaluate the robustness of genome scale metabolic network models — case study with the filamentous fungus &lt;i&gt;penicillium rubens&lt;/i&gt;. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.09.10.675325">[DOI]</a>.</p></li>
<li id="id780"><p>Érica Mangaravite, Christina Cleo Vinson, Eduardo Luís Menezes de Almeida, and Thomas Christopher Rhys Williams. Genome-scale metabolic models in plant stress physiology: implications for future climate resilience. 199(2):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/plphys/kiaf406">[DOI]</a>.</p></li>
<li id="id781"><p>Albert Fina, Pierre Millard, Cécilia Bergès, Pau Ferrer, Joan Albiol, and Stéphanie Heux. High-throughput stationary 13c-based metabolic flux analysis of pichia pastoris strains. 2697():153–170, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-4779-0_9">[DOI]</a>.</p></li>
<li id="id782"><p>Xavier Benedicto, Åsmund Flobak, Miguel Ponce-de-León, and Alfonso Valencia. Constraint based modeling of drug induced metabolic changes in a cancer cell line. 11(1):111–111, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-025-00586-y">[DOI]</a>.</p></li>
<li id="id783"><p>Isamar-Maydeth Vidal-Silva, Antonio Loza, and Rosa-María Gutiérrez-Ríos. Unlocking microbial potential: advances in omics and bioinformatics for aromatic hydrocarbon degradation. 41(10):384–384, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11274-025-04592-4">[DOI]</a>.</p></li>
<li id="id784"><p>Anna Procopio, Elvira Immacolata Parrotta, Stefania Scalise, Paolo Zaffino, R. Granata, Francesco Amato, Giovanni Cuda, and Carlo Cosentino. Hipscgem01: a genome-scale metabolic model for fibroblast-derived human ipscs. 12(10):1128–1128, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering12101128">[DOI]</a>.</p></li>
<li id="id785"><p>Edda Klipp. Simulation and modeling of metabolic networks. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-99507-8.00008-1">[DOI]</a>.</p></li>
<li id="id1287"><p>M. K. Weldon and Christian Euler. Physiology-informed use of cupriavidus necator in biomanufacturing: a review of advances and challenges. 24(1):30–30, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-025-02643-x">[DOI]</a>.</p></li>
<li id="id1470"><p>Artai R. Moimenta, Diego Troitiño-Jordedo, David Henriques, Alba Contreras-Ruíz, Romain Minebois, Miguel Morard, Eladio Barrio, Amparo Querol, and Eva Balsa‐Canto. An integrated multiphase dynamic genome-scale model explains batch fermentations led by species of the &lt;i&gt;saccharomyces&lt;/i&gt; genus. 10(2):e0161524–e0161524, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01615-24">[DOI]</a>.</p></li>
<li id="id1610"><p>Xianghua Wen, Xin Sun, and Xingyu Liu. Synergizing mechanistic and ai models for deeper insights into algal-bacterial systems in sustainable wastewater treatment. 76():108169–108169, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jwpe.2025.108169">[DOI]</a>.</p></li>
<li id="id1726"><p>Ying Zhang, Zhihao Liu, Jingmin Hu, Qing Yao, Hongfei Xu, Qing Zhang, Shouwen Chen, and Y.Y. Wang. Elucidating metabolic mechanisms underlying the influence of specific growth rate on alkaline protease synthesis in bacillus licheniformis through combined omics and computational modeling analysis. 434():132762–132762, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2025.132762">[DOI]</a>.</p></li>
<li id="id1727"><p>Esraa Gabal, Marwah Azaizeh, and Priyanka Baloni. Investigating lipid and energy dyshomeostasis induced by per- and polyfluoroalkyl substances (pfas) congeners in mouse model using systems biology approaches. 15(8):499–499, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo15080499">[DOI]</a>.</p></li>
<li id="id1999"><p>Mohammad Zim, Christian Euler, and Matthew P. Scott. Constraints on metabolic network analysis in bacterial physiology. 3(2):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1103/prxlife.3.022001">[DOI]</a>.</p></li>
<li id="id2000"><p>Ruijian Shao, Zhe Huang, Su Sun, Hongbo Yu, Shangxian Xie, and Fuying Ma. Constructing metabolic pathway of lignin monomers and their derivatives based on metabolic recombination models and yield models. 434():132786–132786, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2025.132786">[DOI]</a>.</p></li>
<li id="id2001"><p>Joseph Zavorskas, Penny Vlahos, Kristina Wagstrom, and Ranjan Srivastava. Dynamic flux balance analysis reveals climate‐driven shifts in arctic diatom succession and bloom dynamics. 31(7):e70339–e70339, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/gcb.70339">[DOI]</a>.</p></li>
<li id="id2002"><p>W.S. Verwoerd and Longfei Mao. The fba solution space kernel: introduction and illustrative examples. 26(1):182–182, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-025-06216-y">[DOI]</a>.</p></li>
<li id="id2003"><p>Manish Kumar, Ballamoole Krishna Kumar, Veena Shetty, R. Shyama Prasad Rao, and Pavan Gollapalli. Perspective on integrated multi-omics approaches and constraint-based modeling to explore metabolic functionality on the evolution of bacterial antibiotic resistance. 208():107999–107999, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.micpath.2025.107999">[DOI]</a>.</p></li>
<li id="id2004"><p>Alvaro R. Lara, Marie B. Andersen, Anna Madsen, Kathrine Gravlund Fønss, Marta Irla, Hilal Taymaz‐Nikerel, Luz María Martínez, Max Daniel Dicke, Marcel Mann, Jørgen Barsett Magnus, and Guillermo Gosset. Biomanufacturing potential of streamlined cells. 122(12):3309–3318, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.70070">[DOI]</a>.</p></li>
<li id="id2005"><p>P. M. Priyadarshan and Rodomiro Ortíz. Voyage of plant breeding to 2050. ():3–78, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-95-0583-8_1">[DOI]</a>.</p></li>
<li id="id2006"><p>Sumit Tiwari, Lumbini Devi Darapu, and Prateek Gupta. Systems biology approach for plant breeding. ():637–657, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-95-0583-8_22">[DOI]</a>.</p></li>
<li id="id2007"><p>Maria Masid, Ioanna A. Rota, David Barras, Flavia De Carlo, Pierpaolo Ginefra, Mathieu Desbuisson, Yaquelin Ortiz-Miranda, Dmitriy Zamarin, Sohrab P Shah, Nicola Vannini, George Coukos, Denarda Dangaj Laniti, Vassily Hatzimanikatis, and Matthieu Desbuisson. Computational inference of metabolic programs: a case study analyzing the effect of brca1 loss. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.11.18.689039">[DOI]</a>.</p></li>
<li id="id2008"><p>Sébastien Moretti, Anne Niknejad, Marco Pagni, and Florence Mehl. Metanetx: a bridge between metabolic resources for enhanced curation and multi-omics data harmonization. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkaf1286">[DOI]</a>.</p></li>
<li id="id2151"><p>Qiannan Peng, Cheng Zhao, Xiaopeng Wang, Kai Cheng, Congcong Wang, Xihui Xu, and Lu Lin. Modeling bacterial interactions uncovers the importance of outliers in the coastal lignin-degrading consortium. 16(1):639–639, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-025-56012-8">[DOI]</a>.</p></li>
<li id="id2316"><p>Almut Heinken, Timothy Otto Hulshof, Bram Nap, Filippo Martinelli Boneschi, Arianna Basile, Amy O’Brolchain, Neil Francis O’Sullivan, Celine Gallagher, E.L. Magee, Francesca McDonagh, Ian Lalor, Maeve Bergin, Phoebe Evans, Rachel Daly, Ronan Farrell, Robert Delaney, Saoirse Hill, Saoirse Roisin McAuliffe, Trevor Kilgannon, Ronan M. T. Fleming, Cyrille C. Thinnes, and Ines Thiele. A genome-scale metabolic reconstruction resource of 247,092 diverse human microbes spanning multiple continents, age groups, and body sites. 16(2):101196–101196, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2025.101196">[DOI]</a>.</p></li>
<li id="id2327"><p>Biki Bapi Kundu, Jayanth Krishnan, Richard Szubin, Arjun Patel, Bernhard Ø. Palsson, Daniel C. Zielinski, and Caroline M. Ajo‐Franklin. Extracellular respiration is a latent energy metabolism in escherichia coli. 188(11):2907–2924.e23, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2025.03.016">[DOI]</a>.</p></li>
<li id="id2335"><p>Nora Scherer, Daniel Fässler, Oleg Borisov, Yurong Cheng, Pascal Schlosser, Matthias Wuttke, Stefan Haug, Yong Li, Fabian Telkämper, Suraj Patil, Heike Meiselbach, Christina S.F. Wong, Urs Berger, Peggy Sekula, Anselm Hoppmann, Ulla T. Schultheiß, Sahar V. Mozaffari, Yannan Xi, Robert Graham, Miriam Schmidts, Michael Köttgen, Peter J. Oefner, Felix Knauf, Kai‐Uwe Eckardt, Sarah C. Grünert, Karol Estrada, Ines Thiele, Johannes Hertel, and Anna Köttgen. Coupling metabolomics and exome sequencing reveals graded effects of rare damaging heterozygous variants on gene function and human traits. 57(1):193–205, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41588-024-01965-7">[DOI]</a>.</p></li>
<li id="id2336"><p>Anthony L. Shiver, Jiawei Sun, Rebecca N. Culver, Arvie Violette, Coral Wynter, Marta Nieckarz, Samara Paula Mattiello, Prabhjot Kaur Sekhon, Francesca Bottacini, Lisa Friess, Hans K. Carlson, Daniel P.G.H. Wong, Steven K. Higginbottom, Meredith Weglarz, Weigao Wang, Benjamin D. Knapp, Emma R. Guiberson, Juan M. Sánchez, Po‐Hsun Huang, Paulo Alonso Gaona-García, Cullen Buie, Benjamin H. Good, Brian C. DeFelice, Felipe Cava, Joy Scaria, Justin L. Sonnenburg, Douwe van Sinderen, Adam M. Deutschbauer, and Kerwyn Casey Huang. Genome-scale resources in the infant gut symbiont bifidobacterium breve reveal genetic determinants of colonization and host-microbe interactions. 188(7):2003–2021.e19, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2025.02.010">[DOI]</a>.</p></li>
<li id="id2347"><p>Alise Žagare, Jānis Kurlovičs, Catarina Serra-Almeida, Daniele Ferrante, Daniela Frangenberg, Armelle Vitali, Gemma Gomez‐Giro, Christian Jäger, Paul Antony, Rashi Halder, Rejko Krueger, Enrico Glaab, Egils Stalidzāns, Giuseppe Arena, and Jens C. Schwamborn. Insulin resistance compromises midbrain organoid neuronal activity and metabolic efficiency predisposing to parkinson’s disease pathology. 16():20417314241295928–20417314241295928, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1177/20417314241295928">[DOI]</a>.</p></li>
<li id="id2348"><p>Xiaowei Li, Yanyan Wang, Xin Chen, Leon Eisentraut, Chunjun Zhan, Jens Nielsen, and Yun Chen. Modular deregulation of central carbon metabolism for efficient xylose utilization in saccharomyces cerevisiae. 16(1):4551–4551, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-025-59966-x">[DOI]</a>.</p></li>
<li id="id2351"><p>Lijuan Liao, Mengjun Xie, Xiaoshan Zheng, Zhou Zhao, Zixin Deng, and Jiangtao Gao. Molecular insights fast-tracked: ai in biosynthetic pathway research. 42(5):911–936, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d4np00003j">[DOI]</a>.</p></li>
<li id="id2489"><p>Sungwon Jung. Advances in functional analysis of the microbiome: integrating metabolic modeling, metabolite prediction, and pathway inference with next-generation sequencing data. 63(1):e:2411006–e:2411006, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.71150/jm.2411006">[DOI]</a>.</p></li>
<li id="id2495"><p>Steffen Klamt, Jürgen Zanghellini, and Axel von Kamp. Minimal cut sets in metabolic networks: from conceptual foundations to applications in metabolic engineering and biomedicine. 26(2):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbaf188">[DOI]</a>.</p></li>
<li id="id2496"><p>Jun Feng, Qingke Wang, Xiaolong Guo, Jialei Hu, Geng Wang, Li Lü, Zhen Qin, Hongxin Fu, Jufang Wang, and Shang‐Tian Yang. Metabolic engineering of &lt;i&gt;clostridium tyrobutyricum&lt;/i&gt; for high-yield &lt;i&gt;n&lt;/i&gt;-butanol production by increasing intracellular reducing equivalent with nadph-dependent 3-hydroxybutyryl-coa dehydrogenase. 14(6):2341–2353, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.5c00218">[DOI]</a>.</p></li>
<li id="id2497"><p>Lena Peters, Moritz Drechsler, Michael A. Herrera, Jing Liu, Barbara Pees, Johanna Jarstorff, Anna Czerwinski, Francesca Lubbock, Georgia Angelidou, Liesa Salzer, Karlis Arturs Moors, Nicole Paczia, Yi‐Ming Shi, Hinrich Schulenburg, Christoph Kaleta, Michael Witting, Manuel Liebeke, Dominic J. Campopiano, Helge B. Bode, and Katja Dierking. Polyketide synthase-derived sphingolipids mediate microbiota protection against a bacterial pathogen in c. elegans. 16(1):5151–5151, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-025-60234-1">[DOI]</a>.</p></li>
<li id="id2498"><p>Simran Kaur Aulakh, Oliver Lemke, Łukasz Szyrwiel, Stephan Kamrad, Yu Chen, Johannes Hartl, Michael Mülleder, Jens Nielsen, and Markus Ralser. The molecular landscape of cellular metal ion biology. 16(7):101319–101319, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2025.101319">[DOI]</a>.</p></li>
<li id="id2499"><p>Òscar Puiggené, Jaime Muñoz-Triviño, Laura Civil-Ferrer, Line Gille, Helena Schulz-Mirbach, Daniel Bergen, Tobias J. Erb, Birgitta E. Ebert, and Pablo I. Nikel. Systematic engineering of synthetic serine cycles in pseudomonas putida uncovers emergent topologies for methanol assimilation. 43(10):2539–2565, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2025.06.001">[DOI]</a>.</p></li>
<li id="id2500"><p>Shamsur Rehman, Muhammad Ibtisam Nasar, Cristina Sousa Mesquita, Souhaila Al Khodor, Richard A. Notebaart, Sascha Ott, Sunil Mundra, Ramesh P Arasardanam, Khalid Muhammad, and Mohammad Tauqeer Alam. Integrative systems biology approaches for analyzing microbiome dysbiosis and species interactions. 26(4):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbaf323">[DOI]</a>.</p></li>
<li id="id2642"><p>Diego Tec-Campos, Juan D. Tibocha‐Bonilla, Chunguo Jiang, Anurag Passi, Deepan Thiruppathy, Cristal Zúñiga, Camila Posadas, Alejandro Zepeda, and Karsten Zengler. A genome-scale metabolic model for the denitrifying bacterium thauera sp. mz1t accurately predicts degradation of pollutants and production of polymers. 21(1):e1012736–e1012736, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1012736">[DOI]</a>.</p></li>
<li id="id2643"><p>Tim Hensen and Ines Thiele. Metabolic modeling links gut microbiota to metabolic markers of parkinson’s disease. 17(1):2554195–2554195, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2025.2554195">[DOI]</a>.</p></li>
<li id="id2656"><p>Almut Heinken, John M. Asara, Gopalan Gnanaguru, and Charandeep Singh. Systemic regulation of retinal medium-chain fatty acid oxidation repletes tca cycle flux in oxygen-induced retinopathy. 8(1):25–25, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42003-024-07394-w">[DOI]</a>.</p></li>
<li id="id2657"><p>Venkat R. Pannala, Archana Hari, Mohamed AbdulHameed, Michele R. Balik-Meisner, Deepak Mav, Dhiral Phadke, Elizabeth H. Scholl, Ruchir Shah, Scott S. Auerbach, and Anders Wallqvist. Quantifying liver-toxic responses from dose-dependent chemical exposures using a rat genome-scale metabolic model. 204(2):154–168, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/toxsci/kfaf005">[DOI]</a>.</p></li>
<li id="id2658"><p>Òscar Puiggené, Jaime Muñoz-Triviño, Laura Civil-Ferrer, Line Gille, Helena Schulz-Mirbach, Daniel Bergen, Tobias J. Erb, Birgitta E. Ebert, and Pablo I. Nikel. Systematic engineering of synthetic serine cycles in &lt;i&gt;pseudomonas putida&lt;/i&gt; uncovers emergent topologies for methanol assimilation. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.02.17.638773">[DOI]</a>.</p></li>
<li id="id2659"><p>Chloe V. McCreery, Drew Alessi, Katarina Mollo, Alessio Fasano, and Ali R. Zomorrodi. Investigating intestinal epithelium metabolic dysfunction in celiac disease using personalized genome-scale models. 23(1):95–95, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12916-025-03854-0">[DOI]</a>.</p></li>
<li id="id2660"><p>Vikash Pandey. Mineapy: enhancing enrichment network analysis in metabolic networks. 41(3):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaf077">[DOI]</a>.</p></li>
<li id="id2661"><p>Chunjun Zhan, Guangxu Lan, Qingyun Dan, Ning Qin, Allie Pearson, Peter Mellinger, Yuzhong Liu, Zilong Wang, Seokjung Cheong, Chang Dou, Chenyi Li, Robert W. Haushalter, and Jay D. Keasling. Hybrid biological-chemical strategy for converting polyethylene into a recyclable plastic monomer using engineered corynebacterium glutamicum. 90():106–116, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2025.03.007">[DOI]</a>.</p></li>
<li id="id2662"><p>Daniel Fässler, Almut Heinken, and Johannes Hertel. Characterising functional redundancy in microbiome communities via relative entropy. 27():1482–1497, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2025.03.012">[DOI]</a>.</p></li>
<li id="id2663"><p>Zhenqiang Zhao, Rongshuai Zhu, Xuanping Shi, Fengyu Yang, Meijuan Xu, Minglong Shao, Rongzhen Zhang, Youxi Zhao, Jiajia You, and Zhiming Rao. Combining biosensor and metabolic network optimization strategies for enhanced l-threonine production in escherichia coli. 18(1):37–37, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-025-02640-7">[DOI]</a>.</p></li>
<li id="id2664"><p>Dilara Uzuner, Atılay İlgün, Fatma Betül Bozkurt, and Tunahan Çakır. A personalized metabolic modelling approach through integrated analysis of rna-seq-based genomic variants and gene expression levels in alzheimer’s disease. 8(1):502–502, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42003-025-07941-z">[DOI]</a>.</p></li>
<li id="id2665"><p>Devlin C. Moyer, Justin Reimertz, Daniel Segrè, and Juan I. Fuxman Bass. Macaw: a method for semi-automatic detection of errors in genome-scale metabolic models. 26(1):79–79, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13059-025-03533-6">[DOI]</a>.</p></li>
<li id="id2666"><p>Xiangyan Zhao, Hongwei Guo, Hai Huang, Ming Zheng, Xianping Zhang, Jing Li, Congcong Li, Bo Yuan, Chunmei Pan, and Zhoutong Sun. Contamination and biotransformation of deoxynivalenol (don) in common commercial foods: current status, challenges and future perspectives. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gresc.2025.04.008">[DOI]</a>.</p></li>
<li id="id2667"><p>Amal Saeed Alblooshi, Muhammad Ibtisam Nasar, Shamsur Rehman, and Mohammad Tauqeer Alam. Genomic and metabolic network properties in thermophiles and psychrophiles compared to mesophiles. 15(1):19757–19757, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-025-05030-z">[DOI]</a>.</p></li>
<li id="id2784"><p>Eduardo Luís Menezes de Almeida and Wendel Batista da Silveira. Insights into the response and tolerance mechanisms of papiliotrema laurentii to acetic acid stress by rna-seq and genome-scale metabolic modeling analysis. 215():109634–109634, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2025.109634">[DOI]</a>.</p></li>
<li id="id2785"><p>Luke N. Yaeger, David Sychantha, Princeton Luong, Shahrokh Shekarriz, Océane Goncalves, Annamaria Dobrin, Michael R. M. Ranieri, Ryan P. Lamers, Hanjeong Harvey, George C. diCenzo, Michael G. Surette, Jean‐Philippe Côté, Jakob Magolan, and Lori L. Burrows. Perturbation of &lt;i&gt;pseudomonas aeruginosa&lt;/i&gt; peptidoglycan recycling by anti-folates and design of a dual-action inhibitor. 16(3):e0298424–e0298424, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.02984-24">[DOI]</a>.</p></li>
<li id="id2786"><p>Sonal Omer, Subasree Sridhar, Devarajan Karunagaran, and G. K. Suraishkumar. Mechanistic insights into hypoxia‐induced metabolic reprogramming in colorectal cancer through genome‐scale modeling. 41(3):e70002–e70002, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.70002">[DOI]</a>.</p></li>
<li id="id2787"><p>M.E. Emetere. Progress, limitations, and advances of biohydrogen technologies: bringing the technology close to energy participants in developing countries. 16(8):912–928, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/17597269.2025.2467493">[DOI]</a>.</p></li>
<li id="id2789"><p>Maurício Alexander de Moura Ferreira, Eduardo Luís Menezes de Almeida, Wendel Batista da Silveira, and Zoran Nikoloski. Protein-constrained models pinpoints the role of underground metabolism in robustness of metabolic phenotypes. 28(3):112126–112126, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2025.112126">[DOI]</a>.</p></li>
<li id="id2790"><p>Jyoti Jyoti and Marc-Thorsten Hütt. Evaluating changes in attractor sets under small network perturbations to infer reliable microbial interaction networks from abundance patterns. 41(4):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaf095">[DOI]</a>.</p></li>
<li id="id2791"><p>Mustafa Sertbaş and Kutlu Ö. Ülgen. Exploring human brain metabolism via genome-scale metabolic modeling with highlights on multiple sclerosis. 16(7):1346–1360, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acschemneuro.5c00006">[DOI]</a>.</p></li>
<li id="id2792"><p>Pengyang Liu, Yingwei Ai, Muzi Li, Jiacheng Shi, Ning Xiao, Xiaoyu Zhang, Hongbo Yu, Fuying Ma, Su Sun, and Shangxian Xie. Discovery of mannose as an alternative non-nutrient-deficient regulator of lipid accumulation in microalgae. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jare.2025.03.019">[DOI]</a>.</p></li>
<li id="id2793"><p>Ziwei Yang and Takeyuki Tamura. Dbgdel: database-enhanced gene deletion framework for growth-coupled production in genome-scale metabolic models. 22(4):1415–1427, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/tcbbio.2025.3555999">[DOI]</a>.</p></li>
<li id="id2794"><p>Rami Balasubramanian, Debajit Saha, A. Arun, and P. K. Vinod. Hypometabolism in autism spectrum disorder: insights from brain and blood transcriptomics. 62(8):10765–10778, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12035-025-04941-2">[DOI]</a>.</p></li>
<li id="id2795"><p>Pablo Di Giusto, Dong‐Hyuk Choi, Athanasios Antonakoudis, Vikash Gokul Duraikannan, Pierrick Craveur, Nicholas Luke Cowie, Tejaswini Ganapathy, K Ramesh, Santiago Benavides-López, Camila A. Orellana, Natalia E. Jiménez, Leo Alexander Dworkin, James Morrissey, Igor Marín de Mas, Benjamin Strain, Norma A. Valdez‐Cruz, Mauricio A. Trujillo‐Roldán, Jannis Marzluf, Verónica S. Martínez, Leopold Zehetner, Claudia Altamirano, Ana María Vega-Letter, Bradley Priem, Haoyu Chris Cao, Martin Hold, Juchen Ma, Yi Fan Hong, Saratram Gopalakrishnan, Blaise Manga Enuh, Chaimaa Tarzi, Kuin Tian Pang, Claudio Angione, Jürgen Zanghellini, Cleo Kontoravdi, Hooman Hefzi, Michael J. Betenbaugh, Lars K. Nielsen, Meiyappan Lakshmanan, Dong‐Yup Lee, Anne Richelle, and Nathan E. Lewis. A community-consensus reconstruction of chinese hamster metabolism enables structural systems biology analyses to decipher metabolic rewiring in lactate-free cho cells. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.04.10.647063">[DOI]</a>.</p></li>
<li id="id2796"><p>Ian Sofian Yunus, David Carruthers, Yan Chen, Jennife W Gin, Edward E. K. Baidoo, Christopher J. Petzold, Héctor García Martín, Paul D. Adams, Aindrila Mukhopadhyay, and Taek Soon Lee. Predictive genome-wide crispr-mediated gene downregulation for enhanced bioproduction. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.04.25.650723">[DOI]</a>.</p></li>
<li id="id2797"><p>Min Chen, Junfeng Jiang, Tingting Xie, Yingping Zhuang, and Jianye Xia. Allosteric effectors outcompete transcript levels and substrate concentration in regulating central carbon flux during the crabtree effect transition. 20(4):e70024–e70024, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.70024">[DOI]</a>.</p></li>
<li id="id2798"><p>Merve Yarıcı, Furkan Cantürk, Serdar Dursun, Hatice Nur Aydın, and Muhammed Erkan Karabekmez. Rsea: a web server for pathway enrichment analysis of metabolic reaction sets. 122(8):2251–2258, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.29020">[DOI]</a>.</p></li>
<li id="id2799"><p>Srijith Sasikumar, S Pavan Kumar, Nirav Bhatt, and Himanshu Sinha. Genome-scale metabolic modelling identifies reactions mediated by snp-snp interactions associated with yeast sporulation. 11(1):50–50, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-025-00503-3">[DOI]</a>.</p></li>
<li id="id2800"><p>Drew Alessi, Chloe V. McCreery, and Ali R. Zomorrodi. In silico dietary interventions using whole-body metabolic models reveal sex-specific and differential dietary risk profiles for metabolic syndrome. 16():1586750–1586750, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2025.1586750">[DOI]</a>.</p></li>
<li id="id2801"><p>Luisella Spiga, Ryan T. Fansler, Alexandra Grote, Madison Langford-Butler, Asia K. Miller, M J Neal, Owen F. Hale, Yifan Wu, Deepanshu Singla, M. Wade Calcutt, Angelika Rose, Madeline M. Bresson, Alexandra C. Schrimpe‐Rutledge, Brittany Berdy, Simona G. Codreanu, M. Kay Washington, Benjamin P. Bratton, Stacy D. Sherrod, John A. McLean, Karsten Zengler, Cynthia L. Sears, Megan G. Behringer, Andreas Gnirke, Jonathan Livny, Danyvid Olivares‐Villagómez, Ashlee M. Earl, and Wenhan Zhu. An anaerobic pathogen rewires host metabolism to fuel oxidative growth in the inflamed gut. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.05.31.657111">[DOI]</a>.</p></li>
<li id="id2802"><p>Y. Zhang, Guannan Mao, Wei Hu, Zhineng Wu, Zhao Yang, Yanan Wang, Mark Bartlam, and Yingying Wang. Enhanced bde-47 biotransformation via cross-feeding in an aerobic-facultative anaerobic bacterial synthetic community. 282():122047–122047, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.envres.2025.122047">[DOI]</a>.</p></li>
<li id="id2803"><p>Xi Luo, Diana C. El Assal, Yanjun Liu, Samira Ranjbar, and Ronan M. T. Fleming. Constraint-based modeling of bioenergetic differences between synaptic and non-synaptic components of dopaminergic neurons in parkinson’s disease. 19():1594330–1594330, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fncom.2025.1594330">[DOI]</a>.</p></li>
<li id="id2804"><p>Gong‐Hua Li, Feifei Han, Efthymios Kalafatis, Qing‐Peng Kong, and Wenzhong Xiao. Systems modeling reveals shared metabolic dysregulation and potential treatments in me/cfs and long covid. 26(13):6082–6082, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms26136082">[DOI]</a>.</p></li>
<li id="id2805"><p>Boyu Jiang, Nick Quinn-Bohmann, Christian Diener, Vignesh Bose Nathan, Yu Han-Hallett, Lavanya Reddivari, Sean M. Gibbons, and Priyanka Baloni. Understanding disease-associated metabolic changes in human colonic epithelial cells using the icolonepithelium metabolic reconstruction. 21(7):e1013253–e1013253, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1013253">[DOI]</a>.</p></li>
<li id="id2806"><p>Anurag Passi, Diego Tec-Campos, Manish Kumar, Juan D. Tibocha‐Bonilla, Cristal Zúñiga, Beth B. Peacock, Amanda R. Hale, Rodrigo Santibáñez, James Borneman, and Karsten Zengler. Unveiling organ-specific metabolism of &lt;i&gt;citrus clementina&lt;/i&gt;. 122(29):e2503406122–e2503406122, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2503406122">[DOI]</a>.</p></li>
<li id="id2807"><p>Míriam Tarrado‐Castellarnau, Carles Foguet, Josep Tarragó‐Celada, Marc Palobart, Claudia Hernández-Carro, Jordi Perarnau, Erika Zodda, Ibrahim H. Polat, Silvia Marín, Alejandro Suárez‐Bonnet, Juan José Lozano, Mariia Yuneva, Timothy M. Thomson, and Marta Cascante. Glutaminase as a metabolic target of choice to counter acquired resistance to palbociclib by colorectal cancer cells. 44(36):3386–3406, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41388-025-03495-w">[DOI]</a>.</p></li>
<li id="id2808"><p>Massimo Bilancioni and Massimiliano Esposito. Energy transduction in complex networks with multiple resources: the chemistry paradigm. 163(4):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1063/5.0280649">[DOI]</a>.</p></li>
<li id="id2809"><p>Walter Vieri, Veronica Ghini, Paola Turano, Lara Massai, Luigi Messori, and Marco Fondi. Modeling the metabolic response of a2780 ovarian cancer cells to gold-based cytotoxic drugs. 11(1):83–83, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-025-00535-9">[DOI]</a>.</p></li>
<li id="id2810"><p>Wenbo Han, Luchi Xiao, Haocheng Sun, G. M. Xiang, Qiyue Jia, Haoyu Wang, Boyang Ji, Cheng Zhang, Eduard J. Kerkhoven, Jens Nielsen, and Hongzhong Lu. Alphagem enables precise genome-scale metabolic modelling by integrating protein structure alignment with deep-learning-based dark metabolism mining. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.07.21.665674">[DOI]</a>.</p></li>
<li id="id2811"><p>David Armitage, Alexandro Alonso-Sánchez, Samantha R Coy, Zhuli Cheng, Arno Hagenbeek, Karla P López-Martínez, Yong Heng Phua, and Alden R Sears. Adaptive pangenomic remodeling in the &lt;i&gt;azolla&lt;/i&gt; cyanobiont amid a transient microbiome. 19(1):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/ismejo/wraf154">[DOI]</a>.</p></li>
<li id="id2812"><p>Jintao Lu, Beining Wang, Xiqiang Liu, Jung‐Kul Lee, Vipin Chandra Kalia, and Chunjie Gong. Revolutionizing caffeic acid production: advanced microbial metabolic engineering and synthetic biology approaches. 20(8):e70091–e70091, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.70091">[DOI]</a>.</p></li>
<li id="id2813"><p>Devlin C. Moyer, Justin Reimertz, Juan I. Fuxman Bass, and Daniel Segrè. Flux sampling and context-specific genome-scale metabolic models for biotechnological applications. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2025.07.010">[DOI]</a>.</p></li>
<li id="id2814"><p>Almut Heinken, Hussein Awada, Vito Riccardo Tomaso Zanotelli, D. Sean Froese, Rosa‐Maria Guéant‐Rodriguez, and Jean‐Louis Guéant. Personalized genome‐scale modeling reveals metabolic perturbations in fibroblasts of methylmalonic aciduria patients. 48(5):e70077–e70077, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/jimd.70077">[DOI]</a>.</p></li>
<li id="id2815"><p>B.B. BAIMAKHANOVA, А.К. САДАНОВ, И.А. Ратникова, Gul Baimakhanova, Saltanat Orasymbet, Aigul Amitova, Gulzat Aitkaliyeva, and Ardak B. Kakimova. In silico modeling of metabolic pathways in probiotic microorganisms for functional food biotechnology. 11(8):458–458, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation11080458">[DOI]</a>.</p></li>
<li id="id2816"><p>Andreas Kremling. From flux analysis to self contained cellular models. 5():1546072–1546072, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2025.1546072">[DOI]</a>.</p></li>
<li id="id2817"><p>Jeffrey J. Czajka, Joonhoon Kim, Yinjie Tang, Kyle Pomraning, Aindrila Mukhopadhyay, and Héctor García Martín. Fluxretap: a reaction target prioritization genome-scale modeling technique for selecting genetic targets. 41(9):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaf471">[DOI]</a>.</p></li>
<li id="id2818"><p>Jonathan Josephs‐Spaulding, Hannah Clara Rettig, Johannes Zimmermann, Mariam Chkonia, Alexander Mischnik, Sören Franzenburg, Simon Graspeuntner, Jan Rupp, and Christoph Kaleta. Metatranscriptomics-based metabolic modeling of patient-specific urinary microbiome during infection. 11(1):183–183, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41522-025-00823-6">[DOI]</a>.</p></li>
<li id="id2819"><p>Aline Ovalle, Estefanía López, Jimena Sierralta, Nuria Paricio, and Daniel Garrido. A rationally designed microbial consortium modulates neurodegeneration in a drosophila melanogaster model of parkinson’s disease. 11(1):185–185, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41522-025-00797-5">[DOI]</a>.</p></li>
<li id="id2820"><p>Pınar Kocabaş. Modeling meets metabolic engineering: the ipichia consensus model as basis for metabolic studies in komagataella phaffii. 225():109940–109940, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2025.109940">[DOI]</a>.</p></li>
<li id="id2821"><p>Jinhui Zhao, Yanyan Zhou, Han Bao, Xinjie Zhao, Xinxin Wang, Chunxia Zhao, Wangshu Qin, Xin Lu, and Guowang Xu. Moda: a graph convolutional network-based multi-omics integration framework for unraveling hub molecules and disease mechanisms. 26(5):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbaf532">[DOI]</a>.</p></li>
<li id="id3041"><p>Andrea Angarita-Rodríguez, Nicolás Mendoza-Mejía, Janneth González, Jason A. Papin, Andrés Felipe Aristizábal-Pachón, and Andrés Pinzón. Improvement in the prediction power of an astrocyte genome-scale metabolic model using multi-omic data. 4():1500710–1500710, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2024.1500710">[DOI]</a>.</p></li>
<li id="id3042"><p>John Metzcar, Rachael Guenter, Yafei Wang, Kimberly Mantzke Baker, and Kate E Lines. Improving neuroendocrine tumor treatments with mathematical modeling: lessons from other endocrine cancers. 5(1):e240025–e240025, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1530/eo-24-0025">[DOI]</a>.</p></li>
<li id="id3043"><p>Shuvechha Chakraborty, Indumathi Palanikumar, Yash Gune, K. V. Venkatesh, Karthik Raman, and Susan Idicula‐Thomas. An integrated systems biology approach establishes arginine biosynthesis as a metabolic weakness in&lt;i&gt;candida albicans&lt;/i&gt;during host infection. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.01.11.632533">[DOI]</a>.</p></li>
<li id="id3044"><p>Rydberg Supo-Escalante, Germán Plata, Dinara R. Usmanova, and Dennis Vitkup. Percolation and lifestyle transition in microbial metabolism. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.01.12.632617">[DOI]</a>.</p></li>
<li id="id3045"><p>Joshua A. M. Kaste, Ran Ji, Patrick William Sydow, Ruairidh J. H. Sawers, and Megan L. Matthews. Metabolic modeling predicts synergistic growth benefits between arbuscular mycorrhizal fungi and theoretical n&lt;sub&gt;2&lt;/sub&gt;-fixing rhizobia symbiosis in maize. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.01.28.635303">[DOI]</a>.</p></li>
<li id="id3046"><p>Mohamed E. Hasan, Mohammed Osman, Alaa A. Hemeida, and Mahmoud ElHefnawi. Bioinformatics and artificial intelligence approaches in metabolic pathways. ():17–32, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-96-1305-2_2">[DOI]</a>.</p></li>
<li id="id3047"><p>S. Srinivasan, Karthik Raman, and Smita Srivastava. Investigating the impact of carbamazepine on tomato plant metabolism using genome-scale metabolic modelling. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.02.03.635865">[DOI]</a>.</p></li>
<li id="id3048"><p>Thomas A. Pugsley, Guy T. Hanke, and Christopher D. P. Duffy. Evaluating transcriptomic integration for cyanobacterial constraint-based metabolic modelling. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.02.10.637384">[DOI]</a>.</p></li>
<li id="id3049"><p>Cyrille C. Thinnes, Renee Waschkowitz, Eoghan Courtney, Eoghan Culligan, Katie Fahy, Ruby A M Ferrazza, C. Ferris, Angeline Lagali, Rebecca Lane, Colm Maye, Olivia Murphy, David Noone, Scott Ryan, Mihaela Bet, Maripat Corr, H.T. Cummins, David P. Hackett, Eugene Healy, Nina Kulczycka, NP Lang, L. V. Madden, Louise McHugh, Ivana Pyne, C.D. Varley, Niamh Harkin, Roger Meade, Grace O'Donnell, Bram Nap, Filippo Martinelli Boneschi, Almut Heinken, and Ines Thiele. The micromap is a network visualisation resource for microbiome metabolism. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.02.13.637616">[DOI]</a>.</p></li>
<li id="id3051"><p>Shripad Mukund Upasani. Computational genomics: finding out mystery of whole genome. ():125–149, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-443-31550-3.00006-8">[DOI]</a>.</p></li>
<li id="id3052"><p>Yang Cheng, Wenwen Yu, Xinyu Bi, Yanfeng Liu, J Li, Guocheng Du, Jian Chen, Xueqin Lv, and Long Liu. Carveadorncurate: a versatile cloud-based platform for constructing multiscale metabolic models. 43(5):1234–1259, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2025.01.011">[DOI]</a>.</p></li>
<li id="id3053"><p>Mustafa Sertbaş and Kutlu Ö. Ülgen. Genome-scale metabolic modeling of human pancreas with focus on type 2 diabetes. 29(4):125–138, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/omi.2024.0211">[DOI]</a>.</p></li>
<li id="id3054"><p>Riccardo Aucello, Simone Pernice, Dora Tortarolo, Raffaele Calogero, Celia Herrera-Rincon, Giulia Ronchi, Stefano Geuna, Francesca Cordero, Píetro Lió, and Marco Beccuti. Unifiedgreatmod: a new holistic modelling paradigm for studying biological systems on a complete and harmonious scale. 41(3):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaf103">[DOI]</a>.</p></li>
<li id="id3055"><p>David Armitage, Alexandro Alonso-Sánchez, Samantha R Coy, Zhongjun Cheng, Anton Hagenbeek, Karla P López-Martínez, Yong Heng Phua, and Alden R Sears. Adaptive pangenomic remodeling in the azolla cyanobiont amid a transient microbiome. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.03.24.644982">[DOI]</a>.</p></li>
<li id="id3056"><p>Bastien Nihant, Job A.J. Verdonschot, Sonia Balăn, Eva Thielecke, Joost Luiken, Miranda Nabben, Stéphane Heymans, Marian Breuer, and Michiel Adriaens. Metabolic task analysis reveals distinct metabotypes in end-stage dilated cardiomyopathy. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.03.28.645901">[DOI]</a>.</p></li>
<li id="id3057"><p>Elena Matveishina, Bartosz J. Bartmanski, Sara Benito-Vaquerizo, and Maria Zimmermann‐Kogadeeva. Gemsembler: cross-tool structural comparison and ensemble modeling improve metabolic model performance. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.04.09.647174">[DOI]</a>.</p></li>
<li id="id3058"><p>Carlos Coello-Castillo, Freddy Castillo-Alfonso, and Roberto Olivares‐Hernández. Computational metabolic engineering using genome-scale metabolic models and constraint-based methods. ():89–110, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-443-23998-4.00004-x">[DOI]</a>.</p></li>
<li id="id3059"><p>Yadira Boada, Marcelo Flores, Martin T. Stiebritz, Marco Córdova, Francisco Flores, and Alejandro Vignoni. Synthetic biology design principles enable efficient bioproduction of heparosan with low molecular weight and low polydispersion index for the biomedical industry. 10(1):ysaf006–ysaf006, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/synbio/ysaf006">[DOI]</a>.</p></li>
<li id="id3060"><p>Paul Matthay, Thomas Schalck, Kenneth Simoens, Dorien Kerstens, Bert F. Sels, Natalie Verstraeten, Kristel Bernaerts, and Jan Michiels. In silico identification of gene targets to enhance c12 fatty acid production in escherichia coli. 109(1):116–116, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-025-13501-6">[DOI]</a>.</p></li>
<li id="id3061"><p>Cyriel Huijer, Xiang Jiao, Yun Chen, and Rosemary Yu. Improved flux profiling in genome-scale modeling of human cell metabolism. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.05.13.653488">[DOI]</a>.</p></li>
<li id="id3063"><p>Sultana Mohammed Al Zubaidi, Muhammad Ibtisam Nasar, Richard A. Notebaart, Markus Ralser, and Mohammad Tauqeer Alam. An enzyme activation network reveals extensive regulatory crosstalk between metabolic pathways. 21(7):870–888, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s44320-025-00111-7">[DOI]</a>.</p></li>
<li id="id3064"><p>Pedro Fontanarrosa, Chania Clare, Alex J. H. Fedorec, and C. Barnes. Mimic: a python package for simulating, inferring, and predicting microbial community interactions and dynamics. 41(5):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaf174">[DOI]</a>.</p></li>
<li id="id3065"><p>Jia Bao Zhang, Wei Chen, Yongjun Yang, and Zhenzhen Liu. Discovery and green metabolic engineering of a self-sufficient genistein pathway in &lt;i&gt;paenibacillus jilinensis&lt;/i&gt;. 27(25):7575–7585, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d5gc01323b">[DOI]</a>.</p></li>
<li id="id3066"><p>Daniel J. Willard and Robert M. Kelly. Metabolic engineering in hot acid: strategies enabling chemolithotrophy in thermoacidophilic archaea. 91():366–378, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2025.06.005">[DOI]</a>.</p></li>
<li id="id3067"><p>W. A. Santos, Rafael David de Oliveira, José Gregorio Cabrera Gómez, and G. Roux. Analysis of poly-3-hydroxybutyrate production with different microorganisms using the dynamic simulations for evaluation of economic potential approach. 10(26):27756–27774, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acsomega.4c11178">[DOI]</a>.</p></li>
<li id="id3068"><p>María Teresita Castañeda, Sebastián Núñez, Martín Jamilis, and Hernán De Battista. Enhancement of lipid production in rhodosporidium toruloides: designing feeding strategies through dynamic flux balance analysis. 11(6):354–354, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation11060354">[DOI]</a>.</p></li>
<li id="id3069"><p>Emma M. Glass, Glynis L. Kolling, and Jason A. Papin. A systems biology approach to evaluate potential probiotic candidates for women’s vaginal health. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.06.16.659967">[DOI]</a>.</p></li>
<li id="id3070"><p>Xin Zhao, Haotong Li, Tao Zhang, Jiajing Qi, Yahui Cao, Ming Yang, and Weijie Liu. An efficient multi-objective shuffled frog leaping algorithm-based method for metabolic pathway design. ():1–4, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cec65147.2025.11043073">[DOI]</a>.</p></li>
<li id="id3071"><p>S Pavan Kumar and Nirav Bhatt. Thermoptcobra: thermodynamically optimal construction and analysis of metabolic networks for reliable phenotype predictions. 28(8):113005–113005, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2025.113005">[DOI]</a>.</p></li>
<li id="id3072"><p>Zhihao Liu, Minghao Chen, Jingmin Hu, Yonghong Wang, and Yu Chen. Minimization of proteome reallocation explains metabolic transition in hierarchical utilization of carbon sources. 10(7):e0069025–e0069025, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00690-25">[DOI]</a>.</p></li>
<li id="id3073"><p>Seo‐Young Park, Yi Qing Lee, Dong-Seok Kim, Dong Kyung Sung, Kyoung Jin Kim, Jae‐Seung Lee, Joo‐Youn Cho, Won Sik Lee, Se In Sung, and Dong‐Yup Lee. Deciphering dynamic antibiotics-microbiome-metabolome interactions in preterm infants using systems biology. 28(8):113038–113038, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2025.113038">[DOI]</a>.</p></li>
<li id="id3074"><p>V Vipina and Sanu Shameer. Modelling leaf metabolism using a charge- and proton-balanced diel flux balance analysis model. 7(2):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/insilicoplants/diaf008">[DOI]</a>.</p></li>
<li id="id3075"><p>Danni Guo, Yuanyuan Chen, Yahong Wu, Jingmin Cheng, Wei Lai, Wentao Ma, Hang Yang, Lianyi Han, Lan Ma, Huaijie Jia, and Xiao Liu. Multi-omics characterization of the skin microbiota reveals the anti-aging roles of &lt;i&gt;stenotrophomonas maltophilia&lt;/i&gt;. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.07.09.663908">[DOI]</a>.</p></li>
<li id="id3076"><p>Raúl J. Cano, Jaret C. Daniels, Martha Carlin, and Don M. Huber. Enhancing soil health and corn productivity with a co-fermented microbial inoculant (cfmi-8): a field-based evaluation. 13(7):1638–1638, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms13071638">[DOI]</a>.</p></li>
<li id="id3077"><p>Sarfraz Ahmad, Maksud Hasan Shah, Mohammad Ghani Raghib, Vikas Belwal, Shaukat Ali, and Rubina Khan. High-throughput metabolomics for plant phenotyping. ():236–251, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1079/9781800626911.0012">[DOI]</a>.</p></li>
<li id="id3078"><p>Juan Pablo Molina Ortiz, Dale D. McClure, Andrew Holmes, Stuart A. Rice, Mark Read, and Erin R. Shanahan. Genome-scale metabolic modelling of human gut microbes to inform rational community design. 17(1):2534673–2534673, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2025.2534673">[DOI]</a>.</p></li>
<li id="id3079"><p>Lin Zhang, Bin Wang, Ruiqi Zhang, Zhen He, Mingzhi Zhang, Tong Hao, and Jinsheng Sun. Screening of potential drug targets based on the genome-scale metabolic network model of vibrio parahaemolyticus. 47(7):575–575, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cimb47070575">[DOI]</a>.</p></li>
<li id="id3080"><p>Quanxiu Gao, Luhui Zhang, Wei‐Ran Huang, Yue Zhang, Richard Hewage, Mengkai Zhou, Yang Xu, Yang Han, Junxiong Yu, Zejian Wang, Zhengwei Xie, Jing Han, and Hua Xiang. Improving poly(3-hydroxybutyrate-co-3-hydroxyvalerate) synthesis with the construction of a genome-scale metabolic model for haloferax mediterranei. 321(Pt 2):146335–146335, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijbiomac.2025.146335">[DOI]</a>.</p></li>
<li id="id3081"><p>Sasha Yogiswara, Jonas Rombout, Giovanni Micharikopoulos, Sam De Craemer, Beatriz Herrera‐Malaver, Lotte van Landschoot, Sofie Mannaerts, Marcelo do Amaral, Karin Voordeckers, Stijn Spaepen, Jan Steensels, Quinten Deparis, Bart Ghesquière, and Kevin J. Verstrepen. Metabolic engineering of saccharomyces cerevisiae for co-production of ethanol and 3-methyl-1-butanol from sugarcane molasses. 18(1):86–86, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-025-02685-8">[DOI]</a>.</p></li>
<li id="id3082"><p>Rigoberto Rincón-Ballesteros, Francisco Javier Álvarez Padilla, and German Preciat. Characterization of metabolic phenotypes in breast cancer through the integration of genome-scale metabolic models and machine learning. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.07.29.667003">[DOI]</a>.</p></li>
<li id="id3083"><p>Jörg Schwender. Resolving cofactor imbalance in triacylglycerol biosynthesis in oilseeds through glycolytic shunts: a modeling study. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.07.31.667993">[DOI]</a>.</p></li>
<li id="id3084"><p>Kristin Klier, Ameneh Mehrjerd, Daniel Fässler, Maximilien Franck, Antoine Weihs, Kathrin Budde, Martin Bahls, Fabian Frost, Ann‐Kristin Henning, Almut Heinken, Henry Völzke, Marcus Dörr, Matthias Nauck, Hans J. Grabe, Nele Friedrich, and Johannes Hertel. Integrating population-based metabolomics with computational microbiome modelling identifies methanol as a urinary biomarker for protective diet–microbiome–host interactions. 16(18):7067–7081, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d5fo00761e">[DOI]</a>.</p></li>
<li id="id3085"><p>Álvaro Altamirano, Ignacio Tapia, Vicente Acuña, Daniel Garrido, and Pedro A. Saa. Metacone: a scalable framework for exploring the conversion cone of metabolic networks. 120(Pt 1):108607–108607, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiolchem.2025.108607">[DOI]</a>.</p></li>
<li id="id3086"><p>Matthias Mattanovich, Holda A. Anagho, Lidia Argemi‐Muntadas, Michael L. Nielsen, Qian Gao, and Thomas Möritz. Multi-omics analysis of thermogenic lipolysis in brown adipocytes. 28(9):113382–113382, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2025.113382">[DOI]</a>.</p></li>
<li id="id3087"><p>Pau Obregón-Gutierrez, Juan Nogales, Cintya González-Torres, Eva Huerta, Alejandro Rubio, Mariano Domingo, Joaquím Segalés, Karl Kochanowski, Antonio J. Pérez‐Pulido, Virginia Aragón, Florencia Correa‐Fiz, and Marina Sibila. Exploring the pangenome of mycoplasma hyorhinis in search of potential virulence markers. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-7195312/v1">[DOI]</a>.</p></li>
<li id="id3088"><p>Shouyong Jiang, Wenbo Shan, Xiongyan Yang, Chao Yang, and Jichun Li. Evolutionary multi-objective design of growth-coupled microbial cell factories. ():261–274, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-96-9582-9_19">[DOI]</a>.</p></li>
<li id="id3089"><p>Djordje Bajić, Marco van Oort, Minke Gabriëls, and Uroš Gojković. Structuring complexity by mapping the possible in microbial ecosystems. 88():102658–102658, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mib.2025.102658">[DOI]</a>.</p></li>
<li id="id3090"><p>P. B. Manjunatha, Rupa Bhowmick, and Sriram Chandrasekaran. Systems biology methods to study the epigenome and metabolic-epigenetic interactions. ():617–662, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-443-26759-8.00006-0">[DOI]</a>.</p></li>
<li id="id3091"><p>Benjamín R. Elizondo and Pedro A. Saa. Complex kinetic models predict β-carotene production and reveal flux limitations in recombinant &lt;i&gt;saccharomyces cerevisiae&lt;/i&gt; strains. 14(9):3457–3472, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.5c00256">[DOI]</a>.</p></li>
<li id="id3092"><p>Jhang Ho Pak, H.H. Park, K.-H. Baek, Seung-Ho Seo, and Suryang Kwak. Yeast-based precision fermentation for the biosynthesis of terpenoids. 12():100752–100752, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.fufo.2025.100752">[DOI]</a>.</p></li>
<li id="id3093"><p>Eva Balsa‐Canto, Núria Campo-Manzanares, Artai R. Moimenta, Gildas Roudaut, and Diego Troitin ̃o-Jordedo. Quantifying and managing uncertainty in systems biology: mechanistic and data-driven models. 42():100557–100557, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2025.100557">[DOI]</a>.</p></li>
<li id="id3094"><p>Elena Matveishina, Bartosz J. Bartmanski, Sara Benito-Vaquerizo, and Maria Zimmermann‐Kogadeeva. Gemsembler: consensus model assembly and structural comparison of genome-scale metabolic models across tools improve functional performance. 10(10):e0057425–e0057425, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00574-25">[DOI]</a>.</p></li>
<li id="id3095"><p>Francesca Casagli, Andrea Turolla, Damien J. Batstone, Gabriel Capson‐Tojo, Elena Ficara, Joan Garcı́a, Eva González-Flo, Julien Laurent, Tatjana Lorenz, Michaël Pierrelée, Benedek G. Plósz, Gustavo Henrique Ribeiro da Silva, Ángel Robles, Simone Rossi, Estel Rueda-Hernández, Lars Stegemüller, Jean‐Philippe Steyer, Olivier Bernard, and Borja Valverde‐Pérez. Modelling challenges to unlock the power of phototrophic systems for wastewater valorization. 85():108709–108709, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2025.108709">[DOI]</a>.</p></li>
<li id="id3096"><p>Huiyuan Guo, Wenli Zhang, Weihui Xu, Wenjing Chen, Yunlong Hu, and Zhigang Wang. Microbial consortium accelerates di(2-ethylhexyl) phthalate degradation through amino acid exchange-mediated inter-strain synergism. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jes.2025.09.016">[DOI]</a>.</p></li>
<li id="id3097"><p>Huijun Long, Jai W. Mehat, Huihai Wu, Arnoud H. M. van Vliet, and Roberto M. La Ragione. Use of genome-scale metabolic reconstructions of avian pathogenic escherichia coli (apec) phylogroups for the identification of lineage-specific metabolic pathways. 11(9):, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1099/mgen.0.001490">[DOI]</a>.</p></li>
<li id="id3098"><p>Anonymous. Computational biology for cell-free systems. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.pmbts.2025.08.007">[DOI]</a>.</p></li>
<li id="id3099"><p>Tunahan Çakır and Ecehan Abdi̇k. Personalized metabolite biomarker predictions reveal heterogeneous characteristics of parkinson’s disease. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.09.16.676636">[DOI]</a>.</p></li>
<li id="id3100"><p>Vítor Geniselli da Silva, Nick W. Smith, Jane Mullaney, Nicole C. Roy, Clare Wall, and Warren C. McNabb. Mathematical models of the colonic microbiota: an evaluation of accuracy using in vitro fecal fermentation data. 12():1623418–1623418, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fnut.2025.1623418">[DOI]</a>.</p></li>
<li id="id3101"><p>Kashif R. Siddique, Debajyoti Bose, Riya Bhattacharya, Raul V. Rodriguez, and Aritra Ray. Artificial intelligence driven bioinformatics for sustainable bioremediation: integrating computational intelligence with ecological restoration. 4(12):1987–1997, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d5va00240k">[DOI]</a>.</p></li>
<li id="id3102"><p>Qingqing Chen, Xin Qi, Sheng Dong, Xin Ge, Xiao Men, and Haibo Zhang. Recent advances and perspectives in biosynthesis of paclitaxel: key enzymes and intermediates. 331(Pt 1):148049–148049, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijbiomac.2025.148049">[DOI]</a>.</p></li>
<li id="id3103"><p>Juan M. Olaguez-Gonzalez, Isaac Chaírez, and Luz Bretón‐Deval. In-silico assessment of dynamic symbiotic microbial interactions in a reduced microbiota related to the autism spectrum disorder symptoms. 27():4078–4088, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2025.09.006">[DOI]</a>.</p></li>
<li id="id3104"><p>Zahrasadat Seyyedi, Hamed Haddad Kashani, Atefeh Parchebafi, Romina Ghayoumi, Mohammad Mehdi Haghighat Lari, and Elahe Seyed Hosseini. Integrating probiotics and microbiome-derived metabolites into cancer therapy: mechanistic insights, multi-omics strategies, and clinical potential. 10():100340–100340, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.crbiot.2025.100340">[DOI]</a>.</p></li>
<li id="id3105"><p>Martina Du, Jeremy M. Chacón, Heejoon Park, Campbell Putnam, Tomás̆ Gedeon, William R. Harcombe, and Ross P. Carlson. Cellular economics of exchanged metabolites alter ratios of microbial trading partners in a predictable manner. 14(11):4373–4387, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.5c00265">[DOI]</a>.</p></li>
<li id="id3106"><p>Qian Liu, Shuang Gao, Yanan Hou, Jianfeng Liu, Qianqian Yuan, Cong Huang, and Nanqi Ren. Ai- and biotechnology-driven digital design of biohydrogen-producing microbiota. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.eng.2025.09.027">[DOI]</a>.</p></li>
<li id="id3107"><p>Le Minh Thao Doan, Suraj Verma, Noushin Eftekhari, Claudio Angione, and Annalisa Occhipinti. From bulk to single-cell and spatial data: an ai framework to characterise breast cancer metabolic dysregulations across modalities. 198(Pt B):111195–111195, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2025.111195">[DOI]</a>.</p></li>
<li id="id3108"><p>Henry Kurniawan, Sarah Nickels, Alise Žagare, Elisa Zuccoli, Isabel Rosety, Gemma Gomez‐Giro, Enrico Glaab, and Jens C. Schwamborn. The parkinson’s disease-associated lrrk2-g2019s variant restricts serine metabolism, leading to microglial inflammation and dopaminergic neuron degeneration. 22(1):244–244, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12974-025-03577-2">[DOI]</a>.</p></li>
<li id="id3109"><p>Reetesh Kumar, Rupak Nagraik, Sorabh Lakhanpal, Mosleh Mohammad Abomughaid, Niraj Kumar Jha, and Rohan Gupta. Artificial intelligence in gut microbiome research: toward predictive diagnostics for neurodegenerative disorders. 72(4):296–312, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1556/030.2025.02725">[DOI]</a>.</p></li>
<li id="id3110"><p>Deo Shankar, Sarayu Murali, Shagun Shagun, Shyam Kumar Masakapalli, Karthik Raman, and Smita Srivastava. Genome-scale metabolic model guided metabolic flux analysis in the endophyte &lt;i&gt;alternaria burnsii&lt;/i&gt; ncim1409. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.10.28.685040">[DOI]</a>.</p></li>
<li id="id3111"><p>Mohammadreza Moghimi, F. Martinelli, Jonas Widder, Tim Hensen, Luciana Hannibal, Ute Spiekerkoetter, and Ines Thiele. Elucidating a potential role of the infant gut microbiome on the bioavailability of l-tyrosine in phenylketonuria. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.10.28.685037">[DOI]</a>.</p></li>
<li id="id3112"><p>Tim Hensen, Lora Khatib, Lucas Patel, Daniel McDonald, Antonio G. González, Siamak MahmoudianDehkordi, Colette Blach, Rob Knight, Rima Kaddurah‐Daouk, and Ines Thiele. Personalised whole-body modelling links gut microbiota to metabolic perturbations in alzheimer’s disease. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2025.10.28.685084">[DOI]</a>.</p></li>
<li id="id3113"><p>Nan Xu, Jiaqi Kang, Chengkun Zheng, Lin Zhou, Cong Gao, and Minliang Guo. Reconstruction and application of a genome-scale metabolic model for streptococcus suis. 26(1):997–997, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-025-12195-4">[DOI]</a>.</p></li>
<li id="id3114"><p>Ian Sofian Yunus, David N. Carruthers, Yan Chen, Jennifer Gin, Edward E. K. Baidoo, Christopher J. Petzold, Héctor García Martín, Paul D. Adams, Aindrila Mukhopadhyay, and Taek Soon Lee. Predictive crispr-mediated gene downregulation for enhanced production of sustainable aviation fuel precursor in pseudomonas putida. 94():67–76, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2025.11.007">[DOI]</a>.</p></li>
<li id="id3115"><p>Clémence Dupont Thibert, Sylvaine Roy, Sónia Carneiro, Bruno Pereira, Rafael Carreira, Paulo Vilaça, Séverine Collin, Eric Marechal, Elodie Billey, Gilles Curien, Maxime Durot, and Juliette Jouhet. A new genome-scale metabolic model of oleaginous microalgae with refined lipid metabolism clarifies microchloropsis gaditana mutant phenotypes. 441():133619–133619, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2025.133619">[DOI]</a>.</p></li>
<li id="id3116"><p>Andrea Dell’Olio, Joy Iheoma van Geerestein, Josep Rubert, and William T. Scott. Genome-level constraint-based analysis of megasphaera sp. mjr8396c reveals dual metabolic roles in the human gut. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-7827579/v1">[DOI]</a>.</p></li>
<li id="id3117"><p>Linh Khanh Nong, Chandran Sathesh‐Prabu, Sung Kuk Lee, and Dong-Hyuk Kim. Redefining hexr regulatory landscape in pseudomonas putida kt2440 through integrative systems biology. 94():77–89, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2025.11.014">[DOI]</a>.</p></li>
<li id="id3118"><p>Jiangpeng Yu, Kelin Cheng, Shenyang Qu, Jie Wang, Xun Wang, Cheng Zhao, and Wei Li. Genome‐scale metabolic modeling guided &lt;i&gt;escherichia coli&lt;/i&gt; engineering for &lt;i&gt;de novo&lt;/i&gt; biosynthesis of chrysanthemic acid. ():e12736–e12736, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/advs.202512736">[DOI]</a>.</p></li>
<li id="id3119"><p>C. Ferris, Rebecca Lane, Colm Maye, Olivia Murphy, David Noone, David P. Hackett, Eugene Healy, Louise McHugh, Ivana Pyne, and Roger Meade. Micromap: a network visualisation resource for human microbiome metabolism. 11(1):222–222, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41522-025-00853-0">[DOI]</a>.</p></li>
<li id="id3120"><p>German Preciat, Agnieszka B. Wegrzyn, Xi Luo, Ines Thiele, Thomas Hankemeier, and Ronan M. T. Fleming. Xomicstomodel: omics data integration and generation of thermodynamically consistent metabolic models. ():, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-025-01288-9">[DOI]</a>.</p></li>
<li id="id3121"><p>Lydia Rachbauer, Deepika Awashti, Sara Tejedor-Sanz, Justin Panich, Neha Bansal, Diep N. Pham, Emili Toppari, Eric Sundstrom, and Blake A. Simmons. Biomanufacturing from gaseous c1 feedstocks: a perspective on opportunities and challenges. ():100601–100601, 2025. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.nexus.2025.100601">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-1" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-1" name="0-1" role="tabpanel" tabindex="0"><div class="docutils container" id="id3122">
<ol class="arabic simple" start="1">
<li id="id3240"><p>Nayeon Kim, Junyeong Ma, Wonjong Kim, Jungyeon Kim, Peter Belenky, and Insuk Lee. Genome-resolved metagenomics: a game changer for microbiome medicine. 56(7):1501–1512, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s12276-024-01262-7">[DOI]</a>.</p></li>
<li id="id3258"><p>Chaimaa Tarzi, Guido Zampieri, Neil T. Sullivan, and Claudio Angione. Emerging methods for genome-scale metabolic modeling of microbial communities. 35(6):533–548, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tem.2024.02.018">[DOI]</a>.</p></li>
<li id="id3261"><p>Shengbo Wu, Zheping Qu, Danlei Chen, Hao Wu, Qinggele Caiyin, and Jianjun Qiao. Deciphering and designing microbial communities by genome-scale metabolic modelling. 23():1990–2000, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2024.04.055">[DOI]</a>.</p></li>
<li id="id3463"><p>Kathirvel Iyappan and Neela Gayathri Ganesan. Computational strategies to enhance cell-free protein synthesis efficiency. 4(3):2022–2042, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biomedinformatics4030110">[DOI]</a>.</p></li>
<li id="id3468"><p>Eduardo Luís Menezes de Almeida, Eduard J. Kerkhoven, and Wendel Batista da Silveira. Reconstruction of genome-scale metabolic models of non-conventional yeasts: current state, challenges, and perspectives. 29(1):35–67, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-024-00009-5">[DOI]</a>.</p></li>
<li id="id3469"><p>Carolina Deantas-Jahn, Sebastián N. Mendoza, Cuauhtémoc Licona‐Cassani, Camila A. Orellana, and Pedro A. Saa. Metabolic modeling of halomonas campaniensis improves polyhydroxybutyrate production under nitrogen limitation. 108(1):310–310, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-024-13111-8">[DOI]</a>.</p></li>
<li id="id3470"><p>Márcia R. Couto, Joana L. Rodrigues, Adelaide Braga, Óscar Dias, and L. R. Rodrigues. Optimization of chondroitin production in &lt;i&gt;e. coli&lt;/i&gt; using genome scale models. 9(6):597–611, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d3me00199g">[DOI]</a>.</p></li>
<li id="id3568"><p>Zixin Yang, Jinlin Zhu, Wenwei Lu, Fengwei Tian, Hao Zhang, and Wei Chen. Integrating artificial intelligence in exploring multiscale gut microbiota and diet relations for health promotion: a comprehensive review. 61():104541–104541, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.fbio.2024.104541">[DOI]</a>.</p></li>
<li id="id3569"><p>Wei-Ze Kong and Morihisa Fujita. Glycomaple: recent updates and applications in visualization and analysis of glycosylation pathways. 417(5):885–894, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00216-024-05594-1">[DOI]</a>.</p></li>
<li id="id3570"><p>Natchapon Srinak, Porntip Chiewchankaset, Saowalak Kalapanulak, Pornpan Panichnumsin, and Treenut Saithong. Metabolic cross-feeding interactions modulate the dynamic community structure in microbial fuel cell under variable organic loading wastewaters. 20(10):e1012533–e1012533, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1012533">[DOI]</a>.</p></li>
<li id="id3572"><p>Jingyi Qian and Chao Ye. Development and applications of genome-scale metabolic network models. 126():1–26, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.aambs.2024.02.002">[DOI]</a>.</p></li>
<li id="id3668"><p>Cristina Herencias, Virginia Rivero‐Buceta, Sergio Salgado, Natalia Hernández‐Herreros, Fernando Baquero, Rosa del Campo, Juan Nogales, and M. Auxiliadora Prieto. Bdellovibrio’s prey-independent lifestyle is fueled by amino acids as a carbon source. 108(1):422–422, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-024-13250-y">[DOI]</a>.</p></li>
<li id="id3669"><p>Marina Maurizio, María Masid, Kerry Woods, Reto Caldelari, John G. Doench, Arunasalam Naguleswaran, Denis Alain Henri Lucien Joly, Martín González Fernández, Jonas Zemp, Mélanie Borteele, Vassily Hatzimanikatis, Volker Heussler, Sven Rottenberg, and Philipp Olias. Host cell crispr genomics and modelling reveal shared metabolic vulnerabilities in the intracellular development of plasmodium falciparum and related hemoparasites. 15(1):6145–6145, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-50405-x">[DOI]</a>.</p></li>
<li id="id3670"><p>Sandhya Mishra, Anju Patel, Pankaj Bhatt, Shaohua Chen, and Pankaj Kumar Srivastava. Perspective evaluation of synthetic biology approaches for effective mitigation of heavy metal pollution. 262(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s44169-024-00072-2">[DOI]</a>.</p></li>
<li id="id3671"><p>Jesse Bucksot, Katherine Ritchie, Matthew Biancalana, John A. Cole, and Daniel Cook. Pan-cancer, genome-scale metabolic network analysis of over 10,000 patients elucidates relationship between metabolism and survival. 16(13):2302–2302, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cancers16132302">[DOI]</a>.</p></li>
<li id="id3741"><p>Anna Kattel, Valter Aro, Petri‐Jaan Lahtvee, Jekaterina Kazantseva, Arvi Jõers, Ranno Nahku, and Isma Belouah. Exploring the resilience and stability of a defined human gut microbiota consortium: an isothermal microcalorimetric study. 13(4):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/mbo3.1430">[DOI]</a>.</p></li>
<li id="id3742"><p>Sílvia Àvila-Cabré, Joan Albiol, and Pau Ferrer. Leveraging lactate transporters for superior 3-hydroxypropionic (3-hp) acid production from methanol in komagataella phaffii. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-5386323/v1">[DOI]</a>.</p></li>
<li id="id3814"><p>Bruno Veronez de Lima, Ricardo Luna, Daniel Martins Lima, Julio E. Normey‐Rico, and José Ricardo Pérez‐Correa. Advancing wine fermentation: extended kalman filter for early fault detection. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4419796/v1">[DOI]</a>.</p></li>
<li id="id3890"><p>Isaac Klapper, Daniel B. Szyld, Xinli Yu, Karsten Zengler, Tianyu Zhang, and Cristal Zúñiga. A domain decomposition method for solution of a pde-constrained generalized nash equilibrium model of biofilm community metabolism. 84(1):97–113, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1137/22m1511023">[DOI]</a>.</p></li>
<li id="id3891"><p>Gholamreza Abdi, Nil Patil, Mukul Jain, and Mukul Machhindra Barwant. Integration of metabolomics and flux balance analysis: applications and challenges. ():199–237, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-99-8401-5_10">[DOI]</a>.</p></li>
<li id="id3892"><p>Achintya Saha, Megha Jethwa, and Aditi Gangopadhyay. Application of computation in the study of biosynthesis of phytochemicals. ():321–355, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-443-16102-5.00003-1">[DOI]</a>.</p></li>
<li id="id3893"><p>Deepanwita Banerjee, Javier Menasalvas, Yan Chen, Jennifer Gin, Edward E. K. Baidoo, Christopher J. Petzold, Thomas Eng, and Aindrila Mukhopadhyay. Bottlenecks in the implementation of genome scale metabolic model based designs for bioproduction from aromatic carbon sources. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.03.15.585139">[DOI]</a>.</p></li>
<li id="id3894"><p>Maya Raman. Bioinformatics and computational tools. ():107–134, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-97-2562-5_5">[DOI]</a>.</p></li>
<li id="id3895"><p>Jan Zrimec, Sandra M. Correa, Maja Zagorščak, Marko Petek, Carissa Bleker, Katja Stare, Christian Schuy, Sophia Sonnewald, Kristina Gruden, and Zoran Nikoloski. Evaluating plant growth-defence trade-offs by modelling the interaction between primary and secondary metabolism. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.09.15.613124">[DOI]</a>.</p></li>
<li id="id4119"><p>Sonia Alcalá, Lara Villarino, Laura Ruíz-Cañas, José R. Couceiro, Miguel Martínez‐Calvo, Adrián Palencia‐Campos, Diego Navarro, Pablo Cabezas‐Sainz, Iker Rodríguez-Arabaolaza, Alfonso Cordero‐Barreal, Lucía Trilla‐Fuertes, Juan A. Rubiolo, Sandra Batres-Ramos, Mireia Vallespinós, Cristina González-Paramos, Jéssica Rodríguez, Angelo Gámez‐Pozo, Juan Ángel Fresno Vara, Sara Fra‐Fernández, Amparo Benito Berlinches, Nicolás Moreno‐Mata, Ana María Torres Redondo, Alfredo Carrato, Patrick Hermann, Laura Sánchez, Susana Torrente, Miguel Ángel Fernández‐Moreno, José L. Mascareñas, and Bruno Sáinz. Targeting cancer stem cell oxphos with tailored ruthenium complexes as a new anti-cancer strategy. 43(1):33–33, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13046-023-02931-7">[DOI]</a>.</p></li>
<li id="id4120"><p>Pritam Kundu, Satyajit Beura, Suman Mondal, Amit Kumar Das, and Amit Ghosh. Machine learning for the advancement of genome-scale metabolic modeling. 74():108400–108400, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2024.108400">[DOI]</a>.</p></li>
<li id="id4407"><p>Alexander A. Aksenov, Alex Blacutt, Nichole Ginnan, Philippe E. Rolshausen, Alexey V. Melnik, Ali Lotfi, Emily C. Gentry, Manikandan Ramasamy, Cristal Zúñiga, Karsten Zengler, Kranthi K. Mandadi, Greg McCollum, Pieter C. Dorrestein, and M. Caroline Roper. Spatial chemistry of citrus reveals molecules bactericidal to candidatus liberibacter asiaticus. 14(1):20306–20306, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-024-70499-z">[DOI]</a>.</p></li>
<li id="id4583"><p>Aparajitha Srinivasan, Kevin Chen-Xiao, Deepanwita Banerjee, Asun Oka, Venkataramana R. Pidatala, Aymerick Eudes, Blake A. Simmons, Thomas Eng, and Aindrila Mukhopadhyay. Sustainable production of 2,3,5,6-tetramethylpyrazine at high titer in engineered &lt;i&gt;corynebacterium glutamicum&lt;/i&gt;. 51():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/jimb/kuae026">[DOI]</a>.</p></li>
<li id="id4587"><p>Lorenzo Pasotti, Ilaria Massaiu, Paolo Magni, and Cinzia Calvio. Metabolic engineering of bacillus subtilis for the production of poly-γ-glutamic acid from glycerol feedstock. 10(6):319–319, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation10060319">[DOI]</a>.</p></li>
<li id="id4590"><p>Pablo Rodríguez-Mier, Martín Garrido‐Rodríguez, Attila Gábor, and Julio Sáez-Rodríguez. Unified knowledge-driven network inference from omics data. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.10.26.620390">[DOI]</a>.</p></li>
<li id="id4592"><p>Anita L. Ziegler, Lena Ullmann, Manuel Boßmann, Karla Stein, Ulf W. Liebal, Alexander Mitsos, and Lars M. Blank. Itaconic acid production by co‐feeding of &lt;i&gt;ustilago maydis&lt;/i&gt; : a combined approach of experimental data, design of experiments, and metabolic modeling. 121(6):1846–1858, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28693">[DOI]</a>.</p></li>
<li id="id4712"><p>Bruno G. Galuzzi, Luca Milazzo, and Chiara Damiani. Adjusting for false discoveries in constraint-based differential metabolic flux analysis. 150():104597–104597, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbi.2024.104597">[DOI]</a>.</p></li>
<li id="id4723"><p>Carolina Shene, Allison Leyton, Liset Flores, Daniela Chavez, Juan A. Asenjo, and Yusuf Chisti. Genome‐scale metabolic modeling of &lt;i&gt;thraustochytrium&lt;/i&gt; sp. rt2316‐16: effects of nutrients on metabolism. 121(6):1986–2001, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28689">[DOI]</a>.</p></li>
<li id="id4724"><p>Karthik Raman, Miroslav Kratochvíl, Brett G. Olivier, Matthias König, Pratyay Sengupta, Dinesh Kumar Kuppa Baskaran, Tung V N Nguyen, Daniel Lobo, St. Elmo Wilken, Krishna Kumar Tiwari, Aswathy K. Raghu, Indumathi Palanikumar, Lavanya Raajaraam, Maziya Ibrahim, Sanjaay Balakrishnan, Shreyansh Umale, Frank Bergmann, Tanisha Malpani, Venkata Satagopam, Reinhard Schneider, Moritz Emanuel Beber, Sarah Keating, Mihail Anton, Alina Renz, Meiyappan Lakshmanan, Dong‐Yup Lee, Lokanand Koduru, Reihaneh Mostolizadeh, Óscar Dias, Emanuel Cunha, Alexandre Oliveira, Yi Qing Lee, Karsten Zengler, Rodrigo Santibáñez-Palominos, Manish Kumar, Matteo Barberis, Bhanwar Lal Puniya, Tomáš Helikar, Hoang V. Dinh, Patrick F. Suthers, Costas D. Maranas, Isabella Casini, Seyed Babak Loghmani, Nadine Veith, Nantia Leonidou, Feiran Li, Yu Chen, Jens Nielsen, GaRyoung Lee, Sang Mi Lee, Gi Bae Kim, Pedro T. Monteiro, Miguel C. Teixeira, Hyun Uk Kim, Sang Yup Lee, Ulf W. Liebal, Lars M. Blank, Christian Lieven, Chaimaa Tarzi, Claudio Angione, Manga Enuh Blaise, Celik Pinar Aytar, Mikhail A. Kulyashov, llya Akberdin, Dohyeon Kim, Sung Ho Yoon, Zhaohui Xu, Jyotshana Gautam, William T. Scott, Peter J. Schaap, Jasper J. Koehorst, Cristal Zúñiga, Gabriela Canto-Encalada, Sara Benito-Vaquerizo, Ivette Parera Olm, María Suárez‐Diez, Qianqian Yuan, Hongwu Ma, Mohammad Mazharul Islam, Jason A. Papin, Francisco Zorrilla, Kiran Raosaheb Patil, Arianna Basile, Juan Nogales, Granado San Leon, Freddy Castillo-Alfonso, Roberto Olivares‐Hernández, Gabriel Vigueras-Ramírez, Henning Hermjakob, Andreas Dräger, and Rahuman S. Malik‐Sheriff. Frog analysis ensures the reproducibility of genome scale metabolic models. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.09.24.614797">[DOI]</a>.</p></li>
<li id="id4728"><p>Josh A. Taylor, Alain Rapaport, and Denis Dochain. Convex representation of metabolic networks with michaelis–menten kinetics. 86(6):65–65, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11538-024-01293-1">[DOI]</a>.</p></li>
<li id="id4729"><p>Jasmeet Kaur Khanijou, Yan Ting Hee, Clement P M Scipion, Xixian Chen, and Kumar Selvarajoo. Systems biology approach for enhancing limonene yield by re-engineering escherichia coli. 10(1):109–109, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00440-7">[DOI]</a>.</p></li>
<li id="id4730"><p>Thomaz Satuye Prieto-de Lima, Keilor Rojas-Jiménez, and Christopher Vaglio. Strategy for optimizing vitamin b12 production in pseudomonas putida kt2440 using metabolic modeling. 14(11):636–636, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo14110636">[DOI]</a>.</p></li>
<li id="id4733"><p>Yogesh K. Ahlawat, V.J. Srivastava, Maryam Samani, Sarahani Harun, Vinothienii Rajuloo, and Darshna Chaudhary. Computational strategies for maximizing biomass and metabolite yields for bioproduction. ():143–161, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-031-60117-0_11">[DOI]</a>.</p></li>
<li id="id4734"><p>Peng Su, Konstantine Tchourine, and Dennis Vitkup. Ecological scaling of temporal fluctuations with bacterial abundance in gut microbiota depends on functional properties of individual microbial species and bacterial communities. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.28.625948">[DOI]</a>.</p></li>
<li id="id4805"><p>Yang Xia, Huiying Wang, Dongqin Ding, Huan Fang, Huina Dong, and Dawei Zhang. A hybrid rna-protein biosensor for high-throughput screening of adenosylcobalamin biosynthesis. 9(3):513–521, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.synbio.2024.04.008">[DOI]</a>.</p></li>
<li id="id4832"><p>Chenguang Liang, Sue Murray, Yang Li, Richard Lee, Audrey Low, Shruti Sasaki, Austin W.T. Chiang, Wen-Jen Lin, Joel Mathews, Will Barnes, and Nathan E. Lewis. Lipidsim: inferring mechanistic lipid biosynthesis perturbations from lipidomics with a flexible, low-parameter, markov modeling framework. 82():110–122, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2024.01.004">[DOI]</a>.</p></li>
<li id="id4840"><p>Mitzi de la Cruz, Flavio Kunert, Hilal Taymaz‐Nikerel, Juan‐Carlos Sigala, Guillermo Gosset, Jochen Büchs, and Alvaro R. Lara. Increasing the pentose phosphate pathway flux to improve plasmid dna production in engineered e. coli. 12(1):150–150, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms12010150">[DOI]</a>.</p></li>
<li id="id4841"><p>Alvaro R. Lara, Flavio Kunert, Vincent Vandenbroucke, Hilal Taymaz‐Nikerel, Luz María Martínez, Juan‐Carlos Sigala, Frank Delvigne, Guillermo Gosset, and Jochen Büchs. Transport‐controlled growth decoupling for self‐induced protein expression with a glycerol‐repressible genetic circuit. 121(6):1789–1802, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28697">[DOI]</a>.</p></li>
<li id="id4842"><p>Alexander A. Aksenov, Alex Blacutt, Nichole Ginnan, Philippe E. Rolshausen, Alexey V. Melnik, Ali Lotfi, Emily C. Gentry, Manikandan Ramasamy, Cristal Zúñiga, Karsten Zengler, Kranthi K. Mandadi, Pieter C. Dorrestein, and M. Caroline Roper. Spatial chemistry of citrus reveals molecules bactericidal to candidatus liberibacter asiaticus. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4224869/v1">[DOI]</a>.</p></li>
<li id="id4843"><p>Zainab S. Attarbashi, Mohit Nigam, and Lalit Kumar Singh. Metabolic engineering of lipid biosynthesis pathway to enhance the oil content in microalgae. ():37–63, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-97-1451-3_3">[DOI]</a>.</p></li>
<li id="id4844"><p>Priyanka Narad, Romasha Gupta, Sudeepti Kulshrestha, and Abhishek Sengupta. Systems biology in understanding the human gut microbiome and related diseases highlighting metabolic modeling and analysis. ():437–465, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-99-9462-5_17">[DOI]</a>.</p></li>
<li id="id4845"><p>Harishankar Mahto and Dipa Mahato. In silico characterization of natural products in plants. ():161–185, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-97-2166-5_7">[DOI]</a>.</p></li>
<li id="id4846"><p>Carlos Alberto García-Mogollón, Diego F. Mendoza, and Juan Carlos Quintero. Electrostatic ethanol fermentation: experimental study and kinetic-based metabolic modeling. 10(17):e36587–e36587, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.heliyon.2024.e36587">[DOI]</a>.</p></li>
<li id="id4901"><p>Rayna Nolen, Lene H. Petersen, Karl Kaiser, Antonietta Quigg, and David Hala. In silico biomarker analysis of the adverse effects of perfluorooctane sulfonate (pfos) exposure on the metabolic physiology of embryo-larval zebrafish. 4():1367562–1367562, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2024.1367562">[DOI]</a>.</p></li>
<li id="id4933"><p>Abhirami Thumsi, Diego Martínez, Srivatsan J. Swaminathan, Arezoo Esrafili, Abhirami P. Suresh, Madhan Mohan Chandrasekhar Jaggarapu, Kelly Lintecum, Michelle Halim, Shivani Mantri, Yasmine Sleiman, Nicole Appel, Haiwei Gu, Marion Curtis, Cristal Zúñiga, and Abhinav P. Acharya. Inverse‐vaccines for rheumatoid arthritis re‐establish metabolic and immunological homeostasis in joint tissues. 14(5):e2303995–e2303995, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/adhm.202303995">[DOI]</a>.</p></li>
<li id="id4934"><p>Artai R. Moimenta, Diego Troitiño-Jordedo, David Henriques, Alba Contreras-Ruíz, Romain Minebois, Miguel Morard, Eladio Barrio, Amparo Querol, and Eva Balsa‐Canto. A continuous dynamic genome-scale model explains batch fermentations led by species of the&lt;i&gt;saccharomyces&lt;/i&gt;genus. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.05.03.592398">[DOI]</a>.</p></li>
<li id="id5112"><p>Jean‐Marc Schwartz and Zita Soons. Metabolic models. ():510–518, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-95502-7.00025-7">[DOI]</a>.</p></li>
<li id="id5113"><p>Alexander A. Aksenov, Alex Blacutt, Nichole Ginnan, Philippe E. Rolshausen, Alexey V. Melnik, Ali Lotfi, Emily C. Gentry, Manikandan Ramasamy, Cristal Zúñiga, Karsten Zengler, Kranthi K. Mandadi, Pieter C. Dorrestein, and M. Caroline Roper. Spatial chemistry of citrus reveals molecules bactericidal to&lt;i&gt;candidatus&lt;/i&gt;liberibacter asiaticus. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.04.12.589303">[DOI]</a>.</p></li>
<li id="id5114"><p>Kumar Selvarajoo, Jasmeet Kaur Khanijou, Yan Ting Hee, Clement P M Scipion, and Xixian Chen. Systems biology approach for enhancing limonene yield by re-engineering escherichia coli. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4280267/v1">[DOI]</a>.</p></li>
<li id="id5115"><p>W.S. Verwoerd and Longfei Mao. The fba solution space kernel – introduction and illustrative examples. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.06.10.598372">[DOI]</a>.</p></li>
<li id="id5116"><p>Jacob J. Valenzuela, Selva Rupa Christinal Immanuel, James G. Wilson, Serdar Turkarslan, Maryann Ruiz, Sean M. Gibbons, Kristopher A. Hunt, Nejc Stopnišek, Manfred Auer, Marcin Zemla, David A. Stahl, and Nitin S. Baliga. Origin of biogeographically distinct ecotypes during laboratory evolution. 15(1):7451–7451, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-51759-y">[DOI]</a>.</p></li>
<li id="id5117"><p>M. Maton, Fernando N. Santos-Navarro, Jesús Picó, Philippe Bogaerts, and Alain Vande Wouwer. Towards constraint-based burden-aware models for metabolic engineering. 58(14):247–252, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2024.08.344">[DOI]</a>.</p></li>
<li id="id5118"><p>Tyler Hilsabeck and Shane L. Rea. Deep learning of cellular metabolic flux distributions predicts lifespan. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.22.623650">[DOI]</a>.</p></li>
<li id="id5119"><p>Lidan Bai, Jun Sun, Vasile Palade, Chao Li, Hengyang Lu, and Cong Gao. Optimizing metabolite production with neighborhood-based binary quantum-behaved particle swarm optimization and flux balance analysis. 32(1):64–88, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2024.0538">[DOI]</a>.</p></li>
<li id="id5173"><p>Zhepu Ruan, Kai Chen, Weimiao Cao, Lei Meng, Bingang Yang, Mengjun Xu, Youwen Xing, Pengfa Li, Shiri Freilich, Chen Chen, Yanzheng Gao, Jiandong Jiang, and Xihui Xu. Engineering natural microbiomes toward enhanced bioremediation by microbiome modeling. 15(1):4694–4694, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-49098-z">[DOI]</a>.</p></li>
<li id="id5197"><p>Egeria Scoditti, Silvia Sabatini, Fabrizia Carli, and Amalia Gastaldelli. Hepatic glucose metabolism in the steatotic liver. 21(5):319–334, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41575-023-00888-8">[DOI]</a>.</p></li>
<li id="id5205"><p>Yu Chen, Johan Gustafsson, Albert Tafur Rangel, Mihail Anton, Iván Domenzain, Cheewin Kittikunapong, Feiran Li, Le Yuan, Jens Nielsen, and Eduard J. Kerkhoven. Reconstruction, simulation and analysis of enzyme-constrained metabolic models using gecko toolbox 3.0. 19(3):629–667, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-023-00931-7">[DOI]</a>.</p></li>
<li id="id5231"><p>Zachary M. Burcham, Aeriel D. Belk, Bridget B. McGivern, Amina Bouslimani, Parsa Ghadermazi, Cameron Martino, Liat Shenhav, Anru R. Zhang, Pixu Shi, Alexandra L. Emmons, Heather L. Deel, Zhenjiang Zech Xu, Victoria Nieciecki, Qiyun Zhu, Michael Shaffer, Morgan Panitchpakdi, Kelly C. Weldon, Kalen Cantrell, Asa Ben‐Hur, Sasha C. Reed, Greg C. Humphry, Gail Ackermann, Daniel McDonald, Siu Hung Joshua Chan, Melissa Connor, Derek A. Boyd, Jake Smith, Jenna M. S. Watson, Giovanna M. Vidoli, Dawnie Wolfe Steadman, Aaron M. Lynne, Sibyl R. Bucheli, Pieter C. Dorrestein, Kelly Wrighton, David Carter, Rob Knight, and Jessica L. Metcalf. A conserved interdomain microbial network underpins cadaver decomposition despite environmental variables. 9(3):595–613, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-023-01580-y">[DOI]</a>.</p></li>
<li id="id5255"><p>Zhijin Gong, Jiayao Chen, Xinyu Jiao, Hao Gong, Danzi Pan, Lingli Liu, Yang Zhang, and Tianwei Tan. Genome-scale metabolic network models for industrial microorganisms metabolic engineering: current advances and future prospects. 72():108319–108319, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2024.108319">[DOI]</a>.</p></li>
<li id="id5256"><p>Ning Qin, Lingyun Li, Xiaozhen Wan, Xu Ji, Yu Chen, Chaokun Li, Ping Liu, Yijie Zhang, Weijie Yang, Junfeng Jiang, Jianye Xia, Shuobo Shi, Tianwei Tan, Jens Nielsen, Yun Chen, and Zihe Liu. Increased co2 fixation enables high carbon-yield production of 3-hydroxypropionic acid in yeast. 15(1):1591–1591, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-45557-9">[DOI]</a>.</p></li>
<li id="id5258"><p>Deepanwita Banerjee, Ian Sofian Yunus, Xi Wang, Jinho Kim, Aparajitha Srinivasan, Russel Menchavez, Yan Chen, Jennifer Gin, Christopher J. Petzold, Héctor García Martín, Jon Magnuson, Paul D. Adams, Blake A. Simmons, Aindrila Mukhopadhyay, Joonhoon Kim, and Taek Soon Lee. Genome-scale and pathway engineering for the sustainable aviation fuel precursor isoprenol production in pseudomonas putida. 82():157–170, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2024.02.004">[DOI]</a>.</p></li>
<li id="id5260"><p>Clémence Joseph, Haris Zafeiropoulos, Kristel Bernaerts, and Karoline Faust. Predicting microbial interactions with approaches based on flux balance analysis: an evaluation. 25(1):36–36, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-024-05651-7">[DOI]</a>.</p></li>
<li id="id5261"><p>Hongzhong Lu, Luchi Xiao, Wenbin Liao, Xuefeng Yan, and Jens Nielsen. Cell factory design with advanced metabolic modelling empowered by artificial intelligence. 85():61–72, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2024.07.003">[DOI]</a>.</p></li>
<li id="id5262"><p>Pavel Dvořák, Barbora Burýšková, Barbora Popelářová, Birgitta E. Ebert, Tibor Botka, Dalimil Bujdoš, Alberto Sánchez-Pascuala, Hannah Schöttler, Heiko Hayen, Vı́ctor de Lorenzo, Lars M. Blank, and Martin Benešík. Synthetically-primed adaptation of pseudomonas putida to a non-native substrate d-xylose. 15(1):2666–2666, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-46812-9">[DOI]</a>.</p></li>
<li id="id5267"><p>Guangjian Li, Hui Liang, Ruichen Gao, Ling Qin, Pei Xu, Mingtao Huang, Min‐Hua Zong, Yufei Cao, and Wen‐Yong Lou. Yeast metabolism adaptation for efficient terpenoids synthesis via isopentenol utilization. 15(1):9844–9844, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-54298-8">[DOI]</a>.</p></li>
<li id="id5268"><p>Xi Luo, Yanjun Liu, Alexander Balck, Christine Klein, and Ronan M. T. Fleming. Identification of metabolites reproducibly associated with parkinson’s disease via meta-analysis and computational modelling. 10(1):126–126, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41531-024-00732-z">[DOI]</a>.</p></li>
<li id="id5269"><p>Merve Atasoy, William T. Scott, Alberte Regueira, Miguel Mauricio‐Iglesias, Peter J. Schaap, and Hauke Smidt. Biobased short chain fatty acid production - exploring microbial community dynamics and metabolic networks through kinetic and microbial modeling approaches. 73():108363–108363, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2024.108363">[DOI]</a>.</p></li>
<li id="id5270"><p>Elaine Zaunseder, Ulrike Mütze, Jürgen G. Okun, Georg F. Hoffmann, Stefan Kölker, Vincent Heuveline, and Ines Thiele. Personalized metabolic whole-body models for newborns and infants predict growth and biomarkers of inherited metabolic diseases. 36(8):1882–1897.e7, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cmet.2024.05.006">[DOI]</a>.</p></li>
<li id="id5271"><p>Cong Chen, Haiquan Yang, Kaizheng Zhang, Guangbin Ye, Huibo Luo, and Wei Zou. Revealing microbiota characteristics and predicting flavor-producing sub-communities in nongxiangxing baijiu pit mud through metagenomic analysis and metabolic modeling. 188():114507–114507, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.foodres.2024.114507">[DOI]</a>.</p></li>
<li id="id5273"><p>Silvia Sabatini, Partho Sen, Fabrizia Carli, Samantha Pezzica, Chiara Rosso, Erminia Lembo, Ornella Verrastro, Ann K. Daly, Olivier Govaere, Simon Cockell, Tuulia Hyötyläinen, Geltrude Mingrone, Elisabetta Bugianesi, Quentin M. Anstee, Matej Orešič, and Amalia Gastaldelli. Hepatic glucose production rises with the histological severity of metabolic dysfunction-associated steatohepatitis. 5(11):101820–101820, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xcrm.2024.101820">[DOI]</a>.</p></li>
<li id="id5408"><p>Klara Cerk, Pablo Ugalde‐Salas, Chabname Ghassemi Nedjad, Maxime Lecomte, Coralie Muller, David James Sherman, Falk Hildebrand, Simon Labarthe, and Clémence Frioux. Community‐scale models of microbiomes: articulating metabolic modelling and metagenome sequencing. 17(1):e14396–e14396, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.14396">[DOI]</a>.</p></li>
<li id="id5412"><p>Mahnoor Zulfiqar, Vinay Singh, Christoph Steinbeck, and Maria Sorokina. Review on computer-assisted biosynthetic capacities elucidation to assess metabolic interactions and communication within microbial communities. 50(6):1053–1092, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/1040841x.2024.2306465">[DOI]</a>.</p></li>
<li id="id5413"><p>Maria‐Tsampika Manoli, A. Gargantilla Becerra, Carlos del Cerro, Virginia Rivero‐Buceta, M. Auxiliadora Prieto, and Juan Nogales. A model-driven approach to upcycling recalcitrant feedstocks in pseudomonas putida by decoupling pha production from nutrient limitation. 43(4):113979–113979, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2024.113979">[DOI]</a>.</p></li>
<li id="id5421"><p>Filippo Martinelli Boneschi, Almut Heinken, Ann‐Kristin Henning, Maria A. Ulmer, Tim Hensen, Antonio González, Matthias Arnold, Sanjay Asthana, Kathrin Budde, Corinne D. Engelman, Mehrbod Estaki, Hans J. Grabe, Margo B. Heston, Sterling C. Johnson, Gabi Kastenmüller, Cameron Martino, Daniel McDonald, Federico E. Rey, Ingo Kilimann, Olive Peters, Xiao Wang, Eike Jakob Spruth, Anja Schneider, Klaus Fließbach, Jens Wiltfang, Niels Hansen, Wenzel Glanz, Katharina Büerger, Daniel Janowitz, Christoph Laske, Matthias H. Munk, Annika Spottke, Nina Roy, Matthias Nauck, Stefan Teipel, Rob Knight, Rima Kaddurah‐Daouk, Barbara B. Bendlin, Johannes Hertel, and Ines Thiele. Whole-body metabolic modelling reveals microbiome and genomic interactions on reduced urine formate levels in alzheimer’s disease. 14(1):6095–6095, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-024-55960-3">[DOI]</a>.</p></li>
<li id="id5422"><p>Cecilia Noecker and Peter J. Turnbaugh. Emerging tools and best practices for studying gut microbial community metabolism. 6(7):1225–1236, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42255-024-01074-z">[DOI]</a>.</p></li>
<li id="id5429"><p>Manish Kumar, Juan D. Tibocha‐Bonilla, Zoltán Füssy, Chloe Lieng, Sarah M. Schwenck, Alice V. Lévesque, Mahmoud M. Al‐Bassam, Anurag Passi, M J Neal, Cristal Zúñiga, Farrah Kaiyom, Josh L. Espinoza, Hyun Gyu Lim, Shawn W. Polson, Lisa Zeigler Allen, and Karsten Zengler. Mixotrophic growth of a ubiquitous marine diatom. 10(29):eado2623–eado2623, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/sciadv.ado2623">[DOI]</a>.</p></li>
<li id="id5430"><p>Xihui Xu, Can Li, Weimiao Cao, Lulu Yan, Lulu Cao, Qi Han, Minling Gao, Yahua Chen, Zhenguo Shen, Jiandong Jiang, and Chen Chen. Bacterial growth and environmental adaptation via thiamine biosynthesis and thiamine-mediated metabolic interactions. 18(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/ismejo/wrae157">[DOI]</a>.</p></li>
<li id="id5435"><p>Jinlin Zhu, Jialin Yin, Jing Chen, Mingyi Hu, Wenwei Lu, Hongchao Wang, Hao Zhang, and Wei Chen. Integrative analysis with microbial modelling and machine learning uncovers potential alleviators for ulcerative colitis. 16(1):2336877–2336877, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2024.2336877">[DOI]</a>.</p></li>
<li id="id5436"><p>Arturo Vera-Ponce de León, Tim Hensen, Matthias Hoetzinger, Shashank Gupta, Bronson R. Weston, Sander M. Johnsen, Jacob Agerbo Rasmussen, Cecilie Grønlund Clausen, Louisa Pless, Ana Raquel Verissimo, Knut Rudi, Lars Snipen, Christian Karlsen, Morten T. Limborg, Stefan Bertilsson, Ines Thiele, Torgeir R. Hvidsten, Simen R. Sandve, Phillip B. Pope, and Sabina Leanti La Rosa. Genomic and functional characterization of the atlantic salmon gut microbiome in relation to nutrition and health. 9(11):3059–3074, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-024-01830-7">[DOI]</a>.</p></li>
<li id="id5442"><p>Seo‐Young Park, Jinsung Song, Dong‐Hyuk Choi, Uiseon Park, Hyeran Cho, Bee Hak Hong, Yaron Silberberg, and Dong‐Yup Lee. Exploring metabolic effects of dipeptide feed media on cho cell cultures by in silico model-guided flux analysis. 108(1):123–123, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-023-12997-0">[DOI]</a>.</p></li>
<li id="id5443"><p>Meng Luo, Jinlin Zhu, Jiajia Jia, Hao Zhang, and Jianxin Zhao. Progress on network modeling and analysis of gut microecology: a review. 90(3):e0009224–e0009224, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.00092-24">[DOI]</a>.</p></li>
<li id="id5444"><p>Wheaton L. Schroeder, Patrick F. Suthers, T. Willis, E. Mooney, and Costas D. Maranas. Current state, challenges, and opportunities in genome-scale resource allocation models: a mathematical perspective. 14(7):365–365, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo14070365">[DOI]</a>.</p></li>
<li id="id5445"><p>Sara Bertorello, Francesco Cei, Dorian Fink, Elena Niccolai, and Amedeo Amedei. The future exploring of gut microbiome-immunity interactions: from in vivo/vitro models to in silico innovations. 12(9):1828–1828, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms12091828">[DOI]</a>.</p></li>
<li id="id5446"><p>Shengbo Wu, Yongsheng Zhou, Lei Dai, Aidong Yang, and Jianjun Qiao. Assembly of functional microbial ecosystems: from molecular circuits to communities. 48(6):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsre/fuae026">[DOI]</a>.</p></li>
<li id="id5447"><p>Boatema Ofori-Anyinam, Meagan Hamblin, Miranda L Coldren, Barry Li, Gautam Mereddy, Majid B. Shaikh, Avi Shah, Courtney Grady, Navpreet Ranu, Shou‐En Lu, Paul C. Blainey, Shuyi Ma, James J. Collins, and Jason H. Yang. Catalase activity deficiency sensitizes multidrug-resistant mycobacterium tuberculosis to the atp synthase inhibitor bedaquiline. 15(1):9792–9792, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-53933-8">[DOI]</a>.</p></li>
<li id="id5452"><p>Shivang Hina-Nilesh Joshi, Christopher A. Jenkins, David Ulaeto, and Thomas E. Gorochowski. Accelerating genetic sensor development, scale-up, and deployment using synthetic biology. 6():0037–0037, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.34133/bdr.0037">[DOI]</a>.</p></li>
<li id="id5453"><p>Emanuel Lange, Lena Kranert, Jacob Krüger, Dirk Benndorf, and Robert Heyer. Microbiome modeling: a beginner's guide. 15():1368377–1368377, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2024.1368377">[DOI]</a>.</p></li>
<li id="id5454"><p>John R. Casey, Brian J. Bennion, Patrik D’haeseleer, Jeffrey A. Kimbrel, Gianna L. Marschmann, and Ali Navid. Transporter annotations are holding up progress in metabolic modeling. 4():1394084–1394084, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2024.1394084">[DOI]</a>.</p></li>
<li id="id5455"><p>Ross P. Carlson, Ashley E. Beck, Mauricio Garcia Benitez, William R. Harcombe, Radhakrishnan Mahadevan, and Tomás̆ Gedeon. Cell geometry and membrane protein crowding constrain growth rate, overflow metabolism, respiration, and maintenance energy. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.08.21.609071">[DOI]</a>.</p></li>
<li id="id5460"><p>Aswathy K. Raghu, Indumathi Palanikumar, and Karthik Raman. Designing function-specific minimal microbiomes from large microbial communities. 10(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00373-1">[DOI]</a>.</p></li>
<li id="id5461"><p>Magdalena Wutkowska, Vojtěch Tláskal, Sergio Bordel, Lisa Y. Stein, Justus Amuche Nweze, and Anne Daebeler. Leveraging genome-scale metabolic models to understand aerobic methanotrophs. 18(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/ismejo/wrae102">[DOI]</a>.</p></li>
<li id="id5462"><p>Lvjing Wang, Xiaoyu Wang, Hao Wu, Haixia Wang, and Zhenmei Lü. Interspecies synergistic interactions mediated by cofactor exchange enhance stress tolerance by inducing biofilm formation. 9(9):e0088424–e0088424, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00884-24">[DOI]</a>.</p></li>
<li id="id5463"><p>Matthew Lima, Charandatta Muddana, Zhengyang Xiao, Anindita Bandyopadhyay, Pramod P. Wangikar, Himadri B. Pakrasi, and Yinjie Tang. The new chassis in the flask: advances in vibrio natriegens biotechnology research. 77():108464–108464, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2024.108464">[DOI]</a>.</p></li>
<li id="id5464"><p>Danlei Chen, Zhihua Qu, Shujuan Yang, Yujie Li, Shan Yu, Xin Yao Li, Yutong Lu, Aidong Yang, Chunjiang Liu, Shengbo Wu, and Jianjun Qiao. Quorum sensing regulating the productivity and stability of cross-feeding cocultivation. 502():158167–158167, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cej.2024.158167">[DOI]</a>.</p></li>
<li id="id5465"><p>Yonggang Gao and Cheng Zhao. Development and applications of metabolic models in plant multi-omics research. 15():1361183–1361183, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2024.1361183">[DOI]</a>.</p></li>
<li id="id5466"><p>Hatice Büşra Lüleci̇, Dilara Uzuner, Müberra Fatma Cesur, Atılay İlgün, Elif Düz, Ecehan Abdi̇k, Regan Odongo, and Tunahan Çakır. A benchmark of rna-seq data normalization methods for transcriptome mapping on human genome-scale metabolic networks. 10(1):124–124, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00448-z">[DOI]</a>.</p></li>
<li id="id5467"><p>Abdulla Abu-Salah, Müberra Fatma Cesur, Aiesha Anchan, Muhammet Ay, Monica R. Langley, Ahmed Shah, Pablo Reina-Gonzalez, Rachel Strazdins, Tunahan Çakır, and Souvarish Sarkar. Comparative proteomics highlights that genx exposure leads to metabolic defects and inflammation in astrocytes. 58(46):20525–20539, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.est.4c05472">[DOI]</a>.</p></li>
<li id="id5470"><p>Lizhan Zhang, Jianwen Ye, Gang Li, Helen Park, Hao Luo, Yina Lin, Shaowei Li, Weinan Yang, Yuying Guan, Fuqing Wu, Wuzhe Huang, Qiong Wu, Nigel S. Scrutton, Jens Nielsen, and Guo‐Qiang Chen. A long-term growth stable halomonas sp. deleted with multiple transposases guided by its metabolic network model halo-ecgem. 84():95–108, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2024.06.004">[DOI]</a>.</p></li>
<li id="id5471"><p>Xuangao Wu, Hee-Jong Yang, Myeong Seon Ryu, Su‐Jin Jung, Kyooseob Ha, Do-Yeon Jeong, and Sunmin Park. Association of mucin-degrading gut microbiota and dietary patterns with colonic transit time in constipation: a secondary analysis of a randomized clinical trial. 17(1):138–138, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/nu17010138">[DOI]</a>.</p></li>
<li id="id5578"><p>Fernando Cruz, João Capela, Eugénio C. Ferreira, Miguel Rocha, and Óscar Dias. Bioiso: an objective-oriented application for assisting the curation of genome-scale metabolic models. 21(2):215–226, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/tcbb.2023.3339972">[DOI]</a>.</p></li>
<li id="id5579"><p>Loreto Román, Felipe Melis-Arcos, Tomás Pröschle, Pedro A. Saa, and Daniel Garrido. Genome-scale metabolic modeling of the human milk oligosaccharide utilization by &lt;i&gt;bifidobacterium longum&lt;/i&gt; subsp. &lt;i&gt;infantis&lt;/i&gt;. 9(3):e0071523–e0071523, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00715-23">[DOI]</a>.</p></li>
<li id="id5594"><p>Pedro A. Saa, Sebastián Zapararte, Christopher Drovandi, and Lars K. Nielsen. Looplessfluxsampler: an efficient toolbox for sampling the loopless flux solution space of metabolic models. 25(1):3–3, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-023-05616-2">[DOI]</a>.</p></li>
<li id="id5595"><p>Yu Chen, Johan Gustafsson, Jingyu Yang, Jens Nielsen, and Eduard J. Kerkhoven. Single-cell omics analysis with genome-scale metabolic modeling. 86():103078–103078, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2024.103078">[DOI]</a>.</p></li>
<li id="id5596"><p>Telmo Blasco, Francesco Balzerani, Luis V. Valcárcel, Pedro Larrañaga, Concha Bielza, M. Pilar Francino, José Ángel Rufián‐Henares, Francisco J. Planes, and Sergio Pérez-Burillo. Bn-bacarena: bayesian network extension of bacarena for the dynamic simulation of microbial communities. 40(5):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btae266">[DOI]</a>.</p></li>
<li id="id5597"><p>Nicola Cortese, Anna Procopio, Alessio Merola, Paolo Zaffino, and Carlo Cosentino. Applications of genome-scale metabolic models to the study of human diseases: a systematic review. 256():108397–108397, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cmpb.2024.108397">[DOI]</a>.</p></li>
<li id="id5604"><p>Sol Han, Dohyeon Kim, Young-Shin Kim, and Sung Ho Yoon. Genome-scale metabolic network model and phenome of solvent-tolerant pseudomonas putida s12. 25(1):63–63, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-023-09940-y">[DOI]</a>.</p></li>
<li id="id5605"><p>Luis V. Valcárcel, Edurne San José‐Eneriz, Raquel Ordóñez, Iñigo Apaolaza, Danel Olaverri-Mendizabal, Naroa Barrena, Ana Valcárcel, Leire Gárate, Jesús F. San Miguel, Antonio Pineda‐Lucena, Xabier Agirre, Felipe Prósper, and Francisco J. Planes. An automated network-based tool to search for metabolic vulnerabilities in cancer. 15(1):8685–8685, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-024-52725-4">[DOI]</a>.</p></li>
<li id="id5606"><p>Renee Oles, Marvic Carrillo Terrazas, Luke R. Loomis, M J Neal, Mousumi Paulchakrabarti, Simone Zuffa, Chia-Yun Hsu, Adriana Vasquez Ayala, Michael H. Lee, Caitlin Tribelhorn, Pedro Belda‐Ferre, MacKenzie Bryant, Jasmine Zemlin, Jocelyn Young, Parambir S. Dulai, William J. Sandborn, Mamata Sivagnanam, Manuela Raffatellu, David T. Pride, Pieter C. Dorrestein, Karsten Zengler, Biswa Choudhury, Rob Knight, and Hiutung Chu. Pathogenic&lt;i&gt;bacteroides fragilis&lt;/i&gt;strains can emerge from gut-resident commensals. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.06.19.599758">[DOI]</a>.</p></li>
<li id="id5607"><p>Olivia Gallup, Kirill Sechkar, Sebastian Towers, and Harrison Steel. Computational synthetic biology enabled through jax: a showcase. 13(9):3046–3050, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.4c00307">[DOI]</a>.</p></li>
<li id="id5608"><p>Farid Zare and Ronan M. T. Fleming. Integration of proteomic data with genome‐scale metabolic models: a methodological overview. 33(10):e5150–e5150, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/pro.5150">[DOI]</a>.</p></li>
<li id="id5609"><p>Henry Ehlers, Nicolas Brich, Michael Krone, Martin Nöllenburg, Jiacheng Yu, Hiroaki Natsukawa, Xiaoru Yuan, and Hsiang‐Yun Wu. An introduction to and survey of biological network visualization. 126():104115–104115, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cag.2024.104115">[DOI]</a>.</p></li>
<li id="id5611"><p>Danel Olaverri-Mendizabal, Luis V. Valcárcel, Naroa Barrena, C. Rodriguez, and Francisco J. Planes. Review and meta-analysis of the genetic minimal cut set approach for gene essentiality prediction in cancer metabolism. 25(3):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbae115">[DOI]</a>.</p></li>
<li id="id5612"><p>Zhepu Ruan, Mengjun Xu, Youwen Xing, Kaiqing Yang, Xihui Xu, Jiandong Jiang, and Rongliang Qiu. Enhanced growth of wheat in contaminated fields via synthetic microbiome as revealed by genome-scale metabolic modeling. 953():176047–176047, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.scitotenv.2024.176047">[DOI]</a>.</p></li>
<li id="id5613"><p>M J Neal, William Brakewood, Michael J. Betenbaugh, and Karsten Zengler. Pan-genome-scale metabolic modeling of &lt;i&gt;bacillus subtilis&lt;/i&gt; reveals functionally distinct groups. 9(11):e0092324–e0092324, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00923-24">[DOI]</a>.</p></li>
<li id="id5614"><p>Ye Li, Jianlei Sun, Zhenhao Fu, Yubing He, Xiaorui Chen, Shijie Wang, Lele Zhang, Jiansheng Jian, Weihua Yang, Chunli Liu, Xiuxia Liu, Yankun Yang, and Zhonghu Bai. Engineering the l-tryptophan metabolism for efficient de novo biosynthesis of tryptophol in saccharomyces cerevisiae. 17(1):130–130, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-024-02576-4">[DOI]</a>.</p></li>
<li id="id5615"><p>Rola Shaaban, Susheel Bhanu Busi, Paul Wilmes, Jean‐Louis Guéant, and Almut Heinken. Personalized modeling of gut microbiome metabolism throughout the first year of life. 4(1):281–281, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s43856-024-00715-4">[DOI]</a>.</p></li>
<li id="id5670"><p>Freyr Jóhannsson, James T. Yurkovich, Steinn Guðmundsson, Ólafur E. Sigurjónsson, and Óttar Rolfsson. Temperature dependence of platelet metabolism. 14(2):91–91, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo14020091">[DOI]</a>.</p></li>
<li id="id5693"><p>Jose L. Cadavid, Nancy T. Li, and Alison P. McGuigan. Bridging systems biology and tissue engineering: unleashing the full potential of complex 3d &lt;i&gt;in vitro&lt;/i&gt; tissue models of disease. 5(2):021301–021301, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1063/5.0179125">[DOI]</a>.</p></li>
<li id="id5707"><p>Leila Hassani, Mohammad R. Moosavi, Payam Setoodeh, and Habil Zare. Fastknock: an efficient next-generation approach to identify all knockout strategies for strain optimization. 23(1):37–37, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-023-02277-x">[DOI]</a>.</p></li>
<li id="id5708"><p>David M. Versluis, Ruud Schoemaker, Ellen Looijesteijn, Jan M.W. Geurts, and Roeland M. H. Merks. 2′-fucosyllactose helps butyrate producers outgrow competitors in infant gut microbiota simulations. 27(3):109085–109085, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2024.109085">[DOI]</a>.</p></li>
<li id="id5709"><p>Panyawarin Soommat, Nachon Raethong, Ratchaprapa Ruengsang, Roypim Thananusak, Teeraphan Laomettachit, Kobkul Laoteng, Treenut Saithong, and Wanwipa Vongsangnak. Light-exposed metabolic responses of cordyceps militaris through transcriptome-integrated genome-scale modeling. 13(3):139–139, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biology13030139">[DOI]</a>.</p></li>
<li id="id5731"><p>Mustafa Sertbaş and Kutlu Ö. Ülgen. Uncovering the effect of sars-cov-2 on liver metabolism via genome-scale metabolic modeling for reprogramming and therapeutic strategies. 9(13):15535–15546, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acsomega.4c00392">[DOI]</a>.</p></li>
<li id="id5732"><p>Anqiang Ye, Jian-Ning Shen, Yong Li, Xiang Lian, Bin‐Guang Ma, and Feng‐Biao Guo. Reconstruction of the genome-scale metabolic network model of sinorhizobium fredii ccbau45436 for free-living and symbiotic states. 12():1377334–1377334, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2024.1377334">[DOI]</a>.</p></li>
<li id="id5733"><p>Tim Hulshof, Bram Nap, Filippo Martinelli Boneschi, and Ines Thiele. Microbial abundances retrieved from sequencing data—automated ncbi taxonomy (mars): a pipeline to create relative microbial abundance data for the microbiome modelling toolbox and utilizing homosynonyms for efficient mapping to resources. 4(1):vbae068–vbae068, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioadv/vbae068">[DOI]</a>.</p></li>
<li id="id5734"><p>Carlos J Rodriguez-Flores, Naroa Barrena, Danel Olaverri-Mendizabal, Idoia Ochoa, Luis V. Valcárcel, and Francisco J. Planes. Gmcspy: efficient and accurate computation of genetic minimal cut sets in python. 40(6):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btae318">[DOI]</a>.</p></li>
<li id="id5735"><p>Juan Pablo Molina Ortiz, Dale D. McClure, Andrew Holmes, Stuart A. Rice, Mark Read, and Erin R. Shanahan. Genome scale metabolic modelling of human gut microbes to inform rational community design. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.05.28.596116">[DOI]</a>.</p></li>
<li id="id5736"><p>Somdutt Ravindran, Nima Hajinajaf, Pritam Kundu, Jackson Comes, David R. Nielsen, Arul M. Varman, and Amit Ghosh. Genome-scale metabolic model reconstruction and investigation into the fluxome of the fast-growing cyanobacterium &lt;i&gt;synechococcus&lt;/i&gt; sp. pcc 11901. 13(10):3281–3294, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.4c00379">[DOI]</a>.</p></li>
<li id="id5737"><p>Micaela Chacón, Guadalupe Alvarez-Gonzalez, Piya Gosalvitr, Adokiye Berepiki, Karl Fisher, Rosa M. Cuéllar-Franca, and Neil Dixon. Complex waste stream valorization through combined enzymatic hydrolysis and catabolic assimilation by pseudomonas putida. 43(3):647–672, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2024.10.020">[DOI]</a>.</p></li>
<li id="id5746"><p>Eliott Jacopin, Yuki Sakamoto, Kozo Nishida, Kazunari Kaizu, and Koichi Takahashi. An architecture for collaboration in systems biology at the age of the metaverse. 10(1):12–12, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00334-8">[DOI]</a>.</p></li>
<li id="id5747"><p>Roman G. Bielski and M. Ahsanul Islam. Pangenome-scale mathematical modelling of anammox bacteria metabolism. 2(1):70–84, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/synbio2010005">[DOI]</a>.</p></li>
<li id="id5748"><p>Xavier Marbehan, Magali Roger, Frantz Fournier, Pascale Infossi, Emmanuel Guédon, Louis Delecourt, Régine Lebrun, Marie‐Thérèse Giudici‐Orticoni, and Stéphane Delaunay. Combining metabolic flux analysis with proteomics to shed light on the metabolic flexibility: the case of desulfovibrio vulgaris hildenborough. 15():1336360–1336360, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2024.1336360">[DOI]</a>.</p></li>
<li id="id5749"><p>Simran Kaur Aulakh, Oliver Lemke, Łukasz Szyrwiel, Stephan Kamrad, Yu Chen, Johannes Hartl, Michael Muelleder, Jens Nielsen, and Markus Ralser. The molecular landscape of cellular metal ion biology. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.02.29.582718">[DOI]</a>.</p></li>
<li id="id5750"><p>Sun Jin Moon, Anush Chiappino-Pepe, Wentao Dong, Joanne K. Kelleher, Matthew G. Vander Heiden, Gregory Stephanopoulos, and Hadley D. Sikes. Nadph composite index analysis quantifies the relationship between compartmentalized nadph dynamics and growth rates in cancer cells. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.04.27.591477">[DOI]</a>.</p></li>
<li id="id5751"><p>Lu Liu, Weihe Rong, Du Xiang, Qianqian Yuan, Zhaoyu Xu, Chang Ho Yu, Hongzhong Lu, Yanfei Wang, Yan Zhu, Zhijia Liu, and Guokun Wang. Integrating experimental and computational analyses of yeast protein profiles for optimizing the production of high-quality microbial proteins. 196(12):8741–8762, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12010-024-04995-3">[DOI]</a>.</p></li>
<li id="id5752"><p>N. Suhas Jagannathan, Javier Yu Peng Koh, Younghwan Lee, Radoslaw M. Sobota, Aaron T. Irving, Lin‐Fa Wang, Yoko Itahana, Koji Itahana, and Lisa Tucker‐Kellogg. Multi-omic analysis of bat versus human fibroblasts reveals altered central metabolism. 13():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.94007">[DOI]</a>.</p></li>
<li id="id5753"><p>Jungyeon Kim, Gun-Hwi Yeon, Mi Jin Kim, Jung‐Hoon Bae, Jung‐Hoon Sohn, and Bong Hyun Sung. Systems metabolic engineering to elucidate and enhance intestinal metabolic activities of &lt;i&gt;escherichia coli&lt;/i&gt; nissle 1917. 72(32):18234–18246, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.jafc.4c00182">[DOI]</a>.</p></li>
<li id="id5754"><p>Ryan Schildcrout, Kirk Smith, Rupa Bhowmick, Suraj Menon, Minali Kapadia, Emily G. Kurtz, Anya Coffeen-Vandeven, and Sriram Chandrasekaran. Recon8d: a metabolic regulome network from oct-omics and machine learning. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.08.17.608400">[DOI]</a>.</p></li>
<li id="id5755"><p>Huiying Zhao, Sarula Bai, Jian Tan, Ming Liu, Yuchao Zhao, and Linshu Jiang. Can meta-omics revolutionize our understanding of rumen methane emissions? 1():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1017/anr.2024.18">[DOI]</a>.</p></li>
<li id="id5756"><p>Jianhang Zhou, Xiaogang Wu, Yiming Feng, Tangran Huo, Yun‐Peng Zhao, Juejun Pan, and Sitong Liu. Bacterial communication intelligently regulates their interactions in anammox consortia under decreasing temperatures. 954():176239–176239, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.scitotenv.2024.176239">[DOI]</a>.</p></li>
<li id="id5757"><p>Elena Lucy Carter, Nicholas R. Waterfield, Chrystala Constantinidou, and Mohammad Tauqeer Alam. A temperature-induced metabolic shift in the emerging human pathogen &lt;i&gt;photorhabdus asymbiotica&lt;/i&gt;. 9(11):e0097023–e0097023, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00970-23">[DOI]</a>.</p></li>
<li id="id5758"><p>Jungyeon Kim, Y S Kim, Juyoung Kim, Min‐Ju Seo, Soo‐Jin Yeom, and Bong Hyun Sung. Investigation of bottleneck enzyme through flux balance analysis to improve glycolic acid production in escherichia coli. 62(11):1023–1033, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12275-024-00175-4">[DOI]</a>.</p></li>
<li id="id5759"><p>Alexandra Manchel, Michelle Gee, and Rajanikanth Vadigepalli. From sampling to simulating: single-cell multiomics in systems pathophysiological modeling. 27(12):111322–111322, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2024.111322">[DOI]</a>.</p></li>
<li id="id5760"><p>Siqing Fan, Hao Ren, Xueni Fu, Xiangyu Kong, Hao Wu, and Zhenmei Lv. Genome streamlining of &lt;i&gt;pseudomonas putida&lt;/i&gt; b6-2 for bioremediation. 9(12):e0084524–e0084524, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00845-24">[DOI]</a>.</p></li>
<li id="id5761"><p>Sandra M. Correa, Asdrúbal Burgos, Álvaro Cuadros‐Inostroza, Ke Xu, Yariv Brotman, and Zoran Nikoloski. A data-integrative modeling approach accurately characterizes the effects of mutations on arabidopsis lipid metabolism. 197(2):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/plphys/kiae615">[DOI]</a>.</p></li>
<li id="id5762"><p>Tim Hensen, Shahzad Ahmad, Gabi Kastenmüller, Robert Kraaij, Mohsen Ghanbari, M. Arfan Ikram, Rima Kaddurah‐Daouk, and Ines Thiele. In silico metabolic modelling links microbiome-derived metabolites to risk factors of alzheimer’s disease. 2(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/29933935.2024.2443171">[DOI]</a>.</p></li>
<li id="id5769"><p>Zhihao Liu, Minghao Chen, Jingmin Hu, Yonghong Wang, and Yu Chen. Minimization of proteome reallocation explains metabolic transition in hierarchical utilization of carbon sources. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.01.23.576957">[DOI]</a>.</p></li>
<li id="id5770"><p>Yuchao Zhao, Jian Tan, Luoyun Fang, and Linshu Jiang. Harnessing meta-omics to unveil and mitigate methane emissions in ruminants: integrative approaches and future directions. 951():175732–175732, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.scitotenv.2024.175732">[DOI]</a>.</p></li>
<li id="id5771"><p>Nan Xu, Jiaojiao Zuo, Chenghao Li, Cong Gao, and Minliang Guo. Reconstruction and analysis of a genome-scale metabolic model of acinetobacter lwoffii. 25(17):9321–9321, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms25179321">[DOI]</a>.</p></li>
<li id="id5772"><p>Helen Scott and Daniel Segrè. Metabolic flux modeling in marine ecosystems. 17(1):593–620, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1146/annurev-marine-032123-033718">[DOI]</a>.</p></li>
<li id="id5773"><p>Sowmya Manojna Narasimha, Tanisha Malpani, Omkar S. Mohite, J. Saketha Nath, and Karthik Raman. Understanding flux switching in metabolic networks through an analysis of synthetic lethals. 10(1):104–104, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00426-5">[DOI]</a>.</p></li>
<li id="id5774"><p>Karel Olavarría and Diana Z. Sousa. Thermodynamic tools for more efficient biotechnological processes: an example in poly-(3-hydroxybutyrate) production from carbon monoxide. 90():103212–103212, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2024.103212">[DOI]</a>.</p></li>
<li id="id5775"><p>Daniella M. Hernandez, Melissa M Marzouk, Madeline Cole, Marla C. Fortoul, S. Kethireddy, Rehan Contractor, Habibul Islam, Trent Moulder, Ariane R. Kalifa, Estefania Marin Meneses, Maximiliano Barbosa Mendoza, Ruth Thomas, S. Masud, Sheena Pubien, Patricia Milanes, Gabriela Diaz-Tang, Allison J. Lopatkin, and Robert P. Smith. Purine and pyrimidine synthesis differently affect the strength of the inoculum effect for aminoglycoside and β-lactam antibiotics. 12(12):e0189524–e0189524, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.01895-24">[DOI]</a>.</p></li>
<li id="id5776"><p>Miha Moškon and Tadeja Režen. Integration and analysis of omics data using genome-scale metabolic models. 14(11):595–595, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo14110595">[DOI]</a>.</p></li>
<li id="id5821"><p>Gustav Sjöberg, Alīna Reķēna, Matilda Fornstad, Petri‐Jaan Lahtvee, and Antonius J. A. van Maris. Evaluation of enzyme-constrained genome-scale model through metabolic engineering of anaerobic co-production of 2,3-butanediol and glycerol by saccharomyces cerevisiae. 82():49–59, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2024.01.007">[DOI]</a>.</p></li>
<li id="id5822"><p>Muhammed Erkan Karabekmez. Insights into yeast response to chemotherapeutic agent through time series genome‐scale metabolic models. 121(10):3351–3359, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28833">[DOI]</a>.</p></li>
<li id="id5835"><p>Chania Clare, Jack W. Rutter, Alex J. H. Fedorec, Stefanie Frank, and C. Barnes. Bacterial microcompartment utilisation in the human commensal&lt;i&gt;escherichia coli&lt;/i&gt;nissle 1917. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.05.13.593892">[DOI]</a>.</p></li>
<li id="id5864"><p>Aruldoss Immanuel, Ragothaman M. Yennamalli, and Venkatasubramanian Ulaganathan. Targeting the bottlenecks in levan biosynthesis pathway in &lt;i&gt;bacillus subtilis&lt;/i&gt; and strain optimization by computational modeling and omics integration. 28(2):49–58, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/omi.2023.0277">[DOI]</a>.</p></li>
<li id="id5865"><p>Lena Peters, Moritz Drechsler, Barbara Pees, Georgia Angelidou, Liesa Salzer, Karlis Arturs Moors, Nicole Paczia, Hinrich Schulenburg, Yi‐Ming Shi, Christoph Kaleta, Michael Witting, Helge B. Bode, and Katja Dierking. Polyketide synthase-derived sphingolipids determine microbiota-mediated protection against pathogens in&lt;i&gt;c. elegans&lt;/i&gt;. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.02.06.579051">[DOI]</a>.</p></li>
<li id="id5866"><p>Roland Sauter, Suraj Sharma, and Ines Heiland. Accounting for nad concentrations in genome-scale metabolic models captures important metabolic alterations in nad-depleted systems. 14(5):602–602, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biom14050602">[DOI]</a>.</p></li>
<li id="id5867"><p>Richard D. Paul, Johann F. Jadebeck, Anton Stratmann, Wolfgang Wiechert, and Katharina Nöh. Hopsy — a methods marketplace for convex polytope sampling in python. 40(7):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btae430">[DOI]</a>.</p></li>
<li id="id5868"><p>Hatice Büşra Lüleci̇, Attila Jones, and Tunahan Çakır. Multi‐omics analyses highlight molecular differences between clinical and neuropathological diagnoses in alzheimer's disease. 60(5):4922–4936, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/ejn.16482">[DOI]</a>.</p></li>
<li id="id5869"><p>Luhui Zhang, Xinpei Sun, Jianwen Ye, Quan Yuan, Xin Zhang, Fei Sun, Yongpan An, Yutong Chen, Yong Qian, Daqian Yang, Qian Wang, Miaomiao Gao, Tao Chen, Hongwu Ma, Guo‐Qiang Chen, and Zhengwei Xie. Reconstruction and analyses of genome-scale halomonas metabolic network yield a highly efficient pha production. 19():e00251–e00251, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2024.e00251">[DOI]</a>.</p></li>
<li id="id5870"><p>Ronan M. T. Fleming, Hulda S. Haraldsdóttir, German Preciat, Luojiao Huang, Ines Thiele, Amy C. Harms, and Thomas Hankemeier. Conserved moiety fluxomics. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.21.624666">[DOI]</a>.</p></li>
<li id="id5871"><p>Jiahang Li, Wolfram Weckwerth, and Steffen Waldherr. Network structure and fluctuation data improve inference of metabolic interaction strengths with the inverse jacobian. 10(1):137–137, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-024-00457-y">[DOI]</a>.</p></li>
<li id="id5886"><p>Naveed Ziari and Marc K. Hellerstein. Pymida: a graphical user interface for mass isotopomer distribution analysis. 96(6):2303–2308, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.analchem.3c02211">[DOI]</a>.</p></li>
<li id="id5887"><p>Lena Peters, Moritz Drechsler, Barbara Pees, Georgia Angelidou, Liesa Salzer, Karlis Arturs Moors, Nicole Paczia, Hinrich Schulenburg, Yi‐Ming Shi, Christoph Kaleta, Michael Witting, Helge B. Bode, and Katja Dierking. Polyketide synthase-derived sphingolipids determine microbiota-mediated protection against pathogens in c. elegans. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3853431/v1">[DOI]</a>.</p></li>
<li id="id5888"><p>Samvedna Singh, Himanshi Gupta, and Shakti Sahi. Computational methods for detection of host-pathogen interactions. ():103–130, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-95890-5.00013-2">[DOI]</a>.</p></li>
<li id="id5890"><p>Tong Hao, Zhentao Song, Mingzhi Zhang, Lingrui Zhang, Jiarui Yang, Jinɡjinɡ Li, and Jinsheng Sun. Reconstruction of metabolic–protein interaction integrated network of eriocheir sinensis and analysis of ecdysone synthesis. 15(4):410–410, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/genes15040410">[DOI]</a>.</p></li>
<li id="id5891"><p>Daniella M. Hernandez, Melissa M Marzouk, Madeline Cole, Marla C. Fortoul, S. Kethireddy, Rehan Contractor, Habibul Islam, Trent Moulder, Ariane R. Kalifa, Estefania Marin Meneses, Maximiliano Barbosa Mendoza, Ruth Thomas, S. Masud, Sheena Pubien, Patricia Milanes, Gabriela Diaz-Tang, Allison J. Lopatkin, and Robert P. Smith. Purine and pyrimidine synthesis differently affect the strength of the inoculum effect for aminoglycoside and β-lactam antibiotics. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.04.09.588696">[DOI]</a>.</p></li>
<li id="id5892"><p>Ali Kishk, Maria Pires Pacheco, Tony Heurtaux, and Thomas Sauter. Metabolic models predict fotemustine and the combination of eflornithine/rifamycin and adapalene/cannabidiol for the treatment of gliomas. 25(3):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbae199">[DOI]</a>.</p></li>
<li id="id5893"><p>Biki Bapi Kundu, Jayanth Krishnan, Richard Szubin, Arjun Patel, Bernhard Ø. Palsson, Daniel C. Zielinski, and Caroline M. Ajo‐Franklin. Extracellular respiration is a latent energy metabolism in&lt;i&gt;escherichia coli&lt;/i&gt;. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.05.30.596743">[DOI]</a>.</p></li>
<li id="id5894"><p>Ehsan Motamedian, Matiss Ricards Baumanis, Jānis Daugavietis, Ieva Bērziņa, Karlis Shvirksts, Elīna Dāce, Jānis Liepiņš, and Egils Stalidzāns. A hybrid strategy for growth-associated production of sophorolipid using a reconstructed genome-scale metabolic model for the fructophilic yeast starmerella bombicola. 189():541–548, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.psep.2024.06.094">[DOI]</a>.</p></li>
<li id="id5895"><p>Soraya Mirzaei and Mojtaba Tefagh. Gem-based computational modeling for exploring metabolic interactions in a microbial community. 20(6):e1012233–e1012233, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1012233">[DOI]</a>.</p></li>
<li id="id5896"><p>Carlos Focil, Gabriela Canto-Encalada, D. O. Campos, Cristal Zúñiga, and Alejandro Zepeda. Applying genome-scale metabolic modeling tools to understand microbial communities in wastewater treatment. ():297–332, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-443-13609-2.00020-3">[DOI]</a>.</p></li>
<li id="id5897"><p>Renan Tosin, Igor Portis, Leandro Rodrigues, Igor Gonçalves, Catarina Barbosa, Jorge Teixeira, Rafael J. Mendes, Filipe Neves dos Santos, Conceição Santos, Rui C. Martins, and Mário Cunha. Integrating spectral sensing and systems biology for precision viticulture: effects of shade nets on grapevine leaves. 10(8):873–873, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/horticulturae10080873">[DOI]</a>.</p></li>
<li id="id5898"><p>Osbaldo Resendis‐Antonio, Jorge E. Arellano-Villavicencio, Aarón Vázquez-Jiménez, Juan José Oropeza-Valdez, Cristian Padrón-Manrique, Heriberto Prado-Garcı́a, and Armando R. Tovar. Intratumoral heterogeneity and metabolic cross-feeding in a three-dimensional model of breast cancer: an in silico perspective. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4864972/v1">[DOI]</a>.</p></li>
<li id="id5899"><p>Carolina A. Contador, Ailin Liu, Ming‐Sin Ng, Yee‐Shan Ku, Siu Hung Joshua Chan, and Hon‐Ming Lam. Contextualized metabolic modelling revealed factors affecting isoflavone accumulation in soybean seeds. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/pce.15140">[DOI]</a>.</p></li>
<li id="id5900"><p>Boyu Jiang, Nick Quinn-Bohmann, Christian Diener, Vignesh Bose Nathan, Yu Han-Hallett, Lavanya Reddivari, Sean M. Gibbons, and Priyanka Baloni. Understanding disease-associated metabolic changes in human colon epithelial cells using&lt;i&gt;i&lt;/i&gt;colonepithelium metabolic reconstruction. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.10.22.619644">[DOI]</a>.</p></li>
<li id="id5901"><p>Tunahan Çakır, Ecehan Abdi̇k, Dilara Uzuner, and Hatice Büşra Lüleci̇. Genome-scale metabolic modeling for diseases. ():258–265, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-95502-7.00227-x">[DOI]</a>.</p></li>
<li id="id5902"><p>Luojiao Huang, German Preciat, Jesus Alarcon-Gil, Edinson Lucumi Moreno, Agnieszka B. Wegrzyn, Ines Thiele, Emma Schymanski, Amy C. Harms, Ronan M. T. Fleming, and Thomas Hankemeier. Fluxtram: integration of tracer-based metabolomics data into atomically resolved genome-scale metabolic networks for metabolic flux analysis. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.26.625485">[DOI]</a>.</p></li>
<li id="id5903"><p>Chania Clare, Jack W. Rutter, Alex J. H. Fedorec, Stefanie Frank, and C. Barnes. Bacterial microcompartment utilization in the human commensal &lt;i&gt;escherichia coli&lt;/i&gt; nissle 1917. 206(12):e0026924–e0026924, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00269-24">[DOI]</a>.</p></li>
<li id="id5904"><p>Zijian Hu, Jingyi Qian, Yuzhou Wang, and Chao Ye. Current status and applications of genome‐scale metabolic models of oleaginous microorganisms. 3(4):492–511, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/fbe2.12113">[DOI]</a>.</p></li>
<li id="id5909"><p>Artai R. Moimenta, Diego Troitiño-Jordedo, and Eva Balsa‐Canto. Integrating kinetic models with genome-scale constraints to explain yeast-based batch fermentation. ():6311–6316, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cdc56724.2024.10886098">[DOI]</a>.</p></li>
<li id="id6000"><p>Muhammed Erkan Karabekmez. Insights into yeast response to chemotherapeutic agent through time series genome-scale metabolic models. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.01.02.573982">[DOI]</a>.</p></li>
<li id="id6001"><p>Emanuel Lange, Lena Kranert, Jacob Krüger, Dirk Benndorf, and Robert Heyer. Microbiome modeling: a beginner’s guide. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202401.0789.v1">[DOI]</a>.</p></li>
<li id="id6002"><p>Chloe V. McCreery, Drew Alessi, Katarina Mollo, Alessio Fasano, and Ali R. Zomorrodi. Investigating intestinal epithelium metabolic dysfunction in celiac disease using personalized genome-scale models. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.04.29.591234">[DOI]</a>.</p></li>
<li id="id6003"><p>Fernando Silva-Lance, Isabel Montejano-Montelongo, Èric Bautista, Lars K. Nielsen, Pär I. Johansson, and Igor Marín de Mas. Integrating genome-scale metabolic models with patient plasma metabolome to study endothelial metabolism in situ. 25(10):5406–5406, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms25105406">[DOI]</a>.</p></li>
<li id="id6004"><p>Indumathi Palanikumar, Himanshu Sinha, and Karthik Raman. Panera: an innovative framework for surmounting uncertainty in microbial community modeling using pan-genera metabolic models. 27(7):110358–110358, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2024.110358">[DOI]</a>.</p></li>
<li id="id6005"><p>Miguel Angel Ramos-Valdovinos, Prisciluis Caheri Salas-Navarrete, Gerardo Ruíz Amores, Ana Lilia Hernández-Orihuela, and Agustino Martı́nez-Antonio. Qualitative perturbation analysis and machine learning: elucidating bacterial optimization of tryptophan production. 17(7):282–282, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/a17070282">[DOI]</a>.</p></li>
<li id="id6006"><p>Devlin C. Moyer, Justin Reimertz, Daniel Segrè, and Juan I. Fuxman Bass. Semi-automatic detection of errors in genome-scale metabolic models. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.06.24.600481">[DOI]</a>.</p></li>
<li id="id6007"><p>Maria Pires Pacheco, Déborah Gerard, Riley J. Mangan, Alec R. Chapman, Dennis Hecker, Manolis Kellis, Marcel H. Schulz, Lasse Sinkkonen, and Thomas Sauter. Epigenetic control of metabolic identity across cell types. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.07.24.604914">[DOI]</a>.</p></li>
<li id="id6023"><p>David Geller-McGrath. Characterization of microbial primary and secondary metabolism in the marine realm. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1575/1912/70396">[DOI]</a>.</p></li>
<li id="id6115"><p>Shaohua Guan, Zhichao Zhang, Zihan Zhang, and Hualin Shi. Universal scaling relation and criticality in metabolism and growth of escherichia coli. 6(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1103/physrevresearch.6.013035">[DOI]</a>.</p></li>
<li id="id6116"><p>Míriam Tarrado‐Castellarnau, Carles Foguet, Josep Tarragó‐Celada, Marc Palobart, Claudia Hernández-Carro, Jordi Perarnau, Erika Zodda, Ibrahim H. Polat, Silvia Marín, Alejandro Suárez‐Bonnet, Juan José Lozano, Mariia Yuneva, Timothy M. Thomson, and Marta Cascante. Glutaminase as a metabolic target of choice to counter acquired resistance to palbociclib by colorectal cancer cells. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.01.04.574237">[DOI]</a>.</p></li>
<li id="id6117"><p>Mr Goodluck Kam'bwemba - and Dr Johnson Okeniyi -. Information and communications technologies, to raise quality of deceased estate disbursement: administrator general's office, malawi. 6(1):, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.36948/ijfmr.2024.v06i01.11037">[DOI]</a>.</p></li>
<li id="id6118"><p>C. Rodriguez, Naroa Barrena, Danel Olaverri-Mendizabal, Idoia Ochoa, Luis V. Valcárcel, and Francisco J. Planes. Gmcspy: efficient and accurate computation of genetic minimal cut sets in python. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.02.02.578370">[DOI]</a>.</p></li>
<li id="id6119"><p>Aamir Faisal Ansari, Gayathri Sambamoorthy, Thrisha C Alexander, Yugandhar B. S. Reddy, Janhavi S. Raut, and Narendra M. Dixit. Disentangling competitive and cooperative components of the interactions between microbial species. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.02.06.579244">[DOI]</a>.</p></li>
<li id="id6120"><p>James Morrissey, Benjamin Strain, and Cleo Kontoravdi. Flux balance analysis of mammalian cell systems. 2774():119–134, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-3718-0_9">[DOI]</a>.</p></li>
<li id="id6121"><p>Sowmya Manojna, Tanisha Malpani, Omkar S. Mohite, Saketha Nath, and Karthik Raman. Understanding flux switching in metabolic networks through an analysis of synthetic lethals. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.03.05.583461">[DOI]</a>.</p></li>
<li id="id6122"><p>Amandine Paulay, Ghjuvan Micaelu Grimaud, Raphaël Caballero, Béatrice Laroche, Marion Leclerc, Simon Labarthe, and Emmanuelle Maguin. Design of a proteolytic module for improved metabolic modeling of &lt;i&gt;bacteroides caccae&lt;/i&gt;. 9(4):e0015324–e0015324, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00153-24">[DOI]</a>.</p></li>
<li id="id6123"><p>Jonathan Josephs‐Spaulding, Hannah Clara Rettig, Johannes Zimmermann, Mariam Chkonia, Alexander Mischnik, Sören Franzenburg, Simon Graspeuntner, Jan Rupp, and Christoph Kaleta. Metatranscriptomics-based metabolic modeling of patient-specific urinary microbiome during infection. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.03.25.586446">[DOI]</a>.</p></li>
<li id="id6124"><p>Alexandre Maciel-Guerra, Kubra Babaarslan, Michelle Baker, Aura Rahman, Maqsud Muhammad Hossain, Abdus Sadique, Jahidul Alam, Salim Uzzaman, Mohammad Ferdous Rahman Sarker, Nasrin Sultana, Ashraful Islam Khan, Yasmin Ara Begum, Mokibul Hassan Afrad, Nicola Senin, Zakir Hossain Habib, Tahmina Shirin, Firdausi Qadri, and Tania Dottorini. Signature genomic traits of the core and accessory genome of vibrio cholerae o1 drive lineage transmission and disease severity. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4184222/v1">[DOI]</a>.</p></li>
<li id="id6130"><p>Dilara Uzuner, Atılay İlgün, Fatma Betül Bozkurt, and Tunahan Çakır. A personalized metabolic modelling approach through integrated analysis of rna-seq-based genomic variants and gene expression levels in alzheimer’s disease. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.04.24.590807">[DOI]</a>.</p></li>
<li id="id6131"><p>Yadira Boada, Marcelo Flores Vázquez, Marco Córdova, Francisco Flores, and Alejandro Vignoni. Synthetic biology design principles enable efficient bioproduction of heparosan with low polydispersion index for the biomedical industry. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4356018/v1">[DOI]</a>.</p></li>
<li id="id6132"><p>Nam Kyu Kang, Hyun Gi Koh, Donald R. Ort, and Yong‐Su Jin. Bioconversion of co2 into valuable bioproducts via a synthetic modular co-culture of engineered chlamydomonas reinhardtii and escherichia coli. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4823901">[DOI]</a>.</p></li>
<li id="id6133"><p>Holly R. Steach, Siddharth Viswanath, Yixuan He, Xitong Zhang, Natalia Ivanova, Matthew Hirn, Michael Perlmutter, and Smita Krishnaswamy. Inferring metabolic states from single cell transcriptomic data via geometric deep learning. ():235–252, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-3989-4_15">[DOI]</a>.</p></li>
<li id="id6134"><p>Fatima Zulqarnain, Xueheng Zhao, Kenneth D.R. Setchell, Yash Sharma, Phillip Fernandes, Sanjana Srivastava, Aman Shrivastava, Lubaina Ehsan, Varun Jain, Shyam S. Raghavan, Christopher A. Moskaluk, Yael Haberman, Lee A. Denson, Khyati Mehta, Najeeha Talat Iqbal, Najeeb Rahman, Kamran Sadiq, Zubair Ahmad, Romana Idress, Junaid Iqbal, Sheraz Ahmed, Aneeta Hotwani, Fayyaz Umrani, Beatrice Amadi, Paul Kelly, Donald E. Brown, Sean R. Moore, Asad Ali, and Sana Syed. Machine-learning-based integrative –‘omics analyses reveal immunologic and metabolic dysregulation in environmental enteric dysfunction. 27(6):110013–110013, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2024.110013">[DOI]</a>.</p></li>
<li id="id6135"><p>Priyanka Choudhury, Sanjukta Dasgupta, Parthasarathi Bhattacharyya, Sushmita Roychowdhury, and Koel Chaudhury. Understanding pulmonary hypertension: the need for an integrative metabolomics and transcriptomics approach. 20(6):366–389, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d3mo00266g">[DOI]</a>.</p></li>
<li id="id6136"><p>Telmo Blasco. Reconstruction and analysis of gut microbiota metabolic networks in the context of personalized nutrition. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.15581/10171/69490">[DOI]</a>.</p></li>
<li id="id6137"><p>Maurício Alexander de Moura Ferreira and Wendel Batista da Silveira. Multi-omics data and model integration reveal the main mechanisms associated with respiro-fermentative metabolism and ethanol stress responses in kluyveromyces marxianus. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.06.06.597719">[DOI]</a>.</p></li>
<li id="id6138"><p>Kayden KM Low and Maurice HT Ling. Cell modelling and simulation. ():445–455, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-95502-7.00105-6">[DOI]</a>.</p></li>
<li id="id6139"><p>Drew Alessi, Chloe V. McCreery, and Ali R. Zomorrodi. Unraveling the link between nutrition and metabolic syndrome risk through in silico dietary interventions. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.07.04.602078">[DOI]</a>.</p></li>
<li id="id6140"><p>Elías Vera-Sigüenza, Cristina Escribano-Gonzalez, Irene Serrano‐Gonzalo, Kattri‐Liis Eskla, Charlotte Speakman, Alejandro Huerta Uribe, Lisa A. Vettore, Himani Rana, Adam Boufersaoui, Hans Vellama, Ramin Nashebi, Ielyaas Cloete, Jennie Roberts, Supratik Basu, Mark T. Drayson, Christopher M. Bunce, Guy Pratt, Fabian Spill, Oliver D.K. Maddocks, and Daniel A. Tennant. Pyruvate from bone marrow mesenchymal stem cells supports myeloma redox homeostasis and anabolism. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.08.08.607157">[DOI]</a>.</p></li>
<li id="id6141"><p>Maurício Alexander de Moura Ferreira. Model-driven evaluation of microbial physiology: insights from protein allocation. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.47328/ufvbbt.2024.425">[DOI]</a>.</p></li>
<li id="id6142"><p>Eduardo Luís Menezes de Almeida, Maurício Alexander de Moura Ferreira, and Wendel Batista da Silveira. Lista-gem: the genome-scale metabolic reconstruction of lipomyces starkeyi. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-4784404/v1">[DOI]</a>.</p></li>
<li id="id6143"><p>Maurício Alexander de Moura Ferreira and Wendel Batista da Silveira. Multi-omics data and model integration reveal the main mechanisms associated with respiro-fermentative metabolism and ethanol stress responses in kluyveromyces marxianus. 211():109471–109471, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2024.109471">[DOI]</a>.</p></li>
<li id="id6144"><p>Fatma Zehra Sarı and Tunahan Çakır. Deciphering antibiotic-targeted metabolic pathways in acinetobacter baumannii: insights from transcriptomics and genome-scale metabolic modeling. 14(9):1102–1102, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/life14091102">[DOI]</a>.</p></li>
<li id="id6145"><p>Satyajit Beura, Amit Kumar Das, and Amit Ghosh. Protocol for genome-scale differential flux analysis to interrogate metabolic differences from gene expression data. 5(3):103291–103291, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xpro.2024.103291">[DOI]</a>.</p></li>
<li id="id6146"><p>Maurício Alexander de Moura Ferreira, Eduardo Luís Menezes de Almeida, Wendel Batista da Silveira, and Zoran Nikoloski. Integrating promiscuous enzyme activities in protein-constrained models pinpoints the role of underground metabolism in robustness of metabolic phenotypes. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.09.06.611666">[DOI]</a>.</p></li>
<li id="id6147"><p>Mi Ae Park, Sharifah Nora Ahmad Almunawar, Rachel Rui Xia Lim, Sumanto Haldar, Christiani Jeyakumar Henry, and Oleg V. Moskvin. Genomic reconstruction and dietary response assessment of three &lt;i&gt;clostridium leptum&lt;/i&gt; -related bacteria isolated from fecal samples of singapore subjects. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.09.13.612987">[DOI]</a>.</p></li>
<li id="id6148"><p>Moohebat Pourmajidian and Sayed‐Amir Marashi. Investigation of cell metabolism dysfunction in autism spectrum disorders using genome-scale metabolic network analysis of neuronal development. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.10.01.616152">[DOI]</a>.</p></li>
<li id="id6149"><p>Jorge E. Arellano-Villavicencio, Aarón Vázquez-Jiménez, Juan José Oropeza-Valdez, Cristian Padrón-Manrique, Heriberto Prado-Garcı́a, Armando R. Tovar, and Osbaldo Resendis‐Antonio. Intratumoral heterogeneity and metabolic cross-feeding in a three-dimensional breast cancer culture: an in silico perspective. 25(20):10894–10894, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms252010894">[DOI]</a>.</p></li>
<li id="id6150"><p>Yuchen Zhang, Wenkai Lai, Meiling Wang, Lai Shirong, Qing Liu, Qi Luo, and Yang Fenglong. Gut microbial metabolic disorder in depression: insights from computational modeling and mediation analysis. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.10.10.617513">[DOI]</a>.</p></li>
<li id="id6151"><p>Huili Yuan, Yang Bai, and Xiongfei Fu. A global resource constrained model to predict metabolic flux dynamics in fluctuating environments. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.10.12.617971">[DOI]</a>.</p></li>
<li id="id6152"><p>Francis Mairet. In silico identification of switching nodes in metabolic networks. 4():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.24072/pcjournal.480">[DOI]</a>.</p></li>
<li id="id6153"><p>Xin Bush, Erica J. Fratz‐Berilla, Casey Kohnhorst, Robert J. King, Cyrus Agarabi, David N. Powers, and Nicholas Trunfio. Defining golden batches in biomanufacturing processes from internal metabolic activity to detect process changes that may affect product quality. 122(2):298–305, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28873">[DOI]</a>.</p></li>
<li id="id6154"><p>Sergio Sánchez‐García, Adrián Povo‐Retana, Silvia Marín, Sergio Madurga, Marco Fariñas, Nuria Aleixandre, Antonio Castrillo, Juan Vladimir de la Rosa, Carlota Alvarez‐Lucena, Rodrigo Landauro‐Vera, Patricia Prieto, Marta Cascante, and Lisardo Boscá. Immunometabolic effect of nitric oxide on human macrophages challenged with the sars‐cov2‐induced cytokine storm. a fluxomic approach. 14(1):e2401688–e2401688, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/adhm.202401688">[DOI]</a>.</p></li>
<li id="id6155"><p>A. Hacibektascedilla ̃oghacek ̃lu, Ilija Dukovski, and Daniel Segrè. Dynamic metabolic modeling of atp allocation during viral infection. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.12.623198">[DOI]</a>.</p></li>
<li id="id6156"><p>Rami Balasubramanian, Debajit Saha, A. Arun, and P. K. Vinod. Hypometabolism in autism spectrum disorder: insights from brain and blood transcriptomics. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.18.624092">[DOI]</a>.</p></li>
<li id="id6157"><p>Elaine Zaunseder, Faiz Khan Mohammad, Ulrike Mütze, Stefan Kölker, Vincent Heuveline, and Ines Thiele. Dynamic whole-body models for infant metabolism. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.11.25.625291">[DOI]</a>.</p></li>
<li id="id6158"><p>Hadi Nazem‐Bokaee, Erik Hom, Sarah Mathews, and Cécile Gueidan. Analyzing sorbitol biosynthesis using a metabolic network flux model of a lichenized strain of the green microalga &lt;i&gt;diplosphaera chodatii&lt;/i&gt;. 13(1):e0366023–e0366023, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.03660-23">[DOI]</a>.</p></li>
<li id="id6159"><p>Qin-Tian Qiu, Caiyun Zhang, Zhipeng Gao, and Bin‐Guang Ma. Spatial chromosome organization and adaptation of the radiation-resistant extremophile deinococcus radiodurans. 301(1):108068–108068, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbc.2024.108068">[DOI]</a>.</p></li>
<li id="id6160"><p>Muhammed Erkan Karabekmez and Merve Yarıcı. Omics and community modeling applications in genome-scale metabolic models. ():544–555, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-95502-7.00252-9">[DOI]</a>.</p></li>
<li id="id6161"><p>Huijun Long, Jai W. Mehat, Huihai Wu, Arnoud H. M. van Vliet, and Roberto M. La Ragione. Use of genome scale metabolic reconstructions of avian pathogenic &lt;i&gt;escherichia coli&lt;/i&gt; (apec) phylogroups for the identification of lineage-specific metabolic pathways. ():, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2024.12.20.629819">[DOI]</a>.</p></li>
<li id="id6171"><p>Clémence Dupont Thibert, Souvik Roy, Gilles Curien, and Maxime Durot. Biomodtool: from biomass composition data to structured biomass objective functions for genome-scale metabolic models. 5(1):vbaf036–vbaf036, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioadv/vbaf036">[DOI]</a>.</p></li>
<li id="id6183"><p>Alexandre Oliveira, Jorge Ferreira, Vítor Vieira, Bruno Sá, and Miguel Rocha. &lt;i&gt;troppo&lt;/i&gt;: tissue-specific reconstruction and phenotype prediction using omics data. 5(1):vbaf113–vbaf113, 2024. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioadv/vbaf113">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-2" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-2" name="0-2" role="tabpanel" tabindex="0"><div class="docutils container" id="id6243">
<ol class="arabic simple" start="1">
<li id="id6367"><p>Nicolás Gurdo, Daniel C. Volke, Douglas McCloskey, and Pablo I. Nikel. Automating the design-build-test-learn cycle towards next-generation bacterial cell factories. 74():1–15, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.nbt.2023.01.002">[DOI]</a>.</p></li>
<li id="id6373"><p>Partho Sen and Matej Orešič. Integrating omics data in genome-scale metabolic modeling: a methodological perspective for precision medicine. 13(7):855–855, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo13070855">[DOI]</a>.</p></li>
<li id="id6376"><p>Signe Tang Karlsen, Martin Holm Rau, Benjamín J. Sánchez, Kristian Jensen, and Ahmad A. Zeidan. From genotype to phenotype: computational approaches for inferring microbial traits relevant to the food industry. 47(4):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsre/fuad030">[DOI]</a>.</p></li>
<li id="id6377"><p>Sílvia Àvila-Cabré, Míriam Pérez‐Trujillo, Joan Albiol, and Pau Ferrer. Engineering the synthetic β-alanine pathway in komagataella phaffii for conversion of methanol into 3-hydroxypropionic acid. 22(1):237–237, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-023-02241-9">[DOI]</a>.</p></li>
<li id="id6378"><p>Peishun Li, Stefan Roos, Hao Luo, Boyang Ji, and Jens Nielsen. Metabolic engineering of human gut microbiome: recent developments and future perspectives. 79():1–13, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.06.006">[DOI]</a>.</p></li>
<li id="id6381"><p>Elena Lucy Carter, Chrystala Constantinidou, and Mohammad Tauqeer Alam. Applications of genome-scale metabolic models to investigate microbial metabolic adaptations in response to genetic or environmental perturbations. 25(1):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbad439">[DOI]</a>.</p></li>
<li id="id6569"><p>Baljeet Singh Saharan, Twinkle Chaudhary, Balwan Singh Mandal, Dharmender Kumar, Ravinder Kumar, Pardeep Kumar Sadh, and Joginder Singh Duhan. Microbe-plant interactions targeting metal stress: new dimensions for bioremediation applications. 13(2):252–269, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jox13020019">[DOI]</a>.</p></li>
<li id="id6570"><p>Zohreh Mirveis, Orla Howe, Paul A. Cahill, Nitin Patil, and Hugh J. Byrne. Monitoring and modelling the glutamine metabolic pathway: a review and future perspectives. 19(8):67–67, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11306-023-02031-9">[DOI]</a>.</p></li>
<li id="id6571"><p>Sizhe Qiu, Aidong Yang, and Hong Zeng. Flux balance analysis-based metabolic modeling of microbial secondary metabolism: current status and outlook. 19(8):e1011391–e1011391, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011391">[DOI]</a>.</p></li>
<li id="id6576"><p>M. Muthulakshmi, Aparajitha Srinivasan, and Smita Srivastava. Antioxidant green factories: toward sustainable production of vitamin e in plant&lt;i&gt;in vitro&lt;/i&gt;cultures. 8(4):3586–3605, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acsomega.2c05819">[DOI]</a>.</p></li>
<li id="id6578"><p>Zhongcai Li, Qian Liu, Jiahui Sun, Jianjian Sun, Mingjie Li, Yun Zhang, Aihua Deng, Shuwen Liu, and Tingyi Wen. Multivariate modular metabolic engineering for enhanced l-methionine biosynthesis in escherichia coli. 16(1):101–101, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-023-02347-7">[DOI]</a>.</p></li>
<li id="id6582"><p>Miao Guo, Chunfei Wu, Stephen P. Chapman, Xi Yu, Tom Vinestock, Astley Hastings, Pete Smith, and Nilay Shah. Advances in biorenewables-resource-waste systems and modelling. 9():100142–100142, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ccst.2023.100142">[DOI]</a>.</p></li>
<li id="id6583"><p>Dae‐Hee Lee, Haseong Kim, Bong Hyun Sung, Byung‐Kwan Cho, and Seung‐Goo Lee. Biofoundries: bridging automation and biomanufacturing in synthetic biology. 28(6):892–904, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-023-0226-x">[DOI]</a>.</p></li>
<li id="id6586"><p>Enes Fahri Tezcan, Yigit Demirtas, Zeynep Petek Çakar, and Kutlu Ö. Ülgen. Comprehensive genome-scale metabolic model of the human pathogen cryptococcus neoformans: a platform for understanding pathogen metabolism and identifying new drug targets. 3():1121409–1121409, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbinf.2023.1121409">[DOI]</a>.</p></li>
<li id="id6587"><p>Won Jun Kim, Youngjoon Lee, Hyun Uk Kim, Jae Yong Ryu, Jung Eun Yang, and Sang Yup Lee. Genome-wide identification of overexpression and downregulation gene targets based on the sum of covariances of the outgoing reaction fluxes. 14(11):990–1001.e5, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2023.10.005">[DOI]</a>.</p></li>
<li id="id6685"><p>Jingyi Qian, Yuzhou Wang, Xiner Liu, Zijian Hu, Nan Xu, Yuetong Wang, Tian‐Qiong Shi, and Chao Ye. Improving acetoin production through construction of a genome-scale metabolic model. 158():106833–106833, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2023.106833">[DOI]</a>.</p></li>
<li id="id6687"><p>Apostolos Chalkis, Vissarion Fisikopoulos, Marios Papachristou, and Elias Tsigaridas. Truncated log-concave sampling for convex bodies with reflective hamiltonian monte carlo. 49(2):1–25, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/3589505">[DOI]</a>.</p></li>
<li id="id6688"><p>Taghreed Elkasaby, Dao Duy Hanh, Hideo Kawaguchi, Masakazu Toyoshima, Akihiko Kondo, and Chiaki Ogino. Co-utilization of maltose and sodium acetate via engineered corynebacterium glutamicum for improved itaconic acid production. 28(5):790–803, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-023-0091-7">[DOI]</a>.</p></li>
<li id="id6783"><p>Ehsan Motamedian, Kristaps Berzins, Reinis Muiznieks, and Egils Stalidzāns. Optenvelope: a target point guided method for growth-coupled production using knockouts. 18(11):e0294313–e0294313, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0294313">[DOI]</a>.</p></li>
<li id="id6786"><p>Ehsan Motamedian, Kristaps Berzins, Reinis Muiznieks, and Egils Stalidzāns. Optenvelope: a target point guided method for growth-coupled production using knockouts. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.10.532079">[DOI]</a>.</p></li>
<li id="id6836"><p>Marc Griesemer and Ali Navid. Uses of multi-objective flux analysis for optimization of microbial production of secondary metabolites. 11(9):2149–2149, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms11092149">[DOI]</a>.</p></li>
<li id="id6861"><p>Michiel Busschaert, Florence H. Vermeire, and Steffen Waldherr. Modeling cell size distribution with heterogeneous flux balance analysis. 7():1903–1908, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/lcsys.2023.3282699">[DOI]</a>.</p></li>
<li id="id6867"><p>Carmen Bedia, Núria Dalmau, Lars K. Nielsen, Romà Tauler, and Igor Marín de Mas. A multi-level systems biology analysis of aldrin’s metabolic effects on prostate cancer cells. 11(2):11–11, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/proteomes11020011">[DOI]</a>.</p></li>
<li id="id6868"><p>Márta Balázs, Hunor Bartos, Szabolcs Lányi, Zsolt Bodor, and Ildikó Miklóssy. Substrate type and co2 addition significantly influence succinic acid production of basfia succiniciproducens. 45(9):1133–1145, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-023-03406-7">[DOI]</a>.</p></li>
<li id="id7005"><p>Livnat Jerby, Lior Wolf, Carsten Denkert, Gideon Y. Stein, Mika Hilvo, Matej Orešič, Tamar Geiger, and Eytan Ruppin. Supplementary material and methods and figure legends from metabolic associations of reduced proliferation and oxidative stress in advanced breast cancer. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1158/0008-5472.22395270.v1">[DOI]</a>.</p></li>
<li id="id7006"><p>Livnat Jerby, Lior Wolf, Carsten Denkert, Gideon Y. Stein, Mika Hilvo, Matej Orešič, Tamar Geiger, and Eytan Ruppin. Supplementary material and methods and figure legends from metabolic associations of reduced proliferation and oxidative stress in advanced breast cancer. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1158/0008-5472.22395270">[DOI]</a>.</p></li>
<li id="id7007"><p>Sílvia Àvila-Cabré, Míriam Pérez‐Trujillo, Joan Albiol, and Pau Ferrer. Engineering the synthetic β-alanine pathway in pichia pastoris for conversion of methanol into 3-hydroxypropionic acid. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3222058/v1">[DOI]</a>.</p></li>
<li id="id7008"><p>Marina Maurizio, María Masid, Kerry Woods, Reto Caldelari, John G. Doench, Arunasalam Naguleswaran, Denis Alain Henri Lucien Joly, Martín González Fernández, Jonas Zemp, Mélanie Borteele, Vassily Hatzimanikatis, Volker Heussler, Sven Rottenberg, and Philipp Olias. Host metabolic pathways essential for malaria and related hemoparasites in the infection of nucleated cells. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.09.27.559824">[DOI]</a>.</p></li>
<li id="id7009"><p>Cristina Herencias, Virginia Rivero‐Buceta, Sergio Salgado, Fernando Baquero, Rosa del Campo, Juan Nogales, and M. Auxiliadora Prieto. &lt;i&gt;bdellovibrio’s&lt;/i&gt;prey-independent growth is fuelled by amino acids as a carbon source. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.11.24.568592">[DOI]</a>.</p></li>
<li id="id7010"><p>Márcia R. Couto, Joana L. Rodrigues, Adelaide Braga, Óscar Dias, and L. R. Rodrigues. Optimization of chondroitin production in&lt;i&gt;e. coli&lt;/i&gt;using genome scale models. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.22.573033">[DOI]</a>.</p></li>
<li id="id7237"><p>Zhitao Mao, Qianqian Yuan, Haoran Li, Yue Zhang, Yuanyuan Huang, Chunhe Yang, Ruoyu Wang, Yongfu Yang, Yalun Wu, Shihui Yang, Xiaoping Liao, and Hongwu Ma. Cave: a cloud-based platform for analysis and visualization of metabolic pathways. 51(W1):W70–W77, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkad360">[DOI]</a>.</p></li>
<li id="id7242"><p>Bethany L Holland, Megan L. Matthews, Pedro Bota, Lee Sweetlove, Stephen P. Long, and George C. diCenzo. A genome‐scale metabolic reconstruction of soybean and &lt;i&gt;bradyrhizobium diazoefficiens&lt;/i&gt; reveals the cost–benefit of nitrogen fixation. 240(2):744–756, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/nph.19203">[DOI]</a>.</p></li>
<li id="id7522"><p>Charlotte Ramon and Jörg Stelling. Functional comparison of metabolic networks across species. 14(1):1699–1699, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-37429-5">[DOI]</a>.</p></li>
<li id="id7525"><p>Müberra Fatma Cesur, Arianna Basile, Kiran Raosaheb Patil, and Tunahan Çakır. A new metabolic model of&lt;i&gt;drosophila melanogaster&lt;/i&gt;and the integrative analysis of parkinson’s disease. 6(8):e202201695–e202201695, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.26508/lsa.202201695">[DOI]</a>.</p></li>
<li id="id7527"><p>Maximilian Luginsland, Cleo Kontoravdi, Andrew J. Racher, Colin Jaques, and Alexandros Kiparissides. Elucidating lactate metabolism in industrial cho cultures through the combined use of flux balance and principal component analyses. 202():109184–109184, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2023.109184">[DOI]</a>.</p></li>
<li id="id7678"><p>María Santos‐Merino, A. Gargantilla Becerra, Fernando Cruz, and Juan Nogales. Highlighting the potential of synechococcus elongatus pcc 7942 as platform to produce α-linolenic acid through an updated genome-scale metabolic modeling. 14():1126030–1126030, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2023.1126030">[DOI]</a>.</p></li>
<li id="id7679"><p>Sandra Correa Córdoba, Hao Tong, Asdrúbal Burgos, Feng Zhu, Saleh Alseekh, Alisdair R. Fernie, and Zoran Nikoloski. Identification of gene function based on models capturing natural variability of arabidopsis thaliana lipid metabolism. 14(1):4897–4897, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-40644-9">[DOI]</a>.</p></li>
<li id="id7680"><p>Emanuel Cunha, Davide Lagoa, José P. Faria, Filipe Liu, Christopher S. Henry, and Óscar Dias. &lt;i&gt;transyt&lt;/i&gt;, an innovative framework for identifying transport systems. 39(8):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btad466">[DOI]</a>.</p></li>
<li id="id7689"><p>Su-Kyung Kim, Minouk Lee, Yi Qing Lee, Hyun Jun Lee, Mina Rho, Yunkwan Kim, Jung Yeon Seo, Sung Hun Youn, Seung Jin Hwang, Nae Gyu Kang, Choong-Hwan Lee, Seo‐Young Park, and Dong‐Yup Lee. Genome-scale metabolic modeling and in silico analysis of opportunistic skin pathogen cutibacterium acnes. 13():1099314–1099314, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcimb.2023.1099314">[DOI]</a>.</p></li>
<li id="id7690"><p>Yoon-Mi Choi, Dong‐Hyuk Choi, Yi Qing Lee, Lokanand Koduru, Nathan E. Lewis, Meiyappan Lakshmanan, and Dong‐Yup Lee. Mitigating biomass composition uncertainties in flux balance analysis using ensemble representations. 21():3736–3745, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2023.07.025">[DOI]</a>.</p></li>
<li id="id7691"><p>Behnaz Nowrouzi, Pablo Torres-Montero, Eduard J. Kerkhoven, José L. Martínez, and Leonardo Rios‐Solis. Rewiring saccharomyces cerevisiae metabolism for optimised taxol® precursors production. 18():e00229–e00229, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2023.e00229">[DOI]</a>.</p></li>
<li id="id7703"><p>Albert Fina, Pierre Millard, Joan Albiol, Pau Ferrer, and Stéphanie Heux. High throughput 13c-metabolic flux analysis of 3-hydroxypropionic acid producing pichia pastoris reveals limited availability of acetyl-coa and atp due to tight control of the glycolytic flux. 22(1):117–117, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-023-02123-0">[DOI]</a>.</p></li>
<li id="id7707"><p>Anna T. Germann, Andreas Nakielski, Maximilian Dietsch, Tim Petzel, Daniel Moser, Sebastian Triesch, Philipp Westhoff, and Ilka M. Axmann. A systematic overexpression approach reveals native targets to increase squalene production in synechocystis sp. pcc 6803. 14():1024981–1024981, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2023.1024981">[DOI]</a>.</p></li>
<li id="id7710"><p>Andrea Angarita-Rodríguez, Yeimy González‐Giraldo, Juan J. Rubio-Mesa, Andrés Felipe Aristizábal-Pachón, Andrés Pinzón, and Janneth González. Control theory and systems biology: potential applications in neurodegeneration and search for therapeutic targets. 25(1):365–365, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms25010365">[DOI]</a>.</p></li>
<li id="id7813"><p>Kenta Miyoshi, Ryutaro Kawai, Teppei Niide, Yoshihiro Toya, and Hiroshi Shimizu. Functional evaluation of non-oxidative glycolysis in escherichia coli in the stationary phase under microaerobic conditions. 135(4):291–297, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiosc.2023.01.002">[DOI]</a>.</p></li>
<li id="id7814"><p>Diego Tec-Campos, Camila Posadas, Juan D. Tibocha‐Bonilla, Deepan Thiruppathy, Nathan Glonek, Cristal Zúñiga, Alejandro Zepeda, and Karsten Zengler. The genome-scale metabolic model for the purple non-sulfur bacterium rhodopseudomonas palustris bis a53 accurately predicts phenotypes under chemoheterotrophic, chemoautotrophic, photoheterotrophic, and photoautotrophic growth conditions. 19(8):e1011371–e1011371, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011371">[DOI]</a>.</p></li>
<li id="id7832"><p>Neema Jamshidi, Kabir Nigam, and Sanjay K. Nigám. Loss of the kidney urate transporter, urat1, leads to disrupted redox homeostasis in mice. 12(3):780–780, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/antiox12030780">[DOI]</a>.</p></li>
<li id="id7842"><p>Amirhossein Shahbazinia, Saber Salehkaleybar, and Matin Hashemi. Paralingam: parallel causal structure learning for linear non-gaussian acyclic models. 176():114–127, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jpdc.2023.01.007">[DOI]</a>.</p></li>
<li id="id7843"><p>Sarayu Murali, Maziya Ibrahim, Hemalatha Rajendran, Shagun Shagun, Shyam Kumar Masakapalli, Karthik Raman, and Smita Srivastava. Genome-scale metabolic model led engineering of nothapodytes nimmoniana plant cells for high camptothecin production. 14():1207218–1207218, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2023.1207218">[DOI]</a>.</p></li>
<li id="id7853"><p>Mohammadreza Yasemi and Mario Jolicœur. A genome-scale dynamic constraint-based modelling (gdcbm) framework predicts growth dynamics, medium composition and intracellular flux distributions in cho clonal variations. 78():209–222, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.06.005">[DOI]</a>.</p></li>
<li id="id7911"><p>Behnaz Nowrouzi, Pablo Torres-Montero, Eduard J. Kerkhoven, José L. Martínez, and Leonardo Rios‐Solis. Rewiring&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;metabolism for optimised taxol® precursors production. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.06.03.543533">[DOI]</a>.</p></li>
<li id="id7923"><p>Bruno G. Galuzzi, Luca Milazzo, and Chiara Damiani. Best practices in flux sampling of constrained-based models. ():234–248, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-031-25891-6_18">[DOI]</a>.</p></li>
<li id="id7924"><p>Elías Vera-Sigüenza, Cristina Escribano-Gonzalez, Irene Serrano‐Gonzalo, Kattri‐Liis Eskla, Fabian Spill, and Daniel A. Tennant. Mathematical reconstruction of the metabolic network in an in-vitro multiple myeloma model. 19(9):e1011374–e1011374, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011374">[DOI]</a>.</p></li>
<li id="id7925"><p>Mirco Plante. Epistemology of synthetic biology: a new theoretical framework based on its potential objects and objectives. 11():1266298–1266298, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2023.1266298">[DOI]</a>.</p></li>
<li id="id7943"><p>Nela Pragathi Sneha, S. Akila Parvathy Dharshini, Y‐h. Taguchi, and M. Michael Gromiha. Investigating neuron degeneration in huntington’s disease using rna-seq based transcriptome study. 14(9):1801–1801, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/genes14091801">[DOI]</a>.</p></li>
<li id="id7944"><p>Jean‐Pierre Mazat. The metabolic control theory: its development and its application to mitochondrial oxidative phosphorylation. 234():105038–105038, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2023.105038">[DOI]</a>.</p></li>
<li id="id7945"><p>Qianqian Yuan, Fan Wei, Xiaogui Deng, Aonan Li, Zhenkun Shi, Zhitao Mao, Feiran Li, and Hongwu Ma. Reconstruction and metabolic profiling of the genome-scale metabolic network model of pseudomonas stutzeri a1501. 8(4):688–696, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.synbio.2023.10.001">[DOI]</a>.</p></li>
<li id="id7952"><p>Bruno G. Galuzzi and Chiara Damiani. An efficient implementation of flux variability analysis for metabolic networks. ():58–69, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-031-31183-3_5">[DOI]</a>.</p></li>
<li id="id7960"><p>Hananeh Ahmadpanah, Ehsan Motamedian, and Mohammad Mahdi Mardanpour. Metabolic regulation boosts bioelectricity generation in zymomonas mobilis microbial fuel cell, surpassing ethanol production. 13(1):20673–20673, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-023-47846-7">[DOI]</a>.</p></li>
<li id="id8052"><p>Clémence Frioux, Sylvie Huet, Simon Labarthe, Julien Martinelli, Thibault Malou, David H. Sherman, Marie‐Luce Taupin, and Pablo Ugalde‐Salas. Accelerating metabolic models evaluation with statistical metamodels: application to &lt;i&gt;salmonella&lt;/i&gt; infection models. 73():187–217, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1051/proc/202373187">[DOI]</a>.</p></li>
<li id="id8221"><p>Josh A. Taylor, Alain Rapaport, and Denis Dochain. Convex representation of metabolic networks with michaelis-menten kinetics. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.01.17.524421">[DOI]</a>.</p></li>
<li id="id8223"><p>Albert Fina, Pierre Millard, Joan Albiol, Pau Ferrer, and Stéphanie Heux. High throughput 13c-metabolic flux analysis of 3-hydroxypropionic acid producing pichia pastoris reveals limited availability of acetyl-coa and atp due to tight control of the glycolytic flux. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-2678707/v1">[DOI]</a>.</p></li>
<li id="id8224"><p>Jacob J. Valenzuela, Selva Rupa Christinal Immanuel, James G. Wilson, Serdar Turkarslan, Maryann Ruiz, Sean M. Gibbons, Kristopher A. Hunt, Manfred Auer, Marcin Zemla, David A. Stahl, and Nitin S. Baliga. Origin of biogeographically distinct ecotypes during laboratory evolution. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.19.541524">[DOI]</a>.</p></li>
<li id="id8225"><p>Chenguang Liang, Sue Murray, Yang Li, Richard Lee, Audrey Low, Shruti Sasaki, Austin W.T. Chiang, Wen-Jen Lin, Joel Mathews, Will Barnes, and Nathan E. Lewis. Lipidsim: inferring mechanistic lipid biosynthesis perturbations from lipidomics with a flexible, low-parameter, systematic markov modeling framework. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.07.26.550768">[DOI]</a>.</p></li>
<li id="id8226"><p>Clémence FRIOUX and Anne SIEGEL. Combinatorial optimization problems for studying metabolism. ():45–88, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781394229086.ch2">[DOI]</a>.</p></li>
<li id="id8227"><p>Fatma Ece Altınışık Kaya and Fatma Gizem Avcı. Modeling the microbial cells for biotechnological applications. ():121–151, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.4018/978-1-6684-6577-6.ch006">[DOI]</a>.</p></li>
<li id="id8228"><p>Anita L. Ziegler, L. Ullmann, Manuel Boßmann, Karla Stein, Ulf W. Liebal, Alexander Mitsos, and Lars M. Blank. Itaconic acid production by co-feeding of&lt;i&gt;ustilago maydis&lt;/i&gt;: a combined approach of experimental data, design of experiments and metabolic modeling. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.13.562154">[DOI]</a>.</p></li>
<li id="id8229"><p>Aparajitha Srinivasan, Kevin Chen-Xiao, Deepanwita Banerjee, Asun Oka, Venkataramana R. Pidatala, Aymerick Eudes, Blake A. Simmons, Thomas Eng, and Aindrila Mukhopadhyay. Sustainable production of 2,3,5,6-tetramethylpyrazine at high titer in engineered&lt;i&gt;corynebacterium glutamicum&lt;/i&gt;using statistical design of experiments. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.20.563186">[DOI]</a>.</p></li>
<li id="id8230"><p>Guilherme de Oliveira Mendes Ohira. Desenvolvimento de ferramentas e algoritmos em computational strain optmization models. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.11606/d.3.2023.tde-29112023-143720">[DOI]</a>.</p></li>
<li id="id8272"><p>Almut Heinken, Johannes Hertel, Geeta Acharya, Dmitry A. Ravcheev, Malgorzata Nyga, Onyedika Emmanuel Okpala, Marcus Hogan, Stefanía Magnúsdóttir, Filippo Martinelli Boneschi, Bram Nap, German Preciat, Janaka N. Edirisinghe, Christopher S. Henry, Ronan M. T. Fleming, and Ines Thiele. Genome-scale metabolic reconstruction of 7,302 human microorganisms for personalized medicine. 41(9):1320–1331, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41587-022-01628-0">[DOI]</a>.</p></li>
<li id="id8288"><p>Martin Schäfer, Alan R. Pacheco, Rahel Künzler, Miriam Bortfeld‐Miller, Christopher M. Field, Evangelia Vayena, Vassily Hatzimanikatis, and Julia A. Vorholt. Metabolic interaction models recapitulate leaf microbiota ecology. 381(6653):eadf5121–eadf5121, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/science.adf5121">[DOI]</a>.</p></li>
<li id="id8307"><p>Roland Hellinger, Arnar Sigurdsson, Wenxin Wu, Elena V. Romanova, Lingjun Li, Jonathan V. Sweedler, Roderich D. Süßmuth, and Christian W. Gruber. Peptidomics. 3(1):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s43586-023-00205-2">[DOI]</a>.</p></li>
<li id="id8309"><p>Emil Karlsen, Marianne Gylseth, Christian Schulz, and Eivind Almaas. A study of a diauxic growth experiment using an expanded dynamic flux balance framework. 18(1):e0280077–e0280077, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0280077">[DOI]</a>.</p></li>
<li id="id8333"><p>William F. Kindschuh, Federico Baldini, Martin C. Liu, Jingqiu Liao, Yoli Meydan, Harry H. Lee, Almut Heinken, Ines Thiele, Christoph A. Thaiss, Maayan Levy, and Tal Korem. Preterm birth is associated with xenobiotics and predicted by the vaginal metabolome. 8(2):246–259, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-022-01293-8">[DOI]</a>.</p></li>
<li id="id8337"><p>Xia Wang, Ying Teng, Xiaomi Wang, Yongfeng Xu, Ran Li, Yi Sun, Shixiang Dai, Wenbo Hu, Hongzhe Wang, Yanning Li, Yan Fang, and Yongming Luo. Nitrogen transfer and cross-feeding between &lt;i&gt;azotobacter chroococcum&lt;/i&gt; and &lt;i&gt;paracoccus aminovorans&lt;/i&gt; promotes pyrene degradation. 17(12):2169–2181, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41396-023-01522-w">[DOI]</a>.</p></li>
<li id="id8356"><p>Clara Correia‐Melo, Stephan Kamrad, Roland Tengölics, Christoph B. Messner, Pauline Trébulle, StJohn Townsend, Sreejith J. Varma, Anja Freiwald, Benjamin M. Heineike, Kate Campbell, Lucía Herrera-Domínguez, Simran Kaur Aulakh, Łukasz Szyrwiel, Jason Yu, Aleksej Zelezniak, Vadim Demichev, Michael Mülleder, Balázs Papp, Mohammad Tauqeer Alam, and Markus Ralser. Cell-cell metabolite exchange creates a pro-survival metabolic environment that extends lifespan. 186(1):63–79.e21, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2022.12.007">[DOI]</a>.</p></li>
<li id="id8362"><p>Francesc Puig‐Castellví, Romina Pacheco‐Tapia, Maxime Deslande, Manyi Jia, Petros Andrikopoulos, Kanta Chechi, Amélie Bonnefond, Philippe Froguel, and Marc‐Emmanuel Dumas. Advances in the integration of metabolomics and metagenomics for human gut microbiome and their clinical applications. 167():117248–117248, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.trac.2023.117248">[DOI]</a>.</p></li>
<li id="id8363"><p>Ning Qin, Lingyun Li, Xu Ji, Rui Pereira, Yu Chen, Shile Yin, Chaokun Li, Xiaozhen Wan, Danye Qiu, Junfeng Jiang, Hao Luo, Yueping Zhang, Genlai Dong, Yiming Zhang, Shuobo Shi, Henning J. Jessen, Jianye Xia, Yun Chen, Christer Larsson, Tianwei Tan, Zihe Liu, and Jens Nielsen. Flux regulation through glycolysis and respiration is balanced by inositol pyrophosphates in yeast. 186(4):748–763.e15, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2023.01.014">[DOI]</a>.</p></li>
<li id="id8366"><p>Miguel Ponce-de-León, Arnau Montagud, Vincent Noël, Annika Meert, Gerard Pradas, Emmanuel Barillot, Laurence Calzone, and Alfonso Valencia. Physiboss 2.0: a sustainable integration of stochastic boolean and agent-based modelling frameworks. 9(1):54–54, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-023-00314-4">[DOI]</a>.</p></li>
<li id="id8370"><p>William T. Scott, Sara Benito-Vaquerizo, Johannes Zimmermann, Djordje Bajić, Almut Heinken, María Suárez‐Diez, and Peter J. Schaap. A structured evaluation of genome-scale constraint-based modeling tools for microbial consortia. 19(8):e1011363–e1011363, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011363">[DOI]</a>.</p></li>
<li id="id8371"><p>Yiming Zhang, Mo Su, Yu Chen, Zheng Wang, Jens Nielsen, and Zihe Liu. Engineering yeast mitochondrial metabolism for 3-hydroxypropionate production. 16(1):64–64, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-023-02309-z">[DOI]</a>.</p></li>
<li id="id8372"><p>Bouchra Ezzamouri, Dorinês Rosário, Gholamreza Bidkhori, Sunjae Lee, Mathias Uhlén, and Saeed Shoaie. Metabolic modelling of the human gut microbiome in type 2 diabetes patients in response to metformin treatment. 9(1):2–2, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-022-00261-6">[DOI]</a>.</p></li>
<li id="id8374"><p>Yun‐Peng Zhao, Yiming Feng, Jianhang Zhou, Kuo Zhang, Jingqi Sun, Lina Wang, and Sitong Liu. Potential bacterial isolation by dosing metabolites in cross-feedings. 231():119589–119589, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.watres.2023.119589">[DOI]</a>.</p></li>
<li id="id8375"><p>Sandra Gomez Romero and Nanette Boyle. Systems biology and metabolic modeling for cultivated meat: a promising approach for cell culture media optimization and cost reduction. 22(4):3422–3443, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1541-4337.13193">[DOI]</a>.</p></li>
<li id="id8378"><p>Thomas Eng, Deepanwita Banerjee, Javier Menasalvas, Yan Chen, Jennifer Gin, Hemant Choudhary, Edward E. K. Baidoo, Jian-Hua Chen, Axel Ekman, Ramu Kakumanu, Yuzhong Liu, Alex Codik, Carolyn A. Larabell, John M. Gladden, Blake A. Simmons, Jay D. Keasling, Christopher J. Petzold, and Aindrila Mukhopadhyay. Maximizing microbial bioproduction from sustainable carbon sources using iterative systems engineering. 42(9):113087–113087, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2023.113087">[DOI]</a>.</p></li>
<li id="id8380"><p>Arianna Basile, Almut Heinken, Johannes Hertel, Larry Smarr, Weizhong Li, Laura Treu, Giorgio Valle, Stefano Campanaro, and Ines Thiele. Longitudinal flux balance analyses of a patient with episodic colonic inflammation reveals microbiome metabolic dynamics. 15(1):2226921–2226921, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2023.2226921">[DOI]</a>.</p></li>
<li id="id8385"><p>Laura Judith Marcos-Zambrano, Víctor Manuel López-Molina, Burcu Bakır-Güngör, Marcus Frohme, Kanita Karađuzović-Hadžiabdić, Thomas Klammsteiner, Eliana Ibrahimi, Leo Lahti, Tatjana Lončar-Turukalo, Xhilda Dhamo, Andrea Simeon, Alina Nechyporenko, Gianvito Pio, Piotr Przymus, Alexia Sampri, Vladimir Trajkovik, Blanca Lacruz-Pleguezuelos, Oliver Aasmets, Ricardo Araújo, Ioannis Anagnostopoulos, Önder Aydemir, Magali Berland, M. Luz Calle, Michelangelo Ceci, Hatice Duman, Aycan Gündoğdu, Aki S. Havulinna, Kardokh Hama Najib Kaka Bra, Eglantina Kalluçi, Sercan Karav, Daniel Lode, Marta B. Lopes, Patrick May, Bram Nap, Miroslava Nedyalkova, Inês Paciência, Lejla Pašić, Meritxell Pujolassos, Rajesh Shigdel, Antonio Susín, Ines Thiele, Ciprian‐Octavian Truică, Paul Wilmes, Ercüment Yılmaz, Malik Yousef, Marcus J. Claesson, Jaak Truu, and Enrique Carrillo de Santa Pau. A toolbox of machine learning software to support microbiome analysis. 14():1250806–1250806, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2023.1250806">[DOI]</a>.</p></li>
<li id="id8386"><p>Le Gao, Jiao Meng, Wuling Dai, Zhaokun Zhang, Haofan Dong, Qianqian Yuan, Wuyuan Zhang, Shuguang Liu, and Xin Wu. Deciphering cell wall sensors enabling the construction of robust p. pastoris for single-cell protein production. 16(1):178–178, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-023-02428-7">[DOI]</a>.</p></li>
<li id="id8387"><p>Isabel Rosety, Alise Žagare, Cláudia Saraiva, Sarah Nickels, Paul Antony, Catarina Serra-Almeida, Enrico Glaab, Rashi Halder, Sergiy Velychko, Thomas Rauen, Hans R. Schöler, Silvia Bolognin, Thomas Sauter, Javier Jarazo, Rejko Krüger, and Jens C. Schwamborn. Impaired neuron differentiation in gba-associated parkinson’s disease is linked to cell cycle defects in organoids. 9(1):166–166, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41531-023-00616-8">[DOI]</a>.</p></li>
<li id="id8440"><p>Cecilia Noecker, Juan M. Sánchez, Jordan E. Bisanz, Veronica Escalante, Margaret Alexander, Kai Trepka, Almut Heinken, Yuanyuan Liu, Dylan Dodd, Ines Thiele, Brian C. DeFelice, and Peter J. Turnbaugh. Systems biology elucidates the distinctive metabolic niche filled by the human gut microbe eggerthella lenta. 21(5):e3002125–e3002125, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pbio.3002125">[DOI]</a>.</p></li>
<li id="id8480"><p>Philipp Wendering, Marius Arend, Zahra Razaghi‐Moghadam, and Zoran Nikoloski. Data integration across conditions improves turnover number estimates and metabolic predictions. 14(1):1485–1485, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-37151-2">[DOI]</a>.</p></li>
<li id="id8485"><p>Mona Synnøve Bjune, Laurence Lawrence‐Archer, Johnny Laupsa‐Borge, Cathrine Horn Sommersten, Adrian McCann, Robert C. Glastad, Iain G. Johnston, Matthias Kern, Matthias Blüher, Gunnar Mellgren, and Simon N. Dankel. Metabolic role of the hepatic valine/3-hydroxyisobutyrate (3-hib) pathway in fatty liver disease. 91():104569–104569, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ebiom.2023.104569">[DOI]</a>.</p></li>
<li id="id8486"><p>Sourav Chowdhury, Daniel C. Zielinski, Christopher Dalldorf, João V. Rodrigues, Bernhard Ø. Palsson, and Eugene I. Shakhnovich. Empowering drug off-target discovery with metabolic and structural analysis. 14(1):3390–3390, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-38859-x">[DOI]</a>.</p></li>
<li id="id8487"><p>Alise Žagare, German Preciat, Sarah Nickels, Xi Luo, Anna S. Monzel, Gemma Gomez‐Giro, Graham Robertson, Christian Jaeger, Jafar Sharif, Haruhiko Koseki, Nico J. Diederich, Enrico Glaab, Ronan M. T. Fleming, and Jens C. Schwamborn. Omics data integration suggests a potential idiopathic parkinson’s disease signature. 6(1):1179–1179, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42003-023-05548-w">[DOI]</a>.</p></li>
<li id="id8488"><p>Koray Malcı, Rodrigo Santibáñez, Nestor Jonguitud‐Borrego, Jorge H. Santoyo‐Garcia, Eduard J. Kerkhoven, and Leonardo Rios‐Solis. Improved production of taxol® precursors in s. cerevisiae using combinatorial in silico design and metabolic engineering. 22(1):243–243, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-023-02251-7">[DOI]</a>.</p></li>
<li id="id8494"><p>Marius Arend, David Zimmer, Rudan Xu, Frederik Sommer, Timo Mühlhaus, and Zoran Nikoloski. Proteomics and constraint-based modelling reveal enzyme kinetic properties of chlamydomonas reinhardtii on a genome scale. 14(1):4781–4781, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-40498-1">[DOI]</a>.</p></li>
<li id="id8499"><p>Miha Moškon and Tadeja Režen. Context-specific genome-scale metabolic modelling and its application to the analysis of covid-19 metabolic signatures. 13(1):126–126, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo13010126">[DOI]</a>.</p></li>
<li id="id8500"><p>William T. Scott, David Henriques, Eddy J. Smid, Richard A. Notebaart, and Eva Balsa‐Canto. Dynamic genome‐scale modeling of &lt;i&gt;saccharomyces cerevisiae&lt;/i&gt; unravels mechanisms for ester formation during alcoholic fermentation. 120(7):1998–2012, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28421">[DOI]</a>.</p></li>
<li id="id8505"><p>Guido Zampieri, Georgios Efthimiou, and Claudio Angione. Multi-dimensional experimental and computational exploration of metabolism pinpoints complex probiotic interactions. 76():120–132, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.01.008">[DOI]</a>.</p></li>
<li id="id8510"><p>Philip Fernandes, Yash Sharma, Fatima Zulqarnain, Brooklyn McGrew, Aman Shrivastava, Lubaina Ehsan, Dawson Payne, Lillian Dillard, Deborah Powers, Isabelle Aldridge, Jason D. Matthews, Subra Kugathasan, Facundo M. Fernández, David A. Gaul, Jason A. Papin, and Sana Syed. Identifying metabolic shifts in crohn's disease using 'omics-driven contextualized computational metabolic network models. 13(1):203–203, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-26816-5">[DOI]</a>.</p></li>
<li id="id8511"><p>Kengo Watanabe, Tomasz Wilmanski, Priyanka Baloni, Max Robinson, Gonzalo G. Garcia, Michael R. Hoopmann, Mukul K. Midha, David Baxter, Michal Maes, Seamus R. Morrone, Kelly M. Crebs, Charu Kapil, Ulrike Kusebauch, Jack Wiedrick, Jodi Lapidus, Lance Pflieger, Christopher Lausted, Jared C. Roach, Gustavo Glusman, Steven R. Cummings, Nicholas J. Schork, Nathan D. Price, Leroy Hood, Richard A. Miller, Robert L. Moritz, and Noa Rappaport. Lifespan-extending interventions induce consistent patterns of fatty acid oxidation in mouse livers. 6(1):768–768, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42003-023-05128-y">[DOI]</a>.</p></li>
<li id="id8512"><p>Zeqian Li, Ahmed Selim, and Seppe Kuehn. Statistical prediction of microbial metabolic traits from genomes. 19(12):e1011705–e1011705, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011705">[DOI]</a>.</p></li>
<li id="id8520"><p>Benjamin Strain, James Morrissey, Athanasios Antonakoudis, and Cleo Kontoravdi. How reliable are chinese hamster ovary (cho) cell genome‐scale metabolic models? 120(9):2460–2478, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28366">[DOI]</a>.</p></li>
<li id="id8521"><p>Jingwei Cai, Alexis Auster, Sungjoon Cho, and Zijuan Lai. Dissecting the human gut microbiome to better decipher drug liability: a once-forgotten organ takes center stage. 52():171–201, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jare.2023.07.002">[DOI]</a>.</p></li>
<li id="id8527"><p>Andrew Walakira, Cene Skubic, Nejc Nadižar, Damjana Rozman, Tadeja Režen, Miha Mraz, and Miha Moškon. Integrative computational modeling to unravel novel potential biomarkers in hepatocellular carcinoma. 159():106957–106957, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2023.106957">[DOI]</a>.</p></li>
<li id="id8528"><p>Ibrahim Fakih, Jeanne Got, Carlos Eduardo Robles-Rodríguez, Anne Siegel, Évelyne Forano, and Rafael Muñoz‐Tamayo. &lt;b&gt; dynamic genome-based metabolic modeling of the predominant cellulolytic rumen bacterium &lt;i&gt;fibrobacter succinogenes&lt;/i&gt; s85 &lt;/b&gt;. 8(3):e0102722–e0102722, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01027-22">[DOI]</a>.</p></li>
<li id="id8532"><p>Christopher Riccardi, Marzia Calvanese, Veronica Ghini, Tania Alonso-Vásquez, Elena Perrin, Paola Turano, Giorgio Giurato, Alessandro Weisz, Ermenegilda Parrilli, Maria Luisa Tutino, and Marco Fondi. Metabolic robustness to growth temperature of a cold- adapted marine bacterium. 8(2):e0112422–e0112422, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01124-22">[DOI]</a>.</p></li>
<li id="id8540"><p>Elif Esvap and Kutlu Ö. Ülgen. Neuroinflammation, energy and sphingolipid metabolism biomarkers are revealed by metabolic modeling of autistic brains. 11(2):583–583, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biomedicines11020583">[DOI]</a>.</p></li>
<li id="id8541"><p>Gonçalo dos Santos Correia, Julian R. Marchesi, and David A. MacIntyre. Moving beyond dna: towards functional analysis of the vaginal microbiome by non-sequencing-based methods. 73():102292–102292, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mib.2023.102292">[DOI]</a>.</p></li>
<li id="id8546"><p>William T. Scott, Sara Benito-Vaquerizo, Johannes Zimmerman, Djordje Bajić, Almut Heinken, María Suárez‐Diez, and Peter J. Schaap. A structured evaluation of genome-scale constraint-based modeling tools for microbial consortia. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.02.08.527721">[DOI]</a>.</p></li>
<li id="id8547"><p>Nantia Leonidou, Alina Renz, Reihaneh Mostolizadeh, and Andreas Dräger. New workflow predicts drug targets against sars-cov-2 via metabolic changes in infected cells. 19(3):e1010903–e1010903, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1010903">[DOI]</a>.</p></li>
<li id="id8548"><p>Joshua A. M. Kaste and Yair Shachar‐Hill. Model validation and selection in metabolic flux analysis and flux balance analysis. 40(1):e3413–e3413, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.3413">[DOI]</a>.</p></li>
<li id="id8549"><p>Qusheng Jin. Building microbial kinetic models for environmental application: a theoretical perspective. 158():105782–105782, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.apgeochem.2023.105782">[DOI]</a>.</p></li>
<li id="id8552"><p>Patrick E. Gelbach and Stacey D. Finley. Genome-scale modeling predicts metabolic differences between macrophage subtypes in colorectal cancer. 26(9):107569–107569, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2023.107569">[DOI]</a>.</p></li>
<li id="id8553"><p>Bo Wang, Cristal Zúñiga, Michael T. Guarnieri, Karsten Zengler, Michael J. Betenbaugh, and Jamey D. Young. Metabolic engineering of synechococcus elongatus 7942 for enhanced sucrose biosynthesis. 80():12–24, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.09.002">[DOI]</a>.</p></li>
<li id="id8554"><p>Johannes Hertel, Almut Heinken, Daniel Fässler, and Ines Thiele. Causal inference on microbiome-metabolome relations in observational host-microbiome data via in silico in vivo association pattern analyses. 3(10):100615–100615, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.crmeth.2023.100615">[DOI]</a>.</p></li>
<li id="id8555"><p>Aliya Lakhani, Da Hyun Kang, Yea Eun Kang, and Junyoung O. Park. Toward systems-level metabolic analysis in endocrine disorders and cancer. 38(6):619–630, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3803/enm.2023.1814">[DOI]</a>.</p></li>
<li id="id8561"><p>Fidias D. González Camargo, Mary Santamaría-Torres, Mónica P. Cala, Marcela Guevara‐Suarez, Silvia Restrepo, Andrea del Pilar Sánchez‐Camargo, Miguel Fernández‐Niño, María Corujo, Ada Carolina Gallo Molina, Javier Cifuentes, Julian A. Serna, Juan C. Cruz, Carolina Muñoz-Camargo, and Andrés Fernando González Barrios. Genome-scale metabolic reconstruction, non-targeted lc-qtof-ms based metabolomics data, and evaluation of anticancer activity of cannabis sativa leaf extracts. 13(7):788–788, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo13070788">[DOI]</a>.</p></li>
<li id="id8562"><p>Quinn S. Solfisburg, Federico Baldini, Brittany Baldwin-Hunter, George I. Austin, Harry H. Lee, Heekuk Park, Daniel E. Freedberg, Charles J. Lightdale, Tal Korem, and Julian A. Abrams. The salivary microbiome and predicted metabolite production are associated with barrett's esophagus and high-grade dysplasia or adenocarcinoma. 33(3):371–380, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1158/1055-9965.epi-23-0652">[DOI]</a>.</p></li>
<li id="id8570"><p>Hao Tong, Roosa A. E. Laitinen, and Zoran Nikoloski. Predicting plasticity of rosette growth and metabolic fluxes in &lt;i&gt;arabidopsis thaliana&lt;/i&gt;. 240(1):426–438, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/nph.19154">[DOI]</a>.</p></li>
<li id="id8571"><p>Almut Heinken, Timothy Otto Hulshof, Bram Nap, Filippo Martinelli Boneschi, Arianna Basile, Amy O’Brolchain, Neil Francis O’Sullivan, Celine Gallagher, E.L. Magee, Francesca McDonagh, Ian Lalor, Maeve Bergin, Phoebe Evans, Rachel Daly, Ronan Farrell, Rose Marie Delaney, Saoirse Hill, Saoirse Roisin McAuliffe, Trevor Kilgannon, Ronan M. T. Fleming, Cyrille C. Thinnes, and Ines Thiele. Apollo: a genome-scale metabolic reconstruction resource of 247,092 diverse human microbes spanning multiple continents, age groups, and body sites. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.02.560573">[DOI]</a>.</p></li>
<li id="id8572"><p>Weiren Liu, Huqiang Wang, Qianfu Zhao, Chenyang Tao, Wei‐Feng Qu, Yushan Hou, Run Huang, Zimei Sun, Gui‐Qi Zhu, Xi‐Fei Jiang, Yuan Fang, Jun Gao, Xiaoling Wu, Zhixiang Yang, Rongyu Ping, Jiafeng Chen, Rui Yang, Tianhao Chu, Jian Zhou, Jia Fan, Zheng Tang, Dong Yang, and Ying‐Hong Shi. Multiomics analysis reveals metabolic subtypes and identifies diacylglycerol kinase α (dgka) as a potential therapeutic target for intrahepatic cholangiocarcinoma. 44(2):226–250, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/cac2.12513">[DOI]</a>.</p></li>
<li id="id8579"><p>Matti Gralka. Searching for principles of microbial ecology across levels of biological organization. 63(6):1520–1531, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/icb/icad060">[DOI]</a>.</p></li>
<li id="id8580"><p>Yawen Li, Jingyi Qian, Jiacong Huang, Dong‐Sheng Guo, Zhi-Kui Nie, Chao Ye, and Tian‐Qiong Shi. Improving gibberellin ga&lt;sub&gt;3&lt;/sub&gt; production with the construction of a genome-scale metabolic model of &lt;i&gt;fusarium fujikuroi&lt;/i&gt;. 71(48):18890–18897, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.jafc.3c05309">[DOI]</a>.</p></li>
<li id="id8616"><p>Blas Blázquez, David San León, Antonia Rojas, Marta Tortajada, and Juan Nogales. New insights on metabolic features of bacillus subtilis based on multistrain genome-scale metabolic modeling. 24(8):7091–7091, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms24087091">[DOI]</a>.</p></li>
<li id="id8663"><p>Haoyang Yao, Sanjeev Dahal, and Laurence Yang. Novel context-specific genome-scale modelling explores the potential of triacylglycerol production by chlamydomonas reinhardtii. 22(1):13–13, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-022-02004-y">[DOI]</a>.</p></li>
<li id="id8664"><p>Yanhua Liu and Hans V. Westerhoff. Competitive, multi‐objective, and compartmented flux balance analysis for addressing tissue‐specific inborn errors of metabolism. 46(4):573–585, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/jimd.12603">[DOI]</a>.</p></li>
<li id="id8673"><p>Huan Du, Meng Li, and Yang Liu. Towards applications of genome‐scale metabolic model‐based approaches in designing synthetic microbial communities. 11(1):15–30, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.15302/j-qb-022-0313">[DOI]</a>.</p></li>
<li id="id8674"><p>GaRyoung Lee, Sang Mi Lee, and Hyun Uk Kim. A contribution of metabolic engineering to addressing medical problems: metabolic flux analysis. 77():283–293, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.04.008">[DOI]</a>.</p></li>
<li id="id8675"><p>Nattharat Punyasu, Saowalak Kalapanulak, and Treenut Saithong. Co2 recycling by phosphoenolpyruvate carboxylase enables cassava leaf metabolism to tolerate low water availability. 14():1159247–1159247, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2023.1159247">[DOI]</a>.</p></li>
<li id="id8676"><p>Naroa Barrena, Luis V. Valcárcel, Danel Olaverri-Mendizabal, Iñigo Apaolaza, and Francisco J. Planes. Synthetic lethality in large-scale integrated metabolic and regulatory network models of human cells. 9(1):32–32, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-023-00296-3">[DOI]</a>.</p></li>
<li id="id8677"><p>Agris Pentjuss, Emīls Bolmanis, Anastasija Šuleiko, Elina Didrihsone, Artūrs Šuleiko, Konstantīns Dubencovs, Jānis Liepiņš, Andris Kazāks, and Juris Vanags. Pichia pastoris growth—coupled heme biosynthesis analysis using metabolic modelling. 13(1):15816–15816, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-023-42865-w">[DOI]</a>.</p></li>
<li id="id8683"><p>Frederick Clasen, Patrícia M. Nunes, Gholamreza Bidkhori, Nourdine Bah, Stefan Boeing, Saeed Shoaie, and Dimitrios Anastasiou. Systematic diet composition swap in a mouse genome-scale metabolic model reveals determinants of obesogenic diet metabolism in liver cancer. 26(2):106040–106040, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2023.106040">[DOI]</a>.</p></li>
<li id="id8684"><p>Bronson R. Weston and Ines Thiele. A nutrition algorithm to optimize feed and medium composition using genome-scale metabolic models. 76():167–178, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.01.010">[DOI]</a>.</p></li>
<li id="id8685"><p>Eric Libby, Christopher P. Kempes, and Jordan G. Okie. Metabolic compatibility and the rarity of prokaryote endosymbioses. 120(17):e2206527120–e2206527120, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2206527120">[DOI]</a>.</p></li>
<li id="id8686"><p>Lvjing Wang, Xiaoyu Wang, Hao Wu, Haixia Wang, Yihan Wang, and Zhenmei Lü. Metabolic modeling of synthetic microbial communities for bioremediation. 53(24):2092–2111, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/10643389.2023.2212569">[DOI]</a>.</p></li>
<li id="id8687"><p>Anna Stikane, Matiss Ricards Baumanis, Reinis Muiznieks, and Egils Stalidzāns. Impact of waste as a substrate on biomass formation, and optimization of spent microbial biomass re-use by sustainable metabolic engineering. 9(6):531–531, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation9060531">[DOI]</a>.</p></li>
<li id="id8688"><p>Adrián Povo‐Retana, Marco Fariñas, Rodrigo Landauro‐Vera, Marina Mojena, Carlota Alvarez‐Lucena, Miguel Ángel Fernández‐Moreno, Antonio Castrillo, Juan Vladimir de la Rosa, Sergio Sánchez‐García, Carles Foguet, Francesc Mas, Silvia Marín, Marta Cascante, and Lisardo Boscá. Immunometabolic actions of trabectedin and lurbinectedin on human macrophages: relevance for their anti-tumor activity. 14():1211068–1211068, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fimmu.2023.1211068">[DOI]</a>.</p></li>
<li id="id8689"><p>M J Neal, Deepan Thiruppathy, and Karsten Zengler. Genome-scale metabolic modeling of the human gut bacterium bacteroides fragilis strain 638r. 19(10):e1011594–e1011594, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011594">[DOI]</a>.</p></li>
<li id="id8695"><p>Nunthaphan Vikromvarasiri, Shuhei Noda, Tomokazu Shirai, and Akihiko Kondo. Investigation of two metabolic engineering approaches for (r,r)-2,3-butanediol production from glycerol in bacillus subtilis. 17(1):3–3, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13036-022-00320-w">[DOI]</a>.</p></li>
<li id="id8696"><p>Darwin Carranza‐Saavedra, Jesús Torres‐Bacete, Blas Blázquez, Claudia Patricia Sánchez Henao, José E. Zapata, and Juan Nogales. System metabolic engineering of escherichia coli w for the production of 2-ketoisovalerate using unconventional feedstock. 11():1176445–1176445, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2023.1176445">[DOI]</a>.</p></li>
<li id="id8697"><p>Xueting Wang, Ali Mohsin, Yifei Sun, Chao Li, Yingping Zhuang, and Guan Wang. From spatial-temporal multiscale modeling to application: bridging the valley of death in industrial biotechnology. 10(6):744–744, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering10060744">[DOI]</a>.</p></li>
<li id="id8698"><p>Rafael Muñoz‐Tamayo, Mohsen Davoudkhani, Ibrahim Fakih, Carlos Eduardo Robles-Rodríguez, Francesco Rubino, Christopher J. Creevey, and Évelyne Forano. Review: towards the next-generation models of the rumen microbiome for enhancing predictive power and guiding sustainable production strategies. 17():100984–100984, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.animal.2023.100984">[DOI]</a>.</p></li>
<li id="id8708"><p>Takeyuki Tamura, Ai Muto-Fujita, Yukako Tohsato, and Tomoyuki Kosaka. Gene deletion algorithms for minimum reaction network design by mixed-integer linear programming for metabolite production in constraint-based models: gdel_minrn. 30(5):553–568, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2022.0352">[DOI]</a>.</p></li>
<li id="id8709"><p>Patrick E. Gelbach and Stacey D. Finley. Ensemble-based genome-scale modeling predicts metabolic differences between macrophage subtypes in colorectal cancer. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.09.532000">[DOI]</a>.</p></li>
<li id="id8710"><p>Hiroki Suyama, Laurence Don Wai Luu, Ling Zhong, Mark J. Raftery, and Ruiting Lan. Integrating proteomic data with metabolic modeling provides insight into key pathways of bordetella pertussis biofilms. 14():1169870–1169870, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2023.1169870">[DOI]</a>.</p></li>
<li id="id8711"><p>Seo‐Young Park, Dong‐Hyuk Choi, Jinsung Song, Uiseon Park, Hyeran Cho, Bee Hak Hong, Yaron Silberberg, and Dong‐Yup Lee. Debottlenecking and reformulating feed media for improved cho cell growth and titer by data‐driven and model‐guided analyses. 18(12):e2300126–e2300126, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.202300126">[DOI]</a>.</p></li>
<li id="id8712"><p>Ronan M. T. Fleming, Hulda S. Haraldsdóttir, Hoai Minh Le, Phan Tu Vuong, Thomas Hankemeier, and Ines Thiele. Cardinality optimization in constraint-based modelling: application to human metabolism. 39(9):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btad450">[DOI]</a>.</p></li>
<li id="id8713"><p>Isabella Casini, Tim McCubbin, Sofia Esquivel‐Elizondo, Guillermo G. Luque, Daria Evseeva, Christian Fink, Sebastian Beblawy, Nicholas D. Youngblut, Ludmilla Aristilde, Daniel H. Huson, Andreas Dräger, Ruth E. Ley, Esteban Marcellin, Largus T. Angenent, and Bastian Molitor. An integrated systems biology approach reveals differences in formate metabolism in the genus methanothermobacter. 26(10):108016–108016, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2023.108016">[DOI]</a>.</p></li>
<li id="id8714"><p>Kelly S. Hayes, Jean‐Marc Schwartz, Richard K. Grencis, and Ian S. Roberts. A genome-scale metabolic model of parasitic whipworm. 14(1):6937–6937, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-023-42552-4">[DOI]</a>.</p></li>
<li id="id8720"><p>Zhidong Zhang, Qi Guo, Jingyi Qian, Chao Ye, and He Huang. Construction and application of the genome-scale metabolic model of streptomyces radiopugnans. 11():1108412–1108412, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2023.1108412">[DOI]</a>.</p></li>
<li id="id8721"><p>Garhima Arora, Mallar Banerjee, Jimpi Langthasa, Ramray Bhat, and Samrat Chatterjee. Targeting metabolic fluxes reverts metastatic transitions in ovarian cancer. 26(11):108081–108081, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2023.108081">[DOI]</a>.</p></li>
<li id="id8722"><p>Peter Gwynne, Kee-Lee K. Stocks, Elysse S. Karozichian, Aarya Pandit, and Linden T. Hu. Metabolic modeling predicts unique drug targets in &lt;i&gt;borrelia burgdorferi&lt;/i&gt;. 8(6):e0083523–e0083523, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00835-23">[DOI]</a>.</p></li>
<li id="id8723"><p>Thummarat Paklao, Apichat Suratanee, and Kitiporn Plaimas. Icon-gems: integration of co-expression network in genome-scale metabolic models, shedding light through systems biology. 24(1):492–492, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-023-05599-0">[DOI]</a>.</p></li>
<li id="id8724"><p>Alec Eames and Sriram Chandrasekaran. Leveraging metabolic modeling and machine learning to uncover modulators of quiescence depth. 3(1):pgae013–pgae013, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/pnasnexus/pgae013">[DOI]</a>.</p></li>
<li id="id8767"><p>Joshua A. M. Kaste and Yair Shachar‐Hill. Accurate flux predictions using tissue-specific gene expression in plant metabolic modeling. 39(5):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btad186">[DOI]</a>.</p></li>
<li id="id8806"><p>Benedict Borer and Stefanía Magnúsdóttir. The media composition as a crucial element in high-throughput metabolic network reconstruction. 13(2):20220070–20220070, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rsfs.2022.0070">[DOI]</a>.</p></li>
<li id="id8807"><p>Igor Marín de Mas, Helena Herand, Jorge Carrasco Muriel, Lars K. Nielsen, and Pär I. Johansson. A protocol for the automatic construction of highly curated genome-scale models of human metabolism. 10(5):576–576, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering10050576">[DOI]</a>.</p></li>
<li id="id8808"><p>Xiaoxiang Gao, Jianxin Zhao, Wei Chen, and Qixiao Zhai. Food and drug design for gut microbiota-directed regulation: current experimental landscape and future innovation. 194():106867–106867, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.phrs.2023.106867">[DOI]</a>.</p></li>
<li id="id8809"><p>Johann F. Jadebeck, Wolfgang Wiechert, and Katharina Nöh. Practical sampling of constraint-based models: optimized thinning boosts chrr performance. 19(8):e1011378–e1011378, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011378">[DOI]</a>.</p></li>
<li id="id8810"><p>Romeu Viana, Tiago Carreiro, Diogo Couceiro, Óscar Dias, Isabel Rocha, and Miguel C. Teixeira. Metabolic reconstruction of the human pathogen &lt;i&gt;candida auris&lt;/i&gt;: using a cross-species approach for drug target prediction. 23():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsyr/foad045">[DOI]</a>.</p></li>
<li id="id8811"><p>Maurício Alexander de Moura Ferreira, Wendel Batista da Silveira, and Zoran Nikoloski. Parrot: prediction of enzyme abundances using protein-constrained metabolic models. 19(10):e1011549–e1011549, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011549">[DOI]</a>.</p></li>
<li id="id8812"><p>Brandt Bessell, Josh Loecker, Zhongyuan Zhao, Sara Sadat Aghamiri, Sabyasachi Mohanty, Rada Amin, Tomáš Helikar, and Bhanwar Lal Puniya. Como: a pipeline for multi-omics data integration in metabolic modeling and drug discovery. 24(6):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbad387">[DOI]</a>.</p></li>
<li id="id8813"><p>Mikhail A. Kulyashov, Semyon Kolmykov, Т. М. Хлебодарова, and Ilya R. Akberdin. State-of the-art constraint-based modeling of microbial metabolism: from basics to context-specific models with a focus on methanotrophs. 11(12):2987–2987, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms11122987">[DOI]</a>.</p></li>
<li id="id8821"><p>David Henriques, Romain Minebois, David Dos Santos, Eladio Barrio, Amparo Querol, and Eva Balsa‐Canto. A dynamic genome-scale model identifies metabolic pathways associated with cold tolerance in saccharomyces kudriavzevii. 11(3):e0351922–e0351922, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.03519-22">[DOI]</a>.</p></li>
<li id="id8822"><p>Håvard Molversmyr, Ove Øyås, Filip Rotnes, and Jon Olav Vik. Extracting functionally accurate context-specific models of atlantic salmon metabolism. 9(1):19–19, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-023-00280-x">[DOI]</a>.</p></li>
<li id="id8823"><p>Fayaz Soleymani Babadi, Zahra Razaghi‐Moghadam, Fatemeh Zare‐Mirakabad, and Zoran Nikoloski. Prediction of metabolite–protein interactions based on integration of machine learning and constraint-based modeling. 3(1):vbad098–vbad098, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioadv/vbad098">[DOI]</a>.</p></li>
<li id="id8824"><p>Kyoshiro Nonaka, Tatsuya Osamura, and Fumikazu Takahashi. A 4-hydroxybenzoate 3-hydroxylase mutant enables 4-amino-3-hydroxybenzoic acid production from glucose in corynebacterium glutamicum. 22(1):168–168, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-023-02179-y">[DOI]</a>.</p></li>
<li id="id8825"><p>Delphine Nègre, Abdelhalim Larhlimi, and Samuel Bertrand. Reconciliation and evolution of penicillium rubens genome-scale metabolic networks–what about specialised metabolism? 18(8):e0289757–e0289757, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0289757">[DOI]</a>.</p></li>
<li id="id8826"><p>Shweta Shinde, Muhammed Hamdan, Prerna Bhalla, and Aravind Kumar Chandiran. Biocompatible cs&lt;sub&gt;2&lt;/sub&gt;ptx&lt;sub&gt;6&lt;/sub&gt; (x = cl, br, i) vacancy ordered perovskites and &lt;i&gt;shewanella oneidensis mr-1&lt;/i&gt; bacteria hybrid for potential photocatalytic solar fuel production. 4(2):224–230, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acsengineeringau.3c00061">[DOI]</a>.</p></li>
<li id="id8827"><p>Debiao Wu, Feng Xu, Yaying Xu, Mingzhi Huang, Zhimin Li, and Ju Chu. Towards a hybrid model-driven platform based on flux balance analysis and a machine learning pipeline for biosystem design. 9(1):33–42, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.synbio.2023.12.004">[DOI]</a>.</p></li>
<li id="id8841"><p>Thomas Sevrin, Lisa Strasser, Camille Ternet, Philipp Junk, Miriam Caffarini, Stella Prins, Cian D’Arcy, Simona Catozzi, Giorgio Oliviero, Kieran Wynne, Christina Kiel, and Philip J. Luthert. Whole-cell energy modeling reveals quantitative changes of predicted energy flows in ras mutant cancer cell lines. 26(2):105931–105931, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2023.105931">[DOI]</a>.</p></li>
<li id="id8842"><p>Shen Qiu, Hua Yang, Qing‐Peng Kong, Gong‐Hua Li, and Li Li. Metabolic modeling identifies a novel molecular type of glioblastoma associated with good prognosis. 13(2):172–172, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo13020172">[DOI]</a>.</p></li>
<li id="id8843"><p>Guadalupe Alvarez-Gonzalez, Micaela Chacόn, Adokiye Berepiki, Karl Fisher, Piya Gosalvitr, Rosa M. Cuéllar-Franca, and Neil Dixon. Complex waste stream valorisation through combined enzymatic hydrolysis and catabolic assimilation by&lt;i&gt;pseudomonas putida&lt;/i&gt;. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.02.13.528311">[DOI]</a>.</p></li>
<li id="id8844"><p>Jonathan Lo, Chao Wu, Jonathan R. Humphreys, Bin Yang, Zhenxiong Jiang, Xin Wang, Pin‐Ching Maness, Nicolas Tsesmetzis, and Wei Xiong. Thermodynamic and kinetic modeling directs pathway optimization for isopropanol production in a gas-fermenting bacterium. 8(2):e0127422–e0127422, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01274-22">[DOI]</a>.</p></li>
<li id="id8845"><p>Heewon Jung, Hyun‐Seob Song, and Christof Meile. Complab v1.0: a scalable pore-scale model for flow, biogeochemistry, microbial metabolism, and biofilm dynamics. 16(6):1683–1696, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/gmd-16-1683-2023">[DOI]</a>.</p></li>
<li id="id8846"><p>Reinis Muiznieks, Elīna Dāce, and Egils Stalidzāns. Integrated sustainability score implementation as an objective function in sustainable metabolic engineering. 9(6):548–548, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation9060548">[DOI]</a>.</p></li>
<li id="id8847"><p>Maryam Khoshnejat, Ali Mohammad Banaei‐Moghaddam, Ali Akbar Moosavi‐Movahedi, and Kaveh Kavousi. A holistic view of muscle metabolic reprogramming through personalized metabolic modeling in newly diagnosed diabetic patients. 18(6):e0287325–e0287325, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0287325">[DOI]</a>.</p></li>
<li id="id8848"><p>Yang Zhang, Menglei Yang, Yang‐Yang Bao, Weihua Tao, Jinyou Tuo, Boya Liu, Luxi Gan, Shuilin Fu, and Heng Gong. A genome-scale metabolic model of the effect of dissolved oxygen on 1,3-propanediol fermentation by klebsiella pneumoniae. 46(9):1319–1330, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-023-02899-w">[DOI]</a>.</p></li>
<li id="id8849"><p>Takeyuki Tamura. Metnetcomp: database for minimal and maximal gene-deletion strategies for growth-coupled production of genome-scale metabolic networks. 20(6):3748–3758, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/tcbb.2023.3317837">[DOI]</a>.</p></li>
<li id="id8850"><p>Chengjie Hou, Xin Song, Zhiqiang Xiong, Guangqiang Wang, Yongjun Xia, and Lianzhong Ai. Genome‐scale reconstruction of the metabolic network in &lt;scp&gt;&lt;i&gt;streptococcus thermophilus&lt;/i&gt;&lt;/scp&gt; s‐3 and assess urea metabolism. 104(3):1458–1469, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/jsfa.13026">[DOI]</a>.</p></li>
<li id="id8851"><p>James D. Brunner, Laverne Gallegos-Graves, and Marie Kroeger. Inferring microbial interactions with their environment from genomic and metagenomic data. 19(11):e1011661–e1011661, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1011661">[DOI]</a>.</p></li>
<li id="id8861"><p>Chi Hung Vo, Nishu Goyal, Markus Kraft, and Iftekhar A. Karimi. Carbon conversion by methanococcus maripaludis s2 under diazotrophy and a revised genome-scale metabolic model. 278():118910–118910, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ces.2023.118910">[DOI]</a>.</p></li>
<li id="id8862"><p>Pedro de Atauri, Carles Foguet, and Marta Cascante. Control analysis in the identification of key enzymes driving metabolic adaptations: towards drug target discovery. 231():104984–104984, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2023.104984">[DOI]</a>.</p></li>
<li id="id8863"><p>Jorge Carrasco Muriel, Christopher P. Long, and Nikolaus Sonnenschein. Simultaneous application of enzyme and thermodynamic constraints to metabolic models using an updated python implementation of gecko. 11(6):e0170523–e0170523, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.01705-23">[DOI]</a>.</p></li>
<li id="id8864"><p>Yanhua Liu and Hans V. Westerhoff. ‘social’ versus ‘asocial’ cells—dynamic competition flux balance analysis. 9(1):53–53, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-023-00313-5">[DOI]</a>.</p></li>
<li id="id8865"><p>María Teresita Castañeda, Sebastián Núñez, Martín Jamilis, and Hernán De Battista. Computational assessment of lipid production in &lt;i&gt;rhodosporidium toruloides&lt;/i&gt; in two‐stage and one‐stage batch bioprocesses. 121(1):238–249, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28579">[DOI]</a>.</p></li>
<li id="id8866"><p>Daniel E. Schäffer, Wenrui Li, Abdurrahman Elbasir, Dario C. Altieri, Qi Long, and Noam Auslander. Microbial gene expression analysis of healthy and cancerous esophagus uncovers bacterial biomarkers of clinical outcomes. 3(1):128–128, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s43705-023-00338-1">[DOI]</a>.</p></li>
<li id="id8888"><p>Bruno G. Galuzzi, Stefano Izzo, Fabio Giampaolo, Salvatore Cuomo, Marco Vanoni, Lilia Alberghina, Chiara Damiani, and Francesco Piccialli. Coupling constrained-based flux sampling and clustering to tackle cancer metabolic heterogeneity. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/pdp59025.2023.00037">[DOI]</a>.</p></li>
<li id="id8889"><p>Jiahang Li, Wolfram Weckwerth, and Steffen Waldherr. Enzyme fluctuations data improve inference of metabolic interaction networks with an inverse differential jacobian approach. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.11.570118">[DOI]</a>.</p></li>
<li id="id8930"><p>Quinn S. Solfisburg, Federico Baldini, Brittany Baldwin-Hunter, Harry H. Lee, Heekuk Park, Daniel E. Freedberg, Charles J. Lightdale, Tal Korem, and Julian A. Abrams. The salivary microbiome and predicted metabolite production are associated with progression from barrett’s esophagus to esophageal adenocarcinoma. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.06.27.546733">[DOI]</a>.</p></li>
<li id="id8936"><p>Tamara Bintener, Maria Pires Pacheco, Demetra Philippidou, Christiane Margue, Ali Kishk, Greta Del Mistro, Luca Di Leo, María Moscardó Garcia, Rashi Halder, Lasse Sinkkonen, Daniela De Zio, Stephanie Kreis, Dagmar Kulms, and Thomas Sauter. Metabolic modelling-based in silico drug target prediction identifies six novel repurposable drugs for melanoma. 14(7):468–468, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41419-023-05955-1">[DOI]</a>.</p></li>
<li id="id8937"><p>Subasree Sridhar, Prerna Bhalla, Justin Kullu, Sriya Veerapaneni, Swagatika Sahoo, Nirav Bhatt, and G. K. Suraishkumar. A reactive species reactions module for integration into genome-scale metabolic models for improved insights: application to cancer. 80():78–93, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2023.08.006">[DOI]</a>.</p></li>
<li id="id8938"><p>Kelin Chen and Feng‐Sheng Wang. Cell‐specific genome‐scale metabolic modeling of &lt;scp&gt;sars‐cov&lt;/scp&gt; ‐2‐infected lung to identify antiviral enzymes. 13(12):2172–2186, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/2211-5463.13710">[DOI]</a>.</p></li>
<li id="id8939"><p>Ehsan Zangene, Sayed‐Amir Marashi, and Hesam Montazeri. Sl-scan identifies synthetic lethal interactions in cancer using metabolic networks. 13(1):15763–15763, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-023-42992-4">[DOI]</a>.</p></li>
<li id="id8940"><p>Nora Scherer, Daniel Fässler, Oleg Borisov, Yurong Cheng, Pascal Schlosser, Matthias Wuttke, Suraj Patil, Heike Meiselbach, Fabian Telkämper, Urs Berger, Sarah C. Grünert, Peggy Sekula, Ulla T. Schultheiß, Yong Li, Michael Köttgen, Peter J. Oefner, Felix Knauf, Kai‐Uwe Eckardt, Ines Thiele, Miriam Schmidts, Johannes Hertel, and Anna Köttgen. Coupling of metabolomics and exome sequencing reveals graded effects of rare damaging heterozygous variants on gene function and resulting traits and diseases. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.17.23297094">[DOI]</a>.</p></li>
<li id="id8941"><p>Arturo Vera‐Ponce de León, Matthias Hoetzinger, Tim Hensen, Shashank Gupta, Bronson R. Weston, Sander M. Johnsen, Jacob Agerbo Rasmussen, Cecilie Grønlund Clausen, Louisa Pless, Ana Raquel Verissimo, Knut Rudi, Lars Snipen, Christian Karlsen, Morten T. Limborg, Stefan Bertilsson, Ines Thiele, Torgeir R. Hvidsten, Simen R. Sandve, Phillip B. Pope, and Sabina Leanti La Rosa. The salmon microbial genome atlas enables novel insights into bacteria-host interactions via functional mapping. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.10.570985">[DOI]</a>.</p></li>
<li id="id8951"><p>Nantia Leonidou, Alina Renz, Reihaneh Mostolizadeh, and Andreas Dräger. New workflow predicts drug targets against sars-cov-2 via metabolic changes in infected cells. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202203.0290.v3">[DOI]</a>.</p></li>
<li id="id8952"><p>Jorge Carrasco Muriel, Christopher P. Long, and Nikolaus Sonnenschein. Simultaneous application of enzyme and thermodynamic constraints to metabolic models using an updated python implementation of gecko. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.20.533446">[DOI]</a>.</p></li>
<li id="id8953"><p>Deepanwita Banerjee, Ian Sofian Yunus, Xi Wang, Jinho Kim, Aparajitha Srinivasan, Russel Menchavez, Yan Chen, Jennifer Gin, Christopher J. Petzold, Héctor García Martín, Paul D. Adams, Aindrila Mukhopadhyay, Joonhoon Kim, and Taek Soon Lee. Genome-scale and pathway engineering for the sustainable aviation fuel precursor isoprenol production in&lt;i&gt;pseudomonas putida&lt;/i&gt;. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.04.29.538800">[DOI]</a>.</p></li>
<li id="id8954"><p>Reinis Muiznieks, Elīna Dāce, and Egils Stalidzāns. Sustainable metabolic engineering design development employing a complex objective function. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202305.0094.v1">[DOI]</a>.</p></li>
<li id="id8955"><p>Xingcun Fan, Lingfeng Cao, and Xuefeng Yan. Sensitivity analysis and adaptive mutation strategy differential evolution algorithm for optimizing enzymes' turnover numbers in metabolic models. 120(8):2301–2313, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28493">[DOI]</a>.</p></li>
<li id="id8979"><p>Garhima Arora, Jimpi Langthasa, Mallar Banerjee, Ramray Bhat, and Samrat Chatterjee. Targeting metabolic fluxes reverts metastatic transitions in ovarian cancer. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.02.538518">[DOI]</a>.</p></li>
<li id="id8980"><p>Dipanka Tanu Sarmah, Abhijit Paul, Umang Berry, Milan Surjit, Nandadulal Bairagi, and Samrat Chatterjee. Bag6 is a novel player in controlling nonalcoholic steatohepatitis: result from a comprehensive&lt;i&gt;in-silico&lt;/i&gt;study. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.04.539506">[DOI]</a>.</p></li>
<li id="id8981"><p>Zeqian Li, Ahmed Selim, and Seppe Kuehn. Statistical prediction of microbial metabolic traits from genomes. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.06.30.547261">[DOI]</a>.</p></li>
<li id="id8982"><p>Himel Nahreen Khaleque, Hadi Nazem‐Bokaee, Yosephine Gumulya, Ross P. Carlson, and Anna H. Kaksonen. Simulating compatible solute biosynthesis using a metabolic flux model of the biomining acidophile, acidithiobacillus ferrooxidans atcc 23270. 175(1-2):104115–104115, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.resmic.2023.104115">[DOI]</a>.</p></li>
<li id="id8983"><p>David Healy, Shuo Wang, Ghjuvan Micaelu Grimaud, Alicja K. Warda, R. Paul Ross, Catherine Stanton, and Eugene Dempsey. Longitudinal observational study protocol – preterm infants: microbiome establishment, neuro-crosstalk and origins (pimento). 13(9):e075060–e075060, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1136/bmjopen-2023-075060">[DOI]</a>.</p></li>
<li id="id8984"><p>Holly R. Steach, Siddharth Viswanath, Yixuan He, Xitong Zhang, Natalia Ivanova, Matthew Hirn, Michael Perlmutter, and Smita Krishnaswamy. Inferring metabolic states from single cell transcriptomic data via geometric deep learning. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.05.570153">[DOI]</a>.</p></li>
<li id="id9000"><p>Isabel Valenbreder, Sonia Balăn, Marian Breuer, and Michiel Adriaens. Mapping out the gut microbiota-dependent trimethylamine n-oxide super pathway for systems biology applications. 3():1074749–1074749, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2023.1074749">[DOI]</a>.</p></li>
<li id="id9001"><p>Dário Neves, Daniel Meinen, Tobias B. Alter, Lars M. Blank, and Birgitta E. Ebert. Expanding &lt;b&gt; &lt;i&gt;pseudomonas taiwanensis&lt;/i&gt; &lt;/b&gt; vlb120's acyl‐coa portfolio: propionate production in mineral salt medium. 17(1):e14309–e14309, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.14309">[DOI]</a>.</p></li>
<li id="id9002"><p>Eliott Jacopin, Yuki Sakamoto, Kozo Nishida, Kazunari Kaizu, and Koichi Takahashi. A novel architecture for collaboration in systems biology at the age of the metaverse. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3229264/v1">[DOI]</a>.</p></li>
<li id="id9003"><p>Juan Ricardo Velasco-Álvarez, Nimbe Torres, Isaac Chaírez, and José Luis Castrejón-Flores. Microbiome distribution modeling using gradient descent strategies for mock, in vitro and clinical community distributions. 18(8):e0290082–e0290082, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0290082">[DOI]</a>.</p></li>
<li id="id9004"><p>Jing Yu, Xiaowei Wang, Qianqian Yuan, Jiaxin Shi, Jingyi Cai, Zhichao Li, and Hongwu Ma. Elucidating the impact of in vitro cultivation on nicotiana tabacum metabolism through combined in silico modeling and multiomics analysis. 14():1281348–1281348, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2023.1281348">[DOI]</a>.</p></li>
<li id="id9005"><p>Egils Stalidzāns, Reinis Muiznieks, Konstantīns Dubencovs, Elīna Sīle, Kristaps Berzins, Artūrs Šuleiko, and Juris Vanags. A fermentation state marker rule design task in metabolic engineering. 10(12):1427–1427, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering10121427">[DOI]</a>.</p></li>
<li id="id9006"><p>J Schönborn, Irfan Akhtar, Andrea Droste, Stephanie Wesemann, and Mathias Beller. Physiological constraints limiting the growth of&lt;i&gt;drosophila&lt;/i&gt;larvae. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.20.572459">[DOI]</a>.</p></li>
<li id="id9010"><p>Lamiaa H.S. Hassan, Imran Ahmad, Mostafa El Sheekh, and Norhayati Abdullah. In silico models on algal cultivation and processing. ():989–1016, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.4018/979-8-3693-3026-5.ch042">[DOI]</a>.</p></li>
<li id="id9080"><p>Ronan M. T. Fleming, Xi Luo, Yanjun Liu, Alexander Balck, and Christine Klein. Identification of metabolites reproducibly associated with parkinson’s disease via meta-analysis and computational modelling. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3209421/v1">[DOI]</a>.</p></li>
<li id="id9081"><p>N Ofori-Anyinam, Michael R. Hamblin, ML Coldren, Bin Li, Gautam Mereddy, Majid B. Shaikh, Aditi Shah, Navpreet Ranu, Shou‐En Lu, PC Blainey, Si Ma, JJ Collins, and Jason H. Yang. Katg catalase deficiency confers bedaquiline hyper-susceptibility to isoniazid resistant&lt;i&gt;mycobacterium tuberculosis&lt;/i&gt;. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.17.562707">[DOI]</a>.</p></li>
<li id="id9082"><p>Elaine Zaunseder, Ulrike Mütze, Jürgen G. Okun, Georg F. Hoffmann, Stefan Kölker, Vincent Heuveline, and Ines Thiele. Personalised metabolic whole-body models for newborns and infants predict growth and biomarkers of inherited metabolic diseases. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.10.20.563364">[DOI]</a>.</p></li>
<li id="id9083"><p>Gabriela Canto-Encalada, Jenna L. Armstrong, Carlos Focil-Espinosa, Ademikanra Adekunle-Fiyin, Ila Peeler, William Gebbie, J. Nuñez‐Garcia, Alexis Saldivar, Diego Martínez, and Cristal Zúñiga. Ten quick tips for accurate reconstruction of prokaryotic and eukaryotic genome-scale metabolic models. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202311.0461.v1">[DOI]</a>.</p></li>
<li id="id9107"><p>Seo‐Young Park, Dong‐Hyuk Choi, Jinsung Song, Uiseon Park, Hyeran Cho, Bee Hak Hong, Yaron Silberberg, and Dong‐Yup Lee. Debottlenecking and reformulating feed media for improved cho cell growth and titer by data-driven and model-guided analyses. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.09.531884">[DOI]</a>.</p></li>
<li id="id9108"><p>Maurício Alexander de Moura Ferreira, Wendel Batista da Silveira, and Zoran Nikoloski. Parrot: prediction of enzyme abundances using protein-constrained metabolic models. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.22541/au.166117417.77605988/v2">[DOI]</a>.</p></li>
<li id="id9109"><p>Agris Pentjuss, Emīls Bolmanis, Anastasija Šuleiko, Elina Didrihsone, Artūrs Šuleiko, Konstantīns Dubencovs, Jānis Liepiņš, Andris Kazāks, and Juris Vanags. &lt;i&gt;pichia pastoris&lt;/i&gt;growth - coupled heme biosynthesis analysis using metabolic modelling. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.13.540629">[DOI]</a>.</p></li>
<li id="id9110"><p>Francis Mairet. &lt;i&gt;in silico&lt;/i&gt; identification of switching nodes in metabolic networks. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.17.541195">[DOI]</a>.</p></li>
<li id="id9111"><p>Kristina Grausa, Shahida Anusha Siddiqui, Norbert Lameyer, Karin Wiesotzki, Sergiy Smetana, and Agris Pentjuss. Metabolic modeling of hermetia illucens larvae resource allocation for high-value fatty acid production. 13(6):724–724, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo13060724">[DOI]</a>.</p></li>
<li id="id9112"><p>Dipanka Tanu Sarmah, Abhijit Paul, Shivam Kumar, Nandadulal Bairagi, and Samrat Chatterjee. A data-driven multilayer approach for the identification of potential therapeutic targets in non-alcoholic steatohepatitis. 624():128955–128955, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.physa.2023.128955">[DOI]</a>.</p></li>
<li id="id9113"><p>Yanhua Liu and Hans V. Westerhoff. ‘social’ versus ‘asocial’ cells— dynamic competition flux balance analysis. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3059897/v1">[DOI]</a>.</p></li>
<li id="id9114"><p>Fangzhou Xiao, Jing Shuang Li, and John C. Doyle. Flux exponent control predicts metabolic dynamics from network structure. ():1189–1194, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.23919/acc55779.2023.10156281">[DOI]</a>.</p></li>
<li id="id9115"><p>Jingning Chen, Huang Yi-ran, and Cheng Zhong. Minimizing enzyme mass to decompose flux distribution for identifying biologically relevant elementary flux modes. 231():104981–104981, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2023.104981">[DOI]</a>.</p></li>
<li id="id9116"><p>Zihe Liu, Ning Qin, Lingyun Li, Xiaozhen Wan, Xu Ji, Yu Chen, Chaokun Li, Ping Liu, Yijie Zhang, Weijie Yang, Junfeng Jang, Jianye Xia, Shuobo Shi, Tianwei Tan, Jens Nielsen, and Yun Chen. Increased co2 fixation enables high carbon-yield production of the acrylic acid precursor 3-hydroxypropionic acid in yeast. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3278694/v1">[DOI]</a>.</p></li>
<li id="id9117"><p>Karel Olavarría, Marco V. Becker, Diana Z. Sousa, Mark C.M. van Loosdrecht, and A. Wahl. Design and thermodynamic analysis of a pathway enabling anaerobic production of poly-3-hydroxybutyrate in escherichia coli. 8(4):629–639, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.synbio.2023.09.005">[DOI]</a>.</p></li>
<li id="id9118"><p>Lingyun Li, Ning Qin, Xin Chen, Yu Chen, Xu Ji, Yijie Zhang, Jens Nielsen, and Zihe Li. Mechanistic understanding of the &quot;gatekeeper&quot; phosphofructokinase of the glycolytic pathway on the central carbon metabolism. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3418319/v1">[DOI]</a>.</p></li>
<li id="id9119"><p>Famke Bäuerle, Gwendolyn O. Döbel, Laura Camus, Simon Heilbronner, and Andreas Dräger. Genome-scale metabolic models consistently predict in vitro characteristics of corynebacterium striatum. 3():1214074–1214074, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbinf.2023.1214074">[DOI]</a>.</p></li>
<li id="id9120"><p>Richard C. Law, Kanchan Pande, Samantha O’Keeffe, Glenn Nurwono, Rachel Ki, Pin‐Kuang Lai, and Junyoung O. Park. Accurate and rapid determination of metabolic flux by deep learning of isotope patterns. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.11.06.565907">[DOI]</a>.</p></li>
<li id="id9142"><p>Juan D. Tibocha‐Bonilla, Manish Kumar, Karsten Zengler, and Cristal Zúñiga. Integrating metabolic modeling and high‐throughput data to characterize diatoms metabolism. ():165–191, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781119751939.ch7">[DOI]</a>.</p></li>
<li id="id9143"><p>Sebastián Nelson Mendoza Farías. Metabolic modeling of microorganisms and dynamics of microbial communities in the food industry. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.5463/thesis.540">[DOI]</a>.</p></li>
<li id="id9204"><p>Christof Meile. Comment on egusphere-2022-1016. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/egusphere-2022-1016-ac1">[DOI]</a>.</p></li>
<li id="id9205"><p>Pınar Kocabaş. Comparative flux balance analyses of serine alkaline protease overproduction in bacillus subtilis at genome and small scale. 12(1):8–18, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.18036/estubtdc.1184733">[DOI]</a>.</p></li>
<li id="id9206"><p>Naroa Barrena, Luis V. Valcárcel, Danel Olaverri-Mendizabal, Iñigo Apaolaza, and Francisco J. Planes. Synthetic lethality in large-scale integrated metabolic and regulatory network models of human cells. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.01.27.525829">[DOI]</a>.</p></li>
<li id="id9207"><p>Pranas Grigaitis. Constrain and conquer. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.5463/thesis.15">[DOI]</a>.</p></li>
<li id="id9208"><p>Lisardo Boscá, Adrián Povo‐Retana, Marco Fariñas, Rodrigo Landauro‐Vera, Marina Mojena, Carlota Alvarez‐Lucena, Miguel Ángel Fernández‐Moreno, Antonio Castrillo, Juan de la Rosa, Sergio Sánchez‐García, Carles Foguet, Francesc Mas, Silvia Marín, and Marta Cascante. Immunometabolic actions of trabectedin and lurbinectedin on human macrophages: relevance for their antitumor activity. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-2625385/v1">[DOI]</a>.</p></li>
<li id="id9209"><p>David M. Versluis, Ruud Schoemaker, Ellen Looijesteijn, Jan M.W. Geurts, and Roeland M. H. Merks. 2’-fucosyllactose helps butyrate producers outgrow competitors in infant gut microbiota simulations. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.10.532059">[DOI]</a>.</p></li>
<li id="id9210"><p>Thomas Eng, Deepanwita Banerjee, Javier Menasalvas, Yan Chen, Jennifer Gin, Hemant Choudhary, Edward E. K. Baidoo, Jian-Hua Chen, Axel Ekman, Ramu Kakumanu, Yuzhong Liu, Alex Codik, Carolyn A. Larabell, John M. Gladden, Blake A. Simmons, Jay D. Keasling, Christopher J. Petzold, and Aindrila Mukhopadhyay. Ensemble and iterative engineering for maximized bioconversion to the blue pigment, indigoidine from non-canonical sustainable carbon sources. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.16.532821">[DOI]</a>.</p></li>
<li id="id9211"><p>Fangzhou Xiao, Jing Shuang Li, and John C. Doyle. Flux exponent control predicts metabolic dynamics from network structure. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.03.23.533708">[DOI]</a>.</p></li>
<li id="id9212"><p>Haoran Zheng, Yupeng Qi, and Yanlong Zhao. A target prediction method to inhibit the replication process of sars-cov-2 via metabolic differential analysis. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-2668079/v1">[DOI]</a>.</p></li>
<li id="id9213"><p>Famke Bäuerle, Gwendolyn O. Gusak, Laura Camus, Simon Heilbronner, and Andreas Dräger. Genome-scale metabolic models consistently predict&lt;i&gt;in vitro&lt;/i&gt;characteristics of&lt;i&gt;corynebacterium striatum&lt;/i&gt;. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.04.28.538764">[DOI]</a>.</p></li>
<li id="id9214"><p>N. Suhas Jagannathan, Javier Yu Peng Koh, Younghwan Lee, Radoslaw M. Sobota, Aaron T. Irving, Lin‐Fa Wang, Yoko Itahana, Koji Itahana, and Lisa Tucker‐Kellogg. Multi-omic analysis of bat versus human fibroblasts reveals altered central metabolism. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.08.537600">[DOI]</a>.</p></li>
<li id="id9215"><p>Pavel Dvořák, Barbora Burýšková, Barbora Popelářová, Birgitta E. Ebert, Tibor Botka, Dalimil Bujdoš, Alberto Sánchez-Pascuala, Hannah Schöttler, Heiko Hayen, Vı́ctor de Lorenzo, Lars M. Blank, and Martin Benešík. Genomic and metabolic plasticity drive alternative scenarios for adapting&lt;i&gt;pseudomonas putida&lt;/i&gt;to non-native substrate d-xylose. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.05.19.541448">[DOI]</a>.</p></li>
<li id="id9216"><p>Aswathy K. Raghu, Indumathi Palanikumar, and Karthik Raman. Designing function-specific minimal microbiomes from large microbial communities. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.06.12.544531">[DOI]</a>.</p></li>
<li id="id9217"><p>Sara Benito-Vaquerizo. Model-driven design of synthetic microbial communities for the upcycling of one-carbon feedstocks. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/630256">[DOI]</a>.</p></li>
<li id="id9218"><p>Joe Jongpyo Lim, Julia Yue Cui, and Yuliang Wang. Genome-scale metabolic model predicts mechanistic differences in metabolic gradient of spatially resolved small intestine villi. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.06.22.546161">[DOI]</a>.</p></li>
<li id="id9219"><p>Leila Hassani, Mohammad R. Moosavi, Payam Setoodeh, and Habil Zare. Fastknock: an efficient next-generation approach to identify all knockout strategies for strain optimization. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-3126389/v1">[DOI]</a>.</p></li>
<li id="id9220"><p>Karel Olavarría, Marco V. Becker, Diana Z. Sousa, and A. Wahl. Design and thermodynamic analysis of a pathway enabling anaerobic production of poly-3-hydroxybutyrate in escherichia coli. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4495353">[DOI]</a>.</p></li>
<li id="id9221"><p>Francesco Balzerani. Enzyme promiscuity approaches to predict metabolic pathways in the human gut microbiota. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.15581/10171/66822">[DOI]</a>.</p></li>
<li id="id9222"><p>Gong‐Hua Li, Feifei Han, Rong‐Hua Luo, Peng Li, Chia-Jung Chang, Weihong Xu, Xin‐Yan Long, Jing‐Fei Huang, Yong‐Tang Zheng, Qing‐Peng Kong, and Wenzhong Xiao. Predictive systems biology modeling: unraveling host metabolic disruptions and potential drug targets in acute viral infections. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.07.24.550423">[DOI]</a>.</p></li>
<li id="id9223"><p>Shaohua Guan, Zhichao Zhang, Zihan Zhang, and Hualin Shi. Universal scaling relation and criticality in metabolism and growth of escherichia coli. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.2308.04776">[DOI]</a>.</p></li>
<li id="id9224"><p>Anukriti Verma, Bhawna Rathi, Shivani Sharda, and Pallavi Somvanshi. Machine learning and systems level insights into the host–pathogen interactions of post-infectious reactive arthritis. ():729–744, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-99-3485-0_58">[DOI]</a>.</p></li>
<li id="id9225"><p>Silvia Sabatini, Partho Sen, Fabrizia Carli, Samantha Pezzica, Chiara Rosso, Erminia Lembo, Ornella Verrastro, Ann K. Daly, Olivier Govaere, Simon Cockel, Tuulia Hyötyläinen, Geltrude Mingrone, Elisabetta Bugianesi, Quentin M. Anstee, Matej Orešič, and Amalia Gastaldelli. Higher hepatic glucose production and gluconeogenesis are features of severe metabolic dysfunction-associated steatohepatitis even in absence of t2d. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4563031">[DOI]</a>.</p></li>
<li id="id9227"><p>Eduardo Luís Menezes de Almeida, Maurício Alexander de Moura Ferreira, and Wendel Batista da Silveira. &lt;i&gt;lista&lt;/i&gt;-gem: the genome-scale metabolic reconstruction of&lt;i&gt;lipomyces starkeyi&lt;/i&gt;. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.09.25.559328">[DOI]</a>.</p></li>
<li id="id9228"><p>Mike Sportiello, Alexis Poindexter, Emma C. Reilly, Adam Geber, Kris Lambert Emo, Taylor N. Jones, and David J. Topham. Mouse memory cd8 t cell subsets defined by tissue-resident memory integrin expression exhibit distinct metabolic profiles. 7(10):652–669, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.4049/immunohorizons.2300040">[DOI]</a>.</p></li>
<li id="id9230"><p>Anonymous. Fe-flux: feasible flux boundary modification framework for constraint based metabolic models. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.59287/as-proceedings.349">[DOI]</a>.</p></li>
<li id="id9231"><p>Clémence Dupont-Thibert, Sylvaine Roy, Sónia Carneiro, Bruno Pereira, Rafael Careira, Paulo Vilaça, Séverine Collin, Éric Maréchal, Élodie Billey, Gilles Curien, Maxime Durot, and Juliette Jouhet. A new genome-scale metabolic model of oleaginous microalgae with refined lipid metabolism elucidates microchloropsis gaditana mutant phenotypes. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.06.570374">[DOI]</a>.</p></li>
<li id="id9232"><p>Shilin Ouyang, Zihao Li, Jiamin Hu, Miyuan Cao, Xiting Wang, Hongxia Sun, Yu Feng, and Longfei Mao. Sysrecon: a systematic data-driven tool for genome-scale metabolic reconstructions. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.13.571370">[DOI]</a>.</p></li>
<li id="id9233"><p>Azliati Azizan, Alma Amangeldi, and Yelena Oleinikova. (4(43)):, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.53729/mv-as.2023.04">[DOI]</a>.</p></li>
<li id="id9234"><p>Арман Исимов, Gulnur Bolatovna Admanova, Саржигитова Асылай Саржигитова, Назерке Кемалова, and Б. БАҚЫТЖАНҚЫЗЫ. Python как инструмент для реконструкции и анализа метаболической модели в масштабе генома. (4(43)):32–62, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.53729/mv-as.2023.04.03">[DOI]</a>.</p></li>
<li id="id9235"><p>Richard D. Paul, Johann F. Jadebeck, Anton Stratmann, Wolfgang Wiechert, and Katharina Nöh. Hopsy - a methods marketplace for convex polytope sampling in python. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2023.12.22.573091">[DOI]</a>.</p></li>
<li id="id9246"><p>Kyoshiro Nonaka, Tatsuya Osamura, and Fumikazu Takahashi. Additional file 3 of a 4-hydroxybenzoate 3-hydroxylase mutant enables 4-amino-3-hydroxybenzoic acid production from glucose in corynebacterium glutamicum. ():, 2023. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.24054451">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-3" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-3" name="0-3" role="tabpanel" tabindex="0"><div class="docutils container" id="id9364">
<ol class="arabic simple" start="1">
<li id="id9483"><p>Adam Amara, Clément Frainay, Fabien Jourdan, Thomas Naake, Steffen Neumann, Elva María Novoa‐del‐Toro, Reza M. Salek, Liesa Salzer, Sarah Scharfenberg, and Michael Witting. Networks and graphs discovery in metabolomics data analysis and interpretation. 9():841373–841373, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmolb.2022.841373">[DOI]</a>.</p></li>
<li id="id9493"><p>Xiaoping Liao, Hongwu Ma, and Yinjie Tang. Artificial intelligence: a solution to involution of design–build–test–learn cycle. 75():102712–102712, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2022.102712">[DOI]</a>.</p></li>
<li id="id9496"><p>J. Ashwini John, Melvin S. Samuel, Muthusamy Govarthanan, and Selvarajan Ethiraj. A comprehensive review on strategic study of cellulase producing marine actinobacteria for biofuel applications. 214(Pt 3):114018–114018, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.envres.2022.114018">[DOI]</a>.</p></li>
<li id="id9668"><p>Gong‐Hua Li, Feifei Han, Fu‐Hui Xiao, Kang‐Su‐Yun Gu, Shen Qiu, Weihong Xu, Wenxing Li, Yanli Wang, Bin Liang, Jing‐Fei Huang, Wenzhong Xiao, and Qing‐Peng Kong. System‐level metabolic modeling facilitates unveiling metabolic signature in exceptional longevity. 21(4):e13595–e13595, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/acel.13595">[DOI]</a>.</p></li>
<li id="id9677"><p>Andreas Wagner. Competition for nutrients increases invasion resistance during assembly of microbial communities. 31(15):4188–4203, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/mec.16565">[DOI]</a>.</p></li>
<li id="id9698"><p>Mohammad Mazharul Islam, Andrea Goertzen, Pankaj K. Singh, and Rajib Saha. Exploring the metabolic landscape of pancreatic ductal adenocarcinoma cells using genome-scale metabolic modeling. 25(6):104483–104483, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2022.104483">[DOI]</a>.</p></li>
<li id="id9701"><p>Sang Mi Lee, GaRyoung Lee, and Hyun Uk Kim. Machine learning-guided evaluation of extraction and simulation methods for cancer patient-specific metabolic models. 20():3041–3052, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2022.06.027">[DOI]</a>.</p></li>
<li id="id9772"><p>Kirk Smith, Fangzhou Shen, Ho‐Joon Lee, and Sriram Chandrasekaran. Metabolic signatures of regulation by phosphorylation and acetylation. 25(1):103730–103730, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.103730">[DOI]</a>.</p></li>
<li id="id9788"><p>Prabir Kumar Das, Ansuman Sahoo, and Veeranki Venkata Dasu. Current status, and the developments of hosts and expression systems for the production of recombinant human cytokines. 59():107969–107969, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2022.107969">[DOI]</a>.</p></li>
<li id="id9802"><p>Elif Gencturk and Kutlu Ö. Ülgen. Understanding hmf inhibition on yeast growth coupled with ethanol production for the improvement of bio-based industrial processes. 121():425–438, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.procbio.2022.07.015">[DOI]</a>.</p></li>
<li id="id9892"><p>Kalaivani Paramasivan, Aneesha Abdulla, Nabarupa Gupta, and Sarma Mutturi. &lt;i&gt;in silico&lt;/i&gt; target-based strain engineering of &lt;i&gt;saccharomyces cerevisiae&lt;/i&gt; for terpene precursor improvement. 14(2):25–36, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/intbio/zyac003">[DOI]</a>.</p></li>
<li id="id9894"><p>Manish Kumar, Cristal Zúñiga, Juan D. Tibocha‐Bonilla, Sarah R. Smith, Joanna Coker, Andrew E. Allen, and Karsten Zengler. Constraint-based modeling of diatoms metabolism and quantitative biology approaches. ():775–808, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-92499-7_26">[DOI]</a>.</p></li>
<li id="id9903"><p>Tomokazu Shirai and Akihiko Kondo. In silico design strategies for the production of target chemical compounds using iterative single-level linear programming problems. 12(5):620–620, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biom12050620">[DOI]</a>.</p></li>
<li id="id9909"><p>Carles Ciurans, Josep M. Guerrero, Ivan Martínez-Mongue, Claude‐Gilles Dussap, Igor Marín de Mas, and Francesc Gòdia. Enhancing control systems of higher plant culture chambers via multilevel structural mechanistic modelling. 13():970410–970410, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2022.970410">[DOI]</a>.</p></li>
<li id="id9967"><p>Kusum Dhakar, Hanan Eizenberg, Zeev Ronen, Raphy Zarecki, and Shiri Freilich. Microbial degradation of herbicides in contaminated soils by following computational approaches. ():399–417, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-89778-5.00030-1">[DOI]</a>.</p></li>
<li id="id9987"><p>Mohamad Faiz Dzulkalnine, Mohd Saberi Mohamad, Yee Wen Choon, Muhammad Akmal Remli, and Hany Alashwal. Optimizing ethanol production in escherichia coli using a hybrid of particle swarm optimization and artificial bee colony. ():140–146, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/3571560.3571581">[DOI]</a>.</p></li>
<li id="id9993"><p>Daniela Calvetti and Erkki Somersalo. Brain energy metabolism. ():540–558, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1006-0_100673">[DOI]</a>.</p></li>
<li id="id10070"><p>Eliza Atkinson, Alice Boo, Huadong Peng, Guy‐Bart Stan, and Rodrigo Ledesma‐Amaro. Principles, tools, and applications of synthetic consortia toward microbiome engineering. ():195–218, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527825462.ch7">[DOI]</a>.</p></li>
<li id="id10111"><p>Naomi van Wijk and Michal Linial. Computational methods for the study of peroxisomes in health and disease. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.5772/intechopen.103178">[DOI]</a>.</p></li>
<li id="id10119"><p>Carmen Bedia, Núria Dalmau, Lars K. Nielsen, Romà Tauler, and Igor Marín de Mas. A multi-level approach to study the metabolic effects of chronic exposure to aldrin in prostate cancer cells. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-1769167/v1">[DOI]</a>.</p></li>
<li id="id10120"><p>Zehra Esra Ilhan and Melissa M. Herbst‐Kralovetz. Microbial drug interactions and human health. ():120–149, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/9781839160950-00120">[DOI]</a>.</p></li>
<li id="id10121"><p>Shrameeta Shinde, Zhenxiong Jiang, and Xin Wang. Exploring the metabolic versatility of cyanobacteria for an emerging carbon-neutral bioeconomy. ():165–187, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-96106-6.00002-2">[DOI]</a>.</p></li>
<li id="id10122"><p>Kang Ning. Microbiome and big-data mining. ():197–222, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-16-6795-4_10">[DOI]</a>.</p></li>
<li id="id10123"><p>Albert Fina, Pau Ferrer, and Joan Albiol. Supplementary materials of annex i. high throughput stationary 13c-based metabolic flux analysis of pichia pastoris strains. phd thesis albert fina romero. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.5565/ddd.uab.cat/263045">[DOI]</a>.</p></li>
<li id="id10309"><p>Catherine Bjerre Collin, Tom Gebhardt, Martin Golebiewski, Tugce Karaderi, Maximilian Hillemanns, Faiz M. Khan, Ali Salehzadeh‐Yazdi, Marc W. Kirschner, Sylvia Krobitsch, and Lars Kuepfer. Computational models for clinical applications in personalized medicine—guidelines and recommendations for data integration and model validation. 12(2):166–166, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jpm12020166">[DOI]</a>.</p></li>
<li id="id10324"><p>Subham Choudhury, Michaël Moret, Pierre Salvy, Daniel Weilandt, Vassily Hatzimanikatis, and Ljubiša Mišković. Reconstructing kinetic models for dynamical studies of metabolism using generative adversarial networks. 4(8):710–719, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42256-022-00519-y">[DOI]</a>.</p></li>
<li id="id10343"><p>Nasser Delangiz, Sajad Aliyar, Neda Pashapoor, Khatereh Nobaharan, Behnam Asgari Lajayer, and Susana Rodríguez‐Couto. Can polymer-degrading microorganisms solve the bottleneck of plastics’ environmental challenges? 294():133709–133709, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.chemosphere.2022.133709">[DOI]</a>.</p></li>
<li id="id10347"><p>Jeffry C. Granados, Kian Falah, Imhoi Koo, Ethan W. Morgan, Gary H. Perdew, Andrew D. Patterson, Neema Jamshidi, and Sanjay K. Nigám. Ahr is a master regulator of diverse pathways in endogenous metabolism. 12(1):16625–16625, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-20572-2">[DOI]</a>.</p></li>
<li id="id10349"><p>Chao Ye, Xinyu Wei, Tian‐Qiong Shi, Xiao‐Man Sun, Nan Xu, Cong Gao, and Wei Zou. Genome-scale metabolic network models: from first-generation to next-generation. 106(13-16):4907–4920, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-022-12066-y">[DOI]</a>.</p></li>
<li id="id10351"><p>Ceri Proffitt, Gholamreza Bidkhori, Sunjae Lee, Abdellah Tebani, Adil Mardinoğlu, Mathias Uhlén, David L. Moyes, and Saeed Shoaie. Genome-scale metabolic modelling of the human gut microbiome reveals changes in the glyoxylate and dicarboxylate metabolism in metabolic disorders. 25(7):104513–104513, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2022.104513">[DOI]</a>.</p></li>
<li id="id10608"><p>Lokanand Koduru, Meiyappan Lakshmanan, Yi Qing Lee, Pooi-Leng Ho, Pei Yu Lim, Wei Xuan Ler, Say Kong Ng, Dongseok Kim, Doo‐Sang Park, M. Shakeera Banu, Dave Siak‐Wei Ow, and Dong‐Yup Lee. Systematic evaluation of genome-wide metabolic landscapes in lactic acid bacteria reveals diet- and strain-specific probiotic idiosyncrasies. 41(10):111735–111735, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2022.111735">[DOI]</a>.</p></li>
<li id="id10612"><p>Kaspar Valgepea, Gert Talbo, Nobuaki Takemori, Ayako Takemori, Christina Ludwig, Vishnuvardhan Mahamkali, Alexander P. Mueller, Ryan Tappel, Michael Köpke, Séan D. Simpson, Lars K. Nielsen, and Esteban Marcellin. Absolute proteome quantification in the gas-fermenting acetogen &lt;i&gt;clostridium autoethanogenum&lt;/i&gt;. 7(2):e0002622–e0002622, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00026-22">[DOI]</a>.</p></li>
<li id="id10622"><p>Lorena Azevedo de Lima, Henri Ingelman, Kush Brahmbhatt, Kristina Reinmets, Craig Barry, Audrey Harris, Esteban Marcellin, Michael Köpke, and Kaspar Valgepea. Faster growth enhances low carbon fuel and chemical production through gas fermentation. 10():879578–879578, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2022.879578">[DOI]</a>.</p></li>
<li id="id10629"><p>Faiz Khan Mohammad, Meghana V. Palukuri, Shruti Shivakumar, Raghunathan Rengaswamy, and Swagatika Sahoo. A computational framework for studying gut-brain axis in autism spectrum disorder. 13():760753–760753, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2022.760753">[DOI]</a>.</p></li>
<li id="id10634"><p>Jun Feng, Xiaolong Guo, Feifei Cai, Hongxin Fu, and Jufang Wang. Model-based driving mechanism analysis for butyric acid production in clostridium tyrobutyricum. 15(1):71–71, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-022-02169-z">[DOI]</a>.</p></li>
<li id="id10638"><p>Neema Jamshidi and Sanjay K. Nigám. Drug transporters oat1 and oat3 have specific effects on multiple organs and gut microbiome as revealed by contextualized metabolic network reconstructions. 12(1):18308–18308, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-21091-w">[DOI]</a>.</p></li>
<li id="id10639"><p>Juan D. Tibocha‐Bonilla, Cristal Zúñiga, Asama Lekbua, Colton J. Lloyd, Kevin Rychel, Katie Short, and Karsten Zengler. Predicting stress response and improved protein overproduction in bacillus subtilis. 8(1):50–50, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-022-00259-0">[DOI]</a>.</p></li>
<li id="id10640"><p>Craig Woods, Christopher M. Humphreys, Claudio Tomi-Andrino, Anne M. Henstra, Michael Köpke, Séan D. Simpson, Klaus Winzer, and Nigel P. Minton. Required gene set for autotrophic growth of &lt;i&gt;clostridium autoethanogenum&lt;/i&gt;. 88(7):e0247921–e0247921, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.02479-21">[DOI]</a>.</p></li>
<li id="id10645"><p>Lidan Bai, Qi You, Chenyang Zhang, Jun Sun, Long Liu, Hengyang Lu, and Qidong Chen. Advances and applications of machine learning and intelligent optimization algorithms in genome-scale metabolic network models. 3(2):193–206, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s43393-022-00115-6">[DOI]</a>.</p></li>
<li id="id10798"><p>Jørn Carlsen, H Henriksen, Igor Marín de Mas, and Pär I. Johansson. An explorative metabolomic analysis of the endothelium in pulmonary hypertension. 12(1):13284–13284, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-17374-x">[DOI]</a>.</p></li>
<li id="id10804"><p>Nahikari López-López, David San León, Sonia de Castro, Roberto Díez‐Martínez, Manuel Iglesias-Bexiga, María‐José Camarasa, Margarita Menéndez, Juan Nogales, and Junkal Garmendia. Interrogation of essentiality in the reconstructed haemophilus influenzae metabolic network identifies lipid metabolism antimicrobial targets: preclinical evaluation of a fabh β-ketoacyl-acp synthase inhibitor. 7(2):e0145921–e0145921, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01459-21">[DOI]</a>.</p></li>
<li id="id10808"><p>Abhijit Paul, Salman Azhar, Phonindra Nath Das, Nandadulal Bairagi, and Samrat Chatterjee. Elucidating the metabolic characteristics of pancreatic β-cells from patients with type 2 diabetes (t2d) using a genome-scale metabolic modeling. 144():105365–105365, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2022.105365">[DOI]</a>.</p></li>
<li id="id10809"><p>Claudio Tomi-Andrino, Alina Pandele, Klaus Winzer, John R. King, Ruman Rahman, and Dong‐Hyun Kim. Metabolic modeling-based drug repurposing in glioblastoma. 12(1):11189–11189, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-14721-w">[DOI]</a>.</p></li>
<li id="id10816"><p>Bingqing He, Chen Cai, Tim McCubbin, Jorge Carrasco Muriel, Nikolaus Sonnenschein, Shihu Hu, Zhiguo Yuan, and Esteban Marcellin. A genome-scale metabolic model of methanoperedens nitroreducens: assessing bioenergetics and thermodynamic feasibility. 12(4):314–314, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo12040314">[DOI]</a>.</p></li>
<li id="id10817"><p>Merve Kutay, Devrim Gözüaçık, and Tunahan Çakır. Cancer recurrence and omics: metabolic signatures of cancer dormancy revealed by transcriptome mapping of genome-scale networks. 26(5):270–279, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/omi.2022.0008">[DOI]</a>.</p></li>
<li id="id10818"><p>Mahsa Mekanik, Reza Fotovat, Ehsan Motamedian, and Vahab Jafarian. Improvement of lutein production in auxenochlorella protothecoides using its genome-scale metabolic model and a system-oriented approach. 195(2):889–904, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12010-022-04186-y">[DOI]</a>.</p></li>
<li id="id10822"><p>Aidin Behravan, Atieh Hashemi, and Sayed‐Amir Marashi. A constraint-based modeling approach to reach an improved chemically defined minimal medium for recombinant antiepex-scfv production by escherichia coli. 179():108339–108339, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2022.108339">[DOI]</a>.</p></li>
<li id="id10823"><p>Kaustubh R. Sawant, Aditya Sarnaik, Prashant Savvashe, Nima Hajinajaf, Parker Poole, Arul M. Varman, Arvind Lali, and Reena Pandit. One cell-two wells bio-refinery: demonstrating cyanobacterial chassis for co-production of heterologous and natural hydrocarbons. 363():127921–127921, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2022.127921">[DOI]</a>.</p></li>
<li id="id10906"><p>Joshua E. Goldford, Ashish B. George, Avi I. Flamholz, and Daniel Segrè. Protein cost minimization promotes the emergence of coenzyme redundancy. 119(14):e2110787119–e2110787119, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2110787119">[DOI]</a>.</p></li>
<li id="id10913"><p>Ryutaro Kawai, Yoshihiro Toya, and Hiroshi Shimizu. Metabolic pathway design for growth-associated phenylalanine production using synthetically designed mutualism. 45(9):1539–1546, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-022-02762-4">[DOI]</a>.</p></li>
<li id="id10914"><p>Kristina Simons, Olivia Thibault, Rayna Nolen, Yu Umeki, Patricia C. Faulkner, Lene H. Petersen, and David Hala. A systems toxicological analysis of oxybenzone effects on the metabolic physiology of embryo-larval zebrafish (danio rerio). 9():100308–100308, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.envadv.2022.100308">[DOI]</a>.</p></li>
<li id="id10951"><p>Subham Choudhury, Michaël Moret, Pierre Salvy, Daniel Weilandt, Vassily Hatzimanikatis, and Ljubiša Mišković. Reconstructing kinetic models for dynamical studies of metabolism using generative adversarial networks. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.01.06.475020">[DOI]</a>.</p></li>
<li id="id10952"><p>Soumendu Mahapatra, Smrutishree Mohanty, Rasmita Mishra, and Punit Prasad. An overview of cancer and the human microbiome. 191(1):83–139, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.pmbts.2022.07.007">[DOI]</a>.</p></li>
<li id="id10961"><p>Amirhossein Shahbazinia, Saber Salehkaleybar, and Matin Hashemi. Paralingam: parallel causal structure learning for linear non-gaussian acyclic models. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4261811">[DOI]</a>.</p></li>
<li id="id10962"><p>Chen Yang, Boyuan Xue, Yiming Zhang, Shaojie Wang, and Haijia Su. Metabolic flux simulation of microbial systems based on optimal planning algorithms. 4(2):146–159, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gce.2022.04.003">[DOI]</a>.</p></li>
<li id="id11007"><p>Ehsan Salehabadi, Ehsan Motamedian, and Seyed Abbas Shojaosadati. Reconstruction of a generic genome-scale metabolic network for chicken: investigating network connectivity and finding potential biomarkers. 17(3):e0254270–e0254270, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0254270">[DOI]</a>.</p></li>
<li id="id11042"><p>Daniela Reyes-González, Haydee De Luna-Valenciano, José Utrilla, Michael Sieber, Rafael Peña‐Miller, and Ayari Fuentes-Hernández. Dynamic proteome allocation regulates the profile of interaction of auxotrophic bacterial consortia. 9(5):212008–212008, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rsos.212008">[DOI]</a>.</p></li>
<li id="id11043"><p>Aidin Behravan, Atieh Hashemi, Sayed‐Amir Marashi, and Hamideh Fouladiha. Genome-scale metabolic model-based engineering of escherichia coli enhances recombinant single-chain antibody fragment production. 44(10):1231–1242, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-022-03301-7">[DOI]</a>.</p></li>
<li id="id11080"><p>Fernández-Niño Miguel and Burgos-Toro Daniela. Engineering microbial biofactories for a sustainable future. ():25–58, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-91601-1.00003-1">[DOI]</a>.</p></li>
<li id="id11315"><p>Dalal Kasir, Sébastien Besseau, Marc Clastre, Audrey Oudin, Monzer Hamzé, Vincent Courdavault, Marwan Osman, and Nicolas Papon. Synthetic biology in the candida (ctg) clade. ():337–375, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-89680-5_12">[DOI]</a>.</p></li>
<li id="id11335"><p>Aidin Behravan, Atieh Hashemi, Sayed‐Amir Marashi, and Hamideh Fouladiha. Genome- scale metabolic model based engineering of escherichia coli enhances recombinant single chain antibody fragment production. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-412220/v2">[DOI]</a>.</p></li>
<li id="id11337"><p>Clémence FRIOUX and Anne SIEGEL. Problèmes d’optimisation combinatoire pour l’étude du métabolisme. ():51–96, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.51926/iste.9029.ch2">[DOI]</a>.</p></li>
<li id="id11338"><p>Carlos E. Mejía-Gómez, Rigoberto Ríos‐Estepa, Luis A González, and Norman Balcázar. An experimental and in silico analysis of lacticaseibacillus paracasei isolated from whey shows an association between lactate production and amino acid catabolism. 94(2):e20211071–e20211071, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1590/0001-3765202220211071">[DOI]</a>.</p></li>
<li id="id11339"><p>Müberra Fatma Cesur, Kiran Raosaheb Patil, and Tunahan Çakır. A genome-scale metabolic model of &lt;i&gt;drosophila melanogaster&lt;/i&gt; for integrative analysis of brain diseases. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.08.22.504778">[DOI]</a>.</p></li>
<li id="id11340"><p>Kristina Simons, Olivia Thibault, Rayna Nolen, Yu Umeki, Patricia Faulkner, Lene H. Petersen, and David Hala. A systems toxicological analysis of oxybenzone effects on the metabolic physiology of embryo-larval zebrafish (danio rerio). ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4200128">[DOI]</a>.</p></li>
<li id="id11341"><p>Sriya Priyadarshani, Paritosh Patel, and Sushree Sangita Sahoo. Role of system biology in microbial system. ():43–56, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-19-3979-2_2">[DOI]</a>.</p></li>
<li id="id11343"><p>Maureen C. McCann and Patrick M. Reed. U.s. scientific leadership: addressing energy, ecosystems, climate, and sustainable prosperity. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.2172/1922095">[DOI]</a>.</p></li>
<li id="id11381"><p>Feiran Li, Le Yuan, Hongzhong Lu, Gang Li, Yu Chen, Martin K. M. Engqvist, Eduard J. Kerkhoven, and Jens Nielsen. Deep learning-based kcat prediction enables improved enzyme-constrained model reconstruction. 5(8):662–672, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41929-022-00798-z">[DOI]</a>.</p></li>
<li id="id11392"><p>Dominik Ternes, Mina Tsenkova, Vitaly I. Pozdeev, Marianne Meyers, Eric Koncina, Sura Atatri, Martine Schmitz, Jessica Karta, Maryse Schmoetten, Almut Heinken, Fabien Rodriguez, Catherine Delbrouck, Anthoula Gaigneaux, Aurélien Ginolhac, Tam T. D. Nguyen, Léa Grandmougin, Audrey Frachet-Bour, Camille Martin‐Gallausiaux, Maria Pires Pacheco, Lorie Neuberger-Castillo, Paulo Miranda, Nikolaus Zuegel, Jean-Yves Ferrand, Manon Gantenbein, Thomas Sauter, Daniel J. Slade, Ines Thiele, Johannes Meiser, Serge Haan, Paul Wilmes, and Elisabeth Letellier. The gut microbial metabolite formate exacerbates colorectal cancer progression. 4(4):458–475, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42255-022-00558-0">[DOI]</a>.</p></li>
<li id="id11399"><p>Ruibing Chen, Jiaoqi Gao, Wei Yu, Xianghui Chen, Xiaoxin Zhai, Yu Chen, Lei Zhang, and Yongjin J. Zhou. Engineering cofactor supply and recycling to drive phenolic acid biosynthesis in yeast. 18(5):520–529, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41589-022-01014-6">[DOI]</a>.</p></li>
<li id="id11414"><p>Iván Domenzain, Benjamín J. Sánchez, Mihail Anton, Eduard J. Kerkhoven, Aarón Millán-Oropeza, Céline Henry, Verena Siewers, John P. Morrissey, Nikolaus Sonnenschein, and Jens Nielsen. Reconstruction of a catalogue of genome-scale metabolic models with enzymatic constraints using gecko 2.0. 13(1):3766–3766, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-31421-1">[DOI]</a>.</p></li>
<li id="id11417"><p>Jason Yu, Clara Correia‐Melo, Francisco Zorrilla, Lucía Herrera-Domínguez, Mary Wu, Johannes Hartl, Kate Campbell, Sonja Blasche, Marco Kreidl, Anna-Sophia Egger, Christoph B. Messner, Vadim Demichev, Anja Freiwald, Michael Mülleder, Michael Howell, Judith Berman, Kiran Raosaheb Patil, Mohammad Tauqeer Alam, and Markus Ralser. Microbial communities form rich extracellular metabolomes that foster metabolic interactions and promote drug tolerance. 7(4):542–555, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-022-01072-5">[DOI]</a>.</p></li>
<li id="id11418"><p>Yameng Xu, Xinglong Wang, Chenyang Zhang, Xuan Zhou, Xianhao Xu, Luyao Han, Xueqin Lv, Yanfeng Liu, Song Liu, Jianghua Li, Guocheng Du, Jian Chen, Rodrigo Ledesma‐Amaro, and Long Liu. De novo biosynthesis of rubusoside and rebaudiosides in engineered yeasts. 13(1):3040–3040, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-30826-2">[DOI]</a>.</p></li>
<li id="id11423"><p>Philipp Keller, Michael Reiter, Patrick Kiefer, Thomas Gassler, Lucas Hemmerle, Philipp Christen, Εlad Noor, and Julia A. Vorholt. Generation of an escherichia coli strain growing on methanol via the ribulose monophosphate cycle. 13(1):5243–5243, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-32744-9">[DOI]</a>.</p></li>
<li id="id11425"><p>Laura M. Helleckes, Johannes Hemmerich, Wolfgang Wiechert, Eric von Lieres, and Alexander Grünberger. Machine learning in bioprocess development: from promise to practice. 41(6):817–835, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2022.10.010">[DOI]</a>.</p></li>
<li id="id11434"><p>Robert D. Hall, John C. D’Auria, A. C. Silva Ferreira, Yves Gibon, Dariusz Kruszka, Puneet Mishra, and H.J. van de Zedde. High-throughput plant phenotyping: a role for metabolomics? 27(6):549–563, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tplants.2022.02.001">[DOI]</a>.</p></li>
<li id="id11435"><p>Xuan Cao, Wei Yu, Yu Chen, Shan Yang, Zongbao K. Zhao, Jens Nielsen, Hongwei Luan, and Yongjin J. Zhou. Engineering yeast for high-level production of diterpenoid sclareol. 75():19–28, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.11.002">[DOI]</a>.</p></li>
<li id="id11436"><p>Priyanka Baloni, Matthias Arnold, Luna Buitrago, Kwangsik Nho, Herman Moreno, Kevin Huynh, Barbara Brauner, Gregory Louie, Alexandra Kueider‐Paisley, Karsten Suhre, Andrew J. Saykin, Kim Ekroos, Peter J. Meikle, Leroy Hood, Nathan D. Price, Matthias Arnold, Colette Blach, Rima Kaddurah‐Daouk, Murali Doraiswamy, Siamak MahmoudianDehkordi, Kathleen A. Welsh‐Bohmer, Brenda L. Plassman, Jan Krumsiek, Richa Batra, Andrew J. Saykin, Jingwen Yan, Shannon L. Risacher, Peter J. Meikle, Tingting Wang, M. Arfan Ikram, Shahzad Ahmad, Thomas Hankemeier, Ivan A. Hernandez, Almut Heinken, Filippo Martinelli Boneschi, Ines Thiele, Johannes Hertel, Tim Hensen, Tim Hulshof, Lindsay A. Farrer, Rhoda Au, Wendy Wei Qiao Qiu, Thor D. Stein, Naama Karu, Kamil Borkowski, John C. Newman, Wei Jia, Guoxiang Xie, Jingye Wang, Runmin Wei, Dan Rader, Mitchel A. Kling, Leslie M. Shaw, P. Murali Doraiswamy, Cory C. Funk, A. Iván Hernández, Gabi Kastenmüller, Rebecca Baillie, Xianlin Han, and Rima Kaddurah‐Daouk. Multi-omic analyses characterize the ceramide/sphingomyelin pathway as a therapeutic target in alzheimer’s disease. 5(1):1074–1074, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42003-022-04011-6">[DOI]</a>.</p></li>
<li id="id11441"><p>Yuan‐Hang Du, Min-Yu Wang, Linhui Yang, Lingling Tong, Dong‐Sheng Guo, and Xiao‐Jun Ji. Optimization and scale-up of fermentation processes driven by models. 9(9):473–473, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering9090473">[DOI]</a>.</p></li>
<li id="id11446"><p>Feiran Li, Yu Chen, Qi Qi, Yanyan Wang, Le Yuan, Mingtao Huang, Ibrahim E. Elsemman, Amir Feizi, Eduard J. Kerkhoven, and Jens Nielsen. Improving recombinant protein production by yeast through genome-scale modeling using proteome constraints. 13(1):2969–2969, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-30689-7">[DOI]</a>.</p></li>
<li id="id11451"><p>Shengbo Wu, Jie Feng, Chunjiang Liu, Hao Wu, Zekai Qiu, Jianjun Ge, Shuyang Sun, Xia Hong, Yukun Li, Xiaona Wang, Aidong Yang, Fei Guo, and Jianjun Qiao. Machine learning aided construction of the quorum sensing communication network for human gut microbiota. 13(1):3079–3079, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-30741-6">[DOI]</a>.</p></li>
<li id="id11459"><p>Benjamin P. Kellman, Anne Richelle, Jeong‐Yeh Yang, Digantkumar Chapla, Austin W.T. Chiang, Julia A. Najera, Chenguang Liang, Annalee Fürst, Bokan Bao, Natália Massaco Koga, Mahmoud A. Mohammad, Anders Bech Bruntse, Morey W. Haymond, Kelley W. Moremen, Lars Bode, and Nathan E. Lewis. Elucidating human milk oligosaccharide biosynthetic genes through network-based multi-omics integration. 13(1):2455–2455, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-29867-4">[DOI]</a>.</p></li>
<li id="id11460"><p>Zhendong Li, Cong Gao, Chao Ye, Liang Guo, Jia Liu, Xiulai Chen, Wei Song, Jing Wu, and Li Liu. Systems engineering of escherichia coli for high-level shikimate production. 75():1–11, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.10.010">[DOI]</a>.</p></li>
<li id="id11463"><p>Kunle Odunsi, Feng Qian, Amit A. Lugade, Han Yu, Melissa A. Geller, Steven P. Fling, Judith C. Kaiser, Andreanne M. Lacroix, Leonard D’Amico, Nirasha Ramchurren, Chihiro Morishima, Mary L. Disis, Lucas Dennis, Patrick Danaher, Sarah Warren, Van Anh Nguyen, Sudharshan Ravi, Takemasa Tsuji, Spencer R. Rosario, Wenjuan Zha, Alan D. Hutson, Song Liu, Shashikant Lele, Emese Zsíros, AJ Robert McGray, Jessie L Chiello, Richard C. Koya, Thinle Chodon, Carl D. Morrison, Vasanta Putluri, Nagireddy Putluri, Donald E. Mager, Rudiyanto Gunawan, Martin A. Cheever, Sebastiano Battaglia, and Junko Matsuzaki. Metabolic adaptation of ovarian tumors in patients treated with an ido1 inhibitor constrains antitumor immune responses. 14(636):eabg8402–eabg8402, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/scitranslmed.abg8402">[DOI]</a>.</p></li>
<li id="id11472"><p>Almut Heinken and Ines Thiele. Microbiome modelling toolbox 2.0: efficient, tractable modelling of microbiome communities. 38(8):2367–2368, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac082">[DOI]</a>.</p></li>
<li id="id11476"><p>Yougan Cheng, Ronny Straube, Abed E. Alnaif, Lu Huang, Tarek A. Leil, and Brian J. Schmidt. Virtual populations for quantitative systems pharmacology models. 2486():129–179, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2265-0_8">[DOI]</a>.</p></li>
<li id="id11480"><p>Maxime R.F. Gosselin, Virginie Mournetas, Małgorzata Borczyk, Suraj Verma, Annalisa Occhipinti, Justyna Róg, Łukasz Bożycki, Michał Korostyński, Samuel C. Robson, Claudio Angione, Christian Pinset, and Dariusz C. Górecki. Loss of full-length dystrophin expression results in major cell-autonomous abnormalities in proliferating myoblasts. 11():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.75521">[DOI]</a>.</p></li>
<li id="id11481"><p>Anoop T. Ambikan, Hong Yang, Shuba Krishnan, Sara Svensson Akusjärvi, Soham Gupta, Magda Lourda, Maike Sperk, Muhammad Arif, Cheng Zhang, Hampus Nordqvist, Sivasankaran Munusamy Ponnan, Anders Sönnerborg, Carl Johan Treutiger, Liam O’Mahony, Adil Mardinoğlu, Rui Benfeitas, and Ujjwal Neogi. Multi-omics personalized network analyses highlight progressive disruption of central metabolism associated with covid-19 severity. 13(8):665–681.e4, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2022.06.006">[DOI]</a>.</p></li>
<li id="id11488"><p>Santosh Lamichhane, Partho Sen, Alex M. Dickens, Marina Amaral Alves, Taina Härkönen, Jarno Honkanen, Tommi Vatanen, Ramnik J. Xavier, Tuulia Hyötyläinen, Mikael Knip, and Matej Orešič. Dysregulation of secondary bile acid metabolism precedes islet autoimmunity and type 1 diabetes. 3(10):100762–100762, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xcrm.2022.100762">[DOI]</a>.</p></li>
<li id="id11489"><p>Nicole Pearcy, Marco Garavaglia, Thomas Millat, James P. Gilbert, Yoseb Song, Hassan Hartman, Craig Woods, Claudio Tomi-Andrino, Rajesh Reddy Bommareddy, Byung‐Kwan Cho, David A. Fell, Mark G. Poolman, John R. King, Klaus Winzer, Jamie Twycross, and Nigel P. Minton. A genome-scale metabolic model of cupriavidus necator h16 integrated with tradis and transcriptomic data reveals metabolic insights for biotechnological applications. 18(5):e1010106–e1010106, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1010106">[DOI]</a>.</p></li>
<li id="id11490"><p>Dalimil Bujdoš, Barbora Popelářová, Daniel C. Volke, Pablo I. Nikel, Nikolaus Sonnenschein, and Pavel Dvořák. Engineering of pseudomonas putida for accelerated co-utilization of glucose and cellobiose yields aerobic overproduction of pyruvate explained by an upgraded metabolic model. 75():29–46, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.10.011">[DOI]</a>.</p></li>
<li id="id11494"><p>Belén Hirmas, Naschla Gasaly, Guillermo Orellana, Marco A. Vega, Pedro A. Saa, Martín Gotteland, and Daniel Garrido. Metabolic modeling and bidirectional culturing of two gut microbes reveal cross-feeding interactions and protective effects on intestinal cells. 7(5):e0064622–e0064622, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00646-22">[DOI]</a>.</p></li>
<li id="id11505"><p>Raymond T. Ng, J. Lee, Priyanka Baloni, Christian Diener, James R. Heath, and Yapeng Su. Constraint-based reconstruction and analyses of metabolic models: open-source python tools and applications to cancer. 12():914594–914594, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fonc.2022.914594">[DOI]</a>.</p></li>
<li id="id11533"><p>Lisha Qu, Xiang Xiu, Guoyun Sun, Chenyang Zhang, Haiquan Yang, Yanfeng Liu, Jianghua Li, Guocheng Du, Xueqin Lv, and Long Liu. Engineered yeast for efficient de novo synthesis of 7‐dehydrocholesterol. 119(5):1278–1289, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28055">[DOI]</a>.</p></li>
<li id="id11542"><p>John R. Casey, Rene Boiteau, Martin K. M. Engqvist, Zoe V. Finkel, Gang Li, Justin D. Liefer, Christian L. Müller, Nathalie Munoz Munoz, and Michael J. Follows. Basin-scale biogeography of marine phytoplankton reflects cellular-scale optimization of metabolism and physiology. 8(3):eabl4930–eabl4930, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/sciadv.abl4930">[DOI]</a>.</p></li>
<li id="id11551"><p>Abhinav Achreja, Tao Yu, Anjali Mittal, Srinadh Choppara, Olamide Animasahun, Minal Nenwani, Fulei Wuchu, Noah Meurs, Aradhana Mohan, Jin Heon Jeon, Itisam Sarangi, Anusha Jayaraman, Sarah Owen, Reva Kulkarni, Michele L. Dziubinski, Frank Weinberg, Hye Kyong Kweon, Chitra Subramanian, Max S. Wicha, Sofía D. Merajver, Sunitha Nagrath, Kathleen R. Cho, Analisa DiFeo, Xiongbin Lu, and Deepak Nagrath. Metabolic collateral lethal target identification reveals mthfd2 paralogue dependency in ovarian cancer. 4(9):1119–1137, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42255-022-00636-3">[DOI]</a>.</p></li>
<li id="id11559"><p>Prerna Bhalla, Raghunathan Rengaswamy, Devarajan Karunagaran, G. K. Suraishkumar, and Swagatika Sahoo. Metabolic modeling of host–microbe interactions for therapeutics in colorectal cancer. 8(1):1–1, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-021-00210-9">[DOI]</a>.</p></li>
<li id="id11560"><p>Antoine Régimbeau, Marko Budinich, Abdelhalim Larhlimi, Juan José Pierella Karlusich, Olivier Aumont, Laurent Mémery, Chris Bowler, and Damien Eveillard. Contribution of genome‐scale metabolic modelling to niche theory. 25(6):1352–1364, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/ele.13954">[DOI]</a>.</p></li>
<li id="id11571"><p>Vetle Simensen, Christian Schulz, Emil Karlsen, Signe Bråtelund, Idun Burgos, Lilja Brekke Thorfinnsdottir, Laura García-Calvo, Per Bruheim, and Eivind Almaas. Experimental determination of escherichia coli biomass composition for constraint-based metabolic modeling. 17(1):e0262450–e0262450, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0262450">[DOI]</a>.</p></li>
<li id="id11572"><p>Philipp Schneider, Pavlos Stephanos Bekiaris, Axel von Kamp, and Steffen Klamt. Straindesign: a comprehensive python package for computational design of metabolic networks. 38(21):4981–4983, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac632">[DOI]</a>.</p></li>
<li id="id11576"><p>João R. C. Ramos, Gil Poiares‐Oliveira, Patrick Dumas, and Rui Oliveira. Genome-scale modeling of chinese hamster ovary cells by hybrid semi-parametric flux balance analysis. 45(11):1889–1904, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-022-02795-9">[DOI]</a>.</p></li>
<li id="id11578"><p>Francesco Balzerani, Daniel Hinojosa-Nogueira, Xabier Cendoya, Telmo Blasco, Sergio Pérez-Burillo, Iñigo Apaolaza, M. Pilar Francino, José Ángel Rufián‐Henares, and Francisco J. Planes. Prediction of degradation pathways of phenolic compounds in the human gut microbiota through enzyme promiscuity methods. 8(1):24–24, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-022-00234-9">[DOI]</a>.</p></li>
<li id="id11579"><p>Ines Thiele and Ronan M. T. Fleming. Whole-body metabolic modelling predicts isoleucine dependency of sars-cov-2 replication. 20():4098–4109, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2022.07.019">[DOI]</a>.</p></li>
<li id="id11580"><p>Partho Sen, Olivier Govaere, Tim Sinioja, Aidan McGlinchey, Dawei Geng, Vlad Ratziu, Elisabetta Bugianesi, Jörn M. Schattenberg, Antonio Vidal‐Puig, Michael Allison, Simon Cockell, Ann K. Daly, Tuulia Hyötyläinen, Quentin M. Anstee, and Matej Orešič. Quantitative modeling of human liver reveals dysregulation of glycosphingolipid pathways in nonalcoholic fatty liver disease. 25(9):104949–104949, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2022.104949">[DOI]</a>.</p></li>
<li id="id11581"><p>Zhepu Ruan, Mengjun Xu, Youwen Xing, Qi Jiang, Bingang Yang, Jiandong Jiang, and Xihui Xu. Interspecies metabolic interactions in a synergistic consortium drive efficient degradation of the herbicide bromoxynil octanoate. 70(37):11613–11622, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.jafc.2c03057">[DOI]</a>.</p></li>
<li id="id11585"><p>Carles Foguet, Yu Xu, Scott C. Ritchie, Samuel A. Lambert, Elodie Persyn, Artika P. Nath, Emma E. Davenport, David J. Roberts, Dirk S. Paul, Emanuele Di Angelantonio, John Danesh, Adam S. Butterworth, Christopher Yau, and Michael Inouye. Genetically personalised organ-specific metabolic models in health and disease. 13(1):7356–7356, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-022-35017-7">[DOI]</a>.</p></li>
<li id="id11593"><p>Satyajit Beura, Pritam Kundu, Amit Kumar Das, and Amit Ghosh. Metagenome-scale community metabolic modelling for understanding the role of gut microbiota in human health. 149():105997–105997, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2022.105997">[DOI]</a>.</p></li>
<li id="id11600"><p>Saratram Gopalakrishnan, Chintan Joshi, Miguel Á. Valderrama-Gómez, Elçin Içten, Pablo Rolandi, William Johnson, Cleo Kontoravdi, and Nathan E. Lewis. Guidelines for extracting biologically relevant context-specific metabolic models using gene expression data. 75():181–191, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.12.003">[DOI]</a>.</p></li>
<li id="id11604"><p>Jong Kwang Hong, Dong‐Hyuk Choi, Seo‐Young Park, Yaron Silberberg, Fumi Shozui, E. Nakamura, Takashi Kayahara, and Dong‐Yup Lee. Data-driven and model-guided systematic framework for media development in cho cell culture. 73():114–123, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.07.003">[DOI]</a>.</p></li>
<li id="id11605"><p>Francine Piubeli. Systems biology: new insight into antibiotic resistance. 10(12):2362–2362, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms10122362">[DOI]</a>.</p></li>
<li id="id11613"><p>Antonino Baez, Ashish K. Sharma, A. L. Bryukhanov, Eric Anderson, Leba Rudack, Roberto Olivares‐Hernández, David N. Quan, and Joseph Shiloach. Iron availability enhances the cellular energetics of aerobic escherichia coli cultures while upregulating anaerobic respiratory chains. 71():11–20, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.nbt.2022.06.004">[DOI]</a>.</p></li>
<li id="id11614"><p>Gabriela Diaz-Tang, Estefania Marin Meneses, Kavish Patel, Sophia Mirkin, Laura García-Diéguez, Camryn Pajon, Ivana Barraza, Vijay Patel, Helana Ghali, Angelica P. Tracey, Christopher A. Blanar, Allison J. Lopatkin, and Robert P. Smith. Growth productivity as a determinant of the inoculum effect for bactericidal antibiotics. 8(50):, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/sciadv.add0924">[DOI]</a>.</p></li>
<li id="id11618"><p>Padhmanand Sudhakar, Tahila Andrighetti, Sare Verstockt, Clara Caenepeel, Marc Ferrante, João Sabino, Bram Verstockt, and Séverine Vermeire. Integrated analysis of microbe-host interactions in crohn’s disease reveals potential mechanisms of microbial proteins on host gene expression. 25(5):103963–103963, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2022.103963">[DOI]</a>.</p></li>
<li id="id11619"><p>João B. Xavier, Jonathan M. Monk, Saugat Poudel, Charles J. Norsigian, Anand V. Sastry, Chen Liao, José Maurício S. Bento, Marc A. Suchard, Mario L. Arrieta‐Ortiz, Eliza J. R. Peterson, Nitin S. Baliga, Thomas Stoeger, Felicia Ruffin, Reese Richardson, Catherine A. Gao, Thomas D. Horvath, Anthony M. Haag, Qinglong Wu, Tor Savidge, and Michael R. Yeaman. Mathematical models to study the biology of pathogens and the infectious diseases they cause. 25(4):104079–104079, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2022.104079">[DOI]</a>.</p></li>
<li id="id11624"><p>Hock Chuan Yeo, Seo‐Young Park, Tessa Rui Min Tan, Say Kong Ng, Meiyappan Lakshmanan, and Dong‐Yup Lee. Combined multivariate statistical and flux balance analyses uncover media bottlenecks to the growth and productivity of chinese hamster ovary cell cultures. 119(7):1740–1754, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28104">[DOI]</a>.</p></li>
<li id="id11625"><p>Jungyeon Kim, Yu Eun Cheong, Sora Yu, Yong-Su Jin, and Kyoung Heon Kim. Strain engineering and metabolic flux analysis of a probiotic yeast saccharomyces boulardii for metabolizing l-fucose, a mammalian mucin component. 21(1):204–204, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-022-01926-x">[DOI]</a>.</p></li>
<li id="id11628"><p>Qusheng Jin, Qiong Wu, Benjamin M. Shapiro, and Shannon E. McKernan. Limited mechanistic link between the monod equation and methanogen growth: a perspective from metabolic modeling. 10(2):e0225921–e0225921, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.02259-21">[DOI]</a>.</p></li>
<li id="id11629"><p>Alan R. Pacheco, Charlie Pauvert, Dileep Kishore, and Daniel Segrè. Toward fair representations of microbial interactions. 7(5):e0065922–e0065922, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00659-22">[DOI]</a>.</p></li>
<li id="id11630"><p>Paula Jouhten, Dimitrios Konstantinidis, Filipa Pereira, Sergej Andrejev, Kristina Grkovska, Sandra Castillo, Payam Ghiachi, Gemma Beltran, Eivind Almaas, Albert Mas, Jonas Warringer, Ramón González, Pilar Morales, and Kiran Raosaheb Patil. Predictive evolution of metabolic phenotypes using model‐designed environments. 18(10):e10980–e10980, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.202210980">[DOI]</a>.</p></li>
<li id="id11637"><p>Minsuk Kim, Jaeyun Sung, and Nicholas Chia. Resource-allocation constraint governs structure and function of microbial communities in metabolic modeling. 70():12–22, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.12.011">[DOI]</a>.</p></li>
<li id="id11638"><p>Jiangong Lu, Yaokang Wu, Deng Chen, Yanfeng Liu, Xueqin Lv, Jianghua Li, Guocheng Du, and Long Liu. Model-based dynamic engineering of escherichia coli for n-acetylglucosamine overproduction. 3():15–24, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotno.2022.02.001">[DOI]</a>.</p></li>
<li id="id11639"><p>Gabriela Canto-Encalada, Diego Tec-Campos, Juan D. Tibocha‐Bonilla, Karsten Zengler, Alejandro Zepeda, and Cristal Zúñiga. Flux balance analysis of the ammonia-oxidizing bacterium nitrosomonas europaea atcc19718 unravels specific metabolic activities while degrading toxic compounds. 18(2):e1009828–e1009828, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009828">[DOI]</a>.</p></li>
<li id="id11640"><p>Shouyong Jiang, Irene Otero‐Muras, Julio R. Banga, Yong Wang, Marcus Kaiser, and Natalio Krasnogor. Optdesign: identifying optimum design strategies in strain engineering for biochemical production. 11(4):1531–1541, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.1c00610">[DOI]</a>.</p></li>
<li id="id11645"><p>Elle M Barnes and Susannah G. Tringe. Exploring the roles of microbes in facilitating plant adaptation to climate change. 479(3):327–335, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bcj20210793">[DOI]</a>.</p></li>
<li id="id11646"><p>Barbara Bourgade, Christopher M. Humphreys, James Millard, Nigel P. Minton, and M. Ahsanul Islam. Design, analysis, and implementation of a novel biochemical pathway for ethylene glycol production in &lt;i&gt;clostridium autoethanogenum&lt;/i&gt;. 11(5):1790–1800, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.1c00624">[DOI]</a>.</p></li>
<li id="id11647"><p>Simone Schito, Rico Zuchowski, Daniel Bergen, Daniel Strohmeier, Bastian Wollenhaupt, Philipp Menke, Johannes Seiffarth, Katharina Nöh, Dietrich Kohlheyer, Michael Bott, Wolfgang Wiechert, Meike Baumgart, and Stephan Noack. Communities of niche-optimized strains (conos) – design and creation of stable, genome-reduced co-cultures. 73():91–103, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2022.06.004">[DOI]</a>.</p></li>
<li id="id11651"><p>Feng‐Sheng Wang, Kelin Chen, and Sz-Wei Chu. Human/sars-cov-2 genome-scale metabolic modeling to discover potential antiviral targets for covid-19. 133():104273–104273, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtice.2022.104273">[DOI]</a>.</p></li>
<li id="id11652"><p>Marilena D. A. Pantziri and Maria I. Klapa. Standardization of human metabolic stoichiometric models: challenges and directions. 2():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fsysb.2022.899980">[DOI]</a>.</p></li>
<li id="id11658"><p>Romeu Viana, Diogo Couceiro, Tiago Carreiro, Óscar Dias, Isabel Rocha, and Miguel C. Teixeira. A genome-scale metabolic model for the human pathogen candida parapsilosis and early identification of putative novel antifungal drug targets. 13(2):303–303, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/genes13020303">[DOI]</a>.</p></li>
<li id="id11659"><p>Defei Liu, Zixiang Xu, Jingen Li, Qian Liu, Qianqian Yuan, Yanmei Guo, Hongwu Ma, and Chaoguang Tian. Reconstruction and analysis of genome‐scale metabolic model for thermophilic fungus &lt;i&gt;myceliophthora thermophila&lt;/i&gt;. 119(7):1926–1937, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28080">[DOI]</a>.</p></li>
<li id="id11660"><p>Giuseppe Magazzù, Guido Zampieri, and Claudio Angione. Clinical stratification improves the diagnostic accuracy of small omics datasets within machine learning and genome-scale metabolic modelling methods. 151(Pt A):106244–106244, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2022.106244">[DOI]</a>.</p></li>
<li id="id11666"><p>Patrick F. Suthers and Costas D. Maranas. Examining organic acid production potential and growth‐coupled strategies in &lt;i&gt;issatchenkia orientalis&lt;/i&gt; using constraint‐based modeling. 38(5):e3276–e3276, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.3276">[DOI]</a>.</p></li>
<li id="id11680"><p>Teresa J. Clark and Jörg Schwender. Elucidation of triacylglycerol overproduction in the c4 bioenergy crop sorghum bicolor by constraint-based analysis. 13():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2022.787265">[DOI]</a>.</p></li>
<li id="id11681"><p>Müberra Fatma Cesur, Tunahan Çakır, and Pınar Pir. Genome-wide analysis of yeast metabolic cycle through metabolic network models reveals superiority of integrated atac-seq data over rna-seq data. 7(3):e0134721–e0134721, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01347-21">[DOI]</a>.</p></li>
<li id="id11726"><p>Evangelia Vayena, Anush Chiappino-Pepe, Homa MohammadiPeyhani, Yannick Francioli, Noushin Hadadi, Meriç Ataman, Jasmin Hafner, Stavros Pavlou, and Vassily Hatzimanikatis. A workflow for annotating the knowledge gaps in metabolic reconstructions using known and hypothetical reactions. 119(46):e2211197119–e2211197119, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2211197119">[DOI]</a>.</p></li>
<li id="id11739"><p>Feng Xu, Ju Lu, Xiang Ke, Minghao Shao, Mingzhi Huang, and Ju Chu. Reconstruction of the genome-scale metabolic model of saccharopolyspora erythraea and its application in the overproduction of erythromycin. 12(6):509–509, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo12060509">[DOI]</a>.</p></li>
<li id="id11744"><p>Iman Shahidi Pour Savizi, Nader Maghsoudi, Ehsan Motamedian, Nathan E. Lewis, and Seyed Abbas Shojaosadati. Valine feeding reduces ammonia production through rearrangement of metabolic fluxes in central carbon metabolism of cho cells. 106(3):1113–1126, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-021-11755-4">[DOI]</a>.</p></li>
<li id="id11745"><p>Shivendra G. Tewari, Bobby Kwan, Rubayet Elahi, Krithika Rajaram, Jaques Reifman, Sean T. Prigge, Akhil B. Vaidya, and Anders Wallqvist. Metabolic adjustments of blood-stage plasmodium falciparum in response to sublethal pyrazoleamide exposure. 12(1):1167–1167, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-04985-7">[DOI]</a>.</p></li>
<li id="id11746"><p>Kristaps Berzins, Reinis Muiznieks, Matiss Ricards Baumanis, Inese Strazdiņa, Karlis Shvirksts, Santa Prikule, Vytautas Galvanauskas, Daniel Pleißner, Agris Pentjuss, Māra Grūbe, Uldis Kalnenieks, and Egils Stalidzāns. Kinetic and stoichiometric modeling-based analysis of docosahexaenoic acid (dha) production potential by crypthecodinium cohnii from glycerol, glucose and ethanol. 20(2):115–115, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/md20020115">[DOI]</a>.</p></li>
<li id="id11747"><p>Miroslava Čuperlović‐Culf, Thao Nguyen-Tran, and Steffany A. L. Bennett. Machine learning and hybrid methods for metabolic pathway modeling. 2553():417–439, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2617-7_18">[DOI]</a>.</p></li>
<li id="id11748"><p>Tamizhini Loganathan and C. George Priya Doss. The influence of machine learning technologies in gut microbiome research and cancer studies - a review. 311(Pt A):121118–121118, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.lfs.2022.121118">[DOI]</a>.</p></li>
<li id="id11755"><p>Natalia I. García‐Tomsig, Marta Robledo, George C. diCenzo, Alessio Mengoni, Vicenta Millán, Alexandra Peregrina, Alejandro Uceta, and José I. Jiménez‐Zurdo. Pervasive rna regulation of metabolism enhances the root colonization ability of nitrogen-fixing symbiotic α-rhizobia. 13(1):e0357621–e0357621, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.03576-21">[DOI]</a>.</p></li>
<li id="id11756"><p>Richard C. Law, Aliya Lakhani, Samantha O’Keeffe, Sevcan Erşan, and Junyoung O. Park. Integrative metabolic flux analysis reveals an indispensable dimension of phenotypes. 75():102701–102701, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2022.102701">[DOI]</a>.</p></li>
<li id="id11757"><p>Abhishek Subramanian, Pooya Zakeri, Mira Mousa, Halima Alnaqbi, Fatima Alshamsi, Leo Bettoni, Ernesto Damiani, Habiba Alsafar, Yvan Saeys, and Peter Carmeliet. Angiogenesis goes computational – the future way forward to discover new angiogenic targets? 20():5235–5255, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2022.09.019">[DOI]</a>.</p></li>
<li id="id11764"><p>Thomas Sauter, Tamara Bintener, Ali Kishk, Luana Presta, Tessy Prohaska, D Guignard, Ni Zeng, Claudia Cipriani, Sundas Arshad, Thomas Pfau, Patricia Martins Conde, and Maria Pires Pacheco. Project-based learning course on metabolic network modelling in computational systems biology. 18(1):e1009711–e1009711, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009711">[DOI]</a>.</p></li>
<li id="id11765"><p>Stefano Camborda, Jan-Niklas Weder, and Nadine Töpfer. Cobramod: a pathway-centric curation tool for constraint-based metabolic models. 38(9):2654–2656, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac119">[DOI]</a>.</p></li>
<li id="id11766"><p>Parsa Ghadermazi, Angela Re, Luca Ricci, and Siu Hung Joshua Chan. Metabolic engineering interventions for sustainable 2,3-butanediol production in gas-fermenting &lt;i&gt;clostridium autoethanogenum&lt;/i&gt;. 7(2):e0111121–e0111121, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01111-21">[DOI]</a>.</p></li>
<li id="id11767"><p>Carolin C. M. Schulte, Vinoy K. Ramachandran, Antonis Papachristodoulou, and Philip S. Poole. Genome-scale metabolic modelling of lifestyle changes in rhizobium leguminosarum. 7(1):e0097521–e0097521, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00975-21">[DOI]</a>.</p></li>
<li id="id11768"><p>Kadir Kocabaş, Alina Arif, Reaz Uddin, and Tunahan Çakır. Dual transcriptome based reconstruction of salmonella-human integrated metabolic network to screen potential drug targets. 17(5):e0268889–e0268889, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0268889">[DOI]</a>.</p></li>
<li id="id11769"><p>Han Bin, Zeyu Dai, and Zhimin Li. Computer-based design of a cell factory for high-yield cytidine production. 11(12):4123–4133, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.2c00431">[DOI]</a>.</p></li>
<li id="id11778"><p>Seyed Babak Loghmani, Nadine Veith, Sven Sahle, Frank Bergmann, Brett G. Olivier, and Ursula Kummer. Inspecting the solution space of genome-scale metabolic models. 12(1):43–43, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo12010043">[DOI]</a>.</p></li>
<li id="id11779"><p>Tadeja Režen, Alexandre Martins, Miha Mraz, Nikolaj Zimic, Damjana Rozman, and Miha Moškon. Integration of omics data to generate and analyse covid-19 specific genome-scale metabolic models. 145():105428–105428, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2022.105428">[DOI]</a>.</p></li>
<li id="id11780"><p>Philipp Wendering and Zoran Nikoloski. Commit: consideration of metabolite leakage and community composition improves microbial community reconstructions. 18(3):e1009906–e1009906, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009906">[DOI]</a>.</p></li>
<li id="id11781"><p>Philipp Wendering and Zoran Nikoloski. Genome-scale modeling specifies the metabolic capabilities of &lt;i&gt;rhizophagus irregularis&lt;/i&gt;. 7(1):e0121621–e0121621, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01216-21">[DOI]</a>.</p></li>
<li id="id11782"><p>Jiangong Lu, Xinyu Bi, Yanfeng Liu, Xueqin Lv, Jianghua Li, Guocheng Du, and Long Liu. In silico cell factory design driven by comprehensive genome-scale metabolic models: development and challenges. 3(2):207–222, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s43393-022-00117-4">[DOI]</a>.</p></li>
<li id="id11783"><p>Raja Murugan, Archanaa Sundararaghavan, Navdeep Kaur Dhami, Abhijit Mukherjee, and G. K. Suraishkumar. Importance of carbon to nitrogen ratio in microbial cement production: insights through experiments and genome-scale metabolic modelling. 186():108573–108573, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2022.108573">[DOI]</a>.</p></li>
<li id="id11787"><p>Juan C. Villada, Maria F. Duran, Chee Kent Lim, Lisa Y. Stein, and Patrick K. H. Lee. Integrative genome-scale metabolic modeling reveals versatile metabolic strategies for methane utilization in methylomicrobium album bg8. 7(2):e0007322–e0007322, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00073-22">[DOI]</a>.</p></li>
<li id="id11788"><p>Iñigo Apaolaza, Edurne San José‐Eneriz, Luis V. Valcárcel, Xabier Agirre, Felipe Prósper, and Francisco J. Planes. A network-based approach to integrate nutrient microenvironment in the prediction of synthetic lethality in cancer metabolism. 18(3):e1009395–e1009395, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009395">[DOI]</a>.</p></li>
<li id="id11789"><p>Laurence Legon, Christophe Corre, Declan G. Bates, and Ahmad A. Mannan. Gcfront: a tool for determining a pareto front of growth-coupled cell factory designs. 38(14):3657–3659, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac376">[DOI]</a>.</p></li>
<li id="id11790"><p>Sebastian Huß, Rika Judd, Kaan Koper, Hiroshi Maéda, and Zoran Nikoloski. An automated workflow that generates atom mappings for large‐scale metabolic models and its application to &lt;i&gt;arabidopsis thaliana&lt;/i&gt;. 111(5):1486–1500, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/tpj.15903">[DOI]</a>.</p></li>
<li id="id11791"><p>Mehdi Dehghan Manshadi, Payam Setoodeh, and Habil Zare. Rapid-sl identifies synthetic lethal sets with an arbitrary cardinality. 12(1):14022–14022, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-18177-w">[DOI]</a>.</p></li>
<li id="id11792"><p>Brendan Fu‐Long Sieow, Ryan De Sotto, Zhi Ren Darren Seet, In Young Hwang, and Matthew Wook Chang. Synthetic biology meets machine learning. 2553():21–39, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2617-7_2">[DOI]</a>.</p></li>
<li id="id11793"><p>K. B. Arun, A. N. Anoopkumar, Raveendran Sindhu, Parameswaran Binod, Embalil Mathachan Aneesh, Aravind Madhavan, and Mukesh Kumar Awasthi. Synthetic biology for sustainable food ingredients production: recent trends. 3(1):137–149, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s43393-022-00150-3">[DOI]</a>.</p></li>
<li id="id11802"><p>Seungyoon Nam and Yong‐Min Lee. Genome-scale metabolic model analysis of metabolic differences between lauren diffuse and intestinal subtypes in gastric cancer. 14(9):2340–2340, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cancers14092340">[DOI]</a>.</p></li>
<li id="id11803"><p>Pattsarun Cheawchanlertfa, Suwalak Chitcharoen, Nachon Raethong, Qing Liu, Pramote Chumnanpuen, Panyawarin Soommat, Yuanda Song, Mattheos Koffas, Kobkul Laoteng, and Wanwipa Vongsangnak. Enhancing genome-scale model by integrative exometabolome and transcriptome: unveiling carbon assimilation towards sphingolipid biosynthetic capability of cordyceps militaris. 8(8):887–887, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jof8080887">[DOI]</a>.</p></li>
<li id="id11815"><p>Menglei Xia, Di Wang, Yiming Xia, Haijiao Shi, Zhongyu Tian, Yu Zheng, and Min Wang. Oxidoreduction potential controlling for increasing the fermentability of enzymatically hydrolyzed steam-exploded corn stover for butanol production. 21(1):130–130, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-022-01824-2">[DOI]</a>.</p></li>
<li id="id11825"><p>Tobias Müller, Simon Schick, Jonathan Beck, Georg A. Sprenger, and Ralf Takors. Synthetic mutualism in engineered &lt;i&gt;e. coli&lt;/i&gt; mutant strains as functional basis for microbial production consortia. 23(1):e2100158–e2100158, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/elsc.202100158">[DOI]</a>.</p></li>
<li id="id11826"><p>Kathryne C. Ford, Joshua A. M. Kaste, Yair Shachar‐Hill, and Michaela A. TerAvest. Flux-balance analysis and mobile crispri-guided deletion of a conditionally essential gene in &lt;i&gt;shewanella oneidensis&lt;/i&gt; mr-1. 11(10):3405–3413, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.2c00323">[DOI]</a>.</p></li>
<li id="id11827"><p>Hatice Büşra Lüleci̇, Dilara Uzuner, Tunahan Çakır, and Madhav Thambisetty. Computational approaches to assess abnormal metabolism in alzheimer’s disease using transcriptomics. 2561():173–189, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2655-9_9">[DOI]</a>.</p></li>
<li id="id11828"><p>Maria Pires Pacheco, Jimmy Ji, Tessy Prohaska, María Moscardó Garcia, and Thomas Sauter. Scfastcormics: a contextualization algorithm to reconstruct metabolic multi-cell population models from single-cell rnaseq data. 12(12):1211–1211, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo12121211">[DOI]</a>.</p></li>
<li id="id11887"><p>Lokanand Koduru, Meiyappan Lakshmanan, Shawn Hoon, Dong‐Yup Lee, Yuan Kun Lee, and Dave Siak‐Wei Ow. Systems biology of gut microbiota-human receptor interactions: toward anti-inflammatory probiotics. 13():846555–846555, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2022.846555">[DOI]</a>.</p></li>
<li id="id11899"><p>Claudia Cipriani, Maria Pires Pacheco, Ali Kishk, Maryem Wachich, Daniel Abankwa, Elisabeth Schaffner‐Reckinger, and Thomas Sauter. Bruceine d identified as a drug candidate against breast cancer by a novel drug selection pipeline and cell viability assay. 15(2):179–179, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ph15020179">[DOI]</a>.</p></li>
<li id="id11900"><p>Clara Correia‐Melo, Stephan Kamrad, Christoph B. Messner, Roland Tengölics, Lucía Herrera-Domínguez, St John Townsend, Mohammad Tauqeer Alam, Anja Freiwald, Kate Campbell, Simran Kaur Aulakh, Łukasz Szyrwiel, Jason Yu, Aleksej Zelezniak, Vadim Demichev, Michael Muelleder, Balázs Papp, and Markus Ralser. Cell-cell metabolite exchange creates a pro-survival metabolic environment that extends lifespan. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.03.07.483228">[DOI]</a>.</p></li>
<li id="id11901"><p>Eunice A. Ferreira, Catarina C. Pacheco, João S. Rodrigues, Filipe Pinto, Pedro Lamosa, David Fuente, Javier F. Urchueguía, and Paula Tamagnini. Heterologous production of glycine betaine using synechocystis sp. pcc 6803-based chassis lacking native compatible solutes. 9():821075–821075, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2021.821075">[DOI]</a>.</p></li>
<li id="id11902"><p>H Henriksen, Igor Marín de Mas, Helena Herand, Joseph D. Krocker, Charles E. Wade, and Pär I. Johansson. Metabolic systems analysis identifies a novel mechanism contributing to shock in patients with endotheliopathy of trauma (eot) involving thromboxane a2 and ltc4. 15():100115–100115, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mbplus.2022.100115">[DOI]</a>.</p></li>
<li id="id11911"><p>Lavanya Raajaraam and Karthik Raman. A computational framework to identify metabolic engineering strategies for the co-production of metabolites. 9():779405–779405, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2021.779405">[DOI]</a>.</p></li>
<li id="id11921"><p>Carles Foguet, Yu Xu, Scott C. Ritchie, Samuel A. Lambert, Elodie Persyn, Artika P. Nath, Emma E. Davenport, David J. Roberts, Dirk S. Paul, Emanuele Di Angelantonio, John Danesh, Adam S. Butterworth, Christopher Yau, and Michael Inouye. Genetically personalised organ-specific metabolic models in health and disease. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.03.25.22272958">[DOI]</a>.</p></li>
<li id="id11922"><p>Yoseb Song, Jiyun Bae, Jongoh Shin, Sangrak Jin, Seulgi Kang, Hyeonsik Lee, Suhyung Cho, and Byung‐Kwan Cho. Systems biology on acetogenic bacteria for utilizing c1 feedstocks. 180():57–90, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2021_199">[DOI]</a>.</p></li>
<li id="id11923"><p>Supreeta Vijayakumar, Giuseppe Magazzù, Pradip Moon, Annalisa Occhipinti, and Claudio Angione. A practical guide to integrating multimodal machine learning and metabolic modeling. 2399():87–122, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1831-8_5">[DOI]</a>.</p></li>
<li id="id11924"><p>Ali Kishk, Maria Pires Pacheco, Tony Heurtaux, Lasse Sinkkonen, Jun Pang, Sabrina Fritah, Simone P. Niclou, and Thomas Sauter. Review of current human genome-scale metabolic models for brain cancer and neurodegenerative diseases. 11(16):2486–2486, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cells11162486">[DOI]</a>.</p></li>
<li id="id11925"><p>Ines Thiele, German Preciat, and Ronan M. T. Fleming. Metaboannotator: an efficient toolbox to annotate metabolites in genome-scale metabolic reconstructions. 38(20):4831–4832, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btac596">[DOI]</a>.</p></li>
<li id="id11926"><p>Sankalpa Venkatraghavan, Sathvik Anantakrishnan, and Karthik Raman. Probing patterning in microbial consortia with a cellular automaton for spatial organisation. 12(1):17159–17159, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-022-20705-7">[DOI]</a>.</p></li>
<li id="id11939"><p>Philipp Wendering, Marius Arend, Zahra Razaghi-Moghadamkashani, and Zoran Nikoloski. Data integration across conditions improves turnover number estimates and metabolic predictions. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.04.01.486742">[DOI]</a>.</p></li>
<li id="id11940"><p>Jingjing Li, Yifei Gou, Jiarui Yang, Lingxuan Zhao, Bin Wang, Tong Hao, and Jinsheng Sun. Genome-scale metabolic network model of eriocheir sinensis icrab4665 and nutritional requirement analysis. 23(1):475–475, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-022-08698-z">[DOI]</a>.</p></li>
<li id="id11941"><p>Petronella R. Hove, Nora Jean Nealon, Siu Hung Joshua Chan, Shea M. Boyer, Hannah B. Haberecht, and Elizabeth P. Ryan. Integrated profiling of gram-positive and gram-negative probiotic genomes, proteomes and metabolomes revealed small molecules with differential growth inhibition of antimicrobial-resistant pathogens. 20(5):788–810, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19390211.2022.2120146">[DOI]</a>.</p></li>
<li id="id11958"><p>Takeyuki Tamura. Trimming gene deletion strategies for growth-coupled production in constraint-based metabolic networks: trimgdel. 20(2):1540–1549, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/tcbb.2022.3185221">[DOI]</a>.</p></li>
<li id="id11959"><p>Cláudia Marlene Bauer, Caroline Schmitz, Melissa Fontes Landell, and Marcelo Maraschin. Biosynthesis of fatty acids and biosurfactants by the yeast yarrowia lipolytica with emphasis on metabolic networks and bioinformatics. 6(1):e2022001–e2022001, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.4322/biori.202201">[DOI]</a>.</p></li>
<li id="id11960"><p>Nirvana Nursimulu, Alan M Moses, and John Parkinson. Architect: a tool for aiding the reconstruction of high-quality metabolic models through improved enzyme annotation. 18(9):e1010452–e1010452, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1010452">[DOI]</a>.</p></li>
<li id="id11961"><p>Kirk Smith, Nicole Rhoads, and Sriram Chandrasekaran. Protocol for carom: a machine learning tool to predict post-translational regulation from metabolic signatures. 3(4):101799–101799, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xpro.2022.101799">[DOI]</a>.</p></li>
<li id="id11980"><p>Alicja K. Warda, Eugene Dempsey, Sofia D. Forssten, C. Anthony Ryan, John F. Cryan, Elaine Patterson, Mairead O’Riordan, Carol‐Anne O'Shea, Finola Keohane, Grainne Meehan, Orlagh O’Connor, R. Paul Ross, and Catherine Stanton. Cross-sectional observational study protocol: missing microbes in infants born by caesarean section (mimic): antenatal antibiotics and mode of delivery. 12(11):e064398–e064398, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1136/bmjopen-2022-064398">[DOI]</a>.</p></li>
<li id="id11981"><p>Freddy Castillo-Alfonso, Alejandro Quintana-Menéndez, Gabriel Vigueras-Ramírez, Alfonso Mauricio Sales-Cruz, Luis Manuel Rosales-Colunga, and Roberto Olivares‐Hernández. Analysis of the propionate metabolism in bacillus subtilis during 3-indolacetic production. 10(12):2352–2352, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms10122352">[DOI]</a>.</p></li>
<li id="id12007"><p>Ezgi Tanıl, Nehir Kızılilsoley, and Emrah Nıkerel. Genome-scale metabolic model guided subtyping lung cancer towards personalized diagnosis. 55(20):641–646, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2022.09.168">[DOI]</a>.</p></li>
<li id="id12008"><p>Kristina Grausa, Shahida Anusha Siddiqui, Norbert Lameyer, Karin Wiesotzki, Sergiy Smetana, and Agris Pentjuss. Metabolic modeling of&lt;i&gt;hermetia illucens&lt;/i&gt;larvae resource allocation for high-value fatty acid production. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.10.25.513765">[DOI]</a>.</p></li>
<li id="id12039"><p>Apostolos Chalkis, Vissarion Fisikopoulos, Elias Tsigaridas, and Haris Zafeiropoulos. Geometric algorithms for sampling the flux space of metabolic networks. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.20382/jocg.v14i1a8">[DOI]</a>.</p></li>
<li id="id12042"><p>Puhua Niu, Maria J. Soto, Byung-Jun Yoon, Edward R. Dougherty, Francis J. Alexander, Ian K. Blaby, and Xiaoning Qian. Protocol for condition-dependent metabolite yield prediction using the trimer pipeline. 3(1):101184–101184, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xpro.2022.101184">[DOI]</a>.</p></li>
<li id="id12043"><p>Kristina Grausa, Ivars Mozga, Kārlis Pleiko, and Agris Pentjuss. Integrative gene expression and metabolic analysis tool igemrna. 12(4):586–586, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biom12040586">[DOI]</a>.</p></li>
<li id="id12049"><p>Seo‐Young Park, Dong‐Hyuk Choi, Jinsung Song, Uiseon Park, Hyeran Cho, Bee Hak Hong, Fumi Shozui, Yaron Silberberg, and Dong‐Yup Lee. Characterizing basal and feed media effects on mammalian cell cultures by systems engineering approaches. 55(7):31–36, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2022.07.418">[DOI]</a>.</p></li>
<li id="id12050"><p>Heewon Jung, Hyun‐Seob Song, and Christof Meile. Complab v1.0: a scalable pore-scale model for flow, biogeochemistry, microbial metabolism, and biofilm dynamics. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/egusphere-2022-1016">[DOI]</a>.</p></li>
<li id="id12056"><p>Emil Ljungqvist and Martin Gustavsson. Genome-scale reconstruction and metabolic modelling of the fast-growing thermophile geobacillus sp. lc300. 15():e00212–e00212, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2022.e00212">[DOI]</a>.</p></li>
<li id="id12067"><p>Gayathri Sambamoorthy and Karthik Raman. Deciphering the evolution of microbial interactions: in silico studies of two-member microbial communities. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.01.14.476316">[DOI]</a>.</p></li>
<li id="id12068"><p>Xingcun Fan, Jingru Zhou, Jianye Xia, and Xuefeng Yan. Genome‐scale metabolic model's multi‐objective solving algorithm based on the inflexion point of pareto front including maximum energy utilization and its application in &lt;i&gt;aspergillus niger&lt;/i&gt; ds03043. 119(6):1539–1555, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.28078">[DOI]</a>.</p></li>
<li id="id12070"><p>Luis V. Valcárcel, Edurne San José‐Eneriz, Raquel Ordóñez, Iñigo Apaolaza, Ana Valcárcel, Leire Gárate, Jesús F. San Miguel, Antonio Pineda‐Lucena, Xabier Agirre, Felipe Prósper, and Francisco J. Planes. Gmcstool: automated network-based tool to search for metabolic vulnerabilities in cancer. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.11.03.514827">[DOI]</a>.</p></li>
<li id="id12071"><p>Arianna Basile, Almut Heinken, Johannes Hertel, Larry Smarr, Weizhong Li, Laura Treu, Giorgio Valle, Stefano Campanaro, and Ines Thiele. Longitudinal flux balance analyses of a patient with crohn’s disease highlight microbiome metabolic alterations. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.12.19.520975">[DOI]</a>.</p></li>
<li id="id12085"><p>Tamara Bintener, Maria Pires Pacheco, Ali Kishk, Jeff Didier, and Thomas Sauter. Drug target prediction using context-specific metabolic models reconstructed from rfastcormics. 2535():221–240, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2513-2_17">[DOI]</a>.</p></li>
<li id="id12092"><p>Ye Zhang, Dehua Liu, and Zhen Chen. Genome-scale modeling and systems metabolic engineering of vibrio natriegens for the production of 1,3-propanediol. 2553():209–220, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-2617-7_11">[DOI]</a>.</p></li>
<li id="id12099"><p>Hamza Umut Karakurt and Pınar Pir. &lt;i&gt;in silico&lt;/i&gt; analysis of metabolic effects of bipolar disorder on prefrontal cortex identified altered gaba, glutamate-glutamine cycle, energy metabolism and amino acid synthesis pathways. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/intbio/zyac012">[DOI]</a>.</p></li>
<li id="id12119"><p>Prerna Bhalla, Subasree Sridhar, Justin Kullu, Sriya Veerapaneni, Swagatika Sahoo, Nirav Bhatt, and G. K. Suraishkumar. Modeling reactive species metabolism in colorectal cancer for identifying metabolic targets and devising therapeutics. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.05.03.490417">[DOI]</a>.</p></li>
<li id="id12120"><p>Saratram Gopalakrishnan, Chintan Joshi, Miguel Valderrama Gomez, Elçin Içten, Pablo Rolandi, William Johnson, Cleo Kontoravdi, and Nathan E. Lewis. Guidelines for extracting biologically relevant context-specific metabolic models using gene expression data. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.12.04.519052">[DOI]</a>.</p></li>
<li id="id12187"><p>Cecilia Noecker, Juan M. Sánchez, Jordan E. Bisanz, Veronica Escalante, Margaret Alexander, Kai Trepka, Almut Heinken, Yuanyuan Liu, Dylan Dodd, Ines Thiele, Brian C. DeFelice, and Peter J. Turnbaugh. Systems biology illuminates alternative metabolic niches in the human gut microbiome. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.09.19.508335">[DOI]</a>.</p></li>
<li id="id12198"><p>Maurício Alexander de Moura Ferreira, Wendel Batista da Silveira, and Zoran Nikoloski. Parrot: prediction of enzyme abundances using protein-constrained metabolic models. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.22541/au.166117417.77605988/v1">[DOI]</a>.</p></li>
<li id="id12199"><p>Upasana Jhariya and Sukdeb Pal. Proteomic, genomic, and metabolomic understanding and designing for bioremediation of environmental contaminants. ():415–435, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-19-4320-1_18">[DOI]</a>.</p></li>
<li id="id12200"><p>Johann F. Jadebeck, Wolfgang Wiechert, and Katharina Nöh. Chrrt: boosting coordinate hit-and-run with rounding by thinning. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.11.17.516802">[DOI]</a>.</p></li>
<li id="id12214"><p>Ines Thiele and Ronan M. T. Fleming. Whole-body metabolic modelling predicts isoleucine dependency of sars-cov-2 replication. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.04.13.488249">[DOI]</a>.</p></li>
<li id="id12215"><p>Eric Libby, Christopher P. Kempes, and Jordan G. Okie. Metabolic compatibility and the rarity of prokaryote endosymbioses. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.04.14.488272">[DOI]</a>.</p></li>
<li id="id12216"><p>Hiroki Suyama, Laurence Don Wai Luu, Ling Zhong, Mark J. Raftery, and Ruiting Lan. Integrating proteomic data with metabolic modelling provides insight into key pathways of&lt;i&gt;bordetella pertussis&lt;/i&gt;biofilms. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.05.27.493021">[DOI]</a>.</p></li>
<li id="id12217"><p>Andisheh Dadashi and Derek Martinez. Flux balance network expansion predicts stage-specific human peri_implantation embryo metabolism. 20(04):2250010–2250010, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s021972002250010x">[DOI]</a>.</p></li>
<li id="id12218"><p>Ning Qin, Lingyun Li, Xu Ji, Rui Pereira, Yu Chen, Shile Yin, Chaokun Li, Xiaozhen Wan, Hao Luo, Yueping Zhang, Genlai Dong, Yiming Zhang, Shuobo Shi, Christer Larsson, Yun Chen, Tianwei Tan, Zihe Liu, and Jens Nielsen. Flux regulation through glycolysis and respiration is balanced by inositol pyrophosphates. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4135724">[DOI]</a>.</p></li>
<li id="id12219"><p>Sebastián N. Mendoza, Pedro A. Saa, Bas Teusink, and Eduardo Agosín. Metabolic modeling of wine fermentation at genome scale. 2399():395–454, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1831-8_16">[DOI]</a>.</p></li>
<li id="id12220"><p>Kengo Watanabe, Tomasz Wilmanski, Priyanka Baloni, Max Robinson, Gonzalo G. Garcia, Michael R. Hoopmann, Mukul K. Midha, David Baxter, Michal Maes, Seamus R. Morrone, Kelly M. Crebs, Charu Kapil, Ulrike Kusebauch, Jack Wiedrick, Jodi Lapidus, Jennifer C. Lovejoy, Andrew T. Magis, Christopher Lausted, Jared C. Roach, Gustavo Glusman, Steven R. Cummings, Nicholas J. Schork, Nathan D. Price, Leroy Hood, Richard A. Miller, Robert L. Moritz, and Noa Rappaport. Systems-level patterns in biological processes are changed under prolongevity interventions and across biological age. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.07.11.22277435">[DOI]</a>.</p></li>
<li id="id12221"><p>Handan Çetin, Zeynep Petek Çakar, and Kutlu Ö. Ülgen. Understanding the adaptive laboratory evolution of multiple stress‐resistant yeast strains by genome‐scale modeling. 39(8):449–465, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/yea.3806">[DOI]</a>.</p></li>
<li id="id12222"><p>Nunthaphan Vikromvarasiri, Shuhei Noda, Tomokazu Shirai, and Akihiko Kondo. Investigation of two metabolic engineering approaches for (r,r)-2,3-butanediol production from glycerol in bacillus subtilis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-1861033/v1">[DOI]</a>.</p></li>
<li id="id12223"><p>Nantia Leonidou, Alina Renz, Reihaneh Mostolizadeh, and Andreas Dräger. New workflow predicts drug targets against sars-cov-2 via metabolic changes in infected cells. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202203.0290.v2">[DOI]</a>.</p></li>
<li id="id12224"><p>Håvard Molversmyr, Ove Øyås, Filip Rotnes, and Jon Olav Vik. Transcriptomics data integration for context-specific modeling of atlantic salmon metabolism: functional evaluation of methods based on metabolic tasks. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.09.23.509266">[DOI]</a>.</p></li>
<li id="id12225"><p>Joshua A. M. Kaste and Yair Shachar‐Hill. Accurate flux predictions using tissue-specific gene expression in plant metabolic modeling. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.09.05.506655">[DOI]</a>.</p></li>
<li id="id12226"><p>Ibrahim Fakih, Jeanne Got, Carlos Eduardo Robles-Rodríguez, Anne Siegel, Évelyne Forano, and Rafael Muñoz‐Tamayo. Dynamic genome-based metabolic modeling of the predominant cellulolytic rumen bacterium&lt;i&gt;fibrobacter succinogenes&lt;/i&gt;s85. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.10.18.512662">[DOI]</a>.</p></li>
<li id="id12227"><p>Marius Arend, David Zimmer, Rudan Xu, Frederick Sommer, Timo Mühlhaus, and Zoran Nikoloski. Proteomics and constraint-based modelling reveal enzyme kinetic properties of&lt;i&gt;chlamydomonas reinhardtii&lt;/i&gt;on a genome scale. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.11.06.515318">[DOI]</a>.</p></li>
<li id="id12259"><p>Kulwadee Thanamit, Franziska Hoerhold, Marcus Oswald, and Rainer Koenig. Linear programming based gene expression model (lpm-gem) predicts the carbon source for bacillus subtilis. 23(1):226–226, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-022-04742-7">[DOI]</a>.</p></li>
<li id="id12305"><p>Armin Dadras, Sayed‐Amir Marashi, Kaveh Kavousi, and Ali Mohammad Banaei‐Moghaddam. Computational modeling of the evolutionary transition from c3 to c4 photosynthesis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.01.20.477074">[DOI]</a>.</p></li>
<li id="id12306"><p>Min Wang, Menglei Xia, Di Wang, Yiming Xia, Haijiao Shi, Zhongyu Tian, and Yu Zheng. Oxidoreduction potential controlling for increasing the fermentability of enzymatically hydrolyzed steam-exploded corn stover for butanol production. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-1409943/v1">[DOI]</a>.</p></li>
<li id="id12307"><p>Mohd Firdaus Abdul‐Wahab, Shruti Pavagadhi, Hitesh Tikariha, and Sanjay Swarup. Microbiome engineering for next‐generation precision agriculture. ():137–153, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527825462.ch5">[DOI]</a>.</p></li>
<li id="id12308"><p>Irina Alecu, Carmen Daniela Sosa‐Miranda, Jagdeep K. Sandhu, Steffany A. L. Bennett, and Miroslava Čuperlović‐Culf. Cell culture metabolomics and lipidomics. ():415–456, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-85062-9.00012-x">[DOI]</a>.</p></li>
<li id="id12309"><p>Prerna Bhalla, Swagatika Sahoo, Raghunathan Rengaswamy, Devarajan Karunagaran, and G. K. Suraishkumar. Constraint-based modeling to understand ros-mediated effects in cancer. ():2209–2229, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-15-9411-3_144">[DOI]</a>.</p></li>
<li id="id12311"><p>Emil Karlsen, Marianne Gylseth, Christian Schulz, and Eivind Almaas. A study of a diauxic growth experiment using an expanded dynamic flux balance framework. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.05.16.492165">[DOI]</a>.</p></li>
<li id="id12312"><p>Pooya Borzou, Jafar Ghaisari, Iman Izadi, Yasin Eshraghi, and Yousof Gheisari. A novel strategy for dynamic modelling of genome-scale interaction networks. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.05.20.491854">[DOI]</a>.</p></li>
<li id="id12313"><p>William T. Scott, David Henriques, Eddy J. Smid, Richard A. Notebaart, and Eva Balsa‐Canto. Dynamic genome-scale modeling of&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;unravels mechanisms for ester formation during alcoholic fermentation. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.05.27.493771">[DOI]</a>.</p></li>
<li id="id12314"><p>Tania Alonso-Vásquez, Christopher Riccardi, Iacopo Passeri, and Marco Fondi. Modelling the metabolic consequences of antimicrobial exposure. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.06.08.495414">[DOI]</a>.</p></li>
<li id="id12315"><p>Lamiaa H.S. Hassan, Imran Ahmad, Mostafa El Sheekh, and Norhayati Abdullah. In silico models on algal cultivation and processing. ():228–255, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.4018/978-1-6684-2438-4.ch009">[DOI]</a>.</p></li>
<li id="id12316"><p>Andreas Kremling. Modeling approaches to microbial metabolism. 2399():455–485, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1831-8_17">[DOI]</a>.</p></li>
<li id="id12317"><p>Thomas Sevrin, Lisa Strasser, Camille Ternet, Philipp Junk, Miriam Caffarini, Simona Catozzi, Giorgio Oliviero, Kieran Wynne, Christina Kiel, and Philip J. Luthert. Whole-cell energy modeling reveals quantitative changes of predicted energy flows in ras mutant cancer cell lines. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.4133892">[DOI]</a>.</p></li>
<li id="id12318"><p>Dalimil Bujdoš, Barbora Popelářová, Daniel C. Volke, Pablo I. Nikel, Nikolaus Sonnenschein, and Pavel Dvořák. Engineering of &lt;i&gt;pseudomonas putida&lt;/i&gt; for accelerated co-utilization of glucose and cellobiose yields aerobic overproduction of pyruvate explained by an upgraded metabolic model. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.07.22.501097">[DOI]</a>.</p></li>
<li id="id12319"><p>Mike Sportiello, Alexis Poindexter, Emma C. Reilly, Adam Geber, Kris Lambert Emo, Taylor N. Jones, and David J. Topham. Mouse memory cd8 t cell subsets defined by tissue resident memory (t&lt;sub&gt;rm&lt;/sub&gt;) integrin expression exhibit distinct metabolic profiles. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.09.21.508875">[DOI]</a>.</p></li>
<li id="id12320"><p>Herbert Yao, Sanjeev Dahal, and Laurence Yang. Novel context-specific genome-scale modelling explores the potential of &lt;i&gt;chlamydomonas reinhardtii&lt;/i&gt; for synthetic biology applications. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.10.07.511370">[DOI]</a>.</p></li>
<li id="id12321"><p>Christopher Riccardi, Marzia Calvanese, Veronica Ghini, Tania Alonso-Vásquez, Elena Perrin, Paola Turano, Giorgio Giurato, Alessandro Weisz, Ermenegilda Parrilli, Maria Luisa Tutino, and Marco Fondi. Metabolic robustness to growth temperature of a cold adapted bacterium. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.10.25.513664">[DOI]</a>.</p></li>
<li id="id12322"><p>Anonymous. Comment on egusphere-2022-1016. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/egusphere-2022-1016-rc1">[DOI]</a>.</p></li>
<li id="id12323"><p>Delphine Nègre, Abdelhalim Larhlimi, and Samuel Bertrand. Reconciliation and evolution of&lt;i&gt;penicillium rubens&lt;/i&gt;genome-scale metabolic networks – what about specialised metabolism? ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2022.11.10.515991">[DOI]</a>.</p></li>
<li id="id12324"><p>Maria De La Fuente Ruiz. Comment on egusphere-2022-1016. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/egusphere-2022-1016-rc2">[DOI]</a>.</p></li>
<li id="id12347"><p>Issalapap Kaothong. Enhanced triacylglycerol production in chlorella vulgaris by flux balance analysis on metabolic network. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.58837/chula.the.2022.18">[DOI]</a>.</p></li>
<li id="id12350"><p>Zhihan WANG, Yao Ruan, Qingye Zhang, and Hongyu Zhang. Metabolic network flux balance analysis of the microgravity influence on the growth of &amp;lt;i&amp;gt;arabidopsis&amp;lt;/i&amp;gt; &amp;lt;i&amp;gt;thaliana&amp;lt;/i&amp;gt;. 42(5):941–941, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.11728/cjss2022.05.210714077">[DOI]</a>.</p></li>
<li id="id12368"><p>Jingjing Li, Yifei Gou, Jiarui Yang, Lingxuan Zhao, Bin Wang, Tong Hao, and Jinsheng Sun. Additional file 2 of genome-scale metabolic network model of eriocheir sinensis icrab4665 and nutritional requirement analysis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.20175677">[DOI]</a>.</p></li>
<li id="id12369"><p>Jingjing Li, Yifei Gou, Jiarui Yang, Lingxuan Zhao, Bin Wang, Tong Hao, and Jinsheng Sun. Additional file 6 of genome-scale metabolic network model of eriocheir sinensis icrab4665 and nutritional requirement analysis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.20175692">[DOI]</a>.</p></li>
<li id="id12370"><p>Jingjing Li, Yifei Gou, Jiarui Yang, Lingxuan Zhao, Bin Wang, Tong Hao, and Jinsheng Sun. Additional file 5 of genome-scale metabolic network model of eriocheir sinensis icrab4665 and nutritional requirement analysis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.20175689">[DOI]</a>.</p></li>
<li id="id12371"><p>Jingjing Li, Yifei Gou, Jiarui Yang, Lingxuan Zhao, Bin Wang, Tong Hao, and Jinsheng Sun. Additional file 3 of genome-scale metabolic network model of eriocheir sinensis icrab4665 and nutritional requirement analysis. ():, 2022. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.20175680">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-4" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-4" name="0-4" role="tabpanel" tabindex="0"><div class="docutils container" id="id12485">
<ol class="arabic simple" start="1">
<li id="id12598"><p>Analeigha V. Colarusso, Isabella M. Goodchild-Michelman, Maya Rayle, and Ali R. Zomorrodi. Computational modeling of metabolism in microbial communities on a genome-scale. 26():46–57, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2021.04.001">[DOI]</a>.</p></li>
<li id="id12757"><p>Behnaz Nowrouzi and Leonardo Rios‐Solis. Redox metabolism for improving whole-cell p450-catalysed terpenoid biosynthesis. 42(8):1213–1237, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07388551.2021.1990210">[DOI]</a>.</p></li>
<li id="id12788"><p>Abhijit Paul, Rajat Anand, Sonali Porey Karmakar, Surender Rawat, Nandadulal Bairagi, and Samrat Chatterjee. Exploring gene knockout strategies to identify potential drug targets using genome-scale metabolic models. 11(1):213–213, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-80561-1">[DOI]</a>.</p></li>
<li id="id12795"><p>Yara Seif and Bernhard Ø. Palsson. Path to improving the life cycle and quality of genome-scale models of metabolism. 12(9):842–859, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2021.06.005">[DOI]</a>.</p></li>
<li id="id12804"><p>Andrew Walakira, Damjana Rozman, Tadeja Režen, Miha Mraz, and Miha Moškon. Guided extraction of genome-scale metabolic models for the integration and analysis of omics data. 19():3521–3530, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2021.06.009">[DOI]</a>.</p></li>
<li id="id12805"><p>Alan R. Pacheco and Daniel Segrè. An evolutionary algorithm for designing microbial communities via environmental modification. 18(179):20210348–20210348, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rsif.2021.0348">[DOI]</a>.</p></li>
<li id="id12810"><p>S. Ranganayaki, Neema Jamshidi, Mohamad Aiyaz, Rashmi Santhoshkumar, Narayanappa Gayathri, Pulleri Kandi Harsha, Balasundaram Padmanabhan, and M. M. Srinivas Bharath. Inhibition of mitochondrial complex ii in neuronal cells triggers unique pathways culminating in autophagy with implications for neurodegeneration. 11(1):1483–1483, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-79339-2">[DOI]</a>.</p></li>
<li id="id12891"><p>Bouchra Ezzamouri, Saeed Shoaie, and Rodrigo Ledesma‐Amaro. Synergies of systems biology and synthetic biology in human microbiome studies. 12():681982–681982, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2021.681982">[DOI]</a>.</p></li>
<li id="id12892"><p>Nicolai Panikov. Genome-scale reconstruction of microbial dynamic phenotype: successes and challenges. 9(11):2352–2352, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms9112352">[DOI]</a>.</p></li>
<li id="id12899"><p>Elisabeth Zelle, Nina Pfelzer, Marco Oldiges, Abigail Koch-Koerfges, Michael Bott, Katharina Nöh, and Wolfgang Wiechert. An energetic profile of corynebacterium glutamicum underpinned by measured biomass yield on atp. 65():66–78, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.03.006">[DOI]</a>.</p></li>
<li id="id12915"><p>Lachlan Munro and Douglas B. Kell. Intelligent host engineering for metabolic flux optimisation in biotechnology. 478(20):3685–3721, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bcj20210535">[DOI]</a>.</p></li>
<li id="id12919"><p>N. T. Devika, Ashok Kumar Jangam, Vinaya Kumar Katneni, Prasanna Kumar Patil, Suganya Nathamuni, and M. S. Shekhar. In silico prediction of novel probiotic species limiting pathogenic vibrio growth using constraint-based genome scale metabolic modeling. 11():752477–752477, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcimb.2021.752477">[DOI]</a>.</p></li>
<li id="id12925"><p>Þóra Björg Sigmarsdóttir, Sarah McGarrity, James T. Yurkovich, Óttar Rolfsson, and Ólafur E. Sigurjónsson. Analyzing metabolic states of adipogenic and osteogenic differentiation in human mesenchymal stem cells via genome scale metabolic model reconstruction. 9():642681–642681, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcell.2021.642681">[DOI]</a>.</p></li>
<li id="id12926"><p>German Preciat, Agnieszka B. Wegrzyn, Ines Thiele, Thomas Hankemeier, and Ronan M. T. Fleming. Xomicstomodel: omics data integration and generation of thermodynamically consistent metabolic models. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.11.08.467803">[DOI]</a>.</p></li>
<li id="id12971"><p>Gurneet S. Sangha, Craig J. Goergen, Steven J. Prior, Sushant M. Ranadive, and Alisa Morss Clyne. Preclinical techniques to investigate exercise training in vascular pathophysiology. 320(4):H1566–H1600, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1152/ajpheart.00719.2020">[DOI]</a>.</p></li>
<li id="id12995"><p>Jitin Singla and Kate L. White. A community approach to whole-cell modeling. 26():33–38, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2021.03.009">[DOI]</a>.</p></li>
<li id="id13006"><p>Hameri Tuure, Georgios Fengos, and Vassily Hatzimanikatis. The effects of model complexity and size on metabolic flux distribution and control: case study in escherichia coli. 22(1):134–134, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-021-04066-y">[DOI]</a>.</p></li>
<li id="id13021"><p>Anush Chiappino-Pepe and Vassily Hatzimanikatis. Phenomapping: a protocol to map cellular phenotypes to metabolic bottlenecks, identify conditional essentiality, and curate metabolic models. 2(1):100280–100280, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xpro.2020.100280">[DOI]</a>.</p></li>
<li id="id13023"><p>Ali Navid. A beginner’s guide to the cobra toolbox. 2349():339–365, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1585-0_15">[DOI]</a>.</p></li>
<li id="id13027"><p>Hunor Bartos, Márta Balázs, Ildikó Hajnalka Kuzman, Szabolcs Lányi, and Ildikó Miklóssy. Production of high added-value chemicals in basfia succiniciproducens: role of medium composition. 13(6):3513–3513, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/su13063513">[DOI]</a>.</p></li>
<li id="id13100"><p>Philippe Bogaerts and Alain Vande Wouwer. How to tackle underdeterminacy in metabolic flux analysis? a tutorial and critical review. 9(9):1577–1577, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/pr9091577">[DOI]</a>.</p></li>
<li id="id13101"><p>Siti Zulaiha. Genetic engineering of microalgae lipid biosynthesis for sustainable biodiesel production. 11(3):072–077, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.30574/wjarr.2021.11.3.0397">[DOI]</a>.</p></li>
<li id="id13106"><p>Amnah Alzahmi, Sarah Daakour, Diana C. El Assal, Bushra Dohai, Amphun Chaiboonchoe, Weiqi Fu, David R. Nelson, Alexandra Mystikou, and Kourosh Salehi‐Ashtiani. High-throughput metabolic profiling for model refinements of microalgae. (178):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3791/61913">[DOI]</a>.</p></li>
<li id="id13107"><p>Amnah Alzahmi, Sarah Daakour, Diana C. El Assal, Bushra Dohai, Amphun Chaiboonchoe, Weiqi Fu, David R. Nelson, Alexandra Mystikou, and Kourosh Salehi‐Ashtiani. High-throughput metabolic profiling for model refinements of microalgae. (178):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3791/61913-v">[DOI]</a>.</p></li>
<li id="id13138"><p>Ali Navid. Curating cobra models of microbial metabolism. 2349():321–338, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1585-0_14">[DOI]</a>.</p></li>
<li id="id13176"><p>Priyanka Narad, G. Naresh, and Abhishek Sengupta. Metabolomics and flux balance analysis. ():337–365, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-323-89775-4.00008-0">[DOI]</a>.</p></li>
<li id="id13189"><p>Bernhard Ø. Palsson. Genome‐scale models. ():23–71, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527823468.ch2">[DOI]</a>.</p></li>
<li id="id13227"><p>Mohammad Mazharul Islam, Andrea Goertzen, Pankaj K. Singh, and Rajib Saha. Exploring the metabolic landscape of pancreatic ductal adenocarcinoma cells using genome-scale metabolic modeling. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.14.452356">[DOI]</a>.</p></li>
<li id="id13294"><p>Till Tiso, Tanja Narančić, Ren Wei, Éric Pollet, Niall Beagan, Katja Schröder, Annett Honak, Mengying Jiang, Shane T. Kenny, Nick Wierckx, Rémi Perrin, Luc Avérous, Wolfgang Zimmermann, Kevin E. O’Connor, and Lars M. Blank. Towards bio-upcycling of polyethylene terephthalate. 66():167–178, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.03.011">[DOI]</a>.</p></li>
<li id="id13322"><p>Dorinês Rosário, Gholamreza Bidkhori, Sunjae Lee, Janis R. Bedarf, Falk Hildebrand, Emmanuelle Le Chatelier, Mathias Uhlén, S. Dusko Ehrlich, Gordon Proctor, Ullrich Wüllner, Adil Mardinoğlu, and Saeed Shoaie. Systematic analysis of gut microbiome reveals the role of bacterial folate and homocysteine metabolism in parkinson’s disease. 34(9):108807–108807, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2021.108807">[DOI]</a>.</p></li>
<li id="id13399"><p>Lekha T. Pazhamala, Himabindu Kudapa, Wolfram Weckwerth, A. Harvey Millar, and Rajeev K. Varshney. Systems biology for crop improvement. 14(2):e20098–e20098, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/tpg2.20098">[DOI]</a>.</p></li>
<li id="id13410"><p>Omid Oftadeh, Pierre Salvy, María Masid, Maxime Curvat, Ljubiša Mišković, and Vassily Hatzimanikatis. A genome-scale metabolic model of saccharomyces cerevisiae that integrates expression constraints and reaction thermodynamics. 12(1):4790–4790, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-021-25158-6">[DOI]</a>.</p></li>
<li id="id13440"><p>Karim Azer, Chanchala Kaddi, Jeffrey S. Barrett, Jane P. F. Bai, Sean T. McQuade, Nathaniel J. Merrill, Benedetto Piccoli, Susana R. Neves, Luca Marchetti, Rosario Lombardo, Silvia Parolo, Selva Rupa Christinal Immanuel, and Nitin S. Baliga. History and future perspectives on the discipline of quantitative systems pharmacology modeling and its applications. 12():637999–637999, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2021.637999">[DOI]</a>.</p></li>
<li id="id13461"><p>David J. Sanders, S Inniss, Gregory Sebepos‐Rogers, Farooq Rahman, and Andrew Smith. The role of the microbiome in gastrointestinal inflammation. 41(6):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bsr20203850">[DOI]</a>.</p></li>
<li id="id13471"><p>Yuan Zhu, Ying Li, Ya Xu, Jian Zhang, Linlin Ma, Qingsheng Qi, and Qian Wang. Development of bifunctional biosensors for sensing and dynamic control of glycolysis flux in metabolic engineering. 68():142–151, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.09.011">[DOI]</a>.</p></li>
<li id="id13475"><p>David Henriques, Romain Minebois, Sebastián N. Mendoza, Laura G. Macías, Roberto Pérez‐Torrado, Eladio Barrio, Bas Teusink, Amparo Querol, and Eva Balsa‐Canto. A multiphase multiobjective dynamic genome-scale model shows different redox balancing among yeast species of the &lt;i&gt;saccharomyces&lt;/i&gt; genus in fermentation. 6(4):e0026021–e0026021, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00260-21">[DOI]</a>.</p></li>
<li id="id13477"><p>Marie Schöpping, Paula Gaspar, Ana Rute Neves, Carl Johan Franzén, and Ahmad A. Zeidan. Identifying the essential nutritional requirements of the probiotic bacteria bifidobacterium animalis and bifidobacterium longum through genome-scale modeling. 7(1):47–47, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-021-00207-4">[DOI]</a>.</p></li>
<li id="id13551"><p>Song‐Min Schinn, Carly Morrison, Wei Wei, Lin Zhang, and Nathan E. Lewis. A genome‐scale metabolic network model and machine learning predict amino acid concentrations in chinese hamster ovary cell cultures. 118(5):2118–2123, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27714">[DOI]</a>.</p></li>
<li id="id13568"><p>Karl Kochanowski, Hiroyuki Okano, Vadim Patsalo, James R. Williamson, Uwe Sauer, and Terence Hwa. Global coordination of metabolic pathways in &lt;i&gt;escherichia coli&lt;/i&gt; by active and passive regulation. 17(4):e10064–e10064, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.202010064">[DOI]</a>.</p></li>
<li id="id13603"><p>Zachary B. Haiman, Daniel C. Zielinski, Yuko Koike, James T. Yurkovich, and Bernhard Ø. Palsson. Masspy: building, simulating, and visualizing dynamic biological models in python using mass action kinetics. 17(1):e1008208–e1008208, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008208">[DOI]</a>.</p></li>
<li id="id13612"><p>Nhung Pham, Maarten Reijnders, María Suárez‐Diez, Bart Nijsse, Jan Springer, Gerrit Eggink, and Peter J. Schaap. Genome-scale metabolic modeling underscores the potential of cutaneotrichosporon oleaginosus atcc 20509 as a cell factory for biofuel production. 14(1):2–2, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-020-01838-1">[DOI]</a>.</p></li>
<li id="id13657"><p>Punit Kumar, Kashyap Kumar Dubey, Zeba Usmani, Minaxi Sharma, and Vijai Kumar Gupta. Biotechnological and industrial applications of &lt;i&gt;streptomyces&lt;/i&gt; metabolites. 16(1):244–264, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bbb.2294">[DOI]</a>.</p></li>
<li id="id13720"><p>Song‐Min Schinn, Carly Morrison, Wei Wei, Lin Zhang, and Nathan E. Lewis. Systematic evaluation of parameters for genome-scale metabolic models of cultured mammalian cells. 66():21–30, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.03.013">[DOI]</a>.</p></li>
<li id="id13721"><p>Kangsan Kim, Donghui Choe, Yoseb Song, Minjeong Kang, Seung‐Goo Lee, Dae‐Hee Lee, and Byung‐Kwan Cho. Engineering bacteroides thetaiotaomicron to produce non-native butyrate based on a genome-scale metabolic model-guided design. 68():174–186, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.10.005">[DOI]</a>.</p></li>
<li id="id13722"><p>Ali Kishk, Maria Pires Pacheco, and Thomas Sauter. Dccov: repositioning of drugs and drug combinations for sars-cov-2 infected lung through constraint-based modeling. 24(11):103331–103331, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.103331">[DOI]</a>.</p></li>
<li id="id13728"><p>Dohyeon Kim, Young-Shin Kim, and Sung Ho Yoon. Development of a genome-scale metabolic model and phenome analysis of the probiotic escherichia coli strain nissle 1917. 22(4):2122–2122, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms22042122">[DOI]</a>.</p></li>
<li id="id13737"><p>Raja Rajeswary Thanmalagan, Aiswarya Jayaprakash, Abhijeet Deb Roy, A. Annamalai, and P.T.V. Lakshmi. A review on applications of plant network biology to understand the drought stress response in economically important cereal crops. 29():100345–100345, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.plgene.2021.100345">[DOI]</a>.</p></li>
<li id="id13742"><p>Ying Li, He Xian, Ya Xu, Yuan Zhu, Zhijie Sun, Qian Wang, and Qingsheng Qi. Fine tuning the glycolytic flux ratio of ep-bifido pathway for mevalonate production by enhancing glucose-6-phosphate dehydrogenase (zwf) and crispri suppressing 6-phosphofructose kinase (pfka) in escherichia coli. 20(1):32–32, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-021-01526-1">[DOI]</a>.</p></li>
<li id="id13745"><p>Hossein Firoozabadi, Mohammad Mahdi Mardanpour, and Ehsan Motamedian. A system-oriented strategy to enhance electron production of synechocystis sp. pcc6803 in bio-photovoltaic devices: experimental and modeling insights. 11(1):12294–12294, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-021-91906-9">[DOI]</a>.</p></li>
<li id="id13753"><p>Zongjie Dai, Yan Zhu, Hongjun Dong, Chunhua Zhao, Yanping Zhang, and Yin Li. Enforcing atp hydrolysis enhanced anaerobic glycolysis and promoted solvent production in clostridium acetobutylicum. 20(1):149–149, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-021-01639-7">[DOI]</a>.</p></li>
<li id="id13754"><p>Lady Johanna Forero-Rodríguez, Jonathan Josephs‐Spaulding, Stefano Flor, Andrés Pinzón, and Christoph Kaleta. Parkinson’s disease and the metal–microbiome–gut–brain axis: a systems toxicology approach. 11(1):71–71, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/antiox11010071">[DOI]</a>.</p></li>
<li id="id13861"><p>Patricia Martins Conde, Thomas Pfau, Maria Pires Pacheco, and Thomas Sauter. A dynamic multi-tissue model to study human metabolism. 7(1):5–5, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-020-00159-1">[DOI]</a>.</p></li>
<li id="id13885"><p>You‐Tyun Wang, Min‐Ru Lin, Wei‐Chen Chen, Wu‐Hsiung Wu, and Feng‐Sheng Wang. Optimization of a modeling platform to predict oncogenes from genome‐scale metabolic networks of non‐small‐cell lung cancers. 11(8):2078–2094, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/2211-5463.13231">[DOI]</a>.</p></li>
<li id="id13900"><p>Vladimir Porokhin, Sara A. Amin, Trevor B. Nicks, Venkatesh Endalur Gopinarayanan, Nikhil U. Nair, and Soha Hassoun. Analysis of metabolic network disruption in engineered microbial hosts due to enzyme promiscuity. 12():e00170–e00170, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2021.e00170">[DOI]</a>.</p></li>
<li id="id13907"><p>Mahdi Jalili, Martin Scharm, Olaf Wolkenhauer, Mehdi Damaghi, and Ali Salehzadeh‐Yazdi. Exploring the metabolic heterogeneity of cancers: a benchmark study of context-specific models. 11(6):496–496, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jpm11060496">[DOI]</a>.</p></li>
<li id="id13912"><p>María Moscardó Garcia, Maria Pires Pacheco, Tamara Bintener, Luana Presta, and Thomas Sauter. Importance of the biomass formulation for cancer metabolic modeling and drug prediction. 24(10):103110–103110, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.103110">[DOI]</a>.</p></li>
<li id="id13918"><p>Hilal Taymaz‐Nikerel and Alvaro R. Lara. Vitreoscilla haemoglobin: a tool to reduce overflow metabolism. 10(1):43–43, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms10010043">[DOI]</a>.</p></li>
<li id="id13924"><p>German Preciat, Agnieszka B. Wegrzyn, Edinson Lucumi Moreno, Cornelius C.W. Willacey, Jennifer Modamio, Fátima Liliana Monteiro, Diana C. El Assal, Alissa Schurink, Miguel A. P. Oliveira, Zhi Zhang, Ben Cousins, Hulda S. Haraldsdóttir, Siham Hachi, Susanne Zach, Germán Leparc, Yin Tat Lee, Bastian Hengerer, Santosh Vempala, Michael A. Saunders, Amy C. Harms, Enrico Glaab, Jens C. Schwamborn, Ines Thiele, Thomas Hankemeier, and Ronan M. T. Fleming. Mechanistic model-driven exometabolomic characterisation of human dopaminergic neuronal metabolism. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.30.450562">[DOI]</a>.</p></li>
<li id="id14006"><p>Ratchaprapa Kamsen, Saowalak Kalapanulak, Porntip Chiewchankaset, and Treenut Saithong. Transcriptome integrated metabolic modeling of carbon assimilation underlying storage root development in cassava. 11(1):8758–8758, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-021-88129-3">[DOI]</a>.</p></li>
<li id="id14015"><p>James M. Clomburg, Ángela Cintolesi, and Ramón González. &lt;i&gt;in silico&lt;/i&gt; and &lt;i&gt;in vivo&lt;/i&gt; analyses reveal key metabolic pathways enabling the fermentative utilization of glycerol in &lt;i&gt;escherichia coli&lt;/i&gt;. 15(1):289–304, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.13938">[DOI]</a>.</p></li>
<li id="id14043"><p>Davide Lagoa, José P. Faria, Filipe Liu, Emanuel Cunha, Christopher S. Henry, and Óscar Dias. &lt;i&gt;transyt&lt;/i&gt;, an innovative framework for identifying transport systems. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.04.29.441738">[DOI]</a>.</p></li>
<li id="id14053"><p>Claudio Tomi-Andrino, Rupert Norman, Thomas Millat, Philippe Soucaille, Klaus Winzer, David A. Barrett, John R. King, and Dong‐Hyun Kim. Physicochemical and metabolic constraints for thermodynamics-based stoichiometric modelling under mesophilic growth conditions. 17(1):e1007694–e1007694, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007694">[DOI]</a>.</p></li>
<li id="id14054"><p>Eunice van Pelt‐KleinJan, Daan H. de Groot, and Bas Teusink. Understanding fba solutions under multiple nutrient limitations. 11(5):257–257, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11050257">[DOI]</a>.</p></li>
<li id="id14063"><p>Olga Yaneth Echeverri-Peña, Diego Salazar-Barreto, Alexander Rodríguez‐López, Janneth González, Carlos Javier Alméciga-Díaz, Cristian H. Verano-Guevara, and Luis A. Barrera. Use of a neuron-glia genome-scale metabolic reconstruction to model the metabolic consequences of the arylsulphatase a deficiency through a systems biology approach. 7(7):e07671–e07671, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.heliyon.2021.e07671">[DOI]</a>.</p></li>
<li id="id14064"><p>Neeraj Sinha, Evert M. van Schothorst, Guido Hooiveld, Jaap Keijer, Vítor A. P. Martins dos Santos, and María Suárez‐Diez. Exploring the associations between transcript levels and fluxes in constraint-based models of metabolism. 22(1):574–574, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-021-04488-8">[DOI]</a>.</p></li>
<li id="id14069"><p>Marc Griesemer and Ali Navid. Mofa: multi-objective flux analysis for the cobra toolbox. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.20.445041">[DOI]</a>.</p></li>
<li id="id14070"><p>Fernando Medeiros Filho, Ana Paula Barbosa do Nascimento, Maiana de Oliveira Cerqueira e Costa, Thiago Castanheira Merigueti, Márcio Argollo de Menezes, Marisa Nicolás, Marcelo Trindade dos Santos, Ana Paula D’Alincourt Carvalho-Assef, and Fabrício Alves Barbosa da Silva. A systematic strategy to find potential therapeutic targets for pseudomonas aeruginosa using integrated computational models. 8():728129–728129, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmolb.2021.728129">[DOI]</a>.</p></li>
<li id="id14071"><p>Priyanka Gehlot, Nidhi Pareek, and Vivekanand Vivekanand. Development of biofertilizers and microbial consortium an approach to sustainable agriculture practices. ():315–348, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-16-3364-5_15">[DOI]</a>.</p></li>
<li id="id14090"><p>Omid Oftadeh, Pierre Salvy, María Masid, Maxime Curvat, Ljubiša Mišković, and Vassily Hatzimanikatis. A genome-scale metabolic model of&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;that integrates expression constraints and reaction thermodynamics. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.17.431671">[DOI]</a>.</p></li>
<li id="id14140"><p>MacLean G. Kohlmeier, Ben A. Bailey-Elkin, Brian L. Mark, and Ivan J. Oresnik. Characterization of the sorbitol dehydrogenase smos from &lt;i&gt;sinorhizobium meliloti&lt;/i&gt; 1021. 77(3):380–390, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1107/s2059798321001017">[DOI]</a>.</p></li>
<li id="id14141"><p>Anna Trigos, Benjamin Goudey, Justin Bedő, Thomas Conway, Noel G. Faux, and Kelly L. Wyres. Collateral sensitivity to β-lactam drugs in drug-resistant tuberculosis is driven by the transcriptional wiring of blai operon genes. 6(3):e0024521–e0024521, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msphere.00245-21">[DOI]</a>.</p></li>
<li id="id14159"><p>Ankur Gupta, Ajay Kumar, Rajat Anand, Nandadulal Bairagi, and Samrat Chatterjee. Genome scale metabolic model driven strategy to delineate host response to &lt;i&gt;mycobacterium tuberculosis&lt;/i&gt; infection. 17(2):296–306, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d0mo00138d">[DOI]</a>.</p></li>
<li id="id14160"><p>David Hala, Patricia C. Faulkner, Kai He, Manoj Kamalanathan, Mikeelee Brink, Kristina Simons, Meltem Apaydin, Béatrice V. Hernout, Lene H. Petersen, Ivan Ivanov, and Xiaoning Qian. An integrated in vivo and in silico analysis of the metabolism disrupting effects of cpi-613 on embryo-larval zebrafish (danio rerio). 248():109084–109084, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cbpc.2021.109084">[DOI]</a>.</p></li>
<li id="id14161"><p>Mahsa Sadat Razavi Borghei, Meysam Mobasheri, and Tabassom Sobati. Genome scale constraint-based metabolic reconstruction of propionibacterium freudenreichii dsm 20271. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-847511/v1">[DOI]</a>.</p></li>
<li id="id14162"><p>Luis Caspeta, Eduard J. Kerkhoven, Alfredo Martı́nez, and Jens Nielsen. The yeastgemmap: a process diagram to assist yeast systems‐metabolic studies. 118(12):4800–4814, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27943">[DOI]</a>.</p></li>
<li id="id14184"><p>Albert Tafur Rangel, Luis H. Reyes, Jorge M. Gómez, and Andrés Fernando González Barrios. Optimization of glycerol consumption in wild‐type &lt;i&gt;escherichia coli&lt;/i&gt; using central carbon modeling as an alternative approach. 15(3):825–839, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bbb.2205">[DOI]</a>.</p></li>
<li id="id14193"><p>Joshua E. Goldford, Ashish B. George, Avi I. Flamholz, and Daniel Segrè. Protein cost minimization promotes the emergence of coenzyme redundancy. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.17.444566">[DOI]</a>.</p></li>
<li id="id14200"><p>Ehsan Salehabadi, Ehsan Motamedian, and Seyed Abbas Shojaosadati. Reconstruction of a generic genome-scale metabolic network for chicken: investigating network connectivity and finding potential biomarkers. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.22.453438">[DOI]</a>.</p></li>
<li id="id14236"><p>Julia Koblitz, Sabine Will, S. Alexander Riemer, Thomas Ulas, Meina Neumann‐Schaal, and Dietmar Schomburg. The metano modeling toolbox mmtb: an intuitive, web-based toolbox introduced by two use cases. 11(2):113–113, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11020113">[DOI]</a>.</p></li>
<li id="id14262"><p>Craig Woods, Christopher M. Humphreys, Claudio Tomi-Andrino, Anne M. Henstra, Michael Köpke, Séan D. Simpson, Klaus Winzer, and Nigel P. Minton. Application of transposon-insertion sequencing to determine gene essentiality in the acetogen &lt;i&gt;clostridium autoethanogenum&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.19.444907">[DOI]</a>.</p></li>
<li id="id14263"><p>Yuki Kuriya, Mai Inoue, Masaki Yamamoto, Masahiro Murata, and Michihiro Araki. Knowledge extraction from literature and enzyme sequences complements fba analysis in metabolic engineering. 16(12):e2000443–e2000443, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.202000443">[DOI]</a>.</p></li>
<li id="id14265"><p>Angela Dixon. Designing predictive mathematical models for the metabolic pathways associated with polyhydroxybutyrate synthesis in escherichia coli. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.26076/9713-2802">[DOI]</a>.</p></li>
<li id="id14289"><p>Ali Kishk, Maria Pires Pacheco, and Thomas Sauter. Dccov: repositioning of drugs and drug combinations for sars-cov-2 infected lung through constraint-based modelling. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.2103.13844">[DOI]</a>.</p></li>
<li id="id14291"><p>Lokanand Koduru, Meiyappan Lakshmanan, Pei Yu Lim, Pooi-Leng Ho, M. Shakeera Banu, Doo‐Sang Park, Dave Siak‐Wei Ow, and Dong‐Yup Lee. Systematic evaluation of genome-wide metabolic landscapes in lactic acid bacteria reveals diet-induced and strain-specific probiotic idiosyncrasies. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.20.449192">[DOI]</a>.</p></li>
<li id="id14295"><p>Hossein Firoozabadi, Mohammad Mahdi Mardanpour, and Ehsan Motamedian. A system-oriented strategy to enhance electron production of synechocystis sp. pcc6803 in bio-photovoltaic devices: experimental and modeling insights. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-324293/v1">[DOI]</a>.</p></li>
<li id="id14298"><p>Yoshitaro Heshiki. Optimization of polyhydroxybutyrate production in recombinant escherichia coli through metabolic modeling and simulation. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.26076/6699-1737">[DOI]</a>.</p></li>
<li id="id14303"><p>David Henriques, Romain Minebois, Sebastián N. Mendoza, Laura G. Macías, Roberto Pérez‐Torrado, Eladio Barrio, Bas Teusink, Amparo Querol, and Eva Balsa‐Canto. A multi-phase multi-objective genome-scale model shows diverse redox balance strategies in yeasts. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.11.430755">[DOI]</a>.</p></li>
<li id="id14317"><p>Nhung Pham. Evaluating and deploying genome-scale metabolic models for microbial cell factories. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/537210">[DOI]</a>.</p></li>
<li id="id14337"><p>Lisa M. Leone. Metabolic modeling of secondary metabolism in plant systems. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.7275/5461849">[DOI]</a>.</p></li>
<li id="id14348"><p>Sumit Kumar Singh. Pathway pioneer: a web-based graphical tool for the organization and flux analysis of metabolic networks. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.26076/0715-92d6">[DOI]</a>.</p></li>
<li id="id14354"><p>Harsh Dosi. Pathway pioneer: a web-based metabolic network layout extension. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.26076/9a78-ee0e">[DOI]</a>.</p></li>
<li id="id14424"><p>Vimaladhasan Senthamizhan, Sunanda Subramaniam, Arjun Raghavan, and Karthik Raman. Castle: a database of synthetic lethal sets predicted from genome-scale metabolic networks. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.08.430024">[DOI]</a>.</p></li>
<li id="id14425"><p>Apostolos Chalkis, Vissarion Fisikopoulos, Marios Papachristou, and Elias Tsigaridas. Truncated log-concave sampling with reflective hamiltonian monte carlo. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.2102.13068">[DOI]</a>.</p></li>
<li id="id14426"><p>David Henriques, Romain Minebois, Sebastián N. Mendoza, Laura G. Macías, Roberto Pérez‐Torrado, Eladio Barrio, Bas Teusink, Amparo Querol, and Eva Balsa‐Canto. A multi-phase multi-objective dynamic genome-scale model shows different redox balancing among yeast species in fermentation. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.08.434513">[DOI]</a>.</p></li>
<li id="id14427"><p>Goldford Je, George Ab, Avi I. Flamholz, and Daniel Segrè. Thermodynamics drives coenzyme redundancy in metabolism. ():, 2021. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id14429"><p>Kristopher A. Hunt, Natasha Mallette, Brent Peyton, and Ross P. Carlson. In silico analysis of functionalized hydrocarbon production using ehrlich pathway and fatty acid derivatives in an endophytic fungus. 7(6):435–435, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jof7060435">[DOI]</a>.</p></li>
<li id="id14431"><p>Hossein Firoozabadi, Mohammad Mahdi Mardanpour, and Ehsan Motamedian. A system-oriented strategy to enhance electron production of synechocystis sp. pcc6803 in bio-photovoltaic devices: experimental and modeling insights. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.22541/au.160978513.33613285/v1">[DOI]</a>.</p></li>
<li id="id14432"><p>Aidin Behravan, Atieh Hashemi, Sayed‐Amir Marashi, and Hamideh Fouladiha. Genome-scale metabolic model based engineering of escherichia coli enhances recombinant single chain antibody fragment production. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-412220/v1">[DOI]</a>.</p></li>
<li id="id14433"><p>Romain Minebois. Study of the differences in the fermentative metabolism of s. cerevisiae, s. uvarum and s. kudriavzevii species. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/176018">[DOI]</a>.</p></li>
<li id="id14473"><p>Ying Li, He Xian, Ya Xu, Yuan Zhu, Zhijie Sun, Qian Wang, and Qingsheng Qi. Additional file 2 of fine tuning the glycolytic flux ratio of ep-bifido pathway for mevalonate production by enhancing glucose-6-phosphate dehydrogenase (zwf) and crispri suppressing 6-phosphofructose kinase (pfka) in escherichia coli. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.13698050">[DOI]</a>.</p></li>
<li id="id14474"><p>Zongjie Dai, Yan Zhu, Hongjun Dong, Chunhua Zhao, Yanping Zhang, and Yin Li. Additional file 2 of enforcing atp hydrolysis enhanced anaerobic glycolysis and promoted solvent production in clostridium acetobutylicum. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.15079523">[DOI]</a>.</p></li>
<li id="id14497"><p>Allon Wagner, Chao Wang, Johannes Fessler, David DeTomaso, Julián Ávila-Pacheco, James J. Kaminski, Sarah Zaghouani, Elena Christian, Pratiksha I. Thakore, Brandon Schellhaass, Elliot H. Akama‐Garren, Kerry A. Pierce, Vasundhara Singh, Noga Ron‐Harel, Vivian Paraskevi Douglas, Lloyd Bod, Alexandra Schnell, Daniel J. Puleston, Raymond A. Sobel, Marcia C. Haigis, Erika L. Pearce, Manoocher Soleimani, Clary B. Clish, Aviv Regev, Vijay K. Kuchroo, and Nir Yosef. Metabolic modeling of single th17 cells reveals regulators of autoimmunity. 184(16):4168–4185.e21, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2021.05.045">[DOI]</a>.</p></li>
<li id="id14503"><p>Laura Judith Marcos-Zambrano, Kanita Karađuzović-Hadžiabdić, Tatjana Lončar-Turukalo, Piotr Przymus, Vladimir Trajkovik, Oliver Aasmets, Magali Berland, Aleksandra Gruca, Jasminka Hasić Telalović, Karel Hron, Thomas Klammsteiner, Mikhail Kolev, Leo Lahti, Marta B. Lopes, Vı́ctor Moreno, Irina Naskinova, Elin Org, Inês Paciência, Γεώργιος Παπουτσόγλου, Rajesh Shigdel, Blaž Stres, Baiba Vilne, Malik Yousef, Eftim Zdravevski, Ioannis Tsamardinos, Enrique Carrillo de Santa Pau, Marcus J. Claesson, Isabel Moreno‐Indias, and Jaak Truu. Applications of machine learning in human microbiome studies: a review on feature selection, biomarker identification, disease prediction and treatment. 12():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2021.634511">[DOI]</a>.</p></li>
<li id="id14519"><p>Farideh Hosseinkhani, Almut Heinken, Ines Thiele, Petrus W. Lindenburg, Amy C. Harms, and Thomas Hankemeier. The contribution of gut bacterial metabolites in the human immune signaling pathway of non-communicable diseases. 13(1):1–22, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2021.1882927">[DOI]</a>.</p></li>
<li id="id14524"><p>Ilija Dukovski, Djordje Bajić, Jeremy M. Chacón, Michael Quintin, Jean C. C. Vila, Snorre Sulheim, Alan R. Pacheco, David B. Bernstein, William J. Riehl, Kirill S. Korolev, Álvaro Sánchez, William R. Harcombe, and Daniel Segrè. A metabolic modeling platform for the computation of microbial ecosystems in time and space (comets). 16(11):5030–5082, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-021-00593-3">[DOI]</a>.</p></li>
<li id="id14527"><p>Ruiwen Hu, Hai-Ming Zhao, Xihui Xu, Zhigang Wang, Ke Yu, Longfei Shu, Qingyun Yan, Bo Wu, Ce-Hui Mo, Zhili He, and Cheng Wang. Bacteria-driven phthalic acid ester biodegradation: current status and emerging opportunities. 154():106560–106560, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.envint.2021.106560">[DOI]</a>.</p></li>
<li id="id14528"><p>Anurag Passi, Juan D. Tibocha‐Bonilla, Manish Kumar, Diego Tec-Campos, Karsten Zengler, and Cristal Zúñiga. Genome-scale metabolic modeling enables in-depth understanding of big data. 12(1):14–14, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo12010014">[DOI]</a>.</p></li>
<li id="id14531"><p>David B. Bernstein, Snorre Sulheim, Eivind Almaas, and Daniel Segrè. Addressing uncertainty in genome-scale metabolic model reconstruction and analysis. 22(1):64–64, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13059-021-02289-z">[DOI]</a>.</p></li>
<li id="id14533"><p>Vijay R. Varma, Hatice Büşra Lüleci̇, Anup Mammen Oommen, Sudhir Varma, Chad Blackshear, Michael Griswold, Yang An, Jackson A. Roberts, Richard O’Brien, Olga Pletniková, Juan C. Troncoso, David A. Bennett, Tunahan Çakır, Cristina Legido‐Quigley, and Madhav Thambisetty. Abnormal brain cholesterol homeostasis in alzheimer’s disease—a targeted metabolomic and transcriptomic study. 7(1):11–11, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41514-021-00064-9">[DOI]</a>.</p></li>
<li id="id14537"><p>J. Lee, Yapeng Su, Priyanka Baloni, Daniel Chen, Ana Jimena Pavlovitch-Bedzyk, Dan Yuan, Venkata R. Duvvuri, Raymond T. Ng, Jongchan Choi, Jingyi Xie, Rongyu Zhang, Kim M. Murray, Sergey A. Kornilov, Brett Smith, Andrew T. Magis, Dave S.�B. Hoon, Jennifer Hadlock, Jason D. Goldman, Nathan D. Price, Raphaël Gottardo, Mark M. Davis, Leroy Hood, Philip D. Greenberg, and James R. Heath. Integrated analysis of plasma and single immune cells uncovers metabolic changes in individuals with covid-19. 40(1):110–120, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41587-021-01020-4">[DOI]</a>.</p></li>
<li id="id14540"><p>Carolin C. M. Schulte, Khushboo Borah, Rachel M. Wheatley, Jason J. Terpolilli, Gerhard Saalbach, Nick Crang, Daan H. de Groot, R. George Ratcliffe, Nicholas J. Kruger, Antonis Papachristodoulou, and Philip S. Poole. Metabolic control of nitrogen fixation in rhizobium-legume symbioses. 7(31):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/sciadv.abh2433">[DOI]</a>.</p></li>
<li id="id14543"><p>Santosh Lamichhane, Partho Sen, Marina Amaral Alves, Henrique Caracho Ribeiro, Peppi Raunioniemi, Tuulia Hyötyläinen, and Matej Orešič. Linking gut microbiome and lipid metabolism: moving beyond associations. 11(1):55–55, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11010055">[DOI]</a>.</p></li>
<li id="id14548"><p>Almut Heinken, Johannes Hertel, and Ines Thiele. Metabolic modelling reveals broad changes in gut microbial metabolism in inflammatory bowel disease patients with dysbiosis. 7(1):19–19, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-021-00178-6">[DOI]</a>.</p></li>
<li id="id14564"><p>Akshit Goyal, Tong Wang, Veronika Dubinkina, and Sergei Maslov. Ecology-guided prediction of cross-feeding interactions in the human gut microbiome. 12(1):1335–1335, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-021-21586-6">[DOI]</a>.</p></li>
<li id="id14565"><p>Alexander Chou, Seung Hwan Lee, Fayin Zhu, James M. Clomburg, and Ramón González. An orthogonal metabolic framework for one-carbon utilization. 3(10):1385–1399, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42255-021-00453-0">[DOI]</a>.</p></li>
<li id="id14566"><p>Yu Chen, Eunice van Pelt‐KleinJan, Berdien van Olst, Sieze Douwenga, Sjef Boeren, Herwig Bachmann, Douwe Molenaar, Jens Nielsen, and Bas Teusink. Proteome constraints reveal targets for improving microbial fitness in nutrient‐rich environments. 17(4):e10093–e10093, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.202010093">[DOI]</a>.</p></li>
<li id="id14570"><p>Xueqin Lv, Yaokang Wu, Mengyue Gong, Jieying Deng, Yang Gu, Yanfeng Liu, Jianghua Li, Guocheng Du, Rodrigo Ledesma‐Amaro, Long Liu, and Jian Chen. Synthetic biology for future food: research progress and future directions. 3():100025–100025, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.fufo.2021.100025">[DOI]</a>.</p></li>
<li id="id14573"><p>St. Elmo Wilken, Jonathan M. Monk, Patrick A. Leggieri, Christopher E. Lawson, Thomas S. Lankiewicz, Susanna Seppälä, Chris Daum, Jerry Jenkins, Anna Lipzen, Stephen J. Mondo, Kerrie Barry, Igor V. Grigoriev, John K. Henske, Michael K. Theodorou, Bernhard Ø. Palsson, Linda Petzold, and Michelle O’Malley. Experimentally validated reconstruction and analysis of a genome-scale metabolic model of an anaerobic neocallimastigomycota fungus. 6(1):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00002-21">[DOI]</a>.</p></li>
<li id="id14574"><p>Almut Heinken, Arianna Basile, and Ines Thiele. Advances in constraint-based modelling of microbial communities. 27():100346–100346, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2021.05.007">[DOI]</a>.</p></li>
<li id="id14577"><p>Barbara Bourgade, Nigel P. Minton, and M. Ahsanul Islam. Genetic and metabolic engineering challenges of c1-gas fermenting acetogenic chassis organisms. 45(2):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsre/fuab008">[DOI]</a>.</p></li>
<li id="id14582"><p>Hui-Yin Chang, Sean Colby, Xiuxia Du, Javier D. Gomez, Maximilian J. Helf, Katerina Kechris, Christine R. Kirkpatrick, Shuzhao Li, Gary J. Patti, Ryan Renslow, Shankar Subramaniam, Mukesh Verma, Jianguo Xia, and Jamey D. Young. A practical guide to metabolomics software development. 93(4):1912–1923, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.analchem.0c03581">[DOI]</a>.</p></li>
<li id="id14583"><p>Zihe Liu, Junyang Wang, and Jens Nielsen. Yeast synthetic biology advances biofuel production. 65():33–39, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mib.2021.10.010">[DOI]</a>.</p></li>
<li id="id14586"><p>Padhmanand Sudhakar, Kathleen Machiels, Bram Verstockt, Tamás Korcsmáros, and Séverine Vermeire. Computational biology and machine learning approaches to understand mechanistic microbiome-host interactions. 12():618856–618856, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2021.618856">[DOI]</a>.</p></li>
<li id="id14587"><p>Jiufu Qin, Anastasia Krivoruchko, Boyang Ji, Yu Chen, Mette Kristensen, Emre Özdemir, Jay D. Keasling, Michael K. Jensen, and Jens Nielsen. Engineering yeast metabolism for the discovery and production of polyamines and polyamine analogues. 4(6):498–509, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41929-021-00631-z">[DOI]</a>.</p></li>
<li id="id14589"><p>Nicholas S. Kruyer, Matthew J. Realff, Wenting Sun, Caroline L. Genzale, and Pamela Peralta‐Yahya. Designing the bioproduction of martian rocket propellant via a biotechnology-enabled in situ resource utilization strategy. 12(1):6166–6166, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-021-26393-7">[DOI]</a>.</p></li>
<li id="id14592"><p>Gianvito Pio, Paolo Mignone, Giuseppe Magazzù, Guido Zampieri, Michelangelo Ceci, and Claudio Angione. Integrating genome-scale metabolic modelling and transfer learning for human gene regulatory network reconstruction. 38(2):487–493, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab647">[DOI]</a>.</p></li>
<li id="id14595"><p>Anne Richelle, Benjamin P. Kellman, Alexander T. Wenzel, Austin W.T. Chiang, Tyler Reagan, Jahir M. Gutierrez, Chintan Joshi, Shangzhong Li, Joanne K. Liu, Helen O. Masson, Jooyong Lee, Zerong Li, Laurent Heirendt, Christophe Trefois, Edwin F. Juarez, Tyler Bath, David Borland, Jill P. Mesirov, Kimberly Robasky, and Nathan E. Lewis. Model-based assessment of mammalian cell metabolic functionalities using omics data. 1(3):100040–100040, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.crmeth.2021.100040">[DOI]</a>.</p></li>
<li id="id14596"><p>Telmo Blasco, Sergio Pérez-Burillo, Francesco Balzerani, Daniel Hinojosa-Nogueira, Alberto Lerma-Aguilera, Silvia Pastoriza, Xabier Cendoya, Ángel Rubio, María José Gosalbes, Nuria Jiménez‐Hernández, M. Pilar Francino, Iñigo Apaolaza, José Ángel Rufián‐Henares, and Francisco J. Planes. An extended reconstruction of human gut microbiota metabolism of dietary compounds. 12(1):4728–4728, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-021-25056-x">[DOI]</a>.</p></li>
<li id="id14599"><p>Yu Chen, Feiran Li, Jiwei Mao, Yun Chen, and Jens Nielsen. Yeast optimizes metal utilization based on metabolic network and enzyme kinetics. 118(12):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2020154118">[DOI]</a>.</p></li>
<li id="id14600"><p>Piyush Nanda and Amit Ghosh. Genome scale-differential flux analysis reveals deregulation of lung cell metabolism on sars-cov-2 infection. 17(4):e1008860–e1008860, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008860">[DOI]</a>.</p></li>
<li id="id14603"><p>Mohammad Karim Khaleghi, Iman Shahidi Pour Savizi, Nathan E. Lewis, and Seyed Abbas Shojaosadati. Synergisms of machine learning and constraint‐based modeling of metabolism for analysis and optimization of fermentation parameters. 16(11):e2100212–e2100212, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.202100212">[DOI]</a>.</p></li>
<li id="id14607"><p>Hongzhong Lu, Feiran Li, Le Yuan, Iván Domenzain, Rosemary Yu, Hao Wang, Gang Li, Yu Chen, Boyang Ji, Eduard J. Kerkhoven, and Jens Nielsen. Yeast metabolic innovations emerged via expanded metabolic network and gene positive selection. 17(10):e10427–e10427, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.202110427">[DOI]</a>.</p></li>
<li id="id14651"><p>Thomas Eng, Deepanwita Banerjee, Andrew K. Lau, Emily Bowden, Robin A. Herbert, Jessica Trinh, Jan‐Philip Prahl, Adam M. Deutschbauer, Deepti Tanjore, and Aindrila Mukhopadhyay. Engineering pseudomonas putida for efficient aromatic conversion to bioproduct using high throughput screening in a bioreactor. 66():229–238, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.04.015">[DOI]</a>.</p></li>
<li id="id14658"><p>Pedro A. Saa, Arlés Urrutia, Claudia Silva, Alberto J. M. Martín, and Daniel Garrido. Modeling approaches for probing cross-feeding interactions in the human gut microbiome. 20():79–89, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2021.12.006">[DOI]</a>.</p></li>
<li id="id14660"><p>Rik P. van Rosmalen, Robert W. Smith, Vítor A. P. Martins dos Santos, Christian Fleck, and María Suárez‐Diez. Model reduction of genome-scale metabolic models as a basis for targeted kinetic models. 64():74–84, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.01.008">[DOI]</a>.</p></li>
<li id="id14661"><p>Steven Le Feunteun, Ahmed Al-Razaz, Matthijs Dekker, Erwin George, Béatrice Laroche, and George A. van Aken. Physiologically based modeling of food digestion and intestinal microbiota: state of the art and future challenges. an infogest review. 12(1):149–167, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1146/annurev-food-070620-124140">[DOI]</a>.</p></li>
<li id="id14662"><p>Sudharshan Ravi and Rudiyanto Gunawan. Δfba—predicting metabolic flux alterations using genome-scale metabolic models and differential transcriptomic data. 17(11):e1009589–e1009589, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009589">[DOI]</a>.</p></li>
<li id="id14665"><p>Yurong Cheng, Pascal Schlosser, Johannes Hertel, Peggy Sekula, Peter J. Oefner, Ute Spiekerkoetter, Johanna Mielke, Daniel F. Freitag, Miriam Schmidts, Peter J. Oefner, Florian Kronenberg, Kai‐Uwe Eckardt, Florian Kronenberg, Kai‐Uwe Eckardt, Ines Thiele, Yong Li, and Anna Köttgen. Rare genetic variants affecting urine metabolite levels link population variation to inborn errors of metabolism. 12(1):964–964, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-20877-8">[DOI]</a>.</p></li>
<li id="id14668"><p>Johannes Hertel, Almut Heinken, Filippo Martinelli Boneschi, and Ines Thiele. Integration of constraint-based modeling with fecal metabolomics reveals large deleterious effects of &lt;i&gt;fusobacterium&lt;/i&gt; spp. on community butyrate production. 13(1):1–23, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2021.1915673">[DOI]</a>.</p></li>
<li id="id14670"><p>Shany Ofaim, Snorre Sulheim, Eivind Almaas, Daniel Sher, and Daniel Segrè. Dynamic allocation of carbon storage and nutrient-dependent exudation in a revised genome-scale model of prochlorococcus. 12():586293–586293, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2021.586293">[DOI]</a>.</p></li>
<li id="id14671"><p>Matthew L. Jenior, Jhansi L. Leslie, Deborah Powers, Elizabeth M. Garrett, Kimberly A. Walker, Mary E. Dickenson, William A. Petri, Rita Tamayo, and Jason A. Papin. Novel drivers of virulence in clostridioides difficile identified via context-specific metabolic network analysis. 6(5):e0091921–e0091921, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00919-21">[DOI]</a>.</p></li>
<li id="id14675"><p>Giuseppe Magazzù, Guido Zampieri, and Claudio Angione. Multimodal regularized linear models with flux balance analysis for mechanistic integration of omics data. 37(20):3546–3552, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab324">[DOI]</a>.</p></li>
<li id="id14679"><p>Jingru Zhou, Yingping Zhuang, and Jianye Xia. Integration of enzyme constraints in a genome-scale metabolic model of aspergillus niger improves phenotype predictions. 20(1):125–125, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-021-01614-2">[DOI]</a>.</p></li>
<li id="id14687"><p>Bonnie V. Dougherty, Kristopher D. Rawls, Glynis L. Kolling, Kalyan C. Vinnakota, Anders Wallqvist, and Jason A. Papin. Identifying functional metabolic shifts in heart failure with the integration of omics data and a heart-specific, genome-scale model. 34(10):108836–108836, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2021.108836">[DOI]</a>.</p></li>
<li id="id14689"><p>Mattia G. Gollub, Hans‐Michael Kaltenbach, and Jörg Stelling. Probabilistic thermodynamic analysis of metabolic networks. 37(18):2938–2945, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab194">[DOI]</a>.</p></li>
<li id="id14690"><p>Christian Schulz, Tjaša Kumelj, Emil Karlsen, and Eivind Almaas. Genome-scale metabolic modelling when changes in environmental conditions affect biomass composition. 17(5):e1008528–e1008528, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008528">[DOI]</a>.</p></li>
<li id="id14691"><p>Nam Kyu Kang, Minsik Kim, Kwangryul Baek, Yong Keun Chang, Donald R. Ort, and Yong‐Su Jin. Photoautotrophic organic acid production: glycolic acid production by microalgal cultivation. 433():133636–133636, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cej.2021.133636">[DOI]</a>.</p></li>
<li id="id14696"><p>Michael A. Henson. Interrogation of the perturbed gut microbiota in gouty arthritis patients through in silico metabolic modeling. 21(7):489–501, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/elsc.202100003">[DOI]</a>.</p></li>
<li id="id14698"><p>Mario L. Arrieta‐Ortiz, Selva Rupa Christinal Immanuel, Serdar Turkarslan, Wei‐Ju Wu, Brintha Girinathan, Jay Worley, Nicholas DiBenedetto, Olga Soutourina, Johann Peltier, Bruno Dupuy, Lynn Bry, and Nitin S. Baliga. Predictive regulatory and metabolic network models for systems analysis of clostridioides difficile. 29(11):1709–1723.e5, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.chom.2021.09.008">[DOI]</a>.</p></li>
<li id="id14704"><p>Marouen Ben Guebila and Ines Thiele. Dynamic flux balance analysis of whole-body metabolism for type 1 diabetes. 1(5):348–361, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s43588-021-00074-3">[DOI]</a>.</p></li>
<li id="id14705"><p>William T. Scott, Eddy J. Smid, David E. Block, and Richard A. Notebaart. Metabolic flux sampling predicts strain-dependent differences related to aroma production among commercial wine yeasts. 20(1):204–204, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-021-01694-0">[DOI]</a>.</p></li>
<li id="id14708"><p>Pablo Rodríguez-Mier, Nathalie Poupin, Carlo De Blasio, Laurent Le Cam, and Fabien Jourdan. Dexom: diversity-based enumeration of optimal context-specific metabolic networks. 17(2):e1008730–e1008730, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008730">[DOI]</a>.</p></li>
<li id="id14709"><p>Tobias B. Alter, Lars M. Blank, and Birgitta E. Ebert. Proteome regulation patterns determine escherichia coli wild-type and mutant phenotypes. 6(2):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00625-20">[DOI]</a>.</p></li>
<li id="id14710"><p>Viviënne Mol, Martyn Bennett, Benjamín J. Sánchez, Beata K. Lisowska, Markus J. Herrgård, Alex Toftgaard Nielsen, David J. Leak, and Nikolaus Sonnenschein. Genome-scale metabolic modeling of p. thermoglucosidasius ncimb 11955 reveals metabolic bottlenecks in anaerobic metabolism. 65():123–134, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.03.002">[DOI]</a>.</p></li>
<li id="id14711"><p>Junmin Wang, Alireza Delfarah, Patrick E. Gelbach, Emma J. Fong, Paul Macklin, Shannon M. Mumenthaler, Nicholas A. Graham, and Stacey D. Finley. Elucidating tumor-stromal metabolic crosstalk in colorectal cancer through integration of constraint-based models and lc-ms metabolomics. 69():175–187, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2021.11.006">[DOI]</a>.</p></li>
<li id="id14712"><p>Sven Thiele, Axel von Kamp, Pavlos Stephanos Bekiaris, Philipp Schneider, and Steffen Klamt. Cnapy: a cellnetanalyzer gui in python for analyzing and designing metabolic networks. 38(5):1467–1469, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab828">[DOI]</a>.</p></li>
<li id="id14713"><p>Iván Domenzain, Benjamín J. Sánchez, Mihail Anton, Eduard J. Kerkhoven, Aarón Millán-Oropeza, Céline Henry, Verena Siewers, John P. Morrissey, Nikolaus Sonnenschein, and Jens Nielsen. Reconstruction of a catalogue of genome-scale metabolic models with enzymatic constraints using gecko 2.0. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.05.433259">[DOI]</a>.</p></li>
<li id="id14717"><p>Krzysztof Ciomek and Miłosz Kadziński. Polyrun: a java library for sampling from the bounded convex polytopes. 13():100659–100659, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.softx.2021.100659">[DOI]</a>.</p></li>
<li id="id14718"><p>Nana Y. D. Ankrah, Brandon Barker, Joan Song, Cindy Wu, John G. McMullen, and Angela E. Douglas. Predicted metabolic function of the gut microbiota of drosophila melanogaster. 6(3):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.01369-20">[DOI]</a>.</p></li>
<li id="id14719"><p>Elif Esvap and Kutlu Ö. Ülgen. Advances in genome-scale metabolic modeling toward microbial community analysis of the human microbiome. 10(9):2121–2137, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.1c00140">[DOI]</a>.</p></li>
<li id="id14720"><p>Egils Stalidzāns and Elīna Dāce. Sustainable metabolic engineering for sustainability optimisation of industrial biotechnology. 19():4770–4776, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2021.08.034">[DOI]</a>.</p></li>
<li id="id14723"><p>Priyanka Baloni, Matthias Arnold, Herman Moreno, Kwangsik Nho, Luna Buitrago, Kevin Huynh, Barbara Brauner, Gregory Louie, Alexandra Kueider‐Paisley, Karsten Suhre, Andrew J. Saykin, Kim Ekroos, Peter J. Meikle, Leroy Hood, Nathan D. Price, P. Murali Doraiswamy, Cory C. Funk, Gabi Kastenmüller, Rebecca Baillie, Xianlin Han, and Rima Kaddurah‐Daouk. Multi-omic analyses characterize the ceramide/sphingomyelin pathway as a therapeutic target in alzheimer’s disease. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.16.21260601">[DOI]</a>.</p></li>
<li id="id14724"><p>Elisabeth Yaneske, Guido Zampieri, Loris Bertoldi, Giuseppe Benvenuto, and Claudio Angione. Genome‐scale metabolic modelling of sars‐cov‐2 in cancer cells reveals an increased shift to glycolytic energy production. 595(18):2350–2365, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/1873-3468.14180">[DOI]</a>.</p></li>
<li id="id14731"><p>Cheewin Kittikunapong, Suhui Ye, Patricia Magadán‐Corpas, Álvaro Pérez‐Valero, Claudio J. Villar, Felipe Lombó, and Eduard J. Kerkhoven. Reconstruction of a genome-scale metabolic model of streptomyces albus j1074: improved engineering strategies in natural product synthesis. 11(5):304–304, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11050304">[DOI]</a>.</p></li>
<li id="id14732"><p>Pejman Salahshouri, Modjtaba Emadi‐Baygi, Mahdi Jalili, Faiz M. Khan, Olaf Wolkenhauer, and Ali Salehzadeh‐Yazdi. A metabolic model of intestinal secretions: the link between human microbiota and colorectal cancer progression. 11(7):456–456, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11070456">[DOI]</a>.</p></li>
<li id="id14733"><p>Mihir V. Shah, Hadi Nazem‐Bokaee, James Antoney, Suk Woo Kang, Colin J. Jackson, and Colin Scott. Improved production of the non-native cofactor f420 in escherichia coli. 11(1):21774–21774, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-021-01224-3">[DOI]</a>.</p></li>
<li id="id14737"><p>William T. Scott, Oscar van Mastrigt, David E. Block, Richard A. Notebaart, and Eddy J. Smid. Nitrogenous compound utilization and production of volatile organic compounds among commercial wine yeasts highlight strain-specific metabolic diversity. 9(1):e0048521–e0048521, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/spectrum.00485-21">[DOI]</a>.</p></li>
<li id="id14738"><p>Itziar Frades, Carles Foguet, Marta Cascante, and Marcos J. Araúzo‐Bravo. Genome scale modeling to study the metabolic competition between cells in the tumor microenvironment. 13(18):4609–4609, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cancers13184609">[DOI]</a>.</p></li>
<li id="id14748"><p>Anja Karlstaedt. Stable isotopes for tracing cardiac metabolism in diseases. 8():734364–734364, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcvm.2021.734364">[DOI]</a>.</p></li>
<li id="id14756"><p>Dung Hoang Anh, Nguyễn Thị Hoài Thu, and Eun Yeol Lee. The ethylmalonyl-coa pathway for methane-based biorefineries: a case study of using &lt;i&gt;methylosinus trichosporium&lt;/i&gt; ob3b, an alpha-proteobacterial methanotroph, for producing 2-hydroxyisobutyric acid and 1,3-butanediol from methane. 23(19):7712–7723, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d1gc02866a">[DOI]</a>.</p></li>
<li id="id14757"><p>Theresa L. Whiteside. The role of tumor-derived exosomes (tex) in shaping anti-tumor immune competence. 10(11):3054–3054, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cells10113054">[DOI]</a>.</p></li>
<li id="id14778"><p>Axel Theorell, Johann F. Jadebeck, Katharina Nöh, and Jörg Stelling. Polyround: polytope rounding for random sampling in metabolic networks. 38(2):566–567, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab552">[DOI]</a>.</p></li>
<li id="id14786"><p>Alvar J. Alonso-Lavin, Djordje Bajić, and Juan F. Poyatos. Tolerance to nadh/nad+ imbalance anticipates aging and anti-aging interventions. 24(7):102697–102697, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.102697">[DOI]</a>.</p></li>
<li id="id14839"><p>Nunthaphan Vikromvarasiri, Tomokazu Shirai, and Akihiko Kondo. Metabolic engineering design to enhance (r,r)-2,3-butanediol production from glycerol in bacillus subtilis based on flux balance analysis. 20(1):196–196, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-021-01688-y">[DOI]</a>.</p></li>
<li id="id14842"><p>Wang XinHui. Advances in candida albicans. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.5772/intechopen.87410">[DOI]</a>.</p></li>
<li id="id14845"><p>Varshit Dusad, Denise Thiel, Mauricio Barahona, Hector C. Keun, and Diego A. Oyarzún. Opportunities at the interface of network science and metabolic modeling. 8():591049–591049, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2020.591049">[DOI]</a>.</p></li>
<li id="id14850"><p>Nadeem Joudeh, Athanasios Saragliadis, Christian Schulz, André Voigt, Eivind Almaas, and Dirk Linke. Transcriptomic response analysis of escherichia coli to palladium stress. 12():741836–741836, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2021.741836">[DOI]</a>.</p></li>
<li id="id14853"><p>Ana Vila-Santa, M. Ahsanul Islam, Frederico Castelo Ferreira, Kristala L. J. Prather, and Nuno P. Mira. Prospecting biochemical pathways to implement microbe-based production of the new-to-nature platform chemical levulinic acid. 10(4):724–736, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.0c00518">[DOI]</a>.</p></li>
<li id="id14855"><p>Iván Domenzain, Benjamín J. Sánchez, Mihail Anton, Eduard J. Kerkhoven, Aarón Millán-Oropeza, Céline Henry, Verena Siewers, John P. Morrissey, Nikolaus Sonnenschein, and Jens Nielsen. Reconstruction of a catalogue of genome-scale metabolic models with enzymatic constraints using gecko 2.0. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-351350/v1">[DOI]</a>.</p></li>
<li id="id14856"><p>Ewelina Węglarz‐Tomczak, Thierry D.G.A. Mondeel, Diewertje Piebes, and Hans V. Westerhoff. Simultaneous integration of gene expression and nutrient availability for studying the metabolism of hepatocellular carcinoma cell lines. 11(4):490–490, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biom11040490">[DOI]</a>.</p></li>
<li id="id14857"><p>John P. Thomas, Dezső Módos, Tamás Korcsmáros, and Johanne Brooks-Warburton. Network biology approaches to achieve precision medicine in inflammatory bowel disease. 12():760501–760501, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2021.760501">[DOI]</a>.</p></li>
<li id="id14862"><p>Daniel Noreña-Caro, Cristal Zúñiga, Amber J. Pete, Sven A. Saemundsson, Morgan R. Donaldson, Alexandria J. Adams, Kerry M. Dooley, Karsten Zengler, and Michael G. Benton. Analysis of the cyanobacterial amino acid metabolism with a precise genome-scale metabolic reconstruction of anabaena sp. utex 2576. 171():108008–108008, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2021.108008">[DOI]</a>.</p></li>
<li id="id14863"><p>Indu Jatain, Kashyap Kumar Dubey, Manisha Sharma, Zeba Usmani, Minaxi Sharma, and Vijai Kumar Gupta. Synthetic biology potential for carbon sequestration into biocommodities. 323():129176–129176, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jclepro.2021.129176">[DOI]</a>.</p></li>
<li id="id14864"><p>Maziya Ibrahim and Karthik Raman. Two-species community design of lactic acid bacteria for optimal production of lactate. 19():6039–6049, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2021.11.009">[DOI]</a>.</p></li>
<li id="id14873"><p>Hennicke Kamp, Judith Wahrheit, Stefan Stinchcombe, Tilmann Walk, Franz Stauber, and B. van Ravenzwaay. Succinate dehydrogenase inhibitors: in silico flux analysis and in vivo metabolomics investigations show no severe metabolic consequences for rats and humans. 150():112085–112085, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.fct.2021.112085">[DOI]</a>.</p></li>
<li id="id14875"><p>Peng Liu, Shuai Wang, Chao Li, Yingping Zhuang, Jianye Xia, and Henk Noorman. Dynamic response of &lt;i&gt;aspergillus niger&lt;/i&gt; to periodical glucose pulse stimuli in chemostat cultures. 118(6):2265–2282, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27739">[DOI]</a>.</p></li>
<li id="id14880"><p>Iván Domenzain, Feiran Li, Eduard J. Kerkhoven, and Verena Siewers. Evaluating accessibility, usability and interoperability of genome-scale metabolic models for diverse yeasts species. 21(1):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsyr/foab002">[DOI]</a>.</p></li>
<li id="id14882"><p>Ecehan Abdi̇k and Tunahan Çakır. Systematic investigation of mouse models of parkinson's disease by transcriptome mapping on a brain-specific genome-scale metabolic network. 17(4):492–502, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d0mo00135j">[DOI]</a>.</p></li>
<li id="id14883"><p>Felipe Scott, Luz Yáñez, Raúl Conejeros, Blanca Araya, and Alberto Vergara‐Fernández. Two internal bottlenecks cause the overflow metabolism leading to poly(3-hydroxybutyrate) production in azohydromonas lata dsm1123. 9(4):105665–105665, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jece.2021.105665">[DOI]</a>.</p></li>
<li id="id14884"><p>Benjamin Allen, Nidhi Gupta, Janaka N. Edirisinghe, José P. Faria, and Christopher S. Henry. Application of the metabolic modeling pipeline in kbase to categorize reactions, predict essential genes, and predict pathways in an isolate genome. 2349():291–320, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1585-0_13">[DOI]</a>.</p></li>
<li id="id14892"><p>Kyoichi Ebata, Sawa Yamashiro, Keita Iida, and Mariko Okada. Building patient‐specific models for receptor tyrosine kinase signaling networks. 289(1):90–101, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/febs.15831">[DOI]</a>.</p></li>
<li id="id14893"><p>St. Elmo Wilken, Victor Vera Frazão, Nima P. Saadat, and Oliver Ebenhöh. The view of microbes as energy converters illustrates the trade-off between growth rate and yield. 49(4):1663–1674, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20200977">[DOI]</a>.</p></li>
<li id="id14894"><p>Chenchen Gao, Jiarui Yang, Tong Hao, Jingjing Li, and Jinsheng Sun. Reconstruction of litopenaeus vannamei genome-scale metabolic network model and nutritional requirements analysis of different shrimp commercial varieties. 12():658109–658109, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2021.658109">[DOI]</a>.</p></li>
<li id="id14895"><p>Priyanka Baloni, Cory C. Funk, Ben Readhead, and Nathan D. Price. Systems modeling of metabolic dysregulation in neurodegenerative diseases. 60():59–65, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coph.2021.06.012">[DOI]</a>.</p></li>
<li id="id14896"><p>Rudolfs Petrovs, Egils Stalidzāns, and Agris Pentjuss. Imfler: a web application for interactive metabolic flux analysis and visualization. 28(10):1021–1032, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2021.0056">[DOI]</a>.</p></li>
<li id="id14897"><p>Puhua Niu, Maria J. Soto, Byung-Jun Yoon, Edward R. Dougherty, Francis J. Alexander, Ian K. Blaby, and Xiaoning Qian. Trimer: transcription regulation integrated with metabolic regulation. 24(11):103218–103218, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.103218">[DOI]</a>.</p></li>
<li id="id14898"><p>Miroslav Kratochvíl, Laurent Heirendt, St. Elmo Wilken, Taneli Pusa, Sylvain Arreckx, Alberto Noronha, Marvin van Aalst, Venkata Satagopam, Oliver Ebenhöh, Reinhard Schneider, Christophe Trefois, and Wei Gu. Cobrexa.jl: constraint-based reconstruction and exascale analysis. 38(4):1171–1172, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab782">[DOI]</a>.</p></li>
<li id="id14920"><p>Albert Tafur Rangel, Wendy Ríos, Daisy Mejía, Carmen Ojeda, Ross P. Carlson, Jorge M. Gómez, and Andrés Fernando González Barrios. In silico design for systems-based metabolic engineering for the bioconversion of valuable compounds from industrial by-products. 12():633073–633073, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2021.633073">[DOI]</a>.</p></li>
<li id="id14921"><p>Howard Ramírez-Malule, Víctor A. López-Agudelo, David Gómez-Ríos, Silvia Ochoa, Rigoberto Ríos‐Estepa, Stefan Junne, and Peter Neubauer. Tca cycle and its relationship with clavulanic acid production: a further interpretation by using a reduced genome-scale metabolic model of streptomyces clavuligerus. 8(8):103–103, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/bioengineering8080103">[DOI]</a>.</p></li>
<li id="id14922"><p>Carl Malina, Francesca Di Bartolomeo, Eduard J. Kerkhoven, and Jens Nielsen. Constraint-based modeling of yeast mitochondria reveals the dynamics of protein import and iron-sulfur cluster biogenesis. 24(11):103294–103294, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.isci.2021.103294">[DOI]</a>.</p></li>
<li id="id14934"><p>John I. Hendry, Hoang V. Dinh, Debolina Sarkar, Lin Wang, Anindita Bandyopadhyay, Himadri B. Pakrasi, and Costas D. Maranas. A genome-scale metabolic model of anabaena 33047 to guide genetic modifications to overproduce nylon monomers. 11(3):168–168, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11030168">[DOI]</a>.</p></li>
<li id="id14935"><p>Wentao Tang, Tianwei Hao, and Guanghao Chen. Comparative metabolic modeling of multiple sulfate‐reducing prokaryotes reveals versatile energy conservation mechanisms. 118(7):2676–2693, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27787">[DOI]</a>.</p></li>
<li id="id14944"><p>Omid Ardalani, Ehsan Motamedian, and Javad Hamedi. Reconstruction and validation of genome-scale metabolic model of l. lactis subsp. lactis ncdo 2118 and in silico analysis for succinate and gamma-aminobutyric acid overproduction. 170():107967–107967, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2021.107967">[DOI]</a>.</p></li>
<li id="id14945"><p>Baoyu Xiang, Liping Zhao, and Menghui Zhang. Metagenome-scale metabolic network suggests folate produced by bifidobacterium longum might contribute to high-fiber-diet-induced weight loss in a prader–willi syndrome child. 9(12):2493–2493, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms9122493">[DOI]</a>.</p></li>
<li id="id14961"><p>Zachary Johnson, Dennis D. Krutkin, Pavlo Bohutskyi, and Marina Kalyuzhnaya. Metals and methylotrophy: via global gene expression studies. 650():185–213, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.mie.2021.01.046">[DOI]</a>.</p></li>
<li id="id14986"><p>Gholamreza Bidkhori, Sunjae Lee, Lindsey Edwards, Emmanuelle Le Chatelier, Mathieu Almeida, Bouchra Ezzamouri, Florian Plaza Oñate, Nicolas Ponte, Debbie L. Shawcross, Gordon Proctor, Lars K. Nielsen, Jens Nielsen, Mathias Uhlén, S. Dusko Ehrlich, and Saeed Shoaie. The reactobiome unravels a new paradigm in human gut microbiome metabolism. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.01.428114">[DOI]</a>.</p></li>
<li id="id14991"><p>Vetle Simensen, André Voigt, and Eivind Almaas. High‐quality genome‐scale metabolic model of &lt;i&gt;aurantiochytrium&lt;/i&gt; sp. t66. 118(5):2105–2117, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27726">[DOI]</a>.</p></li>
<li id="id14995"><p>Michael A. Henson. Computational modeling of the gut microbiota reveals putative metabolic mechanisms of recurrent clostridioides difficile infection. 17(2):e1008782–e1008782, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008782">[DOI]</a>.</p></li>
<li id="id14996"><p>Snorre Sulheim, Fredrik A. Fossheim, Alexander Wentzel, and Eivind Almaas. Automatic reconstruction of metabolic pathways from identified biosynthetic gene clusters. 22(1):81–81, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-021-03985-0">[DOI]</a>.</p></li>
<li id="id14997"><p>Despoina Kerselidou, Bushra Dohai, David R. Nelson, Sarah Daakour, Nicolas De Cock, Zahra Al Oula Hassoun, Dae‐Kyum Kim, Julien Olivet, Diana C. El Assal, Ashish Jaiswal, Amnah Alzahmi, Deeya Saha, Charlotte Pain, Filip Matthijssens, Pierre Lemaître, Michaël Herfs, Julien Chapuis, Bart Ghesquière, Didier Vertommen, Verena Kriechbaumer, Kèvin Knoops, Carmen López‐Iglesias, Marc van Zandvoort, Jean‐Charles Lambert, Julien Hanson, Christophe Desmet, Marc Thiry, Kyle J. Lauersen, Marc Vidal, Pieter Van Vlierberghe, Samuel Dequiedt, Kourosh Salehi‐Ashtiani, and Jean‐Claude Twizere. Alternative glycosylation controls endoplasmic reticulum dynamics and tubular extension in mammalian cells. 7(19):, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/sciadv.abe8349">[DOI]</a>.</p></li>
<li id="id14998"><p>Chao-Ting Cheng, Tsun-Yu Wang, Peirong Chen, Wu‐Hsiung Wu, Jinmei Lai, Peter Mu‐Hsin Chang, Yi‐Ren Hong, Chi‐Ying F. Huang, and Feng‐Sheng Wang. Computer-aided design for identifying anticancer targets in genome-scale metabolic models of colon cancer. 10(11):1115–1115, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biology10111115">[DOI]</a>.</p></li>
<li id="id15001"><p>Ewelina Węglarz‐Tomczak, Demi J. Rijlaarsdam, Jakub M. Tomczak, and Stanley Brul. Gem-based metabolic profiling for human bone osteosarcoma under different glucose and glutamine availability. 22(3):1470–1470, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/ijms22031470">[DOI]</a>.</p></li>
<li id="id15006"><p>Supreeta Vijayakumar and Claudio Angione. Protocol for hybrid flux balance, statistical, and machine learning analysis of multi-omic data from the cyanobacterium synechococcus sp. pcc 7002. 2(4):100837–100837, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xpro.2021.100837">[DOI]</a>.</p></li>
<li id="id15007"><p>Chaitra Sarathy, Marian Breuer, Martina Kutmon, Michiel Adriaens, Chris T. Evelo, and Ilja C.W. Arts. Comparison of metabolic states using genome-scale metabolic models. 17(11):e1009522–e1009522, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009522">[DOI]</a>.</p></li>
<li id="id15017"><p>Özlem Altay, Cheng Zhang, Hasan Türkez, Jens Nielsen, Mathias Uhlén, and Adil Mardinoğlu. Revealing the metabolic alterations during biofilm development of burkholderia cenocepacia based on genome-scale metabolic modeling. 11(4):221–221, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11040221">[DOI]</a>.</p></li>
<li id="id15018"><p>Marco Fondi, Stefano Gonzi, Mikołaj Dziurzyński, Paola Turano, Veronica Ghini, Marzia Calvanese, Andrea Colarusso, Concetta Lauro, Ermenegilda Parrilli, and Maria Luisa Tutino. Modelling hcdkl5 heterologous expression in bacteria. 11(8):491–491, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11080491">[DOI]</a>.</p></li>
<li id="id15019"><p>Selva Rupa Christinal Immanuel, Mario L. Arrieta‐Ortiz, Rene A. Ruiz, Min Pan, Adrián López García de Lomana, Eliza J. R. Peterson, and Nitin S. Baliga. Quantitative prediction of conditional vulnerabilities in regulatory and metabolic networks using prime. 7(1):43–43, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-021-00205-6">[DOI]</a>.</p></li>
<li id="id15039"><p>William F. Kindschuh, Federico Baldini, Martin C. Liu, Kristin D. Gerson, Jingqiu Liao, Harry H. Lee, Lauren Anton, Pawel Gajer, Jacques Ravel, Maayan Levy, Michal A. Elovitz, and Tal Korem. Preterm birth is associated with xenobiotics and predicted by the vaginal metabolome. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.14.448190">[DOI]</a>.</p></li>
<li id="id15040"><p>Donald James, P. R. Rennya, Mani Deepika Mallavarapu, Ram Chandra Panigrahi, and Hitendra Kumar Patel. Integrating pan-omics data in a systems approach for crop improvement: opportunities and challenges. ():215–246, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-16-2956-3_8">[DOI]</a>.</p></li>
<li id="id15041"><p>Mario Latendresse, Wai Kit Ong, and Peter D. Karp. Metabolic modeling with metaflux. 2349():259–289, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-1585-0_12">[DOI]</a>.</p></li>
<li id="id15058"><p>Freddy Castillo-Alfonso, Gabriel Vigueras-Ramírez, Luis Manuel Rosales-Colunga, Alberto del Monte‐Martínez, and Roberto Olivares‐Hernández. Propionate as the preferred carbon source to produce 3-indoleacetic acid in&lt;i&gt;b. subtilis&lt;/i&gt;: comparative flux analysis using five carbon sources. 17(4):554–564, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/d1mo00039j">[DOI]</a>.</p></li>
<li id="id15059"><p>Albert A. Antolín and Marta Cascante. Ai delivers michaelis constants as fuel for genome-scale metabolic models. 19(10):e3001415–e3001415, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pbio.3001415">[DOI]</a>.</p></li>
<li id="id15074"><p>Sudharshan Ravi and Rudiyanto Gunawan. Δfba: predicting metabolic flux alterations using genome-scale metabolic models and differential transcriptomic data. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.01.18.427188">[DOI]</a>.</p></li>
<li id="id15075"><p>Sanjeev Dahal, Jiao Zhao, and Laurence Yang. Recent advances in genome-scale modeling of proteome allocation. 26():39–45, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2021.04.002">[DOI]</a>.</p></li>
<li id="id15076"><p>Devlin C. Moyer, Alan R. Pacheco, David B. Bernstein, and Daniel Segrè. Stoichiometric modeling of artificial string chemistries reveals constraints on metabolic network structure. 89(7):472–483, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00239-021-10018-0">[DOI]</a>.</p></li>
<li id="id15077"><p>J. Saraiva, Alexander Bartholomäus, René Kallies, Marta Gomes, Marcos Bicalho, Jonas Coelho Kasmanas, Carsten Vogt, Antonis Chatzinotas, Peter F. Stadler, Óscar Dias, and Ulisses Nunes da Rocha. Ortsuite: from genomes to prediction of microbial interactions within targeted ecosystem processes. 4(12):e202101167–e202101167, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.26508/lsa.202101167">[DOI]</a>.</p></li>
<li id="id15078"><p>M. Águila Ruiz‐Sola, Serena Flori, Yizhong Yuan, Gaëlle Villain, Emanuel Sanz‐Luque, Petra Redekop, Ryutaro Tokutsu, Anika Küken, Angeliki Tsichla, Georgios Kepesidis, Guillaume Allorent, Marius Arend, Fabrizio Iacono, Giovanni Finazzi, Michael Hippler, Zoran Nikoloski, Jun Minagawa, Arthur Grossman, and Dimitris Petroutsos. Light-independent regulation of algal photoprotection by co&lt;sub&gt;2&lt;/sub&gt;availability. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.10.23.465040">[DOI]</a>.</p></li>
<li id="id15155"><p>Dafni Giannari, Cleo Han-Chen Ho, and Radhakrishnan Mahadevan. A gap-filling algorithm for prediction of metabolic interactions in microbial communities. 17(11):e1009060–e1009060, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1009060">[DOI]</a>.</p></li>
<li id="id15161"><p>Jibin Liu, Anchun Cheng, Mingshu Wang, Mafeng Liu, Dekang Zhu, Qiao Yang, Ying Wu, Renyong Jia, Shun Chen, Xinxin Zhao, Shaqiu Zhang, Juan Huang, Xumin Ou, Sai Mao, Qun Gao, Xingjian Wen, Ling Zhang, Yunya Liu, Yanling Yu, Bin Tian, Leichang Pan, Mujeeb Ur Rehman, and Xiaoyue Chen. Comparative genomics and metabolomics analysis of riemerella anatipestifer strain ch-1 and ch-2. 11(1):616–616, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-79733-w">[DOI]</a>.</p></li>
<li id="id15162"><p>Upasana Jhariya, Shweta Srivastava, Sanchita Basu Das, Sakina Bombaywala, Sejal Mahallea, and Nishant A. Dafale. Understanding the role of genetic and protein networking involved in microbial bioremediation. ():187–219, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-86169-8_8">[DOI]</a>.</p></li>
<li id="id15167"><p>Fernando Cruz, João Capela, Eugénio C. Ferreira, Miguel Rocha, and Óscar Dias. &lt;i&gt;bioiso&lt;/i&gt;: an objective-oriented application for assisting the curation of genome-scale metabolic models. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.07.434259">[DOI]</a>.</p></li>
<li id="id15168"><p>William T. Scott, Oscar van Mastrigt, David E. Block, Richard A. Notebaart, and Eddy J. Smid. Nitrogenous compound utilization and production of volatile organic compounds among commercial wine yeasts highlight strain-specific metabolic diversity. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.29.437628">[DOI]</a>.</p></li>
<li id="id15169"><p>Rosemary Yu, Egor Vorontsov, Carina Sihlbom, and Jens Nielsen. Non-random organization of flux control mechanisms in yeast central metabolic pathways. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.12.15.472747">[DOI]</a>.</p></li>
<li id="id15176"><p>Anoop T. Ambikan, Hong Yang, Shuba Krishnan, Sara Svensson Akusjärvi, Soham Gupta, Magda Lourda, Maike Sperk, Muhammad Arif, Cheng Zhang, Hampus Nordqvist, Sivasankaran Munusamy Ponnan, Anders Sönnerborg, Carl Johan Treutiger, Liam O’Mahony, Adil Mardinoğlu, Rui Benfeitas, and Ujjwal Neogi. Multiomics personalized network analyses highlight progressive immune disruption of central metabolism associated with covid-19 severity. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.3988390">[DOI]</a>.</p></li>
<li id="id15190"><p>Jingru Zhou, Yingping Zhuang, and Jianye Xia. Integration of enzyme constraints in a genome-scale metabolic model of aspergillus niger improves phenotype predictions. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-448829/v1">[DOI]</a>.</p></li>
<li id="id15199"><p>Nan Xu, Qiyuan Yang, Xiaojing Yang, Mingqi Wang, and Minliang Guo. Reconstruction and analysis of a genome‐scale metabolic model for &lt;i&gt;agrobacterium tumefaciens&lt;/i&gt;. 22(3):348–360, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/mpp.13032">[DOI]</a>.</p></li>
<li id="id15200"><p>Alexander L. R. Lubbock and Carlos F. Lopez. Programmatic modeling for biological systems. 27():100343–100343, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2021.05.004">[DOI]</a>.</p></li>
<li id="id15201"><p>Sigurður Trausti Karvelsson, Qiong Wang, Bylgja Hilmarsdóttir, Arnar Sigurdsson, Siver Andreas Moestue, Gunhild M. Mælandsmo, Skarphéðinn Halldórsson, Steinn Guðmundsson, and Óttar Rolfsson. Argininosuccinate lyase is a metabolic vulnerability in breast development and cancer. 7(1):36–36, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-021-00195-5">[DOI]</a>.</p></li>
<li id="id15203"><p>Selva Rupa Christinal Immanuel, Mario L. Arrieta‐Ortiz, Rene A. Ruiz, Min Pan, Adrián López García de Lomana, Eliza J. R. Peterson, and Nitin S. Baliga. Quantitative prediction of conditional vulnerabilities in regulatory and metabolic networks of&lt;i&gt;mycobacterium tuberculosis&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.01.29.428876">[DOI]</a>.</p></li>
<li id="id15204"><p>Vetle Simensen, Christian Schulz, Emil Karlsen, Signe Bråtelund, Idun Burgos, Lilja Brekke Thorfinnsdottir, Laura García-Calvo, Per Bruheim, and Eivind Almaas. Quantification of macromolecular biomass composition for constraint-based metabolic modeling. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.08.20.457062">[DOI]</a>.</p></li>
<li id="id15205"><p>Maxime R.F. Gosselin, Virginie Mournetas, Małgorzata Borczyk, Suraj Verma, Annalisa Occhipinti, Justyna Róg, Łukasz Bożycki, Michał Korostyński, Samuel C. Robson, Claudio Angione, Christian Pinset, and Dariusz C. Górecki. Loss of full-length dystrophin expression results in major cell-autonomous abnormalities in proliferating myoblasts. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.08.24.457331">[DOI]</a>.</p></li>
<li id="id15211"><p>M. D. Gasparyan, Arnout Van Messem, and Shodhan Rao. Parameter estimation for models of chemical reaction networks from experimental data of reaction rates. 96(2):392–407, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/00207179.2021.1998636">[DOI]</a>.</p></li>
<li id="id15212"><p>Albert Tafur Rangel, Abel García Oviedo, Freddy Cabrera Mojica, Jorge M. Gómez, and Andrés Fernando González Barrios. Development of an integrating systems metabolic engineering and bioprocess modeling approach for rational strain improvement. 178():108268–108268, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2021.108268">[DOI]</a>.</p></li>
<li id="id15216"><p>Matthew L. Jenior, Jhansi L. Leslie, Deborah Powers, Elizabeth M. Garrett, Kimberly A. Walker, Mary E. Dickenson, William A. Petri, Rita Tamayo, and Jason A. Papin. Conserved virulence-linked metabolic reprogramming in &amp;lt;i&amp;gt;clostridioides difficile&amp;lt;/i&amp;gt; identified through genome-scale metabolic network analysis. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.3773788">[DOI]</a>.</p></li>
<li id="id15217"><p>Viviënne Mol, Martyn Bennett, Benjamín J. Sánchez, Beata K. Lisowska, Markus J. Herrgård, Alex Toftgaard Nielsen, David J. Leak, and Nikolaus Sonnenschein. Genome-scale metabolic modelling of&lt;i&gt;p. thermoglucosidasius&lt;/i&gt;ncimb 11955 reveals metabolic bottlenecks in anaerobic metabolism. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.01.429138">[DOI]</a>.</p></li>
<li id="id15218"><p>Takeyuki Tamura. L1 norm minimal mode-based methods for listing reaction network designs for metabolite production. E104.D(5):679–687, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1587/transinf.2020edp7247">[DOI]</a>.</p></li>
<li id="id15219"><p>Takeyuki Tamura. Efficient reaction deletion algorithms for redesign of constraint-based metabolic networks for metabolite production with weak coupling. 14(0):12–21, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.2197/ipsjtbio.14.12">[DOI]</a>.</p></li>
<li id="id15237"><p>Aidan Pavao, Brintha Girinathan, Johann Peltier, Pamela Altamirano Silva, Bruno Dupuy, Leo L. Cheng, and Lynn Bry. Real-time hrmas&lt;sup&gt;13&lt;/sup&gt;c nmr of obligately anaerobic cells identifies new metabolic targets in the pathogen&lt;i&gt;clostridioides difficile&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.04.442336">[DOI]</a>.</p></li>
<li id="id15238"><p>Dafni Giannari, Cleo Han-Chen Ho, and Radhakrishnan Mahadevan. A gap-filling algorithm for prediction of metabolic interactions in microbial communities. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.13.443977">[DOI]</a>.</p></li>
<li id="id15239"><p>Takeyuki Tamura, Ai Muto-Fujita, Yukako Tohsato, and Tomoyuki Kosaka. Gene deletion algorithms for minimum reaction network design by mixed-integer linear programming for metabolite production in constraint-based models: gdel_minrn. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-611371/v1">[DOI]</a>.</p></li>
<li id="id15306"><p>Almut Heinken, Stefanía Magnúsdóttir, Ronan M. T. Fleming, and Ines Thiele. Demeter: efficient simultaneous curation of genome-scale reconstructions guided by experimental data and refined gene annotations. 37(21):3974–3975, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btab622">[DOI]</a>.</p></li>
<li id="id15307"><p>Juan Pablo Molina Ortiz, Dale D. McClure, Erin R. Shanahan, Fariba Dehghani, Andrew Holmes, and Mark Read. Enabling rational gut microbiome manipulations by understanding gut ecology through experimentally-evidenced in silico models. 13(1):1965698–1965698, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2021.1965698">[DOI]</a>.</p></li>
<li id="id15312"><p>Stuart Maudsley, Hanne Leysen, Jaana van Gastel, and Bronwen Martin. Systems pharmacology: enabling multidimensional therapeutics. ():725–769, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-820472-6.00017-7">[DOI]</a>.</p></li>
<li id="id15313"><p>Thomas Eng, Deepanwita Banerjee, Andrew K. Lau, Emily Bowden, Robin A. Herbert, Jessica Trinh, Jan‐Philip Prahl, Adam M. Deutschbauer, Deepti Tanjore, and Aindrila Mukhopadhyay. High throughput fitness profiling reveals loss of gacs-gaca regulation improves indigoidine production in&lt;i&gt;pseudomonas putida&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.02.429437">[DOI]</a>.</p></li>
<li id="id15314"><p>Carolin C. M. Schulte, Khushboo Borah, Rachel M. Wheatley, Jason J. Terpolilli, Gerhard Saalbach, Nick Crang, Daan H. de Groot, R. George Ratcliffe, Nicholas J. Kruger, Antonis Papachristodoulou, and Philip S. Poole. Metabolic constraints on nitrogen fixation by rhizobia in legume nodules. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.16.431433">[DOI]</a>.</p></li>
<li id="id15315"><p>Sourav Chowdhury, Daniel C. Zielinski, Christopher Dalldorf, João V. Rodrigues, Bernhard Ø. Palsson, and Eugene I. Shakhnovich. Empowering systems-guided drug target discovery with metabolic and structural analysis. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.17.444532">[DOI]</a>.</p></li>
<li id="id15316"><p>Junmin Wang, Alireza Delfarah, Patrick E. Gelbach, Emma J. Fong, Paul Macklin, Shannon M. Mumenthaler, Nicholas A. Graham, and Stacey D. Finley. Elucidating tumor-stromal metabolic crosstalk in colorectal cancer through integration of constraint-based models and lc-ms metabolomics. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.28.446227">[DOI]</a>.</p></li>
<li id="id15317"><p>Laurence Legon, Christophe Corre, Declan G. Bates, and Ahmad A. Mannan. Gcfront: a tool for determining a pareto front of growth-coupled cell factory designs. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.10.12.464108">[DOI]</a>.</p></li>
<li id="id15318"><p>Yougan Cheng, Ronny Straube, Abed Alnaif, Lu Huang, Tarek A. Leil, and Brian J. Schmidt. Virtual populations for quantitative systems pharmacology models. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.31219/osf.io/mzqtk">[DOI]</a>.</p></li>
<li id="id15330"><p>Juan D. Tibocha‐Bonilla, Cristal Zúñiga, Jared T. Broddrick, Karsten Zengler, and Rubén Darío Godoy‐Silva. A multiscale model predicts the sensitivity of&lt;i&gt;chlorella vulgaris&lt;/i&gt;to light and nitrogen levels in photobioreactors. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.04.14.439858">[DOI]</a>.</p></li>
<li id="id15331"><p>Rashi Verma, Dibyabhaba Pradhan, Harpreet Singh, Arun K. Jain, and Luqman Ahmad Khan. Metabolic network modeling for rational drug design against &lt;i&gt;candida albicans&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.5772/intechopen.96749">[DOI]</a>.</p></li>
<li id="id15332"><p>Puhua Niu, Maria J. Soto, Byung-Jun Yoon, Edward R. Dougherty, Francis J. Alexander, Ian K. Blaby, and Xiaoning Qian. Trimer: transcription regulation integrated with metabolic regulation. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.04.09.439167">[DOI]</a>.</p></li>
<li id="id15333"><p>Rosemary Yu, Hao Wang, and Jens Nielsen. Amino acid and nucleotide metabolism shape the selection of trophic levels in animals. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.01.450789">[DOI]</a>.</p></li>
<li id="id15334"><p>Eunice A. Ferreira, Catarina C. Pacheco, João S. Rodrigues, Filipe Pinto, Pedro Lamosa, David Fuente, Javier F. Urchueguía, and Paula Tamagnini. Heterologous production of glycine betaine using synechocystis sp. pcc 6803-based chassis lacking native compatible solutes. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-1031622/v1">[DOI]</a>.</p></li>
<li id="id15376"><p>Nana Y. D. Ankrah, Brandon Barker, Joan Song, Cindy Wu, John G. McMullen, and Angela E. Douglas. The predicted metabolic function of the gut microbiota of&lt;i&gt;drosophila melanogaster&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.01.20.427455">[DOI]</a>.</p></li>
<li id="id15378"><p>Alexander B. Alleman, Florence Mus, and John W. Peters. Metabolic model of nitrogen-fixing obligate aerobe &lt;i&gt;azotobacter vinelandii&lt;/i&gt; demonstrates adaptation to oxygen concentration and metal availability. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.16.448589">[DOI]</a>.</p></li>
<li id="id15379"><p>Miroslav Kratochvíl, Laurent Heirendt, St. Elmo Wilken, Taneli Pusa, Sylvain Arreckx, Alberto Noronha, Marvin van Aalst, Venkata Satagopam, Oliver Ebenhöh, Reinhard Schneider, Christophe Trefois, and Wei Gu. Cobrexa.jl: constraint-based reconstruction and exascale analysis. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.04.447038">[DOI]</a>.</p></li>
<li id="id15383"><p>Xiang Gao, Chao Wu, Michael Cantrell, Mélissa Cano, Jianping Yu, and Wei Xiong. What we can learn from measuring metabolic fluxes in cyanobacteria. ():89–122, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527824908.ch4">[DOI]</a>.</p></li>
<li id="id15409"><p>Sankalpa Venkatraghavan, Sathvik Anantakrishnan, and Karthik Raman. Probing patterning in microbial consortia with piccaso: a cellular automaton for spatial organisation. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.02.14.431138">[DOI]</a>.</p></li>
<li id="id15410"><p>Juan C. Villada, Maria F. Duran, Chee Kent Lim, Lisa Y. Stein, and Patrick K. H. Lee. Integrative genome-scale metabolic modeling reveals versatile metabolic strategies for methane utilization in&lt;i&gt;methylomicrobium album&lt;/i&gt;bg8. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.21.436352">[DOI]</a>.</p></li>
<li id="id15411"><p>Johannes Hertel, Almut Heinken, and Ines Thiele. Causal inference on microbiome-metabolome relations via&lt;i&gt;in silico in vivo&lt;/i&gt;association pattern analyses. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.03.15.435397">[DOI]</a>.</p></li>
<li id="id15412"><p>Emil Ljungqvist and Martin Gustavsson. &lt;i&gt;in silico&lt;/i&gt;analysis of the fast-growing thermophile&lt;i&gt;geobacillus&lt;/i&gt;sp. lc300 using a novel genome-scale metabolic model. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.04.07.438430">[DOI]</a>.</p></li>
<li id="id15413"><p>Paula Jouhten, Dimitrios Konstantinidis, Filipa Pereira, Sergej Andrejev, Kristina Grkovska, Payam Ghiachi, Gemma Beltran, Eivind Almaas, Albert Mas, Jonas Warringer, Ramón González, Pilar Morales, and Kiran Raosaheb Patil. Predictive evolution of metabolic phenotypes using model-designed selection niches. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.05.14.443989">[DOI]</a>.</p></li>
<li id="id15414"><p>J. Saraiva, Alexander Bartholomäus, René Kallies, Marta Gomes, Marcos Bicalho, Carsten Vogt, Antonis Chatzinotas, Peter F. Stadler, Óscar Dias, and Ulisses Nunes da Rocha. Ortsuite – from genomes to prediction of microbial interactions within targeted ecosystem processes. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.04.447094">[DOI]</a>.</p></li>
<li id="id15415"><p>Marco Fondi, Stefano Gonzi, Mikołaj Dziurzyński, Paola Turano, Veronica Ghini, Marzia Calvanese, Andrea Colarusso, Concetta Lauro, Ermenegilda Parrilli, and Maria Luisa Tutino. Modelling hcdkl5 heterologous expression in bacteria. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.06.17.448472">[DOI]</a>.</p></li>
<li id="id15416"><p>St. Elmo Wilken, Victor Vera Frazão, Nima P. Saadat, and Oliver Ebenhöh. The view of microbes as energy converters illustrates the trade-off between growth rate and yield. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.04.16.440103">[DOI]</a>.</p></li>
<li id="id15417"><p>Carolin C. M. Schulte, Vinoy K. Ramachandran, Antonis Papachristodoulou, and Philip S. Poole. Genome-scale metabolic modelling of lifestyle changes in &lt;i&gt;rhizobium leguminosarum&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.28.454262">[DOI]</a>.</p></li>
<li id="id15418"><p>Antoine Régimbeau, Marko Budinich, Abdelhalim Larhlimi, Juan José Pierella Karlusich, Olivier Aumont, Laurent Mémery, Chris Bowler, and Damien Eveillard. Contribution of genome scale metabolic modeling to niche theory. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.07.21.453190">[DOI]</a>.</p></li>
<li id="id15419"><p>Kristina Grausa, Ivars Mozga, Kārlis Pleiko, and Agris Pentjuss. Integrative gene expression and metabolic analysis tool igemrna. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.08.02.454732">[DOI]</a>.</p></li>
<li id="id15420"><p>Iñigo Apaolaza, Edurne San José‐Eneriz, Luis V. Valcárcel, Xabier Agirre, Felipe Prósper, and Francisco J. Planes. A network-based approach to integrate nutrient environment in the prediction of synthetic lethality in cancer metabolism. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.09.01.458495">[DOI]</a>.</p></li>
<li id="id15421"><p>Lavanya Raajaraam and Karthik Raman. A computational framework to identify metabolic engineering strategies for the co-production of metabolites. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.09.18.460904">[DOI]</a>.</p></li>
<li id="id15422"><p>Iman Shahidi Pour Savizi, Nader Maghsoudi, Ehsan Motamedian, Nathan E. Lewis, and Seyed Abbas Shojaosadati. Amino acid feeding reduces ammonia production through rearrangement of metabolic fluxes in central carbon metabolism of cho cells. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-918138/v1">[DOI]</a>.</p></li>
<li id="id15423"><p>Philipp Wendering and Zoran Nikoloski. Genome-scale modeling specifies the metabolic capabilities of &lt;i&gt;rhizophagus irregularis&lt;/i&gt;. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2021.10.07.463607">[DOI]</a>.</p></li>
<li id="id15424"><p>Arsh Vyas. Metabolic modeling of cystic fibrosis airway microbiota from patient samples. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.7275/24145939.0">[DOI]</a>.</p></li>
<li id="id15425"><p>Prerna Bhalla, Swagatika Sahoo, Raghunathan Rengaswamy, Devarajan Karunagaran, and G. K. Suraishkumar. Constraint-based modeling to understand ros-mediated effects in cancer. ():1–21, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-15-4501-6_144-1">[DOI]</a>.</p></li>
<li id="id15431"><p>J. Saraiva, Alexander Bartholomäus, René Kallies, Marta Gomes, Marcos Bicalho, Jonas Coelho Kasmanas, Carsten Vogt, Antonis Chatzinotas, Peter F. Stadler, Óscar Dias, and Ulisses Nunes da Rocha. Ortsuite – from genomes to prediction of microbial interactions within targeted ecosystem processes. ():, 2021. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-52281/v2">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-5" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-5" name="0-5" role="tabpanel" tabindex="0"><div class="docutils container" id="id15606">
<ol class="arabic simple" start="1">
<li id="id15637"><p>Maria A. Wörheide, Jan Krumsiek, Gabi Kastenmüller, and Matthias Arnold. Multi-omics integration in biomedical research – a metabolomics-centric review. 1141():144–162, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.aca.2020.10.038">[DOI]</a>.</p></li>
<li id="id15675"><p>Maciek R. Antoniewicz. A guide to metabolic flux analysis in metabolic engineering: methods, tools and applications. 63():2–12, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.11.002">[DOI]</a>.</p></li>
<li id="id15728"><p>Svetlana Volkova, Marta R. A. Matos, Matthias Mattanovich, and Igor Marín de Mas. Metabolic modelling as a framework for metabolomics data integration and analysis. 10(8):303–303, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10080303">[DOI]</a>.</p></li>
<li id="id15733"><p>Shirin Fallahi, Hans J. Skaug, and Guttorm Alendal. A comparison of monte carlo sampling methods for metabolic network models. 15(7):e0235393–e0235393, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0235393">[DOI]</a>.</p></li>
<li id="id15859"><p>Zhuangrong Huang, Jianlin Xu, Andrew Yongky, Caitlin Morris, Ashli Polanco, Michael D. Reily, Michael Borys, Zheng Jian Li, and Seongkyu Yoon. Cho cell productivity improvement by genome-scale modeling and pathway analysis: application to feed supplements. 160():107638–107638, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2020.107638">[DOI]</a>.</p></li>
<li id="id15871"><p>Shreya Anand, Koel Mukherjee, and Padmini Padmanabhan. An insight to flux-balance analysis for biochemical networks. 36(1):32–55, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/02648725.2020.1847440">[DOI]</a>.</p></li>
<li id="id15913"><p>Carolina A. Contador, Siu-Kit Lo, Siu Hung Joshua Chan, and Hon‐Ming Lam. Metabolic analyses of nitrogen fixation in the soybean microsymbiont sinorhizobium fredii using constraint-based modeling. 5(1):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00516-19">[DOI]</a>.</p></li>
<li id="id15930"><p>William T. Scott, Eddy J. Smid, Richard A. Notebaart, and David E. Block. Curation and analysis of a saccharomyces cerevisiae genome-scale metabolic model for predicting production of sensory impact molecules under enological conditions. 8(9):1195–1195, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/pr8091195">[DOI]</a>.</p></li>
<li id="id15944"><p>Naoya Nochino, Yoshihiro Toya, and Hiroshi Shimizu. Transcription factor arca is a flux sensor for the oxygen consumption rate in &lt;i&gt;escherichia coli&lt;/i&gt;. 15(6):e1900353–e1900353, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201900353">[DOI]</a>.</p></li>
<li id="id15994"><p>Zhuangrong Huang and Seongkyu Yoon. Identifying metabolic features and engineering targets for productivity improvement in cho cells by integrated transcriptomics and genome-scale metabolic model. 159():107624–107624, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2020.107624">[DOI]</a>.</p></li>
<li id="id16010"><p>Zhuangrong Huang and Seongkyu Yoon. Integration of time-series transcriptomic data with genome-scale cho metabolic models for mab engineering. 8(3):331–331, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/pr8030331">[DOI]</a>.</p></li>
<li id="id16011"><p>Hamideh Fouladiha, Sayed‐Amir Marashi, Shangzhong Li, Zerong Li, Helen O. Masson, Behrouz Vaziri, and Nathan E. Lewis. Systematically gap-filling the genome-scale metabolic model of cho cells. 43(1):73–87, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-020-03021-w">[DOI]</a>.</p></li>
<li id="id16028"><p>Yu Xu, Roman Holič, and Qiang Hua. Comparison and analysis of published genome-scale metabolic models of yarrowia lipolytica. 25(1):53–61, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-019-0208-1">[DOI]</a>.</p></li>
<li id="id16029"><p>Mikhail A. Kulyashov, S. E. Peltek, and Ilya R. Akberdin. A genome-scale metabolic model of 2,3-butanediol production by thermophilic bacteria geobacillus icigianus. 8(7):1002–1002, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms8071002">[DOI]</a>.</p></li>
<li id="id16035"><p>Ahmad Ahmad, Ruchi Pathania, and Shireesh Srivastava. Biochemical characteristics and a genome-scale metabolic model of an indian euryhaline cyanobacterium with high polyglucan content. 10(5):177–177, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10050177">[DOI]</a>.</p></li>
<li id="id16043"><p>Hüseyin Taş, Adam Amara, Miguel Cueva, Nadine Bongaerts, Alicia Calvo‐Villamañán, Samir Hamadache, and Konstantinos Vavitsas. Are synthetic biology standards applicable in everyday research practice? 13(5):1304–1308, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.13612">[DOI]</a>.</p></li>
<li id="id16049"><p>Masakazu Toyoshima, Chiaki Yamamoto, Yoshifumi Ueno, Yoshihiro Toya, Seiji Akimoto, and Hiroshi Shimizu. Role of type i nadh dehydrogenase in synechocystis sp. pcc 6803 under phycobilisome excited red light. 304():110798–110798, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.plantsci.2020.110798">[DOI]</a>.</p></li>
<li id="id16100"><p>Ushashi Banerjee, Santhosh Sankar, Amit Singh, and Nagasuma Chandra. A multi-pronged computational pipeline for prioritizing drug target strategies for latent tuberculosis. 8():593497–593497, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fchem.2020.593497">[DOI]</a>.</p></li>
<li id="id16115"><p>Gong‐Hua Li, Shao‐Xing Dai, Feifei Han, Wenxing Li, Jing‐Fei Huang, and Wenzhong Xiao. Fastmm: an efficient toolbox for personalized constraint-based metabolic modeling. 21(1):67–67, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-020-3410-4">[DOI]</a>.</p></li>
<li id="id16122"><p>Veronica L. Porubsky, Lucian Smith, and Herbert M. Sauro. Publishing reproducible dynamic kinetic models. 22(3):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbaa152">[DOI]</a>.</p></li>
<li id="id16150"><p>Hamideh Fouladiha, Sayed-Amir Marashi, Shangzhong Li, Zerong Li, Helen O Masson, Behrouz Vaziri, Nathan E. Lewis, Hamideh Fouladiha, Sayed-Amir Marashi, Shangzhong Li, Zerong Li, Helen O Masson, Behrouz Vaziri, and Nathan E. Lewis. Systematically gap-filling the genome-scale metabolic model of cho cells. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.01.27.921296">[DOI]</a>.</p></li>
<li id="id16194"><p>Eliana B. Souto, Elena Sánchez‐López, Joana R. Campos, Raquel da Ana, Marta Espina, María L. García, Patrícia Severino, Fernando Batain, Thais Alves, Kessi Crescencio, Selma B. Souto, and Marco V. Chaud. Retinal drug delivery: rethinking outcomes for the efficient replication of retinal behavior. 10(12):4258–4258, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/app10124258">[DOI]</a>.</p></li>
<li id="id16195"><p>Gargi Bhattacharjee, Nisarg Gohil, and Vijai Singh. An introduction to design of microbial strain using synthetic biology toolboxes for production of biomolecules. ():1–10, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-15-2604-6_1">[DOI]</a>.</p></li>
<li id="id16208"><p>Alan R. Pacheco and Daniel Segrè. An evolutionary algorithm for designing microbial communities via environmental modification. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.11.25.398644">[DOI]</a>.</p></li>
<li id="id16213"><p>Cristina Herencias, M. Auxiliadora Prieto, and Juan Nogales. Providing new insights on the byphasic lifestyle of the predatory bacterium&lt;i&gt;bdellovibrio bacteriovorus&lt;/i&gt;through genome-scale metabolic modeling. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.01.13.904078">[DOI]</a>.</p></li>
<li id="id16295"><p>Nikhil Kirtipal and Asheesh Shanker. Integrating omics and microbial biotechnology for the production of biofuel. ():221–239, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-15-3761-5_9">[DOI]</a>.</p></li>
<li id="id16296"><p>Zsolt Bodor, ANDREA FAZAKAS, Katalin Bodor, Erika R. Bérczi‐Kovács, Ildikó Miklóssy, and Beáta Albert. Using genome-scale model to predict the metabolic engineering impact on escherichia coli metabolism during succinic acid production optimization. 25(3):1666–1676, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.25083/rbl/25.3/1666.1676">[DOI]</a>.</p></li>
<li id="id16307"><p>Feng‐Sheng Wang and Wu‐Hsiung Wu. Computer‐aided design for genetic modulation to improve biofuel production. ():173–189, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781119582694.ch7">[DOI]</a>.</p></li>
<li id="id16311"><p>Caroline S.M. Nakama. Desenvolvimentos de ferramentas matemáticas para modelagem de sistemas biológicos baseados em fluxos metabólicos e estimação de parâmetros de problemas mal condicionados. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.11606/t.3.2020.tde-13042021-112113">[DOI]</a>.</p></li>
<li id="id16347"><p>Rupa Bhowmick, Piyali Ganguli, and Ram Rup Sarkar. T-cell activation and differentiation: role of signaling and metabolic cross-talk. ():153–182, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-15-3350-1_6">[DOI]</a>.</p></li>
<li id="id16366"><p>Diego Salazar Barreto. Predicción de biomarcadores en mucopolisacaridosis empleando una red metabólica humana. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.11144/javeriana.10554.15469">[DOI]</a>.</p></li>
<li id="id16367"><p>William Parra Meneses. Análisis comparativo de la topología de redes metabólicas en una muestra metagenómica de aguas termales en el parque natural de los nevados en colombia. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.11144/javeriana.10554.1531">[DOI]</a>.</p></li>
<li id="id16466"><p>Yoseb Song, Jin Soo Lee, Jongoh Shin, Gyu Min Lee, Sangrak Jin, Seulgi Kang, Jung-Kul Lee, Dong Rip Kim, Eun Yeol Lee, Sun Chang Kim, Suhyung Cho, Donghyuk Kim, and Byung‐Kwan Cho. Functional cooperation of the glycine synthase-reductase and wood–ljungdahl pathways for autotrophic growth of &lt;i&gt;clostridium drakei&lt;/i&gt;. 117(13):7516–7523, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1912289117">[DOI]</a>.</p></li>
<li id="id16481"><p>Jahir M. Gutierrez, Amir Feizi, Shangzhong Li, Thomas Beuchert Kallehauge, Hooman Hefzi, Lise Marie Grav, Daniel Ley, Deniz Baycın Hızal, Michael J. Betenbaugh, Björn G. Voldborg, Helene Faustrup Kildegaard, Gyun Min Lee, Bernhard Ø. Palsson, Jens Nielsen, and Nathan E. Lewis. Genome-scale reconstructions of the mammalian secretory pathway predict metabolic costs and limitations of protein secretion. 11(1):68–68, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-019-13867-y">[DOI]</a>.</p></li>
<li id="id16496"><p>Isabel Bator, Andreas Wittgens, Frank Rosenau, Till Tiso, and Lars M. Blank. Comparison of three xylose pathways in pseudomonas putida kt2440 for the synthesis of valuable products. 7():480–480, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2019.00480">[DOI]</a>.</p></li>
<li id="id16497"><p>Aarti Krishnan, Joachim Kloehn, Matteo Lunghi, Anush Chiappino-Pepe, Benjamin S. Waldman, Damien Nicolas, Emmanuel Varesio, Adrian B. Hehl, Sebastian Lourido, Vassily Hatzimanikatis, and Dominique Soldati‐Favre. Functional and computational genomics reveal unprecedented flexibility in stage-specific toxoplasma metabolism. 27(2):290–306.e11, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.chom.2020.01.002">[DOI]</a>.</p></li>
<li id="id16506"><p>Pavlos Stephanos Bekiaris and Steffen Klamt. Automatic construction of metabolic models with enzyme constraints. 21(1):19–19, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-3329-9">[DOI]</a>.</p></li>
<li id="id16519"><p>Pierre Salvy and Vassily Hatzimanikatis. The etfl formulation allows multi-omics integration in thermodynamics-compliant metabolism and expression models. 11(1):30–30, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-019-13818-7">[DOI]</a>.</p></li>
<li id="id16540"><p>Cristal Zúñiga, Tingting Li, Michael T. Guarnieri, Jackson Jenkins, Chien‐Ting Li, Kerem Bingol, Young‐Mo Kim, Michael J. Betenbaugh, and Karsten Zengler. Synthetic microbial communities of heterotrophs and phototrophs facilitate sustainable growth. 11(1):3803–3803, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-17612-8">[DOI]</a>.</p></li>
<li id="id16545"><p>George C. diCenzo, Michelangelo Tesi, Thomas Pfau, Alessio Mengoni, and Marco Fondi. Genome-scale metabolic reconstruction of the symbiosis between a leguminous plant and a nitrogen-fixing bacterium. 11(1):2574–2574, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-16484-2">[DOI]</a>.</p></li>
<li id="id16550"><p>Arnaud Belcour, Clémence Frioux, Méziane Aïte, Anthony Bretaudeau, Falk Hildebrand, and Anne Siegel. Metage2metabo, microbiota-scale metabolic complementarity for the identification of key species. 9():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.61968">[DOI]</a>.</p></li>
<li id="id16595"><p>Dinka Mandaković, Ángela Cintolesi, Jonathan Maldonado, Sebastián N. Mendoza, Méziane Aïte, Alexis Gaete, Francisco Saitua, Miguel L. Allende, Verónica Cambiazo, Anne Siegel, Alejandro Maass, Maurício González, and Mauricio Latorre. Genome-scale metabolic models of microbacterium species isolated from a high altitude desert environment. 10(1):5560–5560, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-62130-8">[DOI]</a>.</p></li>
<li id="id16599"><p>Kristopher D. Rawls, Bonnie V. Dougherty, Kalyan C. Vinnakota, Venkat R. Pannala, Anders Wallqvist, Glynis L. Kolling, and Jason A. Papin. Predicting changes in renal metabolism after compound exposure with a genome-scale metabolic model. 412():115390–115390, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.taap.2020.115390">[DOI]</a>.</p></li>
<li id="id16602"><p>José Manuel Borrero‐de Acuña, Izabook Gutiérrez‐Urrutia, Cristian Hidalgo-Dumont, Carla Aravena-Carrasco, Matias Orellana‐Saez, Nestor Palominos-Gonzalez, Jozef B. J. H. van Duuren, Viktoria Wagner, Lars Gläser, Judith Becker, Michael Kohlstedt, Flavia C. Zacconi, Christoph Wittmann, and Ignacio Poblete‐Castro. Channelling carbon flux through the &lt;i&gt;meta&lt;/i&gt; ‐cleavage route for improved poly(3‐hydroxyalkanoate) production from benzoate and lignin‐based aromatics in &lt;i&gt;pseudomonas putida&lt;/i&gt; h. 14(6):2385–2402, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.13705">[DOI]</a>.</p></li>
<li id="id16696"><p>Shaozhen Ding, Yu Tian, Pengli Cai, Dachuan Zhang, Xingxiang Cheng, Dandan Sun, Le Yuan, Junni Chen, Weizhong Tu, Dong‐Qing Wei, and Qian‐Nan Hu. Novopathfinder: a webserver of designing novel-pathway with integrating gem-model. 48(W1):W477–W487, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkaa230">[DOI]</a>.</p></li>
<li id="id16701"><p>Hamideh Fouladiha, Sayed‐Amir Marashi, Fatemeh Torkashvand, Fereidoun Mahboudi, Nathan E. Lewis, and Behrouz Vaziri. A metabolic network-based approach for developing feeding strategies for cho cells to increase monoclonal antibody production. 43(8):1381–1389, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-020-02332-6">[DOI]</a>.</p></li>
<li id="id16718"><p>Hao Tong, Anika Küken, and Zoran Nikoloski. Integrating molecular markers into metabolic models improves genomic selection for arabidopsis growth. 11(1):2410–2410, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-16279-5">[DOI]</a>.</p></li>
<li id="id16772"><p>Francesca Antonaros, Veronica Ghini, Francesca Pulina, Giuseppe Ramacieri, Elena Cicchini, Elisa Mannini, Anna Martelli, Agnese Feliciello, Silvia Lanfranchi, Sara Onnivello, Renzo Vianello, Chiara Locatelli, Guido Cocchi, Maria Chiara Pelleri, Lorenza Vitale, Pierluigi Strippoli, Claudio Luchinat, Paola Turano, Allison Piovesan, and Maria Caracausi. Plasma metabolome and cognitive skills in down syndrome. 10(1):10491–10491, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-67195-z">[DOI]</a>.</p></li>
<li id="id16780"><p>Till Tiso, Tanja Narančić, Ren Wei, Éric Pollet, Niall Beagan, Katja Schröder, Annett Honak, Mengying Jiang, Shane T. Kenny, Nick Wierckx, Rémi Perrin, Luc Avérous, Wolfgang Zimmermann, Kevin E. O’Connor, and Lars M. Blank. Bio-upcycling of polyethylene terephthalate. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.03.16.993592">[DOI]</a>.</p></li>
<li id="id16781"><p>Cristal Zúñiga, Beth B. Peacock, Bo Liang, Greg McCollum, Sonia Irigoyen, Diego Tec-Campos, Clarisse Marotz, Nien-Chen Weng, Alejandro Zepeda, Georgios Vidalakis, Kranthi K. Mandadi, James Borneman, and Karsten Zengler. Linking metabolic phenotypes to pathogenic traits among “candidatus liberibacter asiaticus” and its hosts. 6(1):24–24, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-020-00142-w">[DOI]</a>.</p></li>
<li id="id16800"><p>Birthe Halmschlag, Kyra Hoffmann, René Hanke, Sastia Prama Putri, Eiichiro Fukusaki, Jochen Büchs, and Lars M. Blank. Comparison of isomerase and weimberg pathway for γ-pga production from xylose by engineered bacillus subtilis. 7():476–476, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2019.00476">[DOI]</a>.</p></li>
<li id="id16814"><p>Hardy Rolletschek, Jörg Schwender, Christina König, Kent D. Chapman, Trevor B. Romsdahl, Christin Lorenz, Hans‐Peter Braun, Peter Denolf, Katrien Van Audenhove, E. Muñz, Nicolas Heinzel, Stefan Ortleb, Twan Rutten, S. McCorkle, Taras Borysyuk, André Guendel, Hai Shi, Michiel Vander Auwermeulen, Stéphane Bourot, and Ljudmilla Borisjuk. Cellular plasticity in response to suppression of storage proteins in the &lt;i&gt;brassica napus&lt;/i&gt; embryo. 32(7):2383–2401, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1105/tpc.19.00879">[DOI]</a>.</p></li>
<li id="id16815"><p>Emrah Özcan, Merve Seven, Burcu Şirin, Tunahan Çakır, Emrah Nıkerel, Bas Teusink, and Ebru Toksoy Öner. Dynamic co‐culture metabolic models reveal the fermentation dynamics, metabolic capacities and interplays of cheese starter cultures. 118(1):223–237, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27565">[DOI]</a>.</p></li>
<li id="id16821"><p>Juan D. Tibocha‐Bonilla, Manish Kumar, Anne Richelle, Rubén Darío Godoy‐Silva, Karsten Zengler, and Cristal Zúñiga. Dynamic resource allocation drives growth under nitrogen starvation in eukaryotes. 6(1):14–14, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-020-0135-y">[DOI]</a>.</p></li>
<li id="id16822"><p>Emadoddin Karimian and Ehsan Motamedian. Acbm: an integrated agent and constraint based modeling framework for simulation of microbial communities. 10(1):8695–8695, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-65659-w">[DOI]</a>.</p></li>
<li id="id16823"><p>Carolina Shene, Paris Paredes, Liset Flores, Allison Leyton, Juan A. Asenjo, and Yusuf Chisti. Dynamic flux balance analysis of biomass and lipid production by antarctic thraustochytrid &lt;i&gt;oblongichytrium&lt;/i&gt; sp. rt2316‐13. 117(10):3006–3017, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27463">[DOI]</a>.</p></li>
<li id="id16840"><p>Yuki Kuriya and Michihiro Araki. Dynamic flux balance analysis to evaluate the strain production performance on shikimic acid production in escherichia coli. 10(5):198–198, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10050198">[DOI]</a>.</p></li>
<li id="id16869"><p>Lokanand Koduru, Hyang Kim, Meiyappan Lakshmanan, Bijayalaxmi Mohanty, Yi Qing Lee, Choong Hwan Lee, and Dong‐Yup Lee. Genome‐scale metabolic reconstruction and in silico analysis of the rice leaf blight pathogen, &lt;i&gt;xanthomonas oryzae&lt;/i&gt;. 21(4):527–540, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/mpp.12914">[DOI]</a>.</p></li>
<li id="id16870"><p>Gunvor Bjerkelund Røkke, Martin F. Hohmann‐Marriott, and Eivind Almaas. An adjustable algal chloroplast plug-and-play model for genome-scale metabolic models. 15(2):e0229408–e0229408, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0229408">[DOI]</a>.</p></li>
<li id="id16873"><p>Anukriti Verma, Shivani Sharda, Bhawna Rathi, Pallavi Somvanshi, and Bimlesh Dhar Pandey. Elucidating potential molecular signatures through host-microbe interactions for reactive arthritis and inflammatory bowel disease using combinatorial approach. 10(1):15131–15131, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-71674-8">[DOI]</a>.</p></li>
<li id="id16884"><p>Rashi Verma, Dibyabhaba Pradhan, Mohsin Maseet, Harpreet Singh, Arun K. Jain, and Luqman Ahmad Khan. Genome-wide screening and in silico gene knockout to predict potential candidates for drug designing against candida albicans. 80():104196–104196, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meegid.2020.104196">[DOI]</a>.</p></li>
<li id="id16889"><p>Kyle A. Stone, Matthew Hilliard, Kiumars Badr, Alisabeth Bradford, Q. Peter He, and Jin Wang. Comparative study of oxygen-limited and methane-limited growth phenotypes of methylomicrobium buryatense 5gb1. 161():107707–107707, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2020.107707">[DOI]</a>.</p></li>
<li id="id16924"><p>Ali Malek Shahkouhi and Ehsan Motamedian. Reconstruction of a regulated two-cell metabolic model to study biohydrogen production in a diazotrophic cyanobacterium anabaena variabilis atcc 29413. 15(1):e0227977–e0227977, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0227977">[DOI]</a>.</p></li>
<li id="id16925"><p>Na-Rae Lee, Choong Hwan Lee, Dong‐Yup Lee, and Jin‐Byung Park. Genome-scale metabolic network reconstruction and in silico analysis of hexanoic acid producing megasphaera elsdenii. 8(4):539–539, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms8040539">[DOI]</a>.</p></li>
<li id="id16931"><p>David M. Curran, Alexandra Grote, Nirvana Nursimulu, Adam Geber, Denis Voronin, Drew R. Jones, Elodie Ghedin, and John Parkinson. Modeling the metabolic interplay between a parasitic worm and its bacterial endosymbiont allows the identification of novel drug targets. 9():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.51850">[DOI]</a>.</p></li>
<li id="id16946"><p>Hong Zeng and Aidong Yang. Bridging substrate intake kinetics and bacterial growth phenotypes with flux balance analysis incorporating proteome allocation. 10(1):4283–4283, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-61174-0">[DOI]</a>.</p></li>
<li id="id16947"><p>Hoda Nouri, Hamideh Fouladiha, Hamid Moghimi, and Sayed‐Amir Marashi. A reconciliation of genome-scale metabolic network model of zymomonas mobilis zm4. 10(1):7782–7782, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-64721-x">[DOI]</a>.</p></li>
<li id="id16989"><p>Jan Géryk, Daniel Krsička, Markéta Vlčková, Markéta Havlovičová, Milan Maçek, and Radka Kremlíková Pourová. The key role of purine metabolism in the folate-dependent phenotype of autism spectrum disorders: an in silico analysis. 10(5):184–184, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10050184">[DOI]</a>.</p></li>
<li id="id17004"><p>Cristina Herencias, Sergio Salgado, M. Auxiliadora Prieto, and Juan Nogales. Providing new insights on the biphasic lifestyle of the predatory bacterium bdellovibrio bacteriovorus through genome-scale metabolic modeling. 16(9):e1007646–e1007646, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007646">[DOI]</a>.</p></li>
<li id="id17005"><p>Eileen Marie Hanna, Xiaokang Zhang, Marta Eide, Shirin Fallahi, Tomasz Furmanek, Fekadu Yadetie, Daniel C. Zielinski, Anders Goksøyr, and Inge Jonassen. Recodliver0.9: overcoming challenges in genome-scale metabolic reconstruction of a non-model species. 7():591406–591406, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmolb.2020.591406">[DOI]</a>.</p></li>
<li id="id17010"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Elena López‐Camacho, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Rocío López‐Vacas, Jorge M. Arevalillo, Mariana Díaz‐Almirón, Hilario Navarro, Paloma Maı́n, Enrique Espinosa, Pilar Zamora, and Juan Ángel Fresno Vara. Computational models applied to metabolomics data hints at the relevance of glutamine metabolism in breast cancer. 20(1):307–307, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12885-020-06764-x">[DOI]</a>.</p></li>
<li id="id17020"><p>Mohammad Rezazadeh, Valiollah Babaeipour, and Ehsan Motamedian. Reconstruction, verification and in-silico analysis of a genome-scale metabolic model of bacterial cellulose producing komagataeibacter xylinus. 43(6):1017–1026, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-020-02299-4">[DOI]</a>.</p></li>
<li id="id17027"><p>Tobias Ploch, Eric von Lieres, Wolfgang Wiechert, Alexander Mitsos, and Ralf Hannemann. Simulation of differential-algebraic equation systems with optimization criteria embedded in modelica. 140():106920–106920, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compchemeng.2020.106920">[DOI]</a>.</p></li>
<li id="id17049"><p>William Hohenschuh, Ronald E. Hector, Frank Chaplen, and Ganti S. Murthy. Using high-throughput data and dynamic flux balance modeling techniques to identify points of constraint in xylose utilization in saccharomyces cerevisiae. 1(1):58–75, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s43393-020-00003-x">[DOI]</a>.</p></li>
<li id="id17069"><p>Marisa Nicolás, Maiana de Oliveira Cerqueira e Costa, Pablo Ivan Pereira Ramos, Marcelo Trindade dos Santos, Ernesto Pérez‐Rueda, Marcelo A. Martí, Darío Fernández Do Porto, and Adrián G. Turjanski. Integrating omics data to prioritize target genes in pathogenic bacteria. ():217–276, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-51862-2_10">[DOI]</a>.</p></li>
<li id="id17135"><p>Sandra M. Correa, Saleh Alseekh, Lucía Atehortúa, Yariv Brotman, Rigoberto Ríos‐Estepa, Alisdair R. Fernie, and Zoran Nikoloski. Model‐assisted identification of metabolic engineering strategies for &lt;i&gt;jatropha curcas&lt;/i&gt; lipid pathways. 104(1):76–95, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/tpj.14906">[DOI]</a>.</p></li>
<li id="id17147"><p>Rahul Kumar and Petri‐Jaan Lahtvee. Proteome overabundance enables respiration but limitation onsets carbon overflow. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.02.20.957662">[DOI]</a>.</p></li>
<li id="id17162"><p>Gayathri Sambamoorthy and Karthik Raman. &lt;scp&gt;minreact&lt;/scp&gt; : a systematic approach for identifying minimal metabolic networks. 36(15):4309–4315, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaa497">[DOI]</a>.</p></li>
<li id="id17163"><p>Fabián Moreno, José Utrilla, Francisco Bolívar, Juan Nogales, and Adelfo Escalante. Metabolic reconstruction of pseudomonas chlororaphis atcc 9446 to understand its metabolic potential as a phenazine-1-carboxamide-producing strain. 104(23):10119–10132, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-020-10913-4">[DOI]</a>.</p></li>
<li id="id17181"><p>Clémence Frioux, Simon M. Dittami, and Anne Siegel. Using automated reasoning to explore the metabolism of unconventional organisms: a first step to explore host–microbial interactions. 48(3):901–913, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20190667">[DOI]</a>.</p></li>
<li id="id17182"><p>Song‐Min Schinn, Carly Morrison, Wei Wei, Lin Zhang, and Nathan E. Lewis. Systematic evaluation of parameters for genome-scale metabolic models of cultured mammalian cells. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.24.169938">[DOI]</a>.</p></li>
<li id="id17183"><p>Yazdan Asgari and Pegah Khosravi. Flux variability analysis reveals a tragedy of commons in cancer cells. 2(12):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s42452-020-03762-3">[DOI]</a>.</p></li>
<li id="id17189"><p>Inmaculada García‐Romero, Juan Nogales, Eduardo Dı́az, Eduardo Santero, and Belén Floriano. Understanding the metabolism of the tetralin degrader sphingopyxis granuli strain tfa through genome-scale metabolic modelling. 10(1):8651–8651, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-65258-9">[DOI]</a>.</p></li>
<li id="id17202"><p>Narges Pedram, Hamid Rashedi, and Ehsan Motamedian. A systematic strategy using a reconstructed genome-scale metabolic network for pathogen&lt;i&gt;streptococcus&lt;/i&gt;&lt;i&gt;pneumoniae&lt;/i&gt;d39 to find novel potential drug targets. 78(6):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femspd/ftaa051">[DOI]</a>.</p></li>
<li id="id17260"><p>Lucía Trilla‐Fuertes, Ismael Ghanem, Angelo Gámez‐Pozo, Joan Maurel, Laura G-Pastrián, Marta Mendiola, Cristina Peña, Rocío López‐Vacas, Guillermo Prado-Vázquez, Elena López‐Camacho, Andrea Zapater‐Moros, Victoria Heredia-Soto, Míriam Cuatrecasas, Pilar García‐Alfonso, Jaume Capdevila, Carles Conill, Rocio García‐Carbonero, Ricardo Ramos, Claudia Fortes, Carlos Lloréns, Paolo Nanni, Juan Ángel Fresno Vara, and Jaime Feliú. Genetic profile and functional proteomics of anal squamous cell carcinoma: proposal for a molecular classification. 19(4):690–700, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1074/mcp.ra120.001954">[DOI]</a>.</p></li>
<li id="id17273"><p>Jyotshana Gautam and Zhaohui Xu. Construction and validation of a genome-scale metabolic network of thermotoga sp. strain rq7. 193(3):896–911, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12010-020-03470-z">[DOI]</a>.</p></li>
<li id="id17301"><p>Eileen Marie Hanna, Xiaokang Zhang, Marta Eide, Shirin Fallahi, Tomasz Furmanek, Fekadu Yadetie, Daniel C. Zielinski, Anders Goksøyr, and Inge Jonassen. Recodliver0.9: overcoming challenges in genome-scale metabolic reconstruction of a non-model species. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.23.162792">[DOI]</a>.</p></li>
<li id="id17302"><p>Song Min Schinn, Carly Morrison, Wei Wei, Lin Zhang, and Nathan E. Lewis. A genome-scale metabolic network model and machine learning predict amino acid concentrations in chinese hamster ovary cell cultures. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.22541/au.159986472.28149413">[DOI]</a>.</p></li>
<li id="id17303"><p>George C. diCenzo, Marco Galardini, and Marco Fondi. Tn-core: functionally interpreting transposon-sequencing data with metabolic network analysis. 2189():199–215, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-0822-7_15">[DOI]</a>.</p></li>
<li id="id17304"><p>Karel Olavarría, Caspar Quakkelaar, Joachim van Renselaar, Dennis Langerak, Mark C.M. van Loosdrecht, and A. Wahl. Nadh-driven poly-3-hydroxybutyrate accumulation in escherichia coli: data from enzymatic assays and oxygen-limited continuous cultures. 33():106588–106588, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.dib.2020.106588">[DOI]</a>.</p></li>
<li id="id17313"><p>Piao Zou, Yunze Zhang, Jean Bosco Nshimiyimana, Qianwen Cao, Yang Yang, Hui Geng, and Xiong Li. Reconstruction of a context-specific model based on genome-scale metabolic simulation for identification of prochloraz resistance mechanisms in &lt;i&gt;penicillium digitatum&lt;/i&gt;. 27(6):776–785, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/mdr.2020.0018">[DOI]</a>.</p></li>
<li id="id17320"><p>Ying Li, He Xian, Ya Xu, Yuan Zhu, Zhijie Sun, Qian Wang, and Qingsheng Qi. Fine and dynamic tuning the glycolytic flux ratio of an artificial carbon saving pathway for high yield of mevalonate in escherichia coli. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-46741/v1">[DOI]</a>.</p></li>
<li id="id17356"><p>Jake P. N. Hattwell, Janna Hastings, Olivia Casanueva, Horst Joachim Schirra, and Michael Witting. Using genome-scale metabolic networks for analysis, visualization, and integration of targeted metabolomics data. 2104():361–386, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-0239-3_18">[DOI]</a>.</p></li>
<li id="id17382"><p>Song-Min Schinn, Carly Morrison, WEI WEI, Lin Zhang, Nathan E. Lewis, Song-Min Schinn, Carly Morrison, WEI WEI, Lin Zhang, and Nathan E. Lewis. A genome-scale metabolic network model and machine learning predict amino acid concentrations in chinese hamster ovary cell cultures. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.02.279687">[DOI]</a>.</p></li>
<li id="id17409"><p>Gayathri Sambamoorthy and Karthik Raman. Minreact: an efficient algorithm for identifying minimal metabolic networks. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.01.06.896084">[DOI]</a>.</p></li>
<li id="id17414"><p>Nhung Pham, Maarten Reijnders, María Suárez‐Diez, Bart Nijsse, Jan Springer, Gerrit Eggink, and Peter J. Schaap. Genome-scale metabolic modelling underscores the potential of cutaneotrichosporon oleaginosus atcc 20509 as a cell factory for biofuel production. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-26414/v2">[DOI]</a>.</p></li>
<li id="id17449"><p>Nathan Braniff. Optimal experimental design applied to models of microbial gene regulation. ():, 2020. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id17535"><p>Yuzhong Liu, Fei Gan, Pablo Cruz‐Morales, and Jay D. Keasling. Engineering natural product biosynthetic pathways to produce commodity and specialty chemicals. ():352–376, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-409547-2.14825-5">[DOI]</a>.</p></li>
<li id="id17536"><p>Claudio Tomi-Andrino, Rupert Norman, Thomas Millat, Philippe Soucaille, Klaus Winzer, David A. Barrett, John R. King, and Dong‐Hyun Kim. Assessing the impact of physicochemical parameters in the predictive capabilities of thermodynamics-based stoichiometric approaches under mesophilic and thermophilic conditions. ():, 2020. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id17538"><p>Lucía Trilla‐Fuertes, Ricardo Ramos, Natalia Blanca‐López, Elena López‐Camacho, Laura Martín‐Pedraza, Pablo Ryan, Mariana Díaz‐Almirón, Carlos Lloréns, Toni Gabaldón, Andrés Moyá, Juan Ángel Fresno Vara, and Angelo Gámez‐Pozo. Functional characterization of sars-cov-2 infection suggests a complex inflammatory response and metabolic alterations. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.22.164384">[DOI]</a>.</p></li>
<li id="id17539"><p>Vladimir Porokhin, Sara A. Amin, Trevor B. Nicks, Venkatesh Endalur Gopinarayanan, Nikhil U. Nair, and Soha Hassoun. Analysis of metabolic network disruption in engineered microbial hosts due to enzyme promiscuity. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.02.279539">[DOI]</a>.</p></li>
<li id="id17540"><p>Sander Y.A. Rodenburg. Uncovering oomycete metabolism using systems biology. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/528798">[DOI]</a>.</p></li>
<li id="id17542"><p>Claudio Tomi-Andrino, Rupert Norman, Thomas Millat, Philippe Soucaille, Klaus Winzer, David A. Barrett, John R. King, and Dong‐Hyun Kim. Physicochemical and metabolic constraints for thermodynamics-based stoichiometric modelling under mesophilic growth conditions. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.02.03.932855">[DOI]</a>.</p></li>
<li id="id17551"><p>Kiyan Shabestary. Improving cyanobacteria productivity: from theory to assay. ():, 2020. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id17562"><p>Nhung Pham, Maarten Reijnders, María Suárez‐Diez, Bart Nijsse, Jan Springer, Gerrit Eggink, and Peter J. Schaap. Genome-scale metabolic modelling underscores the potential of cutaneotrichosporon oleaginosus atcc 20509 as a cell factory for biofuel production. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-26414/v1">[DOI]</a>.</p></li>
<li id="id17574"><p>Maurice HT Ling. 9(4):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.15406/mojpb.9.4">[DOI]</a>.</p></li>
<li id="id17576"><p>Nhung Pham, Maarten Reijnders, María Suárez‐Diez, Bart Nijsse, Jan Springer, Gerrit Eggink, and Peter J. Schaap. Genome-scale metabolic modelling underscores the potential of cutaneotrichosporon oleaginosus atcc 20509 as a cell factory for biofuel production. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-26414/v3">[DOI]</a>.</p></li>
<li id="id17578"><p>Cynthia Alexandra Martín Jiménez. Reconstrucción a escala genómica de una red metabólica de astrocito humano : aplicación a isquemia. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.11144/javeriana.10554.20625">[DOI]</a>.</p></li>
<li id="id17617"><p>Christian Lieven, Moritz Emanuel Beber, Brett G. Olivier, Frank Bergmann, Meriç Ataman, Parizad Babaei, Jennifer Bartell, Lars M. Blank, Siddharth Chauhan, Kevin Correia, Christian Diener, Andreas Dräger, Birgitta E. Ebert, Janaka N. Edirisinghe, José P. Faria, Adam M. Feist, Georgios Fengos, Ronan M. T. Fleming, Beatriz García-Jiménez, Vassily Hatzimanikatis, Wout van Helvoirt, Christopher S. Henry, Henning Hermjakob, Markus J. Herrgård, Ali Kaafarani, Hyun Uk Kim, Zachary A. King, Steffen Klamt, Edda Klipp, Jasper J. Koehorst, Matthias König, Meiyappan Lakshmanan, Dong‐Yup Lee, Sang Yup Lee, Sunjae Lee, Nathan E. Lewis, Filipe Liu, Hongwu Ma, Daniel Machado, Radhakrishnan Mahadevan, Paulo Maia, Adil Mardinoğlu, Gregory L. Medlock, Jonathan M. Monk, Jens Nielsen, Lars K. Nielsen, Juan Nogales, Intawat Nookaew, Bernhard Ø. Palsson, Jason A. Papin, Kiran Raosaheb Patil, Mark G. Poolman, Nathan D. Price, Osbaldo Resendis‐Antonio, Anne Richelle, Isabel Rocha, Benjamín J. Sánchez, Peter J. Schaap, Rahuman S. Malik‐Sheriff, Saeed Shoaie, Nikolaus Sonnenschein, Bas Teusink, Paulo Vilaça, Jon Olav Vik, Judith A. H. Wodke, Joana C. Xavier, Qianqian Yuan, Maksim Zakhartsev, and Cheng Zhang. Memote for standardized genome-scale metabolic model testing. 38(3):272–276, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41587-020-0446-y">[DOI]</a>.</p></li>
<li id="id17619"><p>Jonathan L. Robinson, Pınar Kocabaş, Hao Wang, Pierre-Etienne Cholley, Daniel Cook, Avlant Nilsson, Mihail Anton, Raphaël Ferreira, Iván Domenzain, Virinchi Billa, Angelo Limeta, Karl Alex Hedin, Johan Gustafsson, Eduard J. Kerkhoven, Thomas Svensson, Bernhard Ø. Palsson, Adil Mardinoğlu, Lena Hansson, Mathias Uhlén, and Jens Nielsen. An atlas of human metabolism. 13(624):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/scisignal.aaz1482">[DOI]</a>.</p></li>
<li id="id17620"><p>M. Fata Moradali and Bernd H. A. Rehm. Bacterial biopolymers: from pathogenesis to advanced materials. 18(4):195–210, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41579-019-0313-3">[DOI]</a>.</p></li>
<li id="id17622"><p>Joana P. Bernardes, Neha Mishra, Florian Tran, Thomas Bahmer, Lena Best, Johanna I. Blase, Dora Bordoni, Jeanette Franzenburg, Ulf Geisen, Jonathan Josephs‐Spaulding, Philipp Köhler, Axel Künstner, Elisa Rosati, Anna C. Aschenbrenner, Petra Bächer, Nathan Baran, Teide Jens Boysen, Burkhard Brandt, Niklas Bruse, Jonathan Dörr, Andreas Dräger, Gunnar Elke, David Ellinghaus, Julia Fischer, Michael Förster, André Franke, Sören Franzenburg, Norbert Frey, Anette Friedrichs, Janina Fuß, Andreas Glück, Jacob Hamm, Finn Hinrichsen, Marc P. Hoeppner, Simon Imm, Ralf Junker, Sina Kaiser, Ying H. Kan, Rainer Knoll, Christoph Lange, Georg Laue, Clemens Lier, Matthias Lindner, Γεώργιος Μαρίνος, Robert Markewitz, Jacob Nattermann, Rainer Noth, Peter Pickkers, Klaus F. Rabe, Alina Renz, Christoph Röcken, Jan Rupp, Annika Schaffarzyk, Alexander Scheffold, Jonas Schulte-Schrepping, Domagoj Schunk, Dirk Skowasch, Thomas Ulas, Klaus‐Peter Wandinger, Michael Wittig, Johannes Zimmermann, Hauke Busch, Bimba F. Hoyer, Christoph Kaleta, Jan Heyckendorf, Matthijs Kox, Jan Rybniker, Stefan Schreiber, Joachim L. Schultze, Philip Rosenstiel, Nicholas E. Banovich, Tushar Desai, Oliver Eickelberg, Muzlifa Haniffa, Péter Horváth, Jonathan A. Kropski, Robert Lafyatis, Joakim Lundeberg, Kerstin B. Meyer, Martijn C. Nawijn, Marko Nikolić, José Ordovas Montañes, Dana Pe’er, Purushothama Rao Tata, Emma L. Rawlins, Aviv Regev, Paul A. Reyfman, Christos Samakovlis, Joachim L. Schultze, Alex K. Shalek, Douglas P. Shepherd, Jason R. Spence, Sarah A. Teichmann, Fabian J. Theis, Alexander M. Tsankov, Maarten van den Berge, Michael von Papen, Jeffrey A. Whitsett, Laure‐Emmanuelle Zaragosi, and Angel Angelov. Longitudinal multi-omics analyses identify responses of megakaryocytes, erythroid cells, and plasmablasts as hallmarks of severe covid-19. 53(6):1296–1314.e9, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.immuni.2020.11.017">[DOI]</a>.</p></li>
<li id="id17627"><p>Kateřina Rohlenová, Jermaine Goveia, Melissa García‐Caballero, Abhishek Subramanian, Joanna Kalucka, Lucas Treps, Kim D. Falkenberg, Laura de Rooij, Yingfeng Zheng, Lin Lin, Liliana Sokol, Laure-Anne Teuwen, Vincent Geldhof, Federico Taverna, Andreas Pircher, Lena‐Christin Conradi, Shawez Khan, Steve Stegen, Dena Panovska, Frederik De Smet, Frank J. T. Staal, René McLaughlin, Stefan Vinckier, Tine Van Bergen, Nadine Ectors, Patrik De Haes, Jian Wang, Lars Bolund, Luc Schoonjans, Tobias K. Karakach, Huanming Yang, Geert Carmeliet, Yizhi Liu, Bernard Thienpont, Mieke Dewerchin, Guy Eelen, Xuri Li, Yonglun Luo, and Peter Carmeliet. Single-cell rna sequencing maps endothelial metabolic plasticity in pathological angiogenesis. 31(4):862–877.e14, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cmet.2020.03.009">[DOI]</a>.</p></li>
<li id="id17628"><p>Xin Fang, Colton J. Lloyd, and Bernhard Ø. Palsson. Reconstructing organisms in silico: genome-scale models and their emerging applications. 18(12):731–743, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41579-020-00440-4">[DOI]</a>.</p></li>
<li id="id17629"><p>Sarah Keating, Dagmar Waltemath, Matthias König, Fengkai Zhang, Andreas Dräger, Claudine Chaouiya, Frank Bergmann, Andrew Finney, Colin S. Gillespie, Tomáš Helikar, Stefan Hoops, Rahuman S. Malik‐Sheriff, Stuart Moodie, Ion I. Moraru, Chris J. Myers, Aurélien Naldi, Brett G. Olivier, Sven Sahle, James C. Schaff, Lucian Smith, Maciej J. Swat, Denis Thieffry, Leandro Watanabe, Darren J. Wilkinson, Michael L. Blinov, Kimberly Begley, James R. Faeder, Harold Gómez, Thomas M. Hamm, Yuichiro Inagaki, Wolfram Liebermeister, Allyson Lister, Daniel Lucio, Eric Mjolsness, Carole J. Proctor, Karthik Raman, Nicolás Rodríguez, Clifford A. Shaffer, Bruce E. Shapiro, Joerg Stelling, Neil Swainston, Naoki Tanimura, John Wagner, Martin Meier‐Schellersheim, Herbert M. Sauro, Bernhard Ø. Palsson, Hamid Bolouri, Hiroaki Kitano, Akira Funahashi, Henning Hermjakob, John C. Doyle, Michael Hucka, Richard R. Adams, Nicholas Allen, Bastian R. Angermann, Marco Antoniotti, Gary D. Bader, Jan Červený, Mélanie Courtot, Chris D. Cox, Piero Dalle Pezze, Emek Demir, William S. Denney, Harish Dharuri, Julien Dorier, Dirk Drasdo, Ali Ebrahim, Johannes Eichner, Johan Elf, Lukas Endler, Chris T. Evelo, Christoph Flamm, Ronan M. T. Fleming, Martina Fröhlich, Mihai Glont, Emanuel Gonçalves, Martin Golebiewski, Hovakim Grabski, Alex Gutteridge, D. Hachmeister, Leonard A. Harris, Ben Heavner, Ron Henkel, William S. Hlavacek, Bin Hu, Daniel R. Hyduke, Hidde de Jong, Nick Juty, Peter D. Karp, Jonathan R. Karr, Douglas B. Kell, Roland Keller, Ilya Kiselev, Steffen Klamt, Edda Klipp, Christian Knüpfer, Fedor Kolpakov, Falko Krause, Martina Kutmon, and Camille Laibe. &lt;scp&gt;sbml&lt;/scp&gt;level 3: an extensible format for the exchange and reuse of biological models. 16(8):e9110–e9110, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20199110">[DOI]</a>.</p></li>
<li id="id17631"><p>Christopher A. Voigt. Synthetic biology 2020–2030: six commercially-available products that are changing our world. 11(1):6379–6379, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-20122-2">[DOI]</a>.</p></li>
<li id="id17632"><p>Jie Zhang, Søren D. Petersen, Tijana Radivojević, Andrés Ramirez, Andrés Pérez-Manríquez, Eduardo Abeliuk, Benjamín J. Sánchez, Zak Costello, Yu Chen, M.J. Fero, Héctor García Martín, Jens Nielsen, Jay D. Keasling, and Michael K. Jensen. Combining mechanistic and machine learning models for predictive engineering and optimization of tryptophan metabolism. 11(1):4880–4880, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-17910-1">[DOI]</a>.</p></li>
<li id="id17637"><p>Federico Baldini, Johannes Hertel, Estelle Sandt, Cyrille C. Thinnes, Lorieza Neuberger-Castillo, Lukas Pavelka, Fay Betsou, Rejko Krüger, Ines Thiele, Gloria Aguayo, Dominic Allen, Wim Ammerlann, Maike K. Aurich, Rudi Balling, Peter Banda, Katy Beaumont, Regina Becker, Daniela Berg, Sylvia Binck, Alexandre Bisdorff, Dheeraj Reddy Bobbili, Kathrin Brockmann, Jessica Calmes, Lorieza Castillo, Nico J. Diederich, Rene Dondelinger, Daniela Esteves, Jean-Yves Ferrand, Ronan M. T. Fleming, Manon Gantenbein, Thomas Gasser, Piotr Gawron, Lars Geffers, Virginie Giarmana, Enrico Glaab, Clarissa P. C. Gomes, Nikolai Goncharenko, Jérôme Graas, Mariela Graziano, Valentin Grouès, Anne Grünewald, Wei Gu, Gaël Hammot, Anne-Ma﻿rie Hanff, Linda Hansen, Maxime Hansen, Hulda S. Haraldsdóttir, Laurent Heirendt, Sylvia Herbrink, Sascha Herzinger, Michaël Heymann, Karsten Hiller, Géraldine Hipp, Joshua Shulman, Laëtitia Huiart, Alexander Hundt, Nadine Jacoby, Jacek Jarosław, Yohan Jaroz, Pierre Kolber, Joachim Kutzera, Zied Landoulsi, Catherine Larue, Roseline Lentz, Inga Liepelt, Robert Liszka, Laura Longhino, Victoria Lorentz, Clare E. Mackay, Walter Maetzler, Katrin Marcus, Guilherme Fernandes Marques, Jan Martens, Conny Mathay, Piotr Matyjaszczyk, Patrick May, Françoise Meisch, Myriam Menster, Maura Minelli, Michel Mittelbronn, Brit Mollenhauer, Kathleen Mommaerts, Carlos S. Moreno, Friedrich Mühlschlegel, Romain Nati, Ulf Nehrbass, Sarah Nickels, Béatrice Nicolai, Jean-Paul Nicolay, Alberto Noronha, Wolfgang H. Oertel, Marek Ostaszewski, Sinthuja Pachchek, Claire Pauly, Magali Perquin, Dorothea Reiter, Isabel Rosety, Kirsten Rump, Venkata Satagopam, and Marc Schlesser. Parkinson’s disease-associated alterations of the gut microbiome predict disease-relevant changes in metabolic functions. 18(1):62–62, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12915-020-00775-7">[DOI]</a>.</p></li>
<li id="id17638"><p>Ines Thiele, Swagatika Sahoo, Almut Heinken, Johannes Hertel, Laurent Heirendt, Maike K. Aurich, and Ronan M. T. Fleming. Personalized whole‐body models integrate metabolism, physiology, and the gut microbiome. 16(5):e8982–e8982, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20198982">[DOI]</a>.</p></li>
<li id="id17642"><p>Priyanka Baloni, Cory C. Funk, Jingwen Yan, James T. Yurkovich, Alexandra Kueider‐Paisley, Kwangsik Nho, Almut Heinken, Wei Jia, Siamak MahmoudianDehkordi, Gregory Louie, Andrew J. Saykin, Matthias Arnold, Gabi Kastenmüller, William J. Griffiths, Ines Thiele, Rima Kaddurah‐Daouk, Nathan D. Price, Rima Kaddurah‐Daouk, Alexandra Kueider‐Paisley, Gregory Louie, P. Murali Doraiswamy, Colette Blach, Arthur Moseley, Paul M. Thompson, Siamak Mahmoudiandehkhordi, Kathleen Welsh-Balmer, Brenda L. Plassman, Andrew J. Saykin, Kwangsik Nho, Gabi Kastenmüller, Matthias Arnold, Sudeepa Bhattacharyya, Xianlin Han, Rebecca Baillie, Oliver Fiehn, Dinesh Kumar Barupal, Peter J. Meikle, Sarkis K. Mazmanian, Mitchel A. Kling, Leslie M. Shaw, John Q. Trojanowski, Jon B. Toledo, Cornelia van Duijin, Thomas Hankemier, Ines Thiele, Almut Heinken, Nathan D. Price, Cory C. Funk, Priyanka Baloni, Wei Jia, David S. Wishart, Roberta Dı́az Brinton, Rui Chang, Lindsay A. Farrer, Rhoda Au, Wendy Qiu, Peter Würtz, Lara M. Mangravite, Jan Krumsiek, John W. Newman, Bin Zhang, and Herman Moreno. Metabolic network analysis reveals altered bile acid synthesis and metabolism in alzheimer’s disease. 1(8):100138–100138, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.xcrm.2020.100138">[DOI]</a>.</p></li>
<li id="id17643"><p>Sébastien Moretti, Van Du T. Tran, Florence Mehl, Mark Ibberson, and Marco Pagni. Metanetx/mnxref: unified namespace for metabolites and biochemical reactions in the context of metabolic models. 49(D1):D570–D574, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkaa992">[DOI]</a>.</p></li>
<li id="id17653"><p>Shan Zhang, Boris Reljić, Chao Liang, Baptiste Kerouanton, Joel Celio Francisco, Jih Hou Peh, Camille Mary, N. Suhas Jagannathan, Volodimir Olexiouk, Claire Tang, Gio Fidelito, Srikanth Nama, Ruey‐Kuang Cheng, Caroline Lei Wee, Loo Chien Wang, Paula Duek, Prabha Sampath, Lydie Lane, Enrico Petretto, Radoslaw M. Sobota, Suresh Jesuthasan, Lisa Tucker‐Kellogg, Bruno Reversade, Gerben Menschaert, Lei Sun, David A. Stroud, and Lena Ho. Mitochondrial peptide brawnin is essential for vertebrate respiratory complex iii assembly. 11(1):1312–1312, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-14999-2">[DOI]</a>.</p></li>
<li id="id17655"><p>Deepanwita Banerjee, Thomas Eng, Andrew K. Lau, Yusuke Sasaki, Brenda Wang, Yan Chen, Jan‐Philip Prahl, Vasanth Singan, Robin A. Herbert, Yuzhong Liu, Deepti Tanjore, Christopher J. Petzold, Jay D. Keasling, and Aindrila Mukhopadhyay. Genome-scale metabolic rewiring improves titers rates and yields of the non-native product indigoidine at scale. 11(1):5385–5385, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-19171-4">[DOI]</a>.</p></li>
<li id="id17666"><p>Rosemary Yu, Kate Campbell, Rui Pereira, Johan Björkeroth, Qi Qi, Egor Vorontsov, Carina Sihlbom, and Jens Nielsen. Nitrogen limitation reveals large reserves in metabolic and translational capacities of yeast. 11(1):1881–1881, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-15749-0">[DOI]</a>.</p></li>
<li id="id17671"><p>Christopher Culley, Supreeta Vijayakumar, Guido Zampieri, and Claudio Angione. A mechanism-aware and multiomic machine-learning pipeline characterizes yeast cell growth. 117(31):18869–18879, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.2002959117">[DOI]</a>.</p></li>
<li id="id17674"><p>Yuliang Wang, Shuyi Ma, and Walter L. Ruzzo. Spatial modeling of prostate cancer metabolic gene expression reveals extensive heterogeneity and selective vulnerabilities. 10(1):3490–3490, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-60384-w">[DOI]</a>.</p></li>
<li id="id17679"><p>Arianna Basile, Stefano Campanaro, Adam Kovalovszki, Guido Zampieri, Alessandro Rossi, İrini Angelidaki, Giorgio Valle, and Laura Treu. Revealing metabolic mechanisms of interaction in the anaerobic digestion microbiome by flux balance analysis. 62():138–149, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.08.013">[DOI]</a>.</p></li>
<li id="id17680"><p>Irene Otero‐Muras and Pablo Carbonell. Automated engineering of synthetic metabolic pathways for efficient biomanufacturing. 63():61–80, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.11.012">[DOI]</a>.</p></li>
<li id="id17690"><p>Aitor Villate, Markel San Nicolás, Mara Gallastegi, Pierre-Antoine Aulas, Maitane Olivares, Aresatz Usobiaga, Néstor Etxebarría, and Oier Aizpurua‐Olaizola. Review: metabolomics as a prediction tool for plants performance under environmental stress. 303():110789–110789, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.plantsci.2020.110789">[DOI]</a>.</p></li>
<li id="id17692"><p>Ailin Liu, Yee‐Shan Ku, Carolina A. Contador, and Hon‐Ming Lam. The impacts of domestication and agricultural practices on legume nutrient acquisition through symbiosis with rhizobia and arbuscular mycorrhizal fungi. 11():583954–583954, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2020.583954">[DOI]</a>.</p></li>
<li id="id17697"><p>Yan Chen, Deepanwita Banerjee, Aindrila Mukhopadhyay, and Christopher J. Petzold. Systems and synthetic biology tools for advanced bioproduction hosts. 64():101–109, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2019.12.007">[DOI]</a>.</p></li>
<li id="id17699"><p>Amin Espah Borujeni, Jing Zhang, Hamid Doosthosseini, Alec A. K. Nielsen, and Christopher A. Voigt. Genetic circuit characterization by inferring rna polymerase movement and ribosome usage. 11(1):5001–5001, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-18630-2">[DOI]</a>.</p></li>
<li id="id17706"><p>Philipp Keller, Εlad Noor, Fabian Meyer, Michael Reiter, Stanislav Anastassov, Patrick Kiefer, and Julia A. Vorholt. Methanol-dependent escherichia coli strains with a complete ribulose monophosphate cycle. 11(1):5403–5403, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-020-19235-5">[DOI]</a>.</p></li>
<li id="id17709"><p>Γεώργιος Μαρίνος, Christoph Kaleta, and Silvio Waschina. Defining the nutritional input for genome-scale metabolic models: a roadmap. 15(8):e0236890–e0236890, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0236890">[DOI]</a>.</p></li>
<li id="id17711"><p>Bing Hu, Miaoxiao Wang, Shuang Geng, Liqun Wen, Mengdi Wu, Yong Nie, Yue‐Qin Tang, and Xiao‐Lei Wu. Metabolic exchange with non-alkane-consuming pseudomonas stutzeri slg510a3-8 improves &lt;i&gt;n&lt;/i&gt; -alkane biodegradation by the alkane degrader &lt;i&gt;dietzia&lt;/i&gt; sp. strain dq12-45-1b. 86(8):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.02931-19">[DOI]</a>.</p></li>
<li id="id17712"><p>Teresa J. Clark, Longyun Guo, John A. Morgan, and Jörg Schwender. Modeling plant metabolism: from network reconstruction to mechanistic models. 71(1):303–326, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1146/annurev-arplant-050718-100221">[DOI]</a>.</p></li>
<li id="id17759"><p>Hock Chuan Yeo, Jongkwang Hong, Meiyappan Lakshmanan, and Dong‐Yup Lee. Enzyme capacity-based genome scale modelling of cho cells. 60():138–147, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.04.005">[DOI]</a>.</p></li>
<li id="id17762"><p>Maureen A. Carey, Andreas Dräger, Moritz Emanuel Beber, Jason A. Papin, and James T. Yurkovich. Community standards to facilitate development and address challenges in metabolic modeling. 16(8):e9235–e9235, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20199235">[DOI]</a>.</p></li>
<li id="id17763"><p>Pritam Chanda, Eduardo Costa, Jie Hu, Shravan Sukumar, John Van Hemert, and Rasna R. Walia. Information theory in computational biology: where we stand today. 22(6):627–627, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/e22060627">[DOI]</a>.</p></li>
<li id="id17764"><p>Denny Popp and Florian Centler. Μbialsim: constraint-based dynamic simulation of complex microbiomes. 8():574–574, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2020.00574">[DOI]</a>.</p></li>
<li id="id17767"><p>L. Şafak Yılmaz, Xuhang Li, Shivani Nanda, Bennett W. Fox, Frank C. Schroeder, and Albertha J.M. Walhout. Modeling tissue‐relevant &lt;i&gt;caenorhabditis elegans&lt;/i&gt; metabolism at network, pathway, reaction, and metabolite levels. 16(10):e9649–e9649, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20209649">[DOI]</a>.</p></li>
<li id="id17768"><p>Javiera López, D.P. Jaramillo Bustos, Conrado Camilo, N Arenas, Pedro A. Saa, and Eduardo Agosín. Engineering saccharomyces cerevisiae for the overproduction of β-ionone and its precursor β-carotene. 8():578793–578793, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2020.578793">[DOI]</a>.</p></li>
<li id="id17770"><p>Sanjeev Dahal, James T. Yurkovich, Hao Xu, Bernhard Ø. Palsson, and Laurence Yang. Synthesizing systems biology knowledge from omics using genome‐scale models. 20(17-18):e1900282–e1900282, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/pmic.201900282">[DOI]</a>.</p></li>
<li id="id17773"><p>Dagmar Waltemath, Martin Golebiewski, Michael L. Blinov, Padraig Gleeson, Henning Hermjakob, Michael Hucka, Esther Thea Inau, Sarah Keating, Matthias König, Olga Krebs, Rahuman S. Malik‐Sheriff, David Nickerson, Ernst Oberortner, Herbert M. Sauro, Falk Schreiber, Lucian Smith, Melanie I. Stefan, Ulrike Wittig, and Chris J. Myers. The first 10 years of the international coordination network for standards in systems and synthetic biology (combine). 17(2-3):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1515/jib-2020-0005">[DOI]</a>.</p></li>
<li id="id17774"><p>Patrick F. Suthers, Charles Foster, Debolina Sarkar, Lin Wang, and Costas D. Maranas. Recent advances in constraint and machine learning-based metabolic modeling by leveraging stoichiometric balances, thermodynamic feasibility and kinetic law formalisms. 63():13–33, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.11.013">[DOI]</a>.</p></li>
<li id="id17777"><p>Thea Os Andersen, Benoît J. Kunath, Live H. Hagen, Magnus Ø. Arntzen, and Phillip B. Pope. Rumen metaproteomics: closer to linking rumen microbial function to animal productivity traits. 186():42–51, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymeth.2020.07.011">[DOI]</a>.</p></li>
<li id="id17778"><p>Almut Heinken, Geeta Acharya, Dmitry A. Ravcheev, Johannes Hertel, Malgorzata Nyga, Onyedika Emmanuel Okpala, Marcus Hogan, Stefanía Magnúsdóttir, Filippo Martinelli Boneschi, German Preciat, Janaka N. Edirisinghe, Christopher S. Henry, Ronan M. T. Fleming, and Ines Thiele. Agora2: large scale reconstruction of the microbiome highlights wide-spread drug-metabolising capacities. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.11.09.375451">[DOI]</a>.</p></li>
<li id="id17780"><p>Tahila Andrighetti, Balázs Bohár, Ney Lemke, Padhmanand Sudhakar, and Tamás Korcsmáros. Microbiolink: an integrated computational pipeline to infer functional effects of microbiome–host interactions. 9(5):1278–1278, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/cells9051278">[DOI]</a>.</p></li>
<li id="id17787"><p>Saeideh Khodaee, Yazdan Asgari, Mehdi Totonchi, and Mohammad Hossein Karimi‐Jafari. Imm1865: a new reconstruction of mouse genome-scale metabolic model. 10(1):6177–6177, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-020-63235-w">[DOI]</a>.</p></li>
<li id="id17788"><p>Bianca M. Esch, Sergej Limar, André Bogdanowski, Christos Gournas, Tushar H. More, Celine Sundag, Stefan Walter, Jürgen J. Heinisch, Christer S. Ejsing, Bruno André, and Florian Fröhlich. Uptake of exogenous serine is important to maintain sphingolipid homeostasis in saccharomyces cerevisiae. 16(8):e1008745–e1008745, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pgen.1008745">[DOI]</a>.</p></li>
<li id="id17794"><p>Chenguang Liang, Austin W.T. Chiang, Anders Holmgaard Hansen, Johnny Arnsdorf, Sanne Schoffelen, James T. Sorrentino, Benjamin P. Kellman, Bokan Bao, Björn G. Voldborg, and Nathan E. Lewis. A markov model of glycosylation elucidates isozyme specificity and glycosyltransferase interactions for glycoengineering. 2():22–36, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.crbiot.2020.01.001">[DOI]</a>.</p></li>
<li id="id17795"><p>Romeu Viana, Óscar Dias, Davide Lagoa, Mónica Galocha, Isabel Rocha, and Miguel C. Teixeira. Genome-scale metabolic model of the human pathogen candida albicans: a promising platform for drug target prediction. 6(3):171–171, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/jof6030171">[DOI]</a>.</p></li>
<li id="id17798"><p>Egils Stalidzāns, Massimiliano Zanin, Paolo Tieri, Filippo Castiglione, Annikka Polster, Stefan Scheiner, Jürgen Pahle, Blaž Stres, Markus List, Jan Baumbach, Manuela Lautizi, Kristel Van Steen, and Harald Schmidt. Mechanistic modeling and multiscale applications for precision medicine: theory and practice. 3(1):36–56, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/nsm.2020.0002">[DOI]</a>.</p></li>
<li id="id17799"><p>Xiaohan Yang, June I. Medford, Kasey Markel, Patrick M. Shih, Henrique C. DePaoli, Cong T. Trinh, Alistair J. McCormick, Raphaël Ployet, Steven G. Hussey, Alexander A. Myburg, Poul Erik Jensen, Md Mahmudul Hassan, Jin Zhang, Wellington Muchero, Udaya C. Kalluri, Hengfu Yin, Renying Zhuo, Paul E. Abraham, Jin‐Gui Chen, David J. Weston, Yinong Yang, Degao Liu, Yi Li, Jessy Labbé, Bing Yang, Jun Hyung Lee, Robert W. Cottingham, Stanton Martin, Mengzhu Lu, Timothy J. Tschaplinski, Guoliang Yuan, Haiwei Lu, Priya Ranjan, Julie C. Mitchell, Stan D. Wullschleger, and Gerald A. Tuskan. Plant biosystems design research roadmap 1.0. 2020():8051764–8051764, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.34133/2020/8051764">[DOI]</a>.</p></li>
<li id="id17805"><p>Yu Chen, Yan Sun, Zhihao Liu, Fengqing Dong, Yuanyuan Li, and Yonghong Wang. Genome‐scale modeling for &lt;i&gt;bacillus coagulans&lt;/i&gt; to understand the metabolic characteristics. 117(11):3545–3558, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27488">[DOI]</a>.</p></li>
<li id="id17806"><p>Kwanjeera Wanichthanarak, Chuthamas Boonchai, Thammaporn Kojonna, Supachitra Chadchawan, Wichian Sangwongchai, and Maysaya Thitisaksakul. Deciphering rice metabolic flux reprograming under salinity stress via in silico metabolic modeling. 18():3555–3566, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2020.11.023">[DOI]</a>.</p></li>
<li id="id17807"><p>Jinxin Zhao, Yan Zhu, Ji-Ru Han, Yu-Wei Lin, Michael Aichem, Jiping Wang, Ke Chen, Tony Velkov, Falk Schreiber, and Jian Li. Genome-scale metabolic modeling reveals metabolic alterations of multidrug-resistant acinetobacter baumannii in a murine bloodstream infection model. 8(11):1793–1793, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms8111793">[DOI]</a>.</p></li>
<li id="id17815"><p>Yan Zhu, Jing Lü, Jinxin Zhao, Xinru Zhang, Heidi H. Yu, Tony Velkov, and Jian Li. Complete genome sequence and genome-scale metabolic modelling of acinetobacter baumannii type strain atcc 19606. 310(3):151412–151412, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijmm.2020.151412">[DOI]</a>.</p></li>
<li id="id17816"><p>Kent M. Rapp, Jackson Jenkins, and Michael J. Betenbaugh. Partners for life: building microbial consortia for the future. 66():292–300, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2020.10.001">[DOI]</a>.</p></li>
<li id="id17828"><p>John R. Casey and Michael J. Follows. A steady-state model of microbial acclimation to substrate limitation. 16(8):e1008140–e1008140, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008140">[DOI]</a>.</p></li>
<li id="id17837"><p>Diego Tec-Campos, Cristal Zúñiga, Anurag Passi, John Del Toro, Juan D. Tibocha‐Bonilla, Alejandro Zepeda, Michael J. Betenbaugh, and Karsten Zengler. Modeling of nitrogen fixation and polymer production in the heterotrophic diazotroph azotobacter vinelandii dj. 11():e00132–e00132, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2020.e00132">[DOI]</a>.</p></li>
<li id="id17864"><p>Tamara Bintener, Maria Pires Pacheco, and Thomas Sauter. Towards the routine use of &lt;i&gt;in silico&lt;/i&gt; screenings for drug discovery using metabolic modelling. 48(3):955–969, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20190867">[DOI]</a>.</p></li>
<li id="id17865"><p>Daniel C. Zielinski, Arjun Patel, and Bernhard Ø. Palsson. The expanding computational toolbox for engineering microbial phenotypes at the genome scale. 8(12):2050–2050, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms8122050">[DOI]</a>.</p></li>
<li id="id17885"><p>Tobias B. Alter, Lars M. Blank, and Birgitta E. Ebert. Protein allocation and enzymatic constraints explain&lt;i&gt;escherichia coli&lt;/i&gt;wildtype and mutant phenotypes. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.02.10.941294">[DOI]</a>.</p></li>
<li id="id17886"><p>Ilija Dukovski, Djordje Bajić, Jeremy M. Chacón, Michael Quintin, Jean C. C. Vila, Snorre Sulheim, Alan R. Pacheco, David B. Bernstein, William J Rieh, Kirill S. Korolev, Álvaro Sánchez, William R. Harcombe, and Daniel Segrè. Computation of microbial ecosystems in time and space (comets): an open source collaborative platform for modeling ecosystems metabolism. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.2009.01734">[DOI]</a>.</p></li>
<li id="id17898"><p>Archanaa Sundararaghavan, Amitava Mukherjee, Swagatika Sahoo, and G. K. Suraishkumar. Mechanism of the oxidative stress‐mediated increase in lipid accumulation by the bacterium, &lt;i&gt;r. opacus&lt;/i&gt; pd630: experimental analysis and genome‐scale metabolic modeling. 117(6):1779–1788, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27330">[DOI]</a>.</p></li>
<li id="id17958"><p>Víctor A. López-Agudelo, Tom A. Mendum, Emma Laing, Huihai Wu, Andrés Baena, Luis F. Barrera, Dany J. V. Beste, and Rigoberto Ríos‐Estepa. A systematic evaluation of mycobacterium tuberculosis genome-scale metabolic networks. 16(6):e1007533–e1007533, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007533">[DOI]</a>.</p></li>
<li id="id17959"><p>S.T.R. Moolamalla and P. K. Vinod. Genome-scale metabolic modelling predicts biomarkers and therapeutic targets for neuropsychiatric disorders. 125():103994–103994, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2020.103994">[DOI]</a>.</p></li>
<li id="id17961"><p>Ratklao Siriwach, Fumio Matsuda, Kentaro Yano, and Masami Yokota Hirai. Drought stress responses in context-specific genome-scale metabolic models of arabidopsis thaliana. 10(4):159–159, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10040159">[DOI]</a>.</p></li>
<li id="id17962"><p>James D. Brunner and Nicholas Chia. Minimizing the number of optimizations for efficient community dynamic flux balance analysis. 16(9):e1007786–e1007786, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007786">[DOI]</a>.</p></li>
<li id="id17965"><p>Frances Blow, Nana Y. D. Ankrah, Noah Clark, Imhoi Koo, Erik L. Allman, Qing Liu, Mallappa Anitha, Andrew D. Patterson, and Angela E. Douglas. Impact of facultative bacteria on the metabolic function of an obligate insect-bacterial symbiosis. 11(4):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.00402-20">[DOI]</a>.</p></li>
<li id="id17969"><p>Fernando Cruz, José P. Faria, Miguel Rocha, Isabel Rocha, and Óscar Dias. A review of methods for the reconstruction and analysis of integrated genome-scale models of metabolism and regulation. 48(5):1889–1903, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20190840">[DOI]</a>.</p></li>
<li id="id17970"><p>Johann F. Jadebeck, Axel Theorell, Samuel Leweke, and Katharina Nöh. Hops: high-performance library for (non-)uniform sampling of convex-constrained models. 37(12):1776–1777, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaa872">[DOI]</a>.</p></li>
<li id="id17972"><p>Thomas C. Keaty and Paul A. Jensen. &lt;scp&gt;gapsplit&lt;/scp&gt; : efficient random sampling for non-convex constraint-based models. 36(8):2623–2625, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btz971">[DOI]</a>.</p></li>
<li id="id17973"><p>Nikolaos Tsiantis and Julio R. Banga. Using optimal control to understand complex metabolic pathways. 21(1):472–472, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-020-03808-8">[DOI]</a>.</p></li>
<li id="id17975"><p>Nana Y. D. Ankrah, Rebecca A. Wilkes, Freya Q. Zhang, Dantong Zhu, Tadeo Kaweesi, Ludmilla Aristilde, and Angela E. Douglas. Syntrophic splitting of central carbon metabolism in host cells bearing functionally different symbiotic bacteria. 14(8):1982–1993, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41396-020-0661-z">[DOI]</a>.</p></li>
<li id="id17980"><p>Sergio Garcia and Cong T. Trinh. Harnessing natural modularity of metabolism with goal attainment optimization to design a modular chassis cell for production of diverse chemicals. 9(7):1665–1681, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.9b00518">[DOI]</a>.</p></li>
<li id="id17982"><p>Priyanka Baloni, Wikum Dinalankara, John C. Earls, Theo Knijnenburg, Donald Geman, Luigi Marchionni, and Nathan D. Price. Identifying personalized metabolic signatures in breast cancer. 11(1):20–20, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo11010020">[DOI]</a>.</p></li>
<li id="id17992"><p>Michel Lavoie, Blanche Saint‐Béat, Jan Strauss, Sébastien Guérin, Antoine Allard, Simon Hardy, Angela Falciatore, and Johann Lavaud. Genome-scale metabolic reconstruction and in silico perturbation analysis of the polar diatom fragilariopsis cylindrus predicts high metabolic robustness. 9(2):30–30, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/biology9020030">[DOI]</a>.</p></li>
<li id="id17993"><p>Agnieszka B. Wegrzyn, Katharina Herzog, Albert Gerding, Marcel Kwiatkowski, Justina C. Wolters, Amalia M. Dolga, A. E. M. van Lint, Ronald J. A. Wanders, Hans R. Waterham, and Barbara M. Bakker. Fibroblast‐specific genome‐scale modelling predicts an imbalance in amino acid metabolism in refsum disease. 287(23):5096–5113, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/febs.15292">[DOI]</a>.</p></li>
<li id="id17995"><p>Priyanka Baloni, Cory C. Funk, Jingwen Yan, James T. Yurkovich, Alexandra Kueider‐Paisley, Kwangsik Nho, Almut Heinken, Wei Jia, Siamak MahmoudianDehkordi, Gregory Louie, Andrew J. Saykin, Matthias Arnold, Gabi Kastenmüller, William J. Griffiths, Ines Thiele, Rima Kaddurah-Daouk, and Nathan D. Price. Metabolic network analysis reveals altered bile acid synthesis and cholesterol metabolism in alzheimer’s disease. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.3596216">[DOI]</a>.</p></li>
<li id="id18000"><p>Sandrien Desmet, Marlies Brouckaert, Wout Boerjan, and Kris Morreel. Seeing the forest for the trees: retrieving plant secondary biochemical pathways from metabolome networks. 19():72–85, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2020.11.050">[DOI]</a>.</p></li>
<li id="id18028"><p>Shouyong Jiang, Yong Wang, Marcus Kaiser, and Natalio Krasnogor. Nihba: a network interdiction approach for metabolic engineering design. 36(11):3482–3492, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaa163">[DOI]</a>.</p></li>
<li id="id18054"><p>Sara Ranjbar, М. Shahmansouri, Pankaj Attri, and Annemie Bogaerts. Effect of plasma-induced oxidative stress on the glycolysis pathway of escherichia coli. 127():104064–104064, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2020.104064">[DOI]</a>.</p></li>
<li id="id18109"><p>Bashir Sajo Mienda and Andreas Dräger. Genome-scale metabolic modeling of escherichia coli and its chassis design for synthetic biology applications. 2189():217–229, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-0822-7_16">[DOI]</a>.</p></li>
<li id="id18111"><p>Mojtaba Tefagh and Stephen Boyd. Swiftcore: a tool for the context-specific reconstruction of genome-scale metabolic networks. 21(1):140–140, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-020-3440-y">[DOI]</a>.</p></li>
<li id="id18113"><p>Chaitra Sarathy, Martina Kutmon, Michael Lenz, Michiel Adriaens, Chris T. Evelo, and Ilja C.W. Arts. Efmviz: a cobra toolbox extension to visualize elementary flux modes in genome-scale metabolic models. 10(2):66–66, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo10020066">[DOI]</a>.</p></li>
<li id="id18114"><p>Ricardo Andrade, Mahdi Doostmohammadi, João Santos, Marie‐France Sagot, Nuno P. Mira, and Susana Vinga. Momo - multi-objective metabolic mixed integer optimization: application to yeast strain engineering. 21(1):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-020-3377-1">[DOI]</a>.</p></li>
<li id="id18115"><p>Alyaa M. Abdel‐Haleem, Vaishnavi Ravikumar, Boyang Ji, Katsuhiko Mineta, Xin Gao, Jens Nielsen, Takashi Gojobori, and Ivan Mijaković. Integrated metabolic modeling, culturing, and transcriptomics explain enhanced virulence of vibrio cholerae during coinfection with enterotoxigenic escherichia coli. 5(5):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00491-20">[DOI]</a>.</p></li>
<li id="id18120"><p>Axel von Kamp and Steffen Klamt. Memo: a method for computing metabolic modules for cell-free production systems. 9(3):556–566, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.9b00434">[DOI]</a>.</p></li>
<li id="id18121"><p>David Botero, Jonathan M. Monk, María Juliana Rodríguez Cubillos, Andrés Rodríguez Cubillos, Mariana Restrepo Benavides, Vivian Bernal‐Galeano, Alejandro Reyes, Andrés Fernando González Barrios, Bernhard Ø. Palsson, Silvia Restrepo, and Adriana Bernal. Genome-scale metabolic model of xanthomonas phaseoli pv. manihotis: an approach to elucidate pathogenicity at the metabolic level. 11():837–837, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2020.00837">[DOI]</a>.</p></li>
<li id="id18124"><p>Yoon-Mi Choi, Yi Qing Lee, Hyun‐Seob Song, and Dong‐Yup Lee. Genome scale metabolic models and analysis for evaluating probiotic potentials. 48(4):1309–1321, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20190668">[DOI]</a>.</p></li>
<li id="id18125"><p>Maryam Khoshnejat, Kaveh Kavousi, Ali Mohammad Banaei‐Moghaddam, and Ali Akbar Moosavi‐Movahedi. Unraveling the molecular heterogeneity in type 2 diabetes: a potential subtype discovery followed by metabolic modeling. 13(1):119–119, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12920-020-00767-0">[DOI]</a>.</p></li>
<li id="id18126"><p>Beatriz García-Jiménez, J Carrasco, Joaquı́n Medina, and Mark D. Wilkinson. Dynamic simulations of microbial communities under perturbations: opportunities for microbiome engineering. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.2.24431/v1">[DOI]</a>.</p></li>
<li id="id18132"><p>Deepanwita Banerjee, Thomas Eng, Andrew K. Lau, Brenda Wang, Yusuke Sasaki, Robin A. Herbert, Yan Chen, Yuzhong Liu, Jan‐Philip Prahl, Vasanth Singan, Deepti Tanjore, Christopher J. Petzold, Jay D. Keasling, and Aindrila Mukhopadhyay. Genome-scale metabolic rewiring to achieve predictable titers rates and yield of a non-native product at scale. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.02.21.954792">[DOI]</a>.</p></li>
<li id="id18133"><p>Alan R. Pacheco and Daniel Segrè. The effects of environmental complexity on microbial community yield and structure. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.07.09.195404">[DOI]</a>.</p></li>
<li id="id18134"><p>Marouen Ben Guebila. Vffva: dynamic load balancing enables large-scale flux variability analysis. 21(1):424–424, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-020-03711-2">[DOI]</a>.</p></li>
<li id="id18135"><p>Yu Chen, Eunice van Pelt‐KleinJan, Berdien van Olst, Sieze Douwenga, Sjef Boeren, Herwig Bachmann, Douwe Molenaar, Jens Nielsen, and Bas Teusink. Proteome constraints reveal targets for improving microbial fitness in nutrient-rich environments. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.10.15.340554">[DOI]</a>.</p></li>
<li id="id18136"><p>Daniel J. Lugar, Sean G. Mack, and Ganesh Sriram. Netred, an algorithm to reduce genome-scale metabolic networks and facilitate the analysis of flux predictions. 65():207–222, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2020.11.003">[DOI]</a>.</p></li>
<li id="id18137"><p>Christian Schulz and Eivind Almaas. Genome-scale reconstructions to assess metabolic phylogeny and organism clustering. 15(12):e0240953–e0240953, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0240953">[DOI]</a>.</p></li>
<li id="id18147"><p>Bianca M. Esch, Sergej Limar, André Bogdanowski, Christos Gournas, Tushar H. More, Celine Sundag, Stefan Walter, Jürgen J. Heinisch, Christer S. Ejsing, Bruno André, and Florian Fröhlich. Uptake of exogenous serine is important to maintain sphingolipid homeostasis in&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.03.30.016220">[DOI]</a>.</p></li>
<li id="id18148"><p>Anne Richelle, Benjamin P. Kellman, Alexander T. Wenzel, Austin W.T. Chiang, Tyler Reagan, Jahir M. Gutierrez, Chintan Joshi, Shangzhong Li, Joanne K. Liu, Helen O. Masson, Jooyong Lee, Zerong Li, Laurent Heirendt, Christophe Trefois, Edwin F. Juarez, Tyler Bath, David Borland, Jill P. Mesirov, Kimberly Robasky, and Nathan E. Lewis. What does your cell really do? model-based assessment of mammalian cells metabolic functionalities using omics data. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.04.26.057943">[DOI]</a>.</p></li>
<li id="id18149"><p>Akshit Goyal, Tong Wang, Veronika Dubinkina, and Sergei Maslov. Ecology-guided prediction of cross-feeding interactions in the human gut microbiome. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.08.140830">[DOI]</a>.</p></li>
<li id="id18150"><p>Sébastien Moretti, Van Du T. Tran, Florence Mehl, Mark Ibberson, and Marco Pagni. Metanetx/mnxref - unified namespace for metabolites and biochemical reactions in the context of metabolic models. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.15.297507">[DOI]</a>.</p></li>
<li id="id18151"><p>Juan F. Poyatos. Genetic buffering and potentiation in metabolism. 16(9):e1008185–e1008185, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1008185">[DOI]</a>.</p></li>
<li id="id18152"><p>Renliang Sun, Yizhou Xu, Hang Zhang, Qiangzhen Yang, Ke Wang, Yongyong Shi, and Zhuo Wang. Mechanistic modeling of gene regulation and metabolism identifies potential targets for hepatocellular carcinoma. 11():595242–595242, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2020.595242">[DOI]</a>.</p></li>
<li id="id18157"><p>Joana Saldida, Anna Paola Muntoni, Daniele De Martino, Georg Hubmann, Bastian Niebel, Anna Mareike Schmidt, Alfredo Braunstein, Andreas Milias‐Argeitis, and Matthias Heinemann. Unbiased metabolic flux inference through combined thermodynamic and&lt;sup&gt;13&lt;/sup&gt;c flux analysis. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.29.177063">[DOI]</a>.</p></li>
<li id="id18158"><p>Benjamin P. Kellman, Anne Richelle, Jeong‐Yeh Yang, Digantkumar Chapla, Austin W.T. Chiang, Julia A. Najera, Bokan Bao, Natália Massaco Koga, Mahmoud A. Mohammad, Anders Bech Bruntse, Morey W. Haymond, Kelley W. Moremen, Lars Bode, and Nathan E. Lewis. Elucidating human milk oligosaccharide biosynthetic genes through network-based multiomics integration. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.02.278663">[DOI]</a>.</p></li>
<li id="id18159"><p>Priyanka Baloni, Wikum Dinalankara, John C. Earls, Theo Knijnenburg, Donald Geman, Luigi Marchionni, and Nathan D. Price. Identifying personalized metabolic signatures in breast cancer. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.20944/preprints202011.0474.v1">[DOI]</a>.</p></li>
<li id="id18194"><p>Jagir R. Hussan and Peter Hunter. Our natural “makeup” reveals more than it hides: modeling the skin and its microbiome. 13(1):e1497–e1497, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/wsbm.1497">[DOI]</a>.</p></li>
<li id="id18273"><p>Bin Wang, Jiarui Yang, Chenchen Gao, Tong Hao, Jinɡjinɡ Li, and Jinsheng Sun. Reconstruction of eriocheir sinensis y-organ genome-scale metabolic network and differential analysis after eyestalk ablation. 11():532492–532492, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2020.532492">[DOI]</a>.</p></li>
<li id="id18275"><p>Gian Marco Messa, Francesco Napolitano, Sarah H. Elsea, Diego di Bernardo, and Xin Gao. A siamese neural network model for the prioritization of metabolic disorders by integrating real and simulated data. 36(Supplement_2):i787–i794, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaa841">[DOI]</a>.</p></li>
<li id="id18279"><p>Joana P. Bernardes, Neha Mishra, Florian Tran, Thomas Bahmer, Lena Best, Johanna I. Blase, Dora Bordoni, Jeanette Franzenburg, Ulf Geisen, Jonathan Josephs‐Spaulding, Philipp Köhler, Axel Künstner, Elisa Rosati, Anna C. Aschenbrenner, Petra Bächer, Nathan Baran, Teide Jens Boysen, Burkhard Brandt, Niklas Bruse, Jonathan Dörr, Andreas Dräger, Gunnar Elke, David Ellinghaus, Julia Fischer, Michael Förster, André Franke, Sören Franzenburg, Norbert Frey, Anette Friedrichs, Janina Fuß, Andreas Glück, Jacob Hamm, Finn Hinrichsen, Marc P. Hoeppner, Simon Imm, Ralf Junker, Sina Kaiser, Ying H. Kan, Rainer Knoll, Christoph Lange, Georg Laue, Clemens Lier, Matthias Lindner, Γεώργιος Μαρίνος, Robert Markewitz, Jacob Nattermann, Rainer Noth, Peter Pickkers, Klaus F. Rabe, Alina Renz, Christoph Röcken, Jan Rupp, Annika Schaffarzyk, Alexander Scheffold, Jonas Schulte-Schrepping, Domagoj Schunck, Dirk Skowasch, Thomas Ulas, Klaus‐Peter Wandinger, Michael Wittig, Johannes Zimmermann, Hauke Busch, Bimba F. Hoyer, Christoph Kaleta, Jan Heyckendorf, Matthijs Kox, Jan Rybniker, Stefan Schreiber, Joachim L. Schultze, and Philip Rosenstiel. Longitudinal multi-omics analysis identifies responses of megakaryocytes, erythroid cells and plasmablasts as hallmarks of severe covid-19 trajectories. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.11.20187369">[DOI]</a>.</p></li>
<li id="id18280"><p>Sean G. Mack and Ganesh Sriram. Netflow: a tool for isolating carbon flows in genome-scale metabolic networks. 12():e00154–e00154, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2020.e00154">[DOI]</a>.</p></li>
<li id="id18287"><p>James D. Brunner and Nicholas Chia. Minimizing the number of optimizations for efficient community dynamic flux balance analysis. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.03.12.988592">[DOI]</a>.</p></li>
<li id="id18294"><p>Zachary B. Haiman, Daniel C. Zielinski, Yuko Koike, James T. Yurkovich, and Bernhard Ø. Palsson. Masspy: building, simulating, and visualizing dynamic biological models in python using mass action kinetics. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.07.31.230334">[DOI]</a>.</p></li>
<li id="id18295"><p>Woosub Shin and Joseph L. Hellerstein. Isolating structural errors in reaction networks in systems biology. 37(3):388–395, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btaa720">[DOI]</a>.</p></li>
<li id="id18296"><p>Petronella R. Hove, Nora Jean Nealon, Siu Hung Joshua Chan, Shea M. Boyer, Hannah B. Haberecht, and Elizabeth P. Ryan. Metabolomics and proteomics of&lt;i&gt;l. rhamnosus&lt;/i&gt;gg and&lt;i&gt;e. coli&lt;/i&gt;nissle probiotic supernatants identify distinct pathways that mediate growth suppression of antimicrobial-resistant pathogens. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.12.21.423897">[DOI]</a>.</p></li>
<li id="id18308"><p>Yizhou Xu, Hang Zhang, Qiangzhen Yang, Renliang Sun, Ke Wang, Yongyong Shi, and Zhuo Wang. Integrated regulatory-metabolic network model reveals critical mechanism and potential targets for hepatocellular carcinoma. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-21615/v1">[DOI]</a>.</p></li>
<li id="id18331"><p>Matthew L. Jenior, Jhansi L. Leslie, Deborah Powers, Elizabeth M. Garrett, Kimberly A. Walker, Mary E. Dickenson, William A. Petri, Rita Tamayo, and Jason A. Papin. Novel drivers of virulence in&lt;i&gt;clostridioides difficile&lt;/i&gt;identified via context-specific metabolic network analysis. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.11.09.373480">[DOI]</a>.</p></li>
<li id="id18336"><p>Johannes Hertel, Almut Heinken, Filippo Martinelli Boneschi, and Ines Thiele. Integration of constraint-based modelling with faecal metabolomics reveals large deleterious effects of&lt;i&gt;fusobacteria&lt;/i&gt;species on community butyrate production. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.09.290494">[DOI]</a>.</p></li>
<li id="id18357"><p>Claudia Patricia Sánchez Henao, Howard Ramírez-Malule, and Víctor A. López-Agudelo. Conceptos básicos de análisis de fluxes metabólicos. 23(1):, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.25100/iyc.v23i1.9509">[DOI]</a>.</p></li>
<li id="id18430"><p>Masoud Ahookhosh, Ronan M. T. Fleming, and Phan Tu Vuong. Finding zeros of hölder metrically subregular mappings via globally convergent levenberg–marquardt methods. 37(1):113–149, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/10556788.2020.1712602">[DOI]</a>.</p></li>
<li id="id18432"><p>Devlin C. Moyer, Alan R. Pacheco, David B. Bernstein, and Daniel Segrè. Stoichiometric modeling of artificial string chemistries. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.16.300491">[DOI]</a>.</p></li>
<li id="id18450"><p>Snorre Sulheim, Fredrik A. Fossheim, Alexander Wentzel, and Eivind Almaas. Automatic reconstruction of metabolic pathways from identified biosynthetic gene clusters. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.11.24.395400">[DOI]</a>.</p></li>
<li id="id18493"><p>Michael A. Henson. Interrogation of the perturbed gut microbiota in gouty arthritis patients through&lt;i&gt;in silico&lt;/i&gt;metabolic modeling. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.02.20187013">[DOI]</a>.</p></li>
<li id="id18494"><p>Maziya Ibrahim and Karthik Raman. Two-species community design of lactic acid bacteria for optimal production of lactate. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.10.24.353805">[DOI]</a>.</p></li>
<li id="id18495"><p>Vetle Simensen, André Voigt, and Eivind Almaas. High-quality genome-scale metabolic model of&lt;i&gt;aurantiochytrium&lt;/i&gt;sp. t66. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.11.06.371070">[DOI]</a>.</p></li>
<li id="id18496"><p>Christian Schulz, Tjaša Kumelj, Emil Karlsen, and Eivind Almaas. Genome-scale metabolic modelling when changes in environmental conditions affect biomass composition. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.12.03.409565">[DOI]</a>.</p></li>
<li id="id18498"><p>Takeyuki Tamura. Efficient reaction deletion algorithms for redesign of constraint-based metabolic networks for metabolite production with weak coupling. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.2.24108/v2">[DOI]</a>.</p></li>
<li id="id18509"><p>Alyaa M. Abdel‐Haleem, Vaishnavi Ravikumar, Boyang Ji, Katsuhiko Mineta, Xin Gao, Jens Nielsen, Takashi Gojobori, and Ivan Mijaković. Integrated metabolic modeling, culturing and transcriptomics explains enhanced virulence of&lt;i&gt;v. cholerae&lt;/i&gt;during co-infection with etec. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.06.16.156158">[DOI]</a>.</p></li>
<li id="id18516"><p>Shan Zhang, Chao Liang, Camille Mary, Baptiste Kerouanton, Joel Celio Francisco, N. Suhas Jagannathan, Volodimir Olexiouk, Jih Hou Peh, Claire Tang, Gio Fidelito, Srikanth Nama, Ruey‐Kuang Cheng, Caroline Lei Wee, Loo Chien Wang, Paula Duek, Prabha Sampath, Lydie Lane, Enrico Petretto, Radoslaw M. Sobota, Suresh Jesuthasan, Lei Sun, Lisa Tucker‐Kellogg, Bruno Reversade, Gerben Menschaert, David A. Stroud, and Lena Ho. Brawnin: a sorf-encoded peptide essential for vertebrate mitochondrial complex iii assembly. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.01.31.926402">[DOI]</a>.</p></li>
<li id="id18517"><p>Michael A. Henson. Computational modeling of the gut microbiota predicts metabolic mechanisms of recurrent&lt;i&gt;clostridioides difficile&lt;/i&gt;infection. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.04.10.036111">[DOI]</a>.</p></li>
<li id="id18518"><p>Daniel Noreña-Caro. Metabolic network analysis of filamentous cyanobacteria. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.31390/gradschool_dissertations.5293">[DOI]</a>.</p></li>
<li id="id18519"><p>P.J. Gawthrop. Bond graphs unify stoichiometric analysis and thermodynamics. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.2007.15917">[DOI]</a>.</p></li>
<li id="id18520"><p>Despoina Kerselidou, Bushra Dohai, David R. Nelson, Sarah Daakour, Nicolas De Cock, Dae‐Kyum Kim, Julien Olivet, Diana C. El Assal, Ashish Jaiswal, Deeya Saha, Charlotte Pain, Filip Matthijssens, Pierre Lemaître, Michaël Herfs, Julien Chapuis, Bart Ghesquière, Didier Vertommen, Verena Kriechbaumer, Kèvin Knoops, Carmen López‐Iglesias, Marc van Zandvoort, Jean‐Charles Lambert, Julien Hanson, Christophe Desmet, Marc Thiry, Kyle J. Lauersen, Marc Vidal, Pieter Van Vlierberghe, Samuel Dequiedt, Kourosh Salehi‐Ashtiani, and Jean‐Claude Twizere. Exostosin-1 glycosyltransferase regulates endoplasmic reticulum architecture and dynamics. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.02.275925">[DOI]</a>.</p></li>
<li id="id18521"><p>Ewelina Węglarz‐Tomczak, Demi J. Rijlaarsdam, Jakub M. Tomczak, and Stanley Brul. Gem-based metabolic profiling for human bone osteosarcoma under different glucose and glutamine availability. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.08.287342">[DOI]</a>.</p></li>
<li id="id18522"><p>Mario L. Arrieta‐Ortiz, Selva Rupa Christinal Immanuel, Serdar Turkarslan, Wei Wu, Brintha Girinathan, Jay Worley, Nicholas DiBenedetto, Olga Soutourina, Johann Peltier, Bruno Dupuy, Lynn Bry, and Nitin S. Baliga. Predictive regulatory and metabolic network models for systems analysis of&lt;i&gt;clostridioides difficile&lt;/i&gt;. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.09.14.297382">[DOI]</a>.</p></li>
<li id="id18523"><p>Harry Beaven and Ioly Kotta‐Loizou. Modelling oxidative stress pathways. ():277–300, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-51862-2_11">[DOI]</a>.</p></li>
<li id="id18524"><p>Christian Schulz and Eivind Almaas. Genome-scale reconstructions to assess metabolic phylogeny and organism clustering. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.10.07.329516">[DOI]</a>.</p></li>
<li id="id18525"><p>Telmo Blasco, Sergio Pérez-Burillo, Francesco Balzerani, Alberto Lerma-Aguilera, Daniel Hinojosa-Nogueira, Silvia Pastoriza, María José Gosalbes, Nuria Jiménez‐Hernández, M. Pilar Francino, José Ángel Rufián‐Henares, Iñigo Apaolaza, and Francisco J. Planes. An extended reconstruction of human gut microbiota metabolism for personalized nutrition. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2020.10.23.350462">[DOI]</a>.</p></li>
<li id="id18526"><p>Bhanwar Lal Puniya, Zhongyuan Zhao, and Tomáš Helikar. Madrid: a pipeline for metabolic drug repurposing identification. ():, 2020. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id18527"><p>Cankut Çubuk. Modeling functional modules using statistical and machine learning methods. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/156175">[DOI]</a>.</p></li>
<li id="id18528"><p>Albert Tafur Rangel, Wendy Lorena Rios Guzman, Carmen Elvira Ojeda Cuella, Daissy Esther Mejia Perez, Ross P. Carlson, Jorge M. Gómez, and Andrés Fernando González Barrios. In silico design for systems-based metabolic engineering in escherichia coli for the bioconversion of aerobic glycerol to succinic acid. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-113854/v1">[DOI]</a>.</p></li>
<li id="id18529"><p>J. Saraiva, Marta Gomes, René Kallies, Carsten Vogt, Antonis Chatzinotas, Peter F. Stadler, Óscar Dias, and Ulisses Nunes da Rocha. Ortsuite – a flexible pipeline for annotation of ecosystem processes and prediction of putative microbial interactions. ():, 2020. <a class="reference external" href="https://doi.org/https://doi.org/10.21203/rs.3.rs-52281/v1">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-6" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-6" name="0-6" role="tabpanel" tabindex="0"><div class="docutils container" id="id18727">
<ol class="arabic simple" start="1">
<li id="id18733"><p>Laurent Heirendt, Sylvain Arreckx, Thomas Pfau, Sebastián N. Mendoza, Anne Richelle, Almut Heinken, Hulda S. Haraldsdóttir, Jacek Wachowiak, Sarah Keating, Vanja Vlasov, Stefanía Magnúsdóttir, Chiam Yu Ng, German Preciat, Alise Žagare, Siu Hung Joshua Chan, Maike K. Aurich, Catherine M. Clancy, Jennifer Modamio, John T. Sauls, Alberto Noronha, Aarash Bordbar, Benjamin Cousins, Diana C. El Assal, Luis V. Valcárcel, Iñigo Apaolaza, Susan Ghaderi, Masoud Ahookhosh, Marouen Ben Guebila, Andrejs Kostromins, Nicolas Sompairac, Hoai Minh Le, Ding Ma, Yuekai Sun, Lin Wang, James T. Yurkovich, Miguel A. P. Oliveira, Phan Tu Vuong, Lemmer P. El Assal, Inna Kuperstein, Andreï Zinovyev, H.S. Hinton, William A Bryant, Francisco J. Aragón Artacho, Francisco J. Planes, Egils Stalidzāns, Alejandro Maass, Santosh Vempala, Michael Hucka, Michael A. Saunders, Costas D. Maranas, Nathan E. Lewis, Thomas Sauter, Bernhard Ø. Palsson, Ines Thiele, and Ronan M. T. Fleming. Creation and analysis of biochemical constraint-based models using the cobra toolbox v.3.0. 14(3):639–702, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-018-0098-2">[DOI]</a>.</p></li>
<li id="id18743"><p>Yasir Suhail, Margo P. Cain, Kiran Vanaja, Paul Kurywchak, Andre Levchenko, Raghu Kalluri, and Kshitiz Gupta. Systems biology of cancer metastasis. 9(2):109–127, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2019.07.003">[DOI]</a>.</p></li>
<li id="id18751"><p>Alan R. Pacheco, Mauricio Moel, and Daniel Segrè. Costless metabolic secretions as drivers of interspecies interactions in microbial ecosystems. 10(1):103–103, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-018-07946-9">[DOI]</a>.</p></li>
<li id="id18775"><p>Manish Kumar, Boyang Ji, Karsten Zengler, and Jens Nielsen. Modelling approaches for studying the microbiome. 4(8):1253–1267, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-019-0491-9">[DOI]</a>.</p></li>
<li id="id18812"><p>Yusuke Minato, Daryl M. Gohl, Joshua M. Thiede, Jeremy M. Chacón, William R. Harcombe, Fumito Maruyama, and Anthony D. Baughn. Genomewide assessment of mycobacterium tuberculosis conditionally essential metabolic pathways. 4(4):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00070-19">[DOI]</a>.</p></li>
<li id="id18833"><p>Charles J. Norsigian, Xin Fang, Yara Seif, Jonathan M. Monk, and Bernhard Ø. Palsson. A workflow for generating multi-strain genome-scale metabolic models of prokaryotes. 15(1):1–14, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-019-0254-3">[DOI]</a>.</p></li>
<li id="id18838"><p>Partho Sen and Matej Orešič. Metabolic modeling of human gut microbiota on a genome scale: an overview. 9(2):22–22, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo9020022">[DOI]</a>.</p></li>
<li id="id18913"><p>Leah Guthrie and Libusha Kelly. Bringing microbiome-drug interaction research into the clinic. 44():708–715, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ebiom.2019.05.009">[DOI]</a>.</p></li>
<li id="id18917"><p>Simonas Marcišauskas, Boyang Ji, and Jens Nielsen. Reconstruction and analysis of a kluyveromyces marxianus genome-scale metabolic model. 20(1):551–551, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-3134-5">[DOI]</a>.</p></li>
<li id="id18954"><p>Christopher P. Long and Maciek R. Antoniewicz. Metabolic flux responses to deletion of 20 core enzymes reveal flexibility and limits of e. coli metabolism. 55():249–257, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2019.08.003">[DOI]</a>.</p></li>
<li id="id18975"><p>Hong Zeng and Aidong Yang. Modelling overflow metabolism in escherichia coli with flux balance analysis incorporating differential proteomic efficiencies of energy pathways. 13(1):3–3, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-018-0677-4">[DOI]</a>.</p></li>
<li id="id18991"><p>Yanfen Fu, Lian He, Jennifer Reeve, David A. C. Beck, and Mary E. Lidstrom. Core metabolism shifts during growth on methanol versus methane in the methanotroph &lt;i&gt;methylomicrobium buryatense&lt;/i&gt; 5gb1. 10(2):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.00406-19">[DOI]</a>.</p></li>
<li id="id18998"><p>Masakazu Toyoshima, Yoshihiro Toya, and Hiroshi Shimizu. Flux balance analysis of cyanobacteria reveals selective use of photosynthetic electron transport components under different spectral light conditions. 143(1):31–43, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11120-019-00678-x">[DOI]</a>.</p></li>
<li id="id19020"><p>Andreas Schwentner, André Feith, Eugenia Münch, Judith Stiefelmaier, Ira Lauer, Lorenzo Favilli, Christoph Massner, Johannes Öhrlein, B. Grund, Andrea T. Hüser, Ralf Takors, and Bastian Blombach. Modular systems metabolic engineering enables balancing of relevant pathways for l-histidine production with corynebacterium glutamicum. 12(1):65–65, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-019-1410-2">[DOI]</a>.</p></li>
<li id="id19026"><p>Fangzhou Shen, Camden Cheek, and Sriram Chandrasekaran. Dynamic network modeling of stem cell metabolism. 1975():305–320, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-9224-9_14">[DOI]</a>.</p></li>
<li id="id19043"><p>Yubing Han and Rachael Hageman Blair. Integration of probabilistic regulatory networks into constraint-based models of metabolism with applications to alzheimer’s disease. 20(1):386–386, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2872-8">[DOI]</a>.</p></li>
<li id="id19044"><p>Joel A. Paulson, Marc Martin-Casas, and Ali Mesbah. Fast uncertainty quantification for dynamic flux balance analysis using non-smooth polynomial chaos expansions. 15(8):e1007308–e1007308, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007308">[DOI]</a>.</p></li>
<li id="id19050"><p>Rotem Katzir, Ibrahim H. Polat, Michal Harel, Shir Katz, Carles Foguet, Vitaly A. Selivanov, Philippe Sabatier, Marta Cascante, Tamar Geiger, and Eytan Ruppin. The landscape of tiered regulation of breast cancer cell metabolism. 9(1):17760–17760, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-54221-y">[DOI]</a>.</p></li>
<li id="id19120"><p>Kentaro Kamata, Yoshihiro Toya, and Hiroshi Shimizu. Effect of precise control of flux ratio between the glycolytic pathways on mevalonate production in &lt;i&gt;escherichia coli&lt;/i&gt;. 116(5):1080–1088, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26923">[DOI]</a>.</p></li>
<li id="id19128"><p>Ankit Gupta, Ahmad Ahmad, Dipesh Chothwe, Midhun K. Madhu, Shireesh Srivastava, and Vineet K. Sharma. Genome-scale metabolic reconstruction and metabolic versatility of an obligate methanotroph &lt;i&gt;methylococcus capsulatus&lt;/i&gt; str. bath. 7():e6685–e6685, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7717/peerj.6685">[DOI]</a>.</p></li>
<li id="id19130"><p>Hong Zeng and Aidong Yang. Quantification of proteomic and metabolic burdens predicts growth retardation and overflow metabolism in recombinant &lt;i&gt;escherichia coli&lt;/i&gt;. 116(6):1484–1495, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26943">[DOI]</a>.</p></li>
<li id="id19140"><p>Kalaivani Paramasivan, Punil Kumar HN, and Sarma Mutturi. Systems-based saccharomyces cerevisiae strain design for improved squalene synthesis. 148():37–45, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2019.04.025">[DOI]</a>.</p></li>
<li id="id19153"><p>Jelili Oyelade, Itunuoluwa Isewon, Olufemi Aromolaran, Efosa Uwoghiren, Titilope M. Dokunmu, Solomon O. Rotimi, Oluwadurotimi Samuel Aworunse, Olawole O. Obembe, and Ezekiel Adebiyi. Computational identification of metabolic pathways of&lt;i&gt;plasmodium falciparum&lt;/i&gt;using the&lt;mml:math xmlns:mml=&quot;http://www.w3.org/1998/math/mathml&quot; id=&quot;m1&quot;&gt;&lt;mml:mi&gt;k&lt;/mml:mi&gt;&lt;/mml:math&gt;-shortest path algorithm. 2019():1–13, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2019/1750291">[DOI]</a>.</p></li>
<li id="id19163"><p>Pandiyan Muthuramalingam, Rajendran Jeyasri, Subramanian Radhesh Krishnan, Shunmugiah Karutha Pandian, Sathishkumar Ramalingam, and Manikandan Ramesh. Integrating the bioinformatics and omics tools for systems analysis of abiotic stress tolerance in oryza sativa (l.). ():59–77, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-13-9624-3_3">[DOI]</a>.</p></li>
<li id="id19205"><p>Sandra Castillo, Kiran Raosaheb Patil, and Paula Jouhten. Yeast genome-scale metabolic models for simulating genotype–phenotype relations. 58():111–133, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-13035-0_5">[DOI]</a>.</p></li>
<li id="id19212"><p>Meisam Yousefi, Sayed‐Amir Marashi, Ali Sharifi‐Zarchi, and Sara Taleahmad. The metabolic network model of primed/naive human embryonic stem cells underlines the importance of oxidation-reduction potential and tryptophan metabolism in primed pluripotency. 9(1):71–71, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13578-019-0334-7">[DOI]</a>.</p></li>
<li id="id19220"><p>Maryam Bagheri, Sayed‐Amir Marashi, and Mohammad Ali Amoozegar. A genome-scale metabolic network reconstruction of extremely halophilic bacterium salinibacter ruber. 14(5):e0216336–e0216336, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0216336">[DOI]</a>.</p></li>
<li id="id19228"><p>Ana M. Untaroiu, Maureen A. Carey, Jennifer L. Güler, and Jason A. Papin. Leveraging the effects of chloroquine on resistant malaria parasites for combination therapies. 20(1):186–186, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2756-y">[DOI]</a>.</p></li>
<li id="id19242"><p>Songül Yaşar Yıldız, Emrah Nıkerel, and Ebru Toksoy Öner. Genome-scale metabolic model of a microbial cell factory ( &lt;i&gt;brevibacillus thermoruber&lt;/i&gt; 423) with multi-industry potentials for exopolysaccharide production. 23(4):237–246, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/omi.2019.0028">[DOI]</a>.</p></li>
<li id="id19250"><p>Yusuke Minato, Daryl M. Gohl, Joshua M. Thiede, Jeremy M. Chacón, William R. Harcombe, Fumito Maruyama, and Anthony D. Baughn. Genome-wide assessment of&lt;i&gt;mycobacterium tuberculosis&lt;/i&gt;conditionally essential metabolic pathways. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/534289">[DOI]</a>.</p></li>
<li id="id19253"><p>Carolina A. Contador, Vida Rodríguez, Bárbara A. Andrews, and Juan A. Asenjo. Use of genome-scale models to get new insights into the marine actinomycete genus salinispora. 13(1):11–11, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-019-0683-1">[DOI]</a>.</p></li>
<li id="id19254"><p>Nutan Chauhan and Shailza Singh. Integrative computational framework for understanding metabolic modulation in leishmania. 7():336–336, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2019.00336">[DOI]</a>.</p></li>
<li id="id19262"><p>Stephania Gómez-Cerón, David Galindo-Betancur, and Howard Ramírez-Malule. Data set of in silico simulation for the production of clavulanic acid and cephamycin c by streptomyces clavuligerus using a genome scale metabolic model. 24():103992–103992, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.dib.2019.103992">[DOI]</a>.</p></li>
<li id="id19264"><p>Kirk Smith, Fangzhou Shen, Ho‐Joon Lee, and Sriram Chandrasekaran. Metabolic signatures of regulation by phosphorylation and acetylation. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/838243">[DOI]</a>.</p></li>
<li id="id19268"><p>Sarah McGarrity, Sigurður Trausti Karvelsson, Ólafur E. Sigurjónsson, and Óttar Rolfsson. Comparative metabolic network flux analysis to identify differences in cellular metabolism. 2088():223–269, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-0159-4_11">[DOI]</a>.</p></li>
<li id="id19328"><p>Hameri Tuure, Georgios Fengos, and Vassily Hatzimanikatis. The effects of model complexity and size on metabolic flux distribution and control. case study in&lt;i&gt;e. coli&lt;/i&gt;. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/666859">[DOI]</a>.</p></li>
<li id="id19377"><p>Zixiang Xu. Egknock: identifying direct gene knockout strategies for microbial strain optimization based on metabolic network with gene-protein-reaction relationships. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/514653">[DOI]</a>.</p></li>
<li id="id19378"><p>Furkan Özden, Metin Can Siper, Necmi Acarsoy, Tugrulcan Elmas, Bryan Marty, Xinjian Qi, A. Ercument Cicek, Furkan Özden, Metin Can Siper, Necmi Acarsoy, Tugrulcan Elmas, Bryan Marty, Xinjian Qi, and A. Ercument Cicek. Dorman: database of reconstructed metabolic networks. 18(4):1474–1480, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/tcbb.2019.2944905">[DOI]</a>.</p></li>
<li id="id19381"><p>Nicolai Panikov. Microbial growth dynamics. ():231–273, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-444-64046-8.00019-7">[DOI]</a>.</p></li>
<li id="id19393"><p>Markus Heinonen, Maria Osmala, Henrik Mannerström, Janne Wallenius, Samuel Kaski, Juho Rousu, and Harri Lähdesmäki. Bayesian metabolic flux analysis reveals intracellular flux couplings. 35(14):i548–i557, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btz315">[DOI]</a>.</p></li>
<li id="id19395"><p>Dikshant Pradhan, Jason A. Papin, and Paul A. Jensen. Efficient enzyme coupling algorithms identify functional pathways in genome-scale metabolic models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/608430">[DOI]</a>.</p></li>
<li id="id19411"><p>Nutan Chauhan and Shailza Singh. Integrative computational framework for understanding metabolic modulation in leishmania. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/512277">[DOI]</a>.</p></li>
<li id="id19412"><p>Md Shamsuddin Sultan Khan, Christopher I. Cazzonelli, Hiroyuki Kurata, Md Bahadur Badsha, Chun Guang Li, Gerald Müench, and Amin Malik Shah Abdul Majid. Metabolic similarity of plant and human: implications for efficacy and regulatory compliance of herbal therapies. 2(1):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.25163/ahi.2046">[DOI]</a>.</p></li>
<li id="id19415"><p>Georgi Georgiev, Tommi Aho, Juha Kesseli, Olli Yli‐Harja, and Stuart Kauffman. Action and power efficiency in self-organization: the case for growth efficiency as a cellular objective in escherichia coli. ():229–244, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-00075-2_8">[DOI]</a>.</p></li>
<li id="id19420"><p>Rui Pereira, Paulo Vilaça, Paulo Maia, Jens Nielsen, and Isabel Rocha. Turnover dependent phenotypic simulation: a quantitative constraint-based simulation method that accommodates all main strain design strategies. 8(5):976–988, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.8b00248">[DOI]</a>.</p></li>
<li id="id19425"><p>Kristin J. Adolfsen and Mark P. Brynildsen. Anti‐virulence therapies through potentiating ros in bacteria. ():239–253, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781119593522.ch10">[DOI]</a>.</p></li>
<li id="id19466"><p>Kyle Medley, Joseph L. Hellerstein, and Herbert M. Sauro. Libsbmljs — enabling web–based sbml tools. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/594804">[DOI]</a>.</p></li>
<li id="id19476"><p>O.W. Davies and Malkhey Verma. 05(02):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.26479/2019.0502.29">[DOI]</a>.</p></li>
<li id="id19526"><p>Hongzhong Lu, Feiran Li, Benjamín J. Sánchez, Zhengming Zhu, Gang Li, Iván Domenzain, Simonas Marcišauskas, Mihail Anton, Dimitra Lappa, Christian Lieven, Moritz Emanuel Beber, Nikolaus Sonnenschein, Eduard J. Kerkhoven, and Jens Nielsen. A consensus s. cerevisiae metabolic model yeast8 and its ecosystem for comprehensively probing cellular metabolism. 10(1):3586–3586, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-019-11581-3">[DOI]</a>.</p></li>
<li id="id19531"><p>Jason H. Yang, Sarah N. Wright, Meagan Hamblin, Douglas McCloskey, Miguel A. Alcantar, Lars Schrübbers, Allison J. Lopatkin, Sangeeta Satish, Amir Nili, Bernhard Ø. Palsson, Graham C. Walker, and James J. Collins. A white-box machine learning approach for revealing antibiotic mechanisms of action. 177(6):1649–1661.e9, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2019.04.016">[DOI]</a>.</p></li>
<li id="id19540"><p>Allison J. Lopatkin, Jonathan Stokes, Erica J. Zheng, Jason H. Yang, Melissa K. Takahashi, Lingchong You, and James J. Collins. Bacterial metabolic state more accurately predicts antibiotic lethality than growth rate. 4(12):2109–2117, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-019-0536-0">[DOI]</a>.</p></li>
<li id="id19553"><p>Juan Nogales, Joshua Mueller, Steinn Guðmundsson, Francisco J. Canalejo, Estrella Duque, Jonathan M. Monk, Adam M. Feist, Juan L. Ramos, Wei Niu, and Bernhard Ø. Palsson. High‐quality genome‐scale metabolic modelling of &lt;i&gt;pseudomonas putida&lt;/i&gt; highlights its broad metabolic capabilities. 22(1):255–269, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.14843">[DOI]</a>.</p></li>
<li id="id19559"><p>Mustafa Saifuddin, Jennifer Bhatnagar, Daniel Segrè, and Adrien C. Finzi. Microbial carbon use efficiency predicted from genome-scale metabolic models. 10(1):3568–3568, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-019-11488-z">[DOI]</a>.</p></li>
<li id="id19583"><p>Rebecca R. Stanway, Ellen Bushell, Anush Chiappino-Pepe, Magali Roques, Theo Sanderson, Blandine Franke–Fayard, Reto Caldelari, Murielle Golomingi, Mary Nyonda, Vikash Pandey, Frank Schwach, Séverine Chevalley, Jai Ramesar, Tom Metcalf, Colin Herd, Paul‐Christian Burda, Julian C. Rayner, Dominique Soldati‐Favre, Chris J. Janse, Vassily Hatzimanikatis, Oliver Billker, and Volker Heussler. Genome-scale identification of essential metabolic processes for targeting the plasmodium liver stage. 179(5):1112–1128.e26, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2019.10.030">[DOI]</a>.</p></li>
<li id="id19591"><p>Boaz Styr, N Gonen, Daniel Zarhin, Antonella Ruggiero, Refaela Atsmon, Neta Gazit, Gabriella Braun, Samuel Frère, Irena Vertkin, Ilana Shapira, Michal Harel, Leore R. Heim, Maxim Katsenelson, Ohad Rechnitz, Saja Fadila, Dori Derdikman, Moran Rubinstein, Tamar Geiger, Eytan Ruppin, and Inna Slutsky. Mitochondrial regulation of the hippocampal firing rate set point and seizure susceptibility. 102(5):1009–1024.e8, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.neuron.2019.03.045">[DOI]</a>.</p></li>
<li id="id19595"><p>Meghan Thommes, Taiyao Wang, Qi Zhao, Ioannis Ch. Paschalidis, and Daniel Segrè. Designing metabolic division of labor in microbial communities. 4(2):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00263-18">[DOI]</a>.</p></li>
<li id="id19604"><p>Helena A. Herrmann, Beth C. Dyson, Lucy Vass, Giles N. Johnson, and Jean‐Marc Schwartz. Flux sampling is a powerful tool to study metabolism under changing environmental conditions. 5(1):32–32, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-019-0109-0">[DOI]</a>.</p></li>
<li id="id19616"><p>Chiara Damiani, Davide Maspero, Marzia Di Filippo, Riccardo Colombo, Dario Pescini, Alex Graudenzi, Hans V. Westerhoff, Lilia Alberghina, Marco Vanoni, and Giancarlo Mauri. Integration of single-cell rna-seq data into population models to characterize cancer metabolism. 15(2):e1006733–e1006733, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006733">[DOI]</a>.</p></li>
<li id="id19630"><p>N. T. Devika and Karthik Raman. Deciphering the metabolic capabilities of bifidobacteria using genome-scale metabolic models. 9(1):18222–18222, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-54696-9">[DOI]</a>.</p></li>
<li id="id19639"><p>Junyoung O. Park, Nian Liu, Kara M. Holinski, David Emerson, Kangjian Qiao, Benjamin M. Woolston, Jingyang Xu, Zbigniew Lazar, M. Ahsanul Islam, Charles Vidoudez, Peter R. Girguis, and Gregory Stephanopoulos. Synergistic substrate cofeeding stimulates reductive metabolism. 1(6):643–651, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s42255-019-0077-0">[DOI]</a>.</p></li>
<li id="id19664"><p>Manali Das, Pradipta Patra, and Amit Ghosh. Metabolic engineering for enhancing microbial biosynthesis of advanced biofuels. 119():109562–109562, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.rser.2019.109562">[DOI]</a>.</p></li>
<li id="id19676"><p>Claudio Angione. Human systems biology and metabolic modelling: a review—from disease metabolism to precision medicine. 2019():1–16, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2019/8304260">[DOI]</a>.</p></li>
<li id="id19679"><p>Lívia S. Zaramela, Cameron Martino, Frederico Alisson‐Silva, Steven D. Rees, Sandra Diaz, Léa Chuzel, Mehul B. Ganatra, Christopher H. Taron, Patrick Secrest, Cristal Zúñiga, Jianbo Huang, Dionicio Siegel, Geoffrey Chang, Ajit Varki, and Karsten Zengler. Gut bacteria responding to dietary change encode sialidases that exhibit preference for red meat-associated carbohydrates. 4(12):2082–2089, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-019-0564-9">[DOI]</a>.</p></li>
<li id="id19680"><p>Cristal Zúñiga, Chien‐Ting Li, Geng Yu, Mahmoud M. Al‐Bassam, Tingting Li, Liqun Jiang, Lívia S. Zaramela, Michael T. Guarnieri, Michael J. Betenbaugh, and Karsten Zengler. Environmental stimuli drive a transition from cooperation to competition in synthetic phototrophic communities. 4(12):2184–2191, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41564-019-0567-6">[DOI]</a>.</p></li>
<li id="id19686"><p>Ilaria Massaiu, Lorenzo Pasotti, Nikolaus Sonnenschein, Erlinda Rama, M. Cavaletti, Paolo Magni, Cinzia Calvio, and Markus J. Herrgård. Integration of enzymatic data in bacillus subtilis genome-scale metabolic model improves phenotype predictions and enables in silico design of poly-γ-glutamic acid production strains. 18(1):3–3, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-018-1052-2">[DOI]</a>.</p></li>
<li id="id19697"><p>Cecilia Noecker, Hsuan-Chao Chiu, Colin P McNally, and Elhanan Borenstein. Defining and evaluating microbial contributions to metabolite variation in microbiome-metabolome association studies. 4(6):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00579-19">[DOI]</a>.</p></li>
<li id="id19715"><p>Lázaro Molina, Ruggero La Rosa, Juan Nogales, and Fernando Rojo. Influence of the crc global regulator on substrate uptake rates and the distribution of metabolic fluxes in &lt;i&gt;pseudomonas putida&lt;/i&gt; kt2440 growing in a complete medium. 21(11):4446–4459, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.14812">[DOI]</a>.</p></li>
<li id="id19750"><p>Emelyne Teo, Sudharshan Ravi, Diogo Barardo, Hyung-Seok Kim, Sheng Fong, Amaury Cazenave‐Gassiot, Tsze Yin Tan, Jianhong Ching, Jean‐Paul Kovalik, Markus R. Wenk, Rudiyanto Gunawan, Philip K. Moore, Barry Halliwell, Nicholas S. Tolwinski, and Jan Gruber. Metabolic stress is a primary pathogenic event in transgenic caenorhabditis elegans expressing pan-neuronal human amyloid beta. 8():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.50069">[DOI]</a>.</p></li>
<li id="id19756"><p>Joshua E. Goldford, Hyman Hartman, Robert Marsland, and Daniel Segrè. Environmental boundary conditions for the origin of life converge to an organo-sulfur metabolism. 3(12):1715–1724, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41559-019-1018-8">[DOI]</a>.</p></li>
<li id="id19775"><p>Lázaro Molina, Ruggero La Rosa, Juan Nogales, and Fernando Rojo. &lt;i&gt;pseudomonas putida&lt;/i&gt; kt2440 metabolism undergoes sequential modifications during exponential growth in a complete medium as compounds are gradually consumed. 21(7):2375–2390, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.14622">[DOI]</a>.</p></li>
<li id="id19791"><p>Minsuk Kim, Beom Gi Park, Eun-Jung Kim, Joonwon Kim, and Byung‐Gee Kim. In silico identification of metabolic engineering strategies for improved lipid production in yarrowia lipolytica by genome-scale metabolic modeling. 12(1):187–187, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-019-1518-4">[DOI]</a>.</p></li>
<li id="id19799"><p>Jae Sung Cho, Changdai Gu, Tae Hee Han, Jae Yong Ryu, and Sang Yup Lee. Reconstruction of context-specific genome-scale metabolic models using multiomics data to study metabolic rewiring. 15():1–11, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2019.02.009">[DOI]</a>.</p></li>
<li id="id19809"><p>Matias Orellana‐Saez, Nicolas Dos Santos Pacheco, José Costa, Katterinne N. Mendez, Matthieu J. Miossec, Claudio Meneses, Eduardo Castro‐Nallar, Andrés E. Marcoleta, and Ignacio Poblete‐Castro. In-depth genomic and phenotypic characterization of the antarctic psychrotolerant strain pseudomonas sp. mpc6 reveals unique metabolic features, plasticity, and biotechnological potential. 10():1154–1154, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2019.01154">[DOI]</a>.</p></li>
<li id="id19820"><p>Laura J. Dunphy, P. H. Yen, and Jason A. Papin. Integrated experimental and computational analyses reveal differential metabolic functionality in antibiotic-resistant pseudomonas aeruginosa. 8(1):3–14.e3, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2018.12.002">[DOI]</a>.</p></li>
<li id="id19821"><p>Alessandra Biz, Scott Proulx, Zhiqing Xu, Kavya Siddartha, Alex Mulet Indrayanti, and Radhakrishnan Mahadevan. Systems biology based metabolic engineering for non-natural chemicals. 37(6):107379–107379, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2019.04.001">[DOI]</a>.</p></li>
<li id="id19838"><p>Emrah Özcan, Sinem Selvin Selvi, Emrah Nıkerel, Bas Teusink, Ebru Toksoy Öner, and Tunahan Çakır. A genome-scale metabolic network of the aroma bacterium leuconostoc mesenteroides subsp. cremoris. 103(7):3153–3165, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-019-09630-4">[DOI]</a>.</p></li>
<li id="id19857"><p>Mochamad Nurcholis, Noppon Lertwattanasakul, Nadchanok Rodrussamee, Tomoyuki Kosaka, Masayuki Murata, and Mamoru Yamada. Integration of comprehensive data and biotechnological tools for industrial applications of kluyveromyces marxianus. 104(2):475–488, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-019-10224-3">[DOI]</a>.</p></li>
<li id="id19884"><p>David B. Bernstein, Floyd E. Dewhirst, and Daniel Segrè. Metabolic network percolation quantifies biosynthetic capabilities across the human oral microbiome. 8():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.39733">[DOI]</a>.</p></li>
<li id="id19885"><p>Uldis Kalnenieks, Elīna Balodīte, Steffi Strähler, Inese Strazdiņa, Julia Rex, Agris Pentjuss, Katsuya Fuchino, Per Bruheim, Reinis Rutkis, Katherine M. Pappas, Robert K. Poole, Oliver Sawodny, and Katja Bettenbrock. Improvement of acetaldehyde production in zymomonas mobilis by engineering of its aerobic metabolism. 10():2533–2533, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2019.02533">[DOI]</a>.</p></li>
<li id="id19898"><p>H Henriksen, Sarah McGarrity, Rósa Sólveig Sigurðardóttir, Travis Nemkov, Angelo D’Alessandro, Bernhard Ø. Palsson, Jakob Stensballe, Charles E. Wade, Óttar Rolfsson, and Pär I. Johansson. Metabolic systems analysis of shock-induced endotheliopathy (shine) in trauma. 272(6):1140–1148, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1097/sla.0000000000003307">[DOI]</a>.</p></li>
<li id="id19913"><p>Maximilian Lularevic, Andrew J. Racher, Colin Jaques, and Alexandros Kiparissides. Improving the accuracy of flux balance analysis through the implementation of carbon availability constraints for intracellular reactions. 116(9):2339–2352, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27025">[DOI]</a>.</p></li>
<li id="id19914"><p>Chien‐Ting Li, Jacob Stephan Yelsky, Yiqun Chen, Cristal Zúñiga, Richard Eng, Liqun Jiang, Alison J. Shapiro, Kai-Wen Huang, Karsten Zengler, and Michael J. Betenbaugh. Utilizing genome-scale models to optimize nutrient supply for sustained algal growth and lipid productivity. 5(1):33–33, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-019-0110-7">[DOI]</a>.</p></li>
<li id="id19919"><p>Jennifer L. Greene, James Daniell, Michael Köpke, Linda J. Broadbelt, and Keith E. J. Tyo. Kinetic ensemble model of gas fermenting clostridium autoethanogenum for improved ethanol production. 148():46–56, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2019.04.021">[DOI]</a>.</p></li>
<li id="id19920"><p>Douaa Mugahid, Tugce Sengül, Xintian You, Yongbo Wang, Leif Steil, Nora Bergmann, Michaël Radkë, Andreas Ofenbauer, Manuela Gesell-Salazar, András Balogh, Stefan Kempa, Baris Tursun, Charles T. Robbins, Uwe Völker, W. Chen, Lonnie A. Nelson, and Michael Gotthardt. Proteomic and transcriptomic changes in hibernating grizzly bears reveal metabolic and signaling pathways that protect against muscle atrophy. 9(1):19976–19976, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-56007-8">[DOI]</a>.</p></li>
<li id="id19933"><p>Oveis Jamialahmadi, Sameereh Hashemi‐Najafabadi, Ehsan Motamedian, Stefano Romeo, and Fatemeh Bagheri. A benchmark-driven approach to reconstruct metabolic networks for studying cancer metabolism. 15(4):e1006936–e1006936, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006936">[DOI]</a>.</p></li>
<li id="id19934"><p>Nathan D. Schwalm, Wais Mojadedi, Elliot S. Gerlach, Marcus S. Benyamin, Matthew Perisin, and Katherine L. Akingbade. Developing a microbial consortium for enhanced metabolite production from simulated food waste. 5(4):98–98, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/fermentation5040098">[DOI]</a>.</p></li>
<li id="id19941"><p>Màrius Tomàs‐Gamisans, Cristiane Conte Paim de Andrade, Francisco Maresca, Sergi Monforte, Pau Ferrer, and Joan Albiol. Redox engineering by ectopic overexpression of nadh kinase in recombinant pichia pastoris ( &lt;i&gt;komagataella phaffii&lt;/i&gt; ): impact on cell physiology and recombinant production of secreted proteins. 86(6):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.02038-19">[DOI]</a>.</p></li>
<li id="id19948"><p>Ragothaman Prathiviraj, Sheela Berchmans, and P. Chellapandi. Analysis of modularity in proteome-wide protein interaction networks of methanothermobacter thermautotrophicus strain δh and metal-loving bacteria. 10(3):179–190, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s42485-019-00019-5">[DOI]</a>.</p></li>
<li id="id19955"><p>David Hoksza, Piotr Gawron, Marek Ostaszewski, Jan Hasenauer, and Reinhard Schneider. Closing the gap between formats for storing layout information in systems biology. 21(4):1249–1260, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbz067">[DOI]</a>.</p></li>
<li id="id19960"><p>Julia Tröndle, Kristin Schoppel, Arne Bleidt, Natalia Trachtmann, Georg A. Sprenger, and Dirk Weuster‐Botz. Metabolic control analysis of l-tryptophan production with escherichia coli based on data from short-term perturbation experiments. 307():15–28, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2019.10.009">[DOI]</a>.</p></li>
<li id="id19969"><p>Beatriz Peñalver Bernabé, Ines Thiele, Eugene Galdones, Anaar Siletz, Sriram Chandrasekaran, Teresa K. Woodruff, Linda J. Broadbelt, and Lonnie D. Shea. Dynamic genome-scale cell-specific metabolic models reveal novel inter-cellular and intra-cellular metabolic communications during ovarian follicle development. 20(1):307–307, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2825-2">[DOI]</a>.</p></li>
<li id="id19982"><p>Anna S. Blazier and Jason A. Papin. Reconciling high-throughput gene essentiality data with metabolic network reconstructions. 15(4):e1006507–e1006507, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006507">[DOI]</a>.</p></li>
<li id="id19983"><p>Adam J. Meyer, Ishtiaq Saaem, Adam D. Silverman, Vanessa A. Varaljay, Rebecca L. Mickol, Steven M. Blum, Alexander V. Tobias, Nathan D. Schwalm, Wais Mojadedi, Elizabeth L. Onderko, Cassandra Bristol, Shangtao Liu, Katelin Pratt, Arturo Casini, Raissa Eluere, Felix Moser, Carrie A. Drake, Maneesh K. Gupta, Nancy Kelley‐Loughnane, Julius B. Lucks, Katherine L. Akingbade, Matthew W. Lux, Sarah M. Glaven, Wendy J. Crookes‐Goodson, Michael C. Jewett, D. Benjamin Gordon, and Christopher A. Voigt. Organism engineering for the bioproduction of the triaminotrinitrobenzene (tatb) precursor phloroglucinol (pg). 8(12):2746–2755, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.9b00393">[DOI]</a>.</p></li>
<li id="id19989"><p>Kirubhakaran Puvendran and Guhan Jayaraman. Enhancement of acetyl-coa by acetate co-utilization in recombinant lactococcus lactis cultures enables the production of high molecular weight hyaluronic acid. 103(17):6989–7001, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-019-09987-6">[DOI]</a>.</p></li>
<li id="id20036"><p>Shoval Lagziel, Won Dong Lee, and Tomer Shlomi. Studying metabolic flux adaptations in cancer through integrated experimental-computational approaches. 17(1):51–51, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12915-019-0669-x">[DOI]</a>.</p></li>
<li id="id20061"><p>Rupert Norman, Thomas Millat, Sarah Schatschneider, Anne M. Henstra, Ronja Breitkopf, Bart Pander, Florence J. Annan, Pawel Piatek, Hassan Hartman, Mark G. Poolman, David A. Fell, Klaus Winzer, Nigel P. Minton, and Charlie Hodgman. Genome‐scale model of &lt;i&gt;c. autoethanogenum&lt;/i&gt; reveals optimal bioprocess conditions for high‐value chemical production from carbon monoxide. 3(2):32–40, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1049/enb.2018.5003">[DOI]</a>.</p></li>
<li id="id20066"><p>Romina Lasry Testa, Claudio Delpino, Vanina Estrada, and M. Soledad Díaz. In silico strategies to couple production of bioethanol with growth in cyanobacteria. 116(8):2061–2073, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26998">[DOI]</a>.</p></li>
<li id="id20100"><p>Aparajitha Srinivasan, S. Vijayakumar, Karthik Raman, and Smita Srivastava. Rational metabolic engineering for enhanced alpha-tocopherol production in helianthus annuus cell culture. 151():107256–107256, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2019.107256">[DOI]</a>.</p></li>
<li id="id20101"><p>Abinaya Badri, Karthik Raman, and Guhan Jayaraman. Uncovering novel pathways for enhancing hyaluronan synthesis in recombinant lactococcus lactis: genome-scale metabolic modeling and experimental validation. 7(6):343–343, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/pr7060343">[DOI]</a>.</p></li>
<li id="id20102"><p>Sucheta Gokhale and Anirban Bhaduri. Provitamin d3 modulation through prebiotics supplementation: simulation based assessment. 9(1):19267–19267, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-55699-2">[DOI]</a>.</p></li>
<li id="id20108"><p>Eun-Youn Kim, Daniel Ashlock, and Sung Ho Yoon. Identification of critical connectors in the directed reaction-centric graphs of microbial metabolic networks. 20(1):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2897-z">[DOI]</a>.</p></li>
<li id="id20119"><p>Sabina Żołędowska, Luana Presta, Marco Fondi, Francesca Decorosi, Luciana Giovannetti, Alessio Mengoni, and Ewa Łojkowska. Metabolic modeling of pectobacterium parmentieri scc3193 provides insights into metabolic pathways of plant pathogenic bacteria. 7(4):101–101, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms7040101">[DOI]</a>.</p></li>
<li id="id20123"><p>Porntip Chiewchankaset, Wanatsanan Siriwat, Malinee Suksangpanomrung, Opas Boonseng, Asawin Meechai, Morakot Tanticharoen, Saowalak Kalapanulak, and Treenut Saithong. Understanding carbon utilization routes between high and low starch-producing cultivars of cassava through flux balance analysis. 9(1):2964–2964, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-39920-w">[DOI]</a>.</p></li>
<li id="id20124"><p>Josi Buerger, Luisa S. Gronenberg, Hans Jasper Genee, and Morten Otto Alexander Sommer. Wiring cell growth to product formation. 59():85–92, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2019.02.014">[DOI]</a>.</p></li>
<li id="id20139"><p>Deepanwita Banerjee and Anu Raghunathan. Constraints-based analysis identifies nad+ recycling through metabolic reprogramming in antibiotic resistant chromobacterium violaceum. 14(1):e0210008–e0210008, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0210008">[DOI]</a>.</p></li>
<li id="id20152"><p>Mahsa Mekanik, Ehsan Motamedian, Reza Fotovat, and Vahab Jafarian. Reconstruction of a genome-scale metabolic model for auxenochlorella protothecoides to study hydrogen production under anaerobiosis using multiple optimal solutions. 44(5):2580–2591, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijhydene.2018.12.049">[DOI]</a>.</p></li>
<li id="id20153"><p>Karel Olavarría, Albert Fina, Mariana I. Velasco, Mark C.M. van Loosdrecht, and A. Wahl. Metabolism of sucrose in a non-fermentative escherichia coli under oxygen limitation. 103(15):6245–6256, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-019-09909-6">[DOI]</a>.</p></li>
<li id="id20178"><p>Emelyne Teo, Sudharshan Ravi, Diogo Barardo, Hyung-Seok Kim, Sheng Fong, Amaury Cazenave‐Gassiot, Tsze Yin Tan, Jianhong Ching, Jean‐Paul Kovalik, Markus R. Wenk, Rudiyanto Gunawan, Philip K. Moore, Barry Halliwell, Nicholas S. Tolwinski, and Jan Gruber. Metabolic stress is a primary pathogenic event in transgenic caenorhabditis elegans expressing pan-neuronal human amyloid beta. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/708537">[DOI]</a>.</p></li>
<li id="id20184"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Mariana Díaz‐Almirón, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Rocío López‐Vacas, Paolo Nanni, Pilar Zamora, Enrique Espinosa, and Juan Ángel Fresno Vara. Computational metabolism modeling predicts risk of distant relapse-free survival in breast cancer patients. 15(30):3483–3490, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.2217/fon-2018-0698">[DOI]</a>.</p></li>
<li id="id20202"><p>Kristopher D. Rawls, Edik M. Blais, Bonnie V. Dougherty, Kalyan C. Vinnakota, Venkat R. Pannala, Anders Wallqvist, Glynis L. Kolling, and Jason A. Papin. Genome-scale characterization of toxicity-induced metabolic alterations in primary hepatocytes. 172(2):279–291, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/toxsci/kfz197">[DOI]</a>.</p></li>
<li id="id20207"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Mariana Díaz‐Almirón, Jorge M. Arevalillo, María Ferrer-Gómez, Hilario Navarro, Paloma Maı́n, Enrique Espinosa, Álvaro Pinto, and Juan Ángel Fresno Vara. Biological molecular layer classification of muscle-invasive bladder cancer opens new treatment opportunities. 19(1):636–636, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12885-019-5858-z">[DOI]</a>.</p></li>
<li id="id20211"><p>Sebastián N. Mendoza, Brett G. Olivier, Douwe Molenaar, and Bas Teusink. A systematic assessment of current genome-scale metabolic reconstruction tools. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/558411">[DOI]</a>.</p></li>
<li id="id20212"><p>Elham Iranmanesh, Mohammad Ali Asadollahi, and Davoud Biria. Improving l-phenylacetylcarbinol production in saccharomyces cerevisiae by in silico aided metabolic engineering. 308():27–34, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2019.11.008">[DOI]</a>.</p></li>
<li id="id20223"><p>Nicolas Sompairac, Jennifer Modamio, Emmanuel Barillot, Ronan M. T. Fleming, Andreï Zinovyev, and Inna Kuperstein. Metabolic and signalling network maps integration: application to cross-talk studies and omics data analysis in cancer. 20(S4):140–140, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2682-z">[DOI]</a>.</p></li>
<li id="id20224"><p>Priyanka Baloni, Vineet Sangar, James T. Yurkovich, Max Robinson, Scott L. Taylor, Christine M. Karbowski, Hisham K. Hamadeh, Yudong D. He, and Nathan D. Price. Genome-scale metabolic model of the rat liver predicts effects of diet restriction. 9(1):9807–9807, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-46245-1">[DOI]</a>.</p></li>
<li id="id20235"><p>Nikolay Martyushenko and Eivind Almaas. Modelexplorer - software for visual inspection and inconsistency correction of genome-scale metabolic reconstructions. 20(1):56–56, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2615-x">[DOI]</a>.</p></li>
<li id="id20253"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Mariana Díaz‐Almirón, Claudia Fortes, María Ferrer-Gómez, Rocío López‐Vacas, Verónica Parra Blanco, Iván Márquez‐Rodas, Ainara Soria, Juan Ángel Fresno Vara, and Enrique Espinosa. Melanoma proteomics suggests functional differences related to mutational status. 9(1):7217–7217, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-43512-z">[DOI]</a>.</p></li>
<li id="id20254"><p>Hamideh Fouladiha, Sayed‐Amir Marashi, Fatemeh Torkashvand, Fereidoun Mahboudi, Nathan E. Lewis, and Behrouz Vaziri. A metabolic network-based approach for developing feeding strategies for cho cells to increase monoclonal antibody production. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/751347">[DOI]</a>.</p></li>
<li id="id20255"><p>Xingping Sun, Chang Chen, Lu Wang, Hongwei Kang, Yong Shen, and Qingyi Chen. Hybrid optimization algorithm for bayesian network structure learning. 10(10):294–294, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/info10100294">[DOI]</a>.</p></li>
<li id="id20274"><p>Anika Küken, Jeanne M. O. Eloundou-Mbebi, Georg Basler, and Zoran Nikoloski. Cellular determinants of metabolite concentration ranges. 15(1):e1006687–e1006687, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006687">[DOI]</a>.</p></li>
<li id="id20294"><p>Cheng Zhang, Sunjae Lee, Gholamreza Bidkhori, Rui Benfeitas, Alen Lovrić, Shuqi Chen, Mathias Uhlén, Jens Nielsen, and Adil Mardinoğlu. Rmetd2: a tool for integration of relative transcriptomics data into genome-scale metabolic models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/663096">[DOI]</a>.</p></li>
<li id="id20301"><p>Krishna Kanhaiya and Dwitiya Tyagi-Tiwari. Identification of drug targets in breast cancer metabolic network. 27(6):975–986, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2019.0258">[DOI]</a>.</p></li>
<li id="id20379"><p>Shaozhen Ding, Pengli Cai, Le Yuan, Yu Tian, Weizhong Tu, Dachuan Zhang, Xingxiang Cheng, Dandan Sun, Junni Chen, and Qian‐Nan Hu. Cf-targeter: a rational biological cell factory targeting platform for biosynthetic target chemicals. 8(10):2280–2286, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.9b00070">[DOI]</a>.</p></li>
<li id="id20380"><p>Freyr Jóhannsson, Níels Árni Árnason, Ragna Landrö, Sveinn Guðmundsson, Ólafur E. Sigurjónsson, and Óttar Rolfsson. Metabolomics study of platelet concentrates photochemically treated with amotosalen and uva light for pathogen inactivation. 60(2):367–377, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/trf.15610">[DOI]</a>.</p></li>
<li id="id20400"><p>Pierre Salvy and Vassily Hatzimanikatis. Etfl: a formulation for flux balance models accounting for expression, thermodynamics, and resource allocation constraints. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/590992">[DOI]</a>.</p></li>
<li id="id20414"><p>Davide Maspero, C. Damiani, Marco Antoniotti, Alex Graudenzi, Marzia Di Filippo, Marco Vanoni, Giulio Caravagna, Riccardo Colombo, Daniele Ramazzotti, and Dario Pescini. The influence of nutrients diffusion on a metabolism-driven model of a multi-cellular system. 171(1-4):279–295, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3233/fi-2020-1883">[DOI]</a>.</p></li>
<li id="id20417"><p>Khadija Mhallem Gziri and Philippe Bogaerts. Determining a unique solution to underdetermined metabolic networks via a systematic path through the most accurate fluxes. 52(1):352–357, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2019.06.087">[DOI]</a>.</p></li>
<li id="id20418"><p>María Teresita Castañeda, Sebastián Núñez, Claudio E. Voget, and Hernán De Battista. In silico optimization of lipid production in rhodosporidium toruloides by gene knockout strategies. 52(1):94–99, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2019.06.043">[DOI]</a>.</p></li>
<li id="id20419"><p>Jorge Ferreira, Vítor Vieira, and Miguel Rocha. Genome-scale metabolic models. ():420–428, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-801238-3.11514-4">[DOI]</a>.</p></li>
<li id="id20433"><p>Philippe Bogaerts and Marianne Rooman. Discopolis: an algorithm for uniform sampling of metabolic flux distributions via iterative sequences of linear programs. 52(26):269–274, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2019.12.269">[DOI]</a>.</p></li>
<li id="id20439"><p>Lucía Trilla‐Fuertes, Ismael Ghanem, Angelo Gámez‐Pozo, Joan Maurel, Laura G-Pastrián, Marta Mendiola, Cristina Peña, Rocío López‐Vacas, Guillermo Prado-Vázquez, Elena López‐Camacho, Andrea Zapater‐Moros, Victoria Heredia-Soto, Míriam Cuatrecasas, Pilar García‐Alfonso, Jaume Capdevila, Carles Conill, Rocio García‐Carbonero, Ricardo Ramos, Claudia Fortes, Carlos Lloréns, Paolo Nanni, Juan Ángel Fresno Vara, and Jaime Feliú. Genetic profile and functional proteomics of anal squamous cell carcinoma: proposal for a molecular classification. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/19009522">[DOI]</a>.</p></li>
<li id="id20475"><p>Lake‐Ee Quek and Nigel Turner. Using the human genome-scale metabolic model recon 2 for steady-state flux analysis of cancer cell metabolism. 1928():479–489, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-9027-6_25">[DOI]</a>.</p></li>
<li id="id20476"><p>Lorenza Vitale, Allison Piovesan, Francesca Antonaros, Pierluigi Strippoli, Maria Chiara Pelleri, and Maria Caracausi. Dataset of differential gene expression between total normal human thyroid and histologically normal thyroid adjacent to papillary thyroid carcinoma. 24():103835–103835, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.dib.2019.103835">[DOI]</a>.</p></li>
<li id="id20497"><p>Tunahan Çakır. Constraint-based modeling of metabolic interactions in and between astrocytes and neurons. ():393–420, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-00817-8_15">[DOI]</a>.</p></li>
<li id="id20499"><p>Reza Mohammadi, Javad Zahiri, and Mohammad Javad Niroomand. Imet: a graphical user interface software tool to merge metabolic networks. 5(6):e01766–e01766, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.heliyon.2019.e01766">[DOI]</a>.</p></li>
<li id="id20500"><p>Leila Hassani, Mohammad R. Moosavi, and Payam Setoodeh. A graph based approach to analyse metabolic networks for strain engineering. 7():1839–1843, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/iraniancee.2019.8786434">[DOI]</a>.</p></li>
<li id="id20501"><p>Arnaud Belcour, Clémence Frioux, Meziane Aite, Anthony Bretaudeau, Anne Siegel, Arnaud Belcour, Clémence Frioux, Meziane Aite, Anthony Bretaudeau, and Anne Siegel. Metage2metabo: metabolic complementarity applied to genomes of large-scale microbiotas for the identification of keystone species. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/803056">[DOI]</a>.</p></li>
<li id="id20502"><p>Jean‐Christophe Lachance, Sébastien Rodrigue, and Bernhard Ø. Palsson. The use of in silico genome-scale models for the rational design of minimal cells. ():141–175, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-030-31897-0_6">[DOI]</a>.</p></li>
<li id="id20524"><p>Nattharat Punyasu, Saowalak Kalapanulak, and Treenut Saithong. Development of a compartmentalized model for insight into the structured metabolic pathway of carbon metabolism in cassava leaves. 13((04) 2019):605–615, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.21475/ajcs.19.13.04.p1639">[DOI]</a>.</p></li>
<li id="id20526"><p>António E. N. Ferreira, Marta Sousa Silva, and Carlos Cordeiro. Metabolic network inference from time series. ():127–133, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-801238-3.11347-9">[DOI]</a>.</p></li>
<li id="id20527"><p>DM Curran, Alexandra Grote, Nirvana Nursimulu, Adam Geber, Denis Voronin, Drew R. Jones, Elodie Ghedin, and John Parkinson. Modeling the metabolic interplay between a parasitic worm and its bacterial endosymbiont allows the identification of novel drug targets. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/767798">[DOI]</a>.</p></li>
<li id="id20528"><p>George C. diCenzo, Michelangelo Tesi, Thomas Pfau, Alessio Mengoni, and Marco Fondi. A virtual nodule environment (vine) for modelling the inter-kingdom metabolic integration during symbiotic nitrogen fixation. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/765271">[DOI]</a>.</p></li>
<li id="id20529"><p>Woosub Shin and Joseph L. Hellerstein. Detecting and isolating mass balance errors in reaction based models in systems biology. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/816876">[DOI]</a>.</p></li>
<li id="id20544"><p>Allison Piovesan, Francesca Antonaros, Pierluigi Strippoli, Lorenza Vitale, Maria Chiara Pelleri, and Maria Caracausi. Reference quantitative transcriptome dataset for adult caenorhabditis elegans. 25():104152–104152, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.dib.2019.104152">[DOI]</a>.</p></li>
<li id="id20568"><p>Jesse C. J. van Dam. Semantic systems biology of prokaryotes : heterogeneous data integration to understand bacterial metabolism. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/464871">[DOI]</a>.</p></li>
<li id="id20569"><p>Jasper J. Koehorst. A fair approach to genomics. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/463364">[DOI]</a>.</p></li>
<li id="id20571"><p>Antti Larjo. Computational methods for modelling and analysing biological networks. ():, 2019. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id20642"><p>Hadas Lapid, Assaf Marron, Smadar Szekely, and David Harel. Using reactive-system modeling techniques to create executable models of biochemical pathways. ():454–464, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.5220/0007572504540464">[DOI]</a>.</p></li>
<li id="id20644"><p>Porntip Chiewchankaset, Saowalak Kalapanulak, and Treenut Saithong. Extended utilization of constraint-based metabolic model in a long-growing crop. 7(5):259–259, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/pr7050259">[DOI]</a>.</p></li>
<li id="id20647"><p>Mohammad Hossein Moteallehi-Ardakani and Sayed‐Amir Marashi. Safeppp: a simple and fast method to find and analyze extreme points of a metabolic phenotypic phase plane. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/642363">[DOI]</a>.</p></li>
<li id="id20648"><p>Hongzhong Lu, Zhengming Zhu, Eduard J. Kerkhoven, and Jens Nielsen. Falconet: an r package to accelerate automatic visualisation of genome scale metabolic models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/662056">[DOI]</a>.</p></li>
<li id="id20649"><p>Dmitry A. Ravcheev. Peer review #1 of &quot;genome-scale metabolic reconstruction and metabolic versatility of an obligate methanotroph methylococcus capsulatus str. bath (v0.1)&quot;. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6685v0.1/reviews/1">[DOI]</a>.</p></li>
<li id="id20655"><p>Henrique da Costa Oliveira. Impacto da inativação de genes relacionados à síntese de produtos de fermentação (lactato e acetato) na produção de 1,3-propanodiol em linhagens de &amp;gt;i&amp;lt;escherichia&amp;gt;/i&amp;lt; &amp;gt;i&amp;lt;coli&amp;gt;/i&amp;lt; recombinantes. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.11606/d.87.2019.tde-06062019-141357">[DOI]</a>.</p></li>
<li id="id20658"><p>Pedraza Palacios and Laura Paola. Análisis metabólico y termodinámico in silico para la biosíntesis de ácido 3-indolacético (aia) a partir de glicerol en azospirillum brasilense. ():, 2019. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id20662"><p>Takeyuki Tamura. Efficient reaction deletion algorithms for redesign of constraint-based metabolic networks for metabolite production with weak coupling. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/563437">[DOI]</a>.</p></li>
<li id="id20679"><p>Villayat Ali, Priyanka Singh, Sonu Gupta, Vijaykumar Yogesh Muley, and Malkhey Verma. 05(02):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.26479/2019.0502.26">[DOI]</a>.</p></li>
<li id="id20682"><p>Ankit Gupta, Ahmad Ahmad, Dipesh Chothwe, Midhnu Madhu, Shireesh Srivastava, Vineet K. Sharma, and Midhun K. Madhu. Peer review #2 of &quot;genome-scale metabolic reconstruction and metabolic versatility of an obligate methanotroph methylococcus capsulatus str. bath (v0.1)&quot;. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6685v0.1/reviews/2">[DOI]</a>.</p></li>
<li id="id20686"><p>Nida Tabassum Khan, Namra Jameel, and Muhammad Noman Khan. Bioinformatics in metabolic engineering. 7(11):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.20431/2349-0365.0711002">[DOI]</a>.</p></li>
<li id="id20688"><p>Jens Nielsen. Faculty opinions recommendation of a white-box machine learning approach for revealing antibiotic mechanisms of action. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3410/f.735729556.793560891">[DOI]</a>.</p></li>
<li id="id20689"><p>Ankit Gupta, Ahmad Ahmad, Dipesh Chothwe, Midhnu Madhu, Shireesh Srivastava, Vineet K. Sharma, and Midhun K. Madhu. Peer review #2 of &quot;genome-scale metabolic reconstruction and metabolic versatility of an obligate methanotroph methylococcus capsulatus str. bath (v0.2)&quot;. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6685v0.2/reviews/2">[DOI]</a>.</p></li>
<li id="id20746"><p>Almut Heinken, Dmitry A. Ravcheev, Federico Baldini, Laurent Heirendt, Ronan M. T. Fleming, and Ines Thiele. Systematic assessment of secondary bile acid metabolism in gut microbes reveals distinct metabolic capabilities in inflammatory bowel disease. 7(1):75–75, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40168-019-0689-3">[DOI]</a>.</p></li>
<li id="id20747"><p>Quanli Liu, Tao Yu, Xiaowei Li, Yu Chen, Kate Campbell, Jens Nielsen, and Yun Chen. Rewiring carbon metabolism in yeast for high level production of aromatic chemicals. 10(1):4976–4976, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-019-12961-5">[DOI]</a>.</p></li>
<li id="id20751"><p>Guido Zampieri, Supreeta Vijayakumar, Elisabeth Yaneske, and Claudio Angione. Machine and deep learning meet genome-scale metabolic modeling. 15(7):e1007084–e1007084, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007084">[DOI]</a>.</p></li>
<li id="id20754"><p>Stéphane Pinhal, Delphine Ropers, Johannes Geiselmann, and Hidde de Jong. Acetate metabolism and the inhibition of bacterial growth by acetate. 201(13):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00147-19">[DOI]</a>.</p></li>
<li id="id20757"><p>Sebastián N. Mendoza, Brett G. Olivier, Douwe Molenaar, and Bas Teusink. A systematic assessment of current genome-scale metabolic reconstruction tools. 20(1):158–158, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13059-019-1769-1">[DOI]</a>.</p></li>
<li id="id20760"><p>Christopher P. Long and Maciek R. Antoniewicz. High-resolution 13c metabolic flux analysis. 14(10):2856–2877, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41596-019-0204-0">[DOI]</a>.</p></li>
<li id="id20765"><p>Yu Chen and Jens Nielsen. Energy metabolism controls phenotypes by protein efficiency and allocation. 116(35):17592–17597, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1906569116">[DOI]</a>.</p></li>
<li id="id20767"><p>Viridiana Olín‐Sandoval, Jason Yu, Leonor Miller‐Fleming, Mohammad Tauqeer Alam, Stephan Kamrad, Clara Correia‐Melo, Robert Haas, Joanna Segal, David Alejandro Peña Navarro, Lucía Herrera-Domínguez, Oscar Méndez‐Lucio, Jakob Vowinckel, Michael Mülleder, and Markus Ralser. Lysine harvesting is an antioxidant strategy and triggers underground polyamine metabolism. 572(7768):249–253, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41586-019-1442-6">[DOI]</a>.</p></li>
<li id="id20768"><p>Johannes Hertel, Amy C. Harms, Almut Heinken, Federico Baldini, Cyrille C. Thinnes, Enrico Glaab, Daniel A. Vasco, Maik Pietzner, Isobel D. Stewart, Nicholas J. Wareham, Claudia Langenberg, Claudia Trenkwalder, Rejko Krüger, Thomas Hankemeier, Ronan M. T. Fleming, Brit Mollenhauer, and Ines Thiele. Integrated analyses of microbiome and longitudinal metabolome data reveal microbial-host interactions on sulfur metabolism in parkinson’s disease. 29(7):1767–1777.e8, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2019.10.035">[DOI]</a>.</p></li>
<li id="id20784"><p>Alexander Eng and Elhanan Borenstein. Microbial community design: methods, applications, and opportunities. 58():117–128, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2019.03.002">[DOI]</a>.</p></li>
<li id="id20789"><p>Kacy Greenhalgh, Javier Ramiro-García, Almut Heinken, Pit Ullmann, Tamara Bintener, Maria Pires Pacheco, Joanna Bagińska, Pranjul Shah, Audrey Frachet, Rashi Halder, Joëlle V. Fritz, Thomas Sauter, Ines Thiele, Serge Haan, Elisabeth Letellier, and Paul Wilmes. Integrated in vitro and in silico modeling delineates the molecular effects of a synbiotic regimen on colorectal-cancer-derived cells. 27(5):1621–1632.e9, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.celrep.2019.04.001">[DOI]</a>.</p></li>
<li id="id20794"><p>Anne Richelle, Austin W.T. Chiang, Chih‐Chung Kuo, and Nathan E. Lewis. Increasing consensus of context-specific metabolic models by integrating data-inferred cell functions. 15(4):e1006867–e1006867, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006867">[DOI]</a>.</p></li>
<li id="id20803"><p>Maciek R. Antoniewicz. Synthetic methylotrophy: strategies to assimilate methanol for growth and chemicals production. 59():165–174, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2019.07.001">[DOI]</a>.</p></li>
<li id="id20805"><p>Su H. Chu, Mengna Huang, Rachel S. Kelly, Elisa Benedetti, Jalal Siddiqui, Oana A. Zeleznik, Alexandre Pereira, David M. Herrington, Craig E. Wheelock, Jan Krumsiek, Michael J. McGeachie, Steven C. Moore, Peter Kraft, Ewy A. Mathé, and Jessica Lasky‐Su. Integration of metabolomic and other omics data in population-based study designs: an epidemiological perspective. 9(6):117–117, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo9060117">[DOI]</a>.</p></li>
<li id="id20848"><p>Kenan Jijakli and Paul A. Jensen. Metabolic modeling of streptococcus mutans reveals complex nutrient requirements of an oral pathogen. 4(5):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00529-19">[DOI]</a>.</p></li>
<li id="id20879"><p>Anne Richelle, Chintan Joshi, and Nathan E. Lewis. Assessing key decisions for transcriptomic data integration in biochemical networks. 15(7):e1007185–e1007185, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007185">[DOI]</a>.</p></li>
<li id="id20882"><p>Rosina Pryor, Daniel Martínez‐Martínez, Leonor M. Quintaneiro, and Filipe Cabreiro. The role of the microbiome in drug response. 60(1):417–435, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1146/annurev-pharmtox-010919-023612">[DOI]</a>.</p></li>
<li id="id20886"><p>Yiqun Chen, Brian O. McConnell, Venkata Gayatri Dhara, Harnish Mukesh Naik, Chien‐Ting Li, Maciek R. Antoniewicz, and Michael J. Betenbaugh. An unconventional uptake rate objective function approach enhances applicability of genome-scale models for mammalian cells. 5(1):25–25, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-019-0103-6">[DOI]</a>.</p></li>
<li id="id20887"><p>Benjamín J. Sánchez, Feiran Li, Eduard J. Kerkhoven, and Jens Nielsen. Slimer: probing flexibility of lipid metabolism in yeast with an improved constraint-based modeling framework. 13(1):4–4, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-018-0673-8">[DOI]</a>.</p></li>
<li id="id20890"><p>Federico Baldini, Johannes Hertel, Estelle Sandt, Cyrille C. Thinnes, Lorieza Neuberger-Castillo, Lukas Pavelka, Fay Betsou, Rejko Krüger, and Ines Thiele. Parkinson’s disease-associated alterations of the gut microbiome can invoke disease-relevant metabolic changes. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/691030">[DOI]</a>.</p></li>
<li id="id20892"><p>Sander Y.A. Rodenburg, Michael Seidl, Howard S. Judelson, Andrea L. Vu, Francine Govers, and Dick de Ridder. Metabolic model of the&lt;i&gt;phytophthora infestans&lt;/i&gt;-tomato interaction reveals metabolic switches during host colonization. 10(4):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.00454-19">[DOI]</a>.</p></li>
<li id="id20897"><p>Yan Zhu, Jinxin Zhao, Mohd Hafidz Mahamad Maifiah, Tony Velkov, Falk Schreiber, and Jian Li. Metabolic responses to polymyxin treatment in &lt;i&gt;acinetobacter baumannii&lt;/i&gt; atcc 19606: integrating transcriptomics and metabolomics with genome-scale metabolic modeling. 4(1):, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00157-18">[DOI]</a>.</p></li>
<li id="id20906"><p>Masoud Ahookhosh, Francisco J. Aragón Artacho, Ronan M. T. Fleming, and Phan Tu Vuong. Local convergence of the levenberg–marquardt method under hölder metric subregularity. 45(5-6):2771–2806, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10444-019-09708-7">[DOI]</a>.</p></li>
<li id="id20911"><p>Peter C. St. John and Yannick J. Bomble. Approaches to computational strain design in the multiomics era. 10():597–597, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2019.00597">[DOI]</a>.</p></li>
<li id="id20918"><p>Taneli Pusa, Mariana Galvão Ferrarini, Ricardo Andrade, Arnaud Mary, Alberto Marchetti-Spaccamela, Leen Stougie, and Marie‐France Sagot. Moomin – mathematical exploration of ’omics data on a metabolic network. 36(2):514–523, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btz584">[DOI]</a>.</p></li>
<li id="id20925"><p>Guan Wang, Cees Haringa, Wenjun Tang, Henk Noorman, Ju Chu, Yingping Zhuang, and Siliang Zhang. Coupled metabolic‐hydrodynamic modeling enabling rational scale‐up of industrial bioprocesses. 117(3):844–867, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.27243">[DOI]</a>.</p></li>
<li id="id20930"><p>Jorge Fernández-de-Cossio-Díaz and Roberto Mulet. Maximum entropy and population heterogeneity in continuous cell cultures. 15(2):e1006823–e1006823, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006823">[DOI]</a>.</p></li>
<li id="id20957"><p>Marouen Ben Guebila and Ines Thiele. Predicting gastrointestinal drug effects using contextualized metabolic models. 15(6):e1007100–e1007100, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1007100">[DOI]</a>.</p></li>
<li id="id20997"><p>Elaina M. Maldonado, F. Taha, Joyeeta Rahman, and Shamima Rahman. Systems biology approaches toward understanding primary mitochondrial diseases. 10():19–19, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2019.00019">[DOI]</a>.</p></li>
<li id="id21078"><p>Ilaria Granata, Enrico Troiano, Mara Sangiovanni, and Mario Rosario Guarracino. Integration of transcriptomic data in a genome-scale metabolic model to investigate the link between obesity and breast cancer. 20(S4):162–162, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2685-9">[DOI]</a>.</p></li>
<li id="id21085"><p>Nathalie Poupin, Marie Tremblay‐Franco, Aurélien Amiel, Cécile Canlet, Didier Rémond, Laurent Debrauwer, Dominique Dardevet, Ines Thiele, Maike K. Aurich, Fabien Jourdan, Isabelle Savary‐Auzeloux, and Sergio Polakof. Arterio-venous metabolomics exploration reveals major changes across liver and intestine in the obese yucatan minipig. 9(1):12527–12527, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-019-48997-2">[DOI]</a>.</p></li>
<li id="id21088"><p>Alessandro Di Stefano, Marialisa Scatà, Supreeta Vijayakumar, Claudio Angione, Aurelio La Corte, and Píetro Lió. Social dynamics modeling of chrono-nutrition. 15(1):e1006714–e1006714, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006714">[DOI]</a>.</p></li>
<li id="id21112"><p>Bergthor Traustason. Amino acid requirements of the chinese hamster ovary cell metabolism during recombinant protein production. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/796490">[DOI]</a>.</p></li>
<li id="id21133"><p>Wentao Dong, Sun Jin Moon, Joanne K. Kelleher, and Gregory Stephanopoulos. Dissecting mammalian cell metabolism through &lt;sup&gt;13&lt;/sup&gt;c- and &lt;sup&gt;2&lt;/sup&gt;h-isotope tracing: interpretations at the molecular and systems levels. 59(6):2593–2610, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.iecr.9b05154">[DOI]</a>.</p></li>
<li id="id21174"><p>José Ramón Valverde, Sonia Gullón, Clara A. García-Herrero, Iván Campoy, and Rafael P. Mellado. Dynamic metabolic modelling of overproduced protein secretion in streptomyces lividans using adaptive dfba. 19(1):233–233, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12866-019-1591-7">[DOI]</a>.</p></li>
<li id="id21185"><p>Natalie Stanford, Martin Scharm, Paul D. Dobson, Martin Golebiewski, Michael Hucka, Varun B. Kothamachu, David Nickerson, Stuart Owen, Jürgen Pahle, Ulrike Wittig, Dagmar Waltemath, Carole Goble, Pedro Mendes, and Jacky L. Snoep. Data management in computational systems biology: exploring standards, tools, databases, and packaging best practices. 2049():285–314, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-9736-7_17">[DOI]</a>.</p></li>
<li id="id21227"><p>Yoon-Mi Choi, Dong‐Hyuk Choi, Yi Qing Lee, Lokanand Koduru, Nathan E. Lewis, Meiyappan Lakshmanan, and Dong‐Yup Lee. Mitigating biomass composition uncertainties in flux balance analysis using ensemble representations. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/652040">[DOI]</a>.</p></li>
<li id="id21231"><p>Brendan Fu‐Long Sieow, Toni Juhani Nurminen, Hua Ling, and Matthew Wook Chang. Meta‐omics‐ and metabolic modeling‐assisted deciphering of human microbiota metabolism. 14(9):e1800445–e1800445, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201800445">[DOI]</a>.</p></li>
<li id="id21244"><p>Tal Pecht, Anna C. Aschenbrenner, Thomas Ulas, and A. Succurro. Modeling population heterogeneity from microbial communities to immune response in cells. 77(3):415–432, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00018-019-03378-w">[DOI]</a>.</p></li>
<li id="id21252"><p>Kristopher D. Rawls, Bonnie V. Dougherty, and Jason A. Papin. Metabolic network reconstructions to predict drug targets and off-target effects. 2088():315–330, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-0716-0159-4_14">[DOI]</a>.</p></li>
<li id="id21266"><p>Nattapat Isarankura Na Ayudhya, Kobkul Laoteng, Yuanda Song, Asawin Meechai, and Wanwipa Vongsangnak. Metabolic traits specific for lipid-overproducing strain of &lt;i&gt;mucor circinelloides&lt;/i&gt; wj11 identified by genome-scale modeling approach. 7():e7015–e7015, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.7717/peerj.7015">[DOI]</a>.</p></li>
<li id="id21267"><p>Miguel Ponce-de-León, Iñigo Apaolaza, Alfonso Valencia, and Francisco J. Planes. On the inconsistent treatment of gene-protein-reaction rules in context-specific metabolic models. 36(6):1986–1988, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btz832">[DOI]</a>.</p></li>
<li id="id21276"><p>Maureen A. Carey, Andreas Dräger, Jason A. Papin, and James T. Yurkovich. Community standards to facilitate development and address challenges in metabolic modeling. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/700112">[DOI]</a>.</p></li>
<li id="id21277"><p>Freyr Jóhannsson, James T. Yurkovich, Steinn Guðmundsson, Ólafur E. Sigurjónsson, and Óttar Rolfsson. Temperature dependence of platelet metabolism. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/802660">[DOI]</a>.</p></li>
<li id="id21299"><p>Priyanka Baloni, Cory C. Funk, Jingwen Yan, James T. Yurkovich, Alexandra Kueider‐Paisley, Kwangsik Nho, Almut Heinken, Wei Jia, Siamak MahmoudianDehkordi, Gregory Louie, Andrew J. Saykin, Matthias Arnold, Gabi Kastenmüller, William J. Griffiths, Ines Thiele, Rima Kaddurah‐Daouk, and Nathan D. Price. Identifying differences in bile acid pathways for cholesterol clearance in alzheimer’s disease using metabolic networks of human brain regions. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/782987">[DOI]</a>.</p></li>
<li id="id21398"><p>Almut Heinken and Ines Thiele. Systematic interrogation of the distinct metabolic potential in gut microbiomes of inflammatory bowel disease patients with dysbiosis. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/640649">[DOI]</a>.</p></li>
<li id="id21399"><p>Hock Chuan Yeo, Jongkwang Hong, Meiyappan Lakshmanan, and Dong‐Yup Lee. Enzyme capacity-based genome scale modelling of cho cells. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/2019.12.19.883652">[DOI]</a>.</p></li>
<li id="id21407"><p>Thomas C. Keaty and Paul A. Jensen. Gapsplit: efficient random sampling for non-convex constraint-based models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/652917">[DOI]</a>.</p></li>
<li id="id21428"><p>Denny Popp and Florian Centler. &lt;i&gt;µ&lt;/i&gt;bialsim: constraint-based dynamic simulation of complex microbiomes. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/716126">[DOI]</a>.</p></li>
<li id="id21444"><p>André Schultz and Rehan Akbani. Sammi: a semi-automated tool for the visualization of metabolic networks. 36(8):2616–2617, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btz927">[DOI]</a>.</p></li>
<li id="id21449"><p>Yuliang Wang, Shuyi Ma, Walter L. Ruzzo, Yuliang Wang, Shuyi Ma, and Walter L. Ruzzo. Spatial modeling of prostate cancer metabolic gene expression reveals extensive heterogeneity and selective vulnerabilities. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/719294">[DOI]</a>.</p></li>
<li id="id21450"><p>Jie Zhang, Søren D. Petersen, Tijana Radivojević, Andrés Ramirez, Andrés M. Perez, Eduardo Abeliuk, Benjamín J. Sánchez, Zachary Costello, Yu Chen, Mike Fero, Héctor García Martín, Jens Nielsen, Jay D. Keasling, and Michael K. Jensen. Predictive engineering and optimization of tryptophan metabolism in yeast through a combination of mechanistic and machine learning models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/858464">[DOI]</a>.</p></li>
<li id="id21451"><p>Ewelina Węglarz‐Tomczak, Thierry D.G.A. Mondeel, Diewertje Piebes, and Hans V. Westerhoff. Simultaneous integration of gene expression and nutrient availability for studying metabolism of hepatocellular carcinoma. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/674150">[DOI]</a>.</p></li>
<li id="id21456"><p>Paolo Romano, Arnaud Céol, Andreas Dräger, Antonino Fiannaca, Rosalba Giugno, Massimo La Rosa, Luciano Milanesi, Ulrich Pfeffer, Riccardo Rizzo, Soo-Yong Shin, Junfeng Xia, and Alfonso Urso. The 2017 network tools and applications in biology (nettab) workshop: aims, topics and outcomes. 20(S4):125–125, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-019-2681-0">[DOI]</a>.</p></li>
<li id="id21477"><p>Miguel Ponce-de-León, Iñigo Apaolaza, Alfonso Valencia, and Francisco J. Planes. On the inconsistent treatment of gene-protein-reaction rules in context-specific metabolic models. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/593277">[DOI]</a>.</p></li>
<li id="id21552"><p>Partho Sen, Alex M. Dickens, María Asunción López-Bascón, Tuomas Lindeman, Esko Kemppainen, Santosh Lamichhane, Tuukka Rönkkö, Jorma Ilonen, Jorma Toppari, Riitta Veijola, Heikki Hyöty, Tuulia Hyötyläinen, Mikael Knip, and Matej Orešič. Metabolic alterations in human peripheral blood mononuclear cells associate with progression to islet autoimmunity and type 1 diabetes. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/658500">[DOI]</a>.</p></li>
<li id="id21568"><p>Agnieszka B. Wegrzyn, Katharina Herzog, Albert Gerding, Marcel Kwiatkowski, Justina C. Wolters, Amalia M. Dolga, A. E. M. van Lint, Ronald J. A. Wanders, Hans R. Waterham, and Barbara M. Bakker. Fibroblast-specific genome-scale modelling predicts an imbalance in amino acid metabolism in refsum disease. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/776575">[DOI]</a>.</p></li>
<li id="id21569"><p>Rui M. C. Portela, Anne Richelle, Patrick Dumas, and Moritz von Stosch. Time integrated flux analysis: exploiting the concentration measurements directly for cost-effective metabolic network flux analysis. 7(12):620–620, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/microorganisms7120620">[DOI]</a>.</p></li>
<li id="id21570"><p>Kjerstin De Winter, Justien Ghesquière, Wim Teughels, Steffen Waldherr, and Kristel Bernaerts. Towards the construction of gsmn-based community model for an oral biofilm. 52(26):193–199, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2019.12.257">[DOI]</a>.</p></li>
<li id="id21613"><p>Alvar J. Alonso-Lavin, Djordje Bajić, and Juan F. Poyatos. Tolerance to nadh/nad&lt;sup&gt;+&lt;/sup&gt; imbalance anticipates aging and anti-aging interventions. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/719401">[DOI]</a>.</p></li>
<li id="id21624"><p>Shangzhong Li, Anne Richelle, and Nathan E. Lewis. Enhancing product and bioprocess attributes using genome‐scale models of cho metabolism. ():73–95, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527811410.ch4">[DOI]</a>.</p></li>
<li id="id21629"><p>Marouen Ben Guebila. Achr.cu: gpu-accelerated sampling of metabolic networks. 4(37):1363–1363, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.21105/joss.01363">[DOI]</a>.</p></li>
<li id="id21631"><p>Shouyong Jiang, Yong Wang, Marcus Kaiser, and Natalio Krasnogor. Nihba: a network interdiction approach with hybrid benders algorithm for strain design. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/752923">[DOI]</a>.</p></li>
<li id="id21632"><p>Tahila Andrighetti, Balázs Bohár, Ney Lemke, Padhmanand Sudhakar, and Tamás Korcsmáros. Microbiolink: an integrated computational pipeline to infer functional effects of microbiome-host interactions. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/837062">[DOI]</a>.</p></li>
<li id="id21633"><p>Juan F. Poyatos. Genetic buffering and potentiation in metabolism. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/845792">[DOI]</a>.</p></li>
<li id="id21634"><p>Víctor A. López-Agudelo, Emma Laing, Tom A. Mendum, Andrés Baena, Luis F. Barrera, Dany J. V. Beste, and Rigoberto Ríos‐Estepa. A systematic evaluation of &lt;i&gt;mycobacterium tuberculosis&lt;/i&gt; genome-scale metabolic networks. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/837401">[DOI]</a>.</p></li>
<li id="id21635"><p>Marouen Ben Guebila and Ines Thiele. Pan-organ model integration of metabolic and regulatory processes in type 1 diabetes. ():, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/859876">[DOI]</a>.</p></li>
<li id="id21636"><p>Partho Sen, Santosh Lamichhane, Alex M. Dickens, and Matej Orešič. The role of omic technologies in the study of the human gut microbiome. ():469–481, 2019. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-08-100596-5.22922-3">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-7" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-7" name="0-7" role="tabpanel" tabindex="0"><div class="docutils container" id="id21848">
<ol class="arabic simple" start="1">
<li id="id21869"><p>Hao Wang, Simonas Marcišauskas, Benjamín J. Sánchez, Iván Domenzain, Daniel Hermansson, Rasmus Ågren, Jens Nielsen, and Eduard J. Kerkhoven. Raven 2.0: a versatile toolbox for metabolic network reconstruction and a case study on streptomyces coelicolor. 14(10):e1006541–e1006541, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006541">[DOI]</a>.</p></li>
<li id="id21938"><p>Partho Sen, Esko Kemppainen, and Matej Orešič. Perspectives on systems modeling of human peripheral blood mononuclear cells. 4():96–96, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmolb.2017.00096">[DOI]</a>.</p></li>
<li id="id21973"><p>Veronika Boczonadi, Martin King, Anthony C. Smith, Monika Oláhová, Boglárka Bánsági, Andreas Roos, Filmon Eyassu, Christoph H. Borchers, Venkateswaran Ramesh, Hanns Lochmüller, Tuomo Polvikoski, Roger G. Whittaker, Angela Pyle, Helen Griffin, Robert W. Taylor, Patrick F. Chinnery, Alan J. Robinson, Edmund R.S. Kunji, and Rita Horvàth. Mitochondrial oxodicarboxylate carrier deficiency is associated with mitochondrial dna depletion and spinal muscular atrophy–like disease. 20(10):1224–1235, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/gim.2017.251">[DOI]</a>.</p></li>
<li id="id22033"><p>Hai He, Christian Edlich‐Muth, Steffen N. Lindner, and Arren Bar‐Even. Ribulose monophosphate shunt provides nearly all biomass and energy required for growth of &lt;i&gt;e. coli&lt;/i&gt;. 7(6):1601–1611, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.8b00093">[DOI]</a>.</p></li>
<li id="id22045"><p>Ioannis Papapetridis, Maarten D. Verhoeven, Sanne J. Wiersma, Maaike Goudriaan, Antonius J. A. van Maris, and Jack T. Pronk. Laboratory evolution for forced glucose-xylose co-consumption enables identification of mutations that improve mixed-sugar fermentation by xylose-fermenting saccharomyces cerevisiae. 18(6):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsyr/foy056">[DOI]</a>.</p></li>
<li id="id22053"><p>Andreas Schwentner, André Feith, Eugenia Münch, Tobias Busche, Christian Rückert, Jörn Kalinowski, Ralf Takors, and Bastian Blombach. Metabolic engineering to guide evolution – creating a novel mode for l-valine production with corynebacterium glutamicum. 47():31–41, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2018.02.015">[DOI]</a>.</p></li>
<li id="id22086"><p>Kurt Patterson, James Yu, Jenny Landberg, Ivan Chang, Farbod Shavarebi, Virginia Bilanchone, and Suzanne Sandmeyer. Functional genomics for the oleaginous yeast yarrowia lipolytica. 48():184–196, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2018.05.008">[DOI]</a>.</p></li>
<li id="id22097"><p>Hameri Tuure, Georgios Fengos, Meriç Ataman, Ljubiša Mišković, and Vassily Hatzimanikatis. Kinetic models of metabolism that consider alternative steady-state solutions of intracellular fluxes and concentrations. 52():29–41, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2018.10.005">[DOI]</a>.</p></li>
<li id="id22102"><p>Michael Scheunemann, Siobhán M. Brady, and Zoran Nikoloski. Integration of large-scale data for extraction of integrated arabidopsis root cell-type specific models. 8(1):7919–7919, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-26232-8">[DOI]</a>.</p></li>
<li id="id22125"><p>Kristopher D. Rawls, Bonnie V. Dougherty, Edik M. Blais, Ethan Stancliffe, Glynis L. Kolling, Kalyan C. Vinnakota, Venkat R. Pannala, Anders Wallqvist, and Jason A. Papin. A simplified metabolic network reconstruction to promote understanding and development of flux balance analysis tools. 105():64–71, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2018.12.010">[DOI]</a>.</p></li>
<li id="id22129"><p>Alyaa M. Abdel‐Haleem, Hooman Hefzi, Katsuhiko Mineta, Xin Gao, Takashi Gojobori, Bernhard Ø. Palsson, Nathan E. Lewis, and Neema Jamshidi. Functional interrogation of plasmodium genus metabolism identifies species- and stage-specific differences in nutrient essentiality and drug targeting. 14(1):e1005895–e1005895, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005895">[DOI]</a>.</p></li>
<li id="id22132"><p>Hao Wang, Simonas Marcišauskas, Benjamín J. Sánchez, Iván Domenzain, Daniel Hermansson, Rasmus Ågren, Jens Nielsen, and Eduard J. Kerkhoven. Raven 2.0: a versatile toolbox for metabolic network reconstruction and a casestudy on &lt;i&gt;streptomyces coelicolor&lt;/i&gt;. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/321067">[DOI]</a>.</p></li>
<li id="id22146"><p>Francine Piubeli, Manuel Salvador, Montserrat Argandoña, Joaquı́n J. Nieto, Vicente Bernal, José M. Pastor, Manuel Cánovas, and Carmen Vargas. Insights into metabolic osmoadaptation of the ectoines-producer bacterium chromohalobacter salexigens through a high-quality genome scale metabolic model. 17(1):2–2, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-017-0852-0">[DOI]</a>.</p></li>
<li id="id22154"><p>Alan R. Pacheco, Mauricio Moel, and Daniel Segrè. Costless metabolic secretions as drivers of interspecies interactions in microbial ecosystems. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/300046">[DOI]</a>.</p></li>
<li id="id22157"><p>Bashir Sajo Mienda, Rabiu Salihu, Aliyu Adamu, and Idris Shehu. Genome-scale metabolic models as platforms for identification of novel genes as antimicrobial drug targets. 13(4):455–467, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.2217/fmb-2017-0195">[DOI]</a>.</p></li>
<li id="id22163"><p>Yun Zhang, Xiuling Shang, Shujuan Lai, Yu Zhang, Qitiao Hu, Xin Chai, Bo Wang, Shuwen Liu, and Tingyi Wen. Reprogramming one-carbon metabolic pathways to decouple &lt;scp&gt;l&lt;/scp&gt;-serine catabolism from cell growth in &lt;i&gt;corynebacterium glutamicum&lt;/i&gt;. 7(2):635–646, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.7b00373">[DOI]</a>.</p></li>
<li id="id22222"><p>Adam Amara, Eriko Takano, and Rainer Breitling. Development and validation of an updated computational model of streptomyces coelicolor primary and secondary metabolism. 19(1):519–519, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-018-4905-5">[DOI]</a>.</p></li>
<li id="id22223"><p>Yu Zhang, Yun Zhang, Xiuling Shang, Bo Wang, Qitiao Hu, Shuwen Liu, and Tingyi Wen. Reconstruction of tricarboxylic acid cycle in &lt;i&gt;corynebacterium glutamicum&lt;/i&gt; with a genome‐scale metabolic network model for &lt;i&gt;trans&lt;/i&gt;‐4‐hydroxyproline production. 116(1):99–109, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26818">[DOI]</a>.</p></li>
<li id="id22260"><p>Miha Kastelic, D. Kopač, Uroš Novak, and Blaž Likozar. Dynamic metabolic network modeling of mammalian chinese hamster ovary (cho) cell cultures with continuous phase kinetics transitions. 142():124–134, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2018.11.015">[DOI]</a>.</p></li>
<li id="id22269"><p>Mohammad Tajparast and Dominic Frigon. Predicting the accumulation of storage compounds by rhodococcus jostii rha1 in the feast-famine growth cycles using genome-scale flux balance analysis. 13(3):e0191835–e0191835, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0191835">[DOI]</a>.</p></li>
<li id="id22310"><p>Ilyas Kabimoldayev, Anh Duc Nguyen, Laurence Yang, Sunghoon Park, Eun Yeol Lee, and Donghyuk Kim. Basics of genome-scale metabolic modeling and applications on c1-utilization. 365(20):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsle/fny241">[DOI]</a>.</p></li>
<li id="id22350"><p>Samara Silva de Souza, Julia de Vasconcellos Castro, and Luismar Marques Porto. Modeling the core metabolism of komagataeibacter hansenii atcc 23769 to evaluate nanocellulose biosynthesis. 35(3):869–886, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1590/0104-6632.20180353s20170327">[DOI]</a>.</p></li>
<li id="id22355"><p>Benjamin M. Shapiro, Tori M. Hoehler, and Qusheng Jin. Integrating genome-scale metabolic models into the prediction of microbial kinetics in natural environments. 242():102–122, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gca.2018.08.047">[DOI]</a>.</p></li>
<li id="id22356"><p>Clémence Frioux, Torsten Schaub, Sebastian Schellhorn, Anne Siegel, and Philipp Wanko. Hybrid metabolic network completion. 19(1):83–108, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1017/s1471068418000455">[DOI]</a>.</p></li>
<li id="id22362"><p>M N Abd Latif and Elebeoba E. May. A multiscale agent-based model for the investigation of e. coli k12 metabolic response during biofilm formation. 80(11):2917–2956, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11538-018-0494-3">[DOI]</a>.</p></li>
<li id="id22368"><p>W. Pinzon, Hector Vega, Janneth González, and Andrés Pinzón. Mathematical framework behind the reconstruction and analysis of genome scale metabolic models. 26(5):1593–1606, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11831-018-9290-3">[DOI]</a>.</p></li>
<li id="id22372"><p>Ignace Tack, Philippe Nimmegeers, Simen Akkermans, Filip Logist, and Jan Van Impe. A low-complexity metabolic network model for the respiratory and fermentative metabolism of escherichia coli. 13(8):e0202565–e0202565, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0202565">[DOI]</a>.</p></li>
<li id="id22382"><p>Zeyu Zhu, Defne Surujon, Aidan Pavao, José Bento, and Tim van Opijnen. Forecasting bacterial survival-success and adaptive evolution through multi-omics stress-response mapping, network analyses and machine learning. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/387910">[DOI]</a>.</p></li>
<li id="id22413"><p>Zachary P. Metz, Tong Ding, and David J. Baumler. Using genome-scale metabolic models to compare serovars of the foodborne pathogen listeria monocytogenes. 13(6):e0198584–e0198584, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0198584">[DOI]</a>.</p></li>
<li id="id22420"><p>Nicholas A. Jose, Rebecca Lau, Tami L. Swenson, Niels Klitgord, Ferrán García‐Pichel, Benjamin P. Bowen, Richard Baran, and Trent R. Northen. Flux balance modeling to predict bacterial survival during pulsed-activity events. 15(7):2219–2229, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/bg-15-2219-2018">[DOI]</a>.</p></li>
<li id="id22422"><p>G. Mitsue León-Saiki, Narcís Ferrer-Ledo, David Lao‐Martil, Douwe van der Veen, René H. Wijffels, and Dirk E. Martens. Metabolic modelling and energy parameter estimation of tetradesmus obliquus. 35():378–387, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.algal.2018.09.008">[DOI]</a>.</p></li>
<li id="id22430"><p>Philip J. Luthert, Luís Serrano, and Christina Kiel. Opportunities and challenges of whole-cell and -tissue simulations of the outer retina in health and disease. 1(1):131–152, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1146/annurev-biodatasci-080917-013356">[DOI]</a>.</p></li>
<li id="id22440"><p>John E. Pearl, M. Das, and Andrea M. Cooper. Immunological roulette: luck or something more? considering the connections between host and environment in tb. 15(3):226–232, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/cmi.2017.145">[DOI]</a>.</p></li>
<li id="id22448"><p>Zixiang Xu, Jing Guo, Yunxia Yue, Jing Meng, and Xiao Sun. &lt;i&gt;in silico&lt;/i&gt; genome-scale reconstruction and analysis of the &lt;i&gt;shewanella loihica&lt;/i&gt; pv-4 metabolic network. 26(03):373–397, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0218339018500171">[DOI]</a>.</p></li>
<li id="id22487"><p>Daniela Calvetti and Erkki Somersalo. Brain energy metabolism. ():1–19, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4614-7320-6_100673-1">[DOI]</a>.</p></li>
<li id="id22497"><p>Tong Hao, Bin Wang, Lingxuan Zhao, Xin Feng, and Jinsheng Sun. Reconstruction and analysis of a genome-scale metabolic network for &lt;i&gt;eriocheir sinensis&lt;/i&gt; hepatopancreas. 6():79235–79244, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/access.2018.2885005">[DOI]</a>.</p></li>
<li id="id22515"><p>Hameri Tuure, Georgios Fengos, Meriç Ataman, Ljubiša Mišković, and Vassily Hatzimanikatis. Kinetic models of metabolism that consider alternative steady-state solutions of intracellular fluxes and concentrations. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/437822">[DOI]</a>.</p></li>
<li id="id22529"><p>Nilanjan Adhikari, Sk. Abdul Amin, Tarun Jha, and Achintya Saha. Application of computation in the biosynthesis of phytochemicals. ():255–276, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-812364-5.00009-2">[DOI]</a>.</p></li>
<li id="id22530"><p>David Hala, Joshua A. Cullen, Béatrice V. Hernout, and Ivan Ivanov. In silico predicted transcriptional regulatory control of steroidogenesis in spawning female fathead minnows (pimephales promelas). 455():179–190, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtbi.2018.07.020">[DOI]</a>.</p></li>
<li id="id22531"><p>Rafael S. Costa and Susana Vinga. Assessing &lt;i&gt;escherichia coli&lt;/i&gt; metabolism models and simulation approaches in phenotype predictions: validation against experimental data. 34(6):1344–1354, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.2700">[DOI]</a>.</p></li>
<li id="id22535"><p>Ana M. Untaroiu, Maureen A. Carey, Jennifer L. Güler, and Jason A. Papin. Leveraging the effects of chloroquine on resistant malaria parasites for combination therapies. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/428284">[DOI]</a>.</p></li>
<li id="id22551"><p>Xiao Pan. Scale-down of cho cell fed-batch cultures. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/444363">[DOI]</a>.</p></li>
<li id="id22583"><p>Amel Bekkar, Julien Dorier, Isaac Crespo, Anne Niknejad, Alan Bridge, and Ioannis Xénarios. Generation of molecular models and pathways. ():274–285, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781119183952.ch16">[DOI]</a>.</p></li>
<li id="id22584"><p>Ankit Gupta, Ahmad Ahmad, Dipesh Chothwe, Midhun K. Madhu, Shireesh Srivastava, and Vineet K. Sharma. Genome-scale metabolic reconstruction and metabolic versatility of an obligate methanotroph&lt;i&gt;methylococcus capsulatus&lt;/i&gt;str. bath. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/349191">[DOI]</a>.</p></li>
<li id="id22585"><p>Zixiang Xu, Qiaqing Wu, and Dunming Zhu. The enzymes that beyond non-oxidative glycolysis. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/362269">[DOI]</a>.</p></li>
<li id="id22586"><p>Graciela Mitsue León Saiki. Role of starch in tetradesmus obliquus. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/454224">[DOI]</a>.</p></li>
<li id="id22601"><p>Maria Pires Pacheco. Fast recontruction of compact context-specific network models. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.31237/osf.io/8xcwa">[DOI]</a>.</p></li>
<li id="id22646"><p>Ben M. Maoz, Anna Herland, Edward A. Fitzgerald, Thomas Grevesse, Charles Vidoudez, Alan R. Pacheco, Sean P. Sheehy, Tae‐Eun Park, Stephanie Dauth, Robert Mannix, Nikita Budnik, Kevin L. Shores, Alexander Cho, Janna Nawroth, Daniel Segrè, Bogdan Budnik, Donald E. Ingber, and Kevin Kit Parker. A linked organ-on-chip model of the human neurovascular unit reveals the metabolic coupling of endothelial and neuronal cells. 36(9):865–874, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.4226">[DOI]</a>.</p></li>
<li id="id22672"><p>Nam‐Hoon Kim, Murukarthick Jayakodi, Sang‐Choon Lee, Beom‐Soon Choi, Woojong Jang, Junki Lee, Hyun Hee Kim, Nomar Espinosa Waminal, Meiyappan Lakshmanan, Binh van Nguyen, Yun Sun Lee, Hyun‐Seung Park, Hyun Jo Koo, Jee Young Park, Sampath Perumal, Ho Jun Joh, Hana Lee, Jin‐Kyung Kim, In Seo Kim, Kyung‐Hee Kim, Lokanand Koduru, Kyo Bin Kang, Sang Hyun Sung, Yeisoo Yu, Daniel Park, Doil Choi, Eun–Young Seo, Seungill Kim, Young‐Chang Kim, Dong Yun Hyun, Youn‐Il Park, Changsoo Kim, Tae‐Ho Lee, Hyun Uk Kim, Moon‐Soo Soh, Yi Lee, Jun Gyo In, Heui‐Soo Kim, Yong‐Min Kim, Deok‐Chun Yang, Rod A. Wing, Dong‐Yup Lee, Andrew H. Paterson, and Tae‐Jin Yang. Genome and evolution of the shade‐requiring medicinal herb &lt;i&gt;panax ginseng&lt;/i&gt;. 16(11):1904–1917, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/pbi.12926">[DOI]</a>.</p></li>
<li id="id22681"><p>Xihui Xu, Raphy Zarecki, Shlomit Medina, Shany Ofaim, Xiaowei Liu, Chen Chen, Shunli Hu, Dan Brom, Daniella Gat, Seema Porob, Hanan Eizenberg, Zeev Ronen, Jiandong Jiang, and Shiri Freilich. Modeling microbial communities from atrazine contaminated soils promotes the development of biostimulation solutions. 13(2):494–508, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41396-018-0288-5">[DOI]</a>.</p></li>
<li id="id22684"><p>Colton J. Lloyd, Ali Ebrahim, Laurence Yang, Zachary A. King, Edward Catoiu, Edward J. O’Brien, Joanne K. Liu, and Bernhard Ø. Palsson. Cobrame: a computational framework for genome-scale models of metabolism and gene expression. 14(7):e1006302–e1006302, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006302">[DOI]</a>.</p></li>
<li id="id22686"><p>Ai‐Lin Liu, Carolina A. Contador, Kejing Fan, and Hon‐Ming Lam. Interaction and regulation of carbon, nitrogen, and phosphorus metabolisms in root nodules of legumes. 9():1860–1860, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2018.01860">[DOI]</a>.</p></li>
<li id="id22687"><p>Wei-Chin Ho and Jianzhi Zhang. Evolutionary adaptations to new environments generally reverse plastic phenotypic changes. 9(1):350–350, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-017-02724-5">[DOI]</a>.</p></li>
<li id="id22692"><p>Hilal Taymaz‐Nikerel, Muhammed Erkan Karabekmez, Serpil Eraslan, and Betül Kırdar. Doxorubicin induces an extensive transcriptional and metabolic rewiring in yeast cells. 8(1):13672–13672, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-31939-9">[DOI]</a>.</p></li>
<li id="id22696"><p>Miroslava Čuperlović‐Culf. Machine learning methods for analysis of metabolic data and metabolic pathway modeling. 8(1):4–4, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo8010004">[DOI]</a>.</p></li>
<li id="id22707"><p>Arturo Casini, Fangyuan Chang, Raissa Eluere, Andrew King, Eric M. Young, Quentin M. Dudley, Ashty S. Karim, Katelin Pratt, Cassandra Bristol, Anthony L. Forget, Amar Ghodasara, Robert Warden‐Rothman, Rui Gan, Alexander Cristofaro, Amin Espah Borujeni, Min‐Hyung Ryu, Jian Li, Yong-Chan Kwon, He Wang, Evangelos C. Tatsis, Carlos E. Rodríguez López, Sarah E. O’Connor, Marnix H. Medema, Michael A. Fischbach, Michael C. Jewett, Christopher A. Voigt, and D. Benjamin Gordon. A pressure test to make 10 molecules in 90 days: external evaluation of methods to engineer biology. 140(12):4302–4316, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/jacs.7b13292">[DOI]</a>.</p></li>
<li id="id22715"><p>Ludovic Cottret, Clément Frainay, Maxime Chazalviel, Floréal Cabanettes, Yoann Gloaguen, Etienne Camenen, Benjamin Merlet, Stéphanie Heux, Jean‐Charles Portais, Nathalie Poupin, Florence Vinson, and Fabien Jourdan. Metexplore: collaborative edition and exploration of metabolic networks. 46(W1):W495–W502, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gky301">[DOI]</a>.</p></li>
<li id="id22719"><p>Shashi Kant Bhatia, Ravi Kant Bhatia, Yong‐Keun Choi, Eunsung Kan, Yun‐Gon Kim, and Yung‐Hun Yang. Biotechnological potential of microbial consortia and future perspectives. 38(8):1209–1229, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07388551.2018.1471445">[DOI]</a>.</p></li>
<li id="id22720"><p>Méziane Aïte, Marie Chevallier, Clémence Frioux, Camille Trottier, Jeanne Got, María Paz Cortés, Sebastián N. Mendoza, Grégory Carrier, Olivier Dameron, Nicolas Guillaudeux, Mauricio Latorre, Nicolás Loira, Gabriel V. Markov, Alejandro Maass, and Anne Siegel. Traceability, reproducibility and wiki-exploration for “à-la-carte” reconstructions of genome-scale metabolic models. 14(5):e1006146–e1006146, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006146">[DOI]</a>.</p></li>
<li id="id22722"><p>Ilya R. Akberdin, Merlin Thompson, Richard Hamilton, Nalini Desai, Danny Alexander, Calvin A. Henard, Michael T. Guarnieri, and Marina Kalyuzhnaya. Methane utilization in methylomicrobium alcaliphilum 20zr: a systems approach. 8(1):2512–2512, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-20574-z">[DOI]</a>.</p></li>
<li id="id22729"><p>Felix Moser, Amin Espah Borujeni, Amar Ghodasara, E. S. Cameron, Yongjin Park, and Christopher A. Voigt. Dynamic control of endogenous metabolism with combinatorial logic circuits. 14(11):e8605–e8605, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20188605">[DOI]</a>.</p></li>
<li id="id22740"><p>Pierre Salvy, Georgios Fengos, Meriç Ataman, Thomas Pathier, Keng Cher Soh, and Vassily Hatzimanikatis. Pytfa and mattfa: a python package and a matlab toolbox for thermodynamics-based flux analysis. 35(1):167–169, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bty499">[DOI]</a>.</p></li>
<li id="id22745"><p>Elias Englund, Kiyan Shabestary, Elton P. Hudson, and Pia Lindberg. Systematic overexpression study to find target enzymes enhancing production of terpenes in synechocystis pcc 6803, using isoprene as a model compound. 49():164–177, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2018.07.004">[DOI]</a>.</p></li>
<li id="id22747"><p>Andrea De Martino and Daniele De Martino. An introduction to the maximum entropy approach and its application to inference problems in biology. 4(4):e00596–e00596, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.heliyon.2018.e00596">[DOI]</a>.</p></li>
<li id="id22757"><p>Robert Ramírez‐García, Nisarg Gohil, and Vijai Singh. Recent advances, challenges, and opportunities in bioremediation of hazardous materials. ():517–568, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-813912-7.00021-1">[DOI]</a>.</p></li>
<li id="id22759"><p>Osvaldo D. Kim, Miguel Rocha, and Paulo Maia. A review of dynamic modeling approaches and their application in computational strain optimization for metabolic engineering. 9():1690–1690, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2018.01690">[DOI]</a>.</p></li>
<li id="id22764"><p>George C. diCenzo, Maryam Zamani, Alice Checcucci, Marco Fondi, Joel S. Griffitts, Turlough M. Finan, and Alessio Mengoni. Multidisciplinary approaches for studying rhizobium–legume symbioses. 65(1):1–33, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1139/cjm-2018-0377">[DOI]</a>.</p></li>
<li id="id22770"><p>Donghyuk Kim, Sang Woo Seo, Ye Gao, Hojung Nam, Gabriela I. Guzmán, Byung‐Kwan Cho, and Bernhard Ø. Palsson. Systems assessment of transcriptional regulation on central carbon metabolism by cra and crp. 46(6):2901–2917, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gky069">[DOI]</a>.</p></li>
<li id="id22791"><p>Wei Zou, Guangbin Ye, and Kaizheng Zhang. Diversity, function, and application of &lt;i&gt;clostridium&lt;/i&gt; in chinese strong flavor baijiu ecosystem: a review. 83(5):1193–1199, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1750-3841.14134">[DOI]</a>.</p></li>
<li id="id22796"><p>Pranjul Mishra, Na‐Rae Lee, Meiyappan Lakshmanan, Minsuk Kim, Byung‐Gee Kim, and Dong‐Yup Lee. Genome-scale model-driven strain design for dicarboxylic acid production in yarrowia lipolytica. 12(S2):12–12, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-018-0542-5">[DOI]</a>.</p></li>
<li id="id22798"><p>Kiyan Shabestary, Josefine Anfelt, Emil Ljungqvist, Michael Jahn, Lun Yao, and Elton P. Hudson. Targeted repression of essential genes to arrest growth and increase carbon partitioning and biofuel titers in cyanobacteria. 7(7):1669–1675, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.8b00056">[DOI]</a>.</p></li>
<li id="id22799"><p>Dorinês Rosário, Rui Benfeitas, Gholamreza Bidkhori, Cheng Zhang, Mathias Uhlén, Saeed Shoaie, and Adil Mardinoğlu. Understanding the representative gut microbiota dysbiosis in metformin-treated type 2 diabetes patients using genome-scale metabolic modeling. 9():775–775, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2018.00775">[DOI]</a>.</p></li>
<li id="id22805"><p>Emil Karlsen, Christian Schulz, and Eivind Almaas. Automated generation of genome-scale metabolic draft reconstructions based on kegg. 19(1):467–467, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-018-2472-z">[DOI]</a>.</p></li>
<li id="id22817"><p>Beatriz García-Jiménez, José L. Garcı́a, and Juan Nogales. Flycop: metabolic modeling-based analysis and engineering microbial communities. 34(17):i954–i963, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bty561">[DOI]</a>.</p></li>
<li id="id22821"><p>José P. Faria, Miguel Rocha, Isabel Rocha, and Christopher S. Henry. Methods for automated genome-scale metabolic model reconstruction. 46(4):931–936, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20170246">[DOI]</a>.</p></li>
<li id="id22826"><p>Zhongbao Ma, Chao Ye, Wei‐Wei Deng, Mengmeng Xu, Qiong Wang, Gao‐Qiang Liu, Feng Wang, Li Liu, Zhenghong Xu, Guiyang Shi, and Zhongyang Ding. Reconstruction and analysis of a genome-scale metabolic model of ganoderma lucidum for improved extracellular polysaccharide production. 9():3076–3076, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2018.03076">[DOI]</a>.</p></li>
<li id="id22843"><p>L. Kyselova, Diana Kreitmayer, Andreas Kremling, and Katja Bettenbrock. Type and capacity of glucose transport influences succinate yield in two-stage cultivations. 17(1):132–132, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-018-0980-1">[DOI]</a>.</p></li>
<li id="id22862"><p>Maria Caracausi, Veronica Ghini, Chiara Locatelli, Martina Mericio, Allison Piovesan, Francesca Antonaros, Maria Chiara Pelleri, Lorenza Vitale, Rosa Anna Vacca, Federica Bedetti, Maria Chiara Mimmi, Claudio Luchinat, Paola Turano, Pierluigi Strippoli, and Guido Cocchi. Plasma and urinary metabolomic profiles of down syndrome correlate with alteration of mitochondrial metabolism. 8(1):2977–2977, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-20834-y">[DOI]</a>.</p></li>
<li id="id22875"><p>Nana Y. D. Ankrah, Bessem Chouaia, and Angela E. Douglas. The cost of metabolic interactions in symbioses between insects and bacteria with reduced genomes. 9(5):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.01433-18">[DOI]</a>.</p></li>
<li id="id22892"><p>George C. diCenzo, Alex B. Benedict, Marco Fondi, Graham C. Walker, Turlough M. Finan, Alessio Mengoni, and Joel S. Griffitts. Robustness encoded across essential and accessory replicons of the ecologically versatile bacterium sinorhizobium meliloti. 14(4):e1007357–e1007357, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pgen.1007357">[DOI]</a>.</p></li>
<li id="id22899"><p>Elliot Rowe, Bernhard Ø. Palsson, and Zachary A. King. Escher-fba: a web application for interactive flux balance analysis. 12(1):84–84, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-018-0607-5">[DOI]</a>.</p></li>
<li id="id22911"><p>Yan Zhu, Tobias Czauderna, Jinxin Zhao, Matthias Klapperstueck, Mohd Hafidz Mahamad Maifiah, Mei‐Ling Han, Jing Lü, Björn Sommer, Tony Velkov, Trevor Lithgow, Jiangning Song, Falk Schreiber, and Jian Li. Genome-scale metabolic modeling of responses to polymyxins in &lt;i&gt;pseudomonas aeruginosa&lt;/i&gt;. 7(4):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/gigascience/giy021">[DOI]</a>.</p></li>
<li id="id22924"><p>Hongzhong Lu, Weiqiang Cao, Xiaoyun Liu, Yufei Sui, Liming Ouyang, Jianye Xia, Mingzhi Huang, Yingping Zhuang, Siliang Zhang, Henk Noorman, and Ju Chu. Multi-omics integrative analysis with genome-scale metabolic model simulation reveals global cellular adaptation of aspergillus niger under industrial enzyme production condition. 8(1):14404–14404, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-32341-1">[DOI]</a>.</p></li>
<li id="id22925"><p>Elizabeth Brunk, Roger L. Chang, Jing Xia, Hooman Hefzi, James T. Yurkovich, Donghyuk Kim, Evan Buckmiller, Harris H. Wang, Byung‐Kwan Cho, Chen Yang, Bernhard Ø. Palsson, George M. Church, and Nathan E. Lewis. Characterizing posttranslational modifications in prokaryotic metabolism using a multiscale workflow. 115(43):11096–11101, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1811971115">[DOI]</a>.</p></li>
<li id="id22937"><p>Naveen Venayak, Axel von Kamp, Steffen Klamt, and Radhakrishnan Mahadevan. Move identifies metabolic valves to switch between phenotypic states. 9(1):5332–5332, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-018-07719-4">[DOI]</a>.</p></li>
<li id="id22948"><p>Anika Küken, Frederik Sommer, Liliya Yaneva-Roder, Luke C. M. Mackinder, Melanie Höhne, Stefan Geimer, Martin C. Jonikas, Michael Schroda, Mark Stitt, Zoran Nikoloski, and Tabea Mettler‐Altmann. Effects of microcompartmentation on flux distribution and metabolic pools in chlamydomonas reinhardtii chloroplasts. 7():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.37960">[DOI]</a>.</p></li>
<li id="id22949"><p>Wei Niu, Levi Kramer, Joshua Mueller, Kun Liu, and Jiantao Guo. Metabolic engineering of escherichia coli for the de novo stereospecific biosynthesis of 1,2-propanediol through lactic acid. 8():e00082–e00082, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2018.e00082">[DOI]</a>.</p></li>
<li id="id22958"><p>Kok Siong Ang, Meiyappan Lakshmanan, Na‐Rae Lee, and Dong‐Yup Lee. Metabolic modeling of microbial community interactions for health, environmental and biotechnological applications. 19(8):712–722, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.2174/1389202919666180911144055">[DOI]</a>.</p></li>
<li id="id22973"><p>Jun Feng, Jing Yang, Wenwen Yang, Jie Chen, Min Jiang, and Xiang Zou. Metabolome- and genome-scale model analyses for engineering of aureobasidium pullulans to enhance polymalic acid and malic acid production from sugarcane molasses. 11(1):94–94, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-018-1099-7">[DOI]</a>.</p></li>
<li id="id22974"><p>María Teresita Castañeda, Sebastián Núñez, Fabricio Garelli, Claudio E. Voget, and Hernán De Battista. Comprehensive analysis of a metabolic model for lipid production in rhodosporidium toruloides. 280():11–18, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2018.05.010">[DOI]</a>.</p></li>
<li id="id22984"><p>Sergio Garcia and Cong T. Trinh. Multiobjective strain design: a framework for modular cell engineering. 51():110–120, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2018.09.003">[DOI]</a>.</p></li>
<li id="id22991"><p>Sarah McGarrity, Ósk Anuforo, Haraldur Halldórsson, Andreas Bergmann, Skarphéðinn Halldórsson, Sirus Palsson, H Henriksen, Pär I. Johansson, and Óttar Rolfsson. Metabolic systems analysis of lps induced endothelial dysfunction applied to sepsis patient stratification. 8(1):6811–6811, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-25015-5">[DOI]</a>.</p></li>
<li id="id22992"><p>Ulf W. Liebal, Lars M. Blank, and Birgitta E. Ebert. Co2 to succinic acid – estimating the potential of biocatalytic routes. 7():e00075–e00075, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mec.2018.e00075">[DOI]</a>.</p></li>
<li id="id22993"><p>Markus Janasch, Johannes Asplund‐Samuelsson, Ralf Steuer, and Elton P. Hudson. Kinetic modeling of the calvin cycle identifies flux control and stable metabolomes in&lt;i&gt;synechocystis&lt;/i&gt;carbon fixation. 70(3):973–983, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/jxb/ery382">[DOI]</a>.</p></li>
<li id="id23003"><p>Kelly Botero, Silvia Restrepo, and Andrés Pinzón. A genome-scale metabolic model of potato late blight suggests a photosynthesis suppression mechanism. 19(S8):863–863, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-018-5192-x">[DOI]</a>.</p></li>
<li id="id23032"><p>Henrik Cordes, Christoph Thiel, Vanessa Baier, Lars M. Blank, and Lars Kuepfer. Integration of genome-scale metabolic networks into whole-body pbpk models shows phenotype-specific cases of drug-induced metabolic perturbation. 4(1):10–10, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-018-0048-1">[DOI]</a>.</p></li>
<li id="id23033"><p>Alex Graudenzi, Davide Maspero, Marzia Di Filippo, Marco Gnugnoli, Claudio Isella, Giancarlo Mauri, Enzo Médico, Marco Antoniotti, and C. Damiani. Integration of transcriptomic data and metabolic networks in cancer samples reveals highly significant prognostic power. 87():37–49, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbi.2018.09.010">[DOI]</a>.</p></li>
<li id="id23039"><p>Julia Tröndle, Natalia Trachtmann, Georg A. Sprenger, and Dirk Weuster‐Botz. Fed‐batch production of &lt;scp&gt;l&lt;/scp&gt;‐tryptophan from glycerol using recombinant &lt;i&gt;escherichia coli&lt;/i&gt;. 115(12):2881–2892, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26834">[DOI]</a>.</p></li>
<li id="id23047"><p>Felipe Scott, Pamela E. Wilson, Raúl Conejeros, and Vassilios S. Vassiliadis. Simulation and optimization of dynamic flux balance analysis models using an interior point method reformulation. 119():152–170, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compchemeng.2018.08.041">[DOI]</a>.</p></li>
<li id="id23048"><p>Tjaša Kumelj, Snorre Sulheim, Alexander Wentzel, and Eivind Almaas. Predicting strain engineering strategies using iks1317: a genome‐scale metabolic model of &lt;i&gt;streptomyces coelicolor&lt;/i&gt;. 14(4):e1800180–e1800180, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201800180">[DOI]</a>.</p></li>
<li id="id23053"><p>Wei Zou, Guangbin Ye, Jing Zhang, Changqing Zhao, Xingxiu Zhao, and Kaizheng Zhang. Genome-scale metabolic reconstruction and analysis for &lt;i&gt;clostridium kluyveri&lt;/i&gt;. 61(8):605–613, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1139/gen-2017-0177">[DOI]</a>.</p></li>
<li id="id23067"><p>Hanseol Kim, Sinyeon Kim, and Sung Ho Yoon. Metabolic network reconstruction and phenome analysis of the industrial microbe, escherichia coli bl21(de3). 13(9):e0204375–e0204375, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0204375">[DOI]</a>.</p></li>
<li id="id23068"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Jorge M. Arevalillo, Mariana Díaz‐Almirón, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Hilario Navarro, Rosa Aras‐López, Irene Dapía, Rocío López‐Vacas, Paolo Nanni, Sara Llorente-Armijo, Pedro Arias, Alberto M. Borobia, Paloma Maı́n, Jaime Feliú, Enrique Espinosa, and Juan Ángel Fresno Vara. Molecular characterization of breast cancer cell response to metabolic drugs. 9(11):9645–9660, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.18632/oncotarget.24047">[DOI]</a>.</p></li>
<li id="id23075"><p>Ayca Cankorur‐Cetinkaya, Nathalie Narraidoo, Ceyda Kasavi, Nigel K.H. Slater, David B. Archer, and Stephen G. Oliver. Process development for the continuous production of heterologous proteins by the industrial yeast, &lt;i&gt;komagataella phaffii&lt;/i&gt;. 115(12):2962–2973, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26846">[DOI]</a>.</p></li>
<li id="id23080"><p>Freyr Jóhannsson, Steinn Guðmundsson, Giuseppe Paglia, Sveinn Guðmundsson, Bernhard Ø. Palsson, Ólafur E. Sigurjónsson, and Óttar Rolfsson. Systems analysis of metabolism in platelet concentrates during storage in platelet additive solution. 475(13):2225–2240, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bcj20170921">[DOI]</a>.</p></li>
<li id="id23089"><p>León F. Toro, Laura Pinilla, Claudio Avignone–Rossa, and Rigoberto Ríos‐Estepa. An enhanced genome-scale metabolic reconstruction of streptomyces clavuligerus identifies novel strain improvement strategies. 41(5):657–669, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-018-1900-9">[DOI]</a>.</p></li>
<li id="id23102"><p>Howard Ramírez-Malule, Stefan Junne, Mariano Nicolás Cruz Bournazou, Peter Neubauer, and Rigoberto Ríos‐Estepa. Streptomyces clavuligerus shows a strong association between tca cycle intermediate accumulation and clavulanic acid biosynthesis. 102(9):4009–4023, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-018-8841-8">[DOI]</a>.</p></li>
<li id="id23109"><p>Jahir M. Gutierrez, Amir Feizi, Shangzhong Li, Thomas Beuchert Kallehauge, Hooman Hefzi, Lise Marie Grav, Daniel Ley, Deniz Baycın Hızal, Michael J. Betenbaugh, Björn G. Voldborg, Helene Faustrup Kildegaard, Gyun Min Lee, Bernhard Ø. Palsson, Jens Nielsen, and Nathan E. Lewis. Genome-scale reconstructions of the mammalian secretory pathway predict metabolic costs and limitations of protein secretion. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/351387">[DOI]</a>.</p></li>
<li id="id23143"><p>Kento Tokuyama, Yoshihiro Toya, Takaaki Horinouchi, Chikara Furusawa, Fumio Matsuda, and Hiroshi Shimizu. Application of adaptive laboratory evolution to overcome a flux limitation in an &lt;i&gt;escherichia coli&lt;/i&gt; production strain. 115(6):1542–1551, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26568">[DOI]</a>.</p></li>
<li id="id23160"><p>Annalisa Occhipinti, Filmon Eyassu, Thahira Rahman, Pattanathu Rahman, and Claudio Angione. In silico engineering of &lt;i&gt;pseudomonas&lt;/i&gt; metabolism reveals new biomarkers for increased biosurfactant production. 6():e6046–e6046, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.7717/peerj.6046">[DOI]</a>.</p></li>
<li id="id23164"><p>George C. diCenzo, Klaudia Dębiec-Andrzejewska, Jan Krzysztoforski, Witold Uhrynowski, Alessio Mengoni, Camilla Fagorzi, Adrian Gόrecki, Łukasz Dziewit, Tomasz Bajda, Grzegorz Rzepa, and Łukasz Drewniak. Genomic and biotechnological characterization of the heavy-metal resistant, arsenic-oxidizing bacterium ensifer sp. m14. 9(8):379–379, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/genes9080379">[DOI]</a>.</p></li>
<li id="id23165"><p>Iñigo Apaolaza, Luis V. Valcárcel, and Francisco J. Planes. &lt;i&gt;gmcs&lt;/i&gt;: fast computation of genetic minimal cut sets in large networks. 35(3):535–537, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bty656">[DOI]</a>.</p></li>
<li id="id23172"><p>José Aguilar-Rodríguez and Andreas Wagner. Metabolic determinants of enzyme evolution in a genome-scale bacterial metabolic network. 10(11):3076–3088, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/gbe/evy234">[DOI]</a>.</p></li>
<li id="id23194"><p>Amornpan Klanchui, Sudarat Dulsawat, Kullapat Chaloemngam, Supapon Cheevadhanarak, Peerada Prommeenate, and Asawin Meechai. An improved genome-scale metabolic model of arthrospira platensis c1 (iak888) and its application in glycogen overproduction. 8(4):84–84, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo8040084">[DOI]</a>.</p></li>
<li id="id23207"><p>Rienk A. Rienksma, Peter J. Schaap, Vítor A. P. Martins dos Santos, and María Suárez‐Diez. Modeling the metabolic state of mycobacterium tuberculosis upon infection. 8():264–264, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcimb.2018.00264">[DOI]</a>.</p></li>
<li id="id23211"><p>Nan Xu, Chao Ye, and Li Liu. Genome-scale biological models for industrial microbial systems. 102(8):3439–3451, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-018-8803-1">[DOI]</a>.</p></li>
<li id="id23220"><p>Carolina Shene, Juan A. Asenjo, and Yusuf Chisti. Metabolic modelling and simulation of the light and dark metabolism of &lt;i&gt;chlamydomonas reinhardtii&lt;/i&gt;. 96(5):1076–1088, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/tpj.14078">[DOI]</a>.</p></li>
<li id="id23228"><p>Matthew Perisin and Christian Sund. Human gut microbe co-cultures have greater potential than monocultures for food waste remediation to commodity chemicals. 8(1):15594–15594, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-018-33733-z">[DOI]</a>.</p></li>
<li id="id23241"><p>Junhua Wang, Cheng Wang, Huanhuan Liu, Haishan Qi, Hong Chen, and Jianping Wen. Metabolomics assisted metabolic network modeling and network wide analysis of metabolites in microbiology. 38(7):1106–1120, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07388551.2018.1462141">[DOI]</a>.</p></li>
<li id="id23242"><p>Mustafa Sertbaş and Kutlu Ö. Ülgen. Unlocking human brain metabolism by genome-scale and multiomics metabolic models: relevance for neurology research, health, and disease. 22(7):455–467, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/omi.2018.0088">[DOI]</a>.</p></li>
<li id="id23243"><p>George C. diCenzo, Alessio Mengoni, and Marco Fondi. Tn-core: a toolbox for integrating tn-seq gene essentiality data and constraint-based metabolic modeling. 8(1):158–169, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.8b00432">[DOI]</a>.</p></li>
<li id="id23259"><p>David Botero, Iván Valdés, M Antonio Jesús Rodríguez, Diana Henao, Giovanna Danies, Andrés González, and Silvia Restrepo. A genome-scale metabolic reconstruction of phytophthora infestans with the integration of transcriptional data reveals the key metabolic patterns involved in the interaction of its host. 9():244–244, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2018.00244">[DOI]</a>.</p></li>
<li id="id23261"><p>Takeyuki Tamura. Grid-based computational methods for the design of constraint-based parsimonious chemical reaction networks to simulate metabolite production: gridprod. 19(1):325–325, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-018-2352-6">[DOI]</a>.</p></li>
<li id="id23267"><p>Yazdan Asgari, Pegah Khosravi, Zahra Zabihinpour, and Mahnaz Habibi. Exploring candidate biomarkers for lung and prostate cancers using gene expression and flux variability analysis. 10(2):113–120, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c7ib00135e">[DOI]</a>.</p></li>
<li id="id23268"><p>Selva Rupa Christinal Immanuel, Deepanwita Banerjee, Mayooreshwar P. Rajankar, and Anu Raghunathan. Integrated constraints based analysis of an engineered violacein pathway in escherichia coli. 171():10–19, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2018.06.002">[DOI]</a>.</p></li>
<li id="id23278"><p>José Ramón Valverde, Sonia Gullón, and Rafael P. Mellado. Modelling the metabolism of protein secretion through the tat route in streptomyces lividans. 18(1):59–59, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12866-018-1199-3">[DOI]</a>.</p></li>
<li id="id23279"><p>Jian Li, Renliang Sun, Xinjuan Ning, Xinran Wang, and Zhuo Wang. Genome-scale metabolic model of actinosynnema pretiosum atcc 31280 and its application for ansamitocin p-3 production improvement. 9(7):364–364, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/genes9070364">[DOI]</a>.</p></li>
<li id="id23280"><p>Swagatika Sahoo, R. Ranjith Kumar, Brandon Nicolay, Omkar S. Mohite, Karthikeyan Sivaraman, Vikas Khetan, Pukhraj Rishi, Suganeswari Ganesan, Krishnakumar Subramanyan, Karthik Raman, Wayne Miles, and Sailaja Elchuri. Metabolite systems profiling identifies exploitable weaknesses in retinoblastoma. 593(1):23–41, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/1873-3468.13294">[DOI]</a>.</p></li>
<li id="id23337"><p>Jason J. Flowers, Matthew A. Richards, Nitin S. Baliga, Birte Meyer, and David A. Stahl. Constraint‐based modelling captures the metabolic versatility of &lt;i&gt;desulfovibrio vulgaris&lt;/i&gt;. 10(2):190–201, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1758-2229.12619">[DOI]</a>.</p></li>
<li id="id23355"><p>Kristopher A. Hunt, Ryan M. Jennings, William P. Inskeep, and Ross P. Carlson. Multiscale analysis of autotroph-heterotroph interactions in a high-temperature microbial community. 14(9):e1006431–e1006431, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006431">[DOI]</a>.</p></li>
<li id="id23362"><p>Ambarish Nag, Peter C. St. John, Michael F. Crowley, and Yannick J. Bomble. Prediction of reaction knockouts to maximize succinate production by actinobacillus succinogenes. 13(1):e0189144–e0189144, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0189144">[DOI]</a>.</p></li>
<li id="id23363"><p>Erick Armingol, Eduardo Tobar-Calfucoy, and Ricardo Cabrera. Understanding the impact of the cofactor swapping of isocitrate dehydrogenase over the growth phenotype of escherichia coli on acetate by using constraint-based modeling. 13(4):e0196182–e0196182, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0196182">[DOI]</a>.</p></li>
<li id="id23364"><p>Sailaja Elchuri, Swetha Rajasekaran, and Wayne Miles. Rna-sequencing of primary retinoblastoma tumors provides new insights and challenges into tumor development. 9():170–170, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2018.00170">[DOI]</a>.</p></li>
<li id="id23365"><p>Muhammad Azharuddin Arif, Mohd Saberi Mohamad, Muhammad Shafie Abd Latif, Safaai Deris, Muhammad Akmal Remli, Kauthar Mohd Daud, Zuwairie Ibrahim, Sigeru Omatu, and Juan M. Corchado. A hybrid of cuckoo search and minimization of metabolic adjustment to optimize metabolites production in genome-scale models. 102():112–119, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2018.09.015">[DOI]</a>.</p></li>
<li id="id23366"><p>Lokanand Koduru, Meiyappan Lakshmanan, and Dong‐Yup Lee. In silico model-guided identification of transcriptional regulator targets for efficient strain design. 17(1):167–167, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-018-1015-7">[DOI]</a>.</p></li>
<li id="id23367"><p>David A. Collins and Marina Kalyuzhnaya. Navigating methane metabolism: enzymes, compartments, and networks. 613():349–383, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.mie.2018.10.010">[DOI]</a>.</p></li>
<li id="id23368"><p>Agnieszka B. Wegrzyn, Sarah Stolle, Rienk A. Rienksma, Vítor A. P. Martins dos Santos, Barbara M. Bakker, and María Suárez‐Diez. Cofactors revisited – predicting the impact of flavoprotein-related diseases on a genome scale. 1865(2):360–370, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bbadis.2018.10.021">[DOI]</a>.</p></li>
<li id="id23373"><p>Santhust Kumar, Saurabh Mahajan, and Sanjay Jain. Feedbacks from the metabolic network to the genetic network reveal regulatory modules in e. coli and b. subtilis. 13(10):e0203311–e0203311, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0203311">[DOI]</a>.</p></li>
<li id="id23387"><p>Feiran Li, Wei Xie, Qianqian Yuan, Hao Luo, Peishun Li, Tao Chen, Xueming Zhao, Zhiwen Wang, and Hongwu Ma. Genome-scale metabolic model analysis indicates low energy production efficiency in marine ammonia-oxidizing archaea. 8(1):106–106, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13568-018-0635-y">[DOI]</a>.</p></li>
<li id="id23388"><p>Cheng Zhang, Gholamreza Bidkhori, Rui Benfeitas, Sunjae Lee, Muhammad Arif, Mathias Uhlén, and Adil Mardinoğlu. Ess: a tool for genome-scale quantification of essentiality score for reaction/genes in constraint-based modeling. 9():1355–1355, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2018.01355">[DOI]</a>.</p></li>
<li id="id23394"><p>Zhendong Zhuang, Mingzhi Huang, and Ju Chu. In silico reconstruction and experimental validation of saccharopolyspora erythraea genome-scale metabolic model izz1342 that accounts for 1685 orfs. 5(1):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40643-018-0212-x">[DOI]</a>.</p></li>
<li id="id23396"><p>Ann Marie Reinhold, Geoffrey C. Poole, Clemente Izurieta, Ashley M. Helton, and Emily S. Bernhardt. Constraint-based simulation of multiple interactive elemental cycles in biogeochemical systems. 50():102–121, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ecoinf.2018.12.008">[DOI]</a>.</p></li>
<li id="id23403"><p>Gaoyang Li, Huansheng Cao, and Ying Xu. Structural and functional analyses of microbial metabolic networks reveal novel insights into genome-scale metabolic fluxes. 20(4):1590–1603, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bby022">[DOI]</a>.</p></li>
<li id="id23413"><p>Alex Graudenzi, Davide Maspero, and C. Damiani. Modeling spatio-temporal dynamics of metabolic networks with cellular automata and constraint-based methods. ():16–29, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-99813-8_2">[DOI]</a>.</p></li>
<li id="id23414"><p>Do‐Gyun Kim, Sung-Won Seo, Byoung‐Kwan Cho, Santosh Lohumi, Soon-Jung Hong, and Wang‐Hee Lee. Review of current approaches for implementing metabolic reconstruction. 43(1):45–58, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.5307/jbe.2018.43.1.045">[DOI]</a>.</p></li>
<li id="id23430"><p>Emrah Özcan and Tunahan Çakır. Genome-scale brain metabolic networks as scaffolds for the systems biology of neurodegenerative diseases: mapping metabolic alterations. 21():195–217, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-94593-4_7">[DOI]</a>.</p></li>
<li id="id23443"><p>Wei Zou, Xianghua Xiong, Jing Zhang, Kaizheng Zhang, Xingxiu Zhao, and Changqing Zhao. Reconstruction and analysis of a genome-scale metabolic model of methylovorus sp. mp688, a high-level pyrroloquinolone quinone producer. 172():37–42, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2018.07.009">[DOI]</a>.</p></li>
<li id="id23470"><p>Benjamín J. Sánchez, Feiran Li, Eduard J. Kerkhoven, and Jens Nielsen. Slimer: probing flexibility of lipid metabolism in yeast with an improved constraint-based modeling framework. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/324863">[DOI]</a>.</p></li>
<li id="id23471"><p>Hadi Nazem‐Bokaee and Costas D. Maranas. A prospective study on the fermentation landscape of gaseous substrates to biorenewables using methanosarcina acetivorans metabolic model. 9():1855–1855, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2018.01855">[DOI]</a>.</p></li>
<li id="id23481"><p>David Ando and Héctor García Martín. Genome-scale 13c fluxomics modeling for metabolic engineering of saccharomyces cerevisiae. 1859():317–345, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-8757-3_19">[DOI]</a>.</p></li>
<li id="id23488"><p>Meghan Thommes, Taiyao Wang, Qi Zhao, Ioannis Ch. Paschalidis, and Daniel Segrè. Designing metabolic division of labor in microbial communities. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/442376">[DOI]</a>.</p></li>
<li id="id23499"><p>Elliot Rowe, Bernhard Ø. Palsson, and Zachary A. King. Escher-fba: a web application for interactive flux balance analysis. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/281600">[DOI]</a>.</p></li>
<li id="id23512"><p>Chiara Damiani, Davide Maspero, Marzia Di Filippo, Riccardo Colombo, Dario Pescini, Alex Graudenzi, Hans V. Westerhoff, Lilia Alberghina, Marco Vanoni, and Giancarlo Mauri. Integration of single-cell rna-seq data into metabolic models to characterize tumour cell populations. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/256644">[DOI]</a>.</p></li>
<li id="id23513"><p>Ilya R. Akberdin, Merlin Thompson, and Marina Kalyuzhnaya. Systems biology and metabolic modeling of c1-metabolism. ():99–115, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-74866-5_7">[DOI]</a>.</p></li>
<li id="id23514"><p>Igor Marín de Mas. Multiomic data integration and analysis via model-driven approaches. ():447–476, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.coac.2018.07.005">[DOI]</a>.</p></li>
<li id="id23519"><p>Neeraj Sinha, María Suárez‐Diez, Guido Hooiveld, Jaap Keijer, Vitor Martin dos Santos, and Evert M. van Schothorst. A constraint-based model analysis of enterocyte mitochondrial adaptation to dietary interventions of lipid type and lipid load. 9():749–749, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2018.00749">[DOI]</a>.</p></li>
<li id="id23520"><p>Anna S. Blazier and Jason A. Papin. Reconciling high-throughput gene essentiality data with metabolic network reconstructions. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/415448">[DOI]</a>.</p></li>
<li id="id23534"><p>Nicolas Sompairac, Jennifer Modamio, Emmanuel Barillot, Ronan M. T. Fleming, Andreï Zinovyev, and Inna Kuperstein. Metabolic and signalling network map integration: application to cross-talk studies and omics data analysis in cancer. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/274902">[DOI]</a>.</p></li>
<li id="id23542"><p>Alessio Mancini, Filmon Eyassu, Maxwell Conway, Annalisa Occhipinti, Píetro Lió, Claudio Angione, and Sandra Pucciarelli. Ciliategem: an open-project and a tool for predictions of ciliate metabolic variations and experimental condition design. 19(S15):442–442, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-018-2422-9">[DOI]</a>.</p></li>
<li id="id23553"><p>Sabina Żołędowska, Luana Presta, Marco Fondi, Francesca Decorosi, Luciana Giovannetti, Alessio Mengoni, and Ewa Łojkowska. Metabolic modeling of &lt;i&gt;pectobacterium parmentieri&lt;/i&gt; scc3193 provides insights into metabolic pathways of plant pathogenic bacteria. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/284968">[DOI]</a>.</p></li>
<li id="id23561"><p>Elizabeth Brunk, Roger L. Chang, Jing Xia, Hooman Hefzi, James T. Yurkovich, Donghyuk Kim, Evan Buckmiller, Harris H. Wang, Yang Chen, Bernhard Ø. Palsson, George M. Church, and Nathan E. Lewis. Systemic post-translational control of bacterial metabolism regulates adaptation in dynamic environments. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.2139/ssrn.3155551">[DOI]</a>.</p></li>
<li id="id23581"><p>Tahereh Ghasemi-Kahrizsangi, Sayed‐Amir Marashi, and Zhaleh Hosseini. Genome-scale metabolic network models of bacillus species suggest that model improvement is necessary for biotechnological applications. 16(3):164–172, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.21859/ijb.1684">[DOI]</a>.</p></li>
<li id="id23594"><p>David B. Bernstein, Floyd E. Dewhirst, and Daniel Segrè. Quantifying biosynthetic network robustness across the human oral microbiome. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/392621">[DOI]</a>.</p></li>
<li id="id23595"><p>Cecilia Noecker, Hsuan-Chao Chiu, Colin P McNally, and Elhanan Borenstein. Defining and evaluating microbial contributions to metabolite variation in microbiome-metabolome association studies. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/402040">[DOI]</a>.</p></li>
<li id="id23616"><p>Miha Moškon, Nikolaj Zimic, and Miha Mraz. Grohar: automated visualization of genome-scale metabolic models and their pathways. 25(5):505–508, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2017.0209">[DOI]</a>.</p></li>
<li id="id23617"><p>Tahereh Ghasemi-Kahrizsangi, Sayed‐Amir Marashi, and Zhaleh Hosseini. Genome-scale metabolic network models of bacillus species suggest that model improvement is necessary for biotechnological applications. 16(3):e1684–e1684, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.15171/ijb.1684">[DOI]</a>.</p></li>
<li id="id23643"><p>Laura J. Dunphy, P. H. Yen, and Jason A. Papin. Network analysis reveals differential metabolic functionality in antibiotic-resistant&lt;i&gt;pseudomonas aeruginosa&lt;/i&gt;. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/303289">[DOI]</a>.</p></li>
<li id="id23644"><p>Meghana V. Palukuri, Shruti Shivakumar, Swagatika Sahoo, and Raghunathan Rengaswamy. Computational framework for exploring the interplay of diet and gut microbiota in autism. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/422931">[DOI]</a>.</p></li>
<li id="id23646"><p>Jeanne M. O. Eloundou-Mbebi, Anika Küken, Georg Basler, and Zoran Nikoloski. Cellular determinants of metabolite concentration ranges. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/439232">[DOI]</a>.</p></li>
<li id="id23663"><p>Agnieszka B. Wegrzyn, Sarah Stolle, Rienk A. Rienksma, Vítor A. P. Martins dos Santos, Barbara M. Bakker, and María Suárez‐Diez. Cofactors revisited – predicting the impact of flavoprotein-related diseases on a genome scale. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/343624">[DOI]</a>.</p></li>
<li id="id23664"><p>Timothy H. Self, David Gilbert, and Monika Heiner. Derivation of a biomass proxy for dynamic analysis of whole genome metabolic models. ():39–58, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-99429-1_3">[DOI]</a>.</p></li>
<li id="id23673"><p>Meiyappan Lakshmanan, Lokanand Koduru, and Dong‐Yup Lee. Software applications for phenotype analysis and strain design of cellular systems. ():771–791, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527803293.ch44">[DOI]</a>.</p></li>
<li id="id23688"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Jorge M. Arevalillo, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Mariana Díaz‐Almirón, Hilario Navarro, Paloma Maı́n, Enrique Espinosa, Pilar Zamora, and Juan Ángel Fresno Vara. Probabilistic graphical models and computational metabolic models applied to the analysis of metabolomics data in breast cancer. ():370221, 2018. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id23754"><p>Abinaya Badri, Karthik Raman, and Guhan Jayaraman. Uncovering novel pathways for enhancing hyaluronan synthesis in recombinant&lt;i&gt;lactococcus lactis&lt;/i&gt;: genome-scale metabolic modelling and experimental validation. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/244699">[DOI]</a>.</p></li>
<li id="id23757"><p>Kyle A. Stone. Exploring phenotypes of a promising industrial methanotroph: an experimental and computational study. ():, 2018. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id23758"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Mariana Díaz‐Almirón, Jorge M. Arevalillo, María Ferrer-Gómez, Hilario Navarro, Paloma Maı́n, Enrique Espinosa, Álvaro Pinto, and Juan Ángel Fresno Vara. Novel molecular classification of muscle-invasive bladder cancer opens new treatment opportunities. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/327114">[DOI]</a>.</p></li>
<li id="id23759"><p>Sergio Garcia and Cong T. Trinh. Multiobjective strain design: a framework for modular cell engineering. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/349399">[DOI]</a>.</p></li>
<li id="id23760"><p>James P. Gilbert, Nicole Pearcy, Rupert Norman, Thomas Millat, Klaus Winzer, John R. King, Charlie Hodgman, Nigel P. Minton, and Jamie Twycross. Gsmodutils: a python based framework for test-driven genome scale metabolic model development. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/430116">[DOI]</a>.</p></li>
<li id="id23761"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Mariana Díaz‐Almirón, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Rocío López‐Vacas, Paolo Nanni, Pilar Zamora, Enrique Espinosa, and Juan Ángel Fresno Vara. Computational metabolism modeling predicts risk of distant relapse-free survival in breast cancer patients. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/468595">[DOI]</a>.</p></li>
<li id="id23762"><p>Tuo Zang. Quantitative characterization of paediatric burn blister fluid. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.5204/thesis.eprints.122968">[DOI]</a>.</p></li>
<li id="id23773"><p>Clémence Frioux, Torsten Schaub, Sebastian Schellhorn, Anne Siegel, and Philipp Wanko. Hybrid metabolic network completion. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1808.04149">[DOI]</a>.</p></li>
<li id="id23786"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Elena López‐Camacho, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Rocío López‐Vacas, Jorge M. Arevalillo, Mariana Díaz‐Almirón, Hilario Navarro, Paloma Maı́n, Enrique Espinosa, Pilar Zamora, and Juan Ángel Fresno Vara. Computational metabolomics hints at the relevance of glutamine metabolism in breast cancer. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/370221">[DOI]</a>.</p></li>
<li id="id23791"><p>Villayat Ali, Priyanka Singh, Sonu Kumar Gupta, Yogesh Mauley, and Malkhey Verma. Metabolic flux analysis in plant metabolic network. ():, 2018. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id23801"><p>Annalisa Occhipinti, Filmon Eyassu, Thahira Rahman, Shafiqur Rahman, Claudio Corresp, Claudio Angione, and Pattanathu Rahman. Peer review #1 of &quot;in silico engineering of pseudomonas metabolism reveals new biomarkers for increased biosurfactant production (v0.2)&quot;. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6046v0.2/reviews/1">[DOI]</a>.</p></li>
<li id="id23802"><p>Annalisa Occhipinti, Filmon Eyassu, Thahira Rahman, Syed Asad Rahman, Claudio Corresp, Claudio Angione, and Pattanathu Rahman. Peer review #2 of &quot;in silico engineering of pseudomonas metabolism reveals new biomarkers for increased biosurfactant production (v0.1)&quot;. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6046v0.1/reviews/2">[DOI]</a>.</p></li>
<li id="id23814"><p>Jean‐Marc Schwartz and Zita Soons. Metabolic models. ():438–444, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-809633-8.20255-0">[DOI]</a>.</p></li>
<li id="id23815"><p>Annalisa Occhipinti, Filmon Eyassu, Thahira Rahman, Shafiqur Rahman, Claudio Corresp, Claudio Angione, and Pattanathu Rahman. Peer review #1 of &quot;in silico engineering of pseudomonas metabolism reveals new biomarkers for increased biosurfactant production (v0.1)&quot;. ():, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.7287/peerj.6046v0.1/reviews/1">[DOI]</a>.</p></li>
<li id="id23858"><p>Elizabeth Brunk, Swagatika Sahoo, Daniel C. Zielinski, Ali Altunkaya, Andreas Dräger, Nathan Mih, Francesco Gatto, Avlant Nilsson, German Preciat, Maike K. Aurich, Andreas Prlić, Anand V. Sastry, Anna Dröfn Daníelsdóttir, Almut Heinken, Alberto Noronha, Peter W. Rose, S.K. Burley, Ronan M. T. Fleming, Jens Nielsen, Ines Thiele, and Bernhard Ø. Palsson. Recon3d enables a three-dimensional view of gene variation in human metabolism. 36(3):272–281, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.4072">[DOI]</a>.</p></li>
<li id="id23863"><p>Alberto Noronha, Jennifer Modamio, Yohan Jarosz, Elisabeth Guerard, Nicolas Sompairac, German Preciat, Anna Dröfn Daníelsdóttir, Max Krecké, Diane Merten, Hulda S. Haraldsdóttir, Almut Heinken, Laurent Heirendt, Stefanía Magnúsdóttir, Dmitry A. Ravcheev, Swagatika Sahoo, Piotr Gawron, Lucia Friscioni, Beatriz García Santa Cruz, Mabel Prendergast, Alberto Puente, Mariana Figueiredo Rodrigues, Akansha Roy, Mouss Rouquaya, Luca Wiltgen, Alise Žagare, Elisabeth John, Maren Krueger, Inna Kuperstein, Andreï Zinovyev, Reinhard Schneider, Ronan M. T. Fleming, and Ines Thiele. The virtual metabolic human database: integrating human and gut microbiome metabolism with nutrition and disease. 47(D1):D614–D624, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gky992">[DOI]</a>.</p></li>
<li id="id23892"><p>Federico Baldini, Almut Heinken, Laurent Heirendt, Stefanía Magnúsdóttir, Ronan M. T. Fleming, and Ines Thiele. The microbiome modeling toolbox: from microbial interactions to personalized microbial communities. 35(13):2332–2334, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bty941">[DOI]</a>.</p></li>
<li id="id23904"><p>Eugen Bauer and Ines Thiele. From network analysis to functional metabolic modeling of the human gut microbiota. 3(3):, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00209-17">[DOI]</a>.</p></li>
<li id="id23917"><p>Marek Ostaszewski, Stephan Gebel, Inna Kuperstein, Alexander Mazein, Andreï Zinovyev, Uğur Doğrusöz, Jan Hasenauer, Ronan M. T. Fleming, Nicolas Le Novère, Piotr Gawron, Thomas Ligon, Anna Niarakis, David Nickerson, Daniel Weindl, Rudi Balling, Emmanuel Barillot, Charles Auffray, and Reinhard Schneider. Community-driven roadmap for integrated disease maps. 20(2):659–670, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bby024">[DOI]</a>.</p></li>
<li id="id24066"><p>Michaela Graf, Julia Zieringer, Thorsten Haas, Alexander Nieß, Bastian Blombach, and Ralf Takors. Physiological response of corynebacterium glutamicum to increasingly nutrient-rich growth conditions. 9():2058–2058, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2018.02058">[DOI]</a>.</p></li>
<li id="id24350"><p>Van Du T. Tran, Sébastien Moretti, Alix T. Coste, Sara Amorim-Vaz, Dominique Sanglard, and Marco Pagni. Condition-specific series of metabolic sub-networks and its application for gene set enrichment analysis. 35(13):2258–2266, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bty929">[DOI]</a>.</p></li>
<li id="id24516"><p>James T. Yurkovich, Miguel A. Alcantar, Zachary B. Haiman, and Bernhard Ø. Palsson. Network-level allosteric effects are elucidated by detailing how ligand-binding events modulate utilization of catalytic potentials. 14(8):e1006356–e1006356, 2018. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1006356">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-8" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-8" name="0-8" role="tabpanel" tabindex="0"><div class="docutils container" id="id24969">
<ol class="arabic simple" start="1">
<li id="id25029"><p>Jennifer Bartell, Anna S. Blazier, P. H. Yen, Juliane Charlotte Thøgersen, Lars Jelsbak, Joanna B. Goldberg, and Jason A. Papin. Reconstruction of the metabolic network of pseudomonas aeruginosa to interrogate virulence factor synthesis. 8(1):14631–14631, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms14631">[DOI]</a>.</p></li>
<li id="id25030"><p>Ali R. Zomorrodi and Daniel Segrè. Genome-driven evolutionary game theory helps understand the rise of metabolic interdependencies in microbial communities. 8(1):1563–1563, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-017-01407-5">[DOI]</a>.</p></li>
<li id="id25042"><p>Xiao Pan, Ciska Dalm, René H. Wijffels, and Dirk E. Martens. Metabolic characterization of a cho cell size increase phase in fed-batch cultures. 101(22):8101–8113, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-017-8531-y">[DOI]</a>.</p></li>
<li id="id25045"><p>Farhana R. Pinu and Silas G. Villas‐Bôas. Extracellular microbial metabolomics: the state of the art. 7(3):43–43, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo7030043">[DOI]</a>.</p></li>
<li id="id25050"><p>D. Joseph Sexton and Martín Schuster. Nutrient limitation determines the fitness of cheaters in bacterial siderophore cooperation. 8(1):230–230, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41467-017-00222-2">[DOI]</a>.</p></li>
<li id="id25065"><p>Troy E. Sandberg, Colton J. Lloyd, Bernhard Ø. Palsson, and Adam M. Feist. Laboratory evolution to alternating substrate environments yields distinct phenotypic and genetic adaptive strategies. 83(13):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.00410-17">[DOI]</a>.</p></li>
<li id="id25073"><p>Yu Zhang, Jingyi Cai, Xiuling Shang, Bo Wang, Shuwen Liu, Xin Chai, Tianwei Tan, Yun Zhang, and Tingyi Wen. A new genome-scale metabolic model of corynebacterium glutamicum and its application. 10(1):169–169, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-017-0856-3">[DOI]</a>.</p></li>
<li id="id25084"><p>Christopher P. Long, Jacqueline E. Gonzalez, Robert M. Cipolla, and Maciek R. Antoniewicz. Metabolism of the fast-growing bacterium vibrio natriegens elucidated by 13c metabolic flux analysis. 44():191–197, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.10.008">[DOI]</a>.</p></li>
<li id="id25097"><p>Andreas Wagner. Information theory, evolutionary innovations and evolvability. 372(1735):20160416–20160416, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rstb.2016.0416">[DOI]</a>.</p></li>
<li id="id25165"><p>Francisco Saitua, Paulina Torres, José Ricardo Pérez‐Correa, and Eduardo Agosín. Dynamic genome-scale metabolic modeling of the yeast pichia pastoris. 11(1):27–27, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0408-2">[DOI]</a>.</p></li>
<li id="id25170"><p>Nicolás Loira, Sebastián N. Mendoza, María Paz Cortés, Natalia Palacios‐Rojas, Dante Travisany, Alex Di Genova, Natalia Gajardo, Nicole Ehrenfeld, and Alejandro Maass. Reconstruction of the microalga nannochloropsis salina genome-scale metabolic model with applications to lipid production. 11(1):66–66, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0441-1">[DOI]</a>.</p></li>
<li id="id25185"><p>Homa Majd, Martin King, Anthony C. Smith, and Edmund R.S. Kunji. Pathogenic mutations of the human mitochondrial citrate carrier slc25a1 lead to impaired citrate export required for lipid, dolichol, ubiquinone and sterol synthesis. 1859(1):1–7, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bbabio.2017.10.002">[DOI]</a>.</p></li>
<li id="id25210"><p>Ab Rauf Shah, Ahmad Ahmad, Shireesh Srivastava, and B. M. Jaffar Ali. Reconstruction and analysis of a genome-scale metabolic model of nannochloropsis gaditana. 26():354–364, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.algal.2017.08.014">[DOI]</a>.</p></li>
<li id="id25221"><p>Maureen A. Carey, Jason A. Papin, and Jennifer L. Güler. Novel plasmodium falciparum metabolic network reconstruction identifies shifts associated with clinical antimalarial resistance. 18(1):543–543, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-017-3905-1">[DOI]</a>.</p></li>
<li id="id25245"><p>Mahesh C. Sharma, Naeem Shaikh, Shailendra Yadav, Sushma Singh, and Prabha Garg. A systematic reconstruction and constraint-based analysis of leishmania donovani metabolic network: identification of potential antileishmanial drug targets. 13(5):955–969, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c6mb00823b">[DOI]</a>.</p></li>
<li id="id25258"><p>R. Adam Thompson and Cong T. Trinh. Overflow metabolism and growth cessation in &lt;i&gt;clostridium thermocellum&lt;/i&gt; dsm1313 during high cellulose loading fermentations. 114(11):2592–2604, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26374">[DOI]</a>.</p></li>
<li id="id25261"><p>Lauren T. Cordova, Robert M. Cipolla, Adti Swarup, Christopher P. Long, and Maciek R. Antoniewicz. 13c metabolic flux analysis of three divergent extremely thermophilic bacteria: geobacillus sp. lc300, thermus thermophilus hb8, and rhodothermus marinus dsm 4252. 44():182–190, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.10.007">[DOI]</a>.</p></li>
<li id="id25270"><p>M. Ahsanul Islam, Noushin Hadadi, Meriç Ataman, Vassily Hatzimanikatis, and Gregory Stephanopoulos. Exploring biochemical pathways for mono-ethylene glycol (meg) synthesis from synthesis gas. 41():173–181, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.04.005">[DOI]</a>.</p></li>
<li id="id25277"><p>ShengShee Thor, J.R. Peterson, and Zaida Luthey‐Schulten. Genome-scale metabolic modeling of archaea lends insight into diversity of metabolic function. 2017():1–18, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2017/9763848">[DOI]</a>.</p></li>
<li id="id25281"><p>Jennifer L. Greene, Andreas Wäechter, Keith E. J. Tyo, and Linda J. Broadbelt. Acceleration strategies to enhance metabolic ensemble modeling performance. 113(5):1150–1162, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bpj.2017.07.018">[DOI]</a>.</p></li>
<li id="id25283"><p>Živa Bricman Rejc, Lidija Magdevska, Tilen Tršelič, Timotej Osolin, Rok Vodopivec, Jakob Mraz, Eva Pavliha, Nikolaj Zimic, Tanja Cvitanović, Damjana Rozman, Miha Moškon, and Miha Mraz. Computational modelling of genome-scale metabolic networks and its application to cho cell cultures. 88():150–160, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiomed.2017.07.005">[DOI]</a>.</p></li>
<li id="id25330"><p>Jorge Fernández-de-Cossio-Díaz, Kalet León, and Roberto Mulet. Characterizing steady states of genome-scale metabolic networks in continuous cell cultures. 13(11):e1005835–e1005835, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005835">[DOI]</a>.</p></li>
<li id="id25356"><p>Deepanwita Banerjee, Dharmeshkumar Parmar, Nivedita Bhattacharya, Avinash Ghanate, Venkateswarlu Panchagnula, and Anu Raghunathan. A scalable metabolite supplementation strategy against antibiotic resistant pathogen chromobacterium violaceum induced by nad+/nadh+ imbalance. 11(1):51–51, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0427-z">[DOI]</a>.</p></li>
<li id="id25361"><p>Garrett W. Birkel, Amit Ghosh, Vinay Kumar, Daniel Weaver, David Ando, Tyler W. H. Backman, Adam P. Arkin, Jay D. Keasling, and Héctor García Martín. The jbei quantitative metabolic modeling library (jqmm): a python library for modeling microbial metabolism. 18(1):205–205, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-017-1615-y">[DOI]</a>.</p></li>
<li id="id25388"><p>Michael A. Rowland, Ahmed Abdelzaher, Preetam Ghosh, and Michael L. Mayo. Crosstalk and the dynamical modularity of feed-forward loops in transcriptional regulatory networks. 112(8):1539–1550, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bpj.2017.02.044">[DOI]</a>.</p></li>
<li id="id25389"><p>András Hartmann, Ana Vila-Santa, Nicolai Kallscheuer, Michael Vogt, Alice Julien-Laferrière, Marie-France Sagot, Jan Marienhagen, and Susana Vinga. Optpipe - a pipeline for optimizing metabolic engineering targets. 11(1):143–143, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0515-0">[DOI]</a>.</p></li>
<li id="id25402"><p>A. J. Mueller, Mandy J. Peffers, Carole J. Proctor, and Peter Clegg. Systems approaches in osteoarthritis: identifying routes to novel diagnostic and therapeutic strategies. 35(8):1573–1588, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/jor.23563">[DOI]</a>.</p></li>
<li id="id25433"><p>Jose A. Gomez and Paul I. Barton. Dynamic flux balance analysis using dfbalab. 1716():353–370, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_16">[DOI]</a>.</p></li>
<li id="id25469"><p>Maryam Iman, Tabassom Sobati, Yunes Panahi, and Meysam Mobasheri. Systems biology approach to bioremediation of nitroaromatics: constraint-based analysis of 2,4,6-trinitrotoluene biotransformation by escherichia coli. 22(8):1242–1242, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/molecules22081242">[DOI]</a>.</p></li>
<li id="id25475"><p>Hamideh Fouladiha, Sayed‐Amir Marashi, Mohammad Ali Shokrgozar, Mehdi Farokhi, and Amir Atashi. Applications of a metabolic network model of mesenchymal stem cells for controlling cell proliferation and differentiation. 70(1):331–338, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10616-017-0148-6">[DOI]</a>.</p></li>
<li id="id25482"><p>Sarah Schatschneider, Jessica Schneider, Jochen Blom, Fabien Létisse, Karsten Niehaus, Alexander Goesmann, and Frank-Jörg Vorhölter. Systems and synthetic biology perspective of the versatile plant-pathogenic and polysaccharide-producing bacterium xanthomonas campestris. 163(8):1117–1144, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1099/mic.0.000473">[DOI]</a>.</p></li>
<li id="id25494"><p>Stephen P. Chapman, Marcelo Trindade dos Santos, Giles N. Johnson, Maurício Vieira Kritz, and Jean‐Marc Schwartz. Cyclic decomposition explains a photosynthetic down regulation for chlamydomonas reinhardtii. 162():119–127, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2017.09.014">[DOI]</a>.</p></li>
<li id="id25498"><p>Yvonne van der Meer. Sustainable bio-based materials: opportunities and challenges. ():2–6, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017.1">[DOI]</a>.</p></li>
<li id="id25502"><p>Paul A. Jensen. Coupling fluxes, enzymes, and regulation in genome-scale metabolic models. 1716():337–351, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_15">[DOI]</a>.</p></li>
<li id="id25539"><p>Clémence Frioux, Torsten Schaub, Sebastian Schellhorn, Anne Siegel, and Philipp Wanko. Hybrid metabolic network completion. ():308–321, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-61660-5_28">[DOI]</a>.</p></li>
<li id="id25540"><p>Níels Árni Árnason and Ólafur E. Sigurjónsson. New strategies to understand platelet storage lesion. 12(4):496–500, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/voxs.12394">[DOI]</a>.</p></li>
<li id="id25550"><p>Karthik Raman, Aditya Pratapa, Omkar S. Mohite, and Shankar Balachandran. Computational prediction of synthetic lethals in genome-scale metabolic models using fast-sl. 1716():315–336, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_14">[DOI]</a>.</p></li>
<li id="id25555"><p>Ali R. Zomorrodi and Daniel Segrè. Intracellular metabolic circuits shape inter-species microbial interactions. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/127332">[DOI]</a>.</p></li>
<li id="id25556"><p>Paulo Vilaça, Paulo Maia, Hugo Giesteira, Isabel Rocha, and Miguel Rocha. Analyzing and designing cell factories with optflux. 1716():37–76, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_2">[DOI]</a>.</p></li>
<li id="id25568"><p>Abraham Alemayehu Labena, Yuan‐Nong Ye, Chuan Dong, Fa-Z Zhang, and Feng‐Biao Guo. Sser: species specific essential reactions database. 11(1):50–50, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0426-0">[DOI]</a>.</p></li>
<li id="id25583"><p>Nick Fyson, Min Kyung Kim, Desmond S. Lun, and Caroline Colijn. Gene-centric constraint of metabolic models. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/116558">[DOI]</a>.</p></li>
<li id="id25617"><p>Daniel Osorio, Janneth González, and Andrés Pinzón. Minval: an r package for minimal validation of stoichiometric reactions. 9(1):114–114, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.32614/rj-2017-031">[DOI]</a>.</p></li>
<li id="id25633"><p>Maureen A. Carey, Jason A. Papin, and Jennifer L. Güler. Novel&lt;i&gt;plasmodium falciparum&lt;/i&gt;metabolic network reconstruction identifies shifts associated with clinical antimalarial resistance. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/119941">[DOI]</a>.</p></li>
<li id="id25634"><p>Paul D. Dobson, Pedro Mendes, Douglas B. Kell, and Neil Swainston. A metabolic reaction balancing web service for computational systems biology. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/187328">[DOI]</a>.</p></li>
<li id="id25646"><p>Kalyani Korla and Nagasuma Chandra. A systems perspective of signalling networks in host–pathogen interactions. 97(1):41–57, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s41745-016-0017-x">[DOI]</a>.</p></li>
<li id="id25647"><p>Elisa de Medeiros. Conceptual design of a self-sufficient hybrid biorefinery for syngas production and fermentation to ethanol. ():25–31, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017.5">[DOI]</a>.</p></li>
<li id="id25648"><p>Aravind Madhavan, Raveendran Sindhu, K. B. Arun, Ashok Pandey, and Parameswaran Binod. Advances and tools in engineering yeast for pharmaceutical production. ():29–49, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-981-10-7434-9_3">[DOI]</a>.</p></li>
<li id="id25649"><p>Janaka N. Edirisinghe, José P. Faria, Nomi L. Harris, Benjamin Allen, and Christopher S. Henry. Reconstruction and analysis of central metabolism in microbes. 1716():111–129, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_5">[DOI]</a>.</p></li>
<li id="id25665"><p>Yvonne van der Meer, Ali R. Mulakhudair, William Zimmerman, Parkia Timoriana, Maxim Muravyov, Н. В. Фомченко, Elisa De Medeiros, John A. Posada, Henk Noorman, Patrícia Osseweijer, and Rubens Maciel Filho. Biotech france 2017 conference proceedings. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017">[DOI]</a>.</p></li>
<li id="id25691"><p>Do‐Gyun Kim, Wang‐Hee Lee, Sung-Won Seo, and Hye‐Sun Park. Metabolic reconstruction for developing a tissue-specific model for cattle glycolysis using cobra. 25(02):231–246, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0218339017500127">[DOI]</a>.</p></li>
<li id="id25698"><p>Juan Camilo Roncallo Sarmiento. Avaliação da expressão de genes da via de weimberg sobre o catabolismo de xilose na linhagem produtora de pha&amp;lt;sub&amp;gt;mcl &amp;lt;/sub&amp;gt;&amp;lt;i&amp;gt;pseudomonas sp.&amp;lt;/i&amp;gt; lfm046. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.11606/d.42.2017.tde-10052017-163350">[DOI]</a>.</p></li>
<li id="id25699"><p>Ruben Heck. Metabolic modeling to understand and redesign microbial systems. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/416473">[DOI]</a>.</p></li>
<li id="id25700"><p>G. J. Sharma. Antioxidant and cytotoxic properties of non-conventional food plant, parkia timoriana (dc.) merr. ():15–21, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017.3">[DOI]</a>.</p></li>
<li id="id25701"><p>Ali R. Mulakhudair. Aerobic fermentation performance of zymomonas mobilis mediated by microbubbles generated by fluidic oscillation. ():7–14, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017.2">[DOI]</a>.</p></li>
<li id="id25702"><p>Maxim Muravyov. A two-step process for improving of refractory sulfide concentrate biooxidation. ():22–24, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.26799/cp-biotechfrance2017.4">[DOI]</a>.</p></li>
<li id="id25703"><p>Maria Siurana Paula. Modelling and multiobjective optimization for simulation of cyanobacterial metabolism. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/90578">[DOI]</a>.</p></li>
<li id="id25714"><p>Nathan E. Lewis, Neema Jamshidi, Ines Thiele, and Bernhard Ø. Palsson. Metabolic systems biology. ():1–23, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-642-27737-5_329-2">[DOI]</a>.</p></li>
<li id="id25720"><p>Nicholas A. Jose, Rebecca Lau, Tami L. Swenson, Niels Klitgord, Ferrán García‐Pichel, Benjamin P. Bowen, Richard Baran, and Trent R. Northen. Flux balance modelling to predict bacterial survival during pulsed activity events. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.5194/bg-2017-403">[DOI]</a>.</p></li>
<li id="id25761"><p>Jonathan M. Monk, Colton J. Lloyd, Elizabeth Brunk, Nathan Mih, Anand V. Sastry, Zachary A. King, Rikiya Takeuchi, Wataru Nomura, Zhen Zhang, Hirotada Mori, Adam M. Feist, and Bernhard Ø. Palsson. Iml1515, a knowledgebase that computes escherichia coli traits. 35(10):904–908, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.3956">[DOI]</a>.</p></li>
<li id="id25764"><p>Benjamín J. Sánchez, Cheng Zhang, Avlant Nilsson, Petri‐Jaan Lahtvee, Eduard J. Kerkhoven, and Jens Nielsen. Improving the phenotype predictions of a yeast genome‐scale metabolic model by incorporating enzymatic constraints. 13(8):935–935, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20167411">[DOI]</a>.</p></li>
<li id="id25770"><p>Pavel Dvořák, Pablo I. Nikel, Jir̆ı́ Damborský, and Vı́ctor de Lorenzo. Bioremediation 3.0: engineering pollutant-removing bacteria in the times of systemic biology. 35(7):845–866, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2017.08.001">[DOI]</a>.</p></li>
<li id="id25780"><p>Sebastian Weis, Ana Rita Carlos, Maria Raquel Moita, Sumnima Singh, Birte Blankenhaus, Sílvia Cardoso, Rasmus Larsen, Sofia Rebelo, Sascha Schäuble, Laura del Barrio, Gilles Mithieux, Fabienne Rajas, Sandro Lindig, Michael Bauer, and Miguel P. Soares. Metabolic adaptation establishes disease tolerance to sepsis. 169(7):1263–1275.e14, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2017.05.031">[DOI]</a>.</p></li>
<li id="id25786"><p>Sjoerd Opdam, Anne Richelle, Benjamin P. Kellman, Shanzhong Li, Daniel C. Zielinski, and Nathan E. Lewis. A systematic evaluation of methods for tailoring genome-scale metabolic models. 4(3):318–329.e6, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2017.01.010">[DOI]</a>.</p></li>
<li id="id25787"><p>Noora Ottman, Mark Davids, María Suárez‐Diez, Sjef Boeren, Peter J. Schaap, Vítor A. P. Martins dos Santos, Hauke Smidt, Clara Belzer, and Willem M. de Vos. Genome-scale model and omics analysis of metabolic capacities of &lt;i&gt;akkermansia muciniphila&lt;/i&gt; reveal a preferential mucin-degrading lifestyle. 83(18):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.01014-17">[DOI]</a>.</p></li>
<li id="id25791"><p>Petri‐Jaan Lahtvee, Benjamín J. Sánchez, Agata Smialowska, Sergo Kasvandik, Ibrahim E. Elsemman, Francesco Gatto, and Jens Nielsen. Absolute quantification of protein and mrna abundances demonstrate variability in gene-specific translation efficiency in yeast. 4(5):495–504.e5, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2017.03.003">[DOI]</a>.</p></li>
<li id="id25797"><p>Ran Chao, Shekhar Mishra, Tong Si, and Huimin Zhao. Engineering biological systems using automated biofoundries. 42():98–108, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.06.003">[DOI]</a>.</p></li>
<li id="id25798"><p>Xiulai Chen, Cong Gao, Liang Guo, Guipeng Hu, Qiuling Luo, Jia Liu, Jens Nielsen, Jian Chen, and Li Liu. Dceo biotechnology: tools to design, construct, evaluate, and optimize the metabolic pathway for biosynthesis of chemicals. 118(1):4–72, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.chemrev.6b00804">[DOI]</a>.</p></li>
<li id="id25811"><p>Kaspar Valgepea, Renato de Souza Pinto Lemgruber, Kieran Meaghan, Robin Palfreyman, Tanus Abdalla, Björn D. Heijstra, James B. Y. H. Behrendorff, Ryan Tappel, Michael Köpke, Séan D. Simpson, Lars K. Nielsen, and Esteban Marcellin. Maintenance of atp homeostasis triggers metabolic shifts in gas-fermenting acetogens. 4(5):505–515.e5, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2017.04.008">[DOI]</a>.</p></li>
<li id="id25814"><p>Jan-Philipp Schwarzhans, Tobias Luttermann, Martina Geier, Jörn Kalinowski, and Karl Friehs. Towards systems metabolic engineering in pichia pastoris. 35(6):681–710, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2017.07.009">[DOI]</a>.</p></li>
<li id="id25820"><p>Cristal Zúñiga, Lívia S. Zaramela, and Karsten Zengler. Elucidation of complexity and prediction of interactions in microbial communities. 10(6):1500–1522, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.12855">[DOI]</a>.</p></li>
<li id="id25830"><p>Aarash Bordbar, James T. Yurkovich, Giuseppe Paglia, Óttar Rolfsson, Ólafur E. Sigurjónsson, and Bernhard Ø. Palsson. Elucidating dynamic metabolic physiology through network integration of quantitative time-course metabolomics. 7(1):46249–46249, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep46249">[DOI]</a>.</p></li>
<li id="id25855"><p>Kaspar Valgepea, Quang K. Loi, James B. Y. H. Behrendorff, Renato de Souza Pinto Lemgruber, Manuel R. Plan, Mark P. Hodson, Michael Köpke, Lars K. Nielsen, and Esteban Marcellin. Arginine deiminase pathway provides atp and boosts growth of the gas-fermenting acetogen clostridium autoethanogenum. 41():202–211, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.04.007">[DOI]</a>.</p></li>
<li id="id25856"><p>Siu Hung Joshua Chan, Jingyi Cai, Lin Wang, Margaret Simons, and Costas D. Maranas. Standardizing biomass reactions and ensuring complete mass balance in genome-scale metabolic models. 33(22):3603–3609, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx453">[DOI]</a>.</p></li>
<li id="id25863"><p>Sami Ben Said and Dani Or. Synthetic microbial ecology: engineering habitats for modular consortia. 8():1125–1125, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.01125">[DOI]</a>.</p></li>
<li id="id25864"><p>Milan J. A. Hoek and Roeland M. H. Merks. Emergence of microbial diversity due to cross-feeding interactions in a spatial model of gut microbial metabolism. 11(1):56–56, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0430-4">[DOI]</a>.</p></li>
<li id="id25867"><p>Alyaa M. Abdel‐Haleem, Nathan E. Lewis, Neema Jamshidi, Katsuhiko Mineta, Xin Gao, and Takashi Gojobori. The emerging facets of non-cancerous warburg effect. 8():279–279, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fendo.2017.00279">[DOI]</a>.</p></li>
<li id="id25871"><p>Nicholas S. Kruyer and Pamela Peralta‐Yahya. Metabolic engineering strategies to bio-adipic acid production. 45():136–143, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2017.03.006">[DOI]</a>.</p></li>
<li id="id25874"><p>Faraaz N.K. Yusufi, Meiyappan Lakshmanan, Ying Swan Ho, Bernard Liat Wen Loo, Pramila Ariyaratne, Yuansheng Yang, Say Kong Ng, Tessa Rui Min Tan, Hock Chuan Yeo, Hsueh Lee Lim, Sze Wai Ng, Ai Ping Hiu, Chung Ping Chow, Corrine Wan, Shuwen Chen, Gavin Teo, Song Gao, Ju Xin Chin, Xiaoan Ruan, Wing‐Kin Sung, Wei‐Shou Hu, Miranda G.S. Yap, Muriel Bardor, Niranjan Nagarajan, and Dong‐Yup Lee. Mammalian systems biotechnology reveals global cellular adaptations in a recombinant cho cell line. 4(5):530–542.e6, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2017.04.009">[DOI]</a>.</p></li>
<li id="id25888"><p>C. Damiani, Riccardo Colombo, Daniela Gaglio, Fabrizia Mastroianni, Dario Pescini, Hans V. Westerhoff, Giancarlo Mauri, Marco Vanoni, and Lilia Alberghina. A metabolic core model elucidates how enhanced utilization of glucose and glutamine, with enhanced glutamine-dependent lactate production, promotes cancer cell growth: the warburq effect. 13(9):e1005758–e1005758, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005758">[DOI]</a>.</p></li>
<li id="id25889"><p>Màrius Tomàs‐Gamisans, Pau Ferrer, and Joan Albiol. Fine‐tuning the &lt;i&gt;p. pastoris&lt;/i&gt; imt1026 genome‐scale metabolic model for improved prediction of growth on methanol or glycerol as sole carbon sources. 11(1):224–237, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1751-7915.12871">[DOI]</a>.</p></li>
<li id="id25890"><p>Santosh Pandit, Vaishnavi Ravikumar, Alyaa M. Abdel‐Haleem, Abderahmane Derouiche, V. R. S. S. Mokkapati, Carina Sihlbom, Katsuhiko Mineta, Takashi Gojobori, Xin Gao, Fredrik Westerlund, and Ivan Mijaković. Low concentrations of vitamin c reduce the synthesis of extracellular polymers and destabilize bacterial biofilms. 8():2599–2599, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.02599">[DOI]</a>.</p></li>
<li id="id25900"><p>Hulda S. Haraldsdóttir, Ben Cousins, Ines Thiele, Ronan M. T. Fleming, and Santosh Vempala. Chrr: coordinate hit-and-run with rounding for uniform sampling of constraint-based models. 33(11):1741–1743, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx052">[DOI]</a>.</p></li>
<li id="id25905"><p>Axel von Kamp, Sven Thiele, Oliver Hädicke, and Steffen Klamt. Use of cellnetanalyzer in biotechnology and metabolic engineering. 261():221–228, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2017.05.001">[DOI]</a>.</p></li>
<li id="id25916"><p>Evan Appleton, Curtis Madsen, Nicholas Roehner, and Douglas Densmore. Design automation in synthetic biology. 9(4):a023978–a023978, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/cshperspect.a023978">[DOI]</a>.</p></li>
<li id="id25937"><p>Christopher P. Long, Jacqueline E. Gonzalez, Adam M. Feist, Bernhard Ø. Palsson, and Maciek R. Antoniewicz. Fast growth phenotype of e. coli k-12 from adaptive laboratory evolution does not require intracellular flux rewiring. 44():100–107, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.09.012">[DOI]</a>.</p></li>
<li id="id25944"><p>Thomas J. J. Mueller, Justin Ungerer, Himadri B. Pakrasi, and Costas D. Maranas. Identifying the metabolic differences of a fast-growth phenotype in synechococcus utex 2973. 7(1):41569–41569, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep41569">[DOI]</a>.</p></li>
<li id="id25995"><p>Nana Y. D. Ankrah, Jun‐Bo Luan, and Angela E. Douglas. Cooperative metabolism in a three-partner insect-bacterial symbiosis revealed by metabolic modeling. 199(15):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00872-16">[DOI]</a>.</p></li>
<li id="id26005"><p>Aleksandr Demidenko, Ilya R. Akberdin, Marco N. Allemann, Eric E. Allen, and Marina Kalyuzhnaya. Fatty acid biosynthesis pathways in methylomicrobium buryatense 5g(b1). 7():2167–2167, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2016.02167">[DOI]</a>.</p></li>
<li id="id26014"><p>Govindprasad Bhutada, Martin Kavšček, Rodrigo Ledesma‐Amaro, Stéphane Thomas, Gerald N. Rechberger, Jean‐Marc Nicaud, and Klaus Natter. Sugar versus fat: elimination of glycogen storage improves lipid accumulation in yarrowia lipolytica. 17(3):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsyr/fox020">[DOI]</a>.</p></li>
<li id="id26015"><p>Cristal Zúñiga, Jennifer Levering, Maciek R. Antoniewicz, Michael T. Guarnieri, Michael J. Betenbaugh, and Karsten Zengler. Predicting dynamic metabolic demands in the photosynthetic eukaryote &lt;i&gt;chlorella vulgaris&lt;/i&gt;. 176(1):450–462, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.17.00605">[DOI]</a>.</p></li>
<li id="id26029"><p>Yougan Cheng, Craig J. Thalhauser, Shepard Smithline, Jyotsna Pagidala, Marko Miladinov, Heather Vezina, Manish Gupta, Tarek A. Leil, and Brian J. Schmidt. Qsp toolbox: computational implementation of integrated workflow components for deploying multi-scale mechanistic models. 19(4):1002–1016, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1208/s12248-017-0100-x">[DOI]</a>.</p></li>
<li id="id26030"><p>Angelo Gámez‐Pozo, Lucía Trilla‐Fuertes, Julia Berges‐Soria, Nathalie Selevsek, Rocío López‐Vacas, Mariana Díaz‐Almirón, Paolo Nanni, Jorge M. Arevalillo, Hilario Navarro, Jonas Grossmann, Francisco Gayá, Rubén Gómez Rioja, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Paloma Maı́n, Jaime Feliú, Purificación Martínez del Prado, Pilar Zamora, Eva Ciruelos, Enrique Espinosa, and Juan Ángel Fresno Vara. Functional proteomics outlines the complexity of breast cancer molecular subtypes. 7(1):10100–10100, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-10493-w">[DOI]</a>.</p></li>
<li id="id26031"><p>Ran Yu, Octavio Perez‐Garcia, Huijie Lü, and Kartik Chandran. Nitrosomonas europaea adaptation to anoxic-oxic cycling: insights from transcription analysis, proteomics and metabolic network modeling. 615():1566–1573, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.scitotenv.2017.09.142">[DOI]</a>.</p></li>
<li id="id26038"><p>Qianqian Yuan, Teng Huang, Peishun Li, Tong Hao, Feiran Li, Hongwu Ma, Zhiwen Wang, Xueming Zhao, Tao Chen, and Igor Goryanin. Pathway-consensus approach to metabolic network reconstruction for pseudomonas putida kt2440 by systematic comparison of published models. 12(1):e0169437–e0169437, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0169437">[DOI]</a>.</p></li>
<li id="id26039"><p>William Morrell, Garrett W. Birkel, Mark Forrer, Teresa Díaz-Faes López, Tyler W. H. Backman, Michael Dussault, Christopher J. Petzold, Edward E. K. Baidoo, Zak Costello, David Ando, Jorge Alonso-Gutiérrez, Kevin W. George, Aindrila Mukhopadhyay, Ian Vaino, Jay D. Keasling, Paul D. Adams, Nathan J. Hillson, and Héctor García Martín. The experiment data depot: a web-based software tool for biological experimental data storage, sharing, and visualization. 6(12):2248–2259, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acssynbio.7b00204">[DOI]</a>.</p></li>
<li id="id26040"><p>Kiyotaka Y. Hara, Jyumpei Kobayashi, Ryosuke Yamada, Daisuke Sasaki, Yuki Kuriya, Yoko Hirono-Hara, Jun Ishii, Michihiro Araki, and Akihiko Kondo. Transporter engineering in biomass utilization by yeast. 17(7):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsyr/fox061">[DOI]</a>.</p></li>
<li id="id26068"><p>Lokanand Koduru, Yu‐Jin Kim, Jeongsu Bang, Meiyappan Lakshmanan, Nam Soo Han, and Dong‐Yup Lee. Genome-scale modeling and transcriptome analysis of leuconostoc mesenteroides unravel the redox governed metabolic states in obligate heterofermentative lactic acid bacteria. 7(1):15721–15721, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-16026-9">[DOI]</a>.</p></li>
<li id="id26071"><p>Zhuo Wang, Samuel A. Danziger, Ben Heavner, Shuyi Ma, Jennifer J. Smith, Li Song, Thurston Herricks, Evangelos Simeonidis, Nitin S. Baliga, John D. Aitchison, and Nathan D. Price. Combining inferred regulatory and reconstructed metabolic networks enhances phenotype prediction in yeast. 13(5):e1005489–e1005489, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005489">[DOI]</a>.</p></li>
<li id="id26086"><p>Anthony C. Smith, Filmon Eyassu, Jean‐Pierre Mazat, and Alan J. Robinson. Mitocore: a curated constraint-based model for simulating human central metabolism. 11(1):114–114, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0500-7">[DOI]</a>.</p></li>
<li id="id26090"><p>Arvand Asghari, Sayed‐Amir Marashi, and Naser Ansari‐Pour. A sperm-specific proteome-scale metabolic network model identifies non-glycolytic genes for energy deficiency in asthenozoospermia. 63(2):100–112, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19396368.2016.1263367">[DOI]</a>.</p></li>
<li id="id26091"><p>Agris Pentjuss, Egils Stalidzāns, Jānis Liepiņš, Agnese Kokina, Jekaterina Martynova, Pēteris Zikmanis, Ivars Mozga, Rita Scherbaka, Hassan Hartman, Mark G. Poolman, David A. Fell, and Armands Vīgants. Model-based biotechnological potential analysis of &lt;i&gt;kluyveromyces marxianus&lt;/i&gt; central metabolism. 44(8):1177–1190, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-017-1946-8">[DOI]</a>.</p></li>
<li id="id26092"><p>Yasutaka Hirokawa, Shingo Matsuo, Hiroyuki Hamada, Fumio Matsuda, and Taizo Hanai. Metabolic engineering of synechococcus elongatus pcc 7942 for improvement of 1,3-propanediol and glycerol production based on in silico simulation of metabolic flux distribution. 16(1):212–212, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-017-0824-4">[DOI]</a>.</p></li>
<li id="id26093"><p>Songsong Wei, Xingxing Jian, Jun Chen, Cheng Zhang, and Qiang Hua. Reconstruction of genome-scale metabolic model of yarrowia lipolytica and its application in overproduction of triacylglycerol. 4(1):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40643-017-0180-6">[DOI]</a>.</p></li>
<li id="id26109"><p>Ding Ma, Laurence Yang, Ronan M. T. Fleming, Ines Thiele, Bernhard Ø. Palsson, and Michael A. Saunders. Reliable and efficient solution of genome-scale models of metabolism and macromolecular expression. 7(1):40863–40863, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep40863">[DOI]</a>.</p></li>
<li id="id26110"><p>Jennifer Levering, Christopher L. Dupont, Andrew E. Allen, Bernhard Ø. Palsson, and Karsten Zengler. Integrated regulatory and metabolic networks of the marine diatom &lt;i&gt;phaeodactylum tricornutum&lt;/i&gt; predict the response to rising co &lt;sub&gt;2&lt;/sub&gt; levels. 2(1):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00142-16">[DOI]</a>.</p></li>
<li id="id26111"><p>German Preciat, Lemmer R. P. El Assal, Alberto Noronha, Ines Thiele, Hulda S. Haraldsdóttir, and Ronan M. T. Fleming. Comparative evaluation of atom mapping algorithms for balanced metabolic reactions: application to recon 3d. 9(1):39–39, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13321-017-0223-1">[DOI]</a>.</p></li>
<li id="id26115"><p>Luana Presta, Emanuele Bosi, Leila Mansouri, Lenie Dijkshoorn, Renato Fani, and Marco Fondi. Constraint-based modeling identifies new putative targets to fight colistin-resistant a. baumannii infections. 7(1):3706–3706, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-03416-2">[DOI]</a>.</p></li>
<li id="id26123"><p>Sander Y.A. Rodenburg, Michael Seidl, Dick de Ridder, and Francine Govers. Genome‐wide characterization of &lt;i&gt;phytophthora infestans&lt;/i&gt; metabolism: a systems biology approach. 19(6):1403–1413, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/mpp.12623">[DOI]</a>.</p></li>
<li id="id26128"><p>Govindprasad Bhutada, Martin Kavšček, Rodrigo Ledesma‐Amaro, Stéphane Thomas, Gerald N. Rechberger, Jean‐Marc Nicaud, and Klaus Natter. Sugar versus fat: elimination of glycogen storage improves lipid accumulation in&lt;i&gt;yarrowia lipolytica&lt;/i&gt;. 17(3):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/femsle/fox020">[DOI]</a>.</p></li>
<li id="id26134"><p>Junhua Wang, Cheng Wang, Kejing Song, and Jianping Wen. Metabolic network model guided engineering ethylmalonyl-coa pathway to improve ascomycin production in streptomyces hygroscopicus var. ascomyceticus. 16(1):169–169, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-017-0787-5">[DOI]</a>.</p></li>
<li id="id26138"><p>Juan Nogales, Steinn Guðmundsson, Estrella Duque, Juan L. Ramos, and Bernhard Ø. Palsson. Expanding the computable reactome in&lt;i&gt;pseudomonas putida&lt;/i&gt;reveals metabolic cycles providing robustness. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/139121">[DOI]</a>.</p></li>
<li id="id26139"><p>Ove Øyås and Jörg Stelling. Genome-scale metabolic networks in time and space. 8():51–58, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2017.12.003">[DOI]</a>.</p></li>
<li id="id26150"><p>Chiara Damiani, Marzia Di Filippo, Dario Pescini, Davide Maspero, Riccardo Colombo, and Giancarlo Mauri. Popfba: tackling intratumour heterogeneity with flux balance analysis. 33(14):i311–i318, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx251">[DOI]</a>.</p></li>
<li id="id26151"><p>Yangyang Zheng, Qianqian Yuan, Xiaoyan Yang, and Hongwu Ma. Engineering escherichia coli for poly-(3-hydroxybutyrate) production guided by genome-scale metabolic network analysis. 106():60–66, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.enzmictec.2017.07.003">[DOI]</a>.</p></li>
<li id="id26152"><p>Maike K. Aurich, Ronan M. T. Fleming, and Ines Thiele. A systems approach reveals distinct metabolic strategies among the nci-60 cancer cell lines. 13(8):e1005698–e1005698, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005698">[DOI]</a>.</p></li>
<li id="id26166"><p>Laurent Heirendt, Ines Thiele, and Ronan M. T. Fleming. Distributedfba.jl: high-level, high-performance flux balance analysis in julia. 33(9):1421–1423, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btw838">[DOI]</a>.</p></li>
<li id="id26167"><p>Weihua Guo, You Lin Xu, and Xueyang Feng. Deepmetabolism: a deep learning system to predict phenotype from genome sequencing. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/135574">[DOI]</a>.</p></li>
<li id="id26182"><p>Cheng Wang, Jiao Liu, Huanhuan Liu, Junhua Wang, and Jianping Wen. A genome-scale dynamic flux balance analysis model of streptomyces tsukubaensis nrrl18488 to predict the targets for increasing fk506 production. 123():45–56, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2017.03.017">[DOI]</a>.</p></li>
<li id="id26194"><p>Claudio Angione. Integrating splice-isoform expression into genome-scale models characterizes breast cancer metabolism. 34(3):494–501, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx562">[DOI]</a>.</p></li>
<li id="id26195"><p>Filmon Eyassu and Claudio Angione. Modelling pyruvate dehydrogenase under hypoxia and its role in cancer metabolism. 4(10):170360–170360, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rsos.170360">[DOI]</a>.</p></li>
<li id="id26199"><p>Yi Yang, Xiao-Pan Hu, and Bin‐Guang Ma. Construction and simulation of the bradyrhizobium diazoefficiens usda110 metabolic network: a comparison between free-living and symbiotic states. 13(3):607–620, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c6mb00553e">[DOI]</a>.</p></li>
<li id="id26200"><p>Manon Morin, Delphine Ropers, Eugenio Cinquemani, Jean‐Charles Portais, Brice Enjalbert, and Muriel Cocaign‐Bousquet. The csr system regulates &lt;i&gt;escherichia coli&lt;/i&gt; fitness by controlling glycogen accumulation and energy levels. 8(5):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.01628-17">[DOI]</a>.</p></li>
<li id="id26209"><p>Chintan Joshi, Christie A. M. Peebles, and Ashok Prasad. Modeling and analysis of flux distribution and bioproduct formation in synechocystis sp. pcc 6803 using a new genome-scale metabolic reconstruction. 27():295–310, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.algal.2017.09.013">[DOI]</a>.</p></li>
<li id="id26220"><p>Konstantinos Vavitsas, Emil Østergaard Rue, Lára Kristín Stefánsdóttir, Thiyagarajan Gnanasekaran, Andreas Blennow, Christoph Crocoll, Steinn Guðmundsson, and Poul Erik Jensen. Responses of synechocystis sp. pcc 6803 to heterologous biosynthetic pathways. 16(1):140–140, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-017-0757-y">[DOI]</a>.</p></li>
<li id="id26222"><p>Tanja Cvitanović, Matthias Reichert, Miha Moškon, Miha Mraz, Frank Lammert, and Damjana Rozman. Large‐scale computational models of liver metabolism: how far from the clinics? 66(4):1323–1334, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/hep.29268">[DOI]</a>.</p></li>
<li id="id26242"><p>Thierry Chénard, Frédéric Guénard, Marie‐Claude Vohl, André C. Carpentier, André Tchernof, and Rafaël Najmanovich. Remodeling adipose tissue through in silico modulation of fat storage for the prevention of type 2 diabetes. 11(1):60–60, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0438-9">[DOI]</a>.</p></li>
<li id="id26262"><p>Aida Kalantari, Tao Chen, Boyang Ji, Ivan Andreas Stančik, Vaishnavi Ravikumar, Damjan Franjević, Claire Saulou‐Bérion, Anne Goelzer, and Ivan Mijaković. Conversion of glycerol to 3-hydroxypropanoic acid by genetically engineered bacillus subtilis. 8():638–638, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.00638">[DOI]</a>.</p></li>
<li id="id26263"><p>Alfredo Braunstein, Anna Paola Muntoni, and Andrea Pagnani. An analytic approximation of the feasible space of metabolic networks. 8(1):14915–14915, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms14915">[DOI]</a>.</p></li>
<li id="id26270"><p>Rui Ye, Mingzhi Huang, Hongzhong Lu, Jiangchao Qian, Weilu Lin, Ju Chu, Yingping Zhuang, and Siliang Zhang. Comprehensive reconstruction and evaluation of pichia pastoris genome-scale metabolic model that accounts for 1243 orfs. 4(1):22–22, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40643-017-0152-x">[DOI]</a>.</p></li>
<li id="id26277"><p>Octavio Perez‐Garcia, Cody Mankelow, Kartik Chandran, Silas G. Villas‐Bôas, and Naresh Singhal. Modulation of nitrous oxide (n&lt;sub&gt;2&lt;/sub&gt;o) accumulation by primary metabolites in denitrifying cultures adapting to changes in environmental c and n. 51(23):13678–13688, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.est.7b03345">[DOI]</a>.</p></li>
<li id="id26284"><p>Sara Saheb Kashaf, Claudio Angione, and Píetro Lió. Making life difficult for clostridium difficile: augmenting the pathogen’s metabolic model with transcriptomic and codon usage data for better therapeutic target characterization. 11(1):25–25, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-017-0395-3">[DOI]</a>.</p></li>
<li id="id26292"><p>Nisarg Gohil, Happy Panchasara, Shreya K. Patel, Robert Ramírez‐García, and Vijai Singh. Book review: recent advances in yeast metabolic engineering. 5():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2017.00071">[DOI]</a>.</p></li>
<li id="id26302"><p>Alejandro Acevedo, Raúl Conejeros, and Germán Aroca. Ethanol production improvement driven by genome-scale metabolic modeling and sensitivity analysis in scheffersomyces stipitis. 12(6):e0180074–e0180074, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0180074">[DOI]</a>.</p></li>
<li id="id26307"><p>Piyush Labhsetwar, Marcelo C. R. Melo, John A. Cole, and Zaida Luthey‐Schulten. Population fba predicts metabolic phenotypes in yeast. 13(9):e1005728–e1005728, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005728">[DOI]</a>.</p></li>
<li id="id26313"><p>R. Axayácatl González-García, Tim McCubbin, Annalena Wille, Manuel R. Plan, Lars K. Nielsen, and Esteban Marcellin. Awakening sleeping beauty: production of propionic acid in escherichia coli through the sbm operon requires the activity of a methylmalonyl-coa epimerase. 16(1):121–121, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-017-0735-4">[DOI]</a>.</p></li>
<li id="id26314"><p>Simón Menéndez-Bravo, Julián Paganini, Claudio Avignone–Rossa, Hugo Gramajo, and Ana Arabolaza. Identification of fadab complexes involved in fatty acid β-oxidation in streptomyces coelicolor and construction of a triacylglycerol overproducing strain. 8():1428–1428, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.01428">[DOI]</a>.</p></li>
<li id="id26323"><p>Joseph Gardner and Nanette Boyle. The use of genome-scale metabolic network reconstruction to predict fluxes and equilibrium composition of n-fixing versus c-fixing cells in a diazotrophic cyanobacterium, trichodesmium erythraeum. 11(1):4–4, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-016-0383-z">[DOI]</a>.</p></li>
<li id="id26324"><p>Michael R. MacGillivray, Amy Ko, Emily Gruber, Miranda Sawyer, Eivind Almaas, and Allen Holder. Robust analysis of fluxes in genome-scale metabolic pathways. 7(1):268–268, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-00170-3">[DOI]</a>.</p></li>
<li id="id26325"><p>Sarah Beate Hintermayer and Dirk Weuster‐Botz. Experimental validation of in silico estimated biomass yields of &lt;i&gt;pseudomonas putida&lt;/i&gt; kt2440. 12(6):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201600720">[DOI]</a>.</p></li>
<li id="id26326"><p>Ayca Cankorur‐Cetinkaya, Duygu Dikicioǧlu, and Stephen G. Oliver. Metabolic modeling to identify engineering targets for &lt;i&gt;komagataella phaffii&lt;/i&gt;: the effect of biomass composition on gene target identification. 114(11):2605–2615, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26380">[DOI]</a>.</p></li>
<li id="id26327"><p>Paul Adamczyk and Jennifer L. Reed. Escherichia coli as a model organism for systems metabolic engineering. 6():80–88, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2017.11.001">[DOI]</a>.</p></li>
<li id="id26338"><p>Supreeta Vijayakumar, Max Conway, Píetro Lió, and Claudio Angione. Optimization of multi-omic genome-scale models: methodologies, hands-on tutorial, and perspectives. 1716():389–408, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_18">[DOI]</a>.</p></li>
<li id="id26339"><p>Aitor Blanco‐Míguez, Florentino Fdez‐Riverola, Borja Sánchez, and Anália Lourenço. Resources and tools for the high-throughput, multi-omic study of intestinal microbiota. 20(3):1032–1056, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbx156">[DOI]</a>.</p></li>
<li id="id26340"><p>Óscar Dias, Miguel Rocha, Eugénio C. Ferreira, and Isabel Rocha. Reconstructing high-quality large-scale metabolic models with merlin. 1716():1–36, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_1">[DOI]</a>.</p></li>
<li id="id26348"><p>Zhaleh Hosseini and Sayed‐Amir Marashi. Discovering missing reactions of metabolic networks by using gene co-expression data. 7(1):41774–41774, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep41774">[DOI]</a>.</p></li>
<li id="id26358"><p>Keith Dufault‐Thompson, Huahua Jian, Rui-Xue Cheng, Jiefu Li, Fengping Wang, and Ying Zhang. A genome-scale model of &lt;i&gt;shewanella piezotolerans&lt;/i&gt; simulates mechanisms of metabolic diversity and energy conservation. 2(2):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/msystems.00165-16">[DOI]</a>.</p></li>
<li id="id26359"><p>Mehrdad Ghadiri, Mahshid Heidari, Sayed‐Amir Marashi, and Seyed Hasan Mousavi. A multiscale agent-based framework integrated with a constraint-based metabolic network model of cancer for simulating avascular tumor growth. 13(9):1888–1897, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c7mb00050b">[DOI]</a>.</p></li>
<li id="id26360"><p>Anirikh Chakrabarti, Mathieu Membrez, Delphine Morin‐Rivron, Jay Siddharth, Chieh Jason Chou, Hugues Henry, Stephen J. Bruce, Sylviane Métairon, Frédéric Raymond, Bertrand Bétrisey, Carole Loyer, Scott J. Parkinson, and Mojgan Masoodi. Transcriptomics-driven lipidomics (tdl) identifies the microbiome-regulated targets of ileal lipid metabolism. 3(1):33–33, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41540-017-0033-0">[DOI]</a>.</p></li>
<li id="id26377"><p>Eun Sung Jung, Hye Min Park, Seung Min Hyun, Jong Cheol Shon, Meiyappan Lakshmanan, Minsoo Noh, Hock Chuan Yeo, Kwang‐Hyeon Liu, Dong‐Yup Lee, Jae Sung Hwang, and Choong Hwan Lee. Integrative metabolomic analysis reveals diet supplementation with green tea alleviates uvb-damaged mouse skin correlated with ascorbate metabolism and urea cycle. 13(7):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11306-017-1218-7">[DOI]</a>.</p></li>
<li id="id26378"><p>David Ando and Héctor García Martín. Two-scale 13c metabolic flux analysis for metabolic engineering. 1671():333–352, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7295-1_21">[DOI]</a>.</p></li>
<li id="id26379"><p>Maria Pires Pacheco and Thomas Sauter. The fastcore family: for the fast reconstruction of compact context-specific metabolic networks models. 1716():101–110, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_4">[DOI]</a>.</p></li>
<li id="id26386"><p>Marco Fondi, Eva Pinatel, Adelfia Talà, Fabrizio Damiano, Clarissa Consolandi, Benedetta Mattorre, Daniela Fico, Mariangela Testini, Giuseppe Egidio De Benedetto, Luisa Siculella, Gianluca De Bellis, Pietro Alifano, and Clelia Peano. Time-resolved transcriptomics and constraint-based modeling identify system-level metabolic features and overexpression targets to increase spiramycin production in streptomyces ambofaciens. 8():835–835, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.00835">[DOI]</a>.</p></li>
<li id="id26387"><p>Michael P. Weiner, Julia Tröndle, Christoph Albermann, Georg A. Sprenger, and Dirk Weuster‐Botz. Metabolic control analysis of l -phenylalanine production from glycerol with engineered e. coli using data from short-term steady-state perturbation experiments. 126():86–100, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2017.06.016">[DOI]</a>.</p></li>
<li id="id26407"><p>Xi Wang, Liang Yu, and Shulin Chen. Up finder: a cobra toolbox extension for identifying gene overexpression strategies for targeted overproduction. 5():54–59, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2017.08.001">[DOI]</a>.</p></li>
<li id="id26424"><p>Marzia Di Filippo, Chiara Damiani, Riccardo Colombo, Dario Pescini, and Giancarlo Mauri. Constraint-based modeling and simulation of cell populations. ():126–137, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-57711-1_11">[DOI]</a>.</p></li>
<li id="id26425"><p>Matteo Mori, Enzo Marinari, and Andrea De Martino. A yield-cost tradeoff governs escherichia coli's decision between fermentation and respiration in carbon-limited growth. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/113183">[DOI]</a>.</p></li>
<li id="id26462"><p>Tjerko Kamminga, Simen‐Jan Slagman, Jetta J. E. Bijlsma, Vítor A. P. Martins dos Santos, María Suárez‐Diez, and Peter J. Schaap. Metabolic modeling of energy balances in &lt;i&gt;mycoplasma hyopneumoniae&lt;/i&gt; shows that pyruvate addition increases growth rate. 114(10):2339–2347, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26347">[DOI]</a>.</p></li>
<li id="id26463"><p>Lisa Katharina Blaß, Christian Weyler, and Elmar Heinzle. Network design and analysis for multi-enzyme biocatalysis. 18(1):366–366, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-017-1773-y">[DOI]</a>.</p></li>
<li id="id26464"><p>William J. Kelly, Sorelle Veigne, Xianhua Li, Shyam Sundar Subramanian, Zuyi Huang, and Eugene Schaefer. Optimizing performance of semi‐continuous cell culture in an ambr15™ microbioreactor using dynamic flux balance modeling. 34(2):420–431, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.2585">[DOI]</a>.</p></li>
<li id="id26470"><p>Steinn Guðmundsson, Lucía Agudo, and Juan Nogales. Applications of genome-scale metabolic models of microalgae and cyanobacteria in biotechnology. ():93–111, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-08-101023-5.00004-2">[DOI]</a>.</p></li>
<li id="id26471"><p>James T. Yurkovich, Benjamin J. Yurkovich, Andreas Dräger, Bernhard Ø. Palsson, and Zachary A. King. A padawan programmer’s guide to developing software libraries. 5(5):431–437, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2017.08.003">[DOI]</a>.</p></li>
<li id="id26475"><p>Yu Chen, Yonghong Wang, and Jens Nielsen. Systematic inference of functional phosphorylation events in yeast metabolism. 33(13):1995–2001, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx110">[DOI]</a>.</p></li>
<li id="id26478"><p>Brendan King, Terry Farrah, Matthew A. Richards, Michael Mundy, Evangelos Simeonidis, and Nathan D. Price. Probannoweb and probannopy: probabilistic annotation and gap-filling of metabolic reconstructions. 34(9):1594–1596, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btx796">[DOI]</a>.</p></li>
<li id="id26481"><p>Chao Ye, Nan Xu, Chuan Dong, Yuannong Ye, Xuan Zou, Xiulai Chen, Feng‐Biao Guo, and Li Liu. Imgmd: a platform for the integration and standardisation of in silico microbial genome-scale metabolic models. 7(1):727–727, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-00820-6">[DOI]</a>.</p></li>
<li id="id26482"><p>Ehsan Motamedian, Elham Taheri, and Fatemeh Bagheri. Proliferation inhibition of cisplatin‐resistant ovarian cancer cells using drugs screened by integrating a metabolic model and transcriptomic data. 50(6):, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/cpr.12370">[DOI]</a>.</p></li>
<li id="id26523"><p>Hugh M. Purdy and Jennifer L. Reed. Evaluating the capabilities of microbial chemical production using genome-scale metabolic models. 2():91–97, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coisb.2017.01.008">[DOI]</a>.</p></li>
<li id="id26531"><p>Colton J. Lloyd, Ali Ebrahim, Laurence Yang, Zachary King, Edward Catoiu, Edward J. O'Brien, Joanne K. Liu, and Bernhard O. Palsson. Cobrame: a computational framework for genome-scale models of metabolism and gene expression. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/106559">[DOI]</a>.</p></li>
<li id="id26533"><p>Xingxing Jian, Ningchuan Li, Qian Chen, and Qiang Hua. Model-guided identification of novel gene amplification targets for improving succinate production in escherichia coli nzn111. 9(10):830–835, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c7ib00077d">[DOI]</a>.</p></li>
<li id="id26562"><p>Jimmy G. Lafontaine Rivera, Matthew K. Theisen, Po‐Wei Chen, and James C. Liao. Kinetically accessible yield (kay) for redirection of metabolism to produce exo-metabolites. 41():144–151, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2017.03.011">[DOI]</a>.</p></li>
<li id="id26563"><p>Paulo Maia, Isabel Rocha, and Miguel Rocha. Identification of robust strain designs via tandem pfba/lmoma phenotype prediction. ():1661–1668, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/3067695.3082542">[DOI]</a>.</p></li>
<li id="id26573"><p>Iñigo Apaolaza, Edurne San José‐Eneriz, Xabier Agirre, Felipe Prósper, and Francisco J. Planes. Cobra methods and metabolic drug targets in cancer. 5(1):e1389672–e1389672, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/23723556.2017.1389672">[DOI]</a>.</p></li>
<li id="id26587"><p>Cecilio Valadez‐Cano, Roberto Olivares‐Hernández, Osbaldo Resendis‐Antonio, Alexander DeLuna, and Luis Delaye. Natural selection drove metabolic specialization of the chromatophore in paulinella chromatophora. 17(1):99–99, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12862-017-0947-6">[DOI]</a>.</p></li>
<li id="id26598"><p>Marco Fondi and Renato Fani. Constraint-based metabolic modelling of marine microbes and communities. 34():1–10, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.margen.2017.06.003">[DOI]</a>.</p></li>
<li id="id26599"><p>Neeraj Sinha, María Suárez‐Diez, Evert M. van Schothorst, Jaap Keijer, Vítor A. P. Martins dos Santos, and Guido Hooiveld. Predicting the murine enterocyte metabolic response to diets that differ in lipid and carbohydrate composition. 7(1):8784–8784, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/s41598-017-07350-1">[DOI]</a>.</p></li>
<li id="id26600"><p>George C. diCenzo and Turlough M. Finan. Techniques for large-scale bacterial genome manipulation and characterization of the mutants with respect to in silico metabolic reconstructions. 1716():291–314, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_13">[DOI]</a>.</p></li>
<li id="id26601"><p>Julia Tröndle, Christoph Albermann, Michael P. Weiner, Georg A. Sprenger, and Dirk Weuster‐Botz. Phosphoenolpyruvate transporter enables targeted perturbation during metabolic analysis of l‐phenylalanine production with &lt;i&gt;escherichia coli&lt;/i&gt;. 13(5):e1700611–e1700611, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201700611">[DOI]</a>.</p></li>
<li id="id26607"><p>Sascha Schäuble, Anne-Kristin Stavrum, Mathias Bockwoldt, Pål Puntervoll, and Ines Heiland. Sbmlmod: a python-based web application and web service for efficient data integration and model simulation. 18(1):314–314, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-017-1722-9">[DOI]</a>.</p></li>
<li id="id26608"><p>Aurélien F. A. Moumbock, Conrad V. Simoben, Ludger A. Wessjohann, Wolfgang Sippl, Stefan Günther, and Fidele Ntie‐Kang. Computational studies and biosynthesis of natural products with promising anticancer properties. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.5772/67650">[DOI]</a>.</p></li>
<li id="id26619"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Jorge M. Arevalillo, Mariana Díaz‐Almirón, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Hilario Navarro, Rosa Aras‐López, Irene Dapía, Rocío López‐Vacas, Paolo Nanni, Sara Llorente-Armijo, Pedro Arias, Alberto M. Borobia, Paloma Maı́n, Jaime Feliú, Enrique Espinosa, and Juan Ángel Fresno Vara. Molecular characterization of breast cancer cell response to metabolic drugs. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/185082">[DOI]</a>.</p></li>
<li id="id26632"><p>Jonathan M. Monk and Emanuele Bosi. Integration of comparative genomics with genome-scale metabolic modeling to investigate strain-specific phenotypical differences. 1716():151–175, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_7">[DOI]</a>.</p></li>
<li id="id26653"><p>Tiziano Vignolini, Alessio Mengoni, and Marco Fondi. Template-assisted metabolic reconstruction and assembly of hybrid bacterial models. 1716():177–196, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_8">[DOI]</a>.</p></li>
<li id="id26654"><p>George C. diCenzo, Alessio Mengoni, and Marco Fondi. Tn-core: context-specific reconstruction of core metabolic models using tn-seq data. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/221325">[DOI]</a>.</p></li>
<li id="id26662"><p>Romina Lasry Testa, Claudio Delpino, Vanina Estrada, and M. Soledad Díaz. Metabolic network design of synechocystis sp. pcc 6803 to obtain bioethanol under autotrophic conditions. ():2857–2862, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-444-63965-3.50478-5">[DOI]</a>.</p></li>
<li id="id26700"><p>George C. diCenzo, Alex B. Benedict, Marco Fondi, Graham C. Walker, Turlough M. Finan, Alessio Mengoni, and Joel S. Griffitts. Robustness encoded across essential and accessory replicons in an ecologically versatile bacterium. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/209916">[DOI]</a>.</p></li>
<li id="id26707"><p>Elizabeth Brunk, Roger L. Chang, Jing Xia, Hooman Hefzi, James T. Yurkovich, Donghyuk Kim, Evan Buckmiller, Harris H. Wang, Yang Chen, Bernhard Ø. Palsson, George M. Church, and Nathan E. Lewis. Systemic post-translational control of bacterial metabolism regulates adaptation in dynamic environments. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/180646">[DOI]</a>.</p></li>
<li id="id26714"><p>Anthony C. Smith, Filmon Eyassu, Jean‐Pierre Mazat, and Alan J. Robinson. Mitocore: a curated constraint-based model for simulating human central metabolism. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/141101">[DOI]</a>.</p></li>
<li id="id26733"><p>Brendan King, Terry Farrah, Matthew A. Richards, Michael Mundy, Evangelos Simeonidis, and Nathan D. Price. Probannoweb and probannopy: probabilistic annotation and gap-filling of metabolic reconstructions. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/151258">[DOI]</a>.</p></li>
<li id="id26734"><p>Takeyuki Tamura. Grid-based computational methods for the design of constraint-based parsimonious chemical reaction networks to simulate metabolite production: gridprod. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/166777">[DOI]</a>.</p></li>
<li id="id26735"><p>Nathan Mih, Elizabeth Brunk, Ke Chen, Edward Catoiu, Anand Sastry, Erol Kavvas, Jonathan M. Monk, Zhen Zhang, and Bernhard Ø. Palsson. Ssbio: a python framework for structural systems biology. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/165506">[DOI]</a>.</p></li>
<li id="id26736"><p>Anu Raghunathan and Neema Jamshidi. Integrated host-pathogen metabolic reconstructions. 1716():197–217, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_9">[DOI]</a>.</p></li>
<li id="id26740"><p>Sander Y.A. Rodenburg, Michael Seidl, Dick de Ridder, and Francine Govers. Genome-wide characterization of &lt;i&gt;phytophthora infestans&lt;/i&gt; metabolism: a systems biology approach. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/171082">[DOI]</a>.</p></li>
<li id="id26761"><p>Kumar Gaurav and Yasha Hasija. Comparative analysis of metabolic network of pathogens. 12(2):139–150, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11515-017-1440-8">[DOI]</a>.</p></li>
<li id="id26763"><p>Chao Ye, Nan Xu, Xiulai Chen, and Li Liu. Metabolic model reconstruction and analysis of an artificial microbial ecosystem. 1716():219–238, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7528-0_10">[DOI]</a>.</p></li>
<li id="id26788"><p>Laurent Heirendt, Sylvain Arreckx, Thomas Pfau, Sebastián N. Mendoza, Anne Richelle, Almut Heinken, Hulda S. Haraldsdóttir, Jacek Wachowiak, Sarah Keating, Vanja Vlasov, Stefanía Magnúsdóttir, Chiam Yu Ng, German Preciat, Alise Žagare, Siu Hung Joshua Chan, Maike K. Aurich, Catherine M. Clancy, Jennifer Modamio, John T. Sauls, Alberto Noronha, Aarash Bordbar, Benjamin Cousins, Diana C. El Assal, Luis V. Valcárcel, Iñigo Apaolaza, Susan Ghaderi, Masoud Ahookhosh, Marouen Ben Guebila, Andrejs Kostromins, Nicolas Sompairac, Hoai Minh Le, Ding Ma, Yuekai Sun, Lin Wang, James T. Yurkovich, Miguel A.P. Oliveira, Phan Tu Vuong, Lemmer P. El Assal, Inna Kuperstein, Andreï Zinovyev, H.S. Hinton, William A Bryant, Francisco J. Aragón Artacho, Francisco J. Planes, Egils Stalidzāns, Alejandro Maass, Santosh Vempala, Michael Hucka, Michael A. Saunders, Costas D. Maranas, Nathan E. Lewis, Thomas Sauter, Bernhard Ø. Palsson, Ines Thiele, and Ronan M. T. Fleming. Creation and analysis of biochemical constraint-based models: the cobra toolbox v3.0. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1710.04038">[DOI]</a>.</p></li>
<li id="id26790"><p>Tobias Österlund, Marija Cvijović, and Erik Kristiansson. Integrative analysis of omics data. ():1–24, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527696130.ch1">[DOI]</a>.</p></li>
<li id="id26791"><p>Bonnie V. Dougherty, Thomas J. Moutinho, and Jason A. Papin. Accelerating the drug development pipeline with genome‐scale metabolic network reconstructions. ():139–162, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527696130.ch5">[DOI]</a>.</p></li>
<li id="id26798"><p>Mingyuan Tian, Prashant Kumar, Sanjan T. P. Gupta, and Jennifer L. Reed. Metabolic modeling for design of cell factories. ():71–107, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527696130.ch3">[DOI]</a>.</p></li>
<li id="id26807"><p>J. R. Watson. Modelling sponge-symbiont metabolism. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.14264/uql.2017.180">[DOI]</a>.</p></li>
<li id="id26865"><p>Gabriel Bosque Chacón. Network analysis and modeling in systems biology. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/79082">[DOI]</a>.</p></li>
<li id="id26867"><p>Luana Presta, Emanuele Bosi, Leila Mansouri, Lenie Dijkshoorn, Renato Fani, and Marco Fondi. Constraint-based modeling identifies new putative targets to fight colistin-resistant&lt;i&gt;a. baumannii&lt;/i&gt;infections. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/111765">[DOI]</a>.</p></li>
<li id="id26869"><p>Anika Küken and Zoran Nikoloski. Genome-scale modeling of photorespiratory pathway manipulation. 1653():195–202, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-7225-8_13">[DOI]</a>.</p></li>
<li id="id26872"><p>Lucía Trilla‐Fuertes, Angelo Gámez‐Pozo, Guillermo Prado-Vázquez, Andrea Zapater‐Moros, Mariana Díaz‐Almirón, Claudia Fortes, Rocío López‐Vacas, Iván Márquez‐Rodas, Ainara Soria, Juan Ángel Fresno Vara, and Enrique Espinosa. Melanoma proteomics unravels major differences related to mutational status. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/198358">[DOI]</a>.</p></li>
<li id="id26873"><p>Tjerko Kamminga. Systems analysis of mycoplasma hyopneumoniae to improve vaccine production. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.18174/420702">[DOI]</a>.</p></li>
<li id="id26874"><p>Anna Trigos, BW Goudey, Justin Bedő, TC Conway, NG Faux, and KL Wyres. Collateral sensitivity to β-lactam drugs in drug-resistant tuberculosis is driven by the transcriptional wiring of blai operon genes. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/227538">[DOI]</a>.</p></li>
<li id="id26876"><p>Е. В. Лихошвай. Современные тенденции развития биологии водных экосистем. 2(4):3–14, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.21072/mbj.2017.02.4.01">[DOI]</a>.</p></li>
<li id="id26888"><p>Dipali Singh. Genome scale metabolic modelling of phaeodactylum tricornutum. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.24384/dd1c-r016">[DOI]</a>.</p></li>
<li id="id26906"><p>Van Du T. Tran, Sébastien Moretti, Alix T. Coste, Sara Amorim-Vaz, Dominique Sanglard, and Marco Pagni. Gene set enrichment analysis using rna-seq-based condition-specific metabolic networks. ():200964, 2017. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id26938"><p>Phillip Yen. History of antibiotic adaptation influences evolutionary dynamics during subsequent treatment in pseudomonas aeruginosa. ():, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.18130/v3k05g">[DOI]</a>.</p></li>
<li id="id27123"><p>William A Bryant, Régis Stentz, Gwénaëlle Le Gall, Michael J.E. Sternberg, Simon R. Carding, and Thomas Wilhelm. In silico analysis of the small molecule content of outer membrane vesicles produced by bacteroides thetaiotaomicron indicates an extensive metabolic link between microbe and host. 8():2440–2440, 2017. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2017.02440">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-9" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-9" name="0-9" role="tabpanel" tabindex="0"><div class="docutils container" id="id28090">
<ol class="arabic simple" start="1">
<li id="id28142"><p>Esteban Marcellin, James B. Y. H. Behrendorff, Shilpa Nagaraju, Sashini DeTissera, Simón Segovia, Robin Palfreyman, James Daniell, Cuauhtémoc Licona‐Cassani, Lake‐Ee Quek, Robert Speight, Mark P. Hodson, Séan D. Simpson, Wayne P. Mitchell, Michael Köpke, and Lars K. Nielsen. Low carbon fuels and commodity chemicals from waste gases – systematic approach to understand energy metabolism in a model acetogen. 18(10):3020–3028, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5gc02708j">[DOI]</a>.</p></li>
<li id="id28147"><p>James A. Cotton, Sasisekhar Bennuru, Alexandra Grote, Bhavana Harsha, Alan Tracey, Robin N. Beech, Stephen R. Doyle, Matthew Dunn, Julie C. Dunning Hotopp, Nancy Holroyd, Taisei Kikuchi, Olivia J. Lambert, Amruta S. Mhashilkar, Prudence Mutowo, Nirvana Nursimulu, José M. C. Ribeiro, Matthew B. Rogers, Eleanor Stanley, Lakshmipuram S. Swapna, Isheng Jason Tsai, Thomas R. Unnasch, Denis Voronin, John Parkinson, Thomas B. Nutman, Elodie Ghedin, Matthew Berriman, and Sara Lustigman. The genome of onchocerca volvulus, agent of river blindness. 2(2):16216–16216, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nmicrobiol.2016.216">[DOI]</a>.</p></li>
<li id="id28158"><p>Jacqueline E. Gonzalez, Christopher P. Long, and Maciek R. Antoniewicz. Comprehensive analysis of glucose and xylose metabolism in escherichia coli under aerobic and anaerobic conditions by 13c metabolic flux analysis. 39():9–18, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.11.003">[DOI]</a>.</p></li>
<li id="id28181"><p>Aditya M. Kunjapur, Jason C. Hyun, and Kristala L. J. Prather. Deregulation of s-adenosylmethionine biosynthesis and regeneration improves methylation in the e. coli de novo vanillin biosynthesis pathway. 15(1):61–61, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-016-0459-x">[DOI]</a>.</p></li>
<li id="id28198"><p>Stefano Andreozzi, Anirikh Chakrabarti, Keng Cher Soh, Anthony P. Burgard, Tae Hoon Yang, Stephen Van Dien, Ljubiša Mišković, and Vassily Hatzimanikatis. Identification of metabolic engineering targets for the enhancement of 1,4-butanediol production in recombinant e. coli using large-scale kinetic models. 35():148–159, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.01.009">[DOI]</a>.</p></li>
<li id="id28206"><p>Tobias Großkopf, Jessika Consuegra, Joël Gaffé, John C. Willison, Richard E. Lenski, Orkun S. Soyer, and Dominique Schneider. Metabolic modelling in a dynamic evolutionary framework predicts adaptive diversification of bacteria in a long-term evolution experiment. 16(1):163–163, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12862-016-0733-x">[DOI]</a>.</p></li>
<li id="id28216"><p>Anna Lechner, Elizabeth Brunk, and Jay D. Keasling. The need for integrated approaches in metabolic engineering. 8(11):a023903–a023903, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/cshperspect.a023903">[DOI]</a>.</p></li>
<li id="id28238"><p>Eliza J. R. Peterson, Shuyi Ma, David R. Sherman, and Nitin S. Baliga. Network analysis identifies rv0324 and rv0880 as regulators of bedaquiline tolerance in mycobacterium tuberculosis. 1(8):16078–16078, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nmicrobiol.2016.78">[DOI]</a>.</p></li>
<li id="id28255"><p>Helena Mendes‐Soares, Michael Mundy, Luis Mendes Soares, and Nicholas Chia. Mminte: an application for predicting metabolic interactions among the microbial species in a community. 17(1):343–343, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-016-1230-3">[DOI]</a>.</p></li>
<li id="id28285"><p>Elizabeth Brunk, Nathan Mih, Jonathan M. Monk, Zhen Zhang, Edward J. O’Brien, Spencer Bliven, Ke Chen, Roger L. Chang, Philip E. Bourne, and Bernhard Ø. Palsson. Systems biology of the structural proteome. 10(1):26–26, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-016-0271-6">[DOI]</a>.</p></li>
<li id="id28309"><p>Keisuke Wada, Yoshihiro Toya, Satomi Banno, Katsunori Yoshikawa, Fumio Matsuda, and Hiroshi Shimizu. 13c-metabolic flux analysis for mevalonate-producing strain of escherichia coli. 123(2):177–182, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiosc.2016.08.001">[DOI]</a>.</p></li>
<li id="id28310"><p>Kyeong Rok Choi, Jae Ho Shin, Jae Sung Cho, Dongsoo Yang, and Sang Yup Lee. Systems metabolic engineering of &lt;i&gt;escherichia coli&lt;/i&gt;. 7(1):, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/ecosalplus.esp-0010-2015">[DOI]</a>.</p></li>
<li id="id28313"><p>L Zieliński, Anthony C. Smith, Alexander Smith, and Alan J. Robinson. Metabolic flexibility of mitochondrial respiratory chain disorders predicted by computer modelling. 31():45–55, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mito.2016.09.003">[DOI]</a>.</p></li>
<li id="id28323"><p>Rui Pereira, Jens Nielsen, and Isabel Rocha. Improving the flux distributions simulated with genome-scale metabolic models of saccharomyces cerevisiae. 3():153–163, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2016.05.002">[DOI]</a>.</p></li>
<li id="id28327"><p>Huili Yuan, C. Y. Maurice Cheung, P.A.J. Hilbers, and N.A.W. van Riel. Flux balance analysis of plant metabolism: the effect of biomass composition and model structure on model predictions. 7():537–537, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2016.00537">[DOI]</a>.</p></li>
<li id="id28329"><p>Eli Bogart and Christopher R. Myers. Multiscale metabolic modeling of c4 plants: connecting nonlinear genome-scale models to leaf-scale metabolism in developing maize leaves. 11(3):e0151722–e0151722, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0151722">[DOI]</a>.</p></li>
<li id="id28349"><p>Atefeh Taherian Fard, Sriganesh Srihari, Jessica C. Mar, and Mark A. Ragan. Not just a colourful metaphor: modelling the landscape of cellular development using hopfield networks. 2(1):16001–16001, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/npjsba.2016.1">[DOI]</a>.</p></li>
<li id="id28365"><p>Janaka N. Edirisinghe, Pamela Weisenhorn, Neal Conrad, Fangfang Xia, Ross Overbeek, Rick Stevens, and Christopher S. Henry. Modeling central metabolism and energy biosynthesis across microbial life. 17(1):568–568, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-016-2887-8">[DOI]</a>.</p></li>
<li id="id28372"><p>Marzia Di Filippo, Riccardo Colombo, Chiara Damiani, Dario Pescini, Daniela Gaglio, Marco Vanoni, Lilia Alberghina, and Giancarlo Mauri. Zooming-in on cancer metabolic rewiring with tissue specific constraint-based models. 62():60–69, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiolchem.2016.03.002">[DOI]</a>.</p></li>
<li id="id28373"><p>Bin Du, Daniel C. Zielinski, Erol Kavvas, Andreas Dräger, Justin Tan, Zhen Zhang, Kayla E. Ruggiero, Garri A. Arzumanyan, and Bernhard Ø. Palsson. Evaluation of rate law approximations in bottom-up kinetic models of metabolism. 10(1):40–40, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-016-0283-2">[DOI]</a>.</p></li>
<li id="id28392"><p>Sandra Castillo, Dorothee Barth, Mikko Arvas, Tiina Pakula, Esa Pitkänen, Peter Blomberg, Tuulikki Seppänen‐Laakso, Heli Nygrén, Dhinakaran Sivasiddarthan, Merja Penttilä, and Merja Oja. Whole-genome metabolic model of trichoderma reesei built by comparative reconstruction. 9(1):252–252, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-016-0665-0">[DOI]</a>.</p></li>
<li id="id28463"><p>Wei-Chin Ho and Jianzhi Zhang. Adaptive genetic robustness of&lt;i&gt;escherichia coli&lt;/i&gt;metabolic fluxes. 33(5):1164–1176, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/molbev/msw002">[DOI]</a>.</p></li>
<li id="id28481"><p>Xian‐Hua Li, Yanhong Liu, Qian Jia, Virginia LaMacchia, Kathryn O’Donoghue, and Zuyi Huang. A systems biology approach to investigate the antimicrobial activity of oleuropein. 43(12):1705–1717, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-016-1841-8">[DOI]</a>.</p></li>
<li id="id28487"><p>Bashir Sajo Mienda. Genome-scale metabolic models as platforms for strain design and biological discovery. 35(9):1863–1873, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07391102.2016.1197153">[DOI]</a>.</p></li>
<li id="id28518"><p>Helena Mendes‐Soares and Nicholas Chia. Community metabolic modeling approaches to understanding the gut microbiome: bridging biochemistry and ecology. 105():102–109, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.freeradbiomed.2016.12.017">[DOI]</a>.</p></li>
<li id="id28525"><p>Janne Wallenius, Hannu Maaheimo, and Tero Eerikäinen. Carbon 13-metabolic flux analysis derived constraint-based metabolic modelling of clostridium acetobutylicum in stressed chemostat conditions. 219():378–386, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2016.07.137">[DOI]</a>.</p></li>
<li id="id28547"><p>Luis Tobalina, Jon Pey, Alberto Rezola, and Francisco J. Planes. Assessment of fba based gene essentiality analysis in cancer with a fast context-specific network reconstruction method. 11(5):e0154583–e0154583, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0154583">[DOI]</a>.</p></li>
<li id="id28557"><p>Amphun Chaiboonchoe, Lila Ghamsari, Bushra Dohai, Patrick P. Ng, Basel Khraiwesh, Ashish Jaiswal, Kenan Jijakli, Joseph Koussa, David R. Nelson, Hong Cai, Xinping Yang, Roger L. Chang, Jason A. Papin, Haiyuan Yu, Balaji Santhanam, and Kourosh Salehi‐Ashtiani. Systems level analysis of the chlamydomonas reinhardtii metabolic network reveals variability in evolutionary co-conservation. 12(8):2394–2407, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c6mb00237d">[DOI]</a>.</p></li>
<li id="id28558"><p>Andreas Wagner. The white-knight hypothesis, or does the environment limit innovations? 32(2):131–140, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tree.2016.10.017">[DOI]</a>.</p></li>
<li id="id28565"><p>В. Б. Паничкин, В. А. Лившиц, I. V. Biryukova, and Sergey V. Mashko. Metabolic engineering of escherichia coli for l-tryptophan production. 52(9):783–809, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1134/s0003683816090052">[DOI]</a>.</p></li>
<li id="id28582"><p>Oveis Jamialahmadi, Ehsan Motamedian, and Sameereh Hashemi‐Najafabadi. Bikegg: a cobra toolbox extension for bridging the bigg and kegg databases. 12(11):3459–3466, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c6mb00532b">[DOI]</a>.</p></li>
<li id="id28589"><p>Nicholas Ribaudo, Xian‐Hua Li, Brett Davis, Thomas K. Wood, and Zuyi Huang. A genome‐scale modeling approach to quantify biofilm component growth of &lt;i&gt;salmonella typhimurium&lt;/i&gt;. 82(1):154–166, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1750-3841.13565">[DOI]</a>.</p></li>
<li id="id28595"><p>Bhanwar Lal Puniya, Deepika Kulshreshtha, Inna Mittal, Ahmed Mobeen, and Srinivasan Ramachandran. Integration of metabolic modeling with gene co-expression reveals transcriptionally programmed reactions explaining robustness in mycobacterium tuberculosis. 6(1):23440–23440, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep23440">[DOI]</a>.</p></li>
<li id="id28608"><p>Rachhpal S. Kahlon. Pseudomonas for industrial biotechnology. ():281–342, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-31198-2_8">[DOI]</a>.</p></li>
<li id="id28609"><p>Ildikó Miklóssy, Zsolt Bodor, Réka Sinkler, Kálmán Csongor Orbán, Szabolcs Lányi, and Beáta Albert. &lt;i&gt;in silico&lt;/i&gt; and &lt;i&gt;in vivo&lt;/i&gt; stability analysis of a heterologous biosynthetic pathway for 1,4-butanediol production in metabolically engineered &lt;i&gt;e. coli&lt;/i&gt;. 35(9):1874–1889, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07391102.2016.1198721">[DOI]</a>.</p></li>
<li id="id28646"><p>Sarah McGarrity, Haraldur Halldórsson, Sirus Palsson, Pär I. Johansson, and Óttar Rolfsson. Understanding the causes and implications of endothelial metabolic variation in cardiovascular disease through genome-scale metabolic modeling. 3():10–10, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcvm.2016.00010">[DOI]</a>.</p></li>
<li id="id28647"><p>Sayed‐Rzgar Hosseini and Andreas Wagner. The potential for non-adaptive origins of evolutionary innovations in central carbon metabolism. 10(1):97–97, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-016-0343-7">[DOI]</a>.</p></li>
<li id="id28701"><p>Anonymous. Current challenges in modeling cellular metabolism. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/978-2-88919-754-5">[DOI]</a>.</p></li>
<li id="id28703"><p>Abinaya Badri, Aparajitha Srinivasan, and Karthik Raman. In silico approaches to metabolic engineering. ():161–200, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-444-63667-6.00008-0">[DOI]</a>.</p></li>
<li id="id28728"><p>Yuqi Zhao, Yanjie Wang, Lei Zou, and Jing‐Fei Huang. Reconstruction and applications of consensus yeast metabolic network based on &lt;scp&gt;rna&lt;/scp&gt; sequencing. 6(4):264–275, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/2211-5463.12033">[DOI]</a>.</p></li>
<li id="id28764"><p>Helen S. Toogood, Shirley Tait, Adrian J. Jervis, Aisling Ní Cheallaigh, Luke Humphreys, Eriko Takano, John M. Gardiner, and Nigel S. Scrutton. Natural product biosynthesis in escherichia coli. 575():247–270, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/bs.mie.2016.02.020">[DOI]</a>.</p></li>
<li id="id28765"><p>Milsee Mol, Vineetha Mandlik, and Shailza Singh. Microbial chassis assisting retrosynthesis. ():1–10, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-81-322-2809-7_1">[DOI]</a>.</p></li>
<li id="id28766"><p>Zhaobin Xu, Nicholas Ribaudo, Xian‐Hua Li, Thomas K. Wood, and Zuyi Huang. A genome-scale modeling approach to investigate the antibiotics-triggered perturbation in the metabolism of &amp;lt;italic&amp;gt;pseudomonas aeruginosa&amp;lt;/italic&amp;gt;. 2(3):39–42, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/lls.2017.2652473">[DOI]</a>.</p></li>
<li id="id28776"><p>Helena Mendes‐Soares, Michael Mundy, Luis Mendes Soares, and Nicholas Chia. Mminte: an application for predicting metabolic interactions among the microbial species in a community. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/059550">[DOI]</a>.</p></li>
<li id="id28790"><p>Almut Heinken, Dmitry A. Ravcheev, and Ines Thiele. Systems biology of bacteria‐host interactions. ():113–137, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781118982907.ch7">[DOI]</a>.</p></li>
<li id="id28792"><p>Francis Isidore G. Totañes, Sanu Shameer, David R. Westhead, Fabien Jourdan, and Glenn A. McConkey. Understanding protozoan parasite metabolism and identifying drug targets through constraint‐based modeling. ():487–512, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527694082.ch21">[DOI]</a>.</p></li>
<li id="id28816"><p>Martins Mednis. Comparative analysis of biochemical network reconstructions. 4(4):, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.22364/bjmc.2016.4.4.27">[DOI]</a>.</p></li>
<li id="id28818"><p>Abel Folch Fortuny. Chemometric approaches for systems biology. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/77148">[DOI]</a>.</p></li>
<li id="id28836"><p>Matthew Biggs. Novel computational tools for linking genotypes to microbial community phenotypes. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.18130/v34p5b">[DOI]</a>.</p></li>
<li id="id28877"><p>Borja Sánchez, Susana Delgado, Aitor Blanco‐Míguez, Anália Lourenço, Miguel Gueimonde, and Abelardo Margollés. Probiotics, gut microbiota, and their influence on host health and disease. 61(1):, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/mnfr.201600240">[DOI]</a>.</p></li>
<li id="id28879"><p>Stefanía Magnúsdóttir, Almut Heinken, Laura Kutt, Dmitry A. Ravcheev, Eugen Bauer, Alberto Noronha, Kacy Greenhalgh, Christian Jäger, Joanna Bagińska, Paul Wilmes, Ronan M. T. Fleming, and Ines Thiele. Generation of genome-scale metabolic reconstructions for 773 members of the human gut microbiota. 35(1):81–89, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.3703">[DOI]</a>.</p></li>
<li id="id28890"><p>Eugeni Belda, Ruben Heck, María José López Sánchez, Stéphane Cruveiller, Valérie Barbe, Claire M. Fraser, Hans‐Peter Klenk, Jörn Petersen, Anne Morgat, Pablo I. Nikel, David Vallenet, Zoé Rouy, Agnieszka Sekowska, Vítor A. P. Martins dos Santos, Vı́ctor de Lorenzo, Antoine Danchin, and Claudine Médigue. The revisited genome of &lt;i&gt;pseudomonas putida&lt;/i&gt; kt2440 enlightens its value as a robust metabolic &lt;i&gt;chassis&lt;/i&gt;. 18(10):3403–3424, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.13230">[DOI]</a>.</p></li>
<li id="id28893"><p>Neil Swainston, Kieran Smallbone, Hooman Hefzi, Paul D. Dobson, Judy McKinley Brewer, Michael Hanscho, Daniel C. Zielinski, Kok Siong Ang, Natalie J. Gardiner, Jahir M. Gutierrez, Sarantos Kyriakopoulos, Meiyappan Lakshmanan, Shangzhong Li, Joanne K. Liu, Verónica S. Martínez, Camila A. Orellana, Lake‐Ee Quek, Alex Thomas, Jürgen Zanghellini, Nicole Borth, Dong‐Yup Lee, Lars K. Nielsen, Douglas B. Kell, Nathan E. Lewis, and Pedro Mendes. Recon 2.2: from reconstruction to model of human metabolism. 12(7):109–109, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11306-016-1051-4">[DOI]</a>.</p></li>
<li id="id28902"><p>Hooman Hefzi, Kok Siong Ang, Michael Hanscho, Aarash Bordbar, David E. Ruckerbauer, Meiyappan Lakshmanan, Camila A. Orellana, Deniz Baycin‐Hizal, Yingxiang Huang, Daniel Ley, Verónica S. Martínez, Sarantos Kyriakopoulos, Natalia E. Jiménez, Daniel C. Zielinski, Lake‐Ee Quek, Tune Wulff, Johnny Arnsdorf, Shangzhong Li, Jae Seong Lee, Giuseppe Paglia, Nicolás Loira, Philipp N. Spahn, Lasse Ebdrup Pedersen, Jahir M. Gutierrez, Zachary A. King, Anne Mathilde Lund, Harish Nagarajan, Alex Thomas, Alyaa M. Abdel‐Haleem, Jürgen Zanghellini, Helene Faustrup Kildegaard, Björn G. Voldborg, Ziomara P. Gerdtzen, Michael J. Betenbaugh, Bernhard Ø. Palsson, Mikael Rørdam Andersen, Lars K. Nielsen, Nicole Borth, Dong‐Yup Lee, and Nathan E. Lewis. A consensus genome-scale reconstruction of chinese hamster ovary cell metabolism. 3(5):434–443.e8, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2016.10.020">[DOI]</a>.</p></li>
<li id="id28921"><p>Matteo Mori, Terence Hwa, Olivier Martin, Andrea De Martino, and Enzo Marinari. Constrained allocation flux balance analysis. 12(6):e1004913–e1004913, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004913">[DOI]</a>.</p></li>
<li id="id28943"><p>André Schultz and Amina A. Qutub. Reconstruction of tissue-specific metabolic networks using corda. 12(3):e1004808–e1004808, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004808">[DOI]</a>.</p></li>
<li id="id28948"><p>John T. Sauls, Dongyang Li, and Suckjoon Jun. Adder and a coarse-grained approach to cell size homeostasis in bacteria. 38():38–44, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ceb.2016.02.004">[DOI]</a>.</p></li>
<li id="id28959"><p>Jared T. Broddrick, Benjamin E. Rubin, David Welkie, Niu Du, Nathan Mih, Spencer Diamond, Jenny J. Lee, Susan S. Golden, and Bernhard Ø. Palsson. Unique attributes of cyanobacterial metabolism revealed by improved genome-scale metabolic modeling and essential gene analysis. 113(51):, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1613446113">[DOI]</a>.</p></li>
<li id="id28966"><p>Cristal Zúñiga, Chien‐Ting Li, Tyler Huelsman, Jennifer Levering, Daniel C. Zielinski, Brian O. McConnell, Christopher P. Long, Eric P. Knoshaug, Michael T. Guarnieri, Maciek R. Antoniewicz, Michael J. Betenbaugh, and Karsten Zengler. Genome-scale metabolic model for the green alga &lt;i&gt;chlorella vulgaris&lt;/i&gt; utex 395 accurately predicts phenotypes under autotrophic, heterotrophic, and mixotrophic growth conditions. 172(1):589–602, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.16.00593">[DOI]</a>.</p></li>
<li id="id28973"><p>Jennifer Levering, Jared T. Broddrick, Christopher L. Dupont, Graham Peers, Karen Beeri, Joshua J. Mayers, Alessandra A. Gallina, Andrew E. Allen, Bernhard Ø. Palsson, and Karsten Zengler. Genome-scale model reveals metabolic basis of biomass partitioning in a model diatom. 11(5):e0155038–e0155038, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0155038">[DOI]</a>.</p></li>
<li id="id29008"><p>Aarash Bordbar, Pär I. Johansson, Giuseppe Paglia, Scott J. Harrison, Kristine Wichuk, Manuela Magnúsdóttir, Sóley Valgeirsdóttir, Mikkel Gybel‐Brask, Sisse Rye Ostrowski, Sirus Palsson, Óttar Rolfsson, Ólafur E. Sigurjónsson, Morten Hansen, Sveinn Guðmundsson, and Bernhard Ø. Palsson. Identified metabolic signature for assessing red blood cell unit quality is associated with endothelial damage markers and clinical outcomes. 56(4):852–862, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/trf.13460">[DOI]</a>.</p></li>
<li id="id29014"><p>John I. Hendry, Charulata B. Prasannan, Aditi Joshi, Santanu Dasgupta, and Pramod P. Wangikar. Metabolic model of synechococcus sp. pcc 7002: prediction of flux distribution and network modification for enhanced biofuel production. 213():190–197, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2016.02.128">[DOI]</a>.</p></li>
<li id="id29019"><p>Ying Li, Xi Wang, Xizhen Ge, and Pingfang Tian. High production of 3-hydroxypropionic acid in klebsiella pneumoniae by systematic optimization of glycerol metabolism. 6(1):26932–26932, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep26932">[DOI]</a>.</p></li>
<li id="id29020"><p>Robert Häsler, Raheleh Sheibani‐Tezerji, Anupam Sinha, Matthias Barann, Ateequr Rehman, Daniela Esser, Konrad Aden, Carolin Knecht, Berenice Brandt, Susanna Nikolaus, Sascha Schäuble, Christoph Kaleta, André Franke, Christoph Fretter, Werner Müller, Marc‐Thorsten Hütt, Michael Krawczak, Stefan Schreiber, and Philip Rosenstiel. Uncoupling of mucosal gene regulation, mrna splicing and adherent microbiota signatures in inflammatory bowel disease. 66(12):2087–2097, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1136/gutjnl-2016-311651">[DOI]</a>.</p></li>
<li id="id29022"><p>Kiyan Shabestary and Elton P. Hudson. Computational metabolic engineering strategies for growth-coupled biofuel production by synechocystis. 3():216–226, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2016.07.003">[DOI]</a>.</p></li>
<li id="id29023"><p>George C. diCenzo, Alice Checcucci, Marco Bazzicalupo, Alessio Mengoni, Carlo Viti, Łukasz Dziewit, Turlough M. Finan, Marco Galardini, and Marco Fondi. Metabolic modelling reveals the specialization of secondary replicons for niche adaptation in sinorhizobium meliloti. 7(1):12219–12219, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms12219">[DOI]</a>.</p></li>
<li id="id29032"><p>Mohammad Tauqeer Alam, Aleksej Zelezniak, Michael Mülleder, Pavel V. Shliaha, Roland F. Schwarz, Floriana Capuano, Jakob Vowinckel, Elahe Radmaneshfar, Antje Krüger, Enrica Calvani, Steve Michel, Stefan T. Börno, Stefan Christen, Kiran Raosaheb Patil, Bernd Timmermann, Kathryn S. Lilley, and Markus Ralser. The metabolic background is a global player in saccharomyces gene expression epistasis. 1(3):15030–15030, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nmicrobiol.2015.30">[DOI]</a>.</p></li>
<li id="id29044"><p>Weiqi Fu, Amphun Chaiboonchoe, Basel Khraiwesh, David R. Nelson, Dina Al-Khairy, Alexandra Mystikou, Amnah Alzahmi, and Kourosh Salehi‐Ashtiani. Algal cell factories: approaches, applications, and potentials. 14(12):225–225, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/md14120225">[DOI]</a>.</p></li>
<li id="id29048"><p>Jasper J. Koehorst, Jesse C. J. van Dam, Ruben Heck, Edoardo Saccenti, Vítor A. P. Martins dos Santos, María Suárez‐Diez, and Peter J. Schaap. Comparison of 432 pseudomonas strains through integration of genomic, functional, metabolic and expression data. 6(1):38699–38699, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep38699">[DOI]</a>.</p></li>
<li id="id29052"><p>Jozef Anné, Anastassios Economou, and Kristel Bernaerts. Protein secretion in gram-positive bacteria: from multiple pathways to biotechnology. 404():267–308, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/82_2016_49">[DOI]</a>.</p></li>
<li id="id29054"><p>Alberto Noronha, Anna Dröfn Daníelsdóttir, Piotr Gawron, Freyr Jóhannsson, Soffía Jónsdóttir, Sindri Jarlsson, Jón Pétur Gunnarsson, Sigurður Brynjólfsson, Reinhard Schneider, Ines Thiele, and Ronan M. T. Fleming. Reconmap: an interactive visualization of human metabolism. 33(4):605–607, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btw667">[DOI]</a>.</p></li>
<li id="id29056"><p>Màrius Tomàs‐Gamisans, Pau Ferrer, and Joan Albiol. Integration and validation of the genome-scale metabolic models of pichia pastoris: a comprehensive update of protein glycosylation pathways, lipid and energy metabolism. 11(1):e0148031–e0148031, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0148031">[DOI]</a>.</p></li>
<li id="id29064"><p>Xiaolin Zhang, Christopher J. Tervo, and Jennifer L. Reed. Metabolic assessment of e. coli as a biofactory for commercial products. 35():64–74, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.01.007">[DOI]</a>.</p></li>
<li id="id29074"><p>Adam P. Arkin, Rick Stevens, Robert W. Cottingham, Sergei Maslov, Christopher S. Henry, Paramvir Dehal, Doreen Ware, Fernando Pérez, Nomi L. Harris, Shane Canon, Michael W. Sneddon, Matthew Henderson, William J. Riehl, Dan Gunter, Dan Murphy-Olson, Stephen Y. Chan, Roy T. Kamimura, Thomas Brettin, Folker Meyer, Dylan Chivian, David J. Weston, Elizabeth M. Glass, Brian H. Davison, Sunita Kumari, Benjamin Allen, Jason K. Baumohl, Aaron A. Best, Ben Bowen, Steven E. Brenner, Christopher Bun, John‐Marc Chandonia, Jer-Ming Chia, Ric Colasanti, Neal Conrad, James J. Davis, Matthew DeJongh, Scott Devoid, Emily Dietrich, Meghan M. Drake, Inna Dubchak, Janaka N. Edirisinghe, Gang Fang, José P. Faria, Paul M Frybarger, Wolfgang Gerlach, Mark Gerstein, James Gurtowski, Holly L Haun, Fei He, Rashmi Jain, Marcin P. Joachimiak, Kevin Keegan, Shinnosuke Kondo, Vivek Kumar, Miriam Land, Marissa Mills, Pavel S. Novichkov, Taeyun Oh, Gary J. Olsen, Bob Olson, Bruce Parrello, Shiran Pasternak, Erik Pearson, Sarah S Poon, Gavin A Price, Srividya Ramakrishnan, Priya Ranjan, Pamela C. Ronald, Michael C. Schatz, Samuel M. D. Seaver, Maulik Shukla, Roman A. Sutormin, Mustafa Syed, James Thomason, Nathan Tintle, Daifeng Wang, Fangfang Xia, Hyunseung Yoo, and Shinjae Yoo. The doe systems biology knowledgebase (kbase). ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/096354">[DOI]</a>.</p></li>
<li id="id29098"><p>Daniel Machado, Markus J. Herrgård, and Isabel Rocha. Stoichiometric representation of gene–protein–reaction associations leverages constraint-based analysis from reaction to gene-level phenotype prediction. 12(10):e1005140–e1005140, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005140">[DOI]</a>.</p></li>
<li id="id29122"><p>Daniel Cuevas, Janaka N. Edirisinghe, Chris Henry, Ross Overbeek, Taylor O’Connell, and Robert A. Edwards. From dna to fba: how to build your own genome-scale metabolic model. 7():907–907, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2016.00907">[DOI]</a>.</p></li>
<li id="id29133"><p>Silvio Waschina, Glen G D’Souza, Christian Kost, and Christoph Kaleta. Metabolic network architecture and carbon source determine metabolite production costs. 283(11):2149–2163, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/febs.13727">[DOI]</a>.</p></li>
<li id="id29137"><p>Matthew A. Richards, Thomas J. Lie, Juan Zhang, Stephen W. Ragsdale, John A. Leigh, and Nathan D. Price. Exploring hydrogenotrophic methanogenesis: a genome scale metabolic reconstruction of methanococcus maripaludis. 198(24):3379–3390, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00571-16">[DOI]</a>.</p></li>
<li id="id29140"><p>Christopher P. Long, Jacqueline E. Gonzalez, Nicholas R. Sandoval, and Maciek R. Antoniewicz. Characterization of physiological responses to 22 gene knockouts in escherichia coli central carbon metabolism. 37():102–113, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.05.006">[DOI]</a>.</p></li>
<li id="id29155"><p>Elias Björnson, Jan Borén, and Adil Mardinoğlu. Personalized cardiovascular disease prediction and treatment—a review of existing strategies and novel systems medicine tools. 7():2–2, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2016.00002">[DOI]</a>.</p></li>
<li id="id29165"><p>Laurence Yang, Ding Ma, Ali Ebrahim, Colton J. Lloyd, Michael A. Saunders, and Bernhard Ø. Palsson. Solveme: fast and reliable solution of nonlinear me models. 17(1):391–391, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-016-1240-1">[DOI]</a>.</p></li>
<li id="id29175"><p>Ankita Juneja, Frank Chaplen, and Ganti S. Murthy. Genome scale metabolic reconstruction of chlorella variabilis for exploring its metabolic potential for biofuels. 213():103–110, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2016.02.118">[DOI]</a>.</p></li>
<li id="id29177"><p>Miroslava Čuperlović‐Culf and Adrian S. Culf. Applied metabolomics in drug discovery. 11(8):759–770, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/17460441.2016.1195365">[DOI]</a>.</p></li>
<li id="id29178"><p>Dávid Malatinszky, Ralf Steuer, and Patrik R. Jones. A comprehensively curated genome-scale two-cell model for the heterocystous cyanobacterium &lt;i&gt;anabaena&lt;/i&gt; sp. pcc 7120. 173(1):509–523, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.16.01487">[DOI]</a>.</p></li>
<li id="id29181"><p>Elsje Pienaar, William M. Matern, Jennifer J. Linderman, Joel S. Bader, and Denise E. Kirschner. Multiscale model of mycobacterium tuberculosis infection maps metabolite and gene perturbations to granuloma sterilization predictions. 84(5):1650–1669, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/iai.01438-15">[DOI]</a>.</p></li>
<li id="id29182"><p>Kumari Sonal Choudhary, Neha Rohatgi, Skarphéðinn Halldórsson, Eiríkur Briem, Þórarinn Guðjónsson, Steinn Guðmundsson, and Óttar Rolfsson. Egfr signal-network reconstruction demonstrates metabolic crosstalk in emt. 12(6):e1004924–e1004924, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004924">[DOI]</a>.</p></li>
<li id="id29197"><p>Hadi Nazem‐Bokaee, Saratram Gopalakrishnan, James G. Ferry, Thomas K. Wood, and Costas D. Maranas. Assessing methanotrophy and carbon fixation for biofuel production by methanosarcina acetivorans. 15(1):10–10, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-015-0404-4">[DOI]</a>.</p></li>
<li id="id29198"><p>Manon Morin, Delphine Ropers, Fabien Létisse, Sandrine Laguerre, Jean‐Charles Portais, Muriel Cocaign‐Bousquet, and Brice Enjalbert. The post‐transcriptional regulatory system csr controls the balance of metabolic pools in upper glycolysis of &lt;i&gt;escherichia coli&lt;/i&gt;. 100(4):686–700, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/mmi.13343">[DOI]</a>.</p></li>
<li id="id29199"><p>Georgios Savoglidis, Aline Xavier da Silveira dos Santos, Isabelle Riezman, Paolo Angelino, Howard Riezman, and Vassily Hatzimanikatis. A method for analysis and design of metabolism using metabolomics data and kinetic models: application on lipidomics using a novel kinetic model of sphingolipid metabolism. 37():46–62, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.04.002">[DOI]</a>.</p></li>
<li id="id29205"><p>Miguel A. Campodonico, Daniela Vaisman, Jean Franco Castro, Valeria Razmilic, Francesca Mercado, Bárbara A. Andrews, Adam M. Feist, and Juan A. Asenjo. Acidithiobacillus ferrooxidans's comprehensive model driven analysis of the electron transfer metabolism and synthetic strain design for biomining applications. 3():84–96, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2016.03.003">[DOI]</a>.</p></li>
<li id="id29206"><p>Marouen Ben Guebila and Ines Thiele. Model-based dietary optimization for late-stage, levodopa-treated, parkinson’s disease patients. 2(1):16013–16013, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/npjsba.2016.13">[DOI]</a>.</p></li>
<li id="id29210"><p>Pedro A. Saa and Lars K. Nielsen. Ll-achrb: a scalable algorithm for sampling the feasible solution space of metabolic networks. 32(15):2330–2337, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btw132">[DOI]</a>.</p></li>
<li id="id29219"><p>Jin Chen and Michael A. Henson. In silico metabolic engineering of clostridium ljungdahlii for synthesis gas fermentation. 38():389–400, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.10.002">[DOI]</a>.</p></li>
<li id="id29225"><p>Mariana Galvão Ferrarini, Franciele Maboni Siqueira, Scheila Gabriele Mucha, Tony Palama, Élodie Jobard, Bénédicte Elena‐Herrmann, Ana Tereza Ribeiro de Vasconcelos, Florence Tardy, Irene Silveira Schrank, Arnaldo Zaha, and Marie‐France Sagot. Insights on the virulence of swine respiratory tract mycoplasmas through genome-scale metabolic modeling. 17(1):353–353, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-016-2644-z">[DOI]</a>.</p></li>
<li id="id29241"><p>Di Liu, Ni Wan, Fuzhong Zhang, Yinjie Tang, and Stephen Gang Wu. Enhancing fatty acid production in &lt;i&gt;escherichia coli&lt;/i&gt; by &lt;i&gt;vitreoscilla&lt;/i&gt; hemoglobin overexpression. 114(2):463–467, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26067">[DOI]</a>.</p></li>
<li id="id29243"><p>Maria Pires Pacheco, Thomas Pfau, and Thomas Sauter. Benchmarking procedures for high-throughput context specific reconstruction algorithms. 6():410–410, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2015.00410">[DOI]</a>.</p></li>
<li id="id29246"><p>R.P. Vivek-Ananth and Areejit Samal. Advances in the integration of transcriptional regulatory information into genome-scale metabolic models. 147():1–10, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2016.06.001">[DOI]</a>.</p></li>
<li id="id29253"><p>Thibault Nidelet, Pascale Brial, Carole Camarasa, and Sylvie Dequin. Diversity of flux distribution in central carbon metabolism of s. cerevisiae strains from diverse environments. 15(1):58–58, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-016-0456-0">[DOI]</a>.</p></li>
<li id="id29254"><p>Juliane Gebauer, Christoph Gentsch, Johannes Mansfeld, Kathrin Schmeißer, Silvio Waschina, Susanne Brandes, Lukas Klimmasch, Nicola Zamboni, Kim Zarse, Stefan Schuster, Michael Ristow, Sascha Schäuble, and Christoph Kaleta. A genome-scale database and reconstruction of caenorhabditis elegans metabolism. 2(5):312–322, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2016.04.017">[DOI]</a>.</p></li>
<li id="id29258"><p>Hongzhong Lu, Weiqiang Cao, Liming Ouyang, Jianye Xia, Mingzhi Huang, Ju Chu, Yingping Zhuang, Siliang Zhang, and Henk Noorman. Comprehensive reconstruction and in silico analysis of &lt;i&gt;aspergillus niger&lt;/i&gt; genome‐scale metabolic network model that accounts for 1210 orfs. 114(3):685–695, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.26195">[DOI]</a>.</p></li>
<li id="id29270"><p>Filomena Sannino, Maria Elisa Giuliani, U. Salvatore, Gennaro Antonio Apuzzo, Donatella de Pascale, Renato Fani, Marco Fondi, Gennaro Marino, Maria Luisa Tutino, and Ermenegilda Parrilli. A novel synthetic medium and expression system for subzero growth and recombinant protein production in pseudoalteromonas haloplanktis tac125. 101(2):725–734, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-016-7942-5">[DOI]</a>.</p></li>
<li id="id29279"><p>Wanwipa Vongsangnak, Amornpan Klanchui, Iyarest Tawornsamretkit, Witthawin Tatiyaborwornchai, Kobkul Laoteng, and Asawin Meechai. Genome-scale metabolic modeling of mucor circinelloides and comparative analysis with other oleaginous species. 583(2):121–129, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gene.2016.02.028">[DOI]</a>.</p></li>
<li id="id29280"><p>Emrah Özcan and Tunahan Çakır. Reconstructed metabolic network models predict flux-level metabolic reprogramming in glioblastoma. 10():156–156, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fnins.2016.00156">[DOI]</a>.</p></li>
<li id="id29286"><p>Douglas McCloskey, Jamey D. Young, Sibei Xu, Bernhard Ø. Palsson, and Adam M. Feist. Modeling method for increased precision and scope of directly measurable fluxes at a genome-scale. 88(7):3844–3852, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/acs.analchem.5b04914">[DOI]</a>.</p></li>
<li id="id29291"><p>Paula Jouhten, Tomasz Boruta, Sergej Andrejev, Filipa Pereira, Isabel Rocha, and Kiran Raosaheb Patil. Yeast metabolic chassis designs for diverse biotechnological products. 6(1):29694–29694, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep29694">[DOI]</a>.</p></li>
<li id="id29294"><p>Xi Wang, Xiaochao Xiong, Na Sa, Sanja Roje, and Shulin Chen. Metabolic engineering of enhanced glycerol-3-phosphate synthesis to increase lipid production in synechocystis sp. pcc 6803. 100(13):6091–6101, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-016-7521-9">[DOI]</a>.</p></li>
<li id="id29302"><p>Daniel Hartleb, Florian Jarre, and Martin J. Lercher. Improved metabolic models for e. coli and mycoplasma genitalium from globalfit, an algorithm that simultaneously matches growth and non-growth data sets. 12(8):e1005036–e1005036, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005036">[DOI]</a>.</p></li>
<li id="id29313"><p>Ruben Heck, Mathias Ganter, Vítor A. P. Martins dos Santos, and Joerg Stelling. Efficient reconstruction of predictive consensus metabolic network models. 12(8):e1005085–e1005085, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1005085">[DOI]</a>.</p></li>
<li id="id29314"><p>Lanqing Dang, Jiao Liu, Cheng Wang, Huanhuan Liu, and Jianping Wen. Enhancement of rapamycin production by metabolic engineering in &lt;i&gt;streptomyces hygroscopicus&lt;/i&gt; based on genome-scale metabolic model. 44(2):259–270, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-016-1880-1">[DOI]</a>.</p></li>
<li id="id29356"><p>Zhihao Wang, Siu Hung Joshua Chan, Suresh Sudarsan, Lars M. Blank, Peter Ruhdal Jensen, and Christian Solem. Elucidation of the regulatory role of the fructose operon reveals a novel target for enhancing the nadph supply in corynebacterium glutamicum. 38():344–357, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2016.08.004">[DOI]</a>.</p></li>
<li id="id29362"><p>Xingxing Jian, Shengguo Zhou, Cheng Zhang, and Qiang Hua. In silico identification of gene amplification targets based on analysis of production and growth coupling. 145():1–8, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2016.05.002">[DOI]</a>.</p></li>
<li id="id29370"><p>Kristopher A. Hunt, Ryan deM. Jennings, William P. Inskeep, and Ross P. Carlson. Stoichiometric modelling of assimilatory and dissimilatory biomass utilisation in a microbial community. 18(12):4946–4960, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.13444">[DOI]</a>.</p></li>
<li id="id29381"><p>Yunze Zhang, Huxing Chen, Jiamei Liu, Ge Geng, Deli Liu, Hui Geng, and Xiong Li. Genome sequencing and biodegradation characteristics of the n-butyl benzyl phthalate degrading bacterium-rhodococcus sp. hs-d2. 128():56–62, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ibiod.2016.08.024">[DOI]</a>.</p></li>
<li id="id29382"><p>R. Adam Thompson, Sanjeev Dahal, Sergio Garcia, Intawat Nookaew, and Cong T. Trinh. Exploring complex cellular phenotypes and model-guided strain design with a novel genome-scale metabolic model of clostridium thermocellum dsm 1313 implementing an adjustable cellulosome. 9(1):194–194, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13068-016-0607-x">[DOI]</a>.</p></li>
<li id="id29389"><p>Hulda S. Haraldsdóttir and Ronan M. T. Fleming. Identification of conserved moieties in metabolic networks by graph theoretical analysis of atom transition networks. 12(11):e1004999–e1004999, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004999">[DOI]</a>.</p></li>
<li id="id29396"><p>Ehsan Motamedian, M. Saeidi, and Seyed Abbas Shojaosadati. Reconstruction of a charge balanced genome-scale metabolic model to study the energy-uncoupled growth of zymomonas mobilis zm1. 12(4):1241–1249, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00588d">[DOI]</a>.</p></li>
<li id="id29397"><p>Fredrick M. Mobegi, Aldert Zomer, Marien I. de Jonge, and Sacha A. F. T. van Hijum. Advances and perspectives in computational prediction of microbial gene essentiality. 16(2):70–79, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bfgp/elv063">[DOI]</a>.</p></li>
<li id="id29426"><p>Xiaoyan Yang, Qianqian Yuan, Yangyang Zheng, Hongwu Ma, Tao Chen, and Xueming Zhao. An engineered non-oxidative glycolysis pathway for acetone production in escherichia coli. 38(8):1359–1365, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-016-2115-2">[DOI]</a>.</p></li>
<li id="id29427"><p>Reza Mohammadi, Jalil Fallah-Mehrabadi, Gholamreza Bidkhori, Javad Zahiri, Mohammad Javad Niroomand, and Ali Masoudi‐Nejad. A systems biology approach to reconcile metabolic network models with application to synechocystis sp. pcc 6803 for biofuel production. 12(8):2552–2561, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c6mb00119j">[DOI]</a>.</p></li>
<li id="id29443"><p>Jun Feng, Jing Yang, Xiaorong Li, Meijin Guo, Bochu Wang, Shang‐Tian Yang, and Xiang Zou. Reconstruction of a genome-scale metabolic model and in silico analysis of the polymalic acid producer aureobasidium pullulans cctcc m2012223. 607():1–8, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gene.2016.12.034">[DOI]</a>.</p></li>
<li id="id29457"><p>Pablo Carbonell. Hepatotoxicity prediction by systems biology modeling of disturbed metabolic pathways using gene expression data. 34(2):219–234, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.14573/altex.1602071">[DOI]</a>.</p></li>
<li id="id29458"><p>Marco Fondi, Emanuele Bosi, Luana Presta, Diletta Natoli, and Renato Fani. Modelling microbial metabolic rewiring during growth in a complex medium. 17(1):970–970, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-016-3311-0">[DOI]</a>.</p></li>
<li id="id29468"><p>Ronan M. T. Fleming, Nikos Vlassis, Ines Thiele, and Michael A. Saunders. Conditions for duality between fluxes and concentrations in biochemical networks. 409():1–10, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtbi.2016.06.033">[DOI]</a>.</p></li>
<li id="id29478"><p>Salma Sohrabi-Jahromi, Sayed‐Amir Marashi, and Shiva Kalantari. A kidney-specific genome-scale metabolic network model for analyzing focal segmental glomerulosclerosis. 27(3-4):158–167, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00335-016-9622-2">[DOI]</a>.</p></li>
<li id="id29519"><p>Luis Tobalina, Jon Pey, and Francisco J. Planes. Direct calculation of minimal cut sets involving a specific reaction knock-out. 32(13):2001–2007, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btw072">[DOI]</a>.</p></li>
<li id="id29532"><p>Xingxing Jian, Ningchuan Li, Cheng Zhang, and Qiang Hua. In silico profiling of cell growth and succinate production in escherichia coli nzn111. 3(1):48–48, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40643-016-0125-5">[DOI]</a>.</p></li>
<li id="id29626"><p>Amornpan Klanchui, Nachon Raethong, Peerada Prommeenate, Wanwipa Vongsangnak, and Asawin Meechai. Cyanobacterial biofuels: strategies and developments on network and modeling. 160():75–102, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2016_42">[DOI]</a>.</p></li>
<li id="id29627"><p>Jason Whitham, Mark J. Schulte, Benjamin G. Bobay, José M. Bruno-Bárcena, Mari S. Chinn, Michael C. Flickinger, Joel J. Pawlak, and Amy M. Grunden. Characterization of clostridium ljungdahlii ota1: a non-autotrophic hyper ethanol-producing strain. 101(4):1615–1630, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-016-7978-6">[DOI]</a>.</p></li>
<li id="id29628"><p>Milan JA van Hoek and Roeland M. H. Merks. Emergence of microbial diversity due to cross-feeding interactions in a spatial model of gut microbial metabolism. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/059956">[DOI]</a>.</p></li>
<li id="id29650"><p>Lili Liu, Zijun Zhang, Taotao Sheng, and Ming Chen. Def: an automated dead-end filling approach based on quasi-endosymbiosis. 33(3):405–413, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btw604">[DOI]</a>.</p></li>
<li id="id29651"><p>André Schultz, Sudhir Mehta, Chuanbo Hu, Fieke W. Hoff, Terzah M. Horton, Steven M. Kornblau, and Amina A. Qutub. Identifying cancer specific metabolic signatures using constraint-based models. 22():485–496, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/9789813207813_0045">[DOI]</a>.</p></li>
<li id="id29662"><p>Cristiana G. O. Dal’Molin and Lars K. Nielsen. Algae genome-scale reconstruction, modelling and applications. ():591–598, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-24945-2_22">[DOI]</a>.</p></li>
<li id="id29663"><p>Regiane Kawasaki, Rafael A. Baraúna, Artur Silva, Marta S. P. Carepo, Rui Oliveira, Rodolfo Marques, Rommel Thiago Jucá Ramos, and Maria Paula Cruz Schneider. Reconstruction of the fatty acid biosynthetic pathway of&lt;i&gt;exiguobacterium antarcticum&lt;/i&gt;b7 based on genomic and bibliomic data. 2016():1–9, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2016/7863706">[DOI]</a>.</p></li>
<li id="id29707"><p>Sanjeev Dahal, Suresh Poudel, and R. Adam Thompson. Genome-scale modeling of thermophilic microorganisms. 160():103–119, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2016_45">[DOI]</a>.</p></li>
<li id="id29710"><p>Semidán Robaina‐Estévez and Zoran Nikoloski. Metabolic network constrains gene regulation of c&lt;sub&gt;4&lt;/sub&gt;photosynthesis: the case of maize. 57(5):933–943, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/pcp/pcw034">[DOI]</a>.</p></li>
<li id="id29714"><p>Do‐Gyun Kim, Byoung‐Kwan Cho, and Wang‐Hee Lee. A novel approach in analyzing agriculture and food systems: review of modeling and its applications. 43(2):163–175, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.7744/kjoas.20160019">[DOI]</a>.</p></li>
<li id="id29715"><p>Xudong Wu and Guohui Li. Prevalent accumulation of non-optimal codons through somatic mutations in human cancers. 11(8):e0160463–e0160463, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0160463">[DOI]</a>.</p></li>
<li id="id29716"><p>Meghana Chitale, Ishita Khan, and Daisuke Kihara. Missing gene identification using functional coherence scores. 6(1):31725–31725, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep31725">[DOI]</a>.</p></li>
<li id="id29717"><p>Donghyuk Kim, Sang Woo Seo, Hojung Nam, Gabriela I. Guzmán, Ye Gao, and Bernhard Ø. Palsson. Systems assessment of transcriptional regulation on central carbon metabolism by cra and crp. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/080929">[DOI]</a>.</p></li>
<li id="id29718"><p>Balaji Balagurunathan, Vishist Kumar Jain, Crystal Jing Ying Tear, Chanyuen Lim, and Hua Zhao. In silico design of anaerobic growth-coupled product formation in escherichia coli: experimental validation using a simple polyol, glycerol. 40(3):361–372, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-016-1703-9">[DOI]</a>.</p></li>
<li id="id29727"><p>Maike K. Aurich, Ronan M. T. Fleming, and Ines Thiele. Metabotools: a comprehensive toolbox for analysis of genome-scale metabolic models. 7():327–327, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2016.00327">[DOI]</a>.</p></li>
<li id="id29739"><p>Thomas Pfau, Nils Christian, Shyam Kumar Masakapalli, Lee Sweetlove, Mark G. Poolman, and Oliver Ebenhöh. The intertwined metabolism of&lt;i&gt;medicago truncatula&lt;/i&gt;and its nitrogen fixing symbiont&lt;i&gt;sinorhizobium meliloti&lt;/i&gt;elucidated by genome-scale metabolic models. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/067348">[DOI]</a>.</p></li>
<li id="id29752"><p>Wouter Daniels, Jeroen Bouvin, Tobias Busche, Jörn Kalinowski, and Kristel Bernaerts. Finding targets for genome reduction in streptomyces lividans tk24 using flux balance analysis. 49(26):252–257, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ifacol.2016.12.134">[DOI]</a>.</p></li>
<li id="id29760"><p>Kevin Schwahn, Anika Küken, Daniel J. Kliebenstein, Alisdair R. Fernie, and Zoran Nikoloski. Observability of plant metabolic networks is reflected in the correlation of metabolic profiles. 172(2):pp.00900.2016–pp.00900.2016, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.16.00900">[DOI]</a>.</p></li>
<li id="id29794"><p>Cameron Cotten and Jennifer L. Reed. Applications of constraint-based models for biochemical production. ():201–226, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-444-63475-7.00008-x">[DOI]</a>.</p></li>
<li id="id29801"><p>Amornpan Klanchui, Wanwipa Vongsangnak, Kobkul Laoteng, and Asawin Meechai. In silico analysis of mucor circinelloides genome-scale model for enhancing lipid production. ():14–18, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/3029375.3029383">[DOI]</a>.</p></li>
<li id="id29818"><p>Fangzhou Shen, Jian Li, Ying Zhu, and Zhuo Wang. Systematic investigation of metabolic reprogramming in different cancers based on tissue-specific metabolic models. 14(05):1644001–1644001, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0219720016440017">[DOI]</a>.</p></li>
<li id="id29827"><p>Neha Chaudhary, Kristin Tøndel, Rakesh Bhatnagar, Vítor A. P. Martins dos Santos, and Jacek Puchałka. Characterizing the optimal flux space of genome-scale metabolic reconstructions through modified latin-hypercube sampling. 12(3):994–1005, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00457h">[DOI]</a>.</p></li>
<li id="id29850"><p>R.P. Vivek-Ananth and Areejit Samal. Advances in the integration of transcriptional regulatory information into genome-scale metabolic models. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/053520">[DOI]</a>.</p></li>
<li id="id29853"><p>Zhuo Wang, Samuel A. Danziger, Ben Heavner, Shuyi Ma, Jennifer J. Smith, Song Li, Thurston Herricks, Evangelos Simeonidis, Nitin S. Baliga, John D. Aitchison, and Nathan D. Price. Combining inferred regulatory and reconstructed metabolic networks enhances phenotype prediction in yeast. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/087148">[DOI]</a>.</p></li>
<li id="id29880"><p>Meng Liang, Xiaowei Zhou, and Chunping Xu. Systems biology in biofuel. 1(11):, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.1515/psr-2016-0047">[DOI]</a>.</p></li>
<li id="id29910"><p>Alberto Noronha, Anna Dröfn Daníelsdóttir, Freyr Jóhannsson, Soffía Jónsdóttir, Sindri Jarlsson, Jón Pétur Gunnarsson, Sigur dhur Brynjólfsson, Piotr Gawron, Reinhard Schneider, Ines Thiele, and Ronan M. T. Fleming. Reconmap: an interactive visualisation of human metabolism. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1606.00042">[DOI]</a>.</p></li>
<li id="id29979"><p>Ding Ma, Laurence Yang, Ronan M. T. Fleming, Ines Thiele, Bernhard Ø. Palsson, and Michael A. Saunders. Quadruple-precision solution of genome-scale models of metabolism and macromolecular expression. ():, 2016. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id29980"><p>R. Adam Thompson. In silico driven metabolic engineering towards enhancing biofuel and biochemical production. ():, 2016. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id29981"><p>Morales Pérez and Yeimy Liceth. Constraint-based metabolic models and their application in industrial biotechnology. ():, 2016. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id29982"><p>Jesse R. Walsh. Computational methods for integrated analysis of omics and pathway data. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.31274/etd-180810-4832">[DOI]</a>.</p></li>
<li id="id30013"><p>Ding Ma, Laurence Yang, Ronan M. T. Fleming, Ines Thiele, Bernhard Ø. Palsson, and Michael A. Saunders. Reliable and efficient solution of genome-scale models of metabolism and macromolecular expression. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1606.00054">[DOI]</a>.</p></li>
<li id="id30047"><p>Edik M. Blais. Reconciled rat and human metabolic networks for comparative toxicogenomics analyses. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.18130/v33s2x">[DOI]</a>.</p></li>
<li id="id30053"><p>Agris Pentjuss. Stoichiometric models creation and analysis procedures for metabolic engineering = stehiometrisko modeļu izveides un analīzes procedūras metaboliskajai inženierijai. ():, 2016. <a class="reference external" href="https://doi.org/https://doi.org/10.22616/lludissert/2016.004">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-10" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-10" name="0-10" role="tabpanel" tabindex="0"><div class="docutils container" id="id31211">
<ol class="arabic simple" start="1">
<li id="id31249"><p>Sheng Ren, Anna A. Hinzman, Emily L. Kang, Rhonda D. Szczesniak, and Long Lu. Computational and statistical analysis of metabolomics data. 11(6):1492–1513, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11306-015-0823-6">[DOI]</a>.</p></li>
<li id="id31273"><p>Goldy De Bhowmick, Lokanand Koduru, and Ramkrishna Sen. Metabolic pathway engineering towards enhancing microalgal lipid biosynthesis for biofuel application—a review. 50():1239–1253, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.rser.2015.04.131">[DOI]</a>.</p></li>
<li id="id31288"><p>Matthew B. Biggs, Gregory L. Medlock, Glynis L. Kolling, and Jason A. Papin. Metabolic network modeling of microbial communities. 7(5):317–334, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/wsbm.1308">[DOI]</a>.</p></li>
<li id="id31303"><p>Evangelos Simeonidis and Nathan D. Price. Genome-scale modeling for metabolic engineering. 42(3):327–338, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-014-1576-3">[DOI]</a>.</p></li>
<li id="id31311"><p>Philipp N. Spahn, Anders Holmgaard Hansen, Henning Gram Hansen, Johnny Arnsdorf, Helene Faustrup Kildegaard, and Nathan E. Lewis. A markov chain model for n-linked protein glycosylation – towards a low-parameter tool for model-driven glycoengineering. 33():52–66, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2015.10.007">[DOI]</a>.</p></li>
<li id="id31373"><p>Shuyi Ma, K Minch, Tige R. Rustad, Samuel J. Hobbs, Suk-Lin Zhou, David R. Sherman, and Nathan D. Price. Integrated modeling of gene regulatory and metabolic networks in mycobacterium tuberculosis. 11(11):e1004543–e1004543, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004543">[DOI]</a>.</p></li>
<li id="id31480"><p>John Varga, Mariette Barbier, Xavier Mulet, Piotr Bielecki, Jennifer Bartell, Joshua P. Owings, Inmaculada Martínez-Ramos, Lauren Hittle, Michael R. Davis, F. Heath Damron, George W. Liechti, Jacek Puchałka, Vítor A. P. Martins dos Santos, Robert K. Ernst, Jason A. Papin, Sebastián Albertí, Antonio Oliver, and Joanna B. Goldberg. Genotypic and phenotypic analyses of a pseudomonas aeruginosa chronic bronchiectasis isolate reveal differences from cystic fibrosis and laboratory strains. 16(1):883–883, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-015-2069-0">[DOI]</a>.</p></li>
<li id="id31516"><p>Lisa Maria Dersch, Véronique Beckers, and Christoph Wittmann. Green pathways: metabolic network analysis of plant systems. 34():1–24, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2015.12.001">[DOI]</a>.</p></li>
<li id="id31532"><p>Andrew Damiani, Q. Peter He, Thomas W. Jeffries, and Jin Wang. Comprehensive evaluation of two genome‐scale metabolic network models for &lt;i&gt;scheffersomyces stipitis&lt;/i&gt;. 112(6):1250–1262, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25535">[DOI]</a>.</p></li>
<li id="id31541"><p>Parizad Babaei, Sayed‐Amir Marashi, and Sedigheh Asad. Genome-scale reconstruction of the metabolic network in pseudomonas stutzeri a1501. 11(11):3022–3032, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00086f">[DOI]</a>.</p></li>
<li id="id31575"><p>Devin H. Currie, Babu Raman, Christopher M. Gowen, Timothy J. Tschaplinski, Miriam Land, Steven D. Brown, Sean F. Covalla, Dawn M. Klingeman, Zamin K. Yang, Nancy L. Engle, Courtney M. Johnson, Miguel Rodríguez, A. Joe Shaw, William R. Kenealy, Lee R. Lynd, Stephen S. Fong, Jonathan R. Mielenz, Brian H. Davison, David A. Hogsett, and Christopher D. Herring. Genome-scale resources for thermoanaerobacterium saccharolyticum. 9(1):30–30, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0159-x">[DOI]</a>.</p></li>
<li id="id31580"><p>Mohammad Tajparast and Dominic Frigon. Genome-scale metabolic model of rhodococcus jostii rha1 (imt1174) to study the accumulation of storage compounds during nitrogen-limited condition. 9(1):43–43, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0190-y">[DOI]</a>.</p></li>
<li id="id31587"><p>Diego Salazar, Alexander Rodríguez‐López, Angélica María Herreño, Hector Barbosa, Juliana Herrera, Andrea Ardila, George E. Barreto, Janneth González, and Carlos Javier Alméciga-Díaz. Systems biology study of mucopolysaccharidosis using a human metabolic reconstruction network. 117(2):129–139, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymgme.2015.08.001">[DOI]</a>.</p></li>
<li id="id31600"><p>Anna‐Lena Hillje, Elisabeth Beckmann, Maria Angeliki S. Pavlou, Christian Jaeger, Maria Pires Pacheco, Thomas Sauter, Jens C. Schwamborn, and Lars Lewejohann. The neural stem cell fate determinant trim32 regulates complex behavioral traits. 9():75–75, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fncel.2015.00075">[DOI]</a>.</p></li>
<li id="id31605"><p>Hamideh Fouladiha, Sayed‐Amir Marashi, and Mohammad Ali Shokrgozar. Reconstruction and validation of a constraint‐based metabolic network model for bone marrow‐derived mesenchymal stem cells. 48(4):475–485, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/cpr.12197">[DOI]</a>.</p></li>
<li id="id31627"><p>Zahra Azimzadeh Irani, Amir Maghsoudi, Seyed Abbas Shojaosadati, and Ehsan Motamedian. Development and in silico analysis of a new nitrogen-limited feeding strategy for fed-batch cultures of pichia pastoris based on a simple ph-control system. 98():1–9, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2015.02.016">[DOI]</a>.</p></li>
<li id="id31629"><p>Carolina A. Contador, V. Rodríguez, Bárbara A. Andrews, and Juan A. Asenjo. Genome-scale reconstruction of salinispora tropica cnb-440 metabolism to study strain-specific adaptation. 108(5):1075–1090, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10482-015-0561-9">[DOI]</a>.</p></li>
<li id="id31677"><p>Hailin Meng, Zhiqiang Xiong, Shujie Song, Jianfeng Wang, and Yong Wang. Construction of polyketide overproducing &lt;i&gt;escherichia coli&lt;/i&gt; strains via synthetic antisense rnas based on in silico fluxome analysis and comparative transcriptome analysis. 11(4):530–541, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201500351">[DOI]</a>.</p></li>
<li id="id31680"><p>Michael Binns, Pedro de Atauri, Anestis Vlysidis, Marta Cascante, and Constantinos Theodoropoulos. Sampling with poling-based flux balance analysis: optimal versus sub-optimal flux space analysis of actinobacillus succinogenes. 16(1):49–49, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-015-0476-5">[DOI]</a>.</p></li>
<li id="id31682"><p>Javier Ferrer Valenzuela, Luis Andrés Pinuer, Apolinaria García, and Rodrigo Bórquez. Metabolic fluxes in lactic acid bacteria—a review. 29(2):185–217, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/08905436.2015.1027913">[DOI]</a>.</p></li>
<li id="id31684"><p>Franziska Erlmeier, Annette Feuchtinger, Daniela Borgmann, Martina Rudelius, Michael Autenrieth, Axel Walch, and Gregor Weirich. Supremacy of modern morphometry in typing renal oncocytoma and malignant look-alikes. 144(2):147–156, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00418-015-1324-4">[DOI]</a>.</p></li>
<li id="id31690"><p>Sayed‐Rzgar Hosseini, Aditya Barve, and Andreas Wagner. Exhaustive analysis of a genotype space comprising 1015 central carbon metabolisms reveals an organization conducive to metabolic innovation. 11(8):e1004329–e1004329, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004329">[DOI]</a>.</p></li>
<li id="id31691"><p>Lisha K. Parambil and Debasis Sarkar. &lt;i&gt;in silico&lt;/i&gt; analysis of bioethanol overproduction by genetically modified microorganisms in coculture fermentation. 2015():1–11, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2015/238082">[DOI]</a>.</p></li>
<li id="id31693"><p>Bin Wang, Qianji Ning, Tong Hao, Ailing Yu, and Jinsheng Sun. Reconstruction and analysis of a genome-scale metabolic model for eriocheir sinensis eyestalks. 12(1):246–252, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00571j">[DOI]</a>.</p></li>
<li id="id31695"><p>Lian He, Stephen Gang Wu, Ni Wan, Adrienne C. Reding, and Yinjie Tang. Simulating cyanobacterial phenotypes by integrating flux balance analysis, kinetics, and a light distribution function. 14(1):206–206, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-015-0396-0">[DOI]</a>.</p></li>
<li id="id31698"><p>Carolina A. Contador, Carolina Shene, A. Olivera, Yasuo Yoshikuni, Alejandro H. Buschmann, Bárbara A. Andrews, and Juan A. Asenjo. Analyzing redox balance in a synthetic yeast platform to improve utilization of brown macroalgae as feedstock. 2():76–84, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2015.06.004">[DOI]</a>.</p></li>
<li id="id31707"><p>Bashir Sajo Mienda and Mohd Shahir Shamsir. &lt;i&gt;in silico&lt;/i&gt;deletion of&lt;i&gt;ptsg&lt;/i&gt;gene in&lt;i&gt;escherichia coli&lt;/i&gt;genome-scale model predicts increased succinate production from glycerol. 33(11):2380–2389, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/07391102.2015.1036461">[DOI]</a>.</p></li>
<li id="id31770"><p>Michael P. Weiner, Julia Tröndle, Christoph Albermann, Georg A. Sprenger, and Dirk Weuster‐Botz. Perturbation experiments: approaches for metabolic pathway analysis in bioreactors. 152():91–136, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2015_326">[DOI]</a>.</p></li>
<li id="id31784"><p>Kourosh Salehi‐Ashtiani, Joseph Koussa, Bushra Dohai, Amphun Chaiboonchoe, Hong Cai, Kelly Dougherty, David R. Nelson, Kenan Jijakli, and Basel Khraiwesh. Toward applications of genomics and metabolic modeling to improve algal biomass productivity. ():173–189, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-16640-7_10">[DOI]</a>.</p></li>
<li id="id31789"><p>Brandon Barker, Lin Xu, and Zhenglong Gu. Dynamic epistasis under varying environmental perturbations. 10(1):e0114911–e0114911, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0114911">[DOI]</a>.</p></li>
<li id="id31794"><p>Katja Tummler, Clemens Kühn, and Edda Klipp. Dynamic metabolic models in context: biomass backtracking. 7(8):940–951, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5ib00050e">[DOI]</a>.</p></li>
<li id="id31805"><p>María Antonia Zamora, Andrés Pinzón, María Mercedes Zambrano, Silvia Restrepo, Linda J. Broadbelt, Matthew Moura, Johana Husserl, and Andrés Fernando González Barrios. A comparison between functional frequency and metabolic flows framed by biogeochemical cycles in metagenomes: the case of “el coquito” hot spring located at colombia's national nevados park. 313():259–265, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ecolmodel.2015.06.041">[DOI]</a>.</p></li>
<li id="id31847"><p>David Hala, Lene H. Petersen, Dalma Martinović‐Weigelt, and Duane B. Huggett. &lt;i&gt;in silico&lt;/i&gt;analysis of perturbed steroidogenesis and gonad growth in fathead minnows (&lt;i&gt;p. promelas&lt;/i&gt;) exposed to 17&lt;b&gt;α&lt;/b&gt;-ethynylestradiol. 61(3):122–138, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3109/19396368.2015.1035817">[DOI]</a>.</p></li>
<li id="id31883"><p>Joachim von Wulffen, Patrick C. F. Buchholz, Oliver Sawodny, and Ronny Feuer. Modeling the metabolism of escherichia coli under oxygen gradients with dynamically changing flux bounds. 18():1–6, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/bibe.2015.7367691">[DOI]</a>.</p></li>
<li id="id31905"><p>Jennifer Bartell. Comparative systems analysis of opportunistic gram-negative pathogens. ():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.18130/v3726f">[DOI]</a>.</p></li>
<li id="id31918"><p>Bashir Sajo Mienda and Mohd Shahir Shamsir. Model-guided in silico overexpression of adhe gene predicts increased ethanol production in escherichia coli from xylose. 3(3):39–43, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.13189/bb.2015.030302">[DOI]</a>.</p></li>
<li id="id31925"><p>Stephen Lincoln, Ian S. Rogers, and Ranjan Srivastava. Metabolic design and engineering through ant colony optimization. ():225–232, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/2739480.2754817">[DOI]</a>.</p></li>
<li id="id31951"><p>Claudio Angione, Giovanni Carapezza, Jole Costanza, Píetro Lió, and Giuseppe Nicosia. Design and strain selection criteria for bacterial communication networks. ():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/013383">[DOI]</a>.</p></li>
<li id="id31999"><p>Zachary A. King, Justin S. Lu, Andreas Dräger, Philip Miller, Stephen Federowicz, Joshua A. Lerman, Ali Ebrahim, Bernhard Ø. Palsson, and Nathan E. Lewis. Bigg models: a platform for integrating, standardizing and sharing genome-scale models. 44(D1):D515–D522, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkv1049">[DOI]</a>.</p></li>
<li id="id32001"><p>Edward J. O’Brien, Jonathan M. Monk, and Bernhard Ø. Palsson. Using genome-scale models to predict biological capabilities. 161(5):971–987, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cell.2015.05.019">[DOI]</a>.</p></li>
<li id="id32019"><p>Peter D. Karp, Mario Latendresse, Suzanne Paley, Markus Krummenacker, Quang Ong, Richard Billington, Anamika Kothari, Daniel Weaver, Thomas Lee, Pallavi Subhraveti, Aaron Spaulding, Carol A. Fulcher, Ingrid M. Keseler, and Ron Caspi. Pathway tools version 19.0 update: software for pathway/genome informatics and systems biology. 17(5):877–890, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbv079">[DOI]</a>.</p></li>
<li id="id32038"><p>Douglas B. Kell, Neil Swainston, Pınar Pir, and Stephen G. Oliver. Membrane transporter engineering in industrial biotechnology and whole cell biocatalysis. 33(4):237–246, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2015.02.001">[DOI]</a>.</p></li>
<li id="id32057"><p>Luca Gerosa, Bart Rudolf Boudewijn Haverkorn van Rijsewijk, Dimitris Christodoulou, Karl Kochanowski, Thomas Schmidt, Εlad Noor, and Uwe Sauer. Pseudo-transition analysis identifies the key regulators of dynamic metabolic adaptations from steady-state data. 1(4):270–282, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2015.09.008">[DOI]</a>.</p></li>
<li id="id32074"><p>Martin Kavšček, Govindprasad Bhutada, Tobias Madl, and Klaus Natter. Optimization of lipid production with a genome-scale model of yarrowia lipolytica. 9(1):72–72, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0217-4">[DOI]</a>.</p></li>
<li id="id32076"><p>Nicholas D. Gold, Christopher M. Gowen, François-Xavier Lussier, Sarat C. Cautha, Radhakrishnan Mahadevan, and Vincent J. J. Martin. Metabolic engineering of a tyrosine-overproducing yeast platform using targeted metabolomics. 14(1):73–73, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-015-0252-2">[DOI]</a>.</p></li>
<li id="id32077"><p>Almut Heinken and Ines Thiele. Anoxic conditions promote species-specific mutualism between gut microbes &lt;i&gt;in silico&lt;/i&gt;. 81(12):4049–4061, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.00101-15">[DOI]</a>.</p></li>
<li id="id32091"><p>Cristiana Gomes de Oliveira Dal’Molin, Lake‐Ee Quek, Pedro A. Saa, and Lars K. Nielsen. A multi-tissue genome-scale metabolic modeling framework for the analysis of whole plant systems. 6():4–4, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2015.00004">[DOI]</a>.</p></li>
<li id="id32099"><p>Erik M. Quandt, Jimmy Gollihar, Zachary D. Blount, Andrew D. Ellington, George Georgiou, and Jeffrey E. Barrick. Fine-tuning citrate synthase flux potentiates and refines metabolic innovation in the lenski evolution experiment. 4():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.7554/elife.09696">[DOI]</a>.</p></li>
<li id="id32104"><p>Josefine Anfelt, Danuta Kaczmarzyk, Kiyan Shabestary, Björn Renberg, Johan Rockberg, Jens Nielsen, Mathias Uhlén, and Elton P. Hudson. Genetic and nutrient modulation of acetyl-coa levels in synechocystis for n-butanol production. 14(1):167–167, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-015-0355-9">[DOI]</a>.</p></li>
<li id="id32112"><p>Almut Heinken and Ines Thiele. Systematic prediction of health-relevant human-microbial co-metabolism through a computational framework. 6(2):120–130, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/19490976.2015.1023494">[DOI]</a>.</p></li>
<li id="id32117"><p>Aarash Bordbar, Douglas McCloskey, Daniel C. Zielinski, Nikolaus Sonnenschein, Neema Jamshidi, and Bernhard Ø. Palsson. Personalized whole-cell kinetic models of metabolism for discovery in genomics and pharmacodynamics. 1(4):283–292, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cels.2015.10.003">[DOI]</a>.</p></li>
<li id="id32137"><p>Haiqin Chen, Guangfei Hao, Lei Wang, Hongchao Wang, Zhennan Gu, Li Liu, Hao Zhang, Wei Chen, and Yong Q. Chen. Identification of a critical determinant that enables efficient fatty acid synthesis in oleaginous fungi. 5(1):11247–11247, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep11247">[DOI]</a>.</p></li>
<li id="id32138"><p>Almut Heinken and Ines Thiele. Systems biology of host–microbe metabolomics. 7(4):195–219, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/wsbm.1301">[DOI]</a>.</p></li>
<li id="id32152"><p>Jérôme Izard, Cindy DC Gomez Balderas, Delphine Ropers, Stéphan Lacour, Xiaohu Song, Yifan Yang, Ariel B. Lindner, Johannes Geiselmann, and Hidde de Jong. A synthetic growth switch based on controlled expression of rna polymerase. 11(11):840–840, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20156382">[DOI]</a>.</p></li>
<li id="id32157"><p>Aarthi Ravikrishnan and Karthik Raman. Critical assessment of genome-scale metabolic networks: the need for a unified standard. 16(6):1057–1068, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbv003">[DOI]</a>.</p></li>
<li id="id32172"><p>Ben Heavner and Nathan D. Price. Comparative analysis of yeast metabolic network models highlights progress, opportunities for metabolic reconstruction. 11(11):e1004530–e1004530, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004530">[DOI]</a>.</p></li>
<li id="id32182"><p>Héctor García Martín, Vinay Kumar, Daniel Weaver, Amit Ghosh, Victor Chubukov, Aindrila Mukhopadhyay, Adam P. Arkin, and Jay D. Keasling. A method to constrain genome-scale models with 13c labeling data. 11(9):e1004363–e1004363, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004363">[DOI]</a>.</p></li>
<li id="id32188"><p>Thomas Bayer, Sofia Milker, Thomas Wiesinger, Florian Rudroff, and Marko D. Mihovilovič. Designer microorganisms for optimized redox cascade reactions – challenges and future perspectives. 357(8):1587–1618, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/adsc.201500202">[DOI]</a>.</p></li>
<li id="id32191"><p>Wout Megchelenbrink, Rotem Katzir, Xiaowen Lu, Eytan Ruppin, and Richard A. Notebaart. Synthetic dosage lethality in the human metabolic network is highly predictive of tumor growth and cancer patient survival. 112(39):12217–12222, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1508573112">[DOI]</a>.</p></li>
<li id="id32202"><p>Takeyuki Tamura, Wei Lu, and Tatsuya Akutsu. Computational methods for modification of metabolic networks. 13():376–381, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2015.05.004">[DOI]</a>.</p></li>
<li id="id32213"><p>Christopher J. Petzold, Leanne Jade G. Chan, Melissa Nhan, and Paul D. Adams. Analytics for metabolic engineering. 3():135–135, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2015.00135">[DOI]</a>.</p></li>
<li id="id32214"><p>John A. Cole, Lars Köhler, Jamila Hedhli, and Zaida Luthey‐Schulten. Spatially-resolved metabolic cooperativity within dense bacterial colonies. 9(1):15–15, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0155-1">[DOI]</a>.</p></li>
<li id="id32216"><p>Chao Ye, Nan Xu, Haiqin Chen, Yong Q. Chen, Wei Chen, and Li Liu. Reconstruction and analysis of a genome-scale metabolic model of the oleaginous fungus mortierella alpina. 9(1):1–1, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-014-0137-8">[DOI]</a>.</p></li>
<li id="id32228"><p>Meiyappan Lakshmanan, Sun‐Hyung Lim, Bijayalaxmi Mohanty, Jae Kwang Kim, Sun‐Hwa Ha, and Dong‐Yup Lee. Unraveling the light-specific metabolic and regulatory signatures of rice through combined in silico modeling and multi-omics analysis. 169(4):pp.01379.2015–pp.01379.2015, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.15.01379">[DOI]</a>.</p></li>
<li id="id32229"><p>Quỳnh Anh Nguyễn, Jens Schneider, Gajendar Komati Reddy, and Volker F. Wendisch. Fermentative production of the diamine putrescine: system metabolic engineering of corynebacterium glutamicum. 5(2):211–231, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo5020211">[DOI]</a>.</p></li>
<li id="id32231"><p>Kristin J. Adolfsen and Mark P. Brynildsen. Futile cycling increases sensitivity toward oxidative stress in escherichia coli. 29():26–35, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2015.02.006">[DOI]</a>.</p></li>
<li id="id32233"><p>Aditya Pratapa, Shankar Balachandran, and Karthik Raman. Fast-sl: an efficient algorithm to identify synthetic lethal sets in metabolic networks. 31(20):3299–3305, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btv352">[DOI]</a>.</p></li>
<li id="id32242"><p>Justin Tan, Cristal Zúñiga, and Karsten Zengler. Unraveling interactions in microbial communities - from co-cultures to microbiomes. 53(5):295–305, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12275-015-5060-1">[DOI]</a>.</p></li>
<li id="id32246"><p>Lucas Marmiesse, Rémi Peyraud, and Ludovic Cottret. Flexflux: combining metabolic flux and regulatory network analyses. 9(1):93–93, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0238-z">[DOI]</a>.</p></li>
<li id="id32248"><p>Ali Ebrahim, Eivind Almaas, Eugen Bauer, Aarash Bordbar, Anthony P. Burgard, Roger L. Chang, Andreas Dräger, Iman Famili, Adam M. Feist, Ronan M. T. Fleming, Stephen S. Fong, Vassily Hatzimanikatis, Markus J. Herrgård, Allen Holder, Michael Hucka, Daniel R. Hyduke, Neema Jamshidi, Sang Yup Lee, Nicolas Le Novère, Joshua A. Lerman, Nathan E. Lewis, Ding Ma, Radhakrishnan Mahadevan, Costas D. Maranas, Harish Nagarajan, Ali Navid, Jens Nielsen, Lars K. Nielsen, Juan Nogales, Alberto Noronha, Csaba Pál, Bernhard Ø. Palsson, Jason A. Papin, Kiran Raosaheb Patil, Nathan D. Price, Jennifer L. Reed, Michael A. Saunders, Ryan S. Senger, Nikolaus Sonnenschein, Yuekai Sun, and Ines Thiele. Do genome‐scale models need exact solvers or clearer standards? 11(10):831–831, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20156157">[DOI]</a>.</p></li>
<li id="id32249"><p>Stanislav Kopřiva, Alexander Calderwood, Silke C. Weckopp, and Anna Kopřivová. Plant sulfur and big data. 241():1–10, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.plantsci.2015.09.014">[DOI]</a>.</p></li>
<li id="id32260"><p>Chao Ye, Weihua Qiao, Xiaobin Yu, Xiao‐Jun Ji, He Huang, Jackie L. Collier, and Li Liu. Reconstruction and analysis of the genome-scale metabolic model of schizochytrium limacinum sr21 for docosahexaenoic acid production. 16(1):799–799, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12864-015-2042-y">[DOI]</a>.</p></li>
<li id="id32263"><p>Bianca Habermann, José Villaveces, and Prasanna S. Koti. Tools for visualization and analysis of molecular networks, pathways, and -omics data. 8():11–11, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.2147/aabc.s63534">[DOI]</a>.</p></li>
<li id="id32267"><p>Jae Yong Ryu, Hyun Uk Kim, and Sang Yup Lee. Reconstruction of genome-scale human metabolic models using omics data. 7(8):859–868, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5ib00002e">[DOI]</a>.</p></li>
<li id="id32269"><p>Zhenquan Lin, Yan Zhang, Qianqian Yuan, Qiaojie Liu, Yifan Li, Zhiwen Wang, Hongwu Ma, Tao Chen, and Xueming Zhao. Metabolic engineering of escherichia coli for poly(3-hydroxybutyrate) production via threonine bypass. 14(1):185–185, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12934-015-0369-3">[DOI]</a>.</p></li>
<li id="id32270"><p>Claudio Angione and Píetro Lió. Predictive analytics of environmental adaptability in multi-omic network models. 5(1):26266–26266, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep15147">[DOI]</a>.</p></li>
<li id="id32278"><p>Ruggero La Rosa, Juan Nogales, and Fernando Rojo. The &lt;scp&gt;crc&lt;/scp&gt; / &lt;scp&gt;crcz&lt;/scp&gt; ‐ &lt;scp&gt;crcy&lt;/scp&gt; global regulatory system helps the integration of gluconeogenic and glycolytic metabolism in &lt;scp&gt; &lt;i&gt;p&lt;/i&gt; &lt;/scp&gt; &lt;i&gt;seudomonas putida&lt;/i&gt;. 17(9):3362–3378, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.12812">[DOI]</a>.</p></li>
<li id="id32279"><p>Jahir M. Gutierrez and Nathan E. Lewis. Optimizing eukaryotic cell hosts for protein production through systems biotechnology and genome‐scale modeling. 10(7):939–949, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201400647">[DOI]</a>.</p></li>
<li id="id32308"><p>Eugen Bauer, Cédric C. Laczny, Stefanía Magnúsdóttir, Paul Wilmes, and Ines Thiele. Phenotypic differentiation of gastrointestinal microbes is reflected in their encoded metabolic repertoires. 3(1):35–35, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s40168-015-0121-6">[DOI]</a>.</p></li>
<li id="id32309"><p>Daniel C. Zielinski, Fabian V. Filipp, Aarash Bordbar, Kasper Jensen, Jeffrey W. Smith, Markus J. Herrgård, Monica L. Mo, and Bernhard Ø. Palsson. Pharmacogenomic and clinical data link non-pharmacokinetic metabolic dysregulation to drug side effect pathogenesis. 6(1):7101–7101, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms8101">[DOI]</a>.</p></li>
<li id="id32314"><p>James D. Winkler, Andrea L. Halweg‐Edwards, and Ryan T. Gill. The laser database: formalizing design rules for metabolic engineering. 2():30–38, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meteno.2015.06.003">[DOI]</a>.</p></li>
<li id="id32316"><p>Henning Knoop and Ralf Steuer. A computational analysis of stoichiometric constraints and trade-offs in cyanobacterial biofuel production. 3():47–47, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2015.00047">[DOI]</a>.</p></li>
<li id="id32330"><p>Minsuk Kim, Jeong Sang Yi, Meiyappan Lakshmanan, Dong‐Yup Lee, and Byung‐Gee Kim. Transcriptomics‐based strain optimization tool for designing secondary metabolite overproducing strains of &lt;i&gt;streptomyces coelicolor&lt;/i&gt;. 113(3):651–660, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25830">[DOI]</a>.</p></li>
<li id="id32344"><p>Ratul Chowdhury, Anupam Chowdhury, and Costas D. Maranas. Using gene essentiality and synthetic lethality information to correct yeast and cho cell genome-scale models. 5(4):536–570, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo5040536">[DOI]</a>.</p></li>
<li id="id32350"><p>Maike K. Aurich and Ines Thiele. Computational modeling of human metabolism and its application to systems biomedicine. 1386():253–281, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-3283-2_12">[DOI]</a>.</p></li>
<li id="id32358"><p>M. Ahsanul Islam, Karsten Zengler, Elizabeth A. Edwards, Radhakrishnan Mahadevan, and Gregory Stephanopoulos. Investigating moorella thermoacetica metabolism with a genome-scale constraint-based metabolic model. 7(8):869–882, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5ib00095e">[DOI]</a>.</p></li>
<li id="id32360"><p>Cheng Zhang, Boyang Ji, Adil Mardinoğlu, Jens Nielsen, and Qiang Hua. Logical transformation of genome-scale metabolic models for gene level applications and analysis. 31(14):2324–2331, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btv134">[DOI]</a>.</p></li>
<li id="id32361"><p>Tomáš Zavřel, Henning Knoop, Ralf Steuer, Patrik R. Jones, Jan Červený, and Martin Trtílek. A quantitative evaluation of ethylene production in the recombinant cyanobacterium synechocystis sp. pcc 6803 harboring the ethylene-forming enzyme by membrane inlet mass spectrometry. 202():142–151, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2015.11.062">[DOI]</a>.</p></li>
<li id="id32363"><p>Komi Nambou, Xingxing Jian, Xin-Kai Zhang, Liujing Wei, Jiajia Lou, Catherine Madzak, and Qiang Hua. Flux balance analysis inspired bioprocess upgrading for lycopene production by a metabolically engineered strain of yarrowia lipolytica. 5(4):794–813, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo5040794">[DOI]</a>.</p></li>
<li id="id32384"><p>Jie Mei, Nan Xu, Chao Ye, Li Liu, and Jianrong Wu. Reconstruction and analysis of a genome-scale metabolic network of corynebacterium glutamicum s9114. 575(2):615–622, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gene.2015.09.038">[DOI]</a>.</p></li>
<li id="id32414"><p>Radhakrishnan Mahadevan, Axel von Kamp, and Steffen Klamt. Genome-scale strain designs based on regulatory minimal cut sets. 31(17):2844–2851, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btv217">[DOI]</a>.</p></li>
<li id="id32422"><p>Michael A. Henson. Genome-scale modelling of microbial metabolism with temporal and spatial resolution. 43(6):1164–1171, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bst20150146">[DOI]</a>.</p></li>
<li id="id32440"><p>Susanne Alff-Tuomala, Laura Salusjärvi, Dorothee Barth, Merja Oja, Merja Penttilä, Juha‐Pekka Pitkänen, Laura Ruohonen, and Paula Jouhten. Xylose-induced dynamic effects on metabolism and gene expression in engineered saccharomyces cerevisiae in anaerobic glucose-xylose cultures. 100(2):969–985, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-015-7038-7">[DOI]</a>.</p></li>
<li id="id32458"><p>Abhishek Subramanian, Jitesh Jhawar, and Ram Rup Sarkar. Dissecting leishmania infantum energy metabolism - a systems perspective. 10(9):e0137976–e0137976, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0137976">[DOI]</a>.</p></li>
<li id="id32470"><p>Natalie Stanford, Pierre Millard, and Neil Swainston. Robokod: microbial strain design for (over)production of target compounds. 3():17–17, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fcell.2015.00017">[DOI]</a>.</p></li>
<li id="id32488"><p>Rupa Bhowmick, Abhishek Subramanian, and Ram Rup Sarkar. Exploring the differences in metabolic behavior of astrocyte and glioblastoma: a flux balance analysis approach. 9(4):159–177, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11693-015-9183-9">[DOI]</a>.</p></li>
<li id="id32508"><p>Julian Brandl and Mikael Rørdam Andersen. Current state of genome-scale modeling in filamentous fungi. 37(6):1131–1139, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-015-1782-8">[DOI]</a>.</p></li>
<li id="id32509"><p>Ahmad A. Mannan, Yoshihiro Toya, Kazuyuki Shimizu, Johnjoe McFadden, Andrzej Kierzek, and Andrea Rocco. Integrating kinetic model of e. coli with genome scale metabolic fluxes overcomes its open system problem and reveals bistability in central metabolism. 10(10):e0139507–e0139507, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0139507">[DOI]</a>.</p></li>
<li id="id32513"><p>Joshua J. Hamilton, Montserrat Calixto Contreras, and Jennifer L. Reed. Thermodynamics and h2 transfer in a methanogenic, syntrophic community. 11(7):e1004364–e1004364, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1004364">[DOI]</a>.</p></li>
<li id="id32533"><p>Ehsan Motamedian, Ghazaleh Ghavami, and Soroush Sardari. Investigation on metabolism of cisplatin resistant ovarian cancer using a genome scale metabolic model and microarray data. 18(3):267–76, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id32538"><p>Yanfen Fu, Jong Moon Yoon, Laura R. Jarboe, and Jacqueline V. Shanks. Metabolic flux analysis of escherichia coli mg1655 under octanoic acid (c8) stress. 99(10):4397–4408, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-015-6387-6">[DOI]</a>.</p></li>
<li id="id32543"><p>Brett G. Olivier, Maciej J. Swat, and Martijn J. Moné. Modeling and simulation tools: from systems biology to systems medicine. 1386():441–463, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-3283-2_19">[DOI]</a>.</p></li>
<li id="id32554"><p>Ding Ma and Michael A. Saunders. Solving multiscale linear programs using the simplex method in quadruple precision. ():223–235, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-17689-5_9">[DOI]</a>.</p></li>
<li id="id32563"><p>Thomas Pfau, Maria Pires Pacheco, and Thomas Sauter. Towards improved genome-scale metabolic network reconstructions: unification, transcript specificity and beyond. 17(6):bbv100–bbv100, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbv100">[DOI]</a>.</p></li>
<li id="id32571"><p>Haythem Latif, Merve Şahin, Janna Tarasova, Yekaterina Tarasova, Vasiliy A. Portnoy, Juan Nogales, and Karsten Zengler. Adaptive evolution of thermotoga maritima reveals plasticity of the abc transporter network. 81(16):5477–5485, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.01365-15">[DOI]</a>.</p></li>
<li id="id32577"><p>James T. Yurkovich and Bernhard Ø. Palsson. Solving puzzles with missing pieces: the power of systems biology. 104(1):2–7, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/jproc.2015.2505338">[DOI]</a>.</p></li>
<li id="id32593"><p>Chao Shi, Jian Yin, Zhe Liu, Jianxin Wu, Qi Zhao, Jian Ren, and Nan Yao. A systematic simulation of the effect of salicylic acid on sphingolipid metabolism. 6():186–186, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2015.00186">[DOI]</a>.</p></li>
<li id="id32595"><p>Zhaobin Xu, Sabina Islam, Thomas K. Wood, and Zuyi Huang. An integrated modeling and experimental approach to study the influence of environmental nutrients on biofilm formation of&lt;i&gt;pseudomonas aeruginosa&lt;/i&gt;. 2015():1–12, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2015/506782">[DOI]</a>.</p></li>
<li id="id32612"><p>Elias W. Krumholz and Igor G. L. Libourel. Sequence-based network completion reveals the integrality of missing reactions in metabolic networks. 290(31):19197–19207, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1074/jbc.m114.634121">[DOI]</a>.</p></li>
<li id="id32614"><p>Georg Basler. Computational prediction of essential metabolic genes using constraint-based approaches. 1279():183–204, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-2398-4_12">[DOI]</a>.</p></li>
<li id="id32616"><p>David Knies, Philipp Wittmüß, Sebastian Appel, Oliver Sawodny, Michael Ederer, and Ronny Feuer. Modeling and simulation of optimal resource management during the diurnal cycle in emiliania huxleyi by genome-scale reconstruction and an extended flux balance analysis approach. 5(4):659–676, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo5040659">[DOI]</a>.</p></li>
<li id="id32618"><p>Matthew Moura, Justin D. Finkle, Sarah C. Stainbrook, Jennifer L. Greene, Linda J. Broadbelt, and Keith E. J. Tyo. Evaluating enzymatic synthesis of small molecule drugs. 33():138–147, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2015.11.006">[DOI]</a>.</p></li>
<li id="id32635"><p>Miguel Ponce-de-León, Jorge Calle-Espinosa, Juli Peretό, and Francisco Montero. Consistency analysis of genome-scale models of bacterial metabolism: a metamodel approach. 10(12):e0143626–e0143626, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0143626">[DOI]</a>.</p></li>
<li id="id32639"><p>Abdul Hakim Mohamed Salleh, Mohd Saberi Mohamad, Safaai Deris, Sigeru Omatu, Florentino Fdez‐Riverola, and Juan M. Corchado. Gene knockout identification for metabolite production improvement using a hybrid of genetic ant colony optimization and flux balance analysis. 20(4):685–693, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-015-0276-9">[DOI]</a>.</p></li>
<li id="id32652"><p>Matthew B. Biggs and Jason A. Papin. Metabolic network-guided binning of metagenomic sequence fragments. 32(6):867–874, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btv671">[DOI]</a>.</p></li>
<li id="id32677"><p>Sayed‐Amir Marashi and Zhaleh Hosseini. On correlated reaction sets and coupled reaction sets in metabolic networks. 13(04):1571003–1571003, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0219720015710031">[DOI]</a>.</p></li>
<li id="id32682"><p>Anand Gavai, Farahaniza Supandi, Hannes Hettling, Paul Murrell, Jack A. M. Leunissen, and Hans van Beek. Using bioconductor package biggr for metabolic flux estimation based on gene expression changes in brain. 10(3):e0119016–e0119016, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0119016">[DOI]</a>.</p></li>
<li id="id32685"><p>Hai Shi and Jörg Schwender. Mathematical models of plant metabolism. 37():143–152, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2015.10.008">[DOI]</a>.</p></li>
<li id="id32688"><p>Lin Liu, Fangzhou Shen, Changpeng Xin, and Zhuo Wang. Multi‐scale modeling of &lt;i&gt;arabidopsis thaliana&lt;/i&gt; response to different co&lt;sub&gt;2&lt;/sub&gt; conditions: from gene expression to metabolic flux. 58(1):2–11, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/jipb.12370">[DOI]</a>.</p></li>
<li id="id32694"><p>Feifei Han, Gong‐Hua Li, Shao‐Xing Dai, and Jing‐Fei Huang. Genome-wide metabolic model to improve understanding of cd4&lt;sup&gt;+&lt;/sup&gt;t cell metabolism, immunometabolism and application in drug design. 12(2):431–443, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00480b">[DOI]</a>.</p></li>
<li id="id32699"><p>Karthik Sekar and Keith E. J. Tyo. Regulatory effects on central carbon metabolism from poly-3-hydroxybutryate synthesis. 28():180–189, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2015.01.003">[DOI]</a>.</p></li>
<li id="id32702"><p>Neema Jamshidi and Anu Raghunathan. Cell scale host-pathogen modeling: another branch in the evolution of constraint-based methods. 6():1032–1032, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2015.01032">[DOI]</a>.</p></li>
<li id="id32703"><p>Óttar Rolfsson and Bernhard Ø. Palsson. Decoding the jargon of bottom‐up metabolic systems biology. 37(6):588–591, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bies.201400187">[DOI]</a>.</p></li>
<li id="id32709"><p>Yali Wang, Nan Xu, Chao Ye, Li Liu, Zhongping Shi, and Jing Wu. Reconstruction and in silico analysis of an actinoplanes sp. se50/110 genome-scale metabolic model for acarbose production. 6():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2015.00632">[DOI]</a>.</p></li>
<li id="id32711"><p>Octavio Perez‐Garcia, Kartik Chandran, Silas G. Villas‐Bôas, and Naresh Singhal. Assessment of nitric oxide (no) redox reactions contribution to nitrous oxide (n&lt;sub&gt;2&lt;/sub&gt;o) formation during nitrification using a multispecies metabolic network model. 113(5):1124–1136, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25880">[DOI]</a>.</p></li>
<li id="id32716"><p>Roberto Gallardo, Alejandro Acevedo, Julián Quintero, Iván Paredes, Raúl Conejeros, and Germán Aroca. In silico analysis of clostridium acetobutylicum atcc 824 metabolic response to an external electron supply. 39(2):295–305, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-015-1513-5">[DOI]</a>.</p></li>
<li id="id32722"><p>Tijana Marinković and Matej Orešič. Modeling strategies to study metabolic pathways in progression to type 1 diabetes – challenges and opportunities. 589():131–137, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.abb.2015.08.011">[DOI]</a>.</p></li>
<li id="id32735"><p>Juan Nogales and Lucía Agudo. A practical protocol for integration of transcriptomics data into genome-scale metabolic reconstructions. ():135–152, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/8623_2015_98">[DOI]</a>.</p></li>
<li id="id32746"><p>Leanne S. Whitmore and Ping Ye. Dissecting germ cell metabolism through network modeling. 10(9):e0137607–e0137607, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0137607">[DOI]</a>.</p></li>
<li id="id32754"><p>Longfei Mao, Averina Nicolae, Miguel A. P. Oliveira, Feng He, Siham Hachi, and Ronan M. T. Fleming. A constraint-based modelling approach to metabolic dysfunction in parkinson's disease. 13():484–491, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2015.08.002">[DOI]</a>.</p></li>
<li id="id32755"><p>Suthat Phaiphinit, Sittiporn Pattaradilokrat, Chidchanok Lursinsap, and Kitiporn Plaimas. In silico multiple-targets identification for heme detoxification in the human malaria parasite plasmodium falciparum. 37():237–244, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meegid.2015.11.025">[DOI]</a>.</p></li>
<li id="id32774"><p>Duschak VG. Synthetic biology: computational modeling bridging the gap between in vitro and in vivo reactions. 03(03):, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.4172/2332-0737.1000127">[DOI]</a>.</p></li>
<li id="id32802"><p>Daichi Yukihira, Yoshinori Fujimura, Hiroyuki Wariishi, and Daisuke Miura. Bacterial metabolism in immediate response to nutritional perturbation with temporal and network view of metabolites. 11(9):2473–2482, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5mb00182j">[DOI]</a>.</p></li>
<li id="id32803"><p>Bashir Sajo Mienda and Mohd Shahir Shamsir. Model-driven &amp;lt;em&amp;gt;in silico glpc&amp;lt;/em&amp;gt; gene knockout predicts increased succinate production from glycerol in &amp;lt;em&amp;gt;escherichia coli&amp;lt;/em&amp;gt;. 2(2):40–48, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3934/bioeng.2015.2.40">[DOI]</a>.</p></li>
<li id="id32824"><p>David E. Ruckerbauer, Christian Jungreuthmayer, and Jürgen Zanghellini. Predicting genetic engineering targets with elementary flux mode analysis: a review of four current methods. 32(6):534–546, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.nbt.2015.03.017">[DOI]</a>.</p></li>
<li id="id32832"><p>Julia Lischke, Christine Lang, Oliver Sawodny, and Ronny Feuer. Impairment of energy metabolism in cardiomyocytes caused by 5-fu catabolites can be compensated by administration of amino acids. 79():5363–5366, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/embc.2015.7319603">[DOI]</a>.</p></li>
<li id="id32846"><p>Nima Abedpour and Markus Kollmann. Resource constrained flux balance analysis predicts selective pressure on the global structure of metabolic networks. 9(1):88–88, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-015-0232-5">[DOI]</a>.</p></li>
<li id="id32858"><p>Na-Rae Lee, Ji-Yeong Yun, Sun-Mee Lee, and Jin‐Byung Park. Cyclohexanone-induced stress metabolism of escherichia coli and corynebacterium glutamicum. 20(6):1088–1098, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-015-0607-x">[DOI]</a>.</p></li>
<li id="id32871"><p>Max Conway, Claudio Angione, and Píetro Lió. Iterative multi level calibration of metabolic networks. 11(1):93–105, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.2174/1574893611666151203222505">[DOI]</a>.</p></li>
<li id="id32900"><p>Pratish Gawand, Fatumina Said Abukar, Naveen Venayak, Siavash Partow, Adilson E. Motter, and Radhakrishnan Mahadevan. Sub-optimal phenotypes of double-knockout mutants of escherichia coli depend on the order of gene deletions. 7(8):930–939, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c5ib00096c">[DOI]</a>.</p></li>
<li id="id32938"><p>Bashir Sajo Mienda and Mohd Shahir Shamsir. An overview of pathway prediction tools for synthetic design of microbial chemical factories. 2(1):1–14, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3934/bioeng.2015.1.1">[DOI]</a>.</p></li>
<li id="id32941"><p>Hadi Nazem‐Bokaee and Ryan S. Senger. Tomi-fba: a genome-scale metabolic flux based algorithm to select optimum hosts and media formulations for expressing pathways of interest. 2(4):335–374, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.3934/bioeng.2015.4.335">[DOI]</a>.</p></li>
<li id="id32950"><p>Daniele De Martino, Matteo Mori, and V. Parisi. Uniform sampling of steady states in metabolic networks: heterogeneous scales and rounding. 10(4):e0122670–e0122670, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0122670">[DOI]</a>.</p></li>
<li id="id32953"><p>Arne C. Reimers. Obstructions to sampling qualitative properties. 10(8):e0135636–e0135636, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0135636">[DOI]</a>.</p></li>
<li id="id32954"><p>Seong-Joo Hong and Choul‐Gyun Lee. Microalgal systems biology for biofuel production. ():3–21, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-20200-6_1">[DOI]</a>.</p></li>
<li id="id32966"><p>Honglei Liu, Yanda Li, and Xiaowo Wang. Op‐synthetic: identification of optimal genetic manipulations for the overproduction of native and non‐native metabolites. 2(3):100–109, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s40484-014-0033-7">[DOI]</a>.</p></li>
<li id="id32993"><p>Jiun Y. Yen, Glenda E. Gillaspy, and Ryan S. Senger. Systems metabolic engineering of &lt;i&gt;arabidopsis&lt;/i&gt; for increased cellulose production. 29(S1):, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1096/fasebj.29.1_supplement.887.26">[DOI]</a>.</p></li>
<li id="id33000"><p>Miguel Á. Valderrama-Gómez, Sander Wagner, and Andreas Kremling. Computer-guided metabolic engineering. ():153–184, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/8623_2015_118">[DOI]</a>.</p></li>
<li id="id33035"><p>John W. May. Cheminformatics for genome-scale metabolic reconstructions. ():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.17863/cam.15987">[DOI]</a>.</p></li>
<li id="id33038"><p>Rui C. Martins and Nuno Fachada. Finite element procedures for enzyme, chemical reaction and 'in-silico' genome scale networks. ():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1508.02506">[DOI]</a>.</p></li>
<li id="id33050"><p>Andrew Damiani. Control engineering perspective on genome-scale metabolic modeling. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33056"><p>Freyr Sverrisson. Mat á styrkgildum umbrotsefna í efnaskiptalíkönum með monte carlo hermun. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33066"><p>Stéphane Pinhal. Mécanismes de l'inhibition de la croissance par l'acétate chez escherichia coli. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33067"><p>João R. C. Ramos. Analysis of metabolic flux distributions in relation to the extracellular environment in avian cells. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33073"><p>Gabriele Fontanarosa, Giulia Menichetti, Enrico Giampieri, Gastone Castellani, Giovanni Martinelli, and Daniel Remondini. Network approaches for analysis and modeling of the human metabolism. 15(02):1540026–1540026, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0219519415400266">[DOI]</a>.</p></li>
<li id="id33091"><p>Md. Bahadur Badsha. Development of complementary elementary modeanalysis for integration of heterogeneous biological data into a complex metabolic network. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33093"><p>E. Mendes and A. FURIGO. Estudo em escala genômica da clostridium acetobutylicum atcc 824 para a bioprodução de hidrogênio e criação de um modelo regulatório. ():12816–12823, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.5151/chemeng-cobeq2014-1792-17528-147104">[DOI]</a>.</p></li>
<li id="id33094"><p>Shuyi Ma. Systems biology of lung diseases: a multi-perspective view from host and pathogen. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33095"><p>Tyler Huelsman. Genome-scale reconstruction of the chlorella vulgaris utex 395 metabolic network. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33096"><p>Yufang Wang and Dewu Ding. Revealing the structure and function of p. pastoris metabolic network using petri nets. 129():360–363, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/dcabes.2015.97">[DOI]</a>.</p></li>
<li id="id33129"><p>Amit Kumar. Understanding physiology of diseases and cell lines using omics based approaches. ():, 2015. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id33137"><p>Ronan M. T. Fleming, Nikos Vlassis, Ines Thiele, and Michael A. Saunders. Conditions for duality between fluxes and concentrations in biochemical networks. ():, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1512.02690">[DOI]</a>.</p></li>
<li id="id33160"><p>Mark Reuter, Duncan J. H. Gaskin, and Aline Métris. A proposed essential gene discovery pipeline: a campylobacter jejuni case study. 1279():167–181, 2015. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-2398-4_11">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-11" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-11" name="0-11" role="tabpanel" tabindex="0"><div class="docutils container" id="id34332">
<ol class="arabic simple" start="1">
<li id="id34334"><p>Edward T. Chouchani, Victoria R. Pell, Edoardo Gaude, Dunja Aksentijević, Stephanie Y. Sundier, Ellen L. Robb, Angela Logan, Sergiy M. Nadtochiy, Emily Ord, Anthony C. Smith, Filmon Eyassu, Rachel Shirley, Chou-Hui Hu, Anna Dare, Andrew M. James, Sebastian Rogatti, Richard C. Hartley, Simon Eaton, Ana S.H. Costa, Paul S. Brookes, Sean M. Davidson, Michael R. Duchen, Kourosh Saeb‐Parsy, Michael J. Shattock, Alan J. Robinson, Lorraine M. Work, Christian Frezza, Thomas Krieg, and Michael P. Murphy. Ischaemic accumulation of succinate controls reperfusion injury through mitochondrial ros. 515(7527):431–435, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nature13909">[DOI]</a>.</p></li>
<li id="id34371"><p>Stephen Sarria, Betty L. Wong, Héctor García Martín, Jay D. Keasling, and Pamela Peralta‐Yahya. Microbial synthesis of pinene. 3(7):466–475, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/sb4001382">[DOI]</a>.</p></li>
<li id="id34397"><p>Matthew N. Benedict, James R. Henriksen, William W. Metcalf, Rachel J. Whitaker, and Nathan D. Price. Itep: an integrated toolkit for exploration of microbial pan-genomes. 15(1):8–8, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2164-15-8">[DOI]</a>.</p></li>
<li id="id34446"><p>Samuel M. D. Seaver, Svetlana Gerdes, Océane Frelin, Claudia Lerma‐Ortiz, Louis Mt Bradbury, Rémi Zallot, Ghulam Hasnain, Thomas D. Niehaus, Basma El Yacoubi, Shiran Pasternak, Robert Olson, Gordon D. Pusch, Ross Overbeek, Rick Stevens, Valérie de Crécy‐Lagard, Doreen Ware, Andrew D. Hanson, and Christopher S. Henry. High-throughput comparison, functional annotation, and metabolic modeling of plant genomes using the plantseed resource. 111(26):9645–9650, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1401329111">[DOI]</a>.</p></li>
<li id="id34449"><p>Javier Carrera, Raíssa Estrela, Jing Luo, Navneet Rai, Αθανάσιος Τσουκαλάς, and Ilias Tagkopoulos. An integrative, multi‐scale, genome‐wide model reveals the phenotypic landscape of &lt;i&gt;&lt;scp&gt;e&lt;/scp&gt;scherichia coli&lt;/i&gt;. 10(7):735–735, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20145108">[DOI]</a>.</p></li>
<li id="id34469"><p>Md. Bahadur Badsha, Ryo Tsuboi, and Hiroyuki Kurata. Complementary elementary modes for fast and efficient analysis of metabolic networks. 90():121–130, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bej.2014.05.022">[DOI]</a>.</p></li>
<li id="id34476"><p>Germán Plata, Christopher S. Henry, and Dennis Vitkup. Long-term phenotypic evolution of bacteria. 517(7534):369–372, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nature13827">[DOI]</a>.</p></li>
<li id="id34477"><p>Michael A. Henson and Timothy J. Hanly. Dynamic flux balance analysis for synthetic microbial communities. 8(5):214–229, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1049/iet-syb.2013.0021">[DOI]</a>.</p></li>
<li id="id34514"><p>Andreas Dräger and Bernhard Ø. Palsson. Improving collaboration by standardization efforts in systems biology. 2():61–61, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2014.00061">[DOI]</a>.</p></li>
<li id="id34532"><p>Óscar Dias, Rui Pereira, Andreas Gombert, Eugénio C. Ferreira, and Isabel Rocha. Iod907, the first genome‐scale metabolic model for the milk yeast &lt;i&gt;kluyveromyces lactis&lt;/i&gt;. 9(6):776–790, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201300242">[DOI]</a>.</p></li>
<li id="id34558"><p>Nadeera Jayasinghe, Ashley E. Franks, Kelly P. Nevin, and Radhakrishnan Mahadevan. Metabolic modeling of spatial heterogeneity of biofilms in microbial fuel cells reveals substrate limitations in electrical current generation. 9(10):1350–1361, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201400068">[DOI]</a>.</p></li>
<li id="id34561"><p>Octavio Perez‐Garcia, Silas G. Villas‐Bôas, Simon Swift, Kartik Chandran, and Naresh Singhal. Clarifying the regulation of no/n2o production in nitrosomonas europaea during anoxic–oxic transition via flux balance analysis of a metabolic network model. 60():267–277, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.watres.2014.04.049">[DOI]</a>.</p></li>
<li id="id34575"><p>Mario Latendresse. Efficiently gap-filling reaction networks. 15(1):225–225, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-15-225">[DOI]</a>.</p></li>
<li id="id34577"><p>Zachary A. King and Adam M. Feist. Optimal cofactor swapping can increase the theoretical yield for chemical production in escherichia coli and saccharomyces cerevisiae. 24():117–128, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.05.009">[DOI]</a>.</p></li>
<li id="id34583"><p>Yanfeng Liu, Hyun‐Dong Shin, Jianghua Li, and Long Liu. Toward metabolic engineering in the context of system biology and synthetic biology: advances and prospects. 99(3):1109–1118, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-014-6298-y">[DOI]</a>.</p></li>
<li id="id34587"><p>Zhiqiang Sun, Hailin Meng, Jing Li, Jianfeng Wang, Qian Li, Yong Wang, and Yansheng Zhang. Identification of novel knockout targets for improving terpenoids biosynthesis in saccharomyces cerevisiae. 9(11):e112615–e112615, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0112615">[DOI]</a>.</p></li>
<li id="id34590"><p>Amphun Chaiboonchoe, Bushra Dohai, Hong Cai, David R. Nelson, Kenan Jijakli, and Kourosh Salehi‐Ashtiani. Microalgal metabolic network model refinement through high-throughput functional metabolic profiling. 2():68–68, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2014.00068">[DOI]</a>.</p></li>
<li id="id34593"><p>Rienk A. Rienksma, María Suárez‐Diez, Lucie Spina, Peter J. Schaap, and Vítor A. P. Martins dos Santos. Systems-level modeling of mycobacterial metabolism for the identification of new (multi-)drug targets. 26(6):610–622, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.smim.2014.09.013">[DOI]</a>.</p></li>
<li id="id34598"><p>Yi Li, Guojian Zhang, and Blaine A. Pfeifer. Current and emerging options for taxol production. 148():405–425, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2014_292">[DOI]</a>.</p></li>
<li id="id34632"><p>Björn H. Junker. Flux analysis in plant metabolic networks: increasing throughput and coverage. 26():183–188, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2014.01.016">[DOI]</a>.</p></li>
<li id="id34663"><p>J.M. González-Martínez, Abel Folch‐Fortuny, Francisco Llaneras, Marta Tortajada, Jesús Picó, and Alberto Ferrer. Metabolic flux understanding of pichia pastoris grown on heterogenous culture media. 134():89–99, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.chemolab.2014.02.003">[DOI]</a>.</p></li>
<li id="id34664"><p>Parizad Babaei, Tahereh Ghasemi-Kahrizsangi, and Sayed‐Amir Marashi. Modeling the differences in biochemical capabilities of&lt;i&gt;pseudomonas&lt;/i&gt;species by flux balance analysis: how good are genome-scale metabolic networks at predicting the differences? 2014():1–11, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2014/416289">[DOI]</a>.</p></li>
<li id="id34689"><p>Benjamín J. Sánchez, José Ricardo Pérez‐Correa, and Eduardo Agosín. Construction of robust dynamic genome-scale metabolic model structures of saccharomyces cerevisiae through iterative re-parameterization. 25():159–173, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.07.004">[DOI]</a>.</p></li>
<li id="id34698"><p>Joseph Koussa, Amphun Chaiboonchoe, and Kourosh Salehi‐Ashtiani. Computational approaches for microalgal biofuel optimization: a review. 2014():1–12, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2014/649453">[DOI]</a>.</p></li>
<li id="id34699"><p>Xiaoyang Wang, Chuanbo Zhang, Meiling Wang, and Wenyu Lu. Genome-scale metabolic network reconstruction of saccharopolyspora spinosa for spinosad production improvement. 13(1):41–41, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-13-41">[DOI]</a>.</p></li>
<li id="id34700"><p>Stephen S. Fong. Computational approaches to metabolic engineering utilizing systems biology and synthetic biology. 11(18):28–34, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.csbj.2014.08.005">[DOI]</a>.</p></li>
<li id="id34703"><p>Lü Jing, Farah Fathiah Muzaffar Shah, Mohd Saberi Mohamad, Nur Laily Hamran, Abdul Hakim Mohamed Salleh, Safaai Deris, and Hany Alashwal. Database and tools for metabolic network analysis. 19(4):568–585, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-014-0172-8">[DOI]</a>.</p></li>
<li id="id34713"><p>Rafael S. Costa, André Veríssimo, and Susana Vinga. Ki mosys: a web-based repository of experimental data for kinetic models of biological systems. 8(1):85–85, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-014-0085-3">[DOI]</a>.</p></li>
<li id="id34734"><p>Gilles Vieira, Marc Carnicer, Jean‐Charles Portais, and Stéphanie Heux. Findpath: a matlab solution for &lt;i&gt;in silico&lt;/i&gt; design of synthetic metabolic pathways. 30(20):2986–2988, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btu422">[DOI]</a>.</p></li>
<li id="id34743"><p>Abel Folch‐Fortuny, Marta Tortajada, J.M. Prats-Montalbán, Francisco Llaneras, Jesús Picó, and Alberto Ferrer. Mcr-als on metabolic networks: obtaining more meaningful pathways. 142():293–303, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.chemolab.2014.10.004">[DOI]</a>.</p></li>
<li id="id34764"><p>Juan Nogales. A practical protocol for genome-scale metabolic reconstructions. ():197–221, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/8623_2014_12">[DOI]</a>.</p></li>
<li id="id34785"><p>Yue‐Dong Gao, Yuqi Zhao, and Jing‐Fei Huang. Metabolic modeling of common&lt;i&gt;escherichia coli&lt;/i&gt;strains in human gut microbiome. 2014():1–11, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2014/694967">[DOI]</a>.</p></li>
<li id="id34790"><p>Michael P. Weiner, Julia Tröndle, Christoph Albermann, Georg A. Sprenger, and Dirk Weuster‐Botz. Improvement of constraint‐based flux estimation during l‐phenylalanine production with &lt;i&gt;escherichia coli&lt;/i&gt; using targeted knock‐out mutants. 111(7):1406–1416, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25195">[DOI]</a>.</p></li>
<li id="id34792"><p>Xinsen Wu, Xiaoyang Wang, and Wenyu Lu. Genome-scale reconstruction of a metabolic network for gluconobacter oxydans 621h. 117():10–14, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2014.01.001">[DOI]</a>.</p></li>
<li id="id34813"><p>Robert Nilsson, Fredric Bauer, Sennai Mesfun, Christian Hulteberg, Joakim Lundgren, Sune Wännström, Ulrika Rova, and Kris A. Berglund. Techno-economics of carbon preserving butanol production using a combined fermentative and catalytic approach. 161():263–269, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2014.03.055">[DOI]</a>.</p></li>
<li id="id34823"><p>Priyanka P. Sadhukhan and Anu Raghunathan. Investigating host–pathogen behavior and their interaction using genome-scale metabolic network models. 1184():523–562, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-1115-8_29">[DOI]</a>.</p></li>
<li id="id34836"><p>Josh Quarterman, Soo Rin Kim, Pan‐Jun Kim, and Yong‐Su Jin. Enhanced hexose fermentation by saccharomyces cerevisiae through integration of stoichiometric modeling and genetic screening. 194():48–57, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2014.11.017">[DOI]</a>.</p></li>
<li id="id34854"><p>Ming‐Tsang Wu and Christina Chan. Prediction of therapeutic microrna based on the human metabolic network. 30(8):1163–1171, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt751">[DOI]</a>.</p></li>
<li id="id34900"><p>Alberto Noronha, Paulo Vilaça, and Miguel Rocha. An integrated network visualization framework towards metabolic engineering applications. 15(1):420–420, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-014-0420-0">[DOI]</a>.</p></li>
<li id="id34909"><p>Lisha K. Parambil and Debasis Sarkar. Probing the bioethanol production potential of scheffersomyces (pichia) stipitis using validated genome-scale model. 36(12):2443–2451, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-014-1629-8">[DOI]</a>.</p></li>
<li id="id34936"><p>Mathieu Almeida and Mihai Pop. High-throughput sequencing as a tool for exploring the human microbiome. ():55–66, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-410472-3.00004-x">[DOI]</a>.</p></li>
<li id="id34937"><p>Sriram Chandrasekaran. Predicting phenotype from genotype through reconstruction and integrative modeling of metabolic and regulatory networks. ():307–325, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-017-9041-3_12">[DOI]</a>.</p></li>
<li id="id34942"><p>K. P. Lisha and Debasis Sarkar. In silico analysis of bioethanol production from glucose/xylose mixtures during fed-batch fermentation of co-culture and mono-culture systems. 19(5):879–891, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-014-0320-1">[DOI]</a>.</p></li>
<li id="id34963"><p>Rafael S. Costa, Son Hoang Nguyen, András Hartmann, and Susana Vinga. Exploring the cellular objective in flux balance constraint-based models. ():211–224, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-12982-2_15">[DOI]</a>.</p></li>
<li id="id34969"><p>Erin R. Shellman, Yu Chen, Xiaoxia Nina Lin, Charles Burant, and Santiago Schnell. Metabolic network motifs can provide novel insights into evolution: the evolutionary origin of eukaryotic organelles as a case study. 53():242–250, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.compbiolchem.2014.09.006">[DOI]</a>.</p></li>
<li id="id34979"><p>Erno Lindfors, Paula Jouhten, Merja Oja, Eija Rintala, Matej Orešič, and Merja Penttilä. Integration of transcription and flux data reveals molecular paths associated with differences in oxygen-dependent phenotypes of saccharomyces cerevisiae. 8(1):16–16, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-8-16">[DOI]</a>.</p></li>
<li id="id34988"><p>Leonid Chindelevitch, Jason Trigg, Aviv Regev, and Bonnie Berger. An exact arithmetic toolbox for a consistent and reproducible structural analysis of metabolic network models. 5(1):4893–4893, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms5893">[DOI]</a>.</p></li>
<li id="id34995"><p>Paula Jouhten and Hannu Maaheimo. Labelling analysis for 13c mfa using nmr spectroscopy. 1191():143–164, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-1170-7_9">[DOI]</a>.</p></li>
<li id="id35031"><p>James R. Collett, Pimphan A. Meyer, and Susanne B. Jones. Preliminary economics for hydrocarbon fuel production from cellulosic sugars. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.2172/1133232">[DOI]</a>.</p></li>
<li id="id35040"><p>Guojian Zhang and Blaine A. Pfeifer. Production of therapeutic products. ():261–276, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9781118794623.ch14">[DOI]</a>.</p></li>
<li id="id35044"><p>DANIELA CAROLINA FLÓREZ PARRA, VANESSA RAMIREZ TOVAR, and Andrés Fernando González Barrios. Elucidating the role of metabolites as quorum sensing signals using phase plane analysis: the case of indole in &lt;i&gt;escherichia coli&lt;/i&gt;. 22(04):523–531, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0218339014500193">[DOI]</a>.</p></li>
<li id="id35053"><p>JULIAN TRIANA DOPICO. Model-based analysis and metabolic design of a cyanobacterium for bio-products synthesis. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/39351">[DOI]</a>.</p></li>
<li id="id35055"><p>Wei Suong Teo. Engineering synthetic promoters as dynamic controllers in saccharomyces cerevisiae. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.32657/10356/61754">[DOI]</a>.</p></li>
<li id="id35059"><p>Binbin Chen. Systems-level characterization and transporter engineering in saccharomyces cerevisiae for improved hydrocarbon tolerance. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.32657/10356/61731">[DOI]</a>.</p></li>
<li id="id35076"><p>Tamás Turányi and Alison S. Tomlin. Reaction pathway analysis. ():53–60, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-662-44562-4_4">[DOI]</a>.</p></li>
<li id="id35125"><p>Victor Chubukov, Luca Gerosa, Karl Kochanowski, and Uwe Sauer. Coordination of microbial metabolism. 12(5):327–340, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nrmicro3238">[DOI]</a>.</p></li>
<li id="id35128"><p>Daniel Machado and Markus J. Herrgård. Systematic evaluation of methods for integration of transcriptomic data into constraint-based models of metabolism. 10(4):e1003580–e1003580, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003580">[DOI]</a>.</p></li>
<li id="id35129"><p>Anna Kubacka, María Suárez‐Diez, David Rojo, Rafael Bargiela, Sergio Ciordia, Inés Zapico, Juan Pablo Albar, Coral Barbas, Vítor A. P. Martins dos Santos, Marcos Fernández‐García, and Manuel Ferrer. Understanding the antimicrobial mechanism of tio2-based nanocomposite films in a pathogenic bacterium. 4(1):4134–4134, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep04134">[DOI]</a>.</p></li>
<li id="id35142"><p>Nikos Vlassis, Maria Pires Pacheco, and Thomas Sauter. Fast reconstruction of compact context-specific metabolic network models. 10(1):e1003424–e1003424, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003424">[DOI]</a>.</p></li>
<li id="id35146"><p>Irina Borodina and Jens Nielsen. Advances in metabolic engineering of yeast&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;for production of chemicals. 9(5):609–620, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201300445">[DOI]</a>.</p></li>
<li id="id35148"><p>Irina Borodina, Kanchana Rueksomtawin Kildegaard, Niels Bjerg Jensen, Thomas Blicher, Jérôme Maury, Svetlana Sherstyk, Konstantin Schneider, Pedro Lamosa, Markus J. Herrgård, Inger Rosenstand, Fredrik Öberg, Jochen Förster, and Jens Nielsen. Establishing a synthetic pathway for high-level production of 3-hydroxypropionic acid in saccharomyces cerevisiae via β-alanine. 27():57–64, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.10.003">[DOI]</a>.</p></li>
<li id="id35151"><p>Almut Heinken, M. Tanweer Khan, Giuseppe Paglia, Dmitry A. Rodionov, Hermie J. M. Harmsen, and Ines Thiele. Functional metabolic map of faecalibacterium prausnitzii, a beneficial human gut microbe. 196(18):3289–3302, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.01780-14">[DOI]</a>.</p></li>
<li id="id35155"><p>Achuthanunni Chokkathukalam, Dong‐Hyun Kim, Michael P. Barrett, Rainer Breitling, and Darren J. Creek. Stable isotope-labeling studies in metabolomics: new insights into structure and dynamics of metabolic networks. 6(4):511–524, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.4155/bio.13.348">[DOI]</a>.</p></li>
<li id="id35157"><p>Joris Beld, D. John Lee, and Michael D. Burkart. Fatty acid biosynthesis revisited: structure elucidation and metabolic engineering. 11(1):38–59, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c4mb00443d">[DOI]</a>.</p></li>
<li id="id35173"><p>Miguel A. Campodonico, Bárbara A. Andrews, Juan A. Asenjo, Bernhard Ø. Palsson, and Adam M. Feist. Generation of an atlas for commodity chemical production in escherichia coli and a novel pathway prediction algorithm, gem-path. 25():140–158, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.07.009">[DOI]</a>.</p></li>
<li id="id35181"><p>Tige R. Rustad, K Minch, Shuyi Ma, Jessica K. Winkler, Samuel J. Hobbs, Mark J. Hickey, William Brabant, Serdar Turkarslan, Nathan D. Price, Nitin S. Baliga, and David R. Sherman. Mapping and manipulating the mycobacterium tuberculosis transcriptome using a transcription factor overexpression-derived regulatory network. 15(11):502–502, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s13059-014-0502-3">[DOI]</a>.</p></li>
<li id="id35187"><p>Jose A. Gomez, Kai Höffner, and Paul I. Barton. Dfbalab: a fast and reliable matlab code for dynamic flux balance analysis. 15(1):409–409, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12859-014-0409-8">[DOI]</a>.</p></li>
<li id="id35210"><p>Ines Thiele, Nikos Vlassis, and Ronan M. T. Fleming. &lt;scp&gt;fast&lt;/scp&gt;g&lt;scp&gt;ap&lt;/scp&gt;f&lt;scp&gt;ill&lt;/scp&gt;: efficient gap filling in metabolic networks. 30(17):2529–2531, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btu321">[DOI]</a>.</p></li>
<li id="id35236"><p>Byoung‐Jin Kim, Won Jun Kim, Dong In Kim, and Sang Yup Lee. Applications of genome-scale metabolic network model in metabolic engineering. 42(3):339–348, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-014-1554-9">[DOI]</a>.</p></li>
<li id="id35239"><p>Swagatika Sahoo, Maike K. Aurich, Jón J. Jónsson, and Ines Thiele. Membrane transporters in a human genome-scale metabolic knowledgebase and their implications for disease. 5():91–91, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2014.00091">[DOI]</a>.</p></li>
<li id="id35240"><p>SemidÃ¡n Robaina EstÃ©vez and Zoran Nikoloski. Generalized framework for context-specific metabolic model extraction methods. 5():491–491, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2014.00491">[DOI]</a>.</p></li>
<li id="id35242"><p>Ibrahim E. Elsemman, Fredrik Karlsson, Saeed Shoaie, Intawat Nookaew, Taysir Hassan A. Soliman, and Jens Nielsen. Genome-scale metabolic reconstructions of bifidobacterium adolescentis l2-32 and faecalibacterium prausnitzii a2-165 and their interaction. 8(1):41–41, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-8-41">[DOI]</a>.</p></li>
<li id="id35272"><p>Matthew N. Benedict, Michael Mundy, Christopher S. Henry, Nicholas Chia, and Nathan D. Price. Likelihood-based gene annotations for gap filling and quality assessment in genome-scale metabolic models. 10(10):e1003882–e1003882, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003882">[DOI]</a>.</p></li>
<li id="id35295"><p>Minsuk Kim, Jeong Sang Yi, Joonwon Kim, Jinu Kim, Min Woo Kim, and Byung‐Gee Kim. Reconstruction of a high‐quality metabolic model enables the identification of gene overexpression targets for enhanced antibiotic production in &lt;i&gt;streptomyces coelicolor&lt;/i&gt; a3(2). 9(9):1185–1194, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201300539">[DOI]</a>.</p></li>
<li id="id35299"><p>Maike K. Aurich, Giuseppe Paglia, Óttar Rolfsson, Sigrún Hrafnsdóttir, Manuela Magnúsdóttir, Magdalena M. Stefaniak, Bernhard Ø. Palsson, Ronan M. T. Fleming, and Ines Thiele. Prediction of intracellular metabolic states from extracellular metabolomic data. 11(3):603–619, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11306-014-0721-3">[DOI]</a>.</p></li>
<li id="id35314"><p>Paolo Cazzaniga, Chiara Damiani, Daniela Besozzi, Riccardo Colombo, Marco S. Nobile, Daniela Gaglio, Dario Pescini, Sara Molinari, Giancarlo Mauri, Lilia Alberghina, and Marco Vanoni. Computational strategies for a system-level understanding of metabolism. 4(4):1034–1087, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo4041034">[DOI]</a>.</p></li>
<li id="id35333"><p>John Blazeck, Jarrett Miller, Anny Pan, Jon Gengler, Clinton Holden, Mariam Jamoussi, and Hal S. Alper. Metabolic engineering of saccharomyces cerevisiae for itaconic acid production. 98(19):8155–8164, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-014-5895-0">[DOI]</a>.</p></li>
<li id="id35336"><p>Javier Cárdenas and Nancy A. Da Silva. Metabolic engineering of saccharomyces cerevisiae for the production of triacetic acid lactone. 25():194–203, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.07.008">[DOI]</a>.</p></li>
<li id="id35338"><p>Tige R. Rustad, K Minch, Shuyi Ma, Jessica K. Winkler, Samuel J. Hobbs, Mark J. Hickey, William Brabant, Serdar Turkarslan, Nathan D. Price, Nitin S. Baliga, and David R. Sherman. Mapping and manipulating the mycobacterium tuberculosis transcriptome using a transcription factor overexpression-derived regulatory network. 15(11):502–502, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/preaccept-1701638048134699">[DOI]</a>.</p></li>
<li id="id35342"><p>Stephen Federowicz, Donghyuk Kim, Ali Ebrahim, Joshua A. Lerman, Harish Nagarajan, Byung-kwan Cho, Karsten Zengler, and Bernhard Ø. Palsson. Determining the control circuitry of redox metabolism at the genome-scale. 10(4):e1004264–e1004264, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pgen.1004264">[DOI]</a>.</p></li>
<li id="id35343"><p>Benjamin VanderSluis, David Hess, Colin Pesyna, Elias W. Krumholz, Tahin Syed, Balázs Szappanos, Corey Nislow, Balázs Papp, Olga G. Troyanskaya, Chad L. Myers, and Amy A. Caudy. Broad metabolic sensitivity profiling of a prototrophic yeast deletion collection. 15(4):R64–R64, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/gb-2014-15-4-r64">[DOI]</a>.</p></li>
<li id="id35348"><p>Hojung Nam, Miguel A. Campodonico, Aarash Bordbar, Daniel R. Hyduke, Sangwoo Kim, Daniel C. Zielinski, and Bernhard Ø. Palsson. A systems approach to predict oncometabolites via context-specific genome-scale metabolic networks. 10(9):e1003837–e1003837, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003837">[DOI]</a>.</p></li>
<li id="id35362"><p>Lake‐Ee Quek, Stefanie Dietmair, Michael Hanscho, Verónica S. Martínez, Nicole Borth, and Lars K. Nielsen. Reducing recon 2 for steady-state flux analysis of hek cell culture. 184():172–178, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2014.05.021">[DOI]</a>.</p></li>
<li id="id35365"><p>Kambiz Baghalian, Mohammad‐Reza Hajirezaei, and Falk Schreiber. Plant metabolic modeling: achieving new insight into metabolism and metabolic engineering. 26(10):3847–3866, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1105/tpc.114.130328">[DOI]</a>.</p></li>
<li id="id35373"><p>Eduard J. Kerkhoven, Petri‐Jaan Lahtvee, and Jens Nielsen. Applications of computational modeling in metabolic engineering of yeast. 15(1):n/a–n/a, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1567-1364.12199">[DOI]</a>.</p></li>
<li id="id35389"><p>Daniel Weaver, Ingrid M. Keseler, Amanda Mackie, Ian T. Paulsen, and Peter D. Karp. A genome-scale metabolic flux model of escherichia coli k–12 derived from the ecocyc database. 8(1):79–79, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-8-79">[DOI]</a>.</p></li>
<li id="id35405"><p>Marco Fondi, Isabel Maida, Elena Perrin, Alessandra Mellera, Stéfano Mocali, Ermenegilda Parrilli, Maria Luisa Tutino, Píetro Lió, and Renato Fani. Genome‐scale metabolic reconstruction and constraint‐based modelling of the antarctic bacterium &lt;scp&gt; &lt;i&gt;p&lt;/i&gt; &lt;/scp&gt; &lt;i&gt;seudoalteromonas haloplanktis&lt;/i&gt; &lt;scp&gt;tac&lt;/scp&gt; 125. 17(3):751–766, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.12513">[DOI]</a>.</p></li>
<li id="id35416"><p>Aarash Bordbar, Harish Nagarajan, Nathan E. Lewis, Haythem Latif, Ali Ebrahim, Stephen Federowicz, Jan Schellenberger, and Bernhard Ø. Palsson. Minimal metabolic pathway structure is consistent with associated biomolecular interactions. 10(7):737–737, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.15252/msb.20145243">[DOI]</a>.</p></li>
<li id="id35428"><p>Omer Eilam, Raphy Zarecki, Matthew Oberhardt, Luke K. Ursell, Martin Kupiec, Rob Knight, Uri Gophna, and Eytan Ruppin. Glycan degradation (glyder) analysis predicts mammalian gut microbiota abundance and host diet-specific adaptations. 5(4):, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/mbio.01526-14">[DOI]</a>.</p></li>
<li id="id35446"><p>Ali Masoudi‐Nejad and Yazdan Asgari. Metabolic cancer biology: structural-based analysis of cancer as a metabolic disease, new sights and opportunities for disease treatment. 30():21–29, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.semcancer.2014.01.007">[DOI]</a>.</p></li>
<li id="id35460"><p>Adam M. Feist, Harish Nagarajan, Amelia‐Elena Rotaru, Pier‐Luc Tremblay, Tian Zhang, Kelly P. Nevin, Derek R. Lovley, and Karsten Zengler. Constraint-based modeling of carbon fixation and the energetics of electron transfer in geobacter metallireducens. 10(4):e1003575–e1003575, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003575">[DOI]</a>.</p></li>
<li id="id35469"><p>Jordan O. Hay, Hai Shi, Nicolas Heinzel, Inga Hebbelmann, Hardy Rolletschek, and Jörg Schwender. Integration of a constraint-based metabolic model of brassica napus developing seeds with 13c-metabolic flux analysis. 5():724–724, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fpls.2014.00724">[DOI]</a>.</p></li>
<li id="id35470"><p>Monica Larocque, Thierry Chénard, and Rafaël Najmanovich. A curated c. difficile strain 630 metabolic network: prediction of essential targets and inhibitors. 8(1):117–117, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-014-0117-z">[DOI]</a>.</p></li>
<li id="id35493"><p>Nan Xu, Jie Liu, Lianzhong Ai, and Li Liu. Reconstruction and analysis of the genome-scale metabolic model of lactobacillus casei lc2w. 554(2):140–147, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gene.2014.10.034">[DOI]</a>.</p></li>
<li id="id35499"><p>Uldis Kalnenieks, Agris Pentjuss, Reinis Rutkis, Egils Stalidzāns, and David A. Fell. Modeling of zymomonas mobilis central metabolism for novel metabolic engineering strategies. 5():42–42, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2014.00042">[DOI]</a>.</p></li>
<li id="id35508"><p>Ángela Cintolesi, James M. Clomburg, and Ramón González. In silico assessment of the metabolic capabilities of an engineered functional reversal of the β-oxidation cycle for the synthesis of longer-chain (c≥4) products. 23():100–115, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2014.02.011">[DOI]</a>.</p></li>
<li id="id35509"><p>Sanu Shameer, Flora Logan-Klumpler, Florence Vinson, Ludovic Cottret, Benjamin Merlet, Fiona Achcar, Michael Boshart, Matthew Berriman, Rainer Breitling, Frédéric Bringaud, Peter Bütikofer, Amy M. Cattanach, Bridget Bannerman-Chukualim, Darren J. Creek, Kathryn Crouch, Harry P. de Koning, Hubert Denise, Charles Ebikeme, Alan H. Fairlamb, Michael A. J. Ferguson, Michael L. Ginger, Christiane Hertz‐Fowler, Eduard J. Kerkhoven, Pascal Mäser, Paul A.M. Michels, Archana Nayak, David W. Nes, Derek P. Nolan, Christian A. Olsen, Fátima Silva-Franco, Terry Smith, Martin C. Taylor, Aloysius G. M. Tielens, Michael D. Urbaniak, Jaap J. van Hellemond, Isabel M. Vincent, Shane R. Wilkinson, Susan Wyllie, Fred R. Opperdoes, Michael P. Barrett, and Fabien Jourdan. Trypanocyc: a community-led biochemical pathways database for trypanosoma brucei. 43(D1):D637–D644, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gku944">[DOI]</a>.</p></li>
<li id="id35520"><p>Swagatika Sahoo, Hulda S. Haraldsdóttir, Ronan M. T. Fleming, and Ines Thiele. Modeling the effects of commonly used drugs on human metabolism. 282(2):297–317, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/febs.13128">[DOI]</a>.</p></li>
<li id="id35527"><p>Paul A. Jensen and Jason A. Papin. Metdraw: automated visualization of genome-scale metabolic network reconstructions and high-throughput data. 30(9):1327–1328, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt758">[DOI]</a>.</p></li>
<li id="id35542"><p>Hong Yang, Elias W. Krumholz, Evan D. Brutinel, Nagendra P. Palani, Michael J. Sadowsky, Andrew Odlyzko, Jeffrey A. Gralnick, and Igor G. L. Libourel. Genome-scale metabolic network validation of shewanella oneidensis using transposon insertion frequency analysis. 10(9):e1003848–e1003848, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003848">[DOI]</a>.</p></li>
<li id="id35550"><p>Mahdieh Hadi and Sayed‐Amir Marashi. Reconstruction of a generic metabolic network model of cancer cells. 10(11):3014–3021, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c4mb00300d">[DOI]</a>.</p></li>
<li id="id35570"><p>Joonhoon Kim and Jennifer L. Reed. Refining metabolic models and accounting for regulatory effects. 29():34–38, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2014.02.009">[DOI]</a>.</p></li>
<li id="id35577"><p>Andrea De Martino, Daniele De Martino, Roberto Mulet, and Andrea Pagnani. Identifying all moiety conservation laws in genome-scale metabolic networks. 9(7):e100750–e100750, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0100750">[DOI]</a>.</p></li>
<li id="id35581"><p>András Pásztor, Pauli T. Kallio, Dávid Malatinszky, M. Akhtar, and Patrik R. Jones. A synthetic o&lt;sub&gt;2&lt;/sub&gt;‐tolerant butanol pathway exploiting native fatty acid biosynthesis in &lt;i&gt;escherichia coli&lt;/i&gt;. 112(1):120–128, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25324">[DOI]</a>.</p></li>
<li id="id35603"><p>Xin Fang, Jaques Reifman, and Anders Wallqvist. Modeling metabolism and stage-specific growth of plasmodium falciparum hb3 during the intraerythrocytic developmental cycle. 10(10):2526–2537, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c4mb00115j">[DOI]</a>.</p></li>
<li id="id35619"><p>Chao Ye, Wei Zou, Nan Xu, and Li Liu. Metabolic model reconstruction and analysis of an artificial microbial ecosystem for vitamin c production. 182-183():61–67, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2014.04.027">[DOI]</a>.</p></li>
<li id="id35620"><p>Nicolás Loira, Anna Zhukova, and David James Sherman. Pantograph: a template-based method for genome-scale metabolic model reconstruction. 13(02):1550006–1550006, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0219720015500067">[DOI]</a>.</p></li>
<li id="id35621"><p>Alfred Fernández‐Castané, Tamás Fehér, Pablo Carbonell, Cyrille Pauthenier, and Jean‐Loup Faulon. Computer-aided design for metabolic engineering. 192():302–313, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2014.03.029">[DOI]</a>.</p></li>
<li id="id35622"><p>Neha Rohatgi, Tine Nielsen, Sara Petersen Bjørn, Ívar Axelsson, Giuseppe Paglia, Björn G. Voldborg, Bernhard Ø. Palsson, and Óttar Rolfsson. Biochemical characterization of human gluconokinase and the proposed metabolic impact of gluconic acid as determined by constraint based metabolic network analysis. 9(6):e98760–e98760, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0098760">[DOI]</a>.</p></li>
<li id="id35635"><p>Shubo Li, Xiang Gao, Nan Xu, Li Liu, and Jian Chen. Enhancement of acetoin production in candida glabrata by in silico-aided metabolic engineering. 13(1):55–55, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-13-55">[DOI]</a>.</p></li>
<li id="id35636"><p>Sarah Schatschneider, Claudia Huber, Heiko Neuweger, Tony Watt, Alfred Pühler, Wolfgang Eisenreich, Christoph Wittmann, Karsten Niehaus, and Frank‐Jörg Vorhölter. Metabolic flux pattern of glucose utilization by xanthomonas campestris pv. campestris: prevalent role of the entner–doudoroff pathway and minor fluxes through the pentose phosphate pathway and glycolysis. 10(10):2663–2676, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c4mb00198b">[DOI]</a>.</p></li>
<li id="id35637"><p>Timo Maarleveld, Joost Boele, Frank J. Bruggeman, and Bas Teusink. A data integration and visualization resource for the metabolic network of &lt;i&gt;synechocystis&lt;/i&gt; sp. pcc 6803. 164(3):1111–1121, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.113.224394">[DOI]</a>.</p></li>
<li id="id35642"><p>Marco Fondi and Píetro Lió. Genome-scale metabolic network reconstruction. 1231():233–256, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-1720-4_15">[DOI]</a>.</p></li>
<li id="id35645"><p>Na‐Rae Lee, Meiyappan Lakshmanan, Shilpi Aggarwal, Ji‐Won Song, Iftekhar A. Karimi, Dong‐Yup Lee, and Jin‐Byung Park. Genome-scale metabolic network reconstruction and in silico flux analysis of the thermophilic bacterium thermus thermophilus hb27. 13(1):61–61, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-13-61">[DOI]</a>.</p></li>
<li id="id35646"><p>Igor Marín de Mas, Esther Aguilar, Anusha Jayaraman, Ibrahim H. Polat, Alfonso Martín-Bernabé, Rohit Bharat, Carles Foguet, Enric Milà, Balázs Papp, Josep J. Centelles, and Marta Cascante. Cancer cell metabolism as new targets for novel designed therapies. 6(16):1791–1810, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.4155/fmc.14.119">[DOI]</a>.</p></li>
<li id="id35652"><p>María Alejandra Álvarez. Plant secondary metabolism. ():15–31, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-05771-2_3">[DOI]</a>.</p></li>
<li id="id35662"><p>Viswanadham Sridhara, Austin G. Meyer, Piyush Rai, Jeffrey E. Barrick, Pradeep Ravikumar, Daniel Segrè, and Claus O. Wilke. Predicting growth conditions from internal metabolic fluxes in an in-silico model of e. coli. 9(12):e114608–e114608, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0114608">[DOI]</a>.</p></li>
<li id="id35667"><p>Daniel Veyel, Alexander Erban, Ines Fehrle, Joachim Kopka, and Michael Schroda. Rationales and approaches for studying metabolism in eukaryotic microalgae. 4(2):184–217, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo4020184">[DOI]</a>.</p></li>
<li id="id35681"><p>Steffen Klamt, Oliver Hädicke, and Axel von Kamp. Stoichiometric and constraint-based analysis of biochemical reaction networks. ():263–316, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-08437-4_5">[DOI]</a>.</p></li>
<li id="id35682"><p>Anders Sandström, Henrik Almqvist, Diogo Portugal-Nunes, Dário Neves, Gunnar Lidén, and Marie F. Gorwa‐Grauslund. Saccharomyces cerevisiae: a potential host for carboxylic acid production from lignocellulosic feedstock? 98(17):7299–7318, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-014-5866-5">[DOI]</a>.</p></li>
<li id="id35717"><p>Raphael Voges, Stephanie Corsten, Wolfgang Wiechert, and Stephan Noack. Absolute quantification of corynebacterium glutamicum glycolytic and anaplerotic enzymes by qconcat. 113():366–377, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jprot.2014.10.008">[DOI]</a>.</p></li>
<li id="id35719"><p>Meiyappan Lakshmanan, Bijayalaxmi Mohanty, Sun‐Hyung Lim, Sun‐Hwa Ha, and Dong‐Yup Lee. Metabolic and transcriptional regulatory mechanisms underlying the anoxic adaptation of rice coleoptile. 6():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/aobpla/plu026">[DOI]</a>.</p></li>
<li id="id35734"><p>Daniel D. Lewis, Fernando Villarreal, Fan Wu, and Cheemeng Tan. Synthetic biology outside the cell: linking computational tools to cell-free systems. 2():66–66, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fbioe.2014.00066">[DOI]</a>.</p></li>
<li id="id35748"><p>Ryan S. Senger, Jiun Y. Yen, and Stephen S. Fong. A review of genome-scale metabolic flux modeling of anaerobiosis in biotechnology. 6():33–42, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.coche.2014.08.003">[DOI]</a>.</p></li>
<li id="id35782"><p>Chintan Joshi and Ashok Prasad. Epistatic interactions among metabolic genes depend upon environmental conditions. 10(10):2578–2589, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c4mb00181h">[DOI]</a>.</p></li>
<li id="id35804"><p>Yan Zhu, Dan Li, Guanhui Bao, Shaohua Wang, Shaoming Mao, Jiangning Song, Yin Li, and Yanping Zhang. Metabolic changes in klebsiella oxytoca in response to low oxidoreduction potential, as revealed by comparative proteomic profiling integrated with flux balance analysis. 80(9):2833–2841, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.03327-13">[DOI]</a>.</p></li>
<li id="id35805"><p>María Victoria Aguilar Pontes, Ronald P. de Vries, and Miaomiao Zhou. (post-)genomics approaches in fungal research. 13(6):424–439, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bfgp/elu028">[DOI]</a>.</p></li>
<li id="id35808"><p>Karel Olavarría, Jessica De Ingeniis, Daniel C. Zielinski, Matías Fuentealba, Rodrigo Muñoz, Douglas McCloskey, Adam M. Feist, and Ricardo Cabrera. Metabolic impact of an nadh-producing glucose-6-phosphate dehydrogenase in escherichia coli. 160(12):2780–2793, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1099/mic.0.082180-0">[DOI]</a>.</p></li>
<li id="id35810"><p>Amit Kumar, Thomas F. Harrelson, Nathan E. Lewis, Emily J. Gallagher, Derek LeRoith, Joseph Shiloach, and Michael J. Betenbaugh. Multi-tissue computational modeling analyzes pathophysiology of type 2 diabetes in mkr mice. 9(7):e102319–e102319, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0102319">[DOI]</a>.</p></li>
<li id="id35813"><p>Cheng Wang, Zhi-Luo Deng, Xie Zhi-ming, Xin‐Yi Chu, Ji-Wei Chang, Kong De, Baoju Li, Hongyu Zhang, and Ling‐Ling Chen. Construction of a genome‐scale metabolic network of the plant pathogen &lt;i&gt;pectobacterium carotovorum&lt;/i&gt; provides new strategies for bactericide discovery. 589(3):285–294, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.febslet.2014.12.010">[DOI]</a>.</p></li>
<li id="id35818"><p>Marianna Taffi, Nicola Paoletti, Claudio Angione, Sandra Pucciarelli, Mauro Marini, and Pietro LiÃ. Bioremediation in marine ecosystems: a computational study combining ecological modeling and flux balance analysis. 5():319–319, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fgene.2014.00319">[DOI]</a>.</p></li>
<li id="id35822"><p>Hae Woo Lee, Andrew Christie, Jason A. Starkey, Erik K. Read, and Seongkyu Yoon. Intracellular metabolic flux analysis of &lt;scp&gt;cho&lt;/scp&gt; cells supplemented with wheat hydrolysates for improved &lt;scp&gt;mab&lt;/scp&gt; production and cell‐growth. 90(2):291–302, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/jctb.4523">[DOI]</a>.</p></li>
<li id="id35836"><p>Ali Salehzadeh‐Yazdi, Yazdan Asgari, Ali Akbar Saboury, and Ali Masoudi‐Nejad. Computational analysis of reciprocal association of metabolism and epigenetics in the budding yeast: a genome-scale metabolic model (gsmm) approach. 9(11):e111686–e111686, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0111686">[DOI]</a>.</p></li>
<li id="id35855"><p>Longfei Mao and W.S. Verwoerd. Computational comparison of mediated current generation capacity of chlamydomonas reinhardtii in photosynthetic and respiratory growth modes. 118(5):565–574, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiosc.2014.04.021">[DOI]</a>.</p></li>
<li id="id35866"><p>Pankaj Narang, Shawez Khan, Anmol Jaywant Hemrom, and Andrew M. Lynn. Metanet - a web-accessible interactive platform for biological metabolic network analysis. 8(1):130–130, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-014-0130-2">[DOI]</a>.</p></li>
<li id="id35883"><p>Keesha E. Erickson, Ryan T. Gill, and Anushree Chatterjee. Constrictor: constraint modification provides insight into design of biochemical networks. 9(11):e113820–e113820, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0113820">[DOI]</a>.</p></li>
<li id="id35885"><p>Maurice HT Ling and Chueh Loo Poh. A predictor for predicting escherichia colitranscriptome and the effects of gene perturbations. 15(1):140–140, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-15-140">[DOI]</a>.</p></li>
<li id="id35891"><p>Rafael Carreira, Pedro Evangelista, Paulo Maia, Paulo Vilaça, Marcellinus Pont, Jean-François Tomb, Isabel Rocha, and Miguel Rocha. Cbfa: phenotype prediction integrating metabolic models with constraints derived from experimental data. 8(1):, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/s12918-014-0123-1">[DOI]</a>.</p></li>
<li id="id35912"><p>N.A.L. Flahaut and Willem M. de Vos. Systems biology and metabolic engineering of lactic acid bacteria for improved fermented foods. ():177–196, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-1-78242-015-6.00008-6">[DOI]</a>.</p></li>
<li id="id35951"><p>Guodong Liu, Antonio Marras, and Jens Nielsen. The future of genome‐scale modeling of yeast through integration of a transcriptional regulatory network. 2(1):30–46, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s40484-014-0027-5">[DOI]</a>.</p></li>
<li id="id35966"><p>Alejandro Acevedo, Germán Aroca, and Raúl Conejeros. Genome-scale nad(h/+) availability patterns as a differentiating feature between saccharomyces cerevisiae and scheffersomyces stipitis in relation to fermentative metabolism. 9(1):e87494–e87494, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0087494">[DOI]</a>.</p></li>
<li id="id35968"><p>Zsolt Bodor, ANDREA FAZAKAS, Erika R. Bérczi‐Kovács, Szabolcs Lányi, and B. Albert. Systems biology and metabolic engineering for obtaining e. coli mutants capable to produce succinate from renewable resources. 19(4):9625–9636, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id35974"><p>Viswanadham Sridhara, Austin G. Meyer, Piyush Rai, Jeffrey E. Barrick, Pradeep Ravikumar, Daniel Segrè, and Claus O. Wilke. Predicting growth conditions from internal metabolic fluxes in an&lt;i&gt;in-silico&lt;/i&gt;model of&lt;i&gt;e. coli&lt;/i&gt;. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/002287">[DOI]</a>.</p></li>
<li id="id35989"><p>Arne C. Müller, Frank J. Bruggeman, Brett G. Olivier, and Leen Stougie. Fast flux module detection using matroid theory. ():192–206, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-05269-4_16">[DOI]</a>.</p></li>
<li id="id35991"><p>Andreas Wagner, Vardan Andriasyan, and Aditya Barve. The organization of metabolic genotype space facilitates adaptive evolution in nitrogen metabolism. 3(1):2–13, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.5167/uzh-107400">[DOI]</a>.</p></li>
<li id="id36020"><p>Pratish Gawand and Radhakrishnan Mahadevan. Engineering&lt;i&gt;escherichia coli&lt;/i&gt;for d-ribose production from glucose-xylose mixtures. 10(2):106–114, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0028">[DOI]</a>.</p></li>
<li id="id36035"><p>Min-Kyu Jeon, Boeun Kim, Min‐Gyu Sung, and Jay H. Lee. On integrating the droop model with the flux balance model for predicting metabolic shifts in microalgae growth. 35():1534–1539, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/iccas.2014.6987810">[DOI]</a>.</p></li>
<li id="id36064"><p>Octavio Perez‐Garcia, Silas G. Villas‐Bôas, and Naresh Singhal. A method to calibrate metabolic network models with experimental datasets. ():183–190, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-07581-5_22">[DOI]</a>.</p></li>
<li id="id36072"><p>Mafalda Galhardo, Lasse Sinkkonen, Philipp Berninger, Jake Lin, Thomas Sauter, and Merja Heinäniemi. Transcriptomics profiling of human sgbs adipogenesis. 2():246–248, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gdata.2014.07.004">[DOI]</a>.</p></li>
<li id="id36088"><p>Zaida Luthey‐Schulten. Stochastic simulations of cellular processes: from single cells to colonies. 106(2):7a–7a, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bpj.2013.11.077">[DOI]</a>.</p></li>
<li id="id36091"><p>Sergio Bordel. Genome-scale metabolic models of yeast, methods for their reconstruction, and other applications. 1152():269–279, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-0563-8_16">[DOI]</a>.</p></li>
<li id="id36093"><p>Ana Rita Brochado and Kiran Raosaheb Patil. Model-guided identification of gene deletion targets for metabolic engineering in saccharomyces cerevisiae. 1152():281–294, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4939-0563-8_17">[DOI]</a>.</p></li>
<li id="id36094"><p>Nathalie Poupin and Fabien Jourdan. Analysing human metabolic networks using metabolomics. ():85–114, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-1-78242-084-2.00005-8">[DOI]</a>.</p></li>
<li id="id36113"><p>Mara Sangiovanni. Model checking of metabolic networks: application to metabolic diseases. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36146"><p>Chao Shi, Jian Yin, Zhe Liu, Jianxin Wu, Qi Zhao, Jian Ren, and Nan Yao. A systematic simulation of the effect of salicylic acid on sphingolipid metabolism. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/011676">[DOI]</a>.</p></li>
<li id="id36166"><p>Marion O. Adebiyi. Computational analysis of anopheles gambiae metabolism to facilitate insecticidal target and complex resistance mechanism discovery. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36168"><p>Fahimeh Salimi. Genome-scale characterization of metabolic interactions in a clostridial co-culture for consolidated bioprocessing. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36169"><p>Pratish Gawand. Metabolic engineering for substrate co-utilization. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36179"><p>Diego Salazar Barreto. Predicción de biomarcadores en mucopolisacaridosis empleando una red metabólica humana. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36185"><p>Benjamin VanderSluis. Computational analysis of genome-scale growth-interaction data in saccharomyces cerevisiae. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36186"><p>Kathleen A. Curran. Optimizing gene expression in saccharomyces cerevisiae for metabolic engineering applications. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36190"><p>Kevin Tsai and Chuan-Hsiung Chang. Diagnostics for stochastic genome-scale modeling via model slicing and debugging. 9(11):e110380–e110380, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0110380">[DOI]</a>.</p></li>
<li id="id36191"><p>Pei-Jung Wu, Wu‐Hsiung Wu, Tzu-Chi Chen, Kuan‐Ting Lin, Jin-Mei Lai, Chi‐Ying F. Huang, and Feng‐Sheng Wang. Reconstruction and analysis of a signal transduction network using hela cell protein–protein interaction data. 45(6):2835–2842, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtice.2014.07.006">[DOI]</a>.</p></li>
<li id="id36193"><p>Zhaobin Xu, Xin Fang, Thomas K. Wood, and Zuyi Huang. Modeling framework for investigating the influence of amino acids on the planktonic-biofilm transition of pseudomonas aeruginosa. 47(3):803–808, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.3182/20140824-6-za-1003.01325">[DOI]</a>.</p></li>
<li id="id36200"><p>Orlando Rocha, Paulo Vilaça, Miguel Rocha, and Rui Mendes. Reg4optflux: an optflux plug-in that comprises meta-heuristics approaches for metabolic engineering using integrated models. 103():226–231, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/nabic.2014.6921882">[DOI]</a>.</p></li>
<li id="id36206"><p>Priyanka Baloni, Soma Ghosh, and Nagasuma Chandra. Systems approaches to study infectious diseases. ():151–172, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-017-9514-2_8">[DOI]</a>.</p></li>
<li id="id36210"><p>Jeevan Amos and Sambasiva Rao. Computational approach to explore plasmodium falciparum 3d7 seryl trna synthetase structure. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36211"><p>Chunli Liu and Tianwei Tan. 156. metabolically engineered escherichia coli for isoprene biosynthesis. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36218"><p>Benjamín J. Sánchez. Development, calibration and validation of a dynamic genome-scale metabolic model of saccharomyces cerevisiae. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.7764/tesisuc/ing/2965">[DOI]</a>.</p></li>
<li id="id36220"><p>Sylvain Prigent. Thèse / université de rennes 1 sous le sceau de l'université européenne de bretagne pour le grade de docteur de l'université de rennes 1 mention : informatique. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36234"><p>Aditya Reddy Kolli. Action potentials as indicators of metabolic perturbations for temporal proteomic analysis. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36248"><p>J. Hettling. Viable models of energy metabolism. ():, 2014. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id36256"><p>Monica Larocque, Thierry Chénard, and Rafaël Najmanovich. A curated c. difficile strain 630 metabolic network: prediction of essential targets and inhibitors. ():, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.1101/006932">[DOI]</a>.</p></li>
<li id="id36290"><p>Maurice HT Ling. 1(5):, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.15406/mojpb.1.5">[DOI]</a>.</p></li>
<li id="id36292"><p>Maurice HT Ling. 1(3):, 2014. <a class="reference external" href="https://doi.org/https://doi.org/10.15406/mojpb.1.3">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-12" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-12" name="0-12" role="tabpanel" tabindex="0"><div class="docutils container" id="id37453">
<ol class="arabic simple" start="1">
<li id="id37467"><p>Mark P. Brynildsen, Jonathan Winkler, Catherine S. Spina, I. Cody MacDonald, and James J. Collins. Potentiating antibacterial activity by predictably enhancing endogenous microbial ros production. 31(2):160–165, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.2458">[DOI]</a>.</p></li>
<li id="id37470"><p>Rasmus Ågren, Li Liu, Saeed Shoaie, Wanwipa Vongsangnak, Intawat Nookaew, and Jens Nielsen. The raven toolbox and its use for generating a genome-scale metabolic model for penicillium chrysogenum. 9(3):e1002980–e1002980, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1002980">[DOI]</a>.</p></li>
<li id="id37485"><p>Alison K. Hottes, Peter L. Freddolino, Anupama Khare, Zachary N. Donnell, Julia Liu, and Saeed Tavazoie. Bacterial adaptation through loss of function. 9(7):e1003617–e1003617, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pgen.1003617">[DOI]</a>.</p></li>
<li id="id37503"><p>Anirikh Chakrabarti, Ljubiša Mišković, Keng Cher Soh, and Vassily Hatzimanikatis. Towards kinetic modeling of genome‐scale metabolic networks without sacrificing stoichiometric, thermodynamic and physiological constraints. 8(9):1043–1057, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201300091">[DOI]</a>.</p></li>
<li id="id37511"><p>Scott Devoid, Ross Overbeek, Matthew DeJongh, Veronika Vonstein, Aaron A. Best, and Christopher S. Henry. Automated genome annotation and metabolic model reconstruction in the seed and model seed. 985():17–45, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_2">[DOI]</a>.</p></li>
<li id="id37512"><p>Gabriel Gelius‐Dietrich, Abdelmoneim Amer Desouki, Claus Jonathan Fritzemeier, and Martin J. Lercher. Sybil – efficient constraint-based modelling in r. 7(1):125–125, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-125">[DOI]</a>.</p></li>
<li id="id37533"><p>Douglas B. Kell. Finding novel pharmaceuticals in the systems biology era using multiple effective drug targets, phenotypic screening and knowledge of transporters: where drug discovery went wrong and how to fix it. 280(23):5957–5980, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/febs.12268">[DOI]</a>.</p></li>
<li id="id37537"><p>Mathias Ganter, Thomas Bernard, Sébastien Moretti, Joerg Stelling, and Marco Pagni. Metanetx.org: a website and repository for accessing, analysing and manipulating metabolic networks. 29(6):815–816, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt036">[DOI]</a>.</p></li>
<li id="id37605"><p>Matthew B. Biggs and Jason A. Papin. Novel multiscale modeling tool applied to pseudomonas aeruginosa biofilm formation. 8(10):e78011–e78011, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0078011">[DOI]</a>.</p></li>
<li id="id37612"><p>Wei Wu, Neema Jamshidi, Satish A. Eraly, Henry C. Liu, Kevin T. Bush, Bernhard Ø. Palsson, and Sanjay K. Nigám. Multispecific drug transporter slc22a8 (oat3) regulates multiple metabolic and signaling pathways. 41(10):1825–1834, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1124/dmd.113.052647">[DOI]</a>.</p></li>
<li id="id37617"><p>Tat‐Ming Lo, Wei Suong Teo, Hua Ling, Binbin Chen, Aram Kang, and Matthew Wook Chang. Microbial engineering strategies to improve cell viability for biochemical production. 31(6):903–914, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2013.02.001">[DOI]</a>.</p></li>
<li id="id37621"><p>Pramod R. Somvanshi and K. V. Venkatesh. A conceptual review on systems biology in health and diseases: from biological networks to modern therapeutics. 8(1):99–116, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11693-013-9125-3">[DOI]</a>.</p></li>
<li id="id37624"><p>Jonathan M. Dreyfuss, Jeremy Zucker, Heather M. Hood, Linda R. Ocasio, Matthew S. Sachs, and James E. Galagan. Reconstruction and validation of a genome-scale metabolic model for the filamentous fungus neurospora crassa using farm. 9(7):e1003126–e1003126, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003126">[DOI]</a>.</p></li>
<li id="id37625"><p>Namrata Tomar and Rajat K. De. Comparing methods for metabolic network analysis and an application to metabolic engineering. 521(1):1–14, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.gene.2013.03.017">[DOI]</a>.</p></li>
<li id="id37647"><p>Tong Hao, Binbin Han, Hongwu Ma, Jing Fu, Hui Wang, Zhiwen Wang, Bincai Tang, Tao Chen, and Xueming Zhao. In silico metabolic engineering of bacillus subtilis for improved production of riboflavin, egl-237, (r,r)-2,3-butanediol and isobutanol. 9(8):2034–2034, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c3mb25568a">[DOI]</a>.</p></li>
<li id="id37670"><p>Sarah Schatschneider, Marcus Persicke, Steven Alexander Watt, Gerd Hublik, Alfred Pühler, Karsten Niehaus, and Frank‐Jörg Vorhölter. Establishment, in silico analysis, and experimental verification of a large-scale metabolic network of the xanthan producing xanthomonas campestris pv. campestris strain b100. 167(2):123–134, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2013.01.023">[DOI]</a>.</p></li>
<li id="id37681"><p>Zixiang Xu, Ping Zheng, Jibin Sun, and Yanhe Ma. Reacknock: identifying reaction deletion strategies for microbial strain optimization based on genome-scale metabolic network. 8(12):e72150–e72150, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0072150">[DOI]</a>.</p></li>
<li id="id37684"><p>Carl Yulun Song, Melissa A. Chiasson, Nirvana Nursimulu, Stacy Hung, James D. Wasmuth, Michael E. Grigg, and John Parkinson. Metabolic reconstruction identifies strain‐specific regulation of virulence in &lt;i&gt;toxoplasma gondii&lt;/i&gt;. 9(1):708–708, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2013.62">[DOI]</a>.</p></li>
<li id="id37693"><p>Matthew C. Gonnerman, Matthew N. Benedict, Adam M. Feist, William W. Metcalf, and Nathan D. Price. Genomically and biochemically accurate metabolic reconstruction of &lt;i&gt;methanosarcina barkeri&lt;/i&gt; fusaro, img746. 8(9):1070–1079, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201200266">[DOI]</a>.</p></li>
<li id="id37739"><p>Zachary A. King and Adam M. Feist. Optimizing cofactor specificity of oxidoreductase enzymes for the generation of microbial production strains—optswap. 9(4):236–246, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0005">[DOI]</a>.</p></li>
<li id="id37740"><p>Hyun Ju Kim, Bo Kyeng Hou, Sung Gun Lee, Joong Su Kim, Dong‐Woo Lee, and Sang Jun Lee. Genome-wide analysis of redox reactions reveals metabolic engineering targets for d-lactate overproduction in escherichia coli. 18():44–52, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2013.03.004">[DOI]</a>.</p></li>
<li id="id37747"><p>Sriram Chandrasekaran and Nathan D. Price. Metabolic constraint-based refinement of transcriptional regulatory networks. 9(12):e1003370–e1003370, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003370">[DOI]</a>.</p></li>
<li id="id37771"><p>Francisco G Vital-Lopez, Anders Wallqvist, and Jaques Reifman. Bridging the gap between gene expression and metabolic phenotype via kinetic models. 7(1):63–63, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-63">[DOI]</a>.</p></li>
<li id="id37806"><p>Zhaobin Xu, Xin Fang, Thomas K. Wood, and Zuyi Huang. A systems-level approach for investigating pseudomonas aeruginosa biofilm formation. 8(2):e57050–e57050, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0057050">[DOI]</a>.</p></li>
<li id="id37833"><p>Steffen Krauser, Christian Weyler, Lisa Katharina Blaß, and Elmar Heinzle. Directed multistep biocatalysis using tailored permeabilized cells. 137():185–234, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2013_240">[DOI]</a>.</p></li>
<li id="id37837"><p>K. P. Lisha and Debasis Sarkar. Dynamic flux balance analysis of batch fermentation: effect of genetic manipulations on ethanol production. 37(4):617–627, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-013-1027-y">[DOI]</a>.</p></li>
<li id="id37849"><p>Kai Zhuang, Bhavik R. Bakshi, and Markus J. Herrgård. Multi‐scale modeling for sustainable chemical production. 8(9):973–984, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201200272">[DOI]</a>.</p></li>
<li id="id37868"><p>Bevan Kai-Sheng Chung, Meiyappan Lakshmanan, Maximilian Klement, Chi Bun Ching, and Dong‐Yup Lee. Metabolic reconstruction and flux analysis of industrial pichia yeasts. 97(5):1865–1873, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-013-4702-7">[DOI]</a>.</p></li>
<li id="id37880"><p>Margaret Simons, Ashish Misra, and Ganesh Sriram. Genome-scale models of plant metabolism. 1083():213–230, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-661-0_13">[DOI]</a>.</p></li>
<li id="id37884"><p>Sumanta Mukherjee, Awanti Sambarey, Prashanthi Karyala, and Nagasuma Chandra. Current trends in modeling host–pathogen interactions. 3(2):109–128, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/widm.1085">[DOI]</a>.</p></li>
<li id="id37892"><p>Ivan Lule, Pieter-Jan D’Huys, Lieve Van Mellaert, Jozef Anné, Kristel Bernaerts, and Jan Van Impe. Metabolic impact assessment for heterologous protein production in streptomyces lividans based on genome-scale metabolic network modeling. 246(1):113–121, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mbs.2013.08.006">[DOI]</a>.</p></li>
<li id="id37908"><p>Sidhartha Chaudhury, Mohamed Diwan M. AbdulHameed, Narender Singh, Gregory J. Tawa, Patrik D’haeseleer, Adam Zemła, Ali Navid, Carol E. Zhou, Matthew C. Franklin, Jonah Cheung, M. Rudolph, J. Love, John Graf, David A. Rozak, Jennifer L. Dankmeyer, Kei Amemiya, Simon Daefler, and Anders Wallqvist. Rapid countermeasure discovery against francisella tularensis based on a metabolic network reconstruction. 8(5):e63369–e63369, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0063369">[DOI]</a>.</p></li>
<li id="id37941"><p>Janne Wallenius, Matti Viikilä, Shrikant A. Survase, Heikki Ojamo, and Tero Eerikäinen. Constraint-based genome-scale metabolic modeling of clostridium acetobutylicum behavior in an immobilized column. 142():603–610, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biortech.2013.05.085">[DOI]</a>.</p></li>
<li id="id37948"><p>Jacob E. Koskimaki, Anna S. Blazier, Andrés F. Clarens, and Jason A. Papin. Computational models of algae metabolism for industrial applications. 9(4):185–195, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0012">[DOI]</a>.</p></li>
<li id="id38022"><p>Wei Tong, Zhen Chen, Zhe Cao, Quanhui Wang, Jiyuan Zhang, Xue Bai, Rong Wang, and Siqi Liu. Robustness analysis of a constraint-based metabolic model links cell growth and proteomics of thermoanaerobacter tengcongensis under temperature perturbation. 9(4):713–713, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c3mb25278g">[DOI]</a>.</p></li>
<li id="id38033"><p>Sarat C. Cautha, Christopher M. Gowen, François-Xavier Lussier, Nicholas D. Gold, Vincent J. J. Martin, and Radhakrishnan Mahadevan. Model-driven design of a saccharomyces cerevisiae platform strain with improved tyrosine production capabilities. 46(31):221–226, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.3182/20131216-3-in-2044.00066">[DOI]</a>.</p></li>
<li id="id38038"><p>Patrick Hyland, Serene Sow Mun Lock, and Radhakrishnan Mahadevan. Prediction of weak acid toxicity in&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;using genome-scale metabolic models. 9(4):229–235, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0004">[DOI]</a>.</p></li>
<li id="id38065"><p>Jordan O. Hay and Jörg Schwender. Flux variability analysis: application to developing oilseed rape embryos using toolboxes for constraint-based modeling. 1090():301–316, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-688-7_18">[DOI]</a>.</p></li>
<li id="id38083"><p>Giuseppe Facchetti and Claudio Altafini. Partial inhibition and bilevel optimization in flux balance analysis. 14(1):344–344, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-14-344">[DOI]</a>.</p></li>
<li id="id38098"><p>N. C. Santos, Maria Olívia Pereira, and Anália Lourenço. Pathogenicity phenomena in three model systems: from network mining to emerging system-level properties. 16(1):169–182, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbt071">[DOI]</a>.</p></li>
<li id="id38113"><p>Hong Yang, Dominic E. Mandy, and Igor G. L. Libourel. Optimal design of isotope labeling experiments. 1083():133–147, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-661-0_9">[DOI]</a>.</p></li>
<li id="id38114"><p>Claudio Angione, Giovanni Carapezza, Jole Costanza, Píetro Lió, and Giuseppe Nicosia. Design and strain selection criteria for bacterial communication networks. 4(4):155–163, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.nancom.2013.08.001">[DOI]</a>.</p></li>
<li id="id38123"><p>Lanay Tierney, Katarzyna M. Tyc, Edda Klipp, and Karl Kuchler. 3 systems biology approaches to understanding and predicting fungal virulence. ():45–74, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-642-39432-4_3">[DOI]</a>.</p></li>
<li id="id38126"><p>Eva Grafahrend‐Belau, Astrid Junker, Falk Schreiber, and Björn H. Junker. Flux balance analysis as an alternative method to estimate fluxes without labeling. 1090():281–299, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-688-7_17">[DOI]</a>.</p></li>
<li id="id38166"><p>Zhaobin Xu, Xin Fang, Thomas K. Wood, and Zuyi Huang. Gene target identification for biofilm-associated pathogens: an application to pseudomonas aeruginosa. 2():3765–3770, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/acc.2013.6580413">[DOI]</a>.</p></li>
<li id="id38196"><p>Paul Jensén. Data integration with constraint-based genome-scale models. ():, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.18130/v37v4m">[DOI]</a>.</p></li>
<li id="id38207"><p>Andrés Mauricio Pinzón Velasco, Silvia Restrepo, and Andrés Fernando González Barrios. Fba analysis, plant-pathogen interactions. ():733–736, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4419-9863-7_1170">[DOI]</a>.</p></li>
<li id="id38238"><p>Ali Ebrahim, Joshua A. Lerman, Bernhard Ø. Palsson, and Daniel R. Hyduke. Cobrapy: constraints-based reconstruction and analysis for python. 7(1):74–74, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-74">[DOI]</a>.</p></li>
<li id="id38239"><p>Ines Thiele, Neil Swainston, Ronan M. T. Fleming, Andreas Hoppe, Swagatika Sahoo, Maike K. Aurich, Hulda S. Haraldsdóttir, Monica L. Mo, Óttar Rolfsson, Miranda D. Stobbe, Stefan Gretar Thorleifsson, Rasmus Ågren, Christian Bölling, Sergio Bordel, Arvind K. Chavali, Paul D. Dobson, Warwick B. Dunn, Lukas Endler, David Hala, Michael Hucka, Duncan Hull, Daniel Jameson, Neema Jamshidi, Jón J. Jónsson, Nick Juty, Sarah Keating, Intawat Nookaew, Nicolas Le Novère, Naglis Malys, Alexander Mazein, Jason A. Papin, Nathan D. Price, E Selkov, Martin I. Sigurðsson, Evangelos Simeonidis, Nikolaus Sonnenschein, Kieran Smallbone, Anatoly Sorokin, Hans van Beek, Dieter Weichart, Igor Goryanin, Jens Nielsen, Hans V. Westerhoff, Douglas B. Kell, Pedro Mendes, and Bernhard Ø. Palsson. A community-driven global reconstruction of human metabolism. 31(5):419–425, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.2488">[DOI]</a>.</p></li>
<li id="id38247"><p>Koyel Mitra, Anne‐Ruxandra Carvunis, Sanath Kumar Ramesh, and Trey Ideker. Integrative approaches for finding modular structure in biological networks. 14(10):719–732, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nrg3552">[DOI]</a>.</p></li>
<li id="id38258"><p>Εlad Noor, Hulda S. Haraldsdóttir, Ron Milo, and Ronan M. T. Fleming. Consistent estimation of gibbs energy using component contributions. 9(7):e1003098–e1003098, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003098">[DOI]</a>.</p></li>
<li id="id38268"><p>Henning Knoop, Marianne Gründel, Yvonne Zilliges, Robert Lehmann, Sabrina Hoffmann, Wolfgang Lockau, and Ralf Steuer. Flux balance analysis of cyanobacterial metabolism: the metabolic network of synechocystis sp. pcc 6803. 9(6):e1003081–e1003081, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1003081">[DOI]</a>.</p></li>
<li id="id38273"><p>Almut Heinken, Swagatika Sahoo, Ronan M. T. Fleming, and Ines Thiele. Systems-level characterization of a host-microbe metabolic symbiosis in the mammalian gut. 4(1):28–40, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.4161/gmic.22370">[DOI]</a>.</p></li>
<li id="id38287"><p>Benjamin Steeb, Beatrice Claudi, Neil A. Burton, Petra Tienz, Alexander Schmidt, Hesso Farhan, Alain Mazé, and Dirk Bumann. Parallel exploitation of diverse host nutrients enhances salmonella virulence. 9(4):e1003301–e1003301, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.ppat.1003301">[DOI]</a>.</p></li>
<li id="id38288"><p>Finja Büchel, Nicolás Rodríguez, Neil Swainston, Clemens Wrzodek, Tobias Czauderna, Roland Keller, Florian Mittag, Michaël Schubert, Mihai Glont, Martin Golebiewski, Martijn van Iersel, Sarah Keating, Matthias Rall, Michael Wybrow, Henning Hermjakob, Michael Hucka, Douglas B. Kell, Wolfgang G. Müller, Pedro Mendes, Andreas Zell, Claudine Chaouiya, Julio Sáez-Rodríguez, Falk Schreiber, Camille Laibe, Andreas Dräger, and Nicolas Le Novère. Path2models: large-scale generation of computational models from biochemical pathway maps. 7(1):116–116, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-116">[DOI]</a>.</p></li>
<li id="id38293"><p>Hnin W. Aung, Susan A. Henry, and Larry P. Walker. Revising the representation of fatty acid, glycerolipid, and glycerophospholipid metabolism in the consensus model of yeast metabolism. 9(4):215–228, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0013">[DOI]</a>.</p></li>
<li id="id38296"><p>Harish Nagarajan, Merve Şahin, Juan Nogales, Haythem Latif, Derek R. Lovley, Ali Ebrahim, and Karsten Zengler. Characterizing acetogenic metabolism using a genome-scale metabolic reconstruction of clostridium ljungdahlii. 12(1):118–118, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-12-118">[DOI]</a>.</p></li>
<li id="id38315"><p>Harish Nagarajan, Mallory Embree, Amelia‐Elena Rotaru, Pravin Malla Shrestha, Adam M. Feist, Bernhard Ø. Palsson, Derek R. Lovley, and Karsten Zengler. Characterization and modelling of interspecies electron transfer mechanisms and microbial community dynamics of a syntrophic association. 4(1):2809–2809, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms3809">[DOI]</a>.</p></li>
<li id="id38323"><p>N.A.L. Flahaut, Anne Wiersma, Bert van de Bunt, Dirk E. Martens, Peter J. Schaap, Lolke Sijtsma, Vítor A. P. Martins dos Santos, and Willem M. de Vos. Genome-scale metabolic model for lactococcus lactis mg1363 and its application to the analysis of flavor formation. 97(19):8729–8739, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-013-5140-2">[DOI]</a>.</p></li>
<li id="id38326"><p>Simon Unthan, Alexander Grünberger, Jan van Ooyen, Jochem Gätgens, Johanna Heinrich, Nicole Paczia, Wolfgang Wiechert, Dietrich Kohlheyer, and Stephan Noack. Beyond growth rate 0.6: what drives &lt;i&gt;corynebacterium glutamicum&lt;/i&gt; to higher growth rates in defined medium. 111(2):359–371, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.25103">[DOI]</a>.</p></li>
<li id="id38332"><p>Roger L. Chang, Kathleen A. Andrews, Donghyuk Kim, Zhanwen Li, Adam Godzik, and Bernhard Ø. Palsson. Structural systems biology evaluation of metabolic thermotolerance in &lt;i&gt;escherichia coli&lt;/i&gt;. 340(6137):1220–1223, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/science.1234012">[DOI]</a>.</p></li>
<li id="id38349"><p>Ben Heavner, Kieran Smallbone, Nathan D. Price, and Larry P. Walker. Version 6 of the consensus yeast metabolic network refines biochemical coverage and improves model performance. 2013():bat059–bat059, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/database/bat059">[DOI]</a>.</p></li>
<li id="id38368"><p>Piyush Labhsetwar, John A. Cole, Elijah Roberts, Nathan D. Price, and Zaida Luthey‐Schulten. Heterogeneity in protein expression induces metabolic variability in a modeled &lt;i&gt;escherichia coli&lt;/i&gt; population. 110(34):14006–14011, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1222569110">[DOI]</a>.</p></li>
<li id="id38409"><p>Joshua J. Hamilton and Jennifer L. Reed. Software platforms to facilitate reconstructing genome‐scale metabolic networks. 16(1):49–59, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1462-2920.12312">[DOI]</a>.</p></li>
<li id="id38425"><p>Yi Wang, Xiangzhen Li, Caroline B. Milne, Holger Janssen, Weiyin Lin, Gloria Phan, Huiying Hu, Yong‐Su Jin, Nathan D. Price, and Hans P. Blaschek. Development of a gene knockout system using mobile group ii introns (targetron) and genetic disruption of acid production pathways in clostridium beijerinckii. 79(19):5853–5863, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.00971-13">[DOI]</a>.</p></li>
<li id="id38434"><p>Jennifer Bartell, P. H. Yen, John Varga, Joanna B. Goldberg, and Jason A. Papin. Comparative metabolic systems analysis of pathogenic burkholderia. 196(2):210–226, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00997-13">[DOI]</a>.</p></li>
<li id="id38462"><p>Judith A. H. Wodke, Jacek Puchałka, María Lluch‐Senar, Josep Marcos, Eva Yus, Miguel Godinho, Ricardo Gutiérrez–Gallego, Vítor A. P. Martins dos Santos, Luís Serrano, Edda Klipp, and Tobias Maier. Dissecting the energy metabolism in &lt;i&gt;mycoplasma pneumoniae&lt;/i&gt; through genome‐scale metabolic modeling. 9(1):653–653, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2013.6">[DOI]</a>.</p></li>
<li id="id38465"><p>Di Huang, Shanshan Li, Menglei Xia, Jianping Wen, and Xiaoqiang Jia. Genome-scale metabolic network guided engineering of streptomyces tsukubaensis for fk506 production improvement. 12(1):52–52, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-12-52">[DOI]</a>.</p></li>
<li id="id38474"><p>Swagatika Sahoo and Ines Thiele. Predicting the impact of diet and enzymopathies on human small intestinal epithelial cells. 22(13):2705–2722, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/hmg/ddt119">[DOI]</a>.</p></li>
<li id="id38493"><p>Kieran Smallbone and Pedro Mendes. Large-scale metabolic models: from reconstruction to differential equations. 9(4):179–184, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0003">[DOI]</a>.</p></li>
<li id="id38508"><p>Alexander Golberg, Edward Vitkin, Gregory Linshiz, Sabaa A. Khan, Nathan J. Hillson, Zohar Yakhini, and Martin L. Yarmush. Proposed design of distributed macroalgal biorefineries: thermodynamics, bioconversion technology, and sustainability implications for developing economies. 8(1):67–82, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bbb.1438">[DOI]</a>.</p></li>
<li id="id38527"><p>Young‐Mo Kim, Brian J. Schmidt, Afshan S. Kidwai, Marcus B. Jones, Brooke L. Deatherage Kaiser, Heather M. Brewer, Hugh Mitchell, Bernhard Ø. Palsson, Jason McDermott, Fred Heffron, Richard Smith, Scott N. Peterson, Charles Ansong, Daniel R. Hyduke, Thomas Metz, and Joshua Adkins. Salmonella modulates metabolism during growth under conditions that induce expression of virulence genes. 9(6):1522–1522, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c3mb25598k">[DOI]</a>.</p></li>
<li id="id38539"><p>Agris Pentjuss, Ilona Odzina, Andrejs Kostromins, David A. Fell, Egils Stalidzāns, and Uldis Kalnenieks. Biotechnological potential of respiring zymomonas mobilis: a stoichiometric analysis of its central metabolism. 165(1):1–10, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2013.02.014">[DOI]</a>.</p></li>
<li id="id38559"><p>Arne C. Müller and Alexander Bockmayr. Fast thermodynamically constrained flux variability analysis. 29(7):903–909, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt059">[DOI]</a>.</p></li>
<li id="id38585"><p>Daniele de Martino, Fabrizio Capuani, Matteo Mori, Andrea De Martino, Enzo Marinari, Daniele de Martino, Fabrizio Capuani, Matteo Mori, Andrea De Martino, and Enzo Marinari. Counting and correcting thermodynamically infeasible flux cycles in genome-scale metabolic networks. 3(4):946–966, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo3040946">[DOI]</a>.</p></li>
<li id="id38615"><p>Jie Liu, Qian Gao, Nan Xu, and Li Liu. Genome-scale reconstruction and in silico analysis of aspergillus terreus metabolism. 9(7):1939–1939, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c3mb70090a">[DOI]</a>.</p></li>
<li id="id38675"><p>Wei Zou, M. Zhou, Li Liu, and Jian Chen. Reconstruction and analysis of the industrial strain bacillus megaterium wsh002 genome-scale in silico metabolic model. 164(4):503–509, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2013.01.019">[DOI]</a>.</p></li>
<li id="id38692"><p>Roger L. Chang, Lei Xie, Philip E. Bourne, and Bernhard Ø. Palsson. Antibacterial mechanisms identified through structural systems pharmacology. 7(1):102–102, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-102">[DOI]</a>.</p></li>
<li id="id38703"><p>Longfei Mao and W.S. Verwoerd. Model-driven elucidation of the inherent capacity of geobacter sulfurreducens for electricity generation. 7(1):14–14, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1754-1611-7-14">[DOI]</a>.</p></li>
<li id="id38749"><p>Manuel Garcia‐Albornoz and Jens Nielsen. Application of genome-scale metabolic models in metabolic engineering. 9(4):203–214, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.0011">[DOI]</a>.</p></li>
<li id="id38753"><p>Shubo Li, Nan Xu, Li Liu, and Jian Chen. Engineering of carboligase activity reaction in candida glabrata for acetoin production. 22():32–39, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2013.12.005">[DOI]</a>.</p></li>
<li id="id38764"><p>Meng Liang, Andrew Damiani, Q. Peter He, and Jin Wang. Elucidating xylose metabolism of &lt;i&gt;scheffersomyces stipitis&lt;/i&gt; for lignocellulosic ethanol production. 2(1):38–48, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/sc400265g">[DOI]</a>.</p></li>
<li id="id38774"><p>Miguel Ponce-de-León, Francisco Montero, and Juli Peretό. Solving gap metabolites and blocked reactions in genome-scale models: application to the metabolic network of blattabacterium cuenoti. 7(1):114–114, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-114">[DOI]</a>.</p></li>
<li id="id38779"><p>Yuekai Sun, Ronan M. T. Fleming, Ines Thiele, and Michael A. Saunders. Robust flux balance analysis of multiscale biochemical reaction networks. 14(1):240–240, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-14-240">[DOI]</a>.</p></li>
<li id="id38781"><p>Debjit Ray and Ping Ye. Characterization of the metabolic requirements in yeast meiosis. 8(5):e63707–e63707, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0063707">[DOI]</a>.</p></li>
<li id="id38795"><p>Shayan Tabe-Bordbar and Sayed‐Amir Marashi. Finding elementary flux modes in metabolic networks based on flux balance analysis and flux coupling analysis: application to the analysis of escherichia coli metabolism. 35(12):2039–2044, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10529-013-1328-x">[DOI]</a>.</p></li>
<li id="id38800"><p>Jiun Y. Yen, Hadi Nazem‐Bokaee, Benjamin G. Freedman, Ahmad I. M. Athamneh, and Ryan S. Senger. Deriving metabolic engineering strategies from genome‐scale modeling with flux ratio constraints. 8(5):581–594, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201200234">[DOI]</a>.</p></li>
<li id="id38814"><p>Pablo Carbonell, Anne‐Gaëlle Planson, and Jean‐Loup Faulon. Retrosynthetic design of heterologous pathways. 985():149–173, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_9">[DOI]</a>.</p></li>
<li id="id38815"><p>Claudia Patiño, Gustavo Jaime-Muñoz, and Osbaldo Resendis‐Antonio. Systems biology of cancer: moving toward the integrative study of the metabolic alterations in cancer cells. 3():481–481, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fphys.2012.00481">[DOI]</a>.</p></li>
<li id="id38817"><p>Colin R. Harwood, Susanne Pohl, Wendy D. Smith, and Anil Wipat. Bacillus subtilis. ():87–117, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-417029-2.00004-2">[DOI]</a>.</p></li>
<li id="id38818"><p>Yu Qiu, Harish Nagarajan, Mallory Embree, Wendy Shieu, Elisa Ann Abate, Katy Juárez, Byung‐Kwan Cho, James G. Elkins, Kelly P. Nevin, Christian Barrett, Derek R. Lovley, Bernhard Ø. Palsson, and Karsten Zengler. Characterizing the interplay between multiple levels of organization within bacterial sigma factor regulatory networks. 4(1):1755–1755, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ncomms2743">[DOI]</a>.</p></li>
<li id="id38830"><p>Soma Ghosh, Priyanka Baloni, Sumanta Mukherjee, Praveen Anand, and Nagasuma Chandra. A multi-level multi-scale approach to study essential genes in mycobacterium tuberculosis. 7(1):132–132, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-132">[DOI]</a>.</p></li>
<li id="id38839"><p>Longfei Mao and W.S. Verwoerd. Orca: a cobra toolbox extension for model-driven discovery and analysis. 30(4):584–585, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt723">[DOI]</a>.</p></li>
<li id="id38876"><p>Ashish Misra, Matthew F. Conway, Joseph Johnnie, Tabish M. Qureshi, Bao Lige, Anne M. Derrick, Eddy C. Agbo, and Ganesh Sriram. Metabolic analyses elucidate non-trivial gene targets for amplifying dihydroartemisinic acid production in yeast. 4():200–200, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2013.00200">[DOI]</a>.</p></li>
<li id="id38902"><p>Yaron Goldstein and Alexander Bockmayr. A lattice-theoretic framework for metabolic pathway analysis. ():178–191, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-642-40708-6_14">[DOI]</a>.</p></li>
<li id="id38915"><p>Luca Marchetti, Vincenzo Manca, Roberto Pagliarini, and Aliccia Bollig‐Fischer. Mp modelling for systems biology: two case studies. ():223–245, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-319-03191-0_7">[DOI]</a>.</p></li>
<li id="id38932"><p>Finja Büchel, Sandra Saliger, Andreas Dräger, Stephanie Hoffmann, Clemens Wrzodek, Andreas Zell, and Philipp J. Kahle. Parkinson’s disease: dopaminergic nerve cell model is consistent with experimental finding of increased extracellular transport of α-synuclein. 14(1):136–136, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2202-14-136">[DOI]</a>.</p></li>
<li id="id38935"><p>John T. Sauls and Joerg M. Buescher. Assimilating genome-scale metabolic reconstructions with modelborgifier. 30(7):1036–1038, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btt747">[DOI]</a>.</p></li>
<li id="id38940"><p>Ashwini Kumar Sharma and Rainer König. Metabolic network modeling approaches for investigating the “hungry cancer”. 23(4):227–234, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.semcancer.2013.05.001">[DOI]</a>.</p></li>
<li id="id38952"><p>Aidi Zhang, Shao‐Xing Dai, and Jing‐Fei Huang. Reconstruction and analysis of human kidney-specific metabolic network based on omics data. 2013():1–11, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2013/187509">[DOI]</a>.</p></li>
<li id="id38956"><p>Nicole L. Fong, Joshua A. Lerman, Irene Lam, Bernhard Ø. Palsson, and Pep Charusanti. Reconciling a&lt;i&gt;salmonella enterica&lt;/i&gt;metabolic model with experimental data confirms that overexpression of the glyoxylate shunt can rescue a lethal&lt;i&gt;ppc&lt;/i&gt;deletion mutant. 342(1):62–69, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/1574-6968.12109">[DOI]</a>.</p></li>
<li id="id38963"><p>Edik M. Blais, Arvind K. Chavali, and Jason A. Papin. Linking genome-scale metabolic modeling and genome annotation. 985():61–83, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_4">[DOI]</a>.</p></li>
<li id="id38975"><p>Cristiana G. O. Dal’Molin, Lake‐Ee Quek, Robin Palfreyman, and Lars K. Nielsen. Plant genome-scale modeling and implementation. 1090():317–332, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-688-7_19">[DOI]</a>.</p></li>
<li id="id38984"><p>Longfei Mao and W.S. Verwoerd. Exploration and comparison of inborn capacity of aerobic and anaerobic metabolisms of&lt;i&gt;&lt;i&gt;saccharomyces cerevisiae&lt;/i&gt;&lt;/i&gt;for microbial electrical current production. 4(6):420–430, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.4161/bioe.26222">[DOI]</a>.</p></li>
<li id="id38985"><p>Arnau Montagud, D. Gamermann, Pedro Fernández de Córdoba, and Javier F. Urchueguía. &lt;i&gt;synechocystis&lt;/i&gt;sp. pcc6803 metabolic models for the enhanced production of hydrogen. 35(2):184–198, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.3109/07388551.2013.829799">[DOI]</a>.</p></li>
<li id="id38986"><p>Longfei Mao and W.S. Verwoerd. Genome-scale stoichiometry analysis to elucidate the innate capability of the cyanobacterium &lt;i&gt;synechocystis&lt;/i&gt; for electricity generation. 40(10):1161–1180, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s10295-013-1308-0">[DOI]</a>.</p></li>
<li id="id39067"><p>Jing Meng, Zixiang Xu, Jing Guo, Yunxia Yue, and Xiao Sun. Analysis of enhanced current-generating mechanism of geobacter sulfurreducens strain via model-driven metabolism simulation. 8(9):e73907–e73907, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0073907">[DOI]</a>.</p></li>
<li id="id39068"><p>Mona Yousofshahi, Ehsan Ullah, Russell Stern, and Soha Hassoun. Mc3: a steady-state model and constraint consistency checker for biochemical networks. 7(1):129–129, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-7-129">[DOI]</a>.</p></li>
<li id="id39097"><p>Agris Pentjuss, Oskars Rubenis, D. Bauze, Lilija Aprupe, and Baiba Lāce. Flux variability analysis approach of autism related metabolism in stoichiometric model of mitochondria. 2(2):37–42, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.11592/bit.131102">[DOI]</a>.</p></li>
<li id="id39101"><p>Ryan S. Senger and Hadi Nazem‐Bokaee. Resolving cell composition through simple measurements, genome-scale modeling, and a genetic algorithm. 985():85–101, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_5">[DOI]</a>.</p></li>
<li id="id39114"><p>Stephan Pabinger and Zlatko Trajanoski. Genome-scale model management and comparison. 985():3–16, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_1">[DOI]</a>.</p></li>
<li id="id39122"><p>Agris Pentjuss and Uldis Kalnenieks. Assessment of zymomonas mobilis biotechnological potential in ethanol production by flux variability analysis. 3(1):1–5, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.11592/bit.140502">[DOI]</a>.</p></li>
<li id="id39187"><p>Meng Liang, Q. Peter He, Thomas W. Jeffries, and Jin Wang. Elucidating xylose metabolism of scheffersomyces stipitis by integrating principal component analysis with flux balance analysis. 56():3777–3782, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/acc.2013.6580415">[DOI]</a>.</p></li>
<li id="id39188"><p>Martins Mednis and Armands Vīgants. Automatic comparison of metabolites names: impact of criteria thresholds. 2(1):1–5, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.11592/bit.130501">[DOI]</a>.</p></li>
<li id="id39189"><p>Zixiang Xu, Xiao Sun, and Jibin Sun. Construction and analysis of the model of energy metabolism in e. coli. 8(1):e55137–e55137, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0055137">[DOI]</a>.</p></li>
<li id="id39194"><p>Jurijs Meitalovs and Egils Stalidzāns. Analysis of synthetic metabolic pathways solution space. ():183–187, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/icsse.2013.6614656">[DOI]</a>.</p></li>
<li id="id39197"><p>Pratish Gawand, Laurence Yang, W.R. Cluett, and Radhakrishnan Mahadevan. Metabolic model refinement using phenotypic microarray data. 985():47–59, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-62703-299-5_3">[DOI]</a>.</p></li>
<li id="id39210"><p>Tatjana Rubina, Martins Mednis, and Egils Stalidzāns. Agreement assessment of biochemical pathway models by structural analysis of their intersection. 40():411–418, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cinti.2013.6705232">[DOI]</a>.</p></li>
<li id="id39237"><p>Brett G. Olivier and Frank Bergmann. Sbml level 3 package: flux balance constraints ('fbc'). ():, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.6084/m9.figshare.664700.v1">[DOI]</a>.</p></li>
<li id="id39240"><p>Nathan D. Price. Genome-scale models for microbial factories. 9(4):177–178, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2013.1586">[DOI]</a>.</p></li>
<li id="id39241"><p>Cesar A. Vargas-García, Carlos Eduardo García Sánchez, Henry Argüello, and Rodrigo Torres. Balance de flujos metabólicos en saccharomyces cerevisiae basado en compartimentalización intracelular. 15(2):18–18, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.15446/rev.colomb.biote.v15n2.41263">[DOI]</a>.</p></li>
<li id="id39244"><p>Jurijs Meitalovs and Egils Stalidzāns. Connectivity analysis of metabolites in synthetic metabolic pathways. ():, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.13140/2.1.1592.7364">[DOI]</a>.</p></li>
<li id="id39258"><p>John A. Cole, Michael J. Hallock, Piyush Labhsetwar, J.R. Peterson, John E. Stone, and Zaida Luthey‐Schulten. Stochastic simulations of cellular processes: from single cells to colonies. ():277–293, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-405926-9.00013-7">[DOI]</a>.</p></li>
<li id="id39283"><p>Dimitrije Jevremović. Scalable computation and analysis of elementary flux modes in metabolic networks. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39303"><p>Bernard Ycart, Frédéric Pont, and Jean‐Jacques Fournié. Simulation of gene regulatory networks. ():, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1310.0730">[DOI]</a>.</p></li>
<li id="id39313"><p>铁 申. Fba and fva analysis of central carbon metabolism in e. coli deletion mutants. 03(03):15–19, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.12677/hjcb.2013.33003">[DOI]</a>.</p></li>
<li id="id39319"><p>Alberto Noronha. Development of a computational platform for the visualization of metabolic models. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39320"><p>Anja Karlstädt. A systems biology approach to model cardiomyocyte metabolism. ():, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.17169/refubium-6170">[DOI]</a>.</p></li>
<li id="id39324"><p>Longfei Mao. Flux balance analysis to model microbial metabolism for electricity generation. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39328"><p>Beverly Sackler. A cross-species study of the evolutionary conservation of essential genes in bacteria. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39329"><p>Yuekai Sun, Ronan M. T. Fleming, Ines Thiele, and Michael A. Saunders. On lifting poorly-scaled ux balance analysis problems. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39330"><p>Fredric Bauer, Kris A. Berglund, Christian Hulteberg, Joakim Lundgren, Sennai Mesfun, Robert Nilsson, Ulrika Rova, and Sune Wännström. Comparative system analysis of carbon preserving fermentations for biofuels production. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39340"><p>Tie Shen, Jiejun Jin, and Bin Rui. Fba and fva analysis of central carbon metabolism in e. coli deletion mutants. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39346"><p>Matteo Mori. Unraveling bacterial growth laws. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39347"><p>Matthew A. Richards. Improving methanosarcina genome-scale models for strain design by incorporating cofactor specificity and free energy constraints. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39361"><p>Eugene Ma. An in silico investigation of the metabolic capabilities of anaeromyxobacter dehalogenans and field-scale applications. ():, 2013. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id39403"><p>Caroline M. Paget. Environmental systems biology of temperature adaptation in yeast. 8(16):999–999, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.21037/atm-20-5205">[DOI]</a>.</p></li>
<li id="id39417"><p>Stephan Pabinger and Zlatko Trajanoski. Memosys: platform for genome-scale metabolic models. ():1–7, 2013. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4614-6418-1_702-5">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-13" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-13" name="0-13" role="tabpanel" tabindex="0"><div class="docutils container" id="id40574">
<ol class="arabic simple" start="1">
<li id="id40599"><p>Kathleen A. Curran, John M. Leavitt, Ashty S. Karim, and Hal S. Alper. Metabolic engineering of muconic acid production in saccharomyces cerevisiae. 15():55–66, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2012.10.003">[DOI]</a>.</p></li>
<li id="id40601"><p>Yuliang Wang, James A. Eddy, and Nathan D. Price. Reconstruction of genome-scale metabolic models for 126 human tissues using mcadre. 6(1):153–153, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-153">[DOI]</a>.</p></li>
<li id="id40614"><p>Marnix H. Medema, Renske van Raaphorst, Eriko Takano, and Rainer Breitling. Computational tools for the synthetic design of biochemical pathways. 10(3):191–202, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nrmicro2717">[DOI]</a>.</p></li>
<li id="id40621"><p>Chuanli Wang, Longyun Guo, Yixue Li, and Zhuo Wang. Systematic comparison of c3 and c4 plants based on metabolic network analysis. 6(Suppl 2):S9–S9, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-s2-s9">[DOI]</a>.</p></li>
<li id="id40630"><p>Milan JA van Hoek and Roeland M. H. Merks. Redox balance is key to explaining full vs. partial switching to low-yield metabolism. 6(1):22–22, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-22">[DOI]</a>.</p></li>
<li id="id40640"><p>Livnat Jerby, Lior Wolf, Carsten Denkert, Gideon Y. Stein, Mika Hilvo, Matej Orešič, Tamar Geiger, and Eytan Ruppin. Metabolic associations of reduced proliferation and oxidative stress in advanced breast cancer. 72(22):5712–5720, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1158/0008-5472.can-12-2215">[DOI]</a>.</p></li>
<li id="id40645"><p>Nicolás Loira, Thierry Dulermo, Jean‐Marc Nicaud, and David James Sherman. A genome-scale metabolic model of the lipid-accumulating yeast yarrowia lipolytica. 6(1):35–35, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-35">[DOI]</a>.</p></li>
<li id="id40646"><p>Ali R. Zomorrodi, Patrick F. Suthers, Sridhar Ranganathan, and Costas D. Maranas. Mathematical optimization applications in metabolic networks. 14(6):672–686, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2012.09.005">[DOI]</a>.</p></li>
<li id="id40661"><p>Arvind K. Chavali, Kevin M. D’Auria, Erik L. Hewlett, Richard D. Pearson, and Jason A. Papin. A metabolic network approach for the identification and prioritization of antimicrobial drug targets. 20(3):113–123, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tim.2011.12.004">[DOI]</a>.</p></li>
<li id="id40673"><p>Meiyappan Lakshmanan, Gerald Choon‐Huat Koh, Bevan Kai-Sheng Chung, and Dong‐Yup Lee. Software applications for flux balance analysis. 15(1):108–122, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbs069">[DOI]</a>.</p></li>
<li id="id40677"><p>Mario Latendresse, Markus Krummenacker, Miles Trupp, and Peter D. Karp. Construction and completion of flux balance models from pathway databases. 28(3):388–396, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btr681">[DOI]</a>.</p></li>
<li id="id40681"><p>Pengcheng Pan and Qiang Hua. Reconstruction and in silico analysis of metabolic network for an oleaginous yeast, yarrowia lipolytica. 7(12):e51535–e51535, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0051535">[DOI]</a>.</p></li>
<li id="id40721"><p>Sandy MacDonald, George Lin, Calum W. Russell, Gavin H. Thomas, and Angela E. Douglas. The central role of the host cell in symbiotic nitrogen metabolism. 279(1740):2965–2973, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rspb.2012.0414">[DOI]</a>.</p></li>
<li id="id40727"><p>Michael J. McAnulty, Jiun Y. Yen, Benjamin G. Freedman, and Ryan S. Senger. Genome-scale modeling using flux ratio constraints to enable metabolic engineering of clostridial metabolism in silico. 6(1):42–42, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-42">[DOI]</a>.</p></li>
<li id="id40729"><p>Anja Karlstädt, Daniela Fliegner, Georgios Kararigas, Hugo Sanchez‐Ruderisch, Vera Regitz‐Zagrosek, and Hermann‐Georg Holzhütter. Cardionet: a human metabolic network suited for the study of cardiomyocyte metabolism. 6(1):114–114, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-114">[DOI]</a>.</p></li>
<li id="id40748"><p>Eva Collakova, Jiun Y. Yen, and Ryan S. Senger. Are we ready for genome-scale modeling in plants? 191-192():53–70, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.plantsci.2012.04.010">[DOI]</a>.</p></li>
<li id="id40750"><p>Balaji Balagurunathan, Sudhakar Jonnalagadda, Lily Tan, and Rajagopalan Srinivasan. Reconstruction and analysis of a genome-scale metabolic model for scheffersomyces stipitis. 11(1):27–27, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-11-27">[DOI]</a>.</p></li>
<li id="id40752"><p>B. Mikael Bergdahl, Dominik Heer, Uwe Sauer, Bärbel Hahn‐Hägerdal, and Ed Wj van Niel. Dynamic metabolomics differentiates between carbon and energy starvation in recombinant saccharomyces cerevisiae fermenting xylose. 5(1):34–34, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1754-6834-5-34">[DOI]</a>.</p></li>
<li id="id40776"><p>Xueyang Feng, You Xu, Yixin Chen, and Yinjie Tang. Microbesflux: a web platform for drafting metabolic models from the kegg database. 6(1):94–94, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-94">[DOI]</a>.</p></li>
<li id="id40783"><p>Gunnar Sigurðsson, Ronan M. T. Fleming, Almut Heinken, and Ines Thiele. A systems biology approach to drug targets in pseudomonas aeruginosa biofilm. 7(4):e34337–e34337, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0034337">[DOI]</a>.</p></li>
<li id="id40785"><p>Thomas Dandekar, Astrid Fieselmann, Saman Majeed, and Zeeshan Ahmed. Software applications toward quantitative metabolic flux analysis and modeling. 15(1):91–107, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbs065">[DOI]</a>.</p></li>
<li id="id40786"><p>Ralf Steuer, Henning Knoop, and Rainer Machné. Modelling cyanobacteria: from metabolism to integrative models of phototrophic growth. 63(6):2259–2274, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/jxb/ers018">[DOI]</a>.</p></li>
<li id="id40787"><p>Andreas Wagner. Metabolic networks and their evolution. 751():29–52, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4614-3567-9_2">[DOI]</a>.</p></li>
<li id="id40789"><p>Khuram Shahzad and Juan J. Loor. Application of top-down and bottom-up systems approaches in ruminant physiology and metabolism. 13(5):379–394, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.2174/138920212801619269">[DOI]</a>.</p></li>
<li id="id40795"><p>Carmen María González-Domenech, Eugeni Belda, Rafael Patiño‐Navarrete, Andrés Moyá, Juli Peretό, and Amparo Latorre. Metabolic stasis in an ancient symbiosis: genome-scale metabolic networks from two blattabacterium cuenoti strains, primary endosymbionts of cockroaches. 12(S1):S5–S5, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2180-12-s1-s5">[DOI]</a>.</p></li>
<li id="id40809"><p>Judith M. Fontana, Elizabeth Alexander, and Mirella Salvatore. Translational research in infectious disease: current paradigms and challenges ahead. 159(6):430–453, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.trsl.2011.12.009">[DOI]</a>.</p></li>
<li id="id40818"><p>Andreas Hoppe. What mrna abundances can tell us about metabolism. 2(3):614–631, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo2030614">[DOI]</a>.</p></li>
<li id="id40820"><p>Xuewen Chen and Yair Shachar‐Hill. Insights into metabolic efficiency from flux analysis. 63(6):2343–2351, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/jxb/ers057">[DOI]</a>.</p></li>
<li id="id40821"><p>Yu-Chieh Liao, Ming-Hsin Tsai, Feng‐Chi Chen, and Chao A. Hsiung. Gemsirv: a software platform for genome-scale metabolic model simulation, reconstruction and visualization. 28(13):1752–1758, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/bts267">[DOI]</a>.</p></li>
<li id="id40831"><p>Pieter-Jan D’Huys, Ivan Lule, Dominique Vercammen, Jozef Anné, Jan Van Impe, and Kristel Bernaerts. Genome-scale metabolic flux analysis of streptomyces lividans growing on a complex medium. 161(1):1–13, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2012.04.010">[DOI]</a>.</p></li>
<li id="id40852"><p>Paula Jouhten, Marilyn G. Wiebe, and Merja Penttilä. Dynamic flux balance analysis of the metabolism of &lt;i&gt;saccharomyces cerevisiae&lt;/i&gt; during the shift from fully respirative or respirofermentative metabolic states to anaerobiosis. 279(18):3338–3354, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1742-4658.2012.08649.x">[DOI]</a>.</p></li>
<li id="id40853"><p>Ana Rita Brochado, Sergej Andrejev, Costas D. Maranas, and Kiran Raosaheb Patil. Impact of stoichiometry representation on simulation of genotype-phenotype relationships in metabolic networks. 8(11):e1002758–e1002758, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1002758">[DOI]</a>.</p></li>
<li id="id40865"><p>Eugeni Belda, Francisco J. Silva, Juli Peretό, and Andrés Moyá. Metabolic networks of sodalis glossinidius: a systems biology approach to reductive evolution. 7(1):e30652–e30652, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0030652">[DOI]</a>.</p></li>
<li id="id40869"><p>Radhakrishnan Mahadevan and Michael A. Henson. Genome-based modeling and design of metabolic interactions in microbial communities. 3(4):e201210008–e201210008, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.5936/csbj.201210008">[DOI]</a>.</p></li>
<li id="id40870"><p>Erin R. Shellman, Charles Burant, and Santiago Schnell. Network motifs provide signatures that characterize metabolism. 9(3):352–352, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c2mb25346a">[DOI]</a>.</p></li>
<li id="id40871"><p>Aaron Brandes, Desmond S. Lun, Kuhn Ip, Jeremy Zucker, Caroline Colijn, Brian Weiner, and James E. Galagan. Inferring carbon sources from gene expression profiles using metabolic flux models. 7(5):e36947–e36947, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0036947">[DOI]</a>.</p></li>
<li id="id40910"><p>Martins Mednis and Maike K. Aurich. Application of string similarity ratio and edit distance in automatic metabolite reconciliation comparing reconstructions and models. 1(1):14–18, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.11592/bit.121102">[DOI]</a>.</p></li>
<li id="id40925"><p>Amornpan Klanchui, Chiraphan Khannapho, Atchara Phodee, Supapon Cheevadhanarak, and Asawin Meechai. I ak692: a genome-scale metabolic model of spirulina platensis c1. 6(1):71–71, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-71">[DOI]</a>.</p></li>
<li id="id40932"><p>Piotr Zakrzewski, Marnix H. Medema, Albert Gevorgyan, Andrzej Kierzek, Rainer Breitling, and Eriko Takano. Multimeteval: comparative and multi-objective analysis of genome-scale metabolic models. 7(12):e51511–e51511, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0051511">[DOI]</a>.</p></li>
<li id="id40933"><p>Hansheng Zhao, Mao Li, Kechi Fang, Wenfeng Chen, and Jing Wang. In silico insights into the symbiotic nitrogen fixation in sinorhizobium meliloti via metabolic reconstruction. 7(2):e31287–e31287, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0031287">[DOI]</a>.</p></li>
<li id="id40936"><p>Rachael Hageman Blair, Daniel J. Kliebenstein, and Gary A. Churchill. What can causal networks tell us about metabolic pathways? 8(4):e1002458–e1002458, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1002458">[DOI]</a>.</p></li>
<li id="id40951"><p>Lin Xu, Brandon Barker, and Zhenglong Gu. Dynamic epistasis for different alleles of the same gene. 109(26):10420–10425, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1121507109">[DOI]</a>.</p></li>
<li id="id40959"><p>Somedutta Barat, Benjamin Steeb, Alain Mazé, and Dirk Bumann. Extensive in vivo resilience of persistent salmonella. 7(7):e42007–e42007, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0042007">[DOI]</a>.</p></li>
<li id="id40983"><p>Tie Shen, Bin Rui, Hong Zhou, Ximing Zhang, Yin Yi, Wen Han, Haoran Zheng, Jihui Wu, and Yunyu Shi. Metabolic flux ratio analysis and multi-objective optimization revealed a globally conserved and coordinated metabolic response of e. coli to paraquat-induced oxidative stress. 9(1):121–132, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c2mb25285f">[DOI]</a>.</p></li>
<li id="id41030"><p>David Hala, Lene H. Petersen, Dalma Martinović‐Weigelt, and Duane B. Huggett. Constraints-based stoichiometric analysis of hypoxic stress on steroidogenesis in fathead minnows, &lt;i&gt;pimephales promelas&lt;/i&gt;. 215(10):1753–1765, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1242/jeb.066027">[DOI]</a>.</p></li>
<li id="id41037"><p>Matthias Bujara and Sven Panke. In silico assessment of cell‐free systems. 109(10):2620–2629, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.24534">[DOI]</a>.</p></li>
<li id="id41048"><p>Juan I. Castrillo, Pınar Pir, and Stephen G. Oliver. Yeast systems biology. ():343–365, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-385944-0.00018-6">[DOI]</a>.</p></li>
<li id="id41050"><p>Payam Setoodeh, A. Jahanmiri, and Reza Eslamloueyan. Hybrid neural modeling framework for simulation and optimization of diauxie-involved fed-batch fermentative succinate production. 81():57–76, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ces.2012.06.031">[DOI]</a>.</p></li>
<li id="id41051"><p>Michaël Bekaert. Reconstruction of danio rerio metabolic model accounting for subcellular compartmentalisation. 7(11):e49903–e49903, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0049903">[DOI]</a>.</p></li>
<li id="id41064"><p>Jianfeng Wang, Zhiqiang Xiong, Hailin Meng, Yiguang Wang, and Yong Wang. Synthetic biology triggers new era of antibiotics development. 64():95–114, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-007-5055-5_5">[DOI]</a>.</p></li>
<li id="id41071"><p>Aaron Brandes, Desmond S. Lun, Kuhn Ip, Jeremy Zucker, Caroline Colijn, Brian Weiner, and James E. Galagan. Correction: inferring carbon sources from gene expression profiles using metabolic flux models. 7(8):, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/annotation/139857f3-5a05-4a23-9bfe-a77aafbce54d">[DOI]</a>.</p></li>
<li id="id41072"><p>Francisco G Vital-Lopez, Vesna Memišević, and Bhaskar Dutta. Tutorial on biological networks. 2(4):298–325, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/widm.1061">[DOI]</a>.</p></li>
<li id="id41085"><p>Tugce Bilgin and Andreas Wagner. Design constraints on a synthetic metabolism. 7(6):e39903–e39903, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0039903">[DOI]</a>.</p></li>
<li id="id41105"><p>Martins Mednis, Zintis Rove, and Vytautas Galvanauskas. Moderator - a software tool for comparison of stoichiometric models. 6():97–100, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/saci.2012.6249983">[DOI]</a>.</p></li>
<li id="id41106"><p>Martins Mednis, Valters Brusbārdis, and Vytautas Galvanauskas. Comparison of genome-scale reconstructions using moderator. 2():79–84, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cinti.2012.6496737">[DOI]</a>.</p></li>
<li id="id41126"><p>Liram Vardi, Eytan Ruppin, and Roded Sharan. A linearized constraint-based approach for modeling signaling networks. 19(2):232–240, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2011.0277">[DOI]</a>.</p></li>
<li id="id41138"><p>Christine Klier. Use of an uncertainty analysis for genome-scale models as a prediction tool for microbial growth processes in subsurface environments. 46(5):2790–2798, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/es203461u">[DOI]</a>.</p></li>
<li id="id41149"><p>Liam G. Fearnley and Lars K. Nielsen. Pathlogic-s: a scalable boolean framework for modelling cellular signalling. 7(8):e41977–e41977, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0041977">[DOI]</a>.</p></li>
<li id="id41153"><p>Jing Ma, Xun Zhang, Choong Yong Ung, Yu Chen, and Baowen Li. Metabolic network analysis revealed distinct routes of deletion effects between essential and non-essential genes. 8(4):1179–1179, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c2mb05376d">[DOI]</a>.</p></li>
<li id="id41164"><p>Jurijs Meitalovs. Software tool for probabilistic metabolic pathways construction. ():405–408, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cinti.2012.6496800">[DOI]</a>.</p></li>
<li id="id41169"><p>Eva Grafahrend‐Belau, Björn H. Junker, and Falk Schreiber. Plant metabolic pathways: databases and pipeline for stoichiometric analysis. ():345–366, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-007-4749-4_17">[DOI]</a>.</p></li>
<li id="id41171"><p>Zheng Li, Alim Dewan, and M. Nazmul Karim. Optimization of bioethanol ethanol production in fed-batch fermentation. 45(15):816–821, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3182/20120710-4-sg-2026.00092">[DOI]</a>.</p></li>
<li id="id41172"><p>Sang Yup Lee, Seung Bum Sohn, Hyun Uk Kim, Jong Myoung Park, Tae Yong Kim, Jeffrey D. Orth, and Bernhard Ø. Palsson. Genome-scale network modeling. ():1–23, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-007-4534-6_1">[DOI]</a>.</p></li>
<li id="id41180"><p>Mariano Bizzarri, Simona Dinicola, and Cesare Manetti. Systems biology approach to metabolomics in cancer studies. ():3–37, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-007-4819-4_1">[DOI]</a>.</p></li>
<li id="id41207"><p>António Roldão, Il‐Kwon Kim, and Jens Nielsen. Bridging omics technologies with synthetic biology in yeast industrial biotechnology. ():271–327, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-94-007-4534-6_9">[DOI]</a>.</p></li>
<li id="id41209"><p>Joo Sang Lee, Takashi Nishikawa, and Adilson E. Motter. Why optimal states recruit fewer reactions in metabolic networks. 32(8):2937–2950, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3934/dcds.2012.32.2937">[DOI]</a>.</p></li>
<li id="id41218"><p>Eleftheria Tzamali, Vangelis Sakkalis, and Kostas Marias. The effects of near optimal growth solutions in genome-scale human cancer metabolic model. 324():626–631, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/bibe.2012.6399774">[DOI]</a>.</p></li>
<li id="id41220"><p>Carol Milena Barreto-Rodriguez, Jessica Paola Ramirez-Angulo, Jorge M. Gómez, Luke E. K. Achenie, and Andrés Fernando González Barrios. Optimization of the bioconversion of glycerol to ethanol using &amp;amp;lt;i&amp;amp;gt;escherichia coli&amp;amp;lt;/i&amp;amp;gt; by implementing a bi-level programming framework for proposing gene transcription control strategies based on genetic algorithms. 03(04):336–343, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.4236/abb.2012.34049">[DOI]</a>.</p></li>
<li id="id41229"><p>Andreas Wagner. Genotype networks and evolutionary innovations in biological systems. ():251–264, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-385944-0.00013-7">[DOI]</a>.</p></li>
<li id="id41236"><p>Hooman Hefzi, Bernhard Ø. Palsson, and Nathan E. Lewis. Reconstruction of genome-scale metabolic networks. ():229–250, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-385944-0.00012-5">[DOI]</a>.</p></li>
<li id="id41245"><p>Porntip Chiewchankaset, Yanee Srimarut, Amornpan Klanchui, Peter Kurdi, Vethachai Plengvidhya, and Asawin Meechai. Systematic identification of lactobacillus plantarum auxotrophs for fermented nham using genome-scale metabolic model. 162(2-3):327–335, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2012.08.019">[DOI]</a>.</p></li>
<li id="id41283"><p>Arnau Montagud. Modelling and analysis of biological systems to obtain biofuels. ():, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.4995/thesis/10251/17319">[DOI]</a>.</p></li>
<li id="id41290"><p>Dewu Ding and Ying Long. Revisiting modules (and centers) in &lt;i&gt;staphylococcus aureus&lt;/i&gt; metabolic network with link clustering. 20(01):57–66, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s021833901150032x">[DOI]</a>.</p></li>
<li id="id41298"><p>Ali Navid. Development of constraint-based system-level models of microbial metabolism. 881():531–549, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-61779-827-6_18">[DOI]</a>.</p></li>
<li id="id41299"><p>Chuanli Wang, Longyun Guo, and Zhuo Wang. Comparative analysis of c3 and c4 plants using constraint-based model. 60():418–425, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/pma.2012.6524867">[DOI]</a>.</p></li>
<li id="id41324"><p>Kourosh Salehi‐Ashtiani and Jason A. Papin. Experimental definition and validation of protein coding transcripts in chlamydomonas reinhardtii. ():, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.2172/1033125">[DOI]</a>.</p></li>
<li id="id41380"><p>Juan Nogales, Steinn Guðmundsson, Eric M. Knight, Bernhard Ø. Palsson, and Ines Thiele. Detailing the optimality of photosynthesis in cyanobacteria through systems biology analysis. 109(7):2678–2683, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1117907109">[DOI]</a>.</p></li>
<li id="id41381"><p>Hojung Nam, Nathan E. Lewis, Joshua A. Lerman, Dae‐Hee Lee, Roger L. Chang, Donghyuk Kim, and Bernhard Ø. Palsson. Network context and selection in the evolution to enzyme specificity. 337(6098):1101–1104, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1126/science.1216861">[DOI]</a>.</p></li>
<li id="id41404"><p>Alisdair R. Fernie and Mark Stitt. On the discordance of metabolomics with proteomics and transcriptomics: coping with increasing complexity in logic, chemistry, and network interactions scientific correspondence. 158(3):1139–1145, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.112.193235">[DOI]</a>.</p></li>
<li id="id41421"><p>ChiamYu Ng, Moo-Young Jung, Jinwon Lee, and Min‐Kyu Oh. Production of 2,3-butanediol in saccharomyces cerevisiae by in silico aided metabolic engineering. 11(1):68–68, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-11-68">[DOI]</a>.</p></li>
<li id="id41424"><p>Dave Lee, Kieran Smallbone, Warwick B. Dunn, Ettore Murabito, Catherine Winder, Douglas B. Kell, Pedro Mendes, and Neil Swainston. Improving metabolic flux predictions using absolute gene expression data. 6(1):73–73, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-73">[DOI]</a>.</p></li>
<li id="id41428"><p>Douglas B. Kell, Paul D. Dobson, Elizabeth Bilsland, and Stephen G. Oliver. The promiscuous binding of pharmaceutical drugs and their transporter-mediated uptake into cells: what we (need to) know and how we can do so. 18(5-6):218–239, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.drudis.2012.11.008">[DOI]</a>.</p></li>
<li id="id41431"><p>Ben Heavner, Kieran Smallbone, Brandon Barker, Pedro Mendes, and Larry P. Walker. Yeast 5 – an expanded reconstruction of the saccharomyces cerevisiae metabolic network. 6(1):55–55, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-55">[DOI]</a>.</p></li>
<li id="id41458"><p>Ines Thiele, Ronan M. T. Fleming, Richard Que, Aarash Bordbar, Dinh Diep, and Bernhard Ø. Palsson. Multiscale modeling of metabolism and macromolecular synthesis in e. coli and its application to the evolution of codon usage. 7(9):e45635–e45635, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0045635">[DOI]</a>.</p></li>
<li id="id41491"><p>Wilbert B. Copeland, Bryan Bartley, Deepak Chandran, Michal Galdzicki, Kyung H. Kim, Sean C. Sleight, Costas D. Maranas, and Herbert M. Sauro. Computational tools for metabolic engineering. 14(3):270–280, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2012.03.001">[DOI]</a>.</p></li>
<li id="id41497"><p>Elhanan Borenstein. Computational systems biology and in silico modeling of the human microbiome. 13(6):769–780, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbs022">[DOI]</a>.</p></li>
<li id="id41502"><p>Jennifer L. Reed. Shrinking the metabolic solution space using experimental datasets. 8(8):e1002662–e1002662, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1002662">[DOI]</a>.</p></li>
<li id="id41512"><p>Joonhoon Kim and Jennifer L. Reed. Relatch: relative optimality in metabolic networks explains robust metabolic and regulatory responses to perturbations. 13(9):R78–R78, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/gb-2012-13-9-r78">[DOI]</a>.</p></li>
<li id="id41539"><p>Cristiana Gomes de Oliveira Dal’Molin and Lars K. Nielsen. Plant genome-scale metabolic reconstruction and modelling. 24(2):271–277, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2012.08.007">[DOI]</a>.</p></li>
<li id="id41574"><p>Ines Thiele, Almut Heinken, and Ronan M. T. Fleming. A systems biology approach to studying the role of microbes in human health. 24(1):4–12, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2012.10.001">[DOI]</a>.</p></li>
<li id="id41581"><p>Joost Boele, Brett G. Olivier, and Bas Teusink. Fame, the flux analysis and modeling environment. 6(1):8–8, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-8">[DOI]</a>.</p></li>
<li id="id41587"><p>Swagatika Sahoo, Leifur Franzson, Jón J. Jónsson, and Ines Thiele. A compendium of inborn errors of metabolism mapped onto the human metabolic network. 8(10):2545–2545, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c2mb25075f">[DOI]</a>.</p></li>
<li id="id41589"><p>Ali Navid and Eivind Almaas. Genome-level transcription data of yersinia pestis analyzed with a new metabolic constraint-based approach. 6(1):150–150, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-150">[DOI]</a>.</p></li>
<li id="id41593"><p>Steven Kelk, Brett G. Olivier, Leen Stougie, and Frank J. Bruggeman. Optimal flux spaces of genome-scale stoichiometric models are determined by a few subnetworks. 2(1):580–580, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/srep00580">[DOI]</a>.</p></li>
<li id="id41598"><p>Abdelhalim Larhlimi, László Dávid, Joachim Selbig, and Alexander Bockmayr. F2c2: a fast tool for the computation of flux coupling in genome-scale metabolic networks. 13(1):57–57, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-13-57">[DOI]</a>.</p></li>
<li id="id41599"><p>Xin Fang, Anders Wallqvist, and Jaques Reifman. Modeling phenotypic metabolic adaptations of mycobacterium tuberculosis h37rv under hypoxia. 8(9):e1002688–e1002688, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1002688">[DOI]</a>.</p></li>
<li id="id41602"><p>Guoqiang Xu, Wei Zou, Xiulai Chen, Nan Xu, Li Liu, and Jian Chen. Fumaric acid production in saccharomyces cerevisiae by in silico aided metabolic engineering. 7(12):e52086–e52086, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0052086">[DOI]</a>.</p></li>
<li id="id41608"><p>Thomas Dandekar, Astrid Fieselmann, Jasmin Popp, and Michael Hensel. Salmonella enterica: a surprisingly well-adapted intracellular lifestyle. 3():164–164, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3389/fmicb.2012.00164">[DOI]</a>.</p></li>
<li id="id41613"><p>Jeffrey D. Orth and Bernhard Ø. Palsson. Gap-filling analysis of the ijo1366 escherichia coli metabolic network reconstruction for discovery of metabolic functions. 6(1):30–30, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-30">[DOI]</a>.</p></li>
<li id="id41616"><p>Chuan Xu, Lili Liu, Zhao Zhang, Danfeng Jin, Juanping Qiu, and Ming Chen. Genome-scale metabolic model in guiding metabolic engineering of microbial improvement. 97(2):519–539, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-012-4543-9">[DOI]</a>.</p></li>
<li id="id41628"><p>Hulda S. Haraldsdóttir, Ines Thiele, and Ronan M. T. Fleming. Quantitative assignment of reaction directionality in a multicompartmental human metabolic reconstruction. 102(8):1703–1711, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bpj.2012.02.032">[DOI]</a>.</p></li>
<li id="id41652"><p>Jari Kämäräinen, Henning Knoop, Natalie Stanford, Fernando Guerrero, M. Akhtar, Eva–Mari Aro, Ralf Steuer, and Patrik R. Jones. Physiological tolerance and stoichiometric potential of cyanobacteria for hydrocarbon fuel production. 162(1):67–74, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2012.07.193">[DOI]</a>.</p></li>
<li id="id41653"><p>Brian J. Schmidt, Jason A. Papin, and Cynthia J. Musante. Mechanistic systems modeling to guide drug discovery and development. 18(3-4):116–127, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.drudis.2012.09.003">[DOI]</a>.</p></li>
<li id="id41654"><p>Nan Xu, Li Liu, Wei Zou, Jie Liu, Qiang Hua, and Jian Chen. Reconstruction and analysis of the genome-scale metabolic network of candida glabrata. 9(2):205–216, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c2mb25311a">[DOI]</a>.</p></li>
<li id="id41687"><p>Keiko Yonekura‐Sakakibara, Atsushi Fukushima, and Kazuki Saito. Transcriptome data modeling for targeted plant metabolic engineering. 24(2):285–290, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2012.10.018">[DOI]</a>.</p></li>
<li id="id41705"><p>Andrejs Kostromins and Egils Stalidzāns. Paint4net: cobra toolbox extension for visualization of stoichiometric models of metabolism. 109(2):233–239, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2012.03.002">[DOI]</a>.</p></li>
<li id="id41722"><p>Wei Zou, Li Liu, Jing Zhang, Haoru Yang, M. Zhou, Qiang Hua, and Jian Chen. Reconstruction and analysis of a genome-scale metabolic model of the vitamin c producing industrial strain ketogulonicigenium vulgare wsh-001. 161(1):42–48, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2012.05.015">[DOI]</a>.</p></li>
<li id="id41734"><p>Nuno Carinhas, Rui Oliveira, Paula M. Alves, Manuel J.T. Carrondo, and Ana P. Teixeira. Systems biotechnology of animal cells: the road to prediction. 30(7):377–385, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2012.03.004">[DOI]</a>.</p></li>
<li id="id41752"><p>Jan Schellenberger, Daniel C. Zielinski, Wing Choi, Sunthosh Madireddi, Vasiliy A. Portnoy, David A. Scott, Jennifer L. Reed, Andrei L. Osterman, and Bernhard Ø. Palsson. Predicting outcomes of steady-state 13c isotope tracing experiments using monte carlo sampling. 6(1):9–9, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-6-9">[DOI]</a>.</p></li>
<li id="id41753"><p>Ting Liu, Wei Zou, Li Liu, and Jian Chen. A constraint-based model of scheffersomyces stipitis for improved ethanol production. 5(1):72–72, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1754-6834-5-72">[DOI]</a>.</p></li>
<li id="id41776"><p>Miguel Ángel Medina. Systems biology for molecular life sciences and its impact in biomedicine. 70(6):1035–1053, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00018-012-1109-z">[DOI]</a>.</p></li>
<li id="id41820"><p>Óttar Rolfsson, Giuseppe Paglia, Manuela Magnúsdóttir, Bernhard Ø. Palsson, and Ines Thiele. Inferring the metabolism of human orphan metabolites from their metabolic network context affirms human gluconokinase activity. 449(2):427–435, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1042/bj20120980">[DOI]</a>.</p></li>
<li id="id41903"><p>Robert Dromms and Mark P. Styczynski. Systematic applications of metabolomics in metabolic engineering. 2(4):1090–1122, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo2041090">[DOI]</a>.</p></li>
<li id="id41905"><p>Carlos Eduardo García Sánchez, Cesar A. Vargas-García, and Rodrigo Torres. Predictive potential of flux balance analysis of saccharomyces cerevisiae using as optimization function combinations of cell compartmental objectives. 7(8):e43006–e43006, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0043006">[DOI]</a>.</p></li>
<li id="id41922"><p>Nathan Crook and Hal S. Alper. Model-based design of synthetic, biological systems. 103():2–11, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ces.2012.12.022">[DOI]</a>.</p></li>
<li id="id41925"><p>Paula Jouhten. Metabolic modelling in the development of cell factories by synthetic biology. 3(4):e201210009–e201210009, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.5936/csbj.201210009">[DOI]</a>.</p></li>
<li id="id41980"><p>Santiago Comba, Ana Arabolaza, and Hugo Gramajo. Emerging engineering principles for yield improvement in microbial cell design. 3(4):e201210016–e201210016, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.5936/csbj.201210016">[DOI]</a>.</p></li>
<li id="id42155"><p>Zeeshan Ahmed, Saman Majeed, and Thomas Dandekar. Computational feature performance and domain specific architecture evaluation of software applications towards metabolic flux analysis. 5(3):165–176, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.2174/2213275911205030165">[DOI]</a>.</p></li>
<li id="id42177"><p>Andrejs Kostromins. Altfluxes: cobra toolbox extension for flux variability analysis of stoichiometric models of metabolism. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42190"><p>Ming‐Tsang Wu and Christina Chan. Human metabolic network: reconstruction, simulation, and applications in systems biology. 2(1):242–253, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.3390/metabo2010242">[DOI]</a>.</p></li>
<li id="id42219"><p>Maike K. Aurich and Ines Thiele. Contextualization procedure and modeling of monocyte specific tlr signaling. 7(12):e49978–e49978, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0049978">[DOI]</a>.</p></li>
<li id="id42327"><p>Emily Gruber, Amy Ko, Michael MacGillvray, and Miranda Sawyer. Robust analysis of metabolic pathways. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42328"><p>Iyarest Tawornsamretkit, Rattana Thanasomboon, Jittrawan Thaiprasit, Dujduan Waraho‐Zhmayev, Supapon Cheevadhanarak, and Asawin Meechai. Analysis of metabolic network of synthetic escherichia coli producing linalool using constraint-based modeling. 11():24–35, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.procs.2012.09.004">[DOI]</a>.</p></li>
<li id="id42332"><p>Ben Heavner, Susan A. Henry, and Larry P. Walker. Evaluating sphingolipid biochemistry in the consensus reconstruction of yeast metabolism. 8(2):72–78, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2012.0002">[DOI]</a>.</p></li>
<li id="id42357"><p>Xinyu Zhang. Enhanced biohydrogen production and substrate utilization by co-culture fermentation with reduced extracellular electron shuttles. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42359"><p>Maikel P. H. Verouden. Fusing prior knowledge with microbial metabolomics. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42367"><p>Ana Rita Brochado, Uffe Hasbro Mortensen, and Kiran Raosaheb Patil. Investigating genotype-phenotype relationships in saccharomyces cerevisiae metabolic network through stoichiometric modeling. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42381"><p>Agris Pentjuss, Ilona Odzina, and A. Gailums. Multi agent genome-scale metabolic reconstruction modeling software schema. 29(1):189–194, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/ejp.4767">[DOI]</a>.</p></li>
<li id="id42397"><p>Peter Yan Li. In silico metabolic network reconstruction of scheffersomyces stipitis. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42422"><p>君 王. Reconstruction of erwinia carotovora subsp. atroseptica scri1043 metabolic network and its application in screening potential targets. 02(01):1–9, 2012. <a class="reference external" href="https://doi.org/https://doi.org/10.12677/hjcb.2012.21001">[DOI]</a>.</p></li>
<li id="id42438"><p>Artículo De Investigación, César Augusto, Vargas García, Henry Argüello, Rodrigo Fernández Gonzalo, and Torres Sáez. Prediction of genome scale of saccharomyces cerevisiae by flux balance analysis. 14(1):93–107, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42469"><p>Louise Mølgaard. Engineering the polyketide cell factory. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id42473"><p>Olusegun Oshota. A systems biology approach to the production of biotechnological products through systematic in silico studies. ():, 2012. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-14" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-14" name="0-14" role="tabpanel" tabindex="0"><div class="docutils container" id="id43695">
<ol class="arabic simple" start="1">
<li id="id43698"><p>Jan Schellenberger, Richard Que, Ronan M. T. Fleming, Ines Thiele, Jeffrey D. Orth, Adam M. Feist, Daniel C. Zielinski, Aarash Bordbar, Nathan E. Lewis, Sorena Rahmanian, Joseph Kang, Daniel R. Hyduke, and Bernhard Ø. Palsson. Quantitative prediction of cellular metabolism with constraint-based models: the cobra toolbox v2.0. 6(9):1290–1307, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nprot.2011.308">[DOI]</a>.</p></li>
<li id="id43702"><p>Jeffrey D. Orth, Tom M Conrad, Jessica Na, Joshua A. Lerman, Hojung Nam, Adam M. Feist, and Bernhard Ø. Palsson. A comprehensive genome‐scale reconstruction of &lt;i&gt;escherichia coli&lt;/i&gt; metabolism—2011. 7(1):535–535, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2011.65">[DOI]</a>.</p></li>
<li id="id43721"><p>Roger L. Chang, Lila Ghamsari, Ani Manichaikul, Erik Hom, Balaji Santhanam, Weiqi Fu, Yun Shen, Tong Hao, Bernhard Ø. Palsson, Kourosh Salehi‐Ashtiani, and Jason A. Papin. Metabolic network reconstruction of &lt;i&gt;chlamydomonas&lt;/i&gt; offers insight into light‐driven algal metabolism. 7(1):518–518, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2011.52">[DOI]</a>.</p></li>
<li id="id43723"><p>Tom M Conrad, Nathan E. Lewis, and Bernhard Ø. Palsson. Microbial laboratory evolution in the era of genome‐scale science. 7(1):509–509, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2011.42">[DOI]</a>.</p></li>
<li id="id43736"><p>Zhi Wang and Jianzhi Zhang. Impact of gene expression noise on organismal fitness and the efficacy of natural selection. 108(16):E67–76, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1100059108">[DOI]</a>.</p></li>
<li id="id43738"><p>Colin Archer, Jihyun F. Kim, Haeyoung Jeong, Jin Hwan Park, Claudia E. Vickers, Sang Yup Lee, and Lars K. Nielsen. The genome sequence of e. coli w (atcc 9637): comparative genome analysis and an improved genome-scale reconstruction of e. coli. 12(1):9–9, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2164-12-9">[DOI]</a>.</p></li>
<li id="id43744"><p>Aarash Bordbar, Adam M. Feist, Renata Usaite-Black, Joseph Woodcock, Bernhard Ø. Palsson, and Iman Famili. A multi-tissue type genome-scale metabolic network for analysis of whole-body systems physiology. 5(1):180–180, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-180">[DOI]</a>.</p></li>
<li id="id43750"><p>Cristiana Gomes de Oliveira Dal’Molin, Lake‐Ee Quek, Robin Palfreyman, and Lars K. Nielsen. Algagem – a genome-scale metabolic reconstruction of algae based on the chlamydomonas reinhardtii genome. 12(Suppl 4):S5–S5, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2164-12-s4-s5">[DOI]</a>.</p></li>
<li id="id43765"><p>Matthew Oberhardt, Jacek Puchałka, Vítor A. P. Martins dos Santos, and Jason A. Papin. Reconciliation of genome-scale metabolic reconstructions for comparative systems analysis. 7(3):e1001116–e1001116, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1001116">[DOI]</a>.</p></li>
<li id="id43769"><p>Yu-Chieh Liao, Tzu‐Wen Huang, Feng-Chi Chen, Pep Charusanti, Jay Sung-Joong Hong, Hwan‐You Chang, Shih-Feng Tsai, Bernhard Ø. Palsson, and Chao A. Hsiung. An experimentally validated genome-scale metabolic reconstruction of &lt;i&gt;klebsiella pneumoniae&lt;/i&gt; mgh 78578, &lt;i&gt;i&lt;/i&gt; yl1228. 193(7):1710–1717, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.01218-10">[DOI]</a>.</p></li>
<li id="id43773"><p>R. Lencastre Fernandes, Marta Nierychlo, Luisa Lundin, A.E. Pedersen, P.E. Puentes Tellez, Abhishek Dutta, Magnus Carlquist, Andrijana Bolic, Daniel Schäpper, A.C. Brunetti, Søren Helmark, A.-L. Heins, Anker Degn Jensen, Ingmar Nopens, Karsten Rottwitt, Nicolas Szita, J.D. van Elsas, Per Halkjær Nielsen, Jan Martinussen, Søren J. Sørensen, Anna Eliasson Lantz, and Krist V. Gernaey. Experimental methods and modeling techniques for description of cell population heterogeneity. 29(6):575–599, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biotechadv.2011.03.007">[DOI]</a>.</p></li>
<li id="id43774"><p>Paul A. Jensen, Kyla Lutz, and Jason A. Papin. Tiger: toolbox for integrating genome-scale metabolic models, expression data, and transcriptional regulatory networks. 5(1):147–147, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-147">[DOI]</a>.</p></li>
<li id="id43778"><p>Aarash Bordbar and Bernhard Ø. Palsson. Using the reconstructed genome‐scale human metabolic network to study physiology and pathology. 271(2):131–141, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1365-2796.2011.02494.x">[DOI]</a>.</p></li>
<li id="id43783"><p>Caroline B. Milne, James A. Eddy, Ravali Raju, Soroush Ardekani, Pan‐Jun Kim, Ryan S. Senger, Yong‐Su Jin, Hans P. Blaschek, and Nathan D. Price. Metabolic network reconstruction and genome-scale model of butanol-producing strain clostridium beijerinckii ncimb 8052. 5(1):130–130, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-130">[DOI]</a>.</p></li>
<li id="id43784"><p>Suvi Santala, Elena Efimova, Virpi Kivinen, Antti Larjo, Tommi Aho, Matti Karp, and Ville Santala. Improved triacylglycerol production in acinetobacter baylyi adp1 by metabolic engineering. 10(1):36–36, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-10-36">[DOI]</a>.</p></li>
<li id="id43788"><p>Pablo Carbonell, Anne‐Gaëlle Planson, Davide Fichera, and Jean‐Loup Faulon. A retrosynthetic biology approach to metabolic pathway design for therapeutic production. 5(1):122–122, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-122">[DOI]</a>.</p></li>
<li id="id43789"><p>Il‐Kwon Kim, António Roldão, Verena Siewers, and Jens Nielsen. A systems-level approach for metabolic engineering of yeast cell factories. 12(2):228–248, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1567-1364.2011.00779.x">[DOI]</a>.</p></li>
<li id="id43804"><p>Neil Swainston, Kieran Smallbone, Pedro Mendes, Douglas B. Kell, and Norman W. Paton. The subliminal toolbox: automating steps in the reconstruction of metabolic networks. 8(2):187–203, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1515/jib-2011-186">[DOI]</a>.</p></li>
<li id="id43807"><p>Jordan O. Hay and Jörg Schwender. Computational analysis of storage synthesis in developing &lt;i&gt;brassica napus&lt;/i&gt; l. (oilseed rape) embryos: flux variability analysis in relation to &lt;sup&gt;13&lt;/sup&gt;c metabolic flux analysis. 67(3):513–525, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1365-313x.2011.04611.x">[DOI]</a>.</p></li>
<li id="id43815"><p>Jordan O. Hay and Jörg Schwender. Metabolic network reconstruction and flux variability analysis of storage synthesis in developing oilseed rape (&lt;i&gt;brassica napus&lt;/i&gt; l.) embryos. 67(3):526–541, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1365-313x.2011.04613.x">[DOI]</a>.</p></li>
<li id="id43816"><p>Andreas Hoppe, Sabrina Hoffmann, Andreas Gerasch, Christoph Gille, and Hermann‐Georg Holzhütter. Fasimu: flexible software for flux-balance computation series in large metabolic networks. 12(1):28–28, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-12-28">[DOI]</a>.</p></li>
<li id="id43818"><p>Steffen Klamt and Axel von Kamp. An application programming interface for cellnetanalyzer. 105(2):162–168, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2011.02.002">[DOI]</a>.</p></li>
<li id="id43826"><p>Siu Hung Joshua Chan and Ping Ji. Decomposing flux distributions into elementary flux modes in genome-scale metabolic networks. 27(16):2256–2262, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btr367">[DOI]</a>.</p></li>
<li id="id43836"><p>Aarash Bordbar, Neema Jamshidi, and Bernhard Ø. Palsson. Iab-rbc-283: a proteomically derived knowledge-base of erythrocyte metabolism that can be used to simulate its physiological and patho-physiological states. 5(1):110–110, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-110">[DOI]</a>.</p></li>
<li id="id43838"><p>Wolfram Weckwerth. Unpredictability of metabolism—the key role of metabolomics science in combination with next-generation genome sequencing. 400(7):1967–1978, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00216-011-4948-9">[DOI]</a>.</p></li>
<li id="id43844"><p>Matthew N. Benedict, Matthew C. Gonnerman, William W. Metcalf, and Nathan D. Price. Genome-scale metabolic reconstruction and hypothesis testing in the methanogenic archaeon methanosarcina acetivorans c2a. 194(4):855–865, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.06040-11">[DOI]</a>.</p></li>
<li id="id43858"><p>Óttar Rolfsson, Bernhard Ø. Palsson, and Ines Thiele. The human metabolic reconstruction recon 1 directs hypotheses of novel human metabolic functions. 5(1):155–155, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-155">[DOI]</a>.</p></li>
<li id="id43861"><p>Sandy MacDonald, Gavin H. Thomas, and Angela E. Douglas. Genetic and metabolic determinants of nutritional phenotype in an insect-bacterial symbiosis. 20(10):2073–2084, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1365-294x.2011.05031.x">[DOI]</a>.</p></li>
<li id="id43864"><p>Anthony C. Smith and Alan J. Robinson. A metabolic model of the mitochondrion and its use in modelling diseases of the tricarboxylic acid cycle. 5(1):102–102, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-102">[DOI]</a>.</p></li>
<li id="id43868"><p>David J. Baumler, Roman G Peplinski, Jennifer L. Reed, Jeremy D. Glasner, and Nicole T. Perna. The evolution of metabolic networks of e. coli. 5(1):182–182, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-182">[DOI]</a>.</p></li>
<li id="id43872"><p>Amit Ghosh, Huimin Zhao, and Nathan D. Price. Genome-scale consequences of cofactor balancing in engineered pentose utilization pathways in saccharomyces cerevisiae. 6(11):e27316–e27316, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0027316">[DOI]</a>.</p></li>
<li id="id43876"><p>Eleftherios Pilalis, Aristotelis Chatziioannou, Brigitte Thomasset, and Fragiskos N. Kolisis. An in silico compartmentalized metabolic model of &lt;i&gt;brassica napus&lt;/i&gt; enables the systemic study of regulatory aspects of plant central metabolism. 108(7):1673–1682, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bit.23107">[DOI]</a>.</p></li>
<li id="id43878"><p>Stefan Gretar Thorleifsson and Ines Thiele. Rbionet: a cobra toolbox extension for &lt;i&gt;r&lt;/i&gt;econstructing high-quality &lt;i&gt;bio&lt;/i&gt;chemical &lt;i&gt;net&lt;/i&gt;works. 27(14):2009–2010, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btr308">[DOI]</a>.</p></li>
<li id="id43884"><p>Gino Baart and Dirk E. Martens. Genome-scale metabolic models: reconstruction and analysis. 799():107–126, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-61779-346-2_7">[DOI]</a>.</p></li>
<li id="id43888"><p>Brett A. Boghigian, John Armando, Daniel Salas, and Blaine A. Pfeifer. Computational identification of gene over-expression targets for metabolic engineering of taxadiene production. 93(5):2063–2073, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00253-011-3725-1">[DOI]</a>.</p></li>
<li id="id43893"><p>Guillaume Cogne, Marco Rügen, Alexander Bockmayr, Mariana Titica, Claude‐Gilles Dussap, Jean‐François Cornet, and Jack Legrand. A model‐based method for investigating bioenergetic processes in autotrophically growing eukaryotic microalgae: application to the green algae &lt;i&gt;chlamydomonas reinhardtii&lt;/i&gt;. 27(3):631–640, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.596">[DOI]</a>.</p></li>
<li id="id43901"><p>Aline Métris, Mark Reuter, Duncan J. H. Gaskin, József Baranyi, and Arnoud H. M. van Vliet. In vivo and in silico determination of essential genes of campylobacter jejuni. 12(1):535–535, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2164-12-535">[DOI]</a>.</p></li>
<li id="id43903"><p>Aditya Vikram Pandit and Radhakrishnan Mahadevan. In silico characterization of microbial electrosynthesis for metabolic engineering of biochemicals. 10(1):76–76, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1475-2859-10-76">[DOI]</a>.</p></li>
<li id="id43913"><p>Eleftheria Tzamali, Panayiota Poirazi, Ioannis G. Tollis, and Martin Reczko. A computational exploration of bacterial metabolic diversity identifying metabolic interactions and growth-efficient strain communities. 5(1):167–167, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-167">[DOI]</a>.</p></li>
<li id="id43919"><p>Osbaldo Resendis‐Antonio, Magdalena Hernández, Emmanuel Salazar, Sandra Contreras, G. Martínez Batallar, Yolanda Mora, and Sergio Encarnación‐Guevara. Systems biology of bacterial nitrogen fixation: high-throughput technology and its integrative description with constraint-based modeling. 5(1):120–120, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-120">[DOI]</a>.</p></li>
<li id="id43929"><p>Pep Charusanti, Sadhana Chauhan, Kathleen McAteer, Joshua A. Lerman, Daniel R. Hyduke, Vladimir L. Motin, Charles Ansong, Joshua Adkins, and Bernhard Ø. Palsson. An experimentally-supported genome-scale metabolic network reconstruction for yersinia pestis co92. 5(1):163–163, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-163">[DOI]</a>.</p></li>
<li id="id43937"><p>Charles R. Haggart, Jennifer Bartell, Jeffrey J. Saucerman, and Jason A. Papin. Whole-genome metabolic network reconstruction and constraint-based modeling⋆. 500():411–433, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-385118-5.00021-9">[DOI]</a>.</p></li>
<li id="id43945"><p>Stephan Pabinger, Robert Rader, Rasmus Ågren, Jens Nielsen, and Zlatko Trajanoski. Memosys: bioinformatics platform for genome-scale metabolic models. 5(1):20–20, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-20">[DOI]</a>.</p></li>
<li id="id43951"><p>Mona Yousofshahi, Kyongbum Lee, and Soha Hassoun. Probabilistic pathway construction. 13(4):435–444, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2011.01.006">[DOI]</a>.</p></li>
<li id="id43955"><p>Kathleen A. Curran, Nathan Crook, and Hal S. Alper. Using flux balance analysis to guide microbial metabolic engineering. 834():197–216, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-61779-483-4_13">[DOI]</a>.</p></li>
<li id="id43958"><p>Daniel R. Hyduke, Daniel R. Hyduke, Jan Schellenberger, Richard Que, Ronan M. T. Fleming, Ines Thiele, Jeffery Orth, Adam M. Feist, Daniel C. Zielinski, Aarash Bordbar, Nathan E. Lewis, Sorena Rahmanian, Joseph Kang, and Bernhard Ø. Palsson. Cobra toolbox 2.0. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/protex.2011.234">[DOI]</a>.</p></li>
<li id="id43962"><p>Hailin Meng, Yong Wang, Qiang Hua, Siliang Zhang, and Xiao‐Ning Wang. In silico analysis and experimental improvement of taxadiene heterologous biosynthesis in escherichia coli. 16(2):205–215, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-010-0329-z">[DOI]</a>.</p></li>
<li id="id43963"><p>Oksana Sorokina, Florence Corellou, David Dauvillée, Anatoly Sorokin, Igor Goryanin, Steven Ball, François‐Yves Bouget, and Andrew J. Millar. Microarray data can predict diurnal changes of starch content in the picoalga ostreococcus. 5(1):36–36, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-36">[DOI]</a>.</p></li>
<li id="id43965"><p>J. Bernadette Moore and Mark C. Willingham. Proteomics and systems biology: current and future applications in the nutritional sciences. 2(4):355–364, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.3945/an.111.000554">[DOI]</a>.</p></li>
<li id="id43968"><p>Karoline Faust, Didier Croes, and Jacques van Helden. Prediction of metabolic pathways from genome-scale metabolic networks. 105(2):109–121, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2011.05.004">[DOI]</a>.</p></li>
<li id="id43975"><p>Mohammad Tauqeer Alam, Marnix H. Medema, Eriko Takano, and Rainer Breitling. Comparative genome-scale metabolic modeling of actinomycetes: the topology of essential core metabolism. 585(14):2389–2394, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.febslet.2011.06.014">[DOI]</a>.</p></li>
<li id="id43983"><p>Tom A. Mendum, Jane Newcombe, Ahmad A. Mannan, Andrzej Kierzek, and Johnjoe McFadden. Interrogation of global mutagenesis data with a genome scale model of neisseria meningitidis to assess gene fitness in vitro and in sera. 12(12):R127–R127, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/gb-2011-12-12-r127">[DOI]</a>.</p></li>
<li id="id43985"><p>Christopher S. Henry, Ross Overbeek, Fangfang Xia, Aaron A. Best, Elizabeth M. Glass, Jack A. Gilbert, Peter A. Larsen, Robert A. Edwards, Terry Disz, Folker Meyer, Veronika Vonstein, Matthew DeJongh, Daniela Bartels, Narayan Desai, Mark D’Souza, Scott Devoid, Kevin Keegan, Robert Olson, Andreas Wilke, Jared Wilkening, and Rick Stevens. Connecting genotype to phenotype in the era of high-throughput sequencing. 1810(10):967–977, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bbagen.2011.03.010">[DOI]</a>.</p></li>
<li id="id44058"><p>Yoshihiro Toya, Nobuaki Kono, Kazuharu Arakawa, and Masaru Tomita. Metabolic flux analysis and visualization. 10(8):3313–3323, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/pr2002885">[DOI]</a>.</p></li>
<li id="id44067"><p>Yuqi Zhao and Jing‐Fei Huang. Reconstruction and analysis of human heart-specific metabolic network based on transcriptome and proteome data. 415(3):450–454, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bbrc.2011.10.090">[DOI]</a>.</p></li>
<li id="id44074"><p>Sean P. Cornelius, Joo Sang Lee, and Adilson E. Motter. Dispensability of &lt;i&gt;escherichia coli&lt;/i&gt; ’s latent pathways. 108(8):3124–3129, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1009772108">[DOI]</a>.</p></li>
<li id="id44078"><p>Intawat Nookaew, Roberto Olivares‐Hernández, Sakarindr Bhumiratana, and Jens Nielsen. Genome-scale metabolic models of saccharomyces cerevisiae. 759():445–463, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-61779-173-4_25">[DOI]</a>.</p></li>
<li id="id44085"><p>Juha Leppävuori, Michael M. Domach, and Lorenz T. Biegler. Parameter estimation in batch bioreactor simulation using metabolic models: sequential solution with direct sensitivities. 50(21):12080–12091, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1021/ie201020g">[DOI]</a>.</p></li>
<li id="id44090"><p>Hailin Meng, Zhiguo Lu, Yong Wang, Xiao‐Ning Wang, and Siliang Zhang. In silico improvement of heterologous biosynthesis of erythromycin precursor 6-deoxyerythronolide b in escherichia coli. 16(3):445–456, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-010-0321-7">[DOI]</a>.</p></li>
<li id="id44094"><p>Andrew L. Hopkins, G. Richard Bickerton, Ian M. Carruthers, Stephen Boyer, Harvey Rubin, and John P. Overington. Rapid analysis of pharmacology for infectious diseases. 11(10):1292–1300, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.2174/156802611795429130">[DOI]</a>.</p></li>
<li id="id44120"><p>Ziomara P. Gerdtzen. Modeling metabolic networks for mammalian cell systems: general considerations, modeling strategies, and available tools. 127():71–108, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/10_2011_120">[DOI]</a>.</p></li>
<li id="id44149"><p>Reza Eslamloueyan and Payam Setoodeh. Optimization of fed-batch recombinant yeast fermentation for ethanol production using a reduced dynamic flux balance model based on artificial neural networks. 198(11):1309–1338, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1080/00986445.2011.560512">[DOI]</a>.</p></li>
<li id="id44154"><p>Neema Jamshidi, Franklin J. Miller, Jess Mandel, Timothy Evans, and Michael Kuo. Individualized therapy of hht driven by network analysis of metabolomic profiles. 5(1):200–200, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-200">[DOI]</a>.</p></li>
<li id="id44160"><p>Renato Umeton, Giovanni Stracquadanio, Anilkumar Sorathiya, Píetro Lió, Alessio Papini, and Giuseppe Nicosia. Design of robust metabolic pathways. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/2024724.2024892">[DOI]</a>.</p></li>
<li id="id44165"><p>Aline Métris, Susan George, and József Baranyi. Modelling osmotic stress by flux balance analysis at the genomic scale. 152(3):123–128, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijfoodmicro.2011.06.016">[DOI]</a>.</p></li>
<li id="id44198"><p>Hannes Link and Dirk Weuster‐Botz. Medium formulation and development. ():119–134, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-08-088504-9.00092-1">[DOI]</a>.</p></li>
<li id="id44207"><p>Xin Fang, Anders Wallqvist, and Jaques Reifman. Modeling synergistic drug inhibition of mycobacterium tuberculosis growth in murine macrophages. 7(9):2622–2622, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/c1mb05106g">[DOI]</a>.</p></li>
<li id="id44212"><p>Daniela Xavier, Sara Vázquez, Clara Higuera, Federico Morán, and Francisco Montero. Tools-4-metatool (t4m): online suite of web-tools to process stoichiometric network analysis data from metatool. 105(2):169–172, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2011.04.004">[DOI]</a>.</p></li>
<li id="id44248"><p>Ali Navid. Applications of system-level models of metabolism for analysis of bacterial physiology and identification of new drug targets. 10(6):354–364, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bfgp/elr034">[DOI]</a>.</p></li>
<li id="id44253"><p>Nicolas Turenne. Role of a web-based software platform for systems biology. 04(03):, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.4172/jcsb.1000101e">[DOI]</a>.</p></li>
<li id="id44279"><p>Ehsan Motamedian and Fereshteh Naeimpoor. Prediction of proton exchange and bacterial growth on various substrates using constraint-based modeling approach. 16(5):875–884, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s12257-011-0115-6">[DOI]</a>.</p></li>
<li id="id44286"><p>Rajeev S. Assary and Linda J. Broadbelt. 2-keto acids to branched-chain alcohols as biofuels: application of reaction network analysis and high-level quantum chemical methods to understand thermodynamic landscapes. 978(1-3):160–165, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.comptc.2011.10.009">[DOI]</a>.</p></li>
<li id="id44312"><p>Hannes Link and Dirk Weuster‐Botz. Medium formulation and development. ():199–213, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-444-64046-8.00074-4">[DOI]</a>.</p></li>
<li id="id44327"><p>Sarah Munro, Stephen H. Zinder, and Larry P. Walker. Original research: comparative constraint-based model development for thermophilic hydrogen production. 7(1):63–82, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/ind.2011.7.063">[DOI]</a>.</p></li>
<li id="id44329"><p>Rajeev S. Assary, Linda J. Broadbelt, and Larry A. Curtiss. Brønsted-evans-polanyi relationships for c–c bond forming and c–c bond breaking reactions in thiamine-catalyzed decarboxylation of 2-keto acids using density functional theory. 18(1):145–150, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00894-011-1062-z">[DOI]</a>.</p></li>
<li id="id44335"><p>Aarash Bordbar and Bernhard Ø. Palsson. Moving toward genome-scale kinetic models: the mass action stoichiometric simulation approach. ():201–220, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4614-0320-3_8">[DOI]</a>.</p></li>
<li id="id44353"><p>Ines Thiele, Ronan M. T. Fleming, Richard Que, Aarash Bordbar, and Bernhard Ø. Palsson. A systems biology approach to the evolution of codon use pattern. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/npre.2011.6312.1">[DOI]</a>.</p></li>
<li id="id44354"><p>Sarah Munro, Leila H. Choe, Stephen H. Zinder, Kelvin H. Lee, and Larry P. Walker. Proteomic and physiological experiments to test &lt;i&gt;thermotoga neapolitana&lt;/i&gt; constraint‐based model hypotheses of carbon source utilization. 28(2):312–318, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.735">[DOI]</a>.</p></li>
<li id="id44443"><p>Ines Thiele, Ines Thiele, Ronan M. T. Fleming, Richard Que, Aarash Bordbar, and Bernhard Ø. Palsson. A systems biology approach to the evolution of codon use pattern. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/npre.2011.6312">[DOI]</a>.</p></li>
<li id="id44516"><p>Jan Schellenberger, Nathan E. Lewis, and Bernhard Ø. Palsson. Elimination of thermodynamically infeasible loops in steady-state metabolic models. 100(3):544–553, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.bpj.2010.12.3707">[DOI]</a>.</p></li>
<li id="id44547"><p>Ines Thiele, Daniel R. Hyduke, Benjamin Steeb, Guy Fankam, Doug K. Allen, Susanna Bazzani, Pep Charusanti, Feng‐Chi Chen, Ronan M. T. Fleming, Chao A. Hsiung, Sigrid C. J. De Keersmaecker, Yu-Chieh Liao, Kathleen Marchal, Monica L. Mo, Emre Özdemir, Anu Raghunathan, Jennifer L. Reed, Sookil Shin, Sara Sigurbjörnsdóttir, Jonas Steinmann, Suresh Sudarsan, Neil Swainston, Inge M Thijs, Karsten Zengler, Bernhard Ø. Palsson, Joshua Adkins, and Dirk Bumann. A community effort towards a knowledge-base and mathematical model of the human pathogen salmonella typhimurium lt2. 5(1):8–8, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-5-8">[DOI]</a>.</p></li>
<li id="id44630"><p>Bouke de Jong, Verena Siewers, and Jens Nielsen. Systems biology of yeast: enabling technology for development of cell factories for production of advanced biofuels. 23(4):624–630, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.copbio.2011.11.021">[DOI]</a>.</p></li>
<li id="id44776"><p>Elias W. Krumholz, Hong Yang, Pamela Weisenhorn, Christopher S. Henry, and Igor G. L. Libourel. Genome-wide metabolic network reconstruction of the picoalga ostreococcus. 63(6):2353–2362, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/jxb/err407">[DOI]</a>.</p></li>
<li id="id44799"><p>Neil Swainston, Kieran Smallbone, Pedro Mendes, Douglas B. Kell, and Norman W. Paton. The subliminal toolbox: automating steps in the reconstruction of metabolic networks. 8(2):186–186, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.2390/biecoll-jib-2011-186">[DOI]</a>.</p></li>
<li id="id44829"><p>Ronan M. T. Fleming, Christopher M. Maes, Michael A. Saunders, Y. Ye, and Bernhard Ø. Palsson. A variational principle for computing nonequilibrium fluxes and potentials in genome-scale biochemical networks. 292():71–77, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtbi.2011.09.029">[DOI]</a>.</p></li>
<li id="id45240"><p>Srinubabu Gedela. Integration, warehousing, and analysis strategies of omics data. 719():399–414, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-61779-027-0_18">[DOI]</a>.</p></li>
<li id="id45247"><p>Yu-Chieh Liao, Chih‐Yu Chen, Min-Hsin Tsai, Yueh-Hsia Tang, Feng‐Chi Chen, and Chao A. Hsiung. Mrbac: a web server for draft metabolic network reconstructions for bacteria. 2(5):284–287, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.4161/bbug.2.5.16113">[DOI]</a>.</p></li>
<li id="id45541"><p>Rafik Salama. Computational biology approaches for studying gene regulatory network discovery and modelling. ():, 2011. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id45546"><p>Wout Megchelenbrink, Martijn A. Huynen, and Elena Marchiori. Flux measurement selection in metabolic networks. ():214–224, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-642-24855-9_19">[DOI]</a>.</p></li>
<li id="id45552"><p>Martin I. Sigurðsson. Bioinformatic and biological analysis of dna methylation in the human genome. ():, 2011. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id45565"><p>Ilona Odzina, Reinis Rutkis, and Uldis Kalnenieks. In silico engineering of biochemical network of zymomonas mobilis adaptation for glycerol conversion into bioethanol. ():272–276, 2011. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id45576"><p>William Parra Meneses. Análisis comparativo de la topología de redes metabólicas en una muestra metagenómica de aguas termales en el parque natural de los nevados en colombia. ():, 2011. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id45622"><p>Ronan M. T. Fleming, Christopher M. Maes, Michael A. Saunders, Yinyu Ye, and Bernhard Ø. Palsson. A variational principle for computing nonequilibrium fluxes and potentials in genome-scale biochemical networks. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.48550/arxiv.1105.1513">[DOI]</a>.</p></li>
<li id="id45704"><p>Christopher M. Gowen. Model-guided systems metabolic engineering of clostridium thermocellum. ():, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.25772/n1jy-5g58">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-15" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-15" name="0-15" role="tabpanel" tabindex="0"><div class="docutils container" id="id46816">
<ol class="arabic simple" start="1">
<li id="id46817"><p>Jeffrey D. Orth, Ines Thiele, and Bernhard Ø. Palsson. What is flux balance analysis? 28(3):245–248, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt.1614">[DOI]</a>.</p></li>
<li id="id46820"><p>Ines Thiele and Bernhard Ø. Palsson. A protocol for generating a high-quality genome-scale metabolic reconstruction. 5(1):93–121, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nprot.2009.203">[DOI]</a>.</p></li>
<li id="id46827"><p>Jan Schellenberger, Junyoung O. Park, Tom M Conrad, and Bernhard Ø. Palsson. Bigg: a biochemical genetic and genomic knowledgebase of large scale metabolic reconstructions. 11(1):213–213, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-11-213">[DOI]</a>.</p></li>
<li id="id46834"><p>Sriram Chandrasekaran and Nathan D. Price. Probabilistic integrative modeling of genome-scale metabolic and regulatory networks in &lt;i&gt;escherichia coli&lt;/i&gt; and &lt;i&gt;mycobacterium tuberculosis&lt;/i&gt;. 107(41):17845–17850, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.1005139107">[DOI]</a>.</p></li>
<li id="id46835"><p>Isabel Rocha, Paulo Maia, Pedro Evangelista, Paulo Vilaça, Simão Soares, José P. Pinto, Jens Nielsen, Kiran Raosaheb Patil, Eugénio C. Ferreira, and Miguel Rocha. Optflux: an open-source software platform for in silico metabolic engineering. 4(1):45–45, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-45">[DOI]</a>.</p></li>
<li id="id46836"><p>Steinn Guðmundsson and Ines Thiele. Computationally efficient flux variability analysis. 11(1):, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-11-489">[DOI]</a>.</p></li>
<li id="id46838"><p>Niels Klitgord and Daniel Segrè. Environments that induce synthetic microbial ecosystems. 6(11):e1001002–e1001002, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1001002">[DOI]</a>.</p></li>
<li id="id46846"><p>Jeffrey D. Orth, Ronan M. T. Fleming, and Bernhard Ø. Palsson. Reconstruction and use of microbial metabolic networks: the core &lt;i&gt;escherichia coli&lt;/i&gt; metabolic model as an educational guide. 4(1):, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/ecosalplus.10.2.1">[DOI]</a>.</p></li>
<li id="id46850"><p>Aarash Bordbar, Nathan E. Lewis, Jan Schellenberger, Bernhard Ø. Palsson, and Neema Jamshidi. Insight into human alveolar macrophage and &lt;i&gt;m. tuberculosis&lt;/i&gt; interactions via metabolic reconstructions. 6(1):422–422, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2010.68">[DOI]</a>.</p></li>
<li id="id46852"><p>Keren Yizhak, Tomer Benyamini, Wolfram Liebermeister, Eytan Ruppin, and Tomer Shlomi. Integrating quantitative proteomics and metabolomics with a genome-scale metabolic network model. 26(12):i255–i260, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btq183">[DOI]</a>.</p></li>
<li id="id46858"><p>Roger L. Chang, Li Xie, Lei Xie, Philip E. Bourne, and Bernhard Ø. Palsson. Drug off-target effects predicted using structural analysis in the context of a metabolic network model. 6(9):e1000938–e1000938, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000938">[DOI]</a>.</p></li>
<li id="id46860"><p>Marnix H. Medema, Axel Trefzer, Andriy Kovalchuk, Marco van den Berg, Ulrike Müller, Wilbert H. M. Heijne, Liang Wu, Mohammad Tauqeer Alam, Catherine M. Ronning, William C. Nierman, Roel A. L. Bovenberg, Rainer Breitling, and Eriko Takano. The sequence of a 1.8-mb bacterial linear plasmid reveals a rich evolutionary reservoir of secondary metabolic pathways. 2():212–224, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/gbe/evq013">[DOI]</a>.</p></li>
<li id="id46861"><p>Henning Knoop, Yvonne Zilliges, Wolfgang Lockau, and Ralf Steuer. The metabolic network of &lt;i&gt;synechocystis&lt;/i&gt; sp. pcc 6803: systemic properties of autotrophic growth. 154(1):410–422, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.110.157198">[DOI]</a>.</p></li>
<li id="id46862"><p>Sergio Bordel, Rasmus Ågren, and Jens Nielsen. Sampling the solution space in genome-scale metabolic networks reveals transcriptional regulation in key enzymes. 6(7):e1000859–e1000859, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000859">[DOI]</a>.</p></li>
<li id="id46869"><p>Martin I. Sigurðsson, Neema Jamshidi, Eirı́kur Steingrı́msson, Ines Thiele, and Bernhard Ø. Palsson. A detailed genome-wide reconstruction of mouse metabolism based on human recon 1. 4(1):140–140, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-140">[DOI]</a>.</p></li>
<li id="id46880"><p>Xuewen Chen, Ana Paula Alonso, Doug K. Allen, Jennifer L. Reed, and Yair Shachar‐Hill. Synergy between 13c-metabolic flux analysis and flux balance analysis for understanding metabolic adaption to anaerobiosis in e. coli. 13(1):38–48, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2010.11.004">[DOI]</a>.</p></li>
<li id="id46883"><p>John Blazeck and Hal S. Alper. Systems metabolic engineering: genome‐scale models and beyond. 5(7):647–659, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.200900247">[DOI]</a>.</p></li>
<li id="id46891"><p>Germán Plata, Tzu‐Lin Hsiao, Kellen Olszewski, Manuel Llinás, and Dennis Vitkup. Reconstruction and flux‐balance analysis of the &lt;i&gt;plasmodium falciparum&lt;/i&gt; metabolic network. 6(1):408–408, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2010.60">[DOI]</a>.</p></li>
<li id="id46898"><p>Fahimeh Salimi, Kai Zhuang, and Radhakrishnan Mahadevan. Genome‐scale metabolic modeling of a clostridial co‐culture for consolidated bioprocessing. 5(7):726–738, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201000159">[DOI]</a>.</p></li>
<li id="id46902"><p>Paul D. Dobson, Kieran Smallbone, Daniel Jameson, Evangelos Simeonidis, Karin Lanthaler, Pınar Pir, Chuan Lü, Neil Swainston, Warwick B. Dunn, Paul G. Fisher, Duncan Hull, Marie Brown, Olusegun Oshota, Natalie Stanford, Douglas B. Kell, Ross D. King, Stephen G. Oliver, Robert Stevens, and Pedro Mendes. Further developments towards a genome-scale metabolic model of yeast. 4(1):145–145, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-145">[DOI]</a>.</p></li>
<li id="id46911"><p>Sarah Kinnings, Li Xie, Kingston H. Fung, Richard M. Jackson, Lei Xie, and Philip E. Bourne. The mycobacterium tuberculosis drugome and its polypharmacological implications. 6(11):e1000976–e1000976, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000976">[DOI]</a>.</p></li>
<li id="id46913"><p>Marija Cvijović, Roberto Olivares‐Hernández, Rasmus Ågren, N. Dahr, Wanwipa Vongsangnak, Intawat Nookaew, Kiran Raosaheb Patil, and Jens Nielsen. Biomet toolbox: genome-wide analysis of metabolism. 38(Web Server):W144–W149, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/nar/gkq404">[DOI]</a>.</p></li>
<li id="id46914"><p>Mohammad Tauqeer Alam, Maria Elena Merlo, David A. Hodgson, Elizabeth M. H. Wellington, Eriko Takano, and Rainer Breitling. Metabolic modeling and analysis of the metabolic switch in streptomyces coelicolor. 11(1):202–202, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2164-11-202">[DOI]</a>.</p></li>
<li id="id46921"><p>Matthew Oberhardt, Joanna B. Goldberg, Michael Hogardt, and Jason A. Papin. Metabolic network analysis of &lt;i&gt;pseudomonas aeruginosa&lt;/i&gt; during chronic cystic fibrosis lung infection. 192(20):5534–5548, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.00900-10">[DOI]</a>.</p></li>
<li id="id46926"><p>Li Liu, Rasmus Ågren, Sergio Bordel, and Jens Nielsen. Use of genome‐scale metabolic models for understanding microbial physiology. 584(12):2556–2564, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.febslet.2010.04.052">[DOI]</a>.</p></li>
<li id="id46970"><p>Osbaldo Resendis‐Antonio, Alberto Checa, and Sergio Encarnación‐Guevara. Modeling core metabolism in cancer cells: surveying the topology underlying the warburg effect. 5(8):e12383–e12383, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0012383">[DOI]</a>.</p></li>
<li id="id46983"><p>Michael D. Barton, Daniela Delneri, Stephen G. Oliver, Magnus Rattray, and Casey Bergman. Evolutionary systems biology of amino acid biosynthetic cost in yeast. 5(8):e11935–e11935, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0011935">[DOI]</a>.</p></li>
<li id="id46996"><p>Stephan Noack, Katharina Nöh, Matthias Moch, Marco Oldiges, and Wolfgang Wiechert. Stationary versus non-stationary 13c-mfa: a comparison using a consistent dataset. 154(2-3):179–190, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jbiotec.2010.07.008">[DOI]</a>.</p></li>
<li id="id47007"><p>Limin Li, Xiaobo Zhou, Wai Ki Ching, and Ping Wang. Predicting enzyme targets for cancer drugs by profiling human metabolic reactions in nci-60 cell lines. 11(1):501–501, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-11-501">[DOI]</a>.</p></li>
<li id="id47008"><p>M. Ahsanul Islam, Elizabeth A. Edwards, and Radhakrishnan Mahadevan. Characterizing the metabolism of dehalococcoides with a constraint-based model. 6(8):e1000887–e1000887, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000887">[DOI]</a>.</p></li>
<li id="id47015"><p>Albert Gevorgyan, Michael E. Bushell, Claudio Avignone–Rossa, and Andrzej Kierzek. Surreyfba: a command line tool and graphics user interface for constraint-based modeling of genome-scale metabolic reaction networks. 27(3):433–434, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btq679">[DOI]</a>.</p></li>
<li id="id47023"><p>K. Srinivasan and Radhakrishnan Mahadevan. Characterization of proton production and consumption associated with microbial metabolism. 10(1):2–2, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1472-6750-10-2">[DOI]</a>.</p></li>
<li id="id47026"><p>John S. Ramsey, Sandy MacDonald, Georg Jander, Atsushi Nakabachi, Gavin H. Thomas, and Angela E. Douglas. Genomic evidence for complementary purine metabolism in the pea aphid, &lt;i&gt;acyrthosiphon pisum&lt;/i&gt; , and its symbiotic bacterium &lt;i&gt;buchnera aphidicola&lt;/i&gt;. 19(s2):241–248, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1365-2583.2009.00945.x">[DOI]</a>.</p></li>
<li id="id47030"><p>Ronan M. T. Fleming and Ines Thiele. Von bertalanffy 1.0: a cobra toolbox extension to thermodynamically constrain metabolic models. 27(1):142–143, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btq607">[DOI]</a>.</p></li>
<li id="id47048"><p>Anu Raghunathan, Sookil Shin, and Simon Daefler. Systems approach to investigating host-pathogen interactions in infections with the biothreat agent francisella. constraints-based model of francisella tularensis. 4(1):118–118, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-118">[DOI]</a>.</p></li>
<li id="id47078"><p>Xin Fang, Anders Wallqvist, and Jaques Reifman. Development and analysis of an in vivo-compatible metabolic network of mycobacterium tuberculosis. 4(1):160–160, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-160">[DOI]</a>.</p></li>
<li id="id47090"><p>Xueyang Feng, Lawrence M. Page, Jacob R. Rubens, Lauren Chircus, Peter Colletti, Himadri B. Pakrasi, and Yinjie Tang. Bridging the gap between fluxomics and industrial biotechnology. 2010():1–13, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1155/2010/460717">[DOI]</a>.</p></li>
<li id="id47101"><p>Brian J. Schmidt, Xiefan Lin‐Schmidt, Austin Chamberlin, Kourosh Salehi‐Ashtiani, and Jason A. Papin. Metabolic systems analysis to advance algal biotechnology. 5(7):660–670, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/biot.201000129">[DOI]</a>.</p></li>
<li id="id47127"><p>Sergey Kozhenkov, Yulia Dubinina, Mayya Sedova, Amarnath Gupta, Julia Ponomarenko, and Michael Baitaluk. Biologicalnetworks 2.0 - an integrative view of genome biology data. 11(1):610–610, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-11-610">[DOI]</a>.</p></li>
<li id="id47145"><p>Rainer Breitling, Robin Donaldson, David Gilbert, and Monika Heiner. Biomodel engineering – from structure to behavior. ():1–12, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-642-11712-1_1">[DOI]</a>.</p></li>
<li id="id47171"><p>Han Min Woo, Stephan Noack, Gerd M. Seibold, Sabine Willbold, Bernhard J. Eikmanns, and Michael Bott. Link between phosphate starvation and glycogen metabolism in &lt;i&gt;corynebacterium glutamicum&lt;/i&gt; , revealed by metabolomics. 76(20):6910–6919, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.01375-10">[DOI]</a>.</p></li>
<li id="id47186"><p>Marija Cvijović, Sergio Bordel, and Jens Nielsen. Mathematical models of cell factories: moving towards the core of industrial biotechnology. 4(5):572–584, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1751-7915.2010.00233.x">[DOI]</a>.</p></li>
<li id="id47214"><p>Paulo Vilaça, Isabel Rocha, and Miguel Rocha. A computational tool for the simulation and optimization of microbial strains accounting integrated metabolic/regulatory information. 103(3):435–441, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2010.11.012">[DOI]</a>.</p></li>
<li id="id47216"><p>Thomas Forth, Glenn A. McConkey, and David R. Westhead. Metnetmaker: a free and open-source tool for the creation of novel metabolic networks in sbml format. 26(18):2352–2353, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btq425">[DOI]</a>.</p></li>
<li id="id47233"><p>Michael D. Barton, Daniela Delneri, Stephen G. Oliver, Magnus Rattray, and Casey Bergman. Correction: evolutionary systems biology of amino acid biosynthetic cost in yeast. 5(10):, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/annotation/b60feca4-9a4f-4311-8a07-4bf16a2f9316">[DOI]</a>.</p></li>
<li id="id47288"><p>Dirk Koschützki, Björn H. Junker, Jörg Schwender, and Falk Schreiber. Structural analysis of metabolic networks based on flux centrality. 265(3):261–269, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtbi.2010.05.009">[DOI]</a>.</p></li>
<li id="id47305"><p>Yong Xi, Yi‐Ping Phoebe Chen, Qian Chen, and Fei Wang. Comparative study of computational methods to detect the correlated reaction sets in biochemical networks. 12(2):132–150, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbp068">[DOI]</a>.</p></li>
<li id="id47370"><p>Sven Dietz and Sven Panke. Microbial systems engineering: first successes and the way ahead. 32(4):356–362, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/bies.200900174">[DOI]</a>.</p></li>
<li id="id47371"><p>Patrick May, Nils Christian, Oliver Ebenhöh, Wolfram Weckwerth, and Dirk Walther. Integration of proteomic and metabolomic profiling as well as metabolic modeling for the functional analysis of metabolic networks. 694():341–363, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-60761-977-2_21">[DOI]</a>.</p></li>
<li id="id47378"><p>Rajeev S. Assary and Linda J. Broadbelt. Computational screening of novel thiamine-catalyzed decarboxylation reactions of 2-keto acids. 34(3):375–388, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s00449-010-0481-z">[DOI]</a>.</p></li>
<li id="id47382"><p>Ashwini G. Bhat, Rohit Vashisht, and Nagasuma Chandra. Modeling metabolic adjustment in mycobacterium tuberculosis upon treatment with isoniazid. 4(4):299–309, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11693-011-9075-6">[DOI]</a>.</p></li>
<li id="id47383"><p>Qi Chen, Zhuo Wang, and Dong‐Qing Wei. Progress in the applications of flux analysis of metabolic networks. 55(22):2315–2322, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/s11434-010-3022-x">[DOI]</a>.</p></li>
<li id="id47412"><p>David Hala, Aneeqa Amin, Armin R. Mikler, and Duane B. Huggett. A constraint-based stoichiometric model of the steroidogenic network of zebrafish (&lt;i&gt;danio rerio&lt;/i&gt;). 18(03):669–685, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1142/s0218339010003469">[DOI]</a>.</p></li>
<li id="id47424"><p>Fahimeh Salimi, Rupasri Mandal, David S. Wishart, and Radhakrishnan Mahadevan. Understanding clostridium acetobutylicum atcc 824 metabolism using genome-scale thermodynamics and metabolomics-based modeling. 43(6):126–131, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.3182/20100707-3-be-2012.0022">[DOI]</a>.</p></li>
<li id="id47425"><p>Nadeera Jayasinghe and Radhakrishnan Mahadevan. Metabolic modeling of spatial heterogeneity of biofilms in microbial fuel cells. 43(6):215–220, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.3182/20100707-3-be-2012.0050">[DOI]</a>.</p></li>
<li id="id47442"><p>Maik Kschischo. A gentle introduction to the thermodynamics of biochemical stoichiometric networks in steady state. 187(1):255–274, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1140/epjst/e2010-01290-3">[DOI]</a>.</p></li>
<li id="id47443"><p>Balaji Veeramani and Joel S. Bader. Predicting functional associations from metabolism using bi-partite network algorithms. 4(1):95–95, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-4-95">[DOI]</a>.</p></li>
<li id="id47444"><p>Mano R. Maurya and Shankar Subramaniam. Computational challenges in systems biology. ():175–223, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-12-372550-9.00008-0">[DOI]</a>.</p></li>
<li id="id47485"><p>Joseph Bressler, Joel S. Bader, and Alan M. Goldberg. Alternatives to conventional toxicology testing. ():247–259, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/b978-0-08-046884-6.00324-9">[DOI]</a>.</p></li>
<li id="id47513"><p>James M. Smith and Marc‐Thorsten Hütt. Network dynamics as an interface between modeling and experiment in systems biology. ():243–276, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9783527630271.ch12">[DOI]</a>.</p></li>
<li id="id47531"><p>Ding De-wu and He Xiao-qing. Analyzing the core of e. coli metabolism with flux balance. 102():195–197, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/cinc.2010.5643859">[DOI]</a>.</p></li>
<li id="id47536"><p>David Hala, Duane B. Huggett, and Dalma Martinović‐Weigelt. &lt;i&gt;in silico&lt;/i&gt; predicted essential genes required for zebrafish ( &lt;i&gt;danio rerio&lt;/i&gt; ) steroid hormone production. ():487–489, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/1854776.1854866">[DOI]</a>.</p></li>
<li id="id47568"><p>琦 陈, 冬青 魏, and 卓 王. 代谢网络流分析进展及应用. 55(14):1302–1309, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1360/csb2010-55-14-1302">[DOI]</a>.</p></li>
<li id="id47618"><p>Kai Zhuang, Mounir Izallalen, Paula Mouser, Hanno Richter, Carla Risso, Radhakrishnan Mahadevan, and Derek R. Lovley. Genome-scale dynamic modeling of the competition between &lt;i&gt;rhodoferax&lt;/i&gt; and &lt;i&gt;geobacter&lt;/i&gt; in anoxic subsurface environments. 5(2):305–316, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/ismej.2010.117">[DOI]</a>.</p></li>
<li id="id47698"><p>Masafumi Nozawa, Sayaka Miura, and Masatoshi Nei. Origins and evolution of microrna genes in drosophila species. 2():180–189, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/gbe/evq009">[DOI]</a>.</p></li>
<li id="id48545"><p>Daniel Machado, Rafael S. Costa, Miguel Rocha, Isabel Rocha, Bruce Tidor, and Eugénio C. Ferreira. Model transformation of metabolic networks using a petri net based framework. 827():101–115, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48602"><p>Laurence Yang, Radhakrishnan Mahadevan, and W.R. Cluett. Designing experiments from noisy metabolomics data to refine constraint-based models. ():5143–5148, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/acc.2010.5530678">[DOI]</a>.</p></li>
<li id="id48644"><p>Hui Wang, Hongwu Ma, and Xueming Zhao. [progress in genome-scale metabolic network: a review]. 26(10):1340–8, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48649"><p>Sarah Munro. Systems biology of thermotoga neapolitana for biological hydrogen production. ():, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48651"><p>Adrián López García de Lomana. Computational approaches to the modelling of topological and dynamical aspects of biochemical networks. ():, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48688"><p>C. D. Machado. Novel modeling formalisms and simulation tools in computational biosystems. ():18–18, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48689"><p>Saeed Sheykhshoaieekhttiarabadi. An automated approach to assigning subcellular localization in genome-scale metabolic network reconstructions. ():, 2010. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id48713"><p>Zaldivar Comenges Jose and Fernanda Strozzi. Analysis and simulation of metabolic networks: application to hepg2. ():, 2010. <a class="reference external" href="https://doi.org/https://doi.org/10.2788/30083">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-16" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-16" name="0-16" role="tabpanel" tabindex="0"><div class="docutils container" id="id49937">
<ol class="arabic simple" start="1">
<li id="id49946"><p>Matthew Oberhardt, Bernhard Ø. Palsson, and Jason A. Papin. Applications of genome‐scale metabolic reconstructions. 5(1):320–320, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2009.77">[DOI]</a>.</p></li>
<li id="id49950"><p>Monica L. Mo, Bernhard Ø. Palsson, and Markus J. Herrgård. Connecting extracellular metabolomic measurements to intracellular flux states in yeast. 3(1):37–37, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-3-37">[DOI]</a>.</p></li>
<li id="id49952"><p>Karthik Raman and Nagasuma Chandra. Flux balance analysis of biological systems: applications and challenges. 10(4):435–449, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbp011">[DOI]</a>.</p></li>
<li id="id49974"><p>Adam M. Feist, Daniel C. Zielinski, Jeffrey D. Orth, Jan Schellenberger, Markus J. Herrgård, and Bernhard Ø. Palsson. Model-driven evaluation of the production potential for growth-coupled products of escherichia coli. 12(3):173–186, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2009.10.003">[DOI]</a>.</p></li>
<li id="id49991"><p>Kenji Nakahigashi, Yoshihiro Toya, Nobuyoshi Ishii, Tomoyoshi Soga, Miki Hasegawa, Hisami Watanabe, Yuki Takai, Masayuki Honma, Hirotada Mori, and Masaru Tomita. Systematic phenome analysis of &lt;i&gt;escherichia coli&lt;/i&gt; multiple‐knockout mutants reveals hidden reactions in central carbon metabolism. 5(1):306–306, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2009.65">[DOI]</a>.</p></li>
<li id="id50000"><p>Bas Teusink, Anne Wiersma, Leo Jacobs, Richard A. Notebaart, and Eddy J. Smid. Understanding the adaptive growth strategy of lactobacillus plantarum by in silico optimisation. 5(6):e1000410–e1000410, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000410">[DOI]</a>.</p></li>
<li id="id50038"><p>Ani Manichaikul, Lila Ghamsari, Erik Hom, Chenwei Lin, Ryan R. Murray, Roger L. Chang, Balaji Santhanam, Tong Hao, Yun Shen, Arvind K. Chavali, Ines Thiele, Xinping Yang, Changyu Fan, Elizabeth Mello, David E. Hill, Marc Vidal, Kourosh Salehi‐Ashtiani, and Jason A. Papin. Metabolic network analysis integrated with transcript verification for sequenced genomes. 6(8):589–592, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nmeth.1348">[DOI]</a>.</p></li>
<li id="id50039"><p>Bernhard Ø. Palsson. Metabolic systems biology. 583(24):3900–3904, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.febslet.2009.09.031">[DOI]</a>.</p></li>
<li id="id50077"><p>Gavin H. Thomas, Jeremy Zucker, Sandy MacDonald, Anatoly Sorokin, Igor Goryanin, and Angela E. Douglas. A fragile metabolic network adapted for cooperation in the symbiotic bacterium buchnera aphidicola. 3(1):24–24, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-3-24">[DOI]</a>.</p></li>
<li id="id50083"><p>Zhi Wang and Jianzhi Zhang. Why is the correlation between gene importance and gene evolutionary rate so weak? 5(1):e1000329–e1000329, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pgen.1000329">[DOI]</a>.</p></li>
<li id="id50088"><p>Zhi Wang and Jianzhi Zhang. Abundant indispensable redundancies in cellular metabolic networks. 1():23–33, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/gbe/evp002">[DOI]</a>.</p></li>
<li id="id50097"><p>Max Hoek and Paulien Hogeweg. Metabolic adaptation after whole genome duplication. 26(11):2441–2453, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/molbev/msp160">[DOI]</a>.</p></li>
<li id="id50112"><p>Matthew Oberhardt, Arvind K. Chavali, and Jason A. Papin. Flux balance analysis: interrogating genome-scale metabolic networks. 500():61–80, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-59745-525-1_3">[DOI]</a>.</p></li>
<li id="id50121"><p>Jacob Feala, Laurence Coquin, Dan Zhou, Gabriel G. Haddad, Giovanni Paternostro, and Andrew D. McCulloch. Metabolism as means for hypoxia adaptation: metabolic profiling and flux balance analysis. 3(1):91–91, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-3-91">[DOI]</a>.</p></li>
<li id="id50124"><p>Christian Barrett, Markus J. Herrgård, and Bernhard Ø. Palsson. Decomposing complex reaction networks using random sampling, principal component analysis and basis rotation. 3(1):30–30, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-3-30">[DOI]</a>.</p></li>
<li id="id50125"><p>Fan Li, Ines Thiele, Neema Jamshidi, and Bernhard Ø. Palsson. Identification of potential pathway mediation targets in toll-like receptor signaling. 5(2):e1000292–e1000292, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000292">[DOI]</a>.</p></li>
<li id="id50141"><p>Eva Grafahrend‐Belau, Christian Klukas, Björn H. Junker, and Falk Schreiber. Fba-simvis: interactive visualization of constraint-based metabolic models. 25(20):2755–2757, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btp408">[DOI]</a>.</p></li>
<li id="id50162"><p>Vicente Acuña, Alberto Marchetti-Spaccamela, Marie-France Sagot, and Leen Stougie. A note on the complexity of finding and enumerating elementary modes. 99(3):210–214, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.biosystems.2009.11.004">[DOI]</a>.</p></li>
<li id="id50259"><p>Osbaldo Resendis‐Antonio. Filling kinetic gaps: dynamic modeling of metabolism where detailed kinetic information is lacking. 4(3):e4967–e4967, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pone.0004967">[DOI]</a>.</p></li>
<li id="id50291"><p>Alfonso Pérez‐Escudero, Marta Rivera-Alba, and Gonzalo G. de Polavieja. Structure of deviations from optimality in biological systems. 106(48):20544–20549, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1073/pnas.0905336106">[DOI]</a>.</p></li>
<li id="id50293"><p>Xin Fang, Anders Wallqvist, and Jaques Reifman. A systems biology framework for modeling metabolic enzyme inhibition of mycobacterium tuberculosis. 3(1):92–92, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-3-92">[DOI]</a>.</p></li>
<li id="id50297"><p>Karthik Raman, Rohit Vashisht, and Nagasuma Chandra. Strategies for efficient disruption of metabolism in mycobacterium tuberculosis from network analysis. 5(12):1740–1740, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/b905817f">[DOI]</a>.</p></li>
<li id="id50302"><p>Hannes Link, Bernd Anselment, and Dirk Weuster‐Botz. Rapid media transition: an experimental approach for steady state analysis of metabolic pathways. 26(1):1–10, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/btpr.290">[DOI]</a>.</p></li>
<li id="id50315"><p>Douglas Densmore, Anne Van Devender, Matthew Johnson, and Nade Sritanyaratana. A platform-based design environment for synthetic biological systems. ():24–29, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/1565799.1565806">[DOI]</a>.</p></li>
<li id="id50319"><p>François Le Fèvre, Serge Smidtas, Cyril Combe, Maxime Durot, Florence d’Alché–Buc, and Vincent Schächter. Cycsim—an online tool for exploring and experimenting with genome-scale metabolic models. 25(15):1987–1988, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btp268">[DOI]</a>.</p></li>
<li id="id50323"><p>Yanping Xi, Yi‐Ping Phoebe Chen, Ming Cao, Weirong Wang, and Fei Wang. Analysis on relationship between extreme pathways and correlated reaction sets. 10(S1):S58–S58, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-10-s1-s58">[DOI]</a>.</p></li>
<li id="id50347"><p>Brett A. Boghigian, Kyongbum Lee, and Blaine A. Pfeifer. Computational analysis of phenotypic space in heterologous polyketide biosynthesis—applications to escherichia coli, bacillus subtilis, and saccharomyces cerevisiae. 262(2):197–207, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.jtbi.2009.10.006">[DOI]</a>.</p></li>
<li id="id50389"><p>Jenni Heino, Daniela Calvetti, and Erkki Somersalo. Metabolica: a statistical research tool for analyzing metabolic networks. 97(2):151–167, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.cmpb.2009.07.007">[DOI]</a>.</p></li>
<li id="id50398"><p>Maikel P. H. Verouden, Richard A. Notebaart, Johan A. Westerhuis, Mariët J. van der Werf, Bas Teusink, and Age K. Smilde. Multi‐way analysis of flux distributions across multiple conditions. 23(7-8):406–420, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/cem.1238">[DOI]</a>.</p></li>
<li id="id50420"><p>Tae Sung Jung, Hock Chuan Yeo, Satty G. Reddy, Wan-Sup Cho, and Dong‐Yup Lee. Webcoli: an interactive and asynchronous web application for &lt;i&gt;in silico&lt;/i&gt; design and analysis of genome-scale &lt;i&gt;e.coli&lt;/i&gt; model. 25(21):2850–2852, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bioinformatics/btp496">[DOI]</a>.</p></li>
<li id="id50497"><p>Balaji Veeramani and Joel S. Bader. Metabolic flux correlations, genetic interactions, and disease. 16(2):291–302, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1089/cmb.2008.14tt">[DOI]</a>.</p></li>
<li id="id50498"><p>Nathan E. Lewis, Neema Jamshidi, Ines Thiele, and Bernhard Ø. Palsson. Metabolic systems biology. ():5535–5552, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-0-387-30440-3_329">[DOI]</a>.</p></li>
<li id="id50500"><p>Zixiang Xu, Xiao Sun, and Shihai Yu. Genome-scale analysis to the impact of gene deletion on the metabolism of e. coli: constraint-based simulation approach. 10(S1):S62–S62, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-10-s1-s62">[DOI]</a>.</p></li>
<li id="id50513"><p>Patrick B. Warren, Sı́lvio M. Duarte Queirós, and Janette L. Jones. Flux networks in metabolic graphs. 6(4):046006–046006, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1088/1478-3975/6/4/046006">[DOI]</a>.</p></li>
<li id="id50594"><p>Adam M. Feist, Ines Thiele, and Bernhard Ø. Palsson. Genome-scale reconstruction, modeling, and simulation of e. coli℉s metabolic network. ():149–176, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4020-9394-4_9">[DOI]</a>.</p></li>
<li id="id50647"><p>Scott Baker, Kenneth S. Bruno, Mark Butcher, James R. Collett, David Culley, Ziyu Dai, Jon Magnuson, and Ellen Panisko. Pnnl fungal biotechnology core doe-obp project. ():, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.2172/986728">[DOI]</a>.</p></li>
<li id="id50728"><p>Peter D. Karp, Suzanne Paley, Markus Krummenacker, Mario Latendresse, J. M. Dale, T. J. Lee, P. Kaipa, Fred Gilham, Aaron Spaulding, Liviu Popescu, Tomer Altman, Ian T. Paulsen, Ingrid M. Keseler, and Ron Caspi. Pathway tools version 13.0: integrated software for pathway/genome informatics and systems biology. 11(1):40–79, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1093/bib/bbp043">[DOI]</a>.</p></li>
<li id="id50822"><p>Brett A. Boghigian, Gargi Seth, Róbert Kiss, and Blaine A. Pfeifer. Metabolic flux analysis and pharmaceutical production. 12(2):81–95, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ymben.2009.10.004">[DOI]</a>.</p></li>
<li id="id51689"><p>Michael Y. Galperin. Sensory transduction network of e. coli. ():133–148, 2009. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-1-4020-9394-4_8">[DOI]</a>.</p></li>
<li id="id51786"><p>Richard A. Notebaart. Integrative bioinformatics of metabolic networks. ():, 2009. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id51945"><p>Alexander James Macdonald. Theoretical and in vitro analysis of iron acquisition in pasteurella multocida a:3. ():, 2009. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-17" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-17" name="0-17" role="tabpanel" tabindex="0"><div class="docutils container" id="id53058">
<ol class="arabic simple" start="1">
<li id="id53066"><p>Adam M. Feist, Markus J. Herrgård, Ines Thiele, Jennifer L. Reed, and Bernhard Ø. Palsson. Reconstruction of biochemical networks in microorganisms. 7(2):129–143, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nrmicro1949">[DOI]</a>.</p></li>
<li id="id53068"><p>Scott A. Becker and Bernhard Ø. Palsson. Context-specific metabolic networks are consistent with experiments. 4(5):e1000082–e1000082, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.1000082">[DOI]</a>.</p></li>
<li id="id53070"><p>Adam M. Feist and Bernhard Ø. Palsson. The growing scope of applications of genome-scale metabolic reconstructions using escherichia coli. 26(6):659–667, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/nbt1401">[DOI]</a>.</p></li>
<li id="id53081"><p>Maxime Durot, Pierre-Yves Bourguignon, and Vincent Schächter. Genome-scale models of bacterial metabolism: reconstruction and applications. 33(1):164–190, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1574-6976.2008.00146.x">[DOI]</a>.</p></li>
<li id="id53087"><p>Karthik Raman, Kalidas Yeturu, and Nagasuma Chandra. Targettb: a target identification pipeline for mycobacterium tuberculosis through an interactome, reactome and genome-scale structural analysis. 2(1):109–109, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-2-109">[DOI]</a>.</p></li>
<li id="id53091"><p>Juan Nogales, Bernhard Ø. Palsson, and Ines Thiele. A genome-scale metabolic reconstruction of pseudomonas putida kt2440: i jn746 as a cell factory. 2(1):79–79, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-2-79">[DOI]</a>.</p></li>
<li id="id53093"><p>Jan Schellenberger and Bernhard Ø. Palsson. Use of randomized sampling for analysis of metabolic networks. 284(9):5457–5461, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1074/jbc.r800048200">[DOI]</a>.</p></li>
<li id="id53109"><p>Eva Grafahrend‐Belau, Falk Schreiber, Dirk Koschützki, and Björn H. Junker. Flux balance analysis of barley seeds: a computational approach to study systemic properties of central metabolism. 149(1):585–598, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1104/pp.108.129635">[DOI]</a>.</p></li>
<li id="id53200"><p>Richard A. Notebaart, Bas Teusink, Roland J. Siezen, and Balázs Papp. Co-regulation of metabolic genes is better explained by flux coupling than by network distance. 4(1):e26–e26, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1371/journal.pcbi.0040026">[DOI]</a>.</p></li>
<li id="id53208"><p>Aneil Mallavarapu, Matthew Thomson, Benjamin Ullian, and Jeremy Gunawardena. Programming with models: modularity and abstraction provide powerful capabilities for systems biology. 6(32):257–270, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1098/rsif.2008.0205">[DOI]</a>.</p></li>
<li id="id53214"><p>Segun Fatumo, Kitiporn Plaimas, Jan‐Philipp Mallm, Gunnar Schramm, Ezekiel Adebiyi, Marcus Oswald, Roland Eils, and Rainer König. Estimating novel potential drug targets of plasmodium falciparum by analysing the metabolic network of knock-out strains in silico. 9(3):351–358, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.meegid.2008.01.007">[DOI]</a>.</p></li>
<li id="id53215"><p>Chenggang Yu, Nela Zavaljevski, Valmik Desai, and Jaques Reifman. Genome‐wide enzyme annotation with precision control: catalytic families (catfam) databases. 74(2):449–460, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/prot.22167">[DOI]</a>.</p></li>
<li id="id53216"><p>Laurence Coquin, Jacob Feala, Andrew D. McCulloch, and Giovanni Paternostro. Metabolomic and flux‐balance analysis of age‐related decline of hypoxia tolerance in&lt;i&gt;drosophila&lt;/i&gt;muscle tissue. 4(1):233–233, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb.2008.71">[DOI]</a>.</p></li>
<li id="id53219"><p>Varun Mazumdar, Evan Snitkin, Salomon Amar, and Daniel Segrè. Metabolic network model of a human oral pathogen. 191(1):74–90, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/jb.01123-08">[DOI]</a>.</p></li>
<li id="id53237"><p>Vasiliy A. Portnoy, Markus J. Herrgård, and Bernhard Ø. Palsson. Aerobic fermentation of &lt;scp&gt;d&lt;/scp&gt; -glucose by an evolved cytochrome oxidase-deficient &lt;i&gt;escherichia coli&lt;/i&gt; strain. 74(24):7561–7569, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1128/aem.00880-08">[DOI]</a>.</p></li>
<li id="id53261"><p>Evan Snitkin, Aimée M. Dudley, Daniel M. Janse, K.K. Wong, George M. Church, and Daniel Segrè. Model-driven analysis of experimentally determined growth phenotypes for 465 yeast gene deletion mutants under 16 different conditions. 9(9):R140–R140, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/gb-2008-9-9-r140">[DOI]</a>.</p></li>
<li id="id53280"><p>Sakiko Okumoto, Hitomi Takanaga, and Wolf B. Frommer. Quantitative imaging for discovery and assembly of the metabo‐regulome. 180(2):271–295, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1111/j.1469-8137.2008.02611.x">[DOI]</a>.</p></li>
<li id="id53288"><p>Jeremiah Wright and Andreas Wagner. The systems biology research tool: evolvable open-source software. 2(1):55–55, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-2-55">[DOI]</a>.</p></li>
<li id="id53294"><p>Monica L. Mo and Bernhard Ø. Palsson. Understanding human metabolic physiology: a genome-to-systems approach. 27(1):37–44, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.tibtech.2008.09.007">[DOI]</a>.</p></li>
<li id="id53407"><p>Dirk Bumann. Has nature already identified all useful antibacterial targets? 11(5):387–392, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.mib.2008.08.002">[DOI]</a>.</p></li>
<li id="id53408"><p>Kitiporn Plaimas, Jan‐Philipp Mallm, Marcus Oswald, Fabian Svara, Victor Sourjik, Roland Eils, and Rainer König. Machine learning based analyses on metabolic networks supports high-throughput knockout screens. 2(1):67–67, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-2-67">[DOI]</a>.</p></li>
<li id="id53609"><p>Scott A. Becker and Bernhard Ø. Palsson. Three factors underlying incorrect in silico predictions of essential metabolic genes. 2(1):14–14, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1752-0509-2-14">[DOI]</a>.</p></li>
<li id="id53699"><p>Pierre-Yves Bourguignon, Jacques van Helden, Christos Ouzounis, and Vincent Schächter. Computational analysis of metabolic networks. ():329–351, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1007/978-3-211-75123-7_16">[DOI]</a>.</p></li>
<li id="id53700"><p>Eleftheria Tzamali and Martin Reczko. The benefit of cooperation: identifying growth-efficient interacting strains of escherichia coli using metabolic flux balance models. ():1–6, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/bibe.2008.4696735">[DOI]</a>.</p></li>
<li id="id53701"><p>Xinghua Shi and Rick Stevens. Swarm. ():25–34, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1145/1383529.1383535">[DOI]</a>.</p></li>
<li id="id53753"><p>Ralf Steuer and Björn H. Junker. Computational models of metabolism: stability and regulation in metabolic networks. ():105–251, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1002/9780470475935.ch3">[DOI]</a>.</p></li>
<li id="id53776"><p>Zixiang Xu, Jianming Xie, Yang Xinan, and Xiao Sun. Genome-scale simulation analysis: the impact of gene deletion on the metabolic flux of e.coli and its flux-backbone. 2():695–699, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/bmei.2008.259">[DOI]</a>.</p></li>
<li id="id53777"><p>Zixiang Xu, Jianming Xie, Xinan Yang, and Xiao Sun. Impact analysis of gene deletion on the metabolic flux distribution of e.coli and on its flux-backbone: genome-scale simulation approach. ():454–457, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1109/icbbe.2008.111">[DOI]</a>.</p></li>
<li id="id54111"><p>Patrik R. Jones. Improving fermentative biomass-derived h2-production by engineering microbial metabolism. 33(19):5122–5130, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1016/j.ijhydene.2008.05.004">[DOI]</a>.</p></li>
<li id="id54704"><p>Zixiang Xu and Xiao Sun. Constrain-based analysis of gene deletion on the metabolic flux redistribution of saccharomyces cerevisiae. 01(02):121–126, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.4236/jbise.2008.12020">[DOI]</a>.</p></li>
<li id="id54716"><p>Michael Bartoň, Balázs Papp, Daniela Delneri, Stephen G. Oliver, Magnus Rattray, and Casey Bergman. Systems biology of energetic and atomic costs in the yeast transcriptome, proteome, and metabolome. ():, 2008. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/npre.2008.1841.2">[DOI]</a>.</p></li>
<li id="id54884"><p>Milan J. A. Hoek. Evolutionary dynamics of metabolic adaptation. ():, 2008. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id54923"><p>Itamar L. de Oliveira. Análise computacional de redes metabólicas com regulação. ():, 2008. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
<li id="id54959"><p>Eva Grafahrend‐Belau, Björn H. Junker, Dirk Koschützki, Christian Klukas, Stéphan Weise, Uwe Scholz, and Falk Schreiber. Towards systems biology of developing barley grains : a framework for modeling metabolism. ():41–44, 2008. <a class="reference external" href="https://doi.org/">[DOI]</a>.</p></li>
</ol>
</div>
</div><div aria-labelledby="tab-0-0-18" class="sphinx-tabs-panel" hidden="true" id="panel-0-0-18" name="0-18" role="tabpanel" tabindex="0"><div class="docutils container" id="id56179">
<ol class="arabic simple" start="1">
<li id="id56184"><p>Adam M. Feist, Christopher S. Henry, Jennifer L. Reed, Markus Krummenacker, Andrew R. Joyce, Peter D. Karp, Linda J. Broadbelt, Vassily Hatzimanikatis, and Bernhard Ø. Palsson. A genome‐scale metabolic reconstruction for &lt;i&gt;escherichia coli&lt;/i&gt; k‐12 mg1655 that accounts for 1260 orfs and thermodynamic information. 3(1):121–121, 2007. <a class="reference external" href="https://doi.org/https://doi.org/10.1038/msb4100155">[DOI]</a>.</p></li>
<li id="id56349"><p>Monica L. Mo, Neema Jamshidi, and Bernhard Ø. Palsson. A genome-scale, constraint-based approach to systems biology of human metabolism. 3(9):598–598, 2007. <a class="reference external" href="https://doi.org/https://doi.org/10.1039/b705597h">[DOI]</a>.</p></li>
</ol>
</div>
</div></div>
</section>
</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="support.html" class="btn btn-default float-right" title="Support" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="cite.html" class="btn btn-default" title="How to cite" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2025, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/sphinx_highlight.js"></script>
    <script type="text/javascript" src="_static/tabs.js"></script>

  


  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>