<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><script type="text/javascript" src="https://cdn.jsdelivr.net/gh/opencobra/cobratoolbox@ffa0229fc0c01c9236bb7e961f65712443277719/latest/_static/js/iframeResizer.contentWindow.min.js"></script><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge,IE=9,chrome=1"><meta name="generator" content="MATLAB 2021a"><title>Flux Balance Analysis (FBA) and its variants</title><style type="text/css">.rtcContent { padding: 30px; } .S0 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 28.8px; min-height: 0px; white-space: normal; color: rgb(213, 80, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 24px; font-weight: normal; text-align: left;  }
.S1 { margin: 20px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: normal; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: bold; text-align: left;  }
.S2 { margin: 2px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: normal; text-align: left;  }
.S3 { margin: 2px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: normal; text-align: center;  }
.S4 { margin: 10px 0px 20px; padding-left: 0px; font-family: Helvetica, Arial, sans-serif; font-size: 14px;  }
.S5 { margin-left: 56px; line-height: 21px; min-height: 0px; text-align: left; white-space: normal;  }
.S6 { margin: 3px 10px 5px 4px; padding: 0px; line-height: 20px; min-height: 0px; white-space: normal; color: rgb(60, 60, 60); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 20px; font-weight: bold; text-align: left;  }
.CodeBlock { background-color: #F7F7F7; margin: 10px 0 10px 0;}
.S7 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 1px solid rgb(233, 233, 233); border-bottom: 0px none rgb(0, 0, 0); border-radius: 4px 4px 0px 0px; padding: 6px 45px 0px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S8 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 0px none rgb(0, 0, 0); border-radius: 0px; padding: 0px 45px 0px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S9 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 1px solid rgb(233, 233, 233); border-radius: 0px 0px 4px 4px; padding: 0px 45px 4px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S10 { margin: 10px 10px 9px 4px; padding: 0px; line-height: 21px; min-height: 0px; white-space: normal; color: rgb(0, 0, 0); font-family: Helvetica, Arial, sans-serif; font-style: normal; font-size: 14px; font-weight: normal; text-align: left;  }
.S11 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 1px solid rgb(233, 233, 233); border-bottom: 1px solid rgb(233, 233, 233); border-radius: 4px; padding: 6px 45px 4px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S12 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 1px solid rgb(233, 233, 233); border-bottom: 1px solid rgb(233, 233, 233); border-radius: 4px 4px 0px 0px; padding: 6px 45px 4px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S13 { color: rgb(64, 64, 64); padding: 10px 0px 6px 17px; background: rgb(255, 255, 255) none repeat scroll 0% 0% / auto padding-box border-box; font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px; overflow-x: hidden; line-height: 17.234px;  }
/* Styling that is common to warnings and errors is in diagnosticOutput.css */.embeddedOutputsErrorElement {    min-height: 18px;    max-height: 250px;    overflow: auto;}
.embeddedOutputsErrorElement.inlineElement {}
.embeddedOutputsErrorElement.rightPaneElement {}
/* Styling that is common to warnings and errors is in diagnosticOutput.css */.embeddedOutputsWarningElement{    min-height: 18px;    max-height: 250px;    overflow: auto;}
.embeddedOutputsWarningElement.inlineElement {}
.embeddedOutputsWarningElement.rightPaneElement {}
/* Copyright 2015-2019 The MathWorks, Inc. *//* In this file, styles are not scoped to rtcContainer since they could be in the Dojo Tooltip */.diagnosticMessage-wrapper {    font-family: Menlo, Monaco, Consolas, "Courier New", monospace;    font-size: 12px;}
.diagnosticMessage-wrapper.diagnosticMessage-warningType {    color: rgb(255,100,0);}
.diagnosticMessage-wrapper.diagnosticMessage-warningType a {    color: rgb(255,100,0);    text-decoration: underline;}
.diagnosticMessage-wrapper.diagnosticMessage-errorType {    color: rgb(230,0,0);}
.diagnosticMessage-wrapper.diagnosticMessage-errorType a {    color: rgb(230,0,0);    text-decoration: underline;}
.diagnosticMessage-wrapper .diagnosticMessage-messagePart,.diagnosticMessage-wrapper .diagnosticMessage-causePart {    white-space: normal;}
.diagnosticMessage-wrapper .diagnosticMessage-stackPart {    white-space: normal;}
.embeddedOutputsTextElement,.embeddedOutputsVariableStringElement {    white-space: normal;    word-wrap:  initial;    min-height: 18px;    max-height: 250px;    overflow: auto;}
.textElement,.rtcDataTipElement .textElement {    padding-top: 3px;}
.embeddedOutputsTextElement.inlineElement,.embeddedOutputsVariableStringElement.inlineElement {}
.inlineElement .textElement {}
.embeddedOutputsTextElement.rightPaneElement,.embeddedOutputsVariableStringElement.rightPaneElement {    min-height: 16px;}
.rightPaneElement .textElement {    padding-top: 2px;    padding-left: 9px;}
.S14 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 1px solid rgb(233, 233, 233); border-bottom: 0px none rgb(0, 0, 0); border-radius: 0px; padding: 6px 45px 0px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }
.S15 { border-left: 1px solid rgb(233, 233, 233); border-right: 1px solid rgb(233, 233, 233); border-top: 0px none rgb(0, 0, 0); border-bottom: 1px solid rgb(233, 233, 233); border-radius: 0px; padding: 0px 45px 4px 13px; line-height: 17.234px; min-height: 18px; white-space: nowrap; color: rgb(0, 0, 0); font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 14px;  }</style></head><body><div class = rtcContent><h1  class = 'S0'><span>Flux Balance Analysis (FBA) and its variants</span></h1><h2  class = 'S1'><span>Author(s): </span><span style=' font-weight: bold;'>Ronan M.T. Fleming,</span><span> Leiden University;</span><span style=' font-weight: bold;'>Vanja Vlasov, LCSB, University of Luxembourg; Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg</span><span> </span></h2><h2  class = 'S1'><span>Reviewer(s): Ines Thiele,</span><span style=' font-weight: bold;'> Catherine Clancy, National University of Ireland, Galway. Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg.</span><span> </span></h2><h2  class = 'S1'><span>INTRODUCTION</span></h2><div  class = 'S2'><span>Flux balance analysis (FBA) evaluates the metabolic flux distribution</span><span texencoding="$^1" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAAAhUlEQVRIS+3VMQrCQBRF0ZNK3IrWgthkCW4gpJKsws3YW9q7gTRZTUgTQkAbyRB0Omd+/+7A5f0/hcgpIvMSAJxxxSHkKuTghBoNBmy/BcyBHi32vwDeDz5xzIDsILYHHXbYYFyqc2gXSlxQvUI3PHD/hCRwUFZvbnYghY8l92DVwF/0YALu1BMnstNhqgAAAABJRU5ErkJggg==" width="8" height="19" /></span><span>, and is one of the most used modelling approaches for metabolic systems. </span></div><div  class = 'S2'><span>The applications of FBA for molecular systems biology include prediction of the growth rates, uptake rates, knockout lethality and product secretion. In FBA, the solution space is constrained by the assumption of a steady-state, under which each internal metabolite is consumed at the same rate as it is produced.</span></div><div  class = 'S2'><span>For the quantitative estimation of the metabolic fluxes, linear programming (LP) can be used to solve the stoichiometric matrix for a given objective function under different constraints. The constraints of the problem depict the space of all eligible possibilities from which an optimal solution can be selected; </span></div><div  class = 'S3'><span texencoding="\begin{array}{ll}
\min\limits _{v} &amp; \ c^{T}v\\
\text{s.t.} &amp; Sv=b,\\
 &amp; l\leq v\leq u,
\end{array}" style="vertical-align:-30px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMMAAACQCAYAAAC1Zd/zAAAWRUlEQVR4Xu2dB7gtVXXHf3as2Bs2NAqWREUsMcTYNRg7iSWKHWzBSOw9UVHsKEpQErsRDYoCtohYUFQsYMNeo8ZubLGb7/dY22/eMOe+mTP3nNnn3LW+733vvXun7L32+u+92qx1JpKSA8mBbRw4U/IhOZAcOJ0DCYaUhORAcCDBkKKQHEgwpAwkB7bnQJ4MKRHJgTwZqpeBywJfBX4DfBL4MXAN4ILAh4BfANcFzgU8FXhc9TOqfIB5MtS7QPcFbgrcD/hpODu+A1wYuATgv88BvBN4PPDueqeyGiNLMNS7Ts8IIf9VDPGawMeAjwN7NIbtifBMoFxX74wqH1mCod4FOm+cCGWEjwSeDjwNeExj2BcH/qfeaazOyKYEg+/+w+qwavKRHg/cGPgr4L2Tj2YNBzAFGHYOPfciwB2Bj64hXzd7ShrJPwR+DVwojOrNfseWf94UYLgDcFRwXr3Y4z9pYw7cEngr8EZA/iUtgANTgMGT4dhwEd4VOHUB81q3Rz4HeChwf+DwdZtcLfOZAgy1zH2VxvEp4KrArhF7WKWxr8xYEwzTLZXBs+sBfwZcGvgs8HLgJ60h7QL8d/z+yjsY7pmBPwGuDnit3iftjCZdG7gd8JIE1vaMSTAsHwxnBw6IP88C3gF8FzgBOB+gsH6/Max7Ai8FXhD3zBrx+YEHAzcAbhYX+e/3tW7Q9tAGMUh3o+VPv943doHhrMANgX2AmwOXj0XSt/13gDvVN8Lf/W8xtfMABwJ3Ai4HfAF4Yew+s2bvbuifD7QuMMLqzvW3kYbwsNhBjbLuBfwobA5//st6Wds5Mnmn8AsIBdZ0C0meC4gLAPu1+KbQ6k69B/CKHvN1Tb8FGH/Qznhe6x6BcFx4p/ToJQUH2mBwJ1HoXSiPXIXtWiF8Ht+nhGDuFvffOoDxOuB7wOeBvwCuFL8XPK9vcFvmK+R3iesEk+kGhR4OmIZQ7j80gPVk4GTA+1UrpP+MZ63KYiqcJwKXDB5+ojFw0yoMnJ07fmekWaG9d2OOnhyqUa8CfreDSctX7/Xau3dc+5pI5dC1nTQDDIUxMn1f4PfAB4F/DGH090ZGNeguE3qsC6O7TyBI5wQ+AlyldRTrH1fYzavx2dIRLTD4MwWi5OJ4AgnAuzV06bLQju2KwJdXZDXdjfcGDgIe2zFmbQhPjM2IJstf11BQuZm1yQCeqpdgSdoBGEro38s8BYqgF8YVV5//N3Pywy2OPgH45wDT2eLvcslOkXHpqdQFBq/7Zuyg7oY3aUWq92wAU1WuxCxqXlRPyCNj3qqGBtAWSZ7w7wF+FptX811uIP8BXKe1Loscz0o8e5YB/SBAFUXyeDdDskkagIfED7QRvtb6vUd0sScuGipU8xKf589ngUGbQ6/Ia0Olat7rfWU8XTpxjYw3feIvgWOA2yxhgK7JV+I9qmXfbrzT0+DfgXctYRwr9Yp5wXCfEGQn2wUGDWkFWVJdOq3FFRdHkM0DBoN25vZLjwIOrpzjzrMI40OA5y9hvBrk2ntniQxXM10l7RBVNTezpBYHlgGGqwGf3sJgKCqLLLgt8OYlSWFRNVUzPQW02bQjNJoz3btjERIMi5fMZi7WMjNO3YA8lbWrTH95Q7ha2/bf4jmwIm9IMCx+ocpHOb7pzmFIt99qfEGvj1+tbRadFG5a85k8nbQT9CIlzeBAgmHxoqGrWRexaorxGO2pJhnUVFB1Omymm/jtETQ1mGdA9NWLn+pqv2EqMOhLv1h4nAyytWkjb5JpB0ahpVUwoB1ncTj4MZMGv7q7cZJbBDj277CrxkqWLmdVtMx07cnJWWDQZWksQTJI1g4EuXj/Gr/fHfhc630GyV4ZPzNprBlt9cd6g/QKeU0JwDUf8fVI1dA3r2rRpJK45s8MXhnEWgXSkyN41eM9LTwF9PebtmKQcbPJNAy9SAIvqQcHZoHhRcAD4n4/Pi+uufJIS5OU73BvDxzdeteTgCfGz/xmwUUvpN/bLEzfbYqFwZ8mCRKPdqOx6r3Xb/3+VmEQ+uNZrtkeU89LkgPbc6ANBr0dCq+ZkgqjZPDmLSH8V4gj36Qxk/MkBdfQ/otDwB8Ru72RZkkfuwlmejMEjn9KbpO/V7d9E3BYpF14KjWrPwgIszu9X5AZEDSZT/pt7Hx+JP+lXNzkwBgOZAr3GO7lvWvFgQTDWi1nTmYMBxIMY7iX964VBxIMa7WcOZkxHEgwjOFe3rtWHEgwrNVy5mTGcCDBMIZ7ee9acSDBsFbLmZMZw4EEwxju5b1rxYEEw1otZ05mDAcSDGO4l/euFQcSDGu1nDmZMRxIMIzhXt67VhxYJBisfGfxMZv0meVqDSTJlG2r4fnhicXGkpIDVXBgkWBwgpZJse6ndVL9BkKylKIfqz83PmypghE5iOTAosEghy1x6GeczW9//djHbxPaX8jliiQHJuPAMsBg6yU/xrGinGTxYPsG+F1wUnKgGg4sAwx+K63dYLlISZXJ0pOqT0nJgWo4sAwwWIDY6tt2rLQcvcUF2g00qmFIDmTrcmAZYHhgGMp+X+3pYI2gpORAdRxYBhis7akr9ZmAxQKSkgNVcmAZYLCTj/0JjDlYRCspOVAlB5YBhionnoNKDrQ5kGBImUgOBAcSDCkKyYEEQ8pAcmB7DuTJkBKRHMiTIWUgOZAnQ8pAcqCTA6kmpWAkB1JNShlIDqSalDKQHEg1KWUgObARB9JmSPlIDqTNkDKQHEibIWUgOZA2Q8pAciBthpSB5EAPDqQB3YNJecnW4ECCYWusc86yBwcSDD2YlJdsDQ4kGLbGOi9ilpYJtdrJbsCV4u+DonriIt638GcmGMax+GzA/lEP6srAuYGvAf8FvCdqRT0KuNq411R59/WjZOitgV2BXwEXAP6vytH2GFSCoQeTZlxyfuBE4KrAK4FnAV8CLgX8NXAwcHbgKGCf+V9T/Z1WTHRDEPw3rH60GwxwmWDYD3jxKjOrNfY3ALcHjgP+pmNezvdwwJNBYKwrfRi4dpQNfcoqT3JZYLgrYJnJK64ysxpj3xn4ESD/7hu1Y9tTOy/wPWBv4F1rMu/2NOTDD4CzRN+Nk1d5nssAg/ryB4FvrxEYbgwcHwt/d+BVM4TAUvzumj9eZSHZYOyejJ6QAuKiwO9XeZ6LBsPVQ43YBfjiGoHBquIaydLRoS51ycHTgEevsoDsYOyHAfcHXgvcZdXnOQsMekUOBPYCfhYdeNzdPgm8BXhvj4nrdnt/9GY4c5SW1OMgfbNRor7Ho6q75HxRTfycMbLHAAr+RqRKYWuva0aTlre3LlbVeFicoK+obsbdA9JhcHng3sBL45JrAXuEd0k5+eqKzGWbztsmf/a2cAu6AxYBVh1wkezPNmSx7NJz5zU7GeSZHhQ9KYX8/ywHgXaDvLxnuB+1IW7SYvx1Q530x3pl9M7UTIJAMEh60FSRXgbcvDFoZefBwBE1T6SMrQsMNwqDz0LBh7QmcWT8Ti9JX1pXMDh/y+vfKxihMKgqvG4DxjwAeFHYEBdsFWI2ZqHqZen+fwAO7cFgffwbva/HI7Zdsi/w+r4Xx3WqR6pJnwoA6EK2CU3p1PRqYE/gl8AVVqE5TRcYDggQaBR6GjTJ3U2/ehskG/FxncGwE/DWhn/9N8Bt42ddPLlMBOX8nWpk2VnLtUZy7XOnOqVKuiPyXdosY8mT241uCBXX8vMi6Pa4AEZ5hrEWVWrJOItgqZq6wODAyy5hPwX7KhRSr9U9+tkBs1pnMMiG88SOfr3giRFY//2JGTz6cgjPbYBjWtfY987d1RSHmsv3Kwd6kLSDVPns3Hpsay67A6fFz+zfV32Tmi4weFyfCpheIBllfdAGi7sjXKw7GJy/0Wh1fHd0SSHQk+ZJ0SaFR1VUY/nZrV++PHZ6VY2a6c+BD8QAPVE8WdrU9LhpH1Ufa5nlTbpc+NE1kqTfhQ77WODnA1dpK4BBlqj2fBTQ0yTZoKVLD9fI1Amhsa3RXcidVHvCGEbt9MRoXew4tQucd5uMvOthU14uNofcLJ0HG8UZTLoy30a3WSGPeI93m5r3pXUCg8e9aswsMhr9kvjlC8OT0r7WSPwTgDeHfeHvdT2rZngCf6UvYye8Tpe5iXo2qrzBjHEoI1cBPO30olVPfYJuHoke5/4t2a3TXbDvCbFOYDAP5zobrOolGl4TdeSuXtf3i1OhKUjuooJgqBE7hTfJk0974awbGMa3DCeCniTV7ZWINXSBwQU8AfAUKOTO9Qzgn+IHtrC1aWEfKmDwebrYVpU8KRUCc45mbQTaW78IQZnlHpV3ukN1Sf5pnA765j0VhtIU3qTbhftU1dngbIlDlbELEp0HgqDtgBk6v6Ve3wUGBf63He5TrzVX/9IdgTd3Cz0oJwE/bc3AyKTH5K/D89JlVOpyFCgaobXmt5T5beQZMfr6kfAEGWnWEdGmW0RQ0yi8wFBg/FuerwJp1xgv+Uy42dtjfg7wUMA4w906JqQnyny1r0eyYzVzngUGAyoOuI36dwO6/3SvlpPDnUCfuAbghwIUzQk+FTBdQdLD4q5x8dAn9TDoWz8FMLXBVGdVhhpJQAvs74Sq5GI2ydPzneEp0veuQHRR8cS4KagqudO2N5Aa51/GZPKhanKXF8lo8wvCbhIwnh5tUoYMLKpCydOhquHCeDMLDBrOplwIivLlku7Ad4QXQQEvZCT1u5HG68/Um7UrCjXTDEzWKy5IheBboSaUwJEBJ0FVI+kZEaj6/83TMsjkXP43QK6HxQxVF1dvUXsjKXNSNXJDUE0y7ULVa1VINdC11pVcUnMc+0VCk7hZuIw1mrvI+0x9L2SKu9muVVAXGFxIvQDu3heKxbpwqDgegW/qGPmTw1j0i69HdvxeVcD0Dnd/AaWP/RtxnV+DaX8oJLpuX1MFZ844CPOQ3PVMRXdj8IMedzjBr/ro6agnyfltRJeNj37ks6fMqpG2gPaN6+VG6fqp+uhGNmthR+nq8ki3c3FBa4OZDDo59fEmTT7IHMBackD7ypQON4cqKMFQxTJsyUFoU2lntaPwkzEjwTAZ67f0i+8UGb560bq8i5MwJ8EwCdu39Eu1tcxb8gvAqsrKJBi2tFxOMnllrsqM3ATDJPKQL62RAwmGGlclxzQJBxIMk7A9X1ojBxIMNa5KjmkSDiQYJmF7vrRGDiQYalyVHNMkHEgwTML2fGmNHEgw1LgqOaZJOJBgmITt+dIaOZBgqHFVckyTcCDBMAnb86U1ciDBUOOq5Jgm4UCCYRK250tr5ECCocZVyTFNwoEEwyRsz5fWyIEEQ42rkmOahAMJhvnYbiU5K4hYIcI//tsatLPKw8z3lrxrqRxIMMzHbjv0WC3bci/WEvoYYLWHpBXmQIJh3OLZo8AKeTZ0sTbUMsiiWz+JinTLeN8i32FZ0XZlwkW+b8NnJxjmZ70F0awOdw7AqtPt7p3zP7n7zksCD4+eDpb3tFbrKpIyZ8Hkx8eJalXyKijBMP8ylM40FlS2xGbfEv1D32iRLasU2idD4Fm9z6LGzTKNQ585xfXWSLKSnlUTreNrgWnnZSnTKijBMP8ylNqr9jq2zORmk7u/tV1tMqldYo+Dg6K9bDW1hnpM2sLUfx/Fpy0yLQhsom5J0iG9AXu8atwlCYb5+WfjEgsNW3D4X+Z/zBnutJuqNYXsk2YNU0FgoWeL+a4SCKzBapVt52JbNCtyFxBYYLo6SjDMtyRWk/5+COtegG2dxpKqjyrEHWBbs3o7+RQQjOndoHpiCXnbAVg0+OnRK6M5XkFtVXSLAo/tsqMtpR2gQ2GXAIENazwJPj+WSVGS0l4eNpP0jxtRu/S94LOPnBXOrfzeixIMvdh0hotK9xqrR2svjNmxbYIiCKw0J9n3QhBY8n0MCHyWoLVngn3XLBcv+W/7QjTJXtY6AeydYIXxecgWwPZksNmNgqiAWlH9KZsEAsdkhyOdCNprkpXPS4fV5ph1dbu5DDq1EwzzLPvppekVMpt+32q+R2wTSns8FCG1QbogsKz/WBC0h+Q62wvDNgMW/LWZSpMEwnHAD6PXwpApFcD5XDcGQWDXHkFgY5NFkN1F94gWAQe0XiAIBIM06NROMMy3VLZwUuVwF7RnxRCyJ7I7lh2QJEGg4NjbYLNB0ByXXUr1SPkejfI2uYvrrbpjz8kIAvts2LvOXguCwGc7l96qSc93NS9T7bMHhH0dHKtl7ZtUNqrBp3aCYfhqNDt6XmNG37aNnqph/CRgt7jI1l+CY9Fxin3DCJ8VLT8+2nQp0H1IPmgH+Fy9XbqWFURLzGtPLYpK5yPrtRqAbL5LMNtMxmaUnnRF9ew1lgRDLzZtd5FN+1RlXAQXY54iuu5u7mraChq2kqAQJG8bPqRed6iW2XbLHdNdtUm6cTVybes7tMGkDS81lu2BvVOA4tCIHywCFKVvXJe9YNu1w2JiBwLP7cWZuCjBMIRbp19bupfavtY+A2PJXs6Cwt53ki5bQaFRu5mkh6U0XDea7Q5ayNPAvtU2nJyXtEdUGxVIjWlPChvDG1Szd9tmkS3P3Eg8hZr2ggAXIKUTUGmm2fu9CYberPrjhUaAzanZP5qbD39C9x16SDSoSwDv5ACFRvpmkMEvO2wau9D4/Hg8VON575ZgjXmffQAfEraEdoWgsF2uoLA54hhSXn2GPQbb9oIbiDGafeIawTno1E4wDFsa/fXFQ+K/NX43m/SAeFIopJJ9pV1odeCxZD6Tp4JGvKeAgmswT8Ha7PRzjWobIeplshuozeIFhUmN84JCYbdLatte0IOku1unhK5hg3tmFg+iBMMgdm07Dez66emg2rFIMiVcULjIrpPuREFx7IiXfjq+vXD39Dl6YhTWzQiGzRrWuYD9Ij4gEAWFer2gGNrtVG+YXrGmvbBztBs25UOwa0Qb9DtiKJ8SDMM4pp1gHzL16/tE21djAwaCFkXuhjaV1z5RxREUfkthGvdQOima1qvXa1A7D71IyyCF9F6RnOdGIihsh2zUuy+5EbkhNe0FTwH/r/FuU3pJh4DBS92+hzQerupkLMScqDM4ChIMfZfh9Ots9m50tdgLHvsGsBa5s5YRNhP3dp0zhVv3rVFc1RS9LQbHlk0lcc+cJSPhQ1K4S2TZj6r06Jla4vcQGuraKa6Fp8Ol4hQ9GjglJui8PQ11A/sdijbadj3lEgz9RUFeuRvrKXl++LLfCPhnmaTxrsvSnXUoHRW5T54Mhw+9eZOvLzlTQzYSjXHVLgXcAOWJoeY5NIFhSrhgOCFOCe2hQgJFwBTSnnEz+yMlGIatsNFV/dwa0aZXL0vFGDbK2VcrEHqRmkKyWc9exnP0SKlq6SJ2Ls0A4Z6AaqyJggd3pJzomPAEKblMPms79TbBsIwlzHfUxoFjwnlgzChPhglX51TAj1zGkG7dqT/71JFgZu0Yellkuo55xtB7rWyiraIbdruvE/NkGMrK8defBuw+8jEaiFODQe+WnpwxpPtziAE95l3eaxDwyHDrFs9TngxjuZr3rxwH3PjNzNVobn/PsW0yeTKs3JrmgEdwQHmfmaKRYBjB2bx1vTiQYFiv9czZjOBAgmEE8/LW9eJAgmG91jNnM4ID/w85nvu+ZQ82OwAAAABJRU5ErkJggg==" width="97.5" height="72" /></span></div><div  class = 'S2'><span>Equation 1: Formula of standard FBA.</span></div><div  class = 'S2'><span>where </span><span texencoding="$c\in\Re^{n}$" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAF0AAAAmCAYAAACmlJfBAAAF4klEQVRoQ+2ZBag0ZRSGn9/u7u5uBVts7MJusbBRMMAW299uDMQWuxsV7O7GVuwObHn0zP0/x9llZnfvnev958Dl7u58Ne+c8573nBlGYwOOwLAB37HZkAb0GpygAb0BvQYEatiy8fQG9BoQqGHLxtOLQR8fGAf4pD+eycgC+lzAXsD8wAwB5mPAecBzOWCXBC4HZgK+BRx3GPBwrx7AyAD6ocAhwI3A7sAYwEnA+sAfwMnA/sCf8f8IYLQCgA8Eju4F8EMd9E2BKwKoKYDP4vOYwMvALPH9SsDrYwOnAa8AevyxwIQxxge0HPBQDngjZxdgYWACYClgMeBMYDpgC+D+dM5QB/1eYMW44fGAH5Kb3xc4PvkupWwH/JL8thJwT/L9BGC/Am+fFXgTOBG4HlgBuBl4NqLDKOmzoQ66nGxS1PTGc5J7XwR4Kvl+AHBcAaBvB7976Rpgo4IxRsAD4dULBk35X9C3BC4bLKD7wOXVDQEBmDr4th1tHh4hX5Za3wFmjMG/AdsDF8f3cYHvoK8VIsVsVrDw68Ds8XsrTzcP7AZcBBwE/BR7nR/39fFgAH0+4BJgobLoxThv6KgKc7YJILIpJsudAMHQPgCmjc+vAaqc1OToz4HR40e5Oo2ObOwjcS/Lh9rxd+lqnqJ7LEsvZvwlgAXiYD45Q8bQq2rLALdF2Mux8qDcq3T7pupiJcbvE3tkQ/X42YB3gZcCGK99CkyZW08uPjJ+uzC8N7+lifYL4BZgvbgormKk56uM/mVlQF8XMKxc1BB8FRgObA2YaPLZvB0OUog8p1KQA+U7va2/Tb4+Jtlk81A1T4TS8JIPQW2emdx9acjHc4E9gN8LDirQJk9xuimu65w60SrAW/FAv8/mtgPdkNKb1wbWCm/M5pmZ/c0QUhKVNQ9veCvXlFb94dmtzvJMEurLAg8GVZhPNGXivOFIcv/GwCg5MIvWVhpKY5MHlztm15CMcwKrAmekE1uB7mY3BOA7JhyYzb0dWA3YEzi9JOImLvlxLGBx4PGS83o1TL2ubjcxCq4048OfOzYw+f0YXqkElFK1D+O8/i8yc8ELIQiy62JzFXAdsEM+QlqBnmlYS9+lC3byQKqCNyogsnpw+fOAcmogTTp7MTbUywVKM/Rnjs/XAjsHP68M3J0c0MSr8/XEikCfPsC0ahOoO3qy0z83pE6WF3/tcM2qktFtlKWW/U9H2KdNrI9C0jlOurw1OZfeq8rS1PsTRaugw6OPmFYEuprzYOCr4Kmi5NHJxhYnZ3UyMZlTVTI6dY2gETV73r4EJg69Lif/nAwwKZrPMpOGFBFdWxHoLmwCMCNv0PUOIxZwLUNYeWj4DgaTx80xJjrVSWp2IOXjzEyWWWHV1dnzoI8aGVjlYvdNjdsrs/nzftDLVJFUe7V2J+tYFClXpTp1+3u5RdTnac/ERpjt4a4tD/okkUhc2JaoVNNLk1ftxtmflp8HyqQQo+vqZENpRw7Xo5Wxeds7ckH2+31J86yrcxfRi9rZ8tekJw/nTTkpVdj8qWr2NtT2yjVVxKNVF+hwvNWkHcTshYTiQM/dJOSjUjZvJlbrkcysOq1YzXG+DFH5pF3L0kcrAt0SXdVi6KlX01anr7DkNf+y6qv0ZlFs3Blep3L4T6+5ymIlx+bbs06z+rQPbtVpnikyHc9EK+VmZiVufWFkWE139DqvCHQLhycjwVj6SwN6gs0ei6FT2xy0DA56i6GqGvDFwNnhUXrh12UWqDBGJ1H6+XLCFkRqF+QSZdGyOtdWuQtSkkVWX1lf4Tx/D21VHM0RvYpFI6T0Sh/AKR02ufLnmhQw5NepeOCqktHc4csLey/Si6/tTOh3AWsGzbU7gsnWXtFkISdNrAoMnaVjK9Pw6njxEhN9w2LjzKp3GsCCrJ2Z3Ku8p7T1YGlvS1dTHm4bPSV76WXMJpi9FFvRRk3XVjfoXd/A/3GBBvQanloDegN6DQjUsGXj6Q3oNSBQw5aNpzeg14BADVv+BfN/HjbxU86bAAAAAElFTkSuQmCC" width="46.5" height="19" /></span><span> is a parameter vector that linearly combines one or more reaction fluxes to form what is termed the objective function,  and where a </span><span texencoding="$b_{i}&lt;0$" style="vertical-align:-6px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAE8AAAAoCAYAAAC/xadkAAAFKklEQVRoQ+2ZZcgtVRSGn2t3NwZXxe7CLrBFxB+K3Y0NdnfXFQNbfyioiIoFdiC2YCsqdjcWJs91zcfccyfPzLl8HGbB9+M7s/fae7+z9rvetWYMnfWNwJi+Z3YT6cBrEAQdeB14DRBoMLWLvAGBNzOwWPwtDiwI7NZgraGbWhR5OwFrAbsC0wJPAOsNHQL/H2hj4EBgfhifRKcEbgcuBn7MO3OVa/sWYOSdApw8ZOBNBlwawJ0HHAP8DSwP3Av8BmwOvJt17jLwZgjkXWQd4KkhA89gOAl4AVg9gEuO6Hm9be8Aq2VFYBl4WwL3AD8BswN/DRF48vkbwOTARsBDGWd7NoA7Fziq93kZeJcABwN3AtuMUuBmBA6Il3tkjT1eDewF/AHMAvyeMfcs4Oi4vvNEEI0MKwPvTWAJYD/gqhobmxRDPfBBwKHAbMD1wB4VFzbavgDmAB4DNsiZtwnwQDzbBbg5Pa4IvAWAj2LwwsAHEeIbhnxx8YeBHypuuK1hHljABG6mWN+sWJgZexZfGngtfjNhHJKzubkDZB/fAOxeFTzf4rWRaeSHNYHrIvMmPgRwK+D5tpAp8OO1OQLYH5ge+D4Ak1py5USFiCpSEVOnrvMrwIpVwbsV2A64DLgNkDRPAJ4BVgHuiOvyKrDcAMFTe8llewPTBGgXAYJmIuvH9gSuiYmHxUvI8/Nr6NxvgDmrgKc0+SpIeN9IFtv2bPacOJT+vErf9nOKgjljg6ytaqYCvgMuBMY1AC1ZziA4Nf7xKnol8+wzYN5QGornEcvjvJVD+/wJPBhX5ZMe7yaRK+K3RYD3WwJPijgW2BGYIl5KAtrPLa2hGD4zfFlB3VTg98MoTc3GVlql4KWdyzFXZjg/HTgO+Ceuk0A3sWXCnxFu5BvJFwRttAVasj+jTf7WlGJGc55JDcqh94BFq4D3KLA+8HUK9V7npnBTuWPNwE3MaNPPfOHEDSvQn2zitGDuZsB98fxE4LScsUZ+EhTuZd0y8Mxk8os8o1Od95oc8HFIlyzO8EofHuq8qpQxGZjhVfJ2cLS7gpteahnEtFQx+bjXLJPLDSDtFmCHMvAshC2KRXwh4PMMr2fHIV8HVsgo2wTAq+8mP615cEl555ifXJO7A8QXa/rKGy7XfxnZ0xtkJGZZUt/6bKJCISthKDYVjTYBnNxrJgdBk5dU5k/nLKyKt0PRrzlfqWTy8CVo1tlmSQv5piaPqyR+AWZNXc+036RxYE1vIJl5RywLPIvlJYHLo1WTHq9otJyxA7EPYH04aHOPWwPHAyvFYt4MxW0Tcb4UoEY1CNbOCQJ5zmdmY7PyBNYLnoJULtN6s6wcaP1oNvSNJdkq7dDwl1RNADYT5Yk2Tf+CaLWjSfqC+FyfiyRa1SLAc6VtVcCuipxnf89qqhA8JyQbkbwFS1sWuDF0l8BZZWSZL0N5oWq3lLGkGYRJF4KYZPn7Q+a8XHMx+dVKw6Jfnj4/pJcvXy6cLsrPzJeTdW03DYGqwrdmtHthaWLomv3Kenq2r9YIVf5vzcPUHe46as0tsgr3Gs6kBdvwCfcJmlFtCThRxCV+y1pSNdYfPzSpCARwUn4sMsoV2RO0jOpuvu74tsGT7x4HtgdsLAy1tQ3eGVHMzzWARsGoexFtg/cI4EcjP5j4xe3tUXfiFjfUJnj6shSzvWOWVTz75WlorU3wBElRqVa0cZn1NWqogGwbvKECp+wwHXhlCBU878DrwGuAQIOpXeR14DVAoMHULvIagPcftJ73KXkJ7JYAAAAASUVORK5CYII=" width="39.5" height="20" /></span><span>, or  </span><span texencoding="$b_{i}&gt;0$" style="vertical-align:-6px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAE8AAAAoCAYAAAC/xadkAAAFMElEQVRoQ+2ZdcgmVRTGf2sndoGB7mIndgeo6yIiJgZ2IigKFmJ3J+quraCCiqgoCnZi/CF2oGK3qNjNT898zM5Ofe/Mu6zDHPj4eGfm3jv3uec85zlnxtDbwAiMGXhkP5AevAZO0IPXg9cAgQZDe88bEnhzAUvH3zLA4sBeDdbq3NAyz9sdWB/YE5gVeBzYuHMI/LehLYBDgEXh3yQ6I3A7cBHwXdGe64TtG4CedzJwUsfAmw64JIA7FzgW+BNYBbgX+BmYALydt+8q8OYI5F1kQ+DJjoGnM5wIvACsE8AlW3S/RttbwFp5HlgF3tbAPcD3wHzAHx0CTz5/DZge2Bx4MGdvzwZw5wBHZ+9XgXcxcChwJ7DdEIHbBvDvTOCdIa6TnvoqYD/gV2Bu4JecdX2fYyJ8Fw4nGnmsCrzXgWWBg4CJQ9yUB3NHhM3NAaJrD8v0ts+A+YFHgU0LFtoSuD/u7QHclH6uDLzFgA/i4aWA98LFNwv54uIPAd+2sEM5dWfghDisvyLbnQ681ML82SlWAF6JiyaMwwrWWChA9vb1wN51wdsHuCYyjfywHnBtZN5kDgE03J5vaYOCuFOAuBzwN3A3IIhtreGrpj2qTEXMnArnF4HV6oJ3a3jDZcBtgKR5PPAMsEaE2bzAy8DKLYGXTCOIOwaIy8dFw+c04KkW1toXuDrmOTz0XNG0P4XO/QpYoA54vvwXkWEPjGShR5h1EzsbOCp+yB1ft7Cp7BQJiB6aoaY9Apwa/wdd0vlOicGGoiFZZJ8Ai4TSUDyPWBHnrR7a53fgAeBg4KPM7CaRK+LaWODdQXdSY5wg7hCev2I8/3SAmBB6jWlGHlEMnxG/rKBuLBn8fpSmZmMrrUrw0pML3JU5kxtCxwGS+yyAQA/bPGxBNLEkICpwfRe5UY6sY3qb/K0pxS4tGWS0zRkSalwd8AyNTYAvU6hn5/fEJV6fNQNPTVNz3ZcicLXaARUelH6/rWK81zwIaSDPZkg5xRPARlXgzQ58A8wUkzp51uSAD0O65HGGIX1EqPM2pEyyvtLhyKCR2QDJfBJgXSo31bW0VLkw3jVvrFyuA2m3ALtWgWchbFFsGC4BfJoz61lRrrwKrJpTtlnKGPq+5Md1d1TynKCZnDwUQTOULgcuSG1uNMsY/p9H9jSC9MQ8S+pb701RKOQlDNswikabAA7OmslB0CRxlXmRdFDF26FoYoZnAppkbURYMipsm3q0PK6S+BGYp4Czk8aBNb2ONJl354FnsaxA9WTtcaVN0Wg5YwdCjrE+HIZlQVM2nR/v9ENLC6of1ag6wQYFTiDPec9sbFaezLLg2QyUy7RslpUDr4sKwBNLslV6Qt1fUrUisZkoT4zG5NLE08zgyiP5zEOyt9a2JVrVIkAdm7Y1Absqcp79PaupUvAc8Fw8YXkmWNpKwA2A2UfgrDLyzMPQQ1TtljKWNHXMdpfC1bkFTc3oxhSvv9WZYMBnFL1WGhb98vR5Ib08fLlQfrX8TDApBc+b44HdgCWjAWi7xtJE172rRk/P9tW6ocrr6q7tw1PtWite9dip2TvcNigq4T5BUwrJr1N4XIJgVUtqtAeqZ1qmCeBoPhatHXrStpSi+39hbYMn3z0G7ALYWOi0tQ2erSM7rwsOqVEwTR1G2+A9DPjRyA8mfnF7c5rabcsv0yZ4zqVwNUOaZRXPfnnqrLUJniApKtWK+xd8jeoUkG2D1ylwqjbTg1eFUMn9HrwevAYINBjae14PXgMEGgztPa8BeP8AkyUAOPI3aRkAAAAASUVORK5CYII=" width="39.5" height="20" /></span><span>, represents some fixed output, or input, of the ith molecular species. </span><span texencoding="$S\in\Re^{m\times n}$" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAH0AAAAmCAYAAADpyZQRAAAIC0lEQVR4Xu2aA6ysVxRG16ttK3VTu03dpm1q2zZTpLZt27Zt20Zqpk5TN+tln5vz/gz+mXvn4s3s5OVl5h5ufPvb+8wgOtJ2GhjUdjfuXJiO0dvQCTpG7xi9DTXQhlfuRHrH6G2ogTa8cifS6xt9GGBy4FPg3/rD+/+IdjT6uMDOwALAdMCvwBvAZcAdBZNNAlwOLAH8AbwNHAdcCfzXAvOOEnsVz1Hcak3ghmbPUNboy4aiZgImAr4GHgUeBL4CTgCWBz5ugSJ6cknPeDXwObAe8BmwHXBobHIPsBHwLbAicBEwXoUD3A2s0ILI3zj23AK4uMrFT44zzwe80oxyyhjdCNgQeAbYF3gBGA2YHzgdmBj4vopymjlTq+ZMDzwHjAGsBVyfbXQFsH58fjUcejXg8Lj3lMARwIzZnH2AowuHNVJFEXUzGzBX6OocYF5gS+C2+NspwFSAASWCJDkQOAjYHLiksL5ztgXWAG7P/rYBsDQwZ5zz5nBk13gYWDtfp57RvYAbfQEY5T8VDjED8BZghCzXKmv10LoHZBFtlN6ZratBns0+vwYsA3yZfWdaeB8YK77TgYy2ogwbCPJBoMUhwJmAax4bKUSnewy4EdgKOL+wiEbX+JsBl8bfTgW2AYR2HScX97wA0PgG4d7ANcBOEbAjR3oaPKee0V8M7zGvCXuV5AnggThkD9mnJcuoKCFbEeKF91x+BkaPL4TvSk4s3G8aY74Dxq9wUr/7JqL19wgaI9poXieifH9gXeCq0O/LFdbRWRxntM4dEa7Bb62iHVOtPMPIFsUMxpuAWYK7dE2rZfQRgV+A4cOLhKZKopd6kGqHKWNBzyGceikhUW8doc5ElVKE11pTzJGbZAPOBnbI8rKRbsQrcha5S1HOjcj0+2qR7j2MYBFS0id/MJpPi/MeCfwJuP/qwIQ1CNlhYfi/AqKF7UpiJP8QCKDBjXLlk0AWeUspo2tsGasGEd6nBfTcopgCritAYRlDpzF6orxhjkYmhTLMs2XFfZ7Motl5OoKkyVJMNBMek+h4ktRc5DUJ0ncHTqyw+UnAroEIKSe7tpFt3n0o5rwDPF/Ys7ickK5jKjnUF8eZikQnnUKnU+Qw7uHnIZylHrw/DiwUi5gDVwH+LqvlEuMWjtwqrP4WVYCpQlZa5A8llqs7xP1UzqjZSA1tCXYGsH32vRxGiEyiHtSHItlbMM5c3FRDyvgNkqQra/x3gSVjsA5lIGnIWizd80jChHdJdCVy55LHAzqhKPlS7KGzyPTlIqauLqlndKPP6BA+FFmuub0nalQvbi6bAHgkCIclVKvFmlsHHik2ErIlSELxHtnm5uFUgs4cZHXSOKuGMG8XZZwo9yRssntlakBSJ+uWxSvyCR1tsuAFOnmuU9FCo7lPilKhvprh1aNopdGTmGLUrY4+e17e1TO6C1g75qWDZZqssLuiArYG3oyoaUVkVztjgmD/Lqu3NLPXsFs2wVyr0owu/40JOC8fU1xfoiZJNAWY8xXzuY41RXT1/M4I1KBWETqFjZYknkP9yvBvKWxgP2G/SEkJITyjaUjGr2Mk0SlNK0a+aUmkGSxljO64o6IMSPPMpTLLZkV4lf0abda0ebnU7JqNzDOyJVJyFGtbc59lVU54JGCKhDKxdAnYSsC9VTaT3S8VBk6Ra8RNE9GWpokEpspjgAuztXSus8Lg1YhxYvXzhEETaswKvJ6tJXewv2DQWh52SVmjO8EIT6TCz0bpeY1oOhtrOSTEmhuFnt4Ue+n3A4sEibo2Nlf55ljlo8i//i+LV5nmRsXSS1L4TwsObcWkMS2Da4ncQO7TlDRidMfqyansMUo02HtN7JwiTcVZjjQjjZZs7mGXzIgWaSxB7S4mMcem2l1itGf2N8utHbPPGiaf28z5+2xOI0b3kHriUwGJflaBefSXvYgw6tzuiOmlkZLNvWTBw0WOK+4tDKdyR6YugU1iHjfXJvH8pocBKZWMboPEnDUEzc9u58tUIgW+Tgl1jYpNCcmLEGUO7A9yV/TBbZeKYDmbtqyTFCUR8cy/A1IqGd1uzl51XswkPhb/HwZJafTylirWrsK7OVNS19dimrK2tomSOlrpTDrmfdkB+4KL9Jh+KhndmtGGvZ5fTezvLhbRKjI0I6mvfzBgfu5NkdHaRUwdRl8NRTYRzKZM8ccSMnzPm0QWLy/oyUZVr92/aHSZrT8qsH+rYiqJOdEacGxgF8BWYTOSSg0VJ5N+uplFmpiT+g42gqxrfcXynd0727ixUVQUa3Sfj33NSmLjygBRD9bKot+AkKLRressU8xnKsKWZVGMSp/9bDfaimyWfetgPskKnbYkzZsiSCtF49gMSuWXe/mDCd/BzdO1mk42W2TtSeQ9dvFERRs8EtwBIUWjp8a98KbhZaw+z+kIEjhbiRpHqLNjZE7vjtj1songjxPc08aET6B2kH7szsJV5totWzRe8fIhRqyM3f5/NZG4+Wadi+/rBkczZWsLrlduyaLRbRn6smbDwt+QeSEdwVahT3cyW3+xYQu1p5oTRp2NkZXLHblrVKMl2+IB57502Y/2idOnVAmld00duGrHEJn8iVh6gLI17QtjtSqnwev03vBG6/RWnsx8ar5Vqf4g0Z5ALbHXrOHKincVxvNoXjV+DZO3L2utJ+HT0OkHCmX37lfj+pPR+5VihubDdIw+NFu3yt06Ru8YvQ010IZX7kR6x+htqIE2vHIn0jtGb0MNtOGV/wfO/qI2IOAybgAAAABJRU5ErkJggg==" width="62.5" height="19" /></span><span> is a stoichiometric matrix for </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">m</span><span> molecular species and </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">n</span><span> reactions, and </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">b</span><span> is a vector of known metabolic exchanges. The output of FBA is a particular flux distribution, </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">v</span><span> , which maximises or minimises the objective function and stands between upper and lower bounds, </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">u</span><span> and </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">l</span><span>, respectively.</span></div><div  class = 'S2'><span>There are multiple different variants of FBA, some of which will be demonstrated here:</span></div><ol  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Standard FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Regularised FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Sparse FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Metabolite dilution FBA (mdFBA)</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Geometric FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Parsimonious enzyme usage Flux Balance Analysis (pFBA)</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Dynamic FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Relaxed FBA</span></li><li  class = 'S5'><span style=' font-weight: bold;'>Flux enrichment analysis (FEA)</span></li></ol><h2  class = 'S6'><span>EQUIPMENT SETUP</span></h2><h2  class = 'S1'><span>Initialise The Cobra Toolbox and set the solver.</span></h2><div  class = 'S2'><span>If necessary, initialise the cobra toolbox:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span > </span><span style="color: rgb(14, 0, 255);">if </span><span >0</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >     initCobraToolbox(false) </span><span style="color: rgb(2, 128, 9);">% false, as we don't want to update</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span > </span><span style="color: rgb(14, 0, 255);">end</span></span></div></div></div><div  class = 'S10'><span>For solving LP problems in a FBA analysis, certain solvers are required and can be set using the </span><span style=' font-family: monospace;'>changeCobraSolver</span><span> function:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% solverOK = changeCobraSolver(solverName, solverType, printLevel, unchecked)</span></span></div></div></div><div  class = 'S10'><span>The present tutoria</span><span>l can ru</span><span>n </span><span>with the GLPK package, which does not require additional installation and configuration. Although, for the analysis of large models is recommended to use the GUROBI package.</span></div><div  class = 'S2'><span>Setup the appropriate solver for the machine you are using by removing the "%" (comment) sign for only the desired solver.</span></div><div class="CodeBlock"><div class="inlineWrapper outputs"><div  class = 'S12'><span style="white-space: normal"><span > changeCobraSolver(</span><span style="color: rgb(170, 4, 249);">'glpk'</span><span >,</span><span style="color: rgb(170, 4, 249);">'all'</span><span >);</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement scrollableOutput" uid="0857C796" data-testid="output_0" data-width="420" data-height="101" data-hashorizontaloverflow="true" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"> &gt; The solver compatibility is not tested with MATLAB R2019b.
 &gt; Solver for LP problems has been set to glpk.
 &gt; The solver compatibility is not tested with MATLAB R2019b.
 &gt; Solver for MILP problems has been set to glpk.
 &gt; Solver glpk not supported for problems of type MIQP. No solver set for this problemtype 
 &gt; Solver glpk not supported for problems of type NLP. No solver set for this problemtype 
 &gt; Solver glpk not supported for problems of type QP. Currently used: qpng </div></div></div></div><div class="inlineWrapper"><div  class = 'S14'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% changeCobraSolver('tomlab_cplex','all');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% changeCobraASolver('ibm_cplex','all');</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% changeCobraSolver ('gurobi', 'all');</span></span></div></div></div><h2  class = 'S1'><span>Model Setup</span></h2><div  class = 'S2'><span>Thi</span><span>s tutorial will use the generic model of the human cellular metabolism</span><span texencoding="$ ^2" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAABF0lEQVRIS+3UsSuFURjH8c/NoJRkUsTIRMmoDAYZZDDJatLNJpudxSSUQWySDMqkJDb/gNVsMSilRKcO3d7e8557u9vtnu2cnud7nuf3/M6paXPV2szXwYAF7GAKb7iJ+4+iZmUarGAfV+jFMkZxj/kcoCcGhqT3GNyPZ0xgDo+NkGIFsxjGZeGmOg6wgeMqwABCn98FwFLUYRUXVYCUr9ZxhJEo6n9cs0YKLb1iKydiWQXjOI8CfrUKCFO5xjZeyui5FvbwgNuUOFWAzTiR06oXmwKsYQy7heRJDOHu77wMsIgTnOGnATAYrTyDzxRgGk/oS5R9iODKln2QlCE3heyP1wXo5G89O/+qt9B0cgjsGqmrQfDBL0GOJif7qwNnAAAAAElFTkSuQmCC" width="8" height="19" /></span><span>, Recon 2.0. Other COBRA models, including </span><span>Recon 3, may also be run with this tutorial. For information on metabolites structures and reactions, and to download the latest COBRA model releases, visit the Virtual Metabolic Human database (VMH, </span><a href = "http://vmh.life)."><span>http://vmh.life).</span></a></div><div  class = 'S2'><span>Before proceeding with the simulations, load the model into the workspace:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(14, 0, 255);">global </span><span >CBTDIR</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%modelFileName = 'Recon2.0model.mat';</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelFileName = </span><span style="color: rgb(170, 4, 249);">'Recon3DModel_301.mat'</span><span >;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelDirectory = getDistributedModelFolder(modelFileName); </span><span style="color: rgb(2, 128, 9);">%Look up the folder for the distributed Models.</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelFileName= [modelDirectory filesep modelFileName]; </span><span style="color: rgb(2, 128, 9);">% Get the full path. Necessary to be sure, that the right model is loaded</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(14, 0, 255);">if </span><span >0</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >    model = readCbModel(modelFileName);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(14, 0, 255);">else</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >    load(</span><span style="color: rgb(170, 4, 249);">'Recon3DModel_301.mat'</span><span >)</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >    model = Recon3DModel;</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(14, 0, 255);">end</span></span></div></div></div><div  class = 'S10'><span>In this tutorial we assume, that the cellular objectives include energy production or optimisation of uptake rates and by-product secretion for various physiological functions of the human body.</span></div><h2  class = 'S6'><span>PROCEDURE</span></h2><h2  class = 'S1'><span style=' text-decoration: underline;'>1. Standard FBA</span></h2><div  class = 'S2'><span>Standard FBA predicts an optimal solution for a cellular objective within a given set of constraints on a metabolic network (see Equation 1). Constraints on the network are set by assigning limits on the uptake, consumption or production of metabolites in reactions.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>The time to determine a FBA solution depends on the size of the genome-scale model and is commonly less than a second for a medium sized model.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Calculating maximal ATP energy production under aerobic conditions:</span></div><div  class = 'S2'><span>For each new simulation, the original model will be copied to a new variable. This preserves the constraints of the original model to perform further simulations with new constraints. Additionally, this method of renaming the model avoids confusion while performing multiple simulations at the same time.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span >modelaerobic = model;</span></span></div></div></div><div  class = 'S10'><span>The ATP demand reacti</span><span>on, i.e., </span><a href = "http://vmh.life/#human/all/DM_atp_c_"><span style=' font-family: monospace;'>DM_atp_c_</span></a><span style=' font-family: monospace;'> </span><span>within the model is a reaction that involves hydrolysis of ATP to ADP, Pi and proton in the cytosol. </span></div><div class="CodeBlock"><div class="inlineWrapper outputs"><div  class = 'S12'><span style="white-space: normal"><span > printRxnFormula(model, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement scrollableOutput" uid="E19D33B1" data-testid="output_1" data-width="420" data-height="18" data-hashorizontaloverflow="true" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">DM_atp_c_	h2o[c] + atp[c] 	-&gt;	h[c] + adp[c] + pi[c] </div></div></div></div></div><div  class = 'S10'><span>We will set this reaction</span><span> as our obj</span><span>ec</span><span>tive with the </span><span style=' font-family: monospace;'>'changeObjective'</span><span> command. Maximising the flux through the ATP demand reaction will result in the network producing a maximal amount of ATP (up to the limit of the reaction).</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span >modelaerobic = changeObjective (modelaerobic, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div></div><div  class = 'S10'><span>The glucose and oxygen, in this case, are provid</span><span>ed in high </span><span>am</span><span>ounts for</span><span> ca</span><span>lc</span><span>ulating the flux through ATP demand. </span></div><div  class = 'S2'><span>The </span><span style=' font-family: monospace;'>'changeRxnBounds' </span><span>function changes the flux constraints of the lower ('</span><span style=' font-family: monospace;'>l</span><span>'), upper ('</span><span style=' font-family: monospace;'>u</span><span>'), or both the bounds ('</span><span style=' font-family: monospace;'>b</span><span>'), of the specified reaction. Here, we will change the maximal uptake of glucose to 20 </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">μ</span><span>mol/min/gDW and of oxygen to 1000 </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">μ</span><span>mol/min/gDW. The uptake of oxygen is effectively unconstrainted (i.e. infinity). </span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelaerobic = changeRxnBounds (modelaerobic, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >, -20, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);  </span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span >modelaerobic = changeRxnBounds (modelaerobic, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, -1000, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >); </span></span></div></div></div><div  class = 'S10'><span>The function </span><span style=' font-family: monospace;'>optimizeCbModel</span><span> calculates one of the op</span><span>timal solu</span><span>tions for a (maximum or minimum) objective reaction wit</span><span>hin</span><span> t</span><span>he defined solution space. In the above example, the maximal flux through the </span><a href = "http://vmh.life/#human/all/DM_atp_c_"><span style=' font-family: monospace;'>DM_atp_c_</span></a><span> is desired. </span></div><div class="CodeBlock"><div class="inlineWrapper outputs"><div  class = 'S12'><span style="white-space: normal"><span >FBAaerobic = optimizeCbModel (modelaerobic, </span><span style="color: rgb(170, 4, 249);">'max'</span><span >)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableStringElement" uid="976F7EB0" data-testid="output_2" data-width="420" data-height="258" data-hashorizontaloverflow="false" style="width: 450px; max-height: 269px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span class="variableNameElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">FBAaerobic = <span class="headerElement" style="white-space: normal; font-style: italic; color: rgb(179, 179, 179); font-size: 12px;">struct with fields:</span></span></div><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">         full: [10600×1 double]
          obj: 1000
        rcost: [10600×1 double]
         dual: [5835×1 double]
        slack: [5835×1 double]
       solver: 'glpk'
    algorithm: 'default'
         stat: 1
     origStat: 5
         time: 0.7090
        basis: []
            f: 1000
            x: [10600×1 double]
            v: [10600×1 double]
            w: [10600×1 double]
            y: [5835×1 double]
            s: [5835×1 double]
</div></div></div></div></div></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Anticipated results</span></li></ul><div  class = 'S2'><span>When oxygen and all carbon sources (internal and external) are provided the flux through ATP demand reaction can reach its maximum rate of 1000 </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">μ</span><span>mol/min/gDW. </span></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Troubleshooting</span></li></ul><div  class = 'S2'><span>If there are multiple carbon sources available in the model, it may be necessary to specify more constraints in order to examine the effect of a single carbon source on ATP production.  </span></div><div  class = 'S2'><span>To avoid this issue, all external carbon sources need to be closed with the exception of the single carbon source of interest.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%Closing the uptake of all energy and oxygen sources</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >[exchBool,uptBool] = findExcRxns(model);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >uptakes = model.rxns(uptBool);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% If you use Recon3.0 model, then:</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelalter = model;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelalter = changeRxnBounds(modelalter, uptakes, 0, 'b');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelalter = changeRxnBounds(modelalter, 'EX_HC00250[e]', -1000, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% The alternative way to do that, in case you were using another large model, </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% that does not contain defined Subsystem is</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% to find uptake exchange reactions with following codes:</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% [selExc, selUpt] = findExcRxns(model);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% uptakes1 = model.rxns(selUpt);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% Selecting from the exchange uptake reactions those </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% which contain at least 1 carbon in the metabolites included in the reaction:</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span > subuptakeModel = extractSubNetwork(model, uptakes);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span > hiCarbonRxns = findCarbonRxns(subuptakeModel,1);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% Closing the uptake of all the carbon sources</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span > modelalter = model;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span > modelalter = changeRxnBounds(modelalter, hiCarbonRxns, 0, </span><span style="color: rgb(170, 4, 249);">'b'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% Closing other oxygen and energy sources. Use the following lines for recon2, or uncomment the lines below for recon3</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span > exoxygen = {</span><span style="color: rgb(170, 4, 249);">'EX_adp[e]'</span><span >,   </span><span style="color: rgb(170, 4, 249);">'EX_amp[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_atp[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_co2[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_coa[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_fad[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_fe2[e]'</span><span >,</span><span style="color: rgb(14, 0, 255);">...</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >    </span><span style="color: rgb(170, 4, 249);">'EX_fe3[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_gdp[e]'</span><span >,   </span><span style="color: rgb(170, 4, 249);">'EX_gmp[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_gtp[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_h[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_h2o[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_h2o2[e]'</span><span >,</span><span style="color: rgb(14, 0, 255);">...</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >    </span><span style="color: rgb(170, 4, 249);">'EX_nad[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_nadp[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_no[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_no2[e]'</span><span >,    </span><span style="color: rgb(170, 4, 249);">'EX_o2s[e]'</span><span >};</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span >modelalter = changeRxnBounds (modelalter, exoxygen, 0, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div></div><div  class = 'S10'><span style=' font-weight: bold;'>Calculating maximum ATP energy production under anaerobic and glucose only conditions:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modelanaerobic = modelalter;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelanaerobic = changeRxnBounds(modelanaerobic, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >,-20,</span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelanaerobic = changeRxnBounds (modelanaerobic, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, 0, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelanaerobic = changeObjective(modelanaerobic,</span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >FBAanaerob = optimizeCbModel(modelanaerobic,</span><span style="color: rgb(170, 4, 249);">'max'</span><span >)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsVariableStringElement" uid="5E730D0A" data-testid="output_3" data-width="420" data-height="258" data-hashorizontaloverflow="false" style="width: 450px; max-height: 269px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span class="variableNameElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">FBAanaerob = <span class="headerElement" style="white-space: normal; font-style: italic; color: rgb(179, 179, 179); font-size: 12px;">struct with fields:</span></span></div><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">         full: [10600×1 double]
          obj: 1.0000e+03
        rcost: [10600×1 double]
         dual: [5835×1 double]
        slack: [5835×1 double]
       solver: 'glpk'
    algorithm: 'default'
         stat: 1
     origStat: 5
         time: 0.6735
        basis: []
            f: 1.0000e+03
            x: [10600×1 double]
            v: [10600×1 double]
            w: [10600×1 double]
            y: [5835×1 double]
            s: [5835×1 double]
</div></div></div></div></div></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Anticipated results</span></li></ul><div  class = 'S2'><span>Compared to the aerobic condition, anaerobic condition with only glucose as an energy source has reduced flux through ATP demand (82 </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">μ</span><span>mol/min/gDW), signifying the need to oxygen to run the oxidative phosphorylation. The results are dependant on the model you are using. For Recon 3.0, under anaerobic conditions with only glucose as an energy source, the flux for ATP demand is 40 </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">μ</span><span>mol/min/gDW.</span></div><h2  class = 'S6'><span style=' text-decoration: underline;'>2. Quadratically regularised FBA</span></h2><div  class = 'S2'><span>Regularised FBA calculates the optimal solution of a linear objective function, and finds the unique optimal flux vector that minimises the Euclidean norm of the flux vector, that is</span></div><div  class = 'S3'><span texencoding="\begin{array}{ll}
\min\limits _{v} &amp; \ c^{T}v + \eta ||v|| \\
\text{s.t.} &amp; Sv=b,\\
 &amp; l\leq v\leq u,\\
\end{array}" style="vertical-align:-30px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOgAAACQCAYAAAAC2CfUAAAavklEQVR4Xu2dB7R1RXXHf4oFEVEEK6CoUVTsoqixExv2FtEoihWiWIhGxUYUUYwNpQgWYo2oKCrYG2jsFQto7MYC2DXYNevH2rPWyeHed8+5p7xz39t7rW993/fenDkz/zP/mT1775l9HlISgURgsgicZ7Ity4YlAokASdAcBInAhBFIgk7442TTEoEkaI6BRGDCCCRBJ/xxsmmtEbgl8DzghguebFqudQOArYH/Bq4NnLlGBY3KJUGX+QTjPHN54LvAn4AvA78ErgNcHPgUcDawO7AV8GzgqeM0a9JveSDwfOASC1rZtNwynS3f7RrAV9eooFG5JOgyn2CcZx4K/APwMOA3cI5B7wxge+Ay8e8LAh8AngZ8ZJxmTfotDwqCitFa0rTcMp3dGfgOcE3gK2tU0KhcEnSZTzDOM6pqEu8P8brrAp8HvgBcr9IEV85/r5Qbp3XTfEtT4jUtt0wvGxEPaFQuCbrMJxjnmYvEylne9kTgucBzgAMrTbg08JNxmjT5tzQlXtNyy3S4EfFWgaBODn9bBoFN+swHgVsDtwBO2aQYLOp2U+I1LbfofbN+v/IEvWjsm9zI3xP43DIobLJnNAT9HPgjsF0YjjYZBI2625R4Tcs1emmt0MoT9B7A8dEp91mqbilrI3B74N3A2wDxm4o8GNgPOAp41QQa1ZR4Tcst06WVJ6gr6InhLrgf8KVlUNhkz7wQeBywL3D0hPr+FODgcPHo6llvaUq8puWW6c/KE3SZTm/2ZzTX7wpcIXyjU8EjCXruL5EEncro7NgOAw5uBFwL2Ak4HXg18OtavTsA/xO/v9qCd54X+LuIYrGsVl/3rVW5AXA34OU9kH2jEvSygaGureOAb9Uw1A/9z8D7gI/WfpcE7UiM9X78AsCj449RL35kQ8I+DGwDSKCfVhqpOnYs8NJ4Zl77LwY8Crg5cJso5L/rA8i9rHtaAxtu1RGMvglq4IU2CY2Hr4+wvdJEgzYOAq4aWKnu/6zW/qaq61rlJJ7fYG/ACe/pwLNq7ykur99G0Ih/FxmcoOcDjFW8F3Bb4IoxcPS9/SPgjP6D8Me9MlplXOEBwH3Cv2Ms4hExS88bA64a/vl4rYBRIM7w944Qt8fHSqPT/qbAL2IP689/33GAjf242ElISSqJDOVTxFySbgs8vIabRNK1Ynjaaxo0WPfVjwD9o+5bX1x7RnKeFFbhRSFxi17XJ0H1+74pVi5XKIMyrh8N8D1OPp+MseGPHwO8ZACClipdOR3vb6+8s/zucsCngUvNiBgalKDOuBLRwePsIQEESaOOqtcXgyy7REvvHGQV2LOAbwB/D1wlfm8H31wB0QEh8e4b5SS4oWxFngAY4laePzwCj53BPhMxlqqEyluirkWDaCq/lzAfA1SfVG1PrTTMlcNggwvH7xycEkkrqXgprrCqwK8D/rKgU+Lqs5Z9wIyybwB8pytVF+mToE7oRwL3B54UZHXCf2TEHLtiGn/8O2BLwFVQPKrSxwpa6hM3J0S3Fy4kdXEiPSTGZNWfPyhBSyPsuEv8X2PWemwQxN8702m0cBZx3+Rg0fQvOZULAZ8Frl5To/TfSUBnR+tWXlEjqD9zkJbYU1dqJwU/WtmblcFn264MfLvLCBvxWVetPeOjOrDr4p7UlbWPqCDx9RtWV6Hq+wx6UG2WwF2kL4KqQVwJ+HqMGTUG6/bbSxS/v+Nsx1gQbPNuM3zofRLU8f29AEfPQ902oEZnwLsLSlVGIWjRsX2xq2UhX2lIMfv7f09UuNxXRb3934Lg54+/y++d/ZwJVcVmEdRyP4yVxlVjj1rEkR/G1VRRDS8+1S4Dbehn1SRUmey3s7FBB0OKmtDJgHsjJ9SqOKn9ZxzJcpJbS9SQ1I7micTyz5/jz7xyutP04S4SJ3excYy4zVEL8Bt7oke5K3BCbHPUxuqaRJ8E3SI0SPtXnwz8mTHRd4gVttqvUQiqWqF6qaiaeYqiKho5Dosf2KAy05QyAlv2p5cM9bf6vPX583kEdQ+rNfKNoQ5Xn/W50p5Ze6xFg2A9fm9o3s2AdwJ3GaEBZZD4KlXqH1fe6appUMGHGrSjEKJB0TWLNJ1Ibwe8J8aLe28nerW0Iu45949tk5NeXfokqHVrI3CVdIJxUivitsyJ1vbVZRIEfUiQy8bNIqh7B8mlqOqeVuuFA0biL0NQ1Q3PRiruVQ7tOnoGft5+FoLMMmwM8XpneO0HrgKefHG2V9zXqmY7wfYhfam4pS1FM1PTeEYcHau2UxVY+4RqZVkAqr/vm6BqIWojGkBfFC9yi+b24U4zXFiFD4MfN1u0grYh6KyDq5uJoEXd9OO5Ir2jD2Y0qKNsE9wiuFpqA3BgaRgqR9gaVLNmkb4JWgIytEI78RfV1kYYpFHsDe4PtU/UpW+C6upx9dRI6WrptsyFx35/cw4yK7eCbnaCVmOPxzyJ4ml+tRfVS63wbw23S92e0IWkfRK0BGTYHj0JHqurijG/Wnm/FlFVs9rdN0G1LOsXdbunaq3fWgNo0Q5ntSEJumIqbjlo7cfcK4xF9Q+r/1N3lrcj9CWfCJeN7glXcfedWm/7lD4Juk+0UTeK+7t6EILGITUQ/braHsYgqJOE2yhXUg2hbq/qQQv1diRBV4ygWiZVx1QxtYi6P6+KgSCSR8Nany6j90agiQEQ7qEcZH1LnwR1VRIbV/q6f1ZPgITVIq3lVEPSGAR9crjFfLd+dye7RbIhCKqvzygMN/p1P5IArGXFNaTNaCJlFYxEtrPs2XVoa9RyL6iLQ6ulg/IRCy6YWjQoZv1e95Pq9ZAnYPoiqEExTiROYrMiptwaaNV176y24So7BkFVa7UcO7mVuIBF32IUgqpCaFFTtFrVnecOqJfF742N1LpWFR3Lr40feP1gNWrGH2uFVV2wTAlaqD7//VBz9B2qFlaluldxgBjNsQqiBdUJxX2hq6qrpaZ79zkGZvQtqoJab+vRNn2+py+CFt+2fk0n7rp6W1TN94dWMK8Pfe9BHcduPQyw0dfbREYhqJtxN+VK1UxfGujZv3Ivzt3DeVxtvEHNmsmVug9Jv5zhU1rEDDio32EqcZ1NjapxH3WTGip3DKOHP57npmkCZJbpjoCrs64jfeJlwu5e67lrKPvpRRNy3wRdpi+DElRVQkLZUQmi6NN5VxDScCzVNdUQA+QVyWTY2DFBun+NVdFoEEWXijGN7i0ks39KLK+/d69kQLKn8p2xXL2rt9b5cbSe+bzE1wVUrlV0VnOFcIatHwlaBtx8ZnoIqG38CnAfWlxG81q54Qk6vc+TLdrsCHgkTj+ue3a1q+rRrjo2SdDNPlqy/6MjoAX6BRF2Z7DCWpIEHf3z5As3OwIl/M+D5/pzFxHUS7wXnXOVyE3KLYN92YMuSv3QqFxeXL3MJ8hnxkTAU1GG2XlY+8YLXuw55mf2WG6ZfqqGG63lHVLVmzHqdTUqlwRd5hPkM2MioDFS15oumHnxr2O2Z9R3JUFHhTtflgi0QyAJ2g6vLJ0IjIpAEnRUuPNliUA7BJKg7fDK0onAqAgkQUeFO1+WCLRDIAnaDq8snQiMikASdFS482WJQDsEkqDt8MrSicCoCAxJUG+A98Jrc2p4+sU7bhWPl3k63cPE3u+SkggkAnMQGJKgvtIrJ72hzVu4S/5IUw54oZXXGHpYOSURSATWiaC+1vQDXmFSvYvHmwQ821m/iSE/VCKQCFQQGHoF9VVe+e8Ba29WVzxpYN5KD36nJAKJwBoIjEFQr8JwH2oqB0V118vCVH1TEoFEYJ0J6nEhL13aKlLpeQFZPalsfqREIBGYgcAYK6g3c2sM8r4jV1HvgE1JBBKBBgiMQVAvIdat4gl2LxRLSQQSgYYIjEFQM26bKk6faDUTccMmZrFEYPMiMAZBNy+62fNEoCMCSdCOAObjicCQCCRBh0Q3604EOiKQBO0IYD6eCAyJQBJ0SHSz7kSgIwJJ0I4A5uOJwJAIJEGHRDfrTgQ6IpAE7QhgPp4IDIlAEnRIdLPuRKAjAknQjgDm44nAkAgkQYdEN+tOBDoikATtCGA+nggMiUASdEh0s+5EoCMCSdCOAObjicCQCCRBh0Q3604EOiKQBO0IYD6eCAyJQBJ0SHSz7kSgIwJJ0I4A5uOJwJAIJEGHRDfrTgQ6IpAE7QjgJn7cFB7e0rgLcJX4+5DIIrCJYem360nQbnieH3hE3Pd7NeDCwPeA9wMnx13ATwKu0e01k3z6JpHO487AFYA/ANsCv5tka1e0UUnQ5T/cxYCPAbsCrwWeD3wL2BG4A3AocAHgeOBey79m8k+aOcBJygnplpNv7Yo1cEyCPhw4ZsXwWau5bwXuDpwE3GlGQft7NOAKKlk3qnwauEGk9Dh4o3Zyvfo1FkHvB5gC4srr1dGe33tR4BeA+D00cs3UX3ER4CxgT+BDPb9/KtWJw8+ALSLv62em0rCN0o4xCOr+65PAjzcQQW8NfDAGwQOA180ZEKZddHX55UYZMLV+qEGoSUjSSwJ/3aD9XLduDU3Qa4cKuAPwzQ1EULO1aQhSTghVd9ZHfA7w5HX7usO/+ChgX+CNwH2Hf93me8M8gmqNPAC4KfDbyJTtKvBl4F3AKQ2g0gT/X5Eb9LyR9kFLn/LDSjrCBlVNrsg2gFnaLhQtOxCQjGuJ6uC1gOtG4uL31gqrJj4+NI3XTK7HsxukUeyKwIOBY6PI9YHrhVXXcfLdFenLJJs5i6D+7D3hInClKKRSlXPgPDD+btohs2nvtcFWUPuu5VILZhH/P88I5j5ULB8Urgj3pHvUANw9tgL+WGuoVtEpi8SUoIqWa9Xb/wBuW2m0Y+dRwCum3JEpt20WQW8VRg2THR1Wa/xx8Tutk01loxLU/ptKcZ8AwgGqmvemNYDZDzgy9qQXryWT0qeq2myaxv2BwxsArA9yrfc1qOKcInsDb25aOMqp2qrifiVIqTvJxMwlo/rrgd2A3wNXyoTNLdGN4rMI+uggpoYPV82quAro96sTd623b2SCbgm8u+L/+xNw1/jZLEwuF4EM/s4tQFmBSlkjcr4eqrDbiUXiu9wDdxU1HCffNlLcTC+OQIWnBllLHfqC3Q4p+oElcEpLBGYRVDDLbGo+T/N6FnGfpKvk9Bbv2cgEFYatY+W7UWBiJI3/PnUORt+OAX0X4J21MjeLVcjwuSmnanQcaLl1X626/iLgxFpfrgqcFj97SCZubsGYStFZBFXV+hJg6JpitMwj1xhwi9680Qlq/40qcs+oEUhxYGrBdkWtiwPabYQGoRfUfvnqWBFVE6csNwY+Hg105XUFrkvV0u1+e6P6ggf9TvOsuDuHn09DgPKX2BM9Bfjfli3aDAQVElXWzwFaeBWTFs/a12lI0dCmQUnDUhFXHPen+linLs8ADopGus+033UxgkrLtuPlUkuMm6ljMEr71vKDGvhsfKkm9CKqZ6pmX23Ruo1EUFU1DSHzxKiil8cvjwgLZr2sEVVPB94R+1V/rxtKFVFN5TstsF2vorrPDJb/KHDzOY1wjFwdUCvQep2yBAJNAhVUZ1TF/FvR/+dq0XQl3UgENe70hmvgfJmKtVILr4Suy8Ni9awOblcbidnWULMeVlw1BPef51vD+HP7MJRpwXWrlL7QJcjpI7MI6qD6MOBqWcQZ/nnAv8QP7g28peE7C0GtT3P7qooahQPTGNt5k5P797Nj8M5zlYidrhHdE9eMVVTfoatnW1kPK+7dwpXitseAluInL22XuBrIJGbdyNi2f5u+/CyCSsI/z3ClWNazjjvNCFZwVtVy+QngNzVUjTBRxfljWDxnGU50P0heDS1Tjecs/VvLImkUzWfDAmvEkMa2utwuAkGMppKsDmL/FvNVEPfJ+nO/Fi63eptfCDwO0A96/xkd0gJsfPb348DBKvR53do4j6A6oQWxPjt+BNAVoKulrLDOmPrsNHJ8Koha7dCzAUPhFC2bzq6Xjv2Jlj19f1+MsDmPZanuTVGcZJxszgg11wFWFbWMD4SFVt+gg3SWFAuoE5VqritSfVKbYv9LmzwA4BZnlvXWqKGXxj5cErvK1sUxZDCG6q+YtlXrp4xN722bR1CNQ4b1SdRyQl7XwPvCeifpihgRc2YcOfJn7sPcpxaphrAZMF/cEQ7MH4WKV5ztOukl+hRFi6STh/5J45J1zNuXX8XEo2XTkysOOK209cmt9Em11klKFdeQPtXmVRFVeL+1bqUS9mnbLxEa123CfaRhaJb4nMf0ingcz1MwKXMQmEVQB5fWN1e57WIAbR/qqerL22fU9awwiHizwBNn/F41ztBBg8sluT7AH0Q5bx1wP+vA1Y3zhol+LeNuXR08Nudk5SFtVwInJFV/tQgtuPZvLbl8HOQWZ1fjVRP3lu6X/V5O3n4/1VZdSkafLTpaJ0a6oIo7yj29BzJSZiDQxIqbwCUCQyDgft1wQSeslBYraIKVCIyBgHt09+31aKox3r0y78gVdGU+1YZq6H3i5I/W61lW/Q3V2S6dSYJ2QS+fXQYB9+7G6XrTRF7RuQDBJOgyQyyf6YKAY27KJ3W69K33Z5OgvUOaFSYC/SGQBO0Py6wpEegdgSRo75BmhYlAfwgkQfvDMmtKBHpHIAnaO6RZYSLQHwJJ0P6wzJoSgd4RSIL2DmlWmAj0h0AStD8ss6ZEoHcEkqC9Q5oVJgL9IZAE7Q/LrCkR6B2BJGjvkGaFiUB/CCRB+8Mya0oEekcgCdo7pFlhItAfAknQ/rDMmhKB3hFIgvYOaVaYCPSHQBK0PyyzpkSgdwSSoMtB6o3q3nzozXb+8d/mrJl31eZyb8mnNj0CSdDlhoCZtM1C5tWZ3hX7ecBb6lISgV4RSIJ2g9Mcmd4Ub5Jj7/4dQ7zo+ddxM/sY7xvyHab8qN/QP+T7Vq7uJOjyn8xLuL0l/YKA2bzeu3xVjZ68LPCEyClq6g1zu6yiOOZM+vS00DzM9pYyB4Ek6PJDo2SQNimU6S+apmNs+0Yvdva2fvO0Ohl4i72JmaopFNrWuR7lvQPXG+XNHmDeH5Nk2S/TjKQkQXsfAyVXyymRAqLvF7hKmgvmAbHPNcfmIYAZulfpLlmTa/1TJNAyUZbEfCNgupDT+wZto9WXK+jyX9RkviZLMmnSM5ev5lxP7hp3xu4VOU8kpsmqTEi0SsQ0Z4vZy7z/dufIdFaIaZKslAYIJEEbgDSjiFm6fhoEuilgSviuotqq+nePSKxsxu1CzC65Q1UtTRdo6kcTHz03crVW2+tEY7Y5Ext1zYbt3tx9pUazHYKYJnF2xfxGV5AiXYS5ZK8Vf5wc62kOnRAuFYm/zKi3spIEXe7TlSzTZuVy/9llZTMxsMT0xnXFvKsS0/R+XYhpXU4k5uy8OWBqQMV/m5e0Ku8OQ5e5O83ctoxsHYl9TQAtOSSNmeoO7omYtslM5BrK3P8rZpSTqHXR7eWE17d2swwunZ5Jgi4Hn2kIHfjvAu64XBXnEMUco4U43wpimsKxKzHrTfI7m4vVlJImLTLBcFW0Qp8E/DxyfbbpUpkErNfJSmKaXVtimux3CPkccL1IB/no2gskpgRV+tJuhuhDozqToI1gOlch07+rLrpamDO1jewRM7uZyhWJ6WA2t2bfxKy265VhCfY9Gp7q4mqnlfieDTsjMc3zun/k+pSY1m1fhlQrVdnNQWpeUdtqCsOqlMmzD+2mIRTDFUuCtsfWhL2uRsp1gC+1rELjz0HALvHcp4KwQ/tR9w5D07yopw8CxwbJmnRJHNxXWq/RVLqZJIfpBN2fDyUlQ7n5XQzaqL7LCcYEy9uGRlC2DUO1ZfB6k6DtIb4/oBrqwHCALJMIyFXA2d+9p8YbRaJK3Pe0b1KjJ1SpT45s1q4+VdGloyHnhuEGaVRhFNopDEIPBbYMoh4e/s0hiOrWwolg1v5zX+CoaNcBwIvadGSKZZOg7b+Kq4zugzcB5rnsKncOou4eFem+kagabvoULZtahhWjklxpiqiavgr4UIcXur9V5ZckGoxcUY8Iop7Vod76o2+JyU2SVvefTjqStmTsduI7tcf3rktVSdD2sBvJYwzpI4Bj2j8+9wktkxqNbhElPhNE1RDVhxgw8PtwDWlg+UJUqoFoz9pg7/K+7YDHxN7UfapEPTKIemaXisP9ZB3bz9h/OqnpQ74XYBknjGW0m45N7PfxJGg7PPUnFsuk/9bA07doeVT1lTjKZ4OoWlm7ivG7rp4aqlwtJZMBEKrbfR+V2wZ4ZFiNLwGcHUT1YMGyRJWAXwniVfefWm51fWl4001kQIQnjlZekqDtPqGr5ssiHlaVcUjx+JpEdeD5nXQtuEqc2OGlX42zq64y1qMFVPdIHwEE85q1FfDw8F86OUhU94kS9YyWfTEeWWt0df95UeC4CCd0AtJQZKDEK1rWPcniSdB2n8V9571jv/YQwHA2gwp0ng8lrhoHxn53iyCqZ1E9ctZWPgEYGOE+UaOR+06tt2OIxNknAuSd3CTqYyN6qen7nRydJKv7T1dL/6+B6gNRkUYvAz50AR1WqVy1V1+tMcDGBE9ekqDtPtFPIkqm7D/dW+n0H3IFKi2sBs9fYcnjZrpyjMZRxdTKaUDB2FKC543RNaKpzXGzEiHkQXkt6YYtep5UY5T7Xr+Fq+iOoW2cAHwxOmi/1Rp0CXmO1z3/78bufNv3JUGbIyZWrlpaKF8Svra3Af4ZUzRQ6b5wBWorx0esryvo0W0f7rl8iRFuM7lpcFJllnQGdXwsVHSbJlk9viZBPxyrqfvrIpJXEhdxf+wEO2lJgrb7PEbJ6IfTUORRsLHUw3atnF/aQar1tjpw+6p7jHo8O6qarLvIvugeKrJbuL4M1j90RjijxjdX2hK7a11Dbk16wSMJ2guMWcmKIfDOMJDp0560JEHH/zyGBnpwuYvo4lnvK080lnnipot4+Hy/LhUs8aw3Mrr31SUz1C0YSzRr9iNJ0N6gbFzRacBVG5eeXVAjyHoT1CgqLahdRFdIGyNRl3f5rIETumR08RSLb9c6B30+CToovFn5hBBwrHtiR8NQ/TzshJr5/5uSBJ3sp8mGDYCA432lwv+SoAOMgqwyEegLgSRoX0hmPYnAAAgkQQcANatMBPpCIAnaF5JZTyIwAAL/B0TxW9ySBNn6AAAAAElFTkSuQmCC" width="116" height="72" /></span></div><div  class = 'S2'><span>If the scalar parameter </span><span style="font-family: STIXGeneral, STIXGeneral-webfont, serif; font-style: italic; font-weight: normal; color: rgb(0, 0, 0);">η</span><span> is small, then </span><span texencoding="c^Tv" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAC4AAAAmCAYAAAC76qlaAAADHklEQVRYR+3XWajVVRQG8J/NBBUYFiWlUGkgJlFRpPQSSEJWRASNiDhkUQ9BVEQzmFTQDD0IUdBE80BJ0EM0KJH4kEQ0WRANpEUpFtHEF6s4/e+55xy559zuhbNfzmH/1177299ae31rTzFJx5RJitsQ+HhHbsj4ZGZ8C2biQ3yL6TgSn+BjHIeD8DYWjPWg/UqVAHwOZ+KzAvUUzsG5yP/stQY/46aJAnwl3sQHBWh3fIf9MQ0/1PxC7MRbEwX4ftjeAuZEbMB6nNwyfyB+xG8TBXgTx/W4BTfW71hxjljfrxxvOk7a5AKG+Xf7jrouTL/9Jm2+r5RIFfmj3xvE3yAYPwMv4AmcNwjQgwJ+Py7DEjz8fwPfCyfhGByNb/AoIjrN8RGOwqH4ugvw2MzDsXgSnzbsD8GleK3K7b+fe0mViModeLnCH2W8Exfj1FLCfxxGOXOY9+uQnXAH0AnlZzfcgFsbC64u0dqBHCK/f49OwPcsVhfjdLze4vSlmnsMF7TM31wAHsLSHtMkTEddcy/Oaqw5vKrSwZiLzd2Ah4HnEdDLsbbh8FWchitwH+ZgGS7BPsV6ovJspVWnM1yER/AlDmtjuAKrS4H/7Ab8KtyOdzC/jbPk/IxqnnokdlSzsPpFfT0APzUsI2bZK8R0zPGcOt3c3liEdWNF1mV9+ppfsAeOx8YW+8xtKhyJSEfgkeqcMo1RGqTfBww87j8vVs/H4y37JU1DZC7uf0a7y5mqMbva1LPHAXS2eAOn4ErcVXumikQHUhh+7QY8YUu/nIoSB3E0HiOaELZTDsNuCI3yXlcPkREYmoxPxbayGlhn14aJB0poorqXl068V+DbEtcuVdIv5wHwIFa1WZVSmRR6uo+huA3XlG6km0x1aYpR1xx/peUWH9HIr32r5qbuvthH4NdWrU60Q0j0oONox3jEJGGKkETmo4Zbq1RFcO7BM90c7+L3pMe9xXhaia6t8GiSPwsJX17mkduv6hB3j9JY7SLOEeYX1l4Rvp6edb00WWMFNZD1Q+ADobWD0yHjQ8Z7ZGCYKj0S1TezScv4X7+sjSdMrmEKAAAAAElFTkSuQmCC" width="23" height="19" /></span><span> dominates and the same optimal objective as in FBA is attained, with the advantage that a unique flux vector is returned - one that minimises the the sum of the squares of the flux vector.</span></div><div  class = 'S2'><span>In practice, for genome-scale, mono-scale models (e.g. not ME models) the following value for the scalar parameter works fine</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span >eta = 1e-6;</span></span></div></div></div><div  class = 'S10'><span>This is a quadratic optimisation problem, so a 'QP' solver is required. The COBRA Toolbox v3 comes shipped with the solver PDCO: Primal-Dual interior method for Convex Objectives (further info here: </span><a href = "https://web.stanford.edu/group/SOL/software/pdco/"><span>link</span></a><span>).</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span >[solverOK, solverInstalled] = changeCobraSolver(</span><span style="color: rgb(170, 4, 249);">'pdco'</span><span >,</span><span style="color: rgb(170, 4, 249);">'QP'</span><span >);</span></span></div></div></div><div  class = 'S10'><span>Solve a quadratically regularised FBA problem:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span >FBAsolution = optimizeCbModel(model,</span><span style="color: rgb(170, 4, 249);">'max'</span><span >,eta);</span></span></div></div></div><h2  class = 'S1'><span style=' text-decoration: underline;'>2. Sparse FBA</span></h2><div  class = 'S2'><span>Sparse FBA calculates the optimal solution of a linear objective function and finds the smallest set of reactions that can carry flux to achieve the objective. Sparse FBA minimises the number of active reactions by keeping same maximal objective;</span></div><div  class = 'S3'><span texencoding="\begin{array}{ll}
\min\limits _{v} &amp; \ ||v||_0 \\
\text{s.t.} &amp; Sv=b,\\
 &amp; l\leq v\leq u,\\
&amp; c^{T}v=\rho^*
\end{array}" style="vertical-align:-41px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMMAAAC6CAYAAAATUvmjAAAdnUlEQVR4Xu2dCfh3RVXHv+644r6BC5BCKJlICEbmkmKY4IIKhARuZColaZJ7oigkIoYiSomaKRrIprGIiImguABqqCUqKKSIiJSYivZ8Xs74zDvv/f1+c/e59z3neXiA33/u3Jkz852Zc+bc872RXFwDroE1GriR68E14Bq4QQMOBp8JrgHTgINh2KlwG0n/KelBkn6w5NW55Zq2/uOS3irppBUV5JZr2o6innMwDDsc95H0bUkPlPTVJa/OLde09d+S9CZJb1tRQW65pu0o6jkHw7DDcV9JTLCtJH1lyatzyzVtPYAEDEesqCC3XNN2FPWcg2HY4cid5LnlmrY+d5LnlmvajqKeczAMOxy5kzy3XNPW507y3HJN21HUc2OCgXf/uiht9N+Y3EmeW65pi3MneW65pu0o6rkxwLChJLwUd5H0FElfKEoj/TYmd5Lnlmva2txJnluuaTuKem4MMDxZ0nGmhUMkvbQojfTbmNxJnluuaWtzJ3luuabtKOq5McDAznCKpDtK2kPShUVppN/G5E7y3HJNW5s7yXPLNW1HUc+NAYaiFDBwY3IneW65ps3PneS55Zq2o6jnHAzDDkfuJF9Vjt31dyQ9WNLXJZ2WdOMmkl4s6QpJ763oYu4kzy03rBZ7epuDoSfFLqh21SQPjy0rt5OkP5K0t6Q7SPqEpEcn73uopPPst0dIOjv5e+4kzy03rBZ7elsVGG4qCQXuKumxkjaVdDtJL5P0NEkbSbpM0hsk/aO1i1ia/SU9XRIDSfwNV/3vWtLue0nin88kZe4s6YmSnirpy7bCbSfplZJ2kHS12RysfD/rSS99VdsFGELbnifp7ZJ+bPZX7Ka+maQzJP2hpBdW3DTnTvLccn3pa9B6UzA83Cb9YyTd2CbbQ2zy/UTSBZKYmJtbK59gwPiQpCslfUPS70u6v/0d8Hw46hHuVCb57lYOMD0n+vtLJD07ep5wAYB1oKTzzR3L8QD5V6trUIW1fFmXYLi3pO9Ye35L0jeTtjEGHKHQF4tKLLmTPLdcS7WU8fiiY9J7JO0l6Ve23f6VTUZafVuLq2Ewvibpekm4SwECcktJn5e0paRPSnqk/X4nSUz2e1jd/Hx0AgZ+u7Wkay28nB0IAO4pCTAiAOiZ1rb7SbqkDFVmtaJLMPBC+r6JpJ0lnZy04A9MVyxc6eVm7iTPLZfV+dILLQIDvv83WuNRZpjooT9vlvQi+x/Op59LOvoqSX9nE5YtG1AF2UDST22yV4GBct+TdE9JZ9l5OB7MbSJgcpQLdxal65r2dQ0G7AUWG46MhyYKYEE7QdJHKhSTO8lzy01B9yvbuAgMz4/OmXeX9P2kpv0kHW6/McBhuw7FWLmDPXFXO0LFVVAfvy8CA0cjtv4P2pEqfpbnQnsA5FtW9rKcAl2D4RhJfybpnZL2jbq5hdkTj1rQ9dxJvqrcxmY7slixo7PIvUDSNeWoPL8lTcHwLJvIYbVLwYAhzURGOC5dnDQJlx8gawIG3IoYjcgBkg7O7+7oJbsGA7svuzAf6exivcPW41KTBY1w8SpZNcnDM8vKsSgRSsMO9Ap7gLDwh0na0Y66oyu8TgOGAEPVhywOhm6+Z8D5wK7w75JwfoQFAhAcu2QidAEGnBucALAFr7N3sTv8SNLrJb22zkQsoayDYdhR6HpnwDOHJ48PhfhgiN0Bdzi7wjJpC4Zb2IUezo30KHaOJJwr9BXnymTEwTDsUHUNBo4jp5rDAWD8jbmbf9kzGHC9n25GO8Z7LHxbzd0GLvjPDqvedm9zMLTTX92nuwbD9nZp+Qs7KnFZiVt6lbTdGYLN+HJJByUvC3ZMese0qk2j/30sMPy3pLuZx4lLtlSWeZNub7fQPLO+G9AcjS6yYxJRA1dlzqi2YCAaAJsAz1GaVAAPH673KndvZvPGKbYIDKFDtIpLMiZvLLjx3mE/4MbjpjMWLsneZz+QFoUBiwVvEF4hynC5l8qlFqqBEbhb8kfCQb5rv1WtTONoMu+tXe8MZNE4ytyrqft7WYvagoHQHIxkbBNCQmIJbvepLVQLk4jRQWJfkK0lfSnpMIpAIciT7HInLvIaSa+2H/hm4QPRH7lMYzIDREIstk3qBiTkFLq5pHPNVRcXeby5DvltkWs2b2oOX6prMDTtQVswsJsTd8YcIEYtlnDhyv1HVcRs0zb3/ly6MxDYxeQlIpLJiOCm+5h1fDNJnBfpKMF5CBP33ebioz6MOFZ7bpqREEZ8vAEH8ITYJv5O+PGJko60sAt2JQAYBEDgv+Z5QMZqRDAfgqGIn5sBSWNzeldegxfMBQx4kM6UVPWl4mGSCN+pipZtoLLhHvEQ7uF0zZvmAgZCbDiWEYbzuESFhNBgy3C8xrCfjDgYhh2quYABrXHZR/Qxn++GSc9pgBD7f8q46xhW8xlvczBkKKnDIgEMq9JL5pZr2jRshr/PSC+5rBwr/xfNgCbEHsGW5IjNF3jLcsk2bXevzzkYelXvOpXjHCAC+AGSfrjk1bnlmrb+UxbT9dEVFawqx4df2Gt8ZEVMFHYmNmMaq9a0nYM+52AYVN3+spI14GAoeXS8bYNqwMEwqLr9ZSVrwMFQ8uh42wbVgINhUHX7y0rWgIOh5NHxtg2qAQfDoOr2l5WsAQdDyaPjbRtUAw6GQdXtLytZA32CgUxuRC+SF5QoVwK4EEK2yYZH4jGSjbm4BorQQJ9goIOkg7nc8qQSt4LwMTm0r4T6rqJeLUJJ3oj1QwN9gwEtEszFZ5zkUgrCxz58m5B+Ibd+aN17WaQGhgAD6Q35GIfcnwjJh0ldyUdCLq6BYjQwBBj4Vhq7gXSRCB+Tk3qS45OLa6AYDQwBBlKHkH37VpbTh+QCZIBzcQ0UpYEhwPAXZijzfTW7A19BubgGitPAEGCA6xlXKl9W8eGHi2ugSA0MAQaYfMiuxp1DSppRpFK8UeunBoYAw/qpWe/15DTgYJjckHmD+9KAg6EvzXq9k9OAg2FyQ+YN7ksDDoa+NOv1Tk4DDobJDZk3uC8NOBj60qzXOzkNOBgmN2Te4L404GDoS7Ne7+Q04GCY3JB5g/vSgIOhL816vZPTgINhckPmDe5LAw6GvjTr9U5OAw6GyQ2ZN7gvDTgY+tKs1zs5DTgYJjdk3uC+NOBg6EuzXu/kNOBgmNyQeYP70oCDoS/Ner2T04CDYXJDVkyDSRNKtpPNJd3f/n2QZU8sppF1GuJgqKOtdcveTNK+lg/qtyXd2mhfz5B0tuWKOkASvM9zk4dZytAnSNpE0v9JuoOk66baUQdD85G7vaRPG6fz+yS9SdI3JW0s6Y+NZxle5OMk7dr8NcU/ScZEFgTA/4jiW7ukgUOC4bmS3jllZSVtP17SkyRBLP4nFf2iv0dJYmc4eEb9TrvyOUm/Z2lDXzflfg4Fhj0kkWbyflNWVtT2DSVdLQn9Pdtyx6Zdu62kKyXtJOkTM+l32g30cJWkmxjvxvlT7ucQYOC8fJ6kK2YEhkdJOtMG/hmS/nnBJCAVP6vmj6c8SZa0nZ2RHRJA3FXSr6bcz77B8CA7Rmwk6b9mBAayimMkIyfYcalqHrxB0t9OeYKsaPuRkv5c0gcl7T71fi4CA16R/SXtIOl/jIGH1e3Lkj4m6VMZHcftdo5xM9zYUkvicUC+F6Woz6iquCK3k0Q28Vtay14miYm/TDhSQO31YCNpOS0pzFHjxbaDvre4Hlc3CIfBppKeKendVuQhkrY27xLz5NsT6cuaM28q/HaquQVZAcME5jjAIMHPVmewYOnZbWY7AzrDg4InJQj/v8hBgN2ALvc29yM2xKMTxT/UjpP8jFcG70zJAggAA4IHjSPSMZIeGzWaufMCSUeX3JHQtiowPNIMPhIFH5504lj7G16SXJkrGOg/6fX3MUUwGTgqfGiJYp4n6e1mQ9wxScTMnQVHL1L3v1DSERkKxse/7H0ZVawpspekD+cWtnIcjzgmfcUAgAsZEprA1PR+SdtI+pmkzaZATlMFhv0MBBiF7AaxsLo9oAIky/Q4ZzBsIOnfIv/6LyTtYr9V6eTedinH3zhGhpU1lOUmF547jlMcSVcJ78JmaSvs3Cx0dSS4lt9il26vMGCEOrhr4UiNcM8CWIqWKjDQ8LBKwKcAr0IQzrW4R79Wo1dzBgNquI2t6NuZTriB5b8vWqCjS2zy7Czp5KQMvHesroQ4lJy+n3mABwk7iCMfzK2nJH3ZQtLF9hv8fcWT1FSBge36QkmEFyDcsj5/yeCuwsXcwUD/uY3mjM+KjjAJ8KSxU6TC5OEoirF8aPLH99hKz1GjZNle0mesgewo7CypxB437KPi71oWeZPua350jCTkejvDvlzS/9YcpfUBDKiEY88XJOFpQiBoqTqHY2TihMDYxugOwkqKPcEdRunyaqMupp3YBfQ7FW7e8bAxX+7WYN4MroNl9wwEXRFvg9ssCFs82zuk5rkyJzCw3XOMWSTcRr/L/gjhO56UVLiJf5Wkk8y+4O+4njlmsAN/K1exI5bDZU6gHkSVD1/QDubIlpLY7fCiFS85l25siWzn/BvBv84qmLtDzAkMxOFsu2RU7xF5TTgjV3FdP8d2hXgisYoCgrpG7BjeJHY+7IWbLjGMH2dOBDxJHLcncddQBQYG8CxJ7AJBWLkOkfTX9sNTjbQwB+0BDNSHi22qwk7JJCDmaNFCgL31U5soi9yj6A53KC7JrWx3wDfPrlBXxvAmPdHcpxyduZwN91Ch7YAE5wEgSB0wdfs3aPkqMDDhf1nhPqXsdyTdq+LijdUCD8q5kq5NesDNJNvkz83zUmVU4nIEKBihpca3hP4t84xw+/p58wRx04wjIpUd7VKTW3iAwYTh3+h8CoJdw33Jf5ibPW3zmyW9SBL3DHtWdAhPFPFql1qwYzF9XgQGLlRocIr6T0rC/Yd7NewcrAT4xDEAP2ugiDv4ekmEKyB4WFg17m7nSTwM+NYvsNAGQp05MpQoABpgf9+OSgxmLOyeHzdPEb53JkSVBE8MiwJHJVbadAEpsf+hTQQfckyu8iJhI/2D2U0Aht0jFeYQF4scodBp3aNhb7pZBAYMZ0IuAEX4cgl34OnmRWCCB+Em9QcWxstvnJuxK4LEYQYE6wUXJJPgcjsmhIsjLpwAVYmCZwSg4v8nTotLJvpyjYEcDwsRqgwu3qJ0IQl94mjEgsAxibALjl5TEY6BjDWu5BCaQ9vvYieJx5jLGKO5SniO0PcghLgT7VqEVIGBgcQLwOp9JxusO9sRhy3wxIqWH2jGIl98vbTi7xwFCO8gsA1A4WO/zMrxNRik6UwSXLf/UoRm1m0EcUiseoSiszDwQQ8rHODn+MjuiCeJ/i2T+9hHP+iZXWZqgi2AfcN4sVAyfhx9cCMTtbAqXB0d4XYOLmhsMIJBR5ccb9LojfQGzFID2FeEdLA4FCEOhiKGYb1sBDYVdlZ6Cz+aMhwMo6l+vX7x0y3CFy9alXdxFOU4GEZR+3r9Umwt4pb4ArCotDIOhvV6Xo7SeeZckRG5DoZR5oO/tEQNOBhKHBVv0ygacDCMonZ/aYkacDCUOCreplE04GAYRe3+0hI14GAocVS8TaNowMEwitr9pSVqwMFQ4qh4m0bRgINhFLX7S0vUgIOhxFHxNo2iAQfDKGr3l5aoAQdDiaPibRpFAw6GUdTuLy1RAw6GEkfF2zSKBhwMo6jdX1qiBhwMJY6Kt2kUDTgYmqmdTHJkECFDBP/w3+SgXZQeptlb/KlBNeBgaKZuGHrIlk26F3IJfVES2R5cJqwBB0O7wYOjgAx5ELqQG2oIIenWTywj3RDv6/MdpBVNMxP2+b6ldTsYmquehGhkh7uFJLJOp+ydzWuufvKekl5inA6k9yRX6xSFOUfC5FfajkpW8iLEwdB8GAIzDQmVSbGZm6K/7htJskWWQngyAB7Z+0hqHKdprFvnGOXJkUQmPbImkseXBNP0i1SmRYiDofkwhNyrcB2TZrJrYfUntyskk9glcBwcZPSyxeQayug0ian/1JJPk2QaEECiTkrSOtyAGa9qV8TB0Fx/EJeQaJiEw69tXs06T8KmSk4heNLIYQoISPRMMt8pgYAcrGTZpi/QopGRO4CABNPFiYOh2ZCQTfqHNll3kAStU1vh6MMR4snSGrJ6mHwCCNpwN3A8IYU8dAAkDX6jcWXE7QXUZEUnKXBblh1sKewAHAobGQggrGEn+EZbJVlKSrg8IJPkHxaiNPU94INHjgznZH7PEgdDlprWKRTYa8gejb3QZsWGBAUQkGkOgfcCEJDyvQ0IqAvQwpkA7xrp4hH+G16IWOCyxgkAdwIZxpsIFMBwMkB2w0RkgpJR/XUdgYA2wXCEEwF7DSHzeWBYjduMq5vFpdau7WBoMuw3pKZnkkH6/fhmVayZlHA8hEkKQTogIK1/WxCkTWKc4cKAZoCEv5CpxAIQPirpR8a1UKdLAXDUy8IACGDtAQQQm/QhsItubRQB+yUvAASAAam1azsYmg0VFE4cOVgF4ayoI3Ais2LBgIQAAiYO3AZdgyBuFyyleKR4D0Z5KqzieKuektkZQADPBtx1cC0AAuqmL9lHk8x3xcU49sEBAa8DbSWtfSxhoaq9azsY6o9GzOj5uwt425bVimH8GkmbWyGovwBH3/cUe5kRvui2/Eyj6WJC5wh6wA6gXrxduJaZiKSYx57qSwLzEflauYCM3wWYIZOBjJKdLhw9s9riYMhS01qFIO3jKMMgMBhNkuiyurGqYStg2CKAApCcWr9JWU9wLIN2ixWTVTUW3LgYudD61iWYhPASYxkO7A0MFEfY/UEfoAi8cVX2ArRrR1rH9pd0WJZmrJCDoY62bigb2Euhr4VnoK3A5Qwo4L5DcNkCCozaLgUPSyBc5zabFTQIuwG81RBONhXsEY6NTEiMaXYKiOG5VIO7rSuB8oyFhF0othcAOAAJTECBTDP7vQ6GbFX9piA3wMTU7Gvk5vVrqH4CDwkGdbjAO99AgZHehXD5BcMmdxcYn1+ySjGed0omVpv3wQP4l2ZLYFcACuhyAQXkiG2E+UodcAym9gILCHc0u1oZwFlr13Yw1Bsa/PXBQ8J/Y/x2LXhA2CmYpAi80gw0Z+C2QjwTuwJGPLsAE5fLPCZW1+HnGNUQIeJlgg0UsnhAQVBjU1Aw2WFJTe0FPEi4u3FK4Brmco/I4lriYKilrjW7Aayf7A4cO/oUQsIBBYPMOOFOBBSntHjpV+3bC1ZP6sETw2Tt4jJsUbNuJem5dj8AEAEF53pAUZftFG8YXrHYXtjQ6IYJ+QDsGNFc+h1dV08Ohnoaw06Ah4zz9bOM9pW7AS6C+hJWQ0jlsU844gAKvqUgjLuunGuk9ZzrMajpB16kIYRJuo8F57GQAArokLn1zhUWIhak2F5gF+D/Md4hpUdwCHB5idv38Khyjk7chRATtY6jwMGQOww3lIPsndvVYC+w7XOB1efKGloYB+5t0jCEG/ctt7gcU/C2cDk2tITAPWKWuAmvE8Idbpb5qAqPHqElfA+BoY6dwliwO2xsu+gJki6wDtJvdkPcwHyHgo22FqecgyF/KqArVmM8JW81X/ZHJPHPkILxjsuSlbWuHGexT+wMR9V9uOPyIWaqzkKCMc6xiwnOBeWn7ZhH0wAGIeGA4SzbJbCHggAUABMEe4bF7DfiYKg3wtyu4ufGiCa8eqgjRr1WLi7NhMCLFE+Sruoeoh48Uhy1cBHTl/iCcBtJHGMJFDy4IuQExwQ7SIhloq61jrcOhiGG0N9RmgZONucBd0a+M4w4OhdK4iOXNoJbd+zPPnEkEFnbRo6xSNc2ddR9lswm2Cq4Ydf6OtF3hrqqbF/+YklbtKwGA3FsMODdwpPTRnB/1jGg27yLZ7kEPNbcusHz5DtDW63685PTAAs/kbkYzen3HGs64zvD5MbUG9xCA8z3hSEaDoYWmvVH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyvNH56UBB8O8xtN700IDDoYWyst8lES4UEPB7kMmOLgFSGVPMisyb5PyhaTDpEAhIRn5Vl1G0ICDoX+lk6qdhFV863utfV1IWkWS58JxwH+TbY5vcuFGhkrKZQQNOBj6V/ohNslDYt9As0T+IrJhB2FHIP9o1wmA++/hTN7gYOh/IOENYEcIQtY3sr/BI00O1SDkASV9pctiDTBf2WVJ3laXVGWlXh0MK1XUeQEGksTB8DBAqO6Sr4EtJZFJHKpeUvV3Kg6GTtW5sjKMZBg1f27cCG0oc1e+bCYFSPS8h6WGZL5eY/RhEK+QL5VUmei0tTgYWquwVgUQkEBPRbJiyM9dVmuAHKk4FUgmTBY/Epc9zTjooA3e3tL0r65pRQkHQ2sV1qoAmlzIQUrIgl2r4SMXhvkHIMApQWJhdgLAcaLtFJ00z8HQiRqzK4GCCfIR+BW4e3DJ0wA0ABBBQr8F9RY803AswHfXmffNwZA3GMtKcXm2naU6hwYWVhjOsSmzzkaSvmt/h1B9mQTuAhgrKYv3CTsjFoxIKK5gvpkqsKCg4shDP+kzqeZZ7WPOBpI0w8cAET28zuiWXeIM0w16+Hr7YfT0km10eHNjyIR+lVz/pxsjDts35H4MUsyDvLfR5qaUrWkbSI4LBwRHAs7ECP+d5gfF9sAG4TwNqd/UBBBAMQXQSQbMgoI9ALMOzEiBkw0DGt2RIxWHAwY0v5EmknIsBld30XnfGZppkVWeyQ8gmLBhZYaiCYooGOoh9Yv5ypi0uFOhYMpJ5c7YXC6J+wfsDMg5YgEIMIByfuZMnSOEfwSyjpzyVWVYtQORe9M6Ai8bfYi5pwEAXHmEqDDhIXCPhV2SHcJdq0013/FzTE7ok2CuZDW7KKqfsAouzuAA4G9wkDHgsFSytSPsHGz1sM5cv6JtMFvyLGWfUVGWrNK8E+raHOHWm7ioNsIxcNUxb1n9W5n3h900vnTkmbB78t9wsHEUSiXcz9TieM7psO8MOVpauwyrMSTiBxk1bVoDNgQ7Rhe3yXsZcAAVVLipcIEH+0xM51S/R8M9AVvpeUYbjH3FXUEsMecDdwsfGK5pbjPU1TXnXM63bOMMZieXPUsaga1wth0XCOuIBfZPJsu2fYQm1FVMZnlsIWymw4xtNH2M+KwD7Udu6dlFBxPfGeqpmvAJ6I/gBNu53qONSsOXjIcF4Vh2RVQLuwE8zvGZu9FLBnyIEPYHWoAiR7ZUoLPd046PRPReOWDbnKykhrKxFcJkhEIV+tu+BYOcowTHCyJcwwTCDuGohidrKoKtQvtxMKDL9MyPa5X7A+5gCGcPnrTB+uc7Q76qw5GFJ3aRdFL+o61Kwt3GrsCFE7sAl04Y4BjNdS+cxvQmHWpHI9ynu1dohEu1oFOAsA7nWistZjzsYMhQkhUhlghScWTIiFOiNInW3NUY7o83V2sdMvHQyzG9Sbif+eqPj5zCHUKsfTxHhGejY/o6uDgY8lUePsrhid3MkE6f5n4Br0+Xq9q55qYlnondCTthamTsUPVCJI9gMwDwWNj10Bk3yXwCy8Xa4OJgyFc5AWKX2TEFJnrcgLFsahOVe4FL8qtdWfI087lz1t5f0vtXPlFeAS7Z3iHpOkl4xeL7FZwE59gtMrYQISujiIOhntq5HWWLx/g72M7ufHG1o4GDQU9XvXpvWLc0xwaOaFOOdGXx4NIR7xAc1iHOih0BYnR2BW7ZSZYwmjgY6que1esAO8ezW7AL4O9/W/J5Z/2aq58gDIOzPkbzFIU5xq5GAgSOd9zR4BTgcpJgxqP6+GqtiaIcDE205s/U0UBwqfLMZh0fIeu0Y2VZB8NKFXmBlhrAzsGteql5k1pW19/jDob+dOs136CBEMuFbbBPyUpxMJQ8OtNvGwGLV0niSzWCDgm3KFYcDMUOzSwaFqJuCWgksBHjuVhxMBQ7NLNoGDsBBjT/JrNg0eJgKHp4vHFDasDBMKS2/V1Fa8DBUPTweOOG1MD/A90W0gZZRRZ1AAAAAElFTkSuQmCC" width="97.5" height="93" /></span></div><div  class = 'S2'><span>Equation 2: Formula of Sparse FBA.</span></div><div  class = 'S2'><span>where the last constraint is optional and represents the requirement to satisfy an optimal objective value </span><span texencoding="\rho^*" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACEAAAAmCAYAAABK4fLXAAACUUlEQVRYR+3YS6iNYRQG4Oe45FpKLpFySwZIbgOXGCgmlIiJa0lKUsRIUsjAgIFMhHIbUIQUUlIUE4SSEKHci6IYoVVra7f793HO2WfvlPPVHvz/37fWu9Z63/f72k3+gdX0D2DQAaI0hY5OdHSiUpH/DSdGojOeVPOkRnTiNHpiXqNBLMEr3MaOBLEZ0zAA58oB1asTlzELy9EJvfAJp3AJixoBIoqbjhXZhQDyAcdwt5HqGIelGIzveIsTeFYriC5JsKiyD97jCm5WBD6MVViHLzmOH9mJ+La2reMYj5O4gb3ojgsYlsFXlgUOYn7GVexMEJswN/edbwuIKRkwkgfbS2sZjufDbFwrkGGQMYhZk0RD4w9zpjPxsyxRdOFFPu/G1gIQQ9OsntfiE1H9RkzGnYpAA/Eu3x2snHVLb21/84ngQUjqASYUBJ2RHIlPMabtLU3cGk7sx3oEqfYVJFiNQ/k+DOhse4PoijfohzF4VJDgaBrSV8Rowg9avZobx/yUYMx8UEHk/niNbtiFba3OnhuaA1GqMrwhpFi5QgmR/CXG4ls9QESVQ5ITBwpUEbLtnWfEvbYCiH3VOjEajzNwHL+3ypL0wEVMwgJcrwVAcyDC80vVT0Sp0lEI74+bUiijBLQmHNU6cQYLcT+dMuYezhm/OAnD+3/VlLlscxGIOPs/om9We6S9klWLUwQi2l+y5xFlZ0PdsBSB2II9Kb04oOq+ikDEJWUOYgxBvrqvShDD8TTZPzVvyw0HsQFrUnqL6569BbbdKAwd/9T86fRv3D9kJ/ZB6XAAAAAASUVORK5CYII=" width="16.5" height="19" /></span><span> derived from any solution to a FBA problem. This approach is used to check for minimal sets of reactions that either should be active or should not be active in a flux balance model that is representative of a biochemical network.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% [vSparse, sparseRxnBool, essentialRxnBool]  = sparseFBA(model, osenseStr,...</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  checkMinimalSet, checkEssentialSet, zeroNormApprox)</span></span></div></div></div><div  class = 'S10'><span>As an optional input, there are different appoximation types of zero-norm (only available when </span><span style=' font-family: monospace;'>minNorm = 'zero'</span><span>). Default is </span><span style=' font-family: monospace;'>cappedL1</span><span>.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% Other types of zero-norm:</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'cappedL1' : Capped-L1 norm</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'exp'      : Exponential function</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'log'      : Logarithmic function</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'SCAD'     : SCAD function</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'lp-'      : :math:`L_p` norm with :math:`p &lt; 0`</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'lp+'      : :math:`L_p` norm with :math:`0 &lt; p &lt; 1`</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'l1'       : L1 norm</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%  * 'all'      : try all approximations and return the best result</span></span></div></div></div><div  class = 'S10'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>The time to determine a </span><span style=' font-family: monospace;'>sparseFBA()</span><span> solution depends on the size of the genome-scale model and is taking from </span><span mathmlencoding="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot; display=&quot;inline&quot;&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;&amp;lt;&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/math&gt;" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAkCAYAAADLsGk3AAAB6ElEQVRYR+3Xy6tOURjH8c/hkDgkl5IkJAaUMHJmTBADA7mEQhkQEuU+ceZuMaf8BcglJQMz1z/BXLkPTbT07Hpz2/XutY73fdu79mTtnmc93+f32+syZECeoQHh0IL0mpKtIq0ihTrQy9aajVNYgl11/L0IMgsncRzT8Rib+w3kKFZhCvZG8X0JMgnf+bm/fcXI/1IkWeEIkr9P19mh5vs7LBpvkJk4hhNIHr+Fg/0EMieKTxAz8BnX4v3SDyDzYmk8jGn4FMVfR1OAir+otRaE9w/FypIAriIBpJ8z51MEZDHOYj8m4yOu4EYBgCKKLMN57MEwPnQAfMvZ/j/kyqLISlzADkwIgMu4idIA2RRJKjzD/MiYvL8Vzwsr8Gv6LIqkI0LaB85gYcxwF2N4M05AWUCqWtORYR/OYWkM3gug14WBsoJUtU7EzvjxV8Tg/QB6VQioCEhVazrMbcNFrInBB7iEl5mBioJ01pruCAloNAYfBtCLTEDvMRdPsLEuZ46L1foA2hCTPYql+23d5P/4vhaVZZMyy+N4/9eQHCBV8nUBsAW3caALkN3YhO2Y2hGfVsrUoKexNfyWOidIlXw10oZ6pwuQrkNKgHRdTJPAFqRJ90rEtoqU6GqTnK0iTbpXInZgFPkBYoheJf3KvD4AAAAASUVORK5CYII=" width="25" height="18" /></span><span> second for a 1,000 reaction model, to </span><span mathmlencoding="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot; display=&quot;inline&quot;&gt;&lt;mrow&gt;&lt;mo&gt;&amp;lt;&lt;/mo&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/mrow&gt;&lt;/math&gt;" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAkCAYAAADLsGk3AAACx0lEQVRYR+3XW8hUVRTA8Z+pIYgmKWgpQqBCpoI9iIYvBipeiIIoMaRSJArCQPH+oL4URF5KITAo8akEyUQFhUJ7SysQwSd9lbQss0DKLBasiXH85jvn6MwwfMyBYeDstdde/3Xb6wwyQJ5BA4RDD6TbItmLSC8ibfJAt6XWCGzCHMzAnziPQzjQnw+6CWQ2PsfEJgZ/jecS7h6RbgF5BBfwOz7CZYzFXKzC4LR8L97uC7RbQPbhSSzA7QZDI1JnMBT/YjyuNMK0GiRy/C2MxvqSdR3evoRZuNpkz/tYl2tLcaxdIKMy5O/gUXyKlSVBwuMvYEM/8ovrjF+NT1oNMgZhfOTtSPyG3fm7URIkxB7CnX7kp2f3CpFIv1OtAhmHtXgTw/FrGr8HVQDKss7HSfyBx/L/rr1Va2RC5n6Ed1gC7EIARMdp1xNp9x4+Tufdd/t9AhvxGh7GdezMVtlOgJrB0ZojC6Y2awhFEZmCzXgFQ/BLHcDNdrm/QW80gsNYlhdmn8c2A5mGLXgpCzEAPkBcSJ0CCIOjgVzEfmyrOqJEFL7B47kxUid697cdikDtmLhfvsTPeL3o7GYRiUKOeyCKrDb7HMEO/FCktEXrH2aHipT6p0hnUY3EWLAiJ9JJqeyrBPq+SPkDrEdaxwQc9fF3GT1FIPVhfjkL/6l8eTSBzpU5qILMmpxyl+BWk30xUP5Uv1YWpLYn5J/HVjydL2Pu2Y6zFYxtJhr306tY2Gxcx2S8ixcfBKR+76IEeiZfHk+g7+4TaDmiLt7IUadRTYwxcSGHE+NKiG+X/5+qEenLxnmp/NlcPJGt+8cKQFELX+RdVbTtWgL91WqQmr4ozijSyO3PyrTM3BjT7+mcGIogYj1G+ns+EVoRkcbDZyIu1INlrGqVTDtAWmVbJT09kEru6oBwLyIdcHKlI3oRqeSuDggPmIj8By42biU3iellAAAAAElFTkSuQmCC" width="25" height="18" /></span><span> seconds for a model with more than 10,000 reactions.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Calculating maximal ATP energy production under anaerobic and glucose only conditions:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modelspar = modelalter;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% For Recon3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelspar = changeRxnBounds (modelspar, 'EX_glc_D[e]', -20, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelspar = changeRxnBounds (modelspar, 'EX_o2[e]', 0, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelspar = changeRxnBounds(modelspar, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >,-20,</span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelspar = changeRxnBounds (modelspar, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, 0, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelspar = changeObjective(modelspar, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >[vSparse, sparseRxnBool, essentialRxnBool] = sparseFBA(modelspar, </span><span style="color: rgb(170, 4, 249);">'max'</span><span >);</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="56726BA3" data-testid="output_4" data-width="420" data-height="18" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">---FBA---</div></div></div></div></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Anticipated results:</span></li></ul><div  class = 'S2'><span>Commonly, a sparse FBA solution will have much smaller number of active reactions compared to a standard FBA on the same model with same objective function. The outputs </span><span style=' font-family: monospace;'>sparseRxnBool </span><span>and </span><span style=' font-family: monospace;'>essentialRxnBool</span><span> return vectors with 1 and 0's, with sparse and essential reactions respectively.</span></div><div  class = 'S2'><span>Display the sparse flux solution, but only the non-zero fluxes. This is convenient with the </span><span style=' font-family: monospace;'>printFluxVector</span><span> function.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >nonZeroFlag = 1;</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >printFluxVector(model, vSparse, nonZeroFlag)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement scrollableOutput" uid="4EAC165E" data-testid="output_5" data-width="420" data-height="1039" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">10FTHFtm            	         250
4ABUTtm             	        -250
ABTArm              	        -250
ACONTm              	         250
AKGDm               	         250
ATPtm               	         250
CSm                 	         250
DCK1m               	        -250
FOLt2               	         500
GALt1r              	         250
GLUDxm              	         250
GLYtm               	        -500
GTHDH               	         500
H2CO3Dm             	         250
H2Otm               	         250
HPYRRy              	         500
ICDHxm              	         250
MTHFCm              	        -250
MTHFD2m             	        -250
NH4t3r              	        -750
PCm                 	         250
PDHm                	         250
SUCOASm             	        -250
THFtm               	        -500
TRIOK               	         500
EX_HC00250[e]       	        -500
EX_HC02160[e]       	         250
EX_HC02161[e]       	        -250
r0021               	        -500
r0027               	         250
r0081               	         500
r0160               	         500
r0178               	        -250
r0193               	        -500
r0377               	         250
r0838               	         250
r0940               	        -500
r1411               	         250
r1434               	         500
RE2605C             	         500
RE2898C             	         500
EX_oh1[e]           	        -500
FOLt2le             	        -500
DM_Lcystin          	         250
r0295               	         500
MLTHFtm             	         250
sink_thf[c]         	        1000
DM_atp_c_           	        1000
ENO                 	        1000
EX_h[e]             	         750
EX_h2o[e]           	         250
FBA                 	         250
GAPD                	        1000
GHMT2r              	        -500
MTHFC               	         250
MTHFD               	         250
PGI                 	         250
PGK                 	       -1000
PGM                 	       -1000
PGMT                	         250
TPI                 	         250
UDPG4E              	        -250
EX_nh4[e]           	        -750
GALK                	         250
PFK                 	         250
PYK                 	        1000
r0392               	        -500
UGLT                	         250
MLTHFte3            	        1000
EX_mlthf[e]         	       -1000
HMR_6607            	        -500
HMR_8475            	        -250
DM_4abut[c]         	         250
DCMPtm              	        -250</div></div></div></div></div><h2  class = 'S6'><span style=' text-decoration: underline;'>3. Metabolite dilution flux balance analysis (mdFBA)</span></h2><div  class = 'S2'><span>This is a variant of FBA for predicting metabolic flux distributions by accounting for growth-associated dilution of all metabolites in a context-dependent manner</span><span texencoding="$^3" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAABBUlEQVRIS+3UPyuFYRzG8c/JgFLMIsVLUGSzGLwCL8FGsolNFhMGi5QBkVJKYbMrXoBFNoM/UcpCd93DcXo8z32eYzo993bX9fvW77qufjUtvlqL89oYMI41jOIdJ1jCZ6NnWR6MYR0beMEMZrGFuRTAXhS+1Ylv0YvhIkAHBvDQILzAN6aLAFm16IzAsMp1s4AuHOMMO1n0vCItYh6D+MJyNPcXJw8QTOvHZIw0/ENCN/WE1CpP4QorWC0DCDNP2MdCWcBjXGW7DGAIdxjBcx6gD+e4xCZe0YND7OK0qAfdcc8JfOAoRniA+2Z7kHSsUmP8E1YBtPNZTypREFU9qDz4lx78AKqqKSfCSoPFAAAAAElFTkSuQmCC" width="8" height="19" /></span><span>.</span></div><div  class = 'S2'><span>A solution from the function </span><span style=' font-family: monospace;'>mdFBA</span><span> supports that all metabolites used in any reaction of the solution can either be produced by the network or taken up from the surrounding medium.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>Since this is a MIXED Integer Problem it can take a long time to solve.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Calculating ATP energy production under aerobic condition using mdFBA:</span></div><div  class = 'S2'><span>In this function, there is an optional output </span><span style=' font-family: monospace;'>newActives</span><span>, th</span><span>at represent reactions that are only active in this analysis.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% The valid solution can be produced with the Recon 3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelmd = model;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelmd = changeRxnBounds(modelmd, 'EX_glc_D[e]',-20,'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelmd = changeRxnBounds (modelmd, 'EX_o2[e]', -1000, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelmd = changeObjective(modelmd, 'DM_atp_c_');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% [sol, newActives] = mdFBA(modelmd)</span></span></div></div></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-weight: bold;'>Troubleshooting:</span></li></ul><div  class = 'S2'><span>When a model do</span><span>es not have a feasible solution, add the input:  </span><span style=' font-family: monospace;'>'getInvalidSolution', true</span><span>.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% clear modelmd</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelnosol = modelalter;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelnosol = changeObjective(modelnosol, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >[sol, newActives] = mdFBA(modelnosol,  </span><span style="color: rgb(170, 4, 249);">'getInvalidSolution'</span><span >, true)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="26869E5A" data-testid="output_6" data-width="420" data-height="115" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">sol =

     []


newActives =

  0×0 empty cell array</div></div></div></div></div><div  class = 'S2'><span>Sometimes when an FBA of a model with the same objective function and constraints is run many times, or using different LP logarithm, we may get different set of solutions for individual reactions. In other words, there are different sets of </span><span style=' font-family: monospace;'>'FBAsolution.v'</span><span> values (fluxes of the reactions) and still get the same objective function value </span><span style=' font-family: monospace;'>'f'</span><span>. That is, the opitmal solution vector is not unique. This can create difficulty when investigating the changes to fluxes between two different conditions. In this case a unique solution is required to compare the changes to fluxes.</span></div><div  class = 'S2'><span>This issue can be solved with either of the following the methods</span><span style=' font-family: monospace;'> </span></div><ul  class = 'S4'><li  class = 'S5'><span style=' font-family: monospace;'>geometricFBA</span><span>, which provides a standard, central and reproducible solution, or</span></li><li  class = 'S5'><span style=' font-family: monospace;'>pFBA</span><span>, which provides a solution based on the minimal fluxes through the model, and classify each gene according to how it contributes to the optimal solution.</span></li></ul><h2  class = 'S1'><span style=' text-decoration: underline;'>4. Geometric FBA</span></h2><div  class = 'S2'><span>The geometric FBA solves the smallest frame that contains all sets of optimal FBA solutions and posts a set of multiple linear programming problems</span><span texencoding="$^4" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAAAyUlEQVRIS+3VIW5CQRCA4e8ZJKnF4TkAht6hCeEGVZWEhAPUYEgqEDUk4DgCvgpRh8HBJZBAXrMkzWvDe7uIir7Vs//O/DOTzdx5sjvv+yeABjaYYV50VsXBBGM8pwAeMcRTCqCJJV7xmQJ4D3WfsY0F9NHGFJ1YQCu8PMApBbAK1g+hZVEZvOCIxbd+RwE+0CtZsFFw8xVWHKQuHgqAXGbekTesscP+GlNlEqNK+C37vwfcdFrFQQ0oGcta4s9tjP4qa4m1xHxoLuTYKCfsY5W7AAAAAElFTkSuQmCC" width="8" height="19" /></span><span>.</span></div><div  class = 'S2'><span>This FBA analysis applies iterations, where by each iteration reduces the permissible solution space. After a finite number of iterations, it resolves one single solution of the flux distribution.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% USAGE:</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% flux = geometricFBA(model, varargin)</span></span></div></div></div><div  class = 'S10'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>The time to determine </span><span>a geometric FBA solution depends on the size of the genome-scale model and the number of iterations. For a model with more than 10,000 reactions and several iterations takes </span><span mathmlencoding="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot; display=&quot;inline&quot;&gt;&lt;mrow&gt;&lt;mo&gt;&amp;geq;&lt;/mo&gt;&lt;mn&gt;30&lt;/mn&gt;&lt;/mrow&gt;&lt;/math&gt;" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEEAAAAkCAYAAADWzlesAAAEWElEQVRoQ+3Yd+jucxQH8Nc1soUIZWTv9YeZEdkkETKyyyqr7L1nWSUrO1uhyM4IGYlIsndERNmzt3u+enru8/3d5z7399x71XP++Y3Pfp/3eZ9zvhOMzIQRBkYghAQjJoxAmCgGIyaMQJgxmTAzDsbOWA1/4W1cjdvx92Sy2ZY4DIuVg2fFPbgU37et7RUOs2E+fDWN0+cCeBxr4nP8icUxU93jZuzTcqfMubwAuAgn1Po18CB+xrZ4t9f6XiAsjA9xDbJhLjRsCwMewxc4CR/XgYvWPbavv7fDQz0uczpOwytYrwBopm2EZ/AO1unFiF4gzIvXsBR+ww04Hx8NEYndsS6O7HHGPOWI/IxTju2aszzeQoDcotjUvc2LBcCFOK57sC07zFLUi1cCxh+4FecVouONR7x4Ln5v2fh5rI8ri/Kd067Fgfi1wviXHnvk3sdXWCyCHzrnTC5FBoy9i6JLl1DdWRd+c7yRGGO/N7AqEhaJ8cbi/S+xIJ7Cpi17bIWHayzvuWVKQGjmBoy9cDKWKZW+D2fj1SGDsVxliLuQsOm0VdA4I8J4RMtdonMBK3Yj9hsEhE7kGzCWrX9GqALGC0MAI5nqkRK8E0ujOo/p9PAZiED2suzThEn0bq2pAaETjD2LGfFU7IkCI7QcD9sEl2F13I9TkbDotANwXf3jqKoH2s7+CXPgGyw0HiB0grFHgRGVjj2HeCUpbxDbpTy6ctfi5PqdOmI7w6fgzJoXiofqbZb0m5QbkU8R9Z9NThj7ecTm9egNanLycUC4rZ/FPeYkFabiy4VXxCElipn6KVbCj7UuRVGySiyFVAqqNkvtsUSFRRgxLiAkJycGm8eHqufg7soiA2IwybJkgFB+3xrJz5s6vH99/X44rhjj0KTFAPw+Gj37d/ogTEh9nscnb8dersc/0EdtPygw8dwHSI6/GMfURtt0VJDRjLNaDkh2a2qQZ7HxoEyIEufxKUtj2SxZ4dFBXzaF6+LxxH0q2P1rbWeKvARHt+yZOuLrGksjFh2bonAI2qnoUtbG8ug8PiBMS0uoJU2mbE75HAuT0+hF7VMM5a69rOkfMpYuNV1pXyCk68rj03SkhQ3dc5HQf3rYHdgNG1YGau5wFQ4qsZy/pfRuGqxkhiWrURsThLnxJNYugUulFgXuztHTEoi58B5ex9ZdByeV5m5pp7sBaqaGtRnr2Y63tdKfDblh6gYwnoxYJd7zTaHbUjSlRtis5ZvABRUmyUy7di2OM9NFRhPyfaEpn8dkwuxIrd309NPC499VB5izojlpefMBZAUcijAhneInLZdJ8ZM0muYorXIySL5KpYCLVsyJHfBSr/WDpMhhgBL9iehF7fPgb8sJT+Pe8mQ/5+5YrXajDXl8epswaRIGNBvOKCD088ChzWnThKn9ipTCJl79X1gbCK3U6fNVUfKmu+xzyfSbNgqHAXuH6eeyIZ08YsKICROpNWLCCISJTPgHI/TWJbWrz68AAAAASUVORK5CYII=" width="32.5" height="18" /></span><span>minutes.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Calculating ATP energy production under anaerobic conditions using geometric FBA:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modelgeo = modelalter;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% For Recon3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelgeo = changeRxnBounds (modelgeo, 'EX_glc_D[e]', -20, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%modelgeo = changeRxnBounds (modelgeo, 'EX_o2[e]', 0, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelgeo = changeRxnBounds(modelgeo, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >,-20,</span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelgeo = changeRxnBounds (modelgeo, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, 0, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelgeo = changeObjective(modelgeo, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% WARNING: Depending on the size of the model running this function might take very long; </span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% FBAgeo = geometricFBA (modelgeo, 'flexRel', 1e-3);</span></span></div></div></div><div  class = 'S10'><span>Display the unique fluxe</span><span>s from reactions, that are non-zero in the geometric FBA solution.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% for i=1:length(FBAgeo)</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%     if abs(FBAgeo(i)) &gt; 1e-3</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%         fprintf('%1.3f \t %s\n', FBAgeo(i), modelgeo.rxns{i})</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%     end</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% end</span></span></div></div></div><div  class = 'S10'><span style=' font-weight: bold;'>Troubleshooting:</span></div><div  class = 'S2'><span>When the algorithm has convergence problems, change one of the optional inputs, </span><span style=' font-family: monospace;'>flexRel</span><span>, into e.g. </span><span style=' font-family: monospace;'>1e-3</span><span>. The default is 0 when there is flexibility to flux bounds.</span></div><div  class = 'S2'><span>Enter the optional parameters as parameter name followed by parameter value, for example:  </span></div><div  class = 'S2'><span style=' font-family: monospace;'>flux = geometricFBA(model, 'epsilon', 1e-9)</span></div><h2  class = 'S6'><span style=' text-decoration: underline;'>5. Parsimonious enzyme usage Flux Balance Analysis (pFBA)</span></h2><div  class = 'S2'><span>The pFBA method was developed to achieve higher flux levels when more enzymes are required</span><span texencoding="$^5" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAABDElEQVRIS+3UvyuFURzH8del8BfcyY/JJqmbbLL4J1Ay2UxSV5LFYBEGg0liNBoYyKpkYJIBq1JEGSTLkx5Px3Me7uLWc9bv+b7P+Xw/n3MqGlyVBvs1F6AV/bhIy86TcIrhzIymsF0EUMMWzlKbn7CEtyKAPaziPOZSSEInLpPTDnCTBwkBVjCXarrGPPZDoBCgC0Powxh6k8ZJ7GQhsSC1YRbLeEU3HosMMXvQOmZQ2MYsoAe3WEhu81WPSUiDnrGItb9I6MALRnESA7TgI6NhHHUM4D0PMIJDHGMa94mdu5jAVczGKjYxiHYc4Q4beCgapFj8v9V/40IQXAI02bde2vjDCymj/B+i/Amnfign8dR1WQAAAABJRU5ErkJggg==" width="8" height="19" /></span><span>. </span></div><div  class = 'S2'><span>After performing the FBA to find the optimal value for the objective function, pFBA gets the answer of an another linear program to determine the flux distribution that minimises the total flux through all metabolic reactions in the model.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>The time to determine a pFBA solution depends on the size of the genome-scale model and is taking from </span><span mathmlencoding="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot; display=&quot;inline&quot;&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;&amp;lt;&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/math&gt;" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAkCAYAAADLsGk3AAAB6ElEQVRYR+3Xy6tOURjH8c/hkDgkl5IkJAaUMHJmTBADA7mEQhkQEuU+ceZuMaf8BcglJQMz1z/BXLkPTbT07Hpz2/XutY73fdu79mTtnmc93+f32+syZECeoQHh0IL0mpKtIq0ihTrQy9aajVNYgl11/L0IMgsncRzT8Rib+w3kKFZhCvZG8X0JMgnf+bm/fcXI/1IkWeEIkr9P19mh5vs7LBpvkJk4hhNIHr+Fg/0EMieKTxAz8BnX4v3SDyDzYmk8jGn4FMVfR1OAir+otRaE9w/FypIAriIBpJ8z51MEZDHOYj8m4yOu4EYBgCKKLMN57MEwPnQAfMvZ/j/kyqLISlzADkwIgMu4idIA2RRJKjzD/MiYvL8Vzwsr8Gv6LIqkI0LaB85gYcxwF2N4M05AWUCqWtORYR/OYWkM3gug14WBsoJUtU7EzvjxV8Tg/QB6VQioCEhVazrMbcNFrInBB7iEl5mBioJ01pruCAloNAYfBtCLTEDvMRdPsLEuZ46L1foA2hCTPYql+23d5P/4vhaVZZMyy+N4/9eQHCBV8nUBsAW3caALkN3YhO2Y2hGfVsrUoKexNfyWOidIlXw10oZ6pwuQrkNKgHRdTJPAFqRJ90rEtoqU6GqTnK0iTbpXInZgFPkBYoheJf3KvD4AAAAASUVORK5CYII=" width="25" height="18" /></span><span> minute for a 1,000 reaction model, to 5 minutes for a model with more than 10,000 reactions.</span></div><div  class = 'S2'><span>The function is:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S11'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% [GeneClasses RxnClasses modelIrrevFM] = pFBA(model, varargin)</span></span></div></div></div><div  class = 'S10'><span>Where 'varagin' includes required inputs:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% * 'geneoption' - 0 = minimize the sum of all fluxes in the network,</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                   1 = only minimize the sum of the flux through</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                   gene-associated fluxes (default),</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                   2 = only minimize the sum of the flux through</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                   non-gene-associated fluxes</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% * 'map' - map structure from readCbMap.m (no map written if empty)</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% * 'mapoutname' - File Name for map</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% * 'skipclass' - 0 = classify genes and reactions (default).</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                 1 = Don't classify genes and reactions. Only return</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%                     model with the minimium flux set as upper bound.</span></span></div></div></div><div  class = 'S10'><span>Given outp</span><span>uts in this function are:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% OUTPUTS:</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% GeneClasses:  Structure with fields for each gene class</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% RxnsClasses:  Structure with fields for each reaction class</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modelIrrevFM: Irreversible model used for minimizing flux with</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%               the minimum flux set as a flux upper bound</span></span></div></div></div><div  class = 'S10'><span style=' font-weight: bold;'>Calculating ATP energy production under anaerobic conditions using pFBA:</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modelp = modelalter;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">%Recon3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelp = changeRxnBounds (modelp, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >, -20, </span><span style="color: rgb(170, 4, 249);">'b'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelp = changeRxnBounds (modelp, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, 0, </span><span style="color: rgb(170, 4, 249);">'b'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modelp = changeObjective(modelp, </span><span style="color: rgb(170, 4, 249);">'DM_atp_c_'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >[GeneClasses RxnClasses modelIrrevFM] = pFBA(modelp,</span><span style="color: rgb(14, 0, 255);">...</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >    </span><span style="color: rgb(170, 4, 249);">'geneoption'</span><span >, 0, </span><span style="color: rgb(170, 4, 249);">'skipclass'</span><span >, 1)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="AEEE734E" data-testid="output_7" data-width="420" data-height="143" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">netFlux	fluxMeasure 	-&gt;	

GeneClasses =

     []


RxnClasses =

     []</div></div><div class="inlineElement eoOutputWrapper embeddedOutputsVariableStringElement scrollableOutput" uid="ABDDD854" data-testid="output_8" data-width="420" data-height="580" data-hashorizontaloverflow="true" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><span class="variableNameElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">modelIrrevFM = <span class="headerElement" style="white-space: normal; font-style: italic; color: rgb(179, 179, 179); font-size: 12px;">struct with fields:</span></span></div><div style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">                      S: [5836×14226 double]
                   mets: {5836×1 cell}
                      b: [5836×1 double]
                 csense: [5836×1 char]
                   rxns: {14226×1 cell}
                     lb: [14226×1 double]
                     ub: [14226×1 double]
                      c: [14226×1 double]
                 osense: -1
                  genes: {2248×1 cell}
                  rules: {14226×1 cell}
             metCharges: [5836×1 int64]
            metFormulas: {5836×1 cell}
              metSmiles: {5836×1 cell}
               metNames: {5836×1 cell}
              metHMDBID: {5836×1 cell}
         metInChIString: {5836×1 cell}
              metKEGGID: {5836×1 cell}
           metPubChemID: {5836×1 cell}
            description: 'Recon3DModel.mat'
                grRules: {14226×1 cell}
             rxnGeneMat: [14226×2248 double]
    rxnConfidenceScores: [14226×1 double]
               rxnNames: {14226×1 cell}
               rxnNotes: {14226×1 cell}
           rxnECNumbers: {14226×1 cell}
          rxnReferences: {14226×1 cell}
              rxnKEGGID: {14226×1 cell}
             subSystems: {14226×1 cell}
             metCHEBIID: {5836×1 cell}
               metPdMap: {5836×1 cell}
            metReconMap: {5836×1 cell}
                modelID: 'Recon3DModel'
                 rxnCOG: {14226×1 cell}
       rxnKeggOrthology: {14226×1 cell}
            rxnReconMap: {14226×1 cell}
                version: 'Recon3D_01'
             PleaseCite: 'Brunk et al, Nat Biotech, 2018; doi:10.1038/nbt.4072'
                  match: [14226×1 double]
        reversibleModel: 0
</div></div></div></div></div></div><div  class = 'S10'><span>Display minimal fluxes of the r</span><span>eactions that are required for producing energy only from only glucose media. This is convenient with the </span><span style=' font-family: monospace;'>printFluxVector</span><span> function.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >nonZeroFlag = 1;</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >printFluxVector(modelIrrevFM, modelIrrevFM.lb, nonZeroFlag)</span></span></div><div  class = 'S13'><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="83DBC2EA" data-testid="output_9" data-width="420" data-height="45" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">DM_atp_c_           	        1000
EX_glc_D[e]_r       	          20
netFlux             	   3.192e+04</div></div></div></div></div><h2  class = 'S6'><span style=' text-decoration: underline;'>6. Dynamic FBA</span></h2><div  class = 'S2'><span>The dynamic FBA is an extension of standard FBA that accounts for cell culture dynamics, implementing both dynamic (nonlinear programming) and static (LP) optimisation of an objective function and applying constraints to the rates of change of flux in addition to the standard FBA constraints</span><span texencoding="$^6" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAABNklEQVRIS+3UvStFYQDH8c9NNotIXqJYZBF/gJLERmYGk2SzEfJWysbGqEiZZBCLyaKIIpmUJIvBaJD06Ny6Hefc62a891lOneecb8/z/f2eJ+OfI/PP/5UIYABD+MQ1dnK95XPQjD3UYBQ3ScLTAO04wxP68JGWVhKgCleoRRee80WdBJjHKtYxU6gnSYAXNGIW/WjFYyRvNw6MAzpwjy9MYh91WMYY5rCWL4UQ2SmOMJzzYUUktBpNeM/OxVcwiBNsYDq23G1MoAfnaYAQ30O03/EYYBFLGMFhGqASr7hDbwywggV04jYNEN5vYgoNeMuBHKAbQXSo9c9IirE+6nyQmd1GGy6jJI7/chZaoiKFNoYzEJ5buCjUg0LF+zVfIjdSXi9lB8lVLqpMZYlliWl3YlFF+gYwXTAnE8GsqgAAAABJRU5ErkJggg==" width="8" height="19" /></span><span>.</span></div><div  class = 'S2'><span>The dynamic FBA metho</span><span>d implemented in this function is essentially the same as the method described by Varma A. and B. O. Palsson</span><span texencoding="$^7" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAmCAYAAADN0BvRAAAA60lEQVRIS+3UP0tCcRTG8Y80i0uEYAiCL8CgMQh8DW01NGS+gqaIoFdQc269Bl2ElqAlaGkpWlqUJsGtrbjgIDfv/cnvbnLPes75cv48PBUFo1Kw34YC7nGWcZspGsu59A2qeMcjPvC7VHyCJ5znAXqYYJSaYAvfOMUwD1DDfMX4h4vGbfzkAbJkcYtdHKUL1tXBF67wEAPYwwt2MIsB3OAA3VX7rbPCGwa4iwG08YkWkjv8i9AEFzhGJ+s9IcAzxriOAdSRaH8frzGAPi7RzDOd0ApBwyoBNtWVg7+PsbRMaCmkUkiJOArr4A++0B4nLmMU/QAAAABJRU5ErkJggg==" width="8" height="19" /></span><span>.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modeldinamic = model;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% For Recon3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc_D[e]', -20, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2[e]', -1000, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac[e]', -1000, 'l');</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modeldinamic = changeRxnBounds (modeldinamic, </span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]'</span><span >, -20, </span><span style="color: rgb(170, 4, 249);">'b'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modeldinamic = changeRxnBounds (modeldinamic, </span><span style="color: rgb(170, 4, 249);">'EX_o2[e]'</span><span >, -1000, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >modeldinamic = changeRxnBounds (modeldinamic, </span><span style="color: rgb(170, 4, 249);">'EX_ac[e]'</span><span >, -1000, </span><span style="color: rgb(170, 4, 249);">'l'</span><span >);</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% For Recon3.0 model</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% smi = {'EX_glc_D[e]' 'EX_ac[e]'}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >smi = {</span><span style="color: rgb(170, 4, 249);">'EX_glc_D[e]' 'EX_ac[e]'</span><span >}; </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% exchange reaction for substrate in environment</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >smc = [10.8]; </span><span style="color: rgb(2, 128, 9);">% Glucose, Acetate concentration (all in mM)</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >Xec = 0.001; </span><span style="color: rgb(2, 128, 9);">% initial biomass</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >dt = 1.0/1000.0; </span><span style="color: rgb(2, 128, 9);">% time steps</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >time = 1.0/dt; </span><span style="color: rgb(2, 128, 9);">% simulation time</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >[concentrationMatrix, excRxnNames, timeVec,</span><span style="color: rgb(14, 0, 255);">...</span></span></div></div><div class="inlineWrapper outputs"><div  class = 'S15'><span style="white-space: normal"><span >    biomassVec] = dynamicFBA(modeldinamic, smi, smc, Xec, dt, time, smi ); </span></span></div><div  class = 'S13'><div class='variableElement' style='font-family: Menlo, Monaco, Consolas, "Courier New", monospace; font-size: 12px; '>ans = 1806</div><div class="inlineElement eoOutputWrapper embeddedOutputsTextElement" uid="90C81AC1" data-testid="output_11" data-width="420" data-height="45" data-hashorizontaloverflow="false" style="width: 450px; max-height: 261px; white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;"><div class="textElement" style="white-space: normal; font-style: normal; color: rgb(64, 64, 64); font-size: 12px;">Step number	Biomass
Dynamic FBA analysis in progress ...
0%      [                                        ]</div></div></div></div></div><h2  class = 'S6'><span style=' text-decoration: underline;'>7. Relaxed FBA</span></h2><div  class = 'S2'><span>Find the minimal set of </span><span>relaxations on bounds and steady-state constraint to make an infeasible FBA problem feasible. </span></div><div  class = 'S2'><span>See tutorial_relaxedFBA.mlx</span></div><h2  class = 'S6'><span style=' text-decoration: underline;'>8. Flux enrichment analysis (FEA)</span></h2><div  class = 'S2'><span>The flux enrichment analysis calculates the likelihood th</span><span>at a set of fluxes would belong to a subsystem or pathway.</span></div><div  class = 'S2'><span style=' font-weight: bold;'>Timing:</span></div><div  class = 'S2'><span>The time to calcu</span><span>late the FEA is </span><span mathmlencoding="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot; display=&quot;inline&quot;&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;&amp;lt;&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/math&gt;" style="vertical-align:-5px"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAkCAYAAADLsGk3AAAB6ElEQVRYR+3Xy6tOURjH8c/hkDgkl5IkJAaUMHJmTBADA7mEQhkQEuU+ceZuMaf8BcglJQMz1z/BXLkPTbT07Hpz2/XutY73fdu79mTtnmc93+f32+syZECeoQHh0IL0mpKtIq0ihTrQy9aajVNYgl11/L0IMgsncRzT8Rib+w3kKFZhCvZG8X0JMgnf+bm/fcXI/1IkWeEIkr9P19mh5vs7LBpvkJk4hhNIHr+Fg/0EMieKTxAz8BnX4v3SDyDzYmk8jGn4FMVfR1OAir+otRaE9w/FypIAriIBpJ8z51MEZDHOYj8m4yOu4EYBgCKKLMN57MEwPnQAfMvZ/j/kyqLISlzADkwIgMu4idIA2RRJKjzD/MiYvL8Vzwsr8Gv6LIqkI0LaB85gYcxwF2N4M05AWUCqWtORYR/OYWkM3gug14WBsoJUtU7EzvjxV8Tg/QB6VQioCEhVazrMbcNFrInBB7iEl5mBioJ01pruCAloNAYfBtCLTEDvMRdPsLEuZ46L1foA2hCTPYql+23d5P/4vhaVZZMyy+N4/9eQHCBV8nUBsAW3caALkN3YhO2Y2hGfVsrUoKexNfyWOidIlXw10oZ6pwuQrkNKgHRdTJPAFqRJ90rEtoqU6GqTnK0iTbpXInZgFPkBYoheJf3KvD4AAAAASUVORK5CYII=" width="25" height="18" /></span><span> second for any size of a model.</span></div><div class="CodeBlock"><div class="inlineWrapper"><div  class = 'S7'><span style="white-space: normal"><span >modelfea = model;</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >res = optimizeCbModel(modelfea,</span><span style="color: rgb(170, 4, 249);">'max'</span><span >); </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% say you are interested in enriching the active reactions</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span >activeReactions = find(res.x) </span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% You can also look for e.g. positive/negative/zeros flux reactions,</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% that depends pretty much on the question.</span></span></div></div><div class="inlineWrapper"><div  class = 'S8'><span style="white-space: normal"><span style="color: rgb(2, 128, 9);">% Now you look for the enrichement of reactions per subsystems</span></span></div></div><div class="inlineWrapper"><div  class = 'S9'><span style="white-space: normal"><span >resultCell = FEA(modelfea, activeReactions, </span><span style="color: rgb(170, 4, 249);">'subSystems'</span><span >)</span></span></div></div></div><h2  class = 'S6'><span>REFERENCES </span></h2><div  class = 'S2'><span>[1] Orth, J. D., Thiele I., and Palsson, B. Ø.  What is flux balance analysis? </span><span style=' font-style: italic;'>Nat. Biotechnol., </span><span>28(3), 245–248 (2010).</span></div><div  class = 'S2'><span>[2] Thiele, I., et al. </span><span>A community-driven global reconstruction of human metabolism. </span><span style=' font-style: italic;'>Nat. Biotechnol., </span><span>31(5), 419–425 (2013).</span></div><div  class = 'S2'><span>[3] Benyamini, T, Folger, O., Ruppin, E., Schlomi, T. Flux balance analysis accounting for metabolite dilution.</span><span style=' font-style: italic;'> Genome Biology</span><span>., 11(4):R43 (2010).</span></div><div  class = 'S2'><span>[4] Smallbone, K., and Simeonidis, E. Flux balance analysis: A geometric perspective. </span><span style=' font-style: italic;'>J Theor Biol</span><span>., 258: 311-315 (2009).</span></div><div  class = 'S2'><span>[5] Lewis, N.E., et al. Omic data from evolved E. coli are consistent with computed optimal growth from genome-scale models. </span><span style=' font-style: italic;'>Mol Syst Biol</span><span>., 6:390 (2010).</span></div><div  class = 'S2'><span>[6] Mahadevan, R., Edwards, J.S., Doyle, F.J. Dynamic Flux Balance Analysis of Diauxic Growth in Escherichia coli. </span><span style=' font-style: italic;'>Biophys J., </span><span>83(3):1331-1340 (2002).</span></div><div  class = 'S2'><span>[7] Varma A. and Palsson, B. Ø. Stoichiometric flux balance models quantitatively predict growth and metabolic by-product secretion in wild-type Escherichia coli W3110. </span><span style=' font-style: italic;'>App Environ Microbiol</span><span>., 60(10):3724-3731 (1994).</span></div><div  class = 'S2'><span>[8]. Laurent Heirendt &amp; Sylvain Arreckx, Thomas Pfau, Sebastian N. Mendoza, Anne Richelle, Almut Heinken, Hulda S. Haraldsdottir, Jacek Wachowiak, Sarah M. Keating, Vanja Vlasov, Stefania Magnusdottir, Chiam Yu Ng, German Preciat, Alise Zagare, Siu H.J. Chan, Maike K. Aurich, Catherine M. Clancy, Jennifer Modamio, John T. Sauls, Alberto Noronha, Aarash Bordbar, Benjamin Cousins, Diana C. El Assal, Luis V. Valcarcel, Inigo Apaolaza, Susan Ghaderi, Masoud Ahookhosh, Marouen Ben Guebila, Andrejs Kostromins, Nicolas Sompairac, Hoai M. Le, Ding Ma, Yuekai Sun, Lin Wang, James T. Yurkovich, Miguel A.P. Oliveira, Phan T. Vuong, Lemmer P. El Assal, Inna Kuperstein, Andrei Zinovyev, H. Scott Hinton, William A. Bryant, Francisco J. Aragon Artacho, Francisco J. Planes, Egils Stalidzans, Alejandro Maass, Santosh Vempala, Michael Hucka, Michael A. Saunders, Costas D. Maranas, Nathan E. Lewis, Thomas Sauter, Bernhard Ø. Palsson, Ines Thiele, Ronan M.T. Fleming, </span><span style=' font-weight: bold;'>Creation and analysis of biochemical constraint-based models: the COBRA Toolbox v3.0</span><span>, Nature Protocols, volume 14, pages 639–702, 2019 </span><a href = "https://doi.org/10.1038/s41596-018-0098-2"><span>doi.org/10.1038/s41596-018-0098-2</span></a><span>.</span></div><div  class = 'S2'></div>
<br>
<!-- 
##### SOURCE BEGIN #####
%% Flux Balance Analysis (FBA) and its variants
%% Author(s): *Ronan M.T. Fleming,* Leiden University;*Vanja Vlasov, LCSB, University of Luxembourg; Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg* 
%% Reviewer(s): Ines Thiele, *Catherine Clancy, National University of Ireland, Galway. Thomas Pfau, Systems Biology Group, LSRU, University of Luxembourg.* 
%% INTRODUCTION
% Flux balance analysis (FBA) evaluates the metabolic flux distribution$$^1$, 
% and is one of the most used modelling approaches for metabolic systems. 
% 
% The applications of FBA for molecular systems biology include prediction of 
% the growth rates, uptake rates, knockout lethality and product secretion. In 
% FBA, the solution space is constrained by the assumption of a steady-state, 
% under which each internal metabolite is consumed at the same rate as it is produced.
% 
% For the quantitative estimation of the metabolic fluxes, linear programming 
% (LP) can be used to solve the stoichiometric matrix for a given objective function 
% under different constraints. The constraints of the problem depict the space 
% of all eligible possibilities from which an optimal solution can be selected; 
% 
% $$\begin{array}{ll}\min\limits _{v} & \ c^{T}v\\\text{s.t.} & Sv=b,\\ & l\leq 
% v\leq u,\end{array}$$
% 
% Equation 1: Formula of standard FBA.
% 
% where $$c\in\Re^{n}$$ is a parameter vector that linearly combines one or 
% more reaction fluxes to form what is termed the objective function,  and where 
% a $$b_{i}<0$$, or  $$b_{i}>0$$, represents some fixed output, or input, of the 
% ith molecular species. $$S\in\Re^{m\times n}$$ is a stoichiometric matrix for 
% $$m$ molecular species and $$n$ reactions, and $$b$ is a vector of known metabolic 
% exchanges. The output of FBA is a particular flux distribution, $$v$ , which 
% maximises or minimises the objective function and stands between upper and lower 
% bounds, $$u$ and $$l$, respectively.
% 
% There are multiple different variants of FBA, some of which will be demonstrated 
% here:
%% 
% # *Standard FBA*
% # *Regularised FBA*
% # *Sparse FBA*
% # *Metabolite dilution FBA (mdFBA)*
% # *Geometric FBA*
% # *Parsimonious enzyme usage Flux Balance Analysis (pFBA)*
% # *Dynamic FBA*
% # *Relaxed FBA*
% # *Flux enrichment analysis (FEA)*
%% EQUIPMENT SETUP
%% Initialise The Cobra Toolbox and set the solver.
% If necessary, initialise the cobra toolbox:

 if 0
     initCobraToolbox(false) % false, as we don't want to update
 end
%% 
% For solving LP problems in a FBA analysis, certain solvers are required and 
% can be set using the |changeCobraSolver| function:

% solverOK = changeCobraSolver(solverName, solverType, printLevel, unchecked)
%% 
% The present tutorial can run with the GLPK package, which does not require 
% additional installation and configuration. Although, for the analysis of large 
% models is recommended to use the GUROBI package.
% 
% Setup the appropriate solver for the machine you are using by removing the 
% "%" (comment) sign for only the desired solver.

 changeCobraSolver('glpk','all');
% changeCobraSolver('tomlab_cplex','all');
% changeCobraASolver('ibm_cplex','all');
% changeCobraSolver ('gurobi', 'all');
%% Model Setup
% This tutorial will use the generic model of the human cellular metabolism$$ 
% ^2$, Recon 2.0. Other COBRA models, including Recon 3, may also be run with 
% this tutorial. For information on metabolites structures and reactions, and 
% to download the latest COBRA model releases, visit the Virtual Metabolic Human 
% database (VMH, <http://vmh.life). http://vmh.life).>
% 
% Before proceeding with the simulations, load the model into the workspace:

global CBTDIR
%modelFileName = 'Recon2.0model.mat';
modelFileName = 'Recon3DModel_301.mat';
modelDirectory = getDistributedModelFolder(modelFileName); %Look up the folder for the distributed Models.
modelFileName= [modelDirectory filesep modelFileName]; % Get the full path. Necessary to be sure, that the right model is loaded
if 0
    model = readCbModel(modelFileName);
else
    load('Recon3DModel_301.mat')
    model = Recon3DModel;
end
%% 
% In this tutorial we assume, that the cellular objectives include energy production 
% or optimisation of uptake rates and by-product secretion for various physiological 
% functions of the human body.
%% PROCEDURE
%% 1. Standard FBA
% Standard FBA predicts an optimal solution for a cellular objective within 
% a given set of constraints on a metabolic network (see Equation 1). Constraints 
% on the network are set by assigning limits on the uptake, consumption or production 
% of metabolites in reactions.
% 
% *Timing:*
% 
% The time to determine a FBA solution depends on the size of the genome-scale 
% model and is commonly less than a second for a medium sized model.
% 
% *Calculating maximal ATP energy production under aerobic conditions:*
% 
% For each new simulation, the original model will be copied to a new variable. 
% This preserves the constraints of the original model to perform further simulations 
% with new constraints. Additionally, this method of renaming the model avoids 
% confusion while performing multiple simulations at the same time.

modelaerobic = model;
%% 
% The ATP demand reaction, i.e., <http://vmh.life/#human/all/DM_atp_c_ |DM_atp_c_|> 
% within the model is a reaction that involves hydrolysis of ATP to ADP, Pi and 
% proton in the cytosol. 

 printRxnFormula(model, 'DM_atp_c_');
%% 
% We will set this reaction as our objective with the |'changeObjective'| command. 
% Maximising the flux through the ATP demand reaction will result in the network 
% producing a maximal amount of ATP (up to the limit of the reaction).

modelaerobic = changeObjective (modelaerobic, 'DM_atp_c_');
%% 
% The glucose and oxygen, in this case, are provided in high amounts for calculating 
% the flux through ATP demand. 
% 
% The |'changeRxnBounds'| function changes the flux constraints of the lower 
% ('|l|'), upper ('|u|'), or both the bounds ('|b|'), of the specified reaction. 
% Here, we will change the maximal uptake of glucose to 20 $\mu$mol/min/gDW and 
% of oxygen to 1000 $\mu$mol/min/gDW. The uptake of oxygen is effectively unconstrainted 
% (i.e. infinity). 

% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and 
%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.
modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l');  
modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); 
%% 
% The function |optimizeCbModel| calculates one of the optimal solutions for 
% a (maximum or minimum) objective reaction within the defined solution space. 
% In the above example, the maximal flux through the <http://vmh.life/#human/all/DM_atp_c_ 
% |DM_atp_c_|> is desired. 

FBAaerobic = optimizeCbModel (modelaerobic, 'max')
%% 
% * *Anticipated results*
%% 
% When oxygen and all carbon sources (internal and external) are provided the 
% flux through ATP demand reaction can reach its maximum rate of 1000 $\mu$mol/min/gDW. 
%% 
% * *Troubleshooting*
%% 
% If there are multiple carbon sources available in the model, it may be necessary 
% to specify more constraints in order to examine the effect of a single carbon 
% source on ATP production.  
% 
% To avoid this issue, all external carbon sources need to be closed with the 
% exception of the single carbon source of interest.

%Closing the uptake of all energy and oxygen sources
[exchBool,uptBool] = findExcRxns(model);
uptakes = model.rxns(uptBool);

% If you use Recon3.0 model, then:
% modelalter = model;
% modelalter = changeRxnBounds(modelalter, uptakes, 0, 'b');
% modelalter = changeRxnBounds(modelalter, 'EX_HC00250[e]', -1000, 'l');

% The alternative way to do that, in case you were using another large model, 
% that does not contain defined Subsystem is
% to find uptake exchange reactions with following codes:
% [selExc, selUpt] = findExcRxns(model);
% uptakes1 = model.rxns(selUpt);

% Selecting from the exchange uptake reactions those 
% which contain at least 1 carbon in the metabolites included in the reaction:
 subuptakeModel = extractSubNetwork(model, uptakes);
 hiCarbonRxns = findCarbonRxns(subuptakeModel,1);
% Closing the uptake of all the carbon sources
 modelalter = model;
 modelalter = changeRxnBounds(modelalter, hiCarbonRxns, 0, 'b');
% Closing other oxygen and energy sources. Use the following lines for recon2, or uncomment the lines below for recon3
 exoxygen = {'EX_adp[e]',   'EX_amp[e]',    'EX_atp[e]',    'EX_co2[e]',    'EX_coa[e]',    'EX_fad[e]',    'EX_fe2[e]',...
    'EX_fe3[e]',    'EX_gdp[e]',   'EX_gmp[e]',    'EX_gtp[e]',    'EX_h[e]',    'EX_h2o[e]',    'EX_h2o2[e]',...
    'EX_nad[e]',    'EX_nadp[e]',    'EX_no[e]',    'EX_no2[e]',    'EX_o2s[e]'};
modelalter = changeRxnBounds (modelalter, exoxygen, 0, 'l');
%% 
% *Calculating maximum ATP energy production under anaerobic and glucose only 
% conditions:*

modelanaerobic = modelalter;
% modelaerobic = changeRxnBounds (modelaerobic, 'EX_glc_D[e]', -20, 'l'); % For Recon 3.0 uncomment these lines and 
%modelaerobic = changeRxnBounds (modelaerobic, 'EX_o2[e]', -1000, 'l'); % comment the lines below.
modelanaerobic = changeRxnBounds(modelanaerobic, 'EX_glc_D[e]',-20,'l');
modelanaerobic = changeRxnBounds (modelanaerobic, 'EX_o2[e]', 0, 'l');
modelanaerobic = changeObjective(modelanaerobic,'DM_atp_c_');
FBAanaerob = optimizeCbModel(modelanaerobic,'max')
%% 
% * *Anticipated results*
%% 
% Compared to the aerobic condition, anaerobic condition with only glucose as 
% an energy source has reduced flux through ATP demand (82 $\mu$mol/min/gDW), 
% signifying the need to oxygen to run the oxidative phosphorylation. The results 
% are dependant on the model you are using. For Recon 3.0, under anaerobic conditions 
% with only glucose as an energy source, the flux for ATP demand is 40 $\mu$mol/min/gDW.
%% 2. Quadratically regularised FBA
% Regularised FBA calculates the optimal solution of a linear objective function, 
% and finds the unique optimal flux vector that minimises the Euclidean norm of 
% the flux vector, that is
% 
% $$\begin{array}{ll}\min\limits _{v} & \ c^{T}v + \eta ||v|| \\\text{s.t.} 
% & Sv=b,\\ & l\leq v\leq u,\\\end{array}$$
% 
% If the scalar parameter $\eta$ is small, then $c^Tv$ dominates and the same 
% optimal objective as in FBA is attained, with the advantage that a unique flux 
% vector is returned - one that minimises the the sum of the squares of the flux 
% vector.
% 
% In practice, for genome-scale, mono-scale models (e.g. not ME models) the 
% following value for the scalar parameter works fine

eta = 1e-6;
%% 
% This is a quadratic optimisation problem, so a 'QP' solver is required. The 
% COBRA Toolbox v3 comes shipped with the solver PDCO: Primal-Dual interior method 
% for Convex Objectives (further info here: <https://web.stanford.edu/group/SOL/software/pdco/ 
% link>).

[solverOK, solverInstalled] = changeCobraSolver('pdco','QP');
%% 
% Solve a quadratically regularised FBA problem:

FBAsolution = optimizeCbModel(model,'max',eta);
%% 2. Sparse FBA
% Sparse FBA calculates the optimal solution of a linear objective function 
% and finds the smallest set of reactions that can carry flux to achieve the objective. 
% Sparse FBA minimises the number of active reactions by keeping same maximal 
% objective;
% 
% $$\begin{array}{ll}\min\limits _{v} & \ ||v||_0 \\\text{s.t.} & Sv=b,\\ & 
% l\leq v\leq u,\\& c^{T}v=\rho^*\end{array}$$
% 
% Equation 2: Formula of Sparse FBA.
% 
% where the last constraint is optional and represents the requirement to satisfy 
% an optimal objective value $\rho^*$ derived from any solution to a FBA problem. 
% This approach is used to check for minimal sets of reactions that either should 
% be active or should not be active in a flux balance model that is representative 
% of a biochemical network.

% [vSparse, sparseRxnBool, essentialRxnBool]  = sparseFBA(model, osenseStr,...
%  checkMinimalSet, checkEssentialSet, zeroNormApprox)
%% 
% As an optional input, there are different appoximation types of zero-norm 
% (only available when |minNorm = 'zero'|). Default is |cappedL1|.

% Other types of zero-norm:
%  * 'cappedL1' : Capped-L1 norm
%  * 'exp'      : Exponential function
%  * 'log'      : Logarithmic function
%  * 'SCAD'     : SCAD function
%  * 'lp-'      : :math:`L_p` norm with :math:`p < 0`
%  * 'lp+'      : :math:`L_p` norm with :math:`0 < p < 1`
%  * 'l1'       : L1 norm
%  * 'all'      : try all approximations and return the best result
%% 
% *Timing:*
% 
% The time to determine a |sparseFBA()| solution depends on the size of the 
% genome-scale model and is taking from $<1$ second for a 1,000 reaction model, 
% to $<2$ seconds for a model with more than 10,000 reactions.
% 
% *Calculating maximal ATP energy production under anaerobic and glucose only 
% conditions:*

modelspar = modelalter;
% For Recon3.0 model
% modelspar = changeRxnBounds (modelspar, 'EX_glc_D[e]', -20, 'l');
% modelspar = changeRxnBounds (modelspar, 'EX_o2[e]', 0, 'l');
modelspar = changeRxnBounds(modelspar, 'EX_glc_D[e]',-20,'l');
modelspar = changeRxnBounds (modelspar, 'EX_o2[e]', 0, 'l');
modelspar = changeObjective(modelspar, 'DM_atp_c_');
[vSparse, sparseRxnBool, essentialRxnBool] = sparseFBA(modelspar, 'max');
%% 
% * *Anticipated results:*
%% 
% Commonly, a sparse FBA solution will have much smaller number of active reactions 
% compared to a standard FBA on the same model with same objective function. The 
% outputs |sparseRxnBool| and |essentialRxnBool| return vectors with 1 and 0's, 
% with sparse and essential reactions respectively.
% 
% Display the sparse flux solution, but only the non-zero fluxes. This is convenient 
% with the |printFluxVector| function.

nonZeroFlag = 1;
printFluxVector(model, vSparse, nonZeroFlag)
%% 3. Metabolite dilution flux balance analysis (mdFBA)
% This is a variant of FBA for predicting metabolic flux distributions by accounting 
% for growth-associated dilution of all metabolites in a context-dependent manner$$^3$.
% 
% A solution from the function |mdFBA| supports that all metabolites used in 
% any reaction of the solution can either be produced by the network or taken 
% up from the surrounding medium.
% 
% *Timing:*
% 
% Since this is a MIXED Integer Problem it can take a long time to solve.
% 
% *Calculating ATP energy production under aerobic condition using mdFBA:*
% 
% In this function, there is an optional output |newActives|, that represent 
% reactions that are only active in this analysis.

% The valid solution can be produced with the Recon 3.0 model
% modelmd = model;
% modelmd = changeRxnBounds(modelmd, 'EX_glc_D[e]',-20,'l');
% modelmd = changeRxnBounds (modelmd, 'EX_o2[e]', -1000, 'l');
% modelmd = changeObjective(modelmd, 'DM_atp_c_');

% [sol, newActives] = mdFBA(modelmd)
%% 
% * *Troubleshooting:*
%% 
% When a model does not have a feasible solution, add the input:  |'getInvalidSolution', 
% true|.

% clear modelmd
modelnosol = modelalter;
modelnosol = changeObjective(modelnosol, 'DM_atp_c_');
[sol, newActives] = mdFBA(modelnosol,  'getInvalidSolution', true)
%% 
% Sometimes when an FBA of a model with the same objective function and constraints 
% is run many times, or using different LP logarithm, we may get different set 
% of solutions for individual reactions. In other words, there are different sets 
% of |'FBAsolution.v'| values (fluxes of the reactions) and still get the same 
% objective function value |'f'|. That is, the opitmal solution vector is not 
% unique. This can create difficulty when investigating the changes to fluxes 
% between two different conditions. In this case a unique solution is required 
% to compare the changes to fluxes.
% 
% This issue can be solved with either of the following the methods 
%% 
% * |geometricFBA|, which provides a standard, central and reproducible solution, 
% or
% * |pFBA|, which provides a solution based on the minimal fluxes through the 
% model, and classify each gene according to how it contributes to the optimal 
% solution.
%% 4. Geometric FBA
% The geometric FBA solves the smallest frame that contains all sets of optimal 
% FBA solutions and posts a set of multiple linear programming problems$$^4$.
% 
% This FBA analysis applies iterations, where by each iteration reduces the 
% permissible solution space. After a finite number of iterations, it resolves 
% one single solution of the flux distribution.

% USAGE:
% flux = geometricFBA(model, varargin)
%% 
% *Timing:*
% 
% The time to determine a geometric FBA solution depends on the size of the 
% genome-scale model and the number of iterations. For a model with more than 
% 10,000 reactions and several iterations takes $\ge 30$minutes.
% 
% *Calculating ATP energy production under anaerobic conditions using geometric 
% FBA:*

modelgeo = modelalter;
% For Recon3.0 model
% modelgeo = changeRxnBounds (modelgeo, 'EX_glc_D[e]', -20, 'l');
%modelgeo = changeRxnBounds (modelgeo, 'EX_o2[e]', 0, 'l');
modelgeo = changeRxnBounds(modelgeo, 'EX_glc_D[e]',-20,'l');
modelgeo = changeRxnBounds (modelgeo, 'EX_o2[e]', 0, 'l');
modelgeo = changeObjective(modelgeo, 'DM_atp_c_');
% WARNING: Depending on the size of the model running this function might take very long; 
% FBAgeo = geometricFBA (modelgeo, 'flexRel', 1e-3);
%% 
% Display the unique fluxes from reactions, that are non-zero in the geometric 
% FBA solution.

% for i=1:length(FBAgeo)
%     if abs(FBAgeo(i)) > 1e-3
%         fprintf('%1.3f \t %s\n', FBAgeo(i), modelgeo.rxns{i})
%     end
% end
%% 
% *Troubleshooting:*
% 
% When the algorithm has convergence problems, change one of the optional inputs, 
% |flexRel|, into e.g. |1e-3|. The default is 0 when there is flexibility to flux 
% bounds.
% 
% Enter the optional parameters as parameter name followed by parameter value, 
% for example:  
% 
% |flux = geometricFBA(model, 'epsilon', 1e-9)|
%% 5. Parsimonious enzyme usage Flux Balance Analysis (pFBA)
% The pFBA method was developed to achieve higher flux levels when more enzymes 
% are required$$^5$. 
% 
% After performing the FBA to find the optimal value for the objective function, 
% pFBA gets the answer of an another linear program to determine the flux distribution 
% that minimises the total flux through all metabolic reactions in the model.
% 
% *Timing:*
% 
% The time to determine a pFBA solution depends on the size of the genome-scale 
% model and is taking from $<1$ minute for a 1,000 reaction model, to 5 minutes 
% for a model with more than 10,000 reactions.
% 
% The function is:

% [GeneClasses RxnClasses modelIrrevFM] = pFBA(model, varargin)
%% 
% Where 'varagin' includes required inputs:

% * 'geneoption' - 0 = minimize the sum of all fluxes in the network,
%                   1 = only minimize the sum of the flux through
%                   gene-associated fluxes (default),
%                   2 = only minimize the sum of the flux through
%                   non-gene-associated fluxes
%
% * 'map' - map structure from readCbMap.m (no map written if empty)
%
% * 'mapoutname' - File Name for map
%
% * 'skipclass' - 0 = classify genes and reactions (default).
%                 1 = Don't classify genes and reactions. Only return
%                     model with the minimium flux set as upper bound.
%% 
% Given outputs in this function are:

% OUTPUTS:
% GeneClasses:  Structure with fields for each gene class
% RxnsClasses:  Structure with fields for each reaction class
% modelIrrevFM: Irreversible model used for minimizing flux with
%               the minimum flux set as a flux upper bound
%% 
% *Calculating ATP energy production under anaerobic conditions using pFBA:*

modelp = modelalter;
%Recon3.0 model
modelp = changeRxnBounds (modelp, 'EX_glc_D[e]', -20, 'b');
modelp = changeRxnBounds (modelp, 'EX_o2[e]', 0, 'b');
modelp = changeObjective(modelp, 'DM_atp_c_');
[GeneClasses RxnClasses modelIrrevFM] = pFBA(modelp,...
    'geneoption', 0, 'skipclass', 1)
%% 
% Display minimal fluxes of the reactions that are required for producing energy 
% only from only glucose media. This is convenient with the |printFluxVector| 
% function.

nonZeroFlag = 1;
printFluxVector(modelIrrevFM, modelIrrevFM.lb, nonZeroFlag)
%% 6. Dynamic FBA
% The dynamic FBA is an extension of standard FBA that accounts for cell culture 
% dynamics, implementing both dynamic (nonlinear programming) and static (LP) 
% optimisation of an objective function and applying constraints to the rates 
% of change of flux in addition to the standard FBA constraints$$^6$.
% 
% The dynamic FBA method implemented in this function is essentially the same 
% as the method described by Varma A. and B. O. Palsson$$^7$.

modeldinamic = model;
% For Recon3.0 model
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc_D[e]', -20, 'l');
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2[e]', -1000, 'l');
% modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac[e]', -1000, 'l');

modeldinamic = changeRxnBounds (modeldinamic, 'EX_glc_D[e]', -20, 'b');
modeldinamic = changeRxnBounds (modeldinamic, 'EX_o2[e]', -1000, 'l');
modeldinamic = changeRxnBounds (modeldinamic, 'EX_ac[e]', -1000, 'l');
% For Recon3.0 model
% smi = {'EX_glc_D[e]' 'EX_ac[e]'}; 
smi = {'EX_glc_D[e]' 'EX_ac[e]'}; 
% exchange reaction for substrate in environment

smc = [10.8]; % Glucose, Acetate concentration (all in mM)

Xec = 0.001; % initial biomass
dt = 1.0/1000.0; % time steps
time = 1.0/dt; % simulation time

[concentrationMatrix, excRxnNames, timeVec,...
    biomassVec] = dynamicFBA(modeldinamic, smi, smc, Xec, dt, time, smi ); 
%% 7. Relaxed FBA
% Find the minimal set of relaxations on bounds and steady-state constraint 
% to make an infeasible FBA problem feasible. 
% 
% See tutorial_relaxedFBA.mlx
%% 8. Flux enrichment analysis (FEA)
% The flux enrichment analysis calculates the likelihood that a set of fluxes 
% would belong to a subsystem or pathway.
% 
% *Timing:*
% 
% The time to calculate the FEA is $<1$ second for any size of a model.

modelfea = model;
res = optimizeCbModel(modelfea,'max'); 
% say you are interested in enriching the active reactions
activeReactions = find(res.x) 
% You can also look for e.g. positive/negative/zeros flux reactions,
% that depends pretty much on the question.
% Now you look for the enrichement of reactions per subsystems
resultCell = FEA(modelfea, activeReactions, 'subSystems')
%% REFERENCES 
% [1] Orth, J. D., Thiele I., and Palsson, B. Ø.  What is flux balance analysis? 
% _Nat. Biotechnol.,_ 28(3), 245–248 (2010).
% 
% [2] Thiele, I., et al. A community-driven global reconstruction of human metabolism. 
% _Nat. Biotechnol.,_ 31(5), 419–425 (2013).
% 
% [3] Benyamini, T, Folger, O., Ruppin, E., Schlomi, T. Flux balance analysis 
% accounting for metabolite dilution. _Genome Biology_., 11(4):R43 (2010).
% 
% [4] Smallbone, K., and Simeonidis, E. Flux balance analysis: A geometric perspective. 
% _J Theor Biol_., 258: 311-315 (2009).
% 
% [5] Lewis, N.E., et al. Omic data from evolved E. coli are consistent with 
% computed optimal growth from genome-scale models. _Mol Syst Biol_., 6:390 (2010).
% 
% [6] Mahadevan, R., Edwards, J.S., Doyle, F.J. Dynamic Flux Balance Analysis 
% of Diauxic Growth in Escherichia coli. _Biophys J.,_ 83(3):1331-1340 (2002).
% 
% [7] Varma A. and Palsson, B. Ø. Stoichiometric flux balance models quantitatively 
% predict growth and metabolic by-product secretion in wild-type Escherichia coli 
% W3110. _App Environ Microbiol_., 60(10):3724-3731 (1994).
% 
% [8]. Laurent Heirendt & Sylvain Arreckx, Thomas Pfau, Sebastian N. Mendoza, 
% Anne Richelle, Almut Heinken, Hulda S. Haraldsdottir, Jacek Wachowiak, Sarah 
% M. Keating, Vanja Vlasov, Stefania Magnusdottir, Chiam Yu Ng, German Preciat, 
% Alise Zagare, Siu H.J. Chan, Maike K. Aurich, Catherine M. Clancy, Jennifer 
% Modamio, John T. Sauls, Alberto Noronha, Aarash Bordbar, Benjamin Cousins, Diana 
% C. El Assal, Luis V. Valcarcel, Inigo Apaolaza, Susan Ghaderi, Masoud Ahookhosh, 
% Marouen Ben Guebila, Andrejs Kostromins, Nicolas Sompairac, Hoai M. Le, Ding 
% Ma, Yuekai Sun, Lin Wang, James T. Yurkovich, Miguel A.P. Oliveira, Phan T. 
% Vuong, Lemmer P. El Assal, Inna Kuperstein, Andrei Zinovyev, H. Scott Hinton, 
% William A. Bryant, Francisco J. Aragon Artacho, Francisco J. Planes, Egils Stalidzans, 
% Alejandro Maass, Santosh Vempala, Michael Hucka, Michael A. Saunders, Costas 
% D. Maranas, Nathan E. Lewis, Thomas Sauter, Bernhard Ø. Palsson, Ines Thiele, 
% Ronan M.T. Fleming, *Creation and analysis of biochemical constraint-based models: 
% the COBRA Toolbox v3.0*, Nature Protocols, volume 14, pages 639–702, 2019 <https://doi.org/10.1038/s41596-018-0098-2 
% doi.org/10.1038/s41596-018-0098-2>.
% 
%
##### SOURCE END #####
-->
</div></body></html>