

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Persephone &mdash; The COBRA Toolbox</title>
  

  
  
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../_static/img/favicon-16x16.png">
  
  
  
    <link rel="manifest" href="../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../index.html"/>
        <link rel="up" title="Analysis" href="../index.html"/>
        <link rel="next" title="Additionalfunctions" href="additionalFunctions/index.html"/>
        <link rel="prev" title="Old" href="../thermo/molFiles/old/index.html"/> 

  
  <script src="../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="/cobratoolbox/stable/_static/img/logo.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},
	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../_static/../" + item.website_url + "' >" + value + "</a>" + "<a href='../../../_static/../" + item.website_url + "' ><img src='../../../_static/link.svg'/></a></span>";
		    }
	    },
		
		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html#solver-compatibility">Solver compatibility</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../sparseFBA/index.html">Sparsefba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fluxEnrichmentAnalysis/index.html">Fluxenrichmentanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../geometricFBA/index.html">Geometricfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../fastForest/index.html">Fastforest</a></li>
<li class="toctree-l3"><a class="reference internal" href="../igemRNA/index.html">Igemrna</a></li>
<li class="toctree-l3"><a class="reference internal" href="../QFCA/index.html">Qfca</a></li>
<li class="toctree-l3"><a class="reference internal" href="../gMCS/index.html">Gmcs</a></li>
<li class="toctree-l3"><a class="reference internal" href="../robustnessAnalysis/index.html">Robustnessanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../relaxedFBA/index.html">Relaxedfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../nutritionAlgorithm/index.html">Nutritionalgorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FVA/index.html">Fva</a></li>
<li class="toctree-l3"><a class="reference internal" href="../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../percentileCalc/index.html">Percentilecalc</a></li>
<li class="toctree-l3"><a class="reference internal" href="../thermo/index.html">Thermo</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Persephone</a><ul>
<li class="toctree-l4"><a class="reference internal" href="additionalFunctions/index.html">Additionalfunctions</a></li>
<li class="toctree-l4"><a class="reference internal" href="SeqC_pipeline/index.html">Seqc_pipeline</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../ROOM/index.html">Room</a></li>
<li class="toctree-l3"><a class="reference internal" href="../uFBA/index.html">Ufba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../findMIIS/index.html">Findmiis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rFBA/index.html">Rfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../mdFBA/index.html">Mdfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rumba/index.html">Rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../parsimoniousFBA/index.html">Parsimoniousfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../KLdistance/index.html">Kldistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../multiSpecies/index.html">Multispecies</a></li>
<li class="toctree-l3"><a class="reference internal" href="../ICONGEMs/index.html">Icongems</a></li>
<li class="toctree-l3"><a class="reference internal" href="../wholeBody/index.html">Wholebody</a></li>
<li class="toctree-l3"><a class="reference internal" href="../FBA/index.html">Fba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../dynamicFBA/index.html">Dynamicfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../rMTA/index.html">Rmta</a></li>
<li class="toctree-l3"><a class="reference internal" href="../MOMA/index.html">Moma</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../dataIntegration/index.html">Dataintegration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../lifecycle.html">Software Development Life Cycle</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../citations.html">Citations</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Analysis</a></li>
      <li class="breadcrumb-item active">Persephone</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="persephone">
<span id="id1"></span><h1>Persephone<a class="headerlink" href="#persephone" title="Permalink to this heading">¶</a></h1>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="additionalFunctions/index.html">Additionalfunctions</a></li>
<li class="toctree-l1"><a class="reference internal" href="SeqC_pipeline/index.html">Seqc_pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="SeqC_pipeline/seqc_proc/index.html">Seqc_proc</a></li>
<li class="toctree-l2"><a class="reference internal" href="SeqC_pipeline/seqc_output/index.html">Seqc_output</a></li>
<li class="toctree-l2"><a class="reference internal" href="SeqC_pipeline/seqc_input/index.html">Seqc_input</a></li>
</ul>
</li>
</ul>
</div>
<span class="target" id="module-src.analysis.persephone"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.analyseWBMs">
<span class="sig-name descname"><span class="pre">analyseWBMs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">mWBMPath</span></em>, <em class="sig-param"><span class="pre">fluxPath</span></em>, <em class="sig-param"><span class="pre">rxnList</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/analyseWBMs.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.analyseWBMs" title="Permalink to this definition">¶</a></dt>
<dd><p>analyseWBMs predicts the optimal fluxes for a list of user-defined
reactions (rxnList). All predicted are further described in
analyseWBMsol.m.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[FBA_results, pathsToFilesForStatistics] = analyseWBMs</strong> (<em>mWBMPath, fluxPath, rxnList</em>)</p>
</dd>
</dl>
<p>INPUTS:
mWBMPath      Path (character array) to WBMs
fluxPath      Path to directory where the results will be stored
rxnList       Cell array of VMH metabolites to investigate.</p>
<blockquote>
<div><p>Example: rxnList = {‘DM_trp_L[bc], DM_met_L[bc],’Brain_trp_L[c],
Heart_met_L[x]}. Note that demand reactions are
automatically added if they are not present in the models.</p>
</div></blockquote>
<p>OPTIONAL INPUTS
rxnSense       Character array containing either ‘max’ or ‘min’</p>
<blockquote>
<div><p>to specify the sense of the objective. Option
to specify differently for each objective- to do so provide
character array the exact length of rxnList.
(OPTIONAL, Default = ‘max’).</p>
</div></blockquote>
<dl class="simple">
<dt>numWorkersOptimization    Number of workers that will perform FBA in parallel. Note</dt><dd><p>that more workers does not necessarily make the function
faster. It is generally not recommended to set numWorkersOptimization
equal to the number of available cores (see:
feature(‘numCores’)) as linear solvers can already
support multi-core linear optimisation, thus resulting in
unnessessary overhead. On computers with 8 cores or less, it
is recommended to set numWorkersOptimization to 1. On a computer with
36 cores, an optimal configuration of numWorkersOptimization=6 was
found.</p>
</dd>
<dt>saveFullRes   Boolean (true/false) indicating if all the complete .v, .y.</dt><dd><p>, and .w vectors are stored in the result. Default = true.
It is recommended to set saveFullRes</p>
</dd>
</dl>
<p>paramFluxProcessing       Structured array with optional parameters:</p>
<blockquote>
<div><p>.NumericalRounding defines how much the predicted flux values are
rounded. A defined value of 1e-6 means that a flux value of
2 + 2.3e-8 is rounded to 2. A flux value of 0 + 1e-15 would be rounded to
exactly zero. This rounding factor will also be applied to the shadow
price values. If microbiome relative abundance data is provided, the same
rounding factor will be applied to the relative abundance data.</p>
<p>Default parameterisation:
- paramFluxProcessing.NumericalRounding = 1e-6;</p>
<p>Example:
- paramFluxProcessing.NumericalRounding = 1e-6;</p>
<p>paramFluxProcessing.NumericalRounding = 1e-6;</p>
<p>.RxnRemovalCutoff defines the minimal number of samples for which a
unique reaction flux could be obtained, before removing the reaction for
further analysis. This parameter can be expressed as
* fraction:  the fraction of samples with unique values,
* SD: the standard deviation across samples, and
* count: the counted number of unique values. If microbiome relative
abundance data is provided, the same removal cutoff factor will be
applied to the relative abundance data.</p>
<p>Default parameterisation:
- paramFluxProcessing.RxnRemovalCutoff = {‘fraction’,0.1};</p>
<p>Examples:
- paramFluxProcessing.RxnRemovalCutoff = {‘fraction’,0.1};
- paramFluxProcessing.RxnRemovalCutoff = {‘SD’,1};
- paramFluxProcessing.RxnRemovalCutoff = {‘count’,30};</p>
<p>paramFluxProcessing.RxnRemovalCutoff = {‘fraction’,0.1};</p>
<p>.RxnEquivalenceThreshold defines the minimal threshold of when
functionally identical flux values are predicted, and are thus part of
the same linear pathways. The threshold for functional equivalence is
expressed as the R2 (r-squared) value after performing a simple linear
regression between two reactions.</p>
<p>Default parameterisation:
- paramFluxProcessing.RxnEquivalenceThreshold = 0.999;</p>
<p>Example:
- paramFluxProcessing.RxnEquivalenceThreshold = 0.999;</p>
<p>paramFluxProcessing.RxnEquivalenceThreshold = 0.999;</p>
<p>.fluxMicrobeCorrelationType defines the method for correlating the
predicted fluxes with microbial relative abundances. Note that this
metric is not used if mWBMs are not present. The available correlation
types are:
* regression_r2:  the R2 (r-squared) value from pairwised linear regression on the
predicted fluxes against microbial relative abundances.
* spearman_rho: the correlation coefficient, rho obtained from pairwise
Spearman nonparametric correlations between predicted fluxes and
microbial relative abundances.</p>
<p>Default parameterisation:
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘regression_r2’;</p>
<p>Examples:
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘regression_r2’;
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘spearman_rho’;</p>
</div></blockquote>
<dl class="simple">
<dt>fluxAnalysisPath:     Character array with path to directory where all</dt><dd><p>results will be saved (Default = pwd)</p>
</dd>
<dt>Solver:               Validated solvers: ‘cplex_direct’,’ibm_cplex’</dt><dd><p>‘tomlab_cplex’, ‘gurobi’, ‘mosek’</p>
</dd>
<dt>analyseGF:            Boolean indiciating whether or not to investigate</dt><dd><p>GF models. In the case of personalisation, a germ
free iWBM will be created for every sample, this
can results in long computation times as there are
double the number of models to solve. If you are
not interested in solving a germ-free iWBM for each
sample, you can set to false. Default is true. When
personalisation is skipped, only one germ-free
model is made for each sex.</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.analyseWBMsol">
<span class="sig-name descname"><span class="pre">analyseWBMsol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">fluxPath</span></em>, <em class="sig-param"><span class="pre">paramFluxProcessing</span></em>, <em class="sig-param"><span class="pre">fluxAnalysisPath</span></em>, <em class="sig-param"><span class="pre">analyseGF</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/analyseWBMsol.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.analyseWBMsol" title="Permalink to this definition">¶</a></dt>
<dd><p>analyseWBMsol loads the FBA solutions produced in
analyseWBMs.m, prepares the FBA results for further
analyses, and produces summary statistics into the flux results.</p>
<dl>
<dt>The function contains six parts:</dt><dd><ul class="simple">
<li><p>PART 1: Loading the flux solutions and corresponding shadow prices</p></li>
<li><p>PART 2: Converting the microbial  biomass shadow prices</p></li>
</ul>
<p>associated with each optimised reaction flux to human readable
tables and producing statistics on microbial influences on the flux
results.
- PART 3: Isolate the microbial component of the fluxes by
subtracting the germ-free fluxes in a sex specific manner. In this
part, descriptive statistics are also obtained on the fluxes across
samples and metabolites are removed based on a user defined
threshold (see thresholds input)
- PART 4: In this part, microbe-flux associations defined by the
species biomass shadow prices (PART 2) are further quantified by
performing Spearman correlations on the fluxes and microbe relative
abundances.
- PART 5: Reactions with identical or near identical fluxes across
samples (user defined) are grouped and collapsed into a single
result.
- PART 6: Save all results</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>processedFluxResPaths = analyseWBMsol</strong> (<em>fluxPath,paramFluxProcessing, fluxAnalysisPath</em>)</p>
</dd>
</dl>
<p>INPUTS:
fluxPath         Character array with path to .mat files produced in</p>
<blockquote>
<div><p>optimiseRxnMultipleWBM.m</p>
</div></blockquote>
<p>paramFluxProcessing       Structured array with optional parameters:</p>
<blockquote>
<div><p>.numericalRounding defines how much the predicted flux values are
rounded. A defined value of 1e-6 means that a flux value of
2 + 2.3e-8 is rounded to 2. A flux value of 0 + 1e-15 would be rounded to
exactly zero. This rounding factor will also be applied to the shadow
price values. If microbiome relative abundance data is provided, the same
rounding factor will be applied to the relative abundance data.</p>
<p>Default parameterisation:
- paramFluxProcessing.numericalRounding = 1e-6;</p>
<p>Example:
- paramFluxProcessing.numericalRounding = 1e-6;</p>
<p>paramFluxProcessing.numericalRounding = 1e-6;</p>
<p>.rxnRemovalCutoff defines the minimal number of samples for which a
unique reaction flux could be obtained, before removing the reaction for
further analysis. This parameter can be expressed as
* fraction:  the fraction of samples with unique values,
* SD: the standard deviation across samples, and
* count: the counted number of unique values. If microbiome relative
abundance data is provided, the same removal cutoff factor will be
applied to the relative abundance data.</p>
<p>Default parameterisation:
- paramFluxProcessing.rxnRemovalCutoff = {‘fraction’,0.1};</p>
<p>Examples:
- paramFluxProcessing.rxnRemovalCutoff = {‘fraction’,0.1};
- paramFluxProcessing.rxnRemovalCutoff = {‘SD’,1};
- paramFluxProcessing.rxnRemovalCutoff = {‘count’,30};</p>
<p>paramFluxProcessing.rxnRemovalCutoff = {‘fraction’,0.1};
RxnEquivalenceThreshold
.rxnEquivalenceThreshold defines the minimal threshold of when
functionally identical flux values are predicted, and are thus part of
the same linear pathways. The threshold for functional equivalence is
expressed as the R2 (r-squared) value after performing a simple linear
regression between two reactions.</p>
<p>Default parameterisation:
- paramFluxProcessing.rxnEquivalenceThreshold = 0.999;</p>
<p>Example:
- paramFluxProcessing.rxnEquivalenceThreshold = 0.999;</p>
<p>paramFluxProcessing.rxnEquivalenceThreshold = 0.999;</p>
<p>.fluxMicrobeCorrelationMetric defines the method for correlating the
predicted fluxes with microbial relative abundances. Note that this
metric is not used if mWBMs are not present. The available correlation
types are:
* regression_r2:  the R2 (r-squared) value from pairwised linear regression on the
predicted fluxes against microbial relative abundances.
* spearman_rho: the correlation coefficient, rho obtained from pairwise
Spearman nonparametric correlations between predicted fluxes and
microbial relative abundances.</p>
<p>Default parameterisation:
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘regression_r2’;</p>
<p>Examples:
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘regression_r2’;
- paramFluxProcessing.fluxMicrobeCorrelationMetric = ‘spearman_rho’;</p>
</div></blockquote>
<dl class="simple">
<dt>fluxAnalysisPath</dt><dd><p>Character array with path to directory where all
results will be saved.</p>
</dd>
<dt>AUTHOR:</dt><dd><ul class="simple">
<li><p>Tim Hensen, July 2024</p></li>
<li><p>Jonas Widder, 11/2024 (small bugfixes)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat script">
<dt class="sig sig-object mat" id="src.analysis.persephone.configTemplatePersephone">
<span class="sig-name descname"><span class="pre">configTemplatePersephone</span></span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/configTemplatePersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.configTemplatePersephone" title="Permalink to this definition">¶</a></dt>
<dd><p>Configuration file %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.createBatchMWBM">
<span class="sig-name descname"><span class="pre">createBatchMWBM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">mgpipePath</span></em>, <em class="sig-param"><span class="pre">mWBMPath</span></em>, <em class="sig-param"><span class="pre">metadataPath</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/createBatchMWBM.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.createBatchMWBM" title="Permalink to this definition">¶</a></dt>
<dd><p>This function creates personalised host-microbiome WBM models (mWBMs) by joining
microbiome community models with unpersonalised WBM models in a
sex-specific manner. The models are parameterised on a predefined diet.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[modelStats, summaryStats, dietInfo, dietGrowthStats] = createBatchMWBM</strong> (<em>mgpipePath, mWBMPath, metadataPath</em>)</p>
</dd>
</dl>
<p>INPUTS
mgpipePath:                   Path to microbiome community models created by the</p>
<blockquote>
<div><p>microbiome modelling toolbox.</p>
</div></blockquote>
<dl class="simple">
<dt>mWBMPath                      Path to directory where the HM models are</dt><dd><p>saved</p>
</dd>
</dl>
<p>OPTIONAL INPUTS
Diet                          Diet option: ‘EUAverageDiet’ (default)
numWorkersCreation                    Number of cores used for parallelisation.</p>
<blockquote>
<div><p>Default = 4.</p>
</div></blockquote>
<dl class="simple">
<dt>checkFeasibility              Flag (true/false) to run the</dt><dd><p>ensureHMfeasibility.m function and check if
the generated models can grow on the diet.
Default = true.</p>
</dd>
<dt>wbmDirectory                  Path to directory with user defined WBMs.</dt><dd><p>This function supports one user adapted male and one user adapted female
WBM or personalised germ free WBMs for each sample with a microbiome
community model. Default is empty. If wbmDirectory is empty,
Harvey/Harvetta version 1.04c are used.</p>
</dd>
</dl>
<p>OUTPUTS
modelStats                    Table with summary statistics on the generated WBMs:</p>
<blockquote>
<div><p>gender, number of reactions, metabolites,constrainsts,
and taxa.</p>
</div></blockquote>
<dl class="simple">
<dt>summaryStats                  Table with the mean and SD of the model</dt><dd><p>statistics in the modelStats variable.</p>
</dd>
<dt>dietInfo                      Table with diet growth information.</dt><dd><p>models can grow on the given diet.</p>
</dd>
</dl>
<p>dietGrowthStats               Table with statistics on</p>
<p>Authors:  Tim Hensen, 2023, 2024</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.createForestPlot">
<span class="sig-name descname"><span class="pre">createForestPlot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">estimates</span></em>, <em class="sig-param"><span class="pre">ci</span></em>, <em class="sig-param"><span class="pre">names</span></em>, <em class="sig-param"><span class="pre">pValues</span></em>, <em class="sig-param"><span class="pre">plotTitle</span></em>, <em class="sig-param"><span class="pre">xTitle</span></em>, <em class="sig-param"><span class="pre">hideLegend</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/createForestPlot.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.createForestPlot" title="Permalink to this definition">¶</a></dt>
<dd><p>createForestPlot generates a forest plot to display confidence intervals for estimates.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>createForestPlot</strong> (<em>estimates, ci, names, pValues, plotTitle, xTitle, hideLegend</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>estimates   - Vector of estimates</strong> (<em>e.g., effect sizes or log fold changes</em>)</p></li>
<li><p><strong>ci          - Matrix of confidence intervals [n x 2] with lower and upper bounds in columns.</strong></p></li>
<li><p><strong>names       - Cell array of labels for each data point.</strong></p></li>
<li><p><strong>pValues     - Vector of p-values corresponding to each estimate.</strong></p></li>
<li><p><strong>plotTitle   - String, title of the plot.</strong></p></li>
<li><p><strong>xTitle      - String, label for the x-axis</strong> (<em>typically “Effect Size” or “Log Fold Change”</em>)</p></li>
<li><p><strong>hideLegend  - Logical, if true, hides the legend</strong> (<em>default is false</em>)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.createMWBM">
<span class="sig-name descname"><span class="pre">createMWBM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">microbiota_model</span></em>, <em class="sig-param"><span class="pre">WBM_model</span></em>, <em class="sig-param"><span class="pre">Diet</span></em>, <em class="sig-param"><span class="pre">saveDir</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/createMWBM.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.createMWBM" title="Permalink to this definition">¶</a></dt>
<dd><p>This function creates personalised host-microbiome WBM models by joining
microbiome community models with unpersonalised WBM models. The models
are parameterised on a predefined diet.</p>
<p>Example: modelHM = createMWBM(microbiota_model, WBM_model, ‘EUAverageDiet’)</p>
<p>INPUTS
microbiota model:             Microbiome community model created by the</p>
<blockquote>
<div><p>microbiome modelling toolbox.</p>
</div></blockquote>
<dl class="simple">
<dt>WBM_model:                    Harvey or Harvetta whole-body metabolic</dt><dd><p>model</p>
</dd>
</dl>
<p>Diet                          Diet option: ‘EUAverageDiet’ (default)</p>
<p>Output
modelHM:                      Personalised host-microbiome WBM model</p>
<dl class="simple">
<dt>Authors:  Tim Hensen, Bronson Weston, 2022</dt><dd><p>Tim Hensen, expanded WBM parameterisation July 2024
Tim Hensen, improved mWBM annotation November 2024</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.createVolcanoPlot">
<span class="sig-name descname"><span class="pre">createVolcanoPlot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">estimates</span></em>, <em class="sig-param"><span class="pre">pValues</span></em>, <em class="sig-param"><span class="pre">names</span></em>, <em class="sig-param"><span class="pre">plotTitle</span></em>, <em class="sig-param"><span class="pre">xTitle</span></em>, <em class="sig-param"><span class="pre">yTitle</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/createVolcanoPlot.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.createVolcanoPlot" title="Permalink to this definition">¶</a></dt>
<dd><p>createVolcanoPlot generates a volcano plot to visualize the relationship
between regression estimates and p-values.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>createVolcanoPlot</strong> (<em>estimates, pValues, names, plotTitle, xTitle, yTitle</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>estimates - [vector] Regression estimates for each data point</strong> (<em>e.g., effect sizes or log fold changes</em>)</p></li>
<li><p><strong>pValues   - [vector] p-values corresponding to each estimate.</strong></p></li>
<li><p><strong>names     - [cell array] Labels or names for each data point.</strong></p></li>
<li><p><strong>plotTitle - [string] Title of the plot.</strong></p></li>
<li><p><strong>xTitle    - [string] Label for the x-axis</strong> (<em>e.g., “Effect Size” or “Log Fold Change”</em>)</p></li>
<li><p><strong>yTitle    - [string] Label for the y-axis</strong> (<em>e.g., “-log10(p-value)”</em>)</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>None      - This function generates a volcano plot in the current figure.</strong></p>
</dd>
</dl>
<p class="rubric">Example</p>
<dl class="simple">
<dt>createVolcanoPlot(estimates, pValues, {‘Metabolite1’, ‘Metabolite2’}, …</dt><dd><p>‘Volcano Plot’, ‘Log Fold Change’, ‘-log10(p-value)’)</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.ensureWBMfeasibility">
<span class="sig-name descname"><span class="pre">ensureWBMfeasibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">mWBMPath</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/ensureWBMfeasibility.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.ensureWBMfeasibility" title="Permalink to this definition">¶</a></dt>
<dd><p>This function checks for the WBMs in the specified mWBMPath to grow on the
given diet, finds a diet that makes all WBMs feasible and propagates this
diet onto the models. The functions works as follows:
1) Load the WBMs from the directory and test which models are feasible on
the diet.
2) If any WBM is not feasible, collect all infeasible WBMs, open all diet
reactions, and test if the WBMs can grow on any diet. If one or more WBMs
cannot grow when opening all diet reactions, the statistics are
collected and the function stops. The user will need to debug the WBMs
manually.
3) If all previously infeasible WBMs can grow when all diet reactions are
opened,missing diet components are searched using getMissingDietPersephone.
If getMissingDietPersephone cannot find a diet after one function call, this
function is run again until a hard limit of 10 iterations is reached. If
this limit is reached, no feasible diet can be found for all models and
the user will need to manually debug the infeasible models. If missing
diet components are found in one WBMs, these component are automatically propagated
to the other WBMs infeasible on the original diet before searching for
more missing diet components
4) If a new diet is found for which all previously infeasible WBMs can grow,
the updated diet is propagated to all models in the mWBMPath.
5) The updated diet is stored as an output variable for this function and
for each model, its feasibility on the original and updated diet is
recorded.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[dietInfo, dietGrowthStats] = ensureHMfeasibility</strong> (<em>mWBMPath, Diet</em>)</p>
</dd>
</dl>
<p>INPUTS
mWBMPath           Path to directory where the HM models are saved
Diet                  Diet option: ‘EUAverageDiet’ (default)</p>
<p>OUTPUTS
dietInfo              Structured variable containing the missing diet</p>
<blockquote>
<div><p>components and the updated diet propagated to all models.
This variable is empty if no updated diet could be found.</p>
</div></blockquote>
<dl class="simple">
<dt>dietGrowthStats       Table indicating which WBMs could initially not grow on</dt><dd><p>the diet, which WBMs could not grow on any diet,
and which WBMs could not grow on an updated diet.</p>
</dd>
</dl>
<p class="rubric">Authors</p>
<ul class="simple">
<li><p>Tim Hensen, 2024</p></li>
<li><p>Bram Nap, 07-2025 remove parfor functionality. Time save is not</p></li>
</ul>
<p>massive and allows the function to be used for other purposes besides
Persephone.</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.fdrBHadjustment">
<span class="sig-name descname"><span class="pre">fdrBHadjustment</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">p</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/fdrBHadjustment.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.fdrBHadjustment" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>IMPORTANT; This function was partially generated by chatGPT o3-mini.</dt><dd><p>The code is manually tested and altered with bug fixes. This function
performs identical to running q = mafdr(p, ‘BHFDR’, true)
using the Matlab bioinformatics toolbox.</p>
</dd>
</dl>
<p>FDR_BH Performs Benjamini-Hochberg FDR correction on a vector of p-values.</p>
<blockquote>
<div><p>q = FDR_BH(p) returns the FDR-corrected p-values q for the input vector p.</p>
<dl class="simple">
<dt>Input:</dt><dd><p>p - a vector of p-values (assumed to be in the interval [0, 1])</p>
</dd>
<dt>Output:</dt><dd><p>q - a vector of FDR corrected p-values, in the same order as p.</p>
</dd>
<dt>The function implements the procedure described in:</dt><dd><p>Benjamini, Y. and Hochberg, Y. (1995) Controlling the false discovery
rate: a practical and powerful approach to multiple testing.</p>
</dd>
<dt>Example:</dt><dd><p>pvals = [0.01, 0.04, 0.03, 0.002, 0.07];
qvals = fdr_bh(pvals);</p>
</dd>
</dl>
</div></blockquote>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.findPvalCategories">
<span class="sig-name descname"><span class="pre">findPvalCategories</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">pValues</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/findPvalCategories.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.findPvalCategories" title="Permalink to this definition">¶</a></dt>
<dd><p>This function inputs a list of p values and outputs the variables
pValueGroups and colourGroups. A maximum of three groups are defined,
FDR&lt;0.05, P&lt;0.05, and P&gt;0.05.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[groupsToPlot, colours] = findPvalCategories</strong> (<em>pValues</em>)</p>
</dd>
</dl>
<p>INPUT
pValues</p>
<p>OUTPUTS
groupsToPlot      Table that categorises the indices of the p-value list.
colours           Cell array with numerical encodings for the</p>
<blockquote>
<div><p>category-associated colours.</p>
</div></blockquote>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.generateStackedBarPlot_PhylumMARScoverage">
<span class="sig-name descname"><span class="pre">generateStackedBarPlot_PhylumMARScoverage</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">input_relAbundances_preMapping</span></em>, <em class="sig-param"><span class="pre">input_relAbundances_postMapping</span></em>, <em class="sig-param"><span class="pre">saveDir</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/generateStackedBarPlot_PhylumMARScoverage.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.generateStackedBarPlot_PhylumMARScoverage" title="Permalink to this definition">¶</a></dt>
<dd><p>Generates stacked bar plots from mean relative abundances of phyla pre- &amp;
post-mapping to a model-database in MARS.</p>
<dl class="field-list simple">
<dt class="field-odd">INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_relAbundances_preMapping</strong> – [chars/string] Path to table which contains phyla taxa and their
mean relative abundances across samples pre-mapped to a model-database
(original dataset). This spreadsheet is a standard output generated in MARS.</p></li>
<li><p><strong>input_relAbundances_postMapping</strong> – [chars/string] Path to table which contains phyla taxa and their
mean relative abundances across samples post-mapped to a model-database
(filtered dataset). This spreadsheet is a standard output generated in MARS.</p></li>
<li><p><strong>saveDir</strong> – [chars/string] Path to the
directory where the stacked bar plot should be saved.</p></li>
<li><p><strong>mappingDatabase_name</strong> – [chars/string] The name of the model-database used for
mapping in MARS, which will be displayed in the title of the graph.
Defaults to ‘’.</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>AUTHOR:</dt><dd><ul class="simple">
<li><p>Jonas Widder, 12/2024 &amp; 01/2025</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.getMissingDietPersephone">
<span class="sig-name descname"><span class="pre">getMissingDietPersephone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">inputModel</span></em>, <em class="sig-param"><span class="pre">missingDietComponents</span></em>, <em class="sig-param"><span class="pre">testInitialFeasibility</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/getMissingDietPersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.getMissingDietPersephone" title="Permalink to this definition">¶</a></dt>
<dd><p>This function determines missing dietary compounds in host-microbiome
WBMs that are infeasible (it assumes that the WBMs and the microbiome
models individually are feasible). Alternatively, this function can
determine missing dietary compounds in microbiome community models
This function first tests whether the model is feasible (which can be
skipped by setting testInitialFeasibility = 0), if infeasible, all diet
exchange reactions that are not already active will be opened and the
feasibility will be tested. If infeasible, the function stops -  then
there is no dietary solution (of course it could be that active diet
constraints are limited but they will not be tested with this script).
Subsequently, diet exchanges are randomly closed (first in batches of 50,
then 10, then 5, then 1). If the inputModel remains feasible this batch as
well as all other 0 diet fluxes in the fba solution will be deemed not
necessary for feasibility, if infeasible the batch set will be kept and
tested for in the next step. when the batch set size is 1, each remaining
diet exchange will be tested for individually.
While still being slow, this approach is much faster then testing each
diet exchange individually.
As the diet exchanges are selected randomly for each batch, running the
function twice may not result in the same final set of
missingDietComponents</p>
<p>INPUT
inputModel                Host-microbiome or microbiome community model structure
missingDietComponents     list of diet exchange reactions that are known</p>
<blockquote>
<div><p>to be missing or that have been identified in a previous run of this
function</p>
</div></blockquote>
<p>testInitialFeasibility    default 1</p>
<p>OUTPUT
missingDietComponents     list of missingDietComponents. If</p>
<blockquote>
<div><p>missingDietComponents was given as an input then this list is a
combination of the input and the newly discovered missingDietComponents</p>
</div></blockquote>
<p>Ines Thiele, Nov 2023
Tim Hensen, September 2025. Added support for microbiome community
models.</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.initPersephone">
<span class="sig-name descname"><span class="pre">initPersephone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">resultPath</span></em>, <em class="sig-param"><span class="pre">paths</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/initPersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.initPersephone" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Initializes the Persephone pipeline by:</dt><dd><ul class="simple">
<li><p>Checking for MATLAB toolbox dependencies.</p></li>
<li><p>Processing and validating the metadata file.</p></li>
<li><p>Cross-referencing metadata with microbiome data (if provided).</p></li>
<li><p>Setting up directory structures for Persephone results.</p></li>
</ul>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[initialised, paths, statToolboxInstalled] = initPersephone</strong> (<em>resultPath, metadataPath, readsTablePath</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>resultPath        -</strong> (<em>char</em>)</p></li>
<li><p><strong>metadataPath     -</strong> (<em>char) Path to the metadata file (.csv or .xlsx</em>)</p></li>
<li><p><strong>readsTablePath   -</strong> (<em>char</em>)</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>initialised           -</strong> (<em>logical</em>)</p></li>
<li><p><strong>paths             -</strong> (<em>struct</em>)</p></li>
<li><p><strong>statToolboxInstalled  -</strong> (<em>logical</em>)</p></li>
<li><p><strong>updatedMetadataPath   -</strong> (<em>char</em>)</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>REQUIREMENTS:</dt><dd><ul class="simple">
<li><p>MATLAB R2020b or newer.</p></li>
<li><p>Parallel Computing Toolbox (mandatory).</p></li>
<li><p>Statistics and Machine Learning Toolbox (recommended for analysis).</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Metadata must contain a column of sample IDs and a column of sex information.</p></li>
<li><p>Metadata and reads table will be synchronized to ensure consistent samples across analyses.</p></li>
<li><p>Any discrepancies in variable naming (e.g., “ID”, “Sex”) are automatically resolved.</p></li>
<li><p>Processed metadata is saved to an updated file.</p></li>
</ul>
<p class="rubric">Example</p>
<p>resultPath = ‘./results’;
metadataPath = ‘./data/metadata.csv’;
readsTablePath = ‘./data/readsTable.csv’;
[initialised, paths, statToolboxInstalled] = initPersephone(resultPath, metadataPath, readsTablePath);</p>
<dl class="simple">
<dt>AUTHOR:</dt><dd><ul class="simple">
<li><p>Tim Hensen, January 2025</p></li>
<li><p>Bram Nap, July 2024: Automatic creation of output folders</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.loadMinimalWBM">
<span class="sig-name descname"><span class="pre">loadMinimalWBM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modPath</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/loadMinimalWBM.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.loadMinimalWBM" title="Permalink to this definition">¶</a></dt>
<dd><p>This function loads the smallest possible combination of WBM model fields
needed to perform FBA. Loading the minimal model can
decrease loading times by 6X.</p>
<p>Author: Tim Hensen, November 2024</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.performRegressions">
<span class="sig-name descname"><span class="pre">performRegressions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">data</span></em>, <em class="sig-param"><span class="pre">metadata</span></em>, <em class="sig-param"><span class="pre">formula</span></em>, <em class="sig-param"><span class="pre">exponentiateLogOdds</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/performRegressions.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.performRegressions" title="Permalink to this definition">¶</a></dt>
<dd><p>This functions performs linear or logistic regressions on either flux
data or microbial relative abundances. The function supports control
variables &amp; moderators in the regression formulae.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>results = performRegressions</strong> (<em>data,metadata,formula</em>)</p>
</dd>
</dl>
<p>INPUT
data:     Table with flux or microbiome data. The first column must be the ID</p>
<blockquote>
<div><p>column.</p>
</div></blockquote>
<p>metadata: Metadata table. The first column must be the ID column.
formula: Must contain either “Flux or “relative_abundance” as predictor.
exponentiateLogOdds: Logical variable on if a logistic regression estimate
should be exponentiately to obtain the odds ratios.</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.performStatsPersephone">
<span class="sig-name descname"><span class="pre">performStatsPersephone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">statPath</span></em>, <em class="sig-param"><span class="pre">pathToProcessedFluxes</span></em>, <em class="sig-param"><span class="pre">metadataPath</span></em>, <em class="sig-param"><span class="pre">response</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/performStatsPersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.performStatsPersephone" title="Permalink to this definition">¶</a></dt>
<dd><p>The performStatistics function is part of the PSCM toolbox. This
function performs regression analyses on processed fluxes and gut
microbiome relative abundance data in large cohort studies. This function needs a
minimal sample size of 50 samples. This function performs different
statistical analyses depending on the user defined inputs.</p>
<p>If the response variable is binary and no confounders are given,
wilcoxon tests will be performed for the fluxes and microbiome
abundances.
If the response variable is binary and confounders are given, multiple
logistic regressions will be performed.
If the response variable is continuous and no confounders are given,
simple linear regressions will be performed.
If the response variable is continuous and confounders are given,
multiple linear regressions will be performed.</p>
<p>The fluxes are z-transformed before statistical testing. The microbiome
relative abundances are log transformed before testing.
If a moderator variable is given and a regression is
performed, a moderation analysis on the moderator with the predictor will be performed.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>performStatistics</strong> (<em>statPath, pathToProcessedFluxes, pathToWbmRelAbundances, metadataPath, response, confounders, moderator, microbeCutoff,threshold</em>)</p>
</dd>
</dl>
<p>INPUT
statPath                          Path (character array) to working directory
pathToProcessedFluxes                 Path to processed flux data
pathToWbmRelAbundances                  Path to microbial relative abundances
metadataPath                      Path to metadata file
response                          Name of response variables (char or string)</p>
<p>OPTIONAL INPUTS
confounders                       Cell array with the names of confounding variables to</p>
<blockquote>
<div><p>be included. Default = empty.</p>
</div></blockquote>
<dl class="simple">
<dt>microbeCutoff                     Cutoff threshold for the number of samples a</dt><dd><p>microbe needs to present to be analysed. Default = 0.1. (10%)</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.persWBM">
<span class="sig-name descname"><span class="pre">persWBM</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">metadata</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/persWBM.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.persWBM" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes a table of physiological parameters for an individual or multiple individuals
(listed in inputs) and adjusts the paramteres of a provided WBM or Harvey/Harvetta to
create a personalised WBM</p>
<p>The calculation of physiological parameteres will be performed based on
the available data e.g., if the user provides Cardiac output, this given
value is assigned, if the user provides stroked volume and no CO, CO is
calculated based on SV. If neither values are provided, CO will be
calculated based on XXX. All details of parameter calculation are
detailed in the personalised model output and in the excel file provided
by this function.</p>
<p>INPUTS</p>
<p>REQUIRED:
metadata                     Can be a struct (for personalising one model)</p>
<blockquote>
<div><p>or a path to an excel file (for batch
personalisation). If it is a structure, it
should follow the format of the individualised
parameters struct obtained from running:
&gt; sex = “male”;
OR
&gt; sex = “female”;
AND
&gt; standardPhysiolDefaultParameters;</p>
</div></blockquote>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.persWBMmetabolomics">
<span class="sig-name descname"><span class="pre">persWBMmetabolomics</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">sex</span></em>, <em class="sig-param"><span class="pre">metabolomicParams</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/persWBMmetabolomics.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.persWBMmetabolomics" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes a table of metabolomic parameters for an individual or multiple individuals
(listed in inputs) and adjusts the paramteres of a provided WBM or Harvey/Harvetta to
create a personalised WBM</p>
<p>The calculation of metabolomic parameteres will be performed based on
the available data</p>
<p>INPUTS</p>
<p>REQUIRED:
metabolomicParamters         Can be a cell array or a table or a path to an</p>
<blockquote>
<div><p>excel file. In any case, this should contain
a list with a minimum of three columns and option
for additional columns for each individual for which a model should be created:</p>
<p><a href="#id2"><span class="problematic" id="id3">|</span></a>”compartment”    <a href="#id4"><span class="problematic" id="id5">|</span></a>”blood”       <a href="#id6"><span class="problematic" id="id7">|</span></a>”urine”   |
<a href="#id8"><span class="problematic" id="id9">|</span></a>”unit”           <a href="#id10"><span class="problematic" id="id11">|</span></a>”mg/dL”       | “mg/dL”  |
<a href="#id12"><span class="problematic" id="id13">|</span></a>”Individual1”    <a href="#id14"><span class="problematic" id="id15">|</span></a>180           <a href="#id16"><span class="problematic" id="id17">|</span></a>50        |
<a href="#id18"><span class="problematic" id="id19">|</span></a>”Individual2”    <a href="#id20"><span class="problematic" id="id21">|</span></a>150           <a href="#id22"><span class="problematic" id="id23">|</span></a>66        |</p>
<p>Sex must be provided for each model. The compartment must be specified and must be
one of the following: csf, u, blParameters that can be personalised include:
All of those metabolites which are found in
the
<a href="#id24"><span class="problematic" id="id25">*</span></a>for each the default unit used by the model
is provided. For x, y, z- unit conversion
will be performed for all known common units of
measurement.</p>
</div></blockquote>
<p>OPTIONAL:
WBM/(s)                     User can provide a WBM (Whole Body Metabolic Model) or a path to multiple models.</p>
<blockquote>
<div><p>If no model is provided, either Harvey or Harvetta will be loaded from the COBRA
toolbox, depending on sex in provided physiological data.
If multiple models are provided, the model ID must
match the model name provided in the physiological data.</p>
</div></blockquote>
<dl class="simple">
<dt>resPath                     Path on which to store personalised model and</dt><dd><p>other outputs.
Default = current directory</p>
</dd>
<dt>Diet                        Diet in the form of text file or named .mat</dt><dd><p>file from the COBRA toolbox (default =
EUAverageDiet)
** This function can also be run in isolation
but if a user wants to personalise both
physiological and metabolomic- this function
should alway be run and NOT
persWBMmetabolomics in isolation!</p>
</dd>
</dl>
<p>OUTPUTS</p>
<dl class="simple">
<dt>iWBM                        Model with updated physiological paramteres</dt><dd><p>(stored as “persModelName.mat”). All updated
paramteres are described in
model.IndividualisedParameters</p>
</dd>
<dt>controlWBM                  This is a WBM with no personalised</dt><dd><p>adjustments. When a WBM or multiple WBMs have
been given as inputs, the controlWBMs are
exact copies of those. When Harvey or
Harvetta are used, controlWBM is copies of
Harvey/Harvetta.</p>
</dd>
<dt>persParameters              Excel file with details of the updated</dt><dd><p>parameter and how it was calculated</p>
</dd>
</dl>
<p>author: Anna Sheehy November 2024
Step One: Read in the available data, check all data is valid
Define the input parser</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.readMetadataForPersephone">
<span class="sig-name descname"><span class="pre">readMetadataForPersephone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">metadataPath</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/readMetadataForPersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.readMetadataForPersephone" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>DESCRIPTION:</dt><dd><p>This function loads a metadata file into a MATLAB table. It accounts for:
- Preserving the full variable names by disabling automatic truncation.
- Capturing variable units if provided in the second row of the file.
- Storing the captured units in the ‘VariableUnits’ property of the table.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>metadata = readMetadataForPersephone</strong> (<em>metadataPath</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>metadataPath</strong> – A string specifying the path to the metadata file
(in CSV or XLSX format).</p>
</dd>
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>metadata</strong> – A MATLAB table containing the processed metadata.
Variable units, if present in the second row of the file,
are stored in the table’s ‘VariableUnits’ property.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>If variable names exceed 64 characters, they are truncated, but the
true variable names are preserved in the ‘VariableDescriptions’ property.</p></li>
<li><p>Warnings related to variable name truncation are suppressed to avoid
unnecessary console output.</p></li>
</ul>
<p class="rubric">AUTHORS</p>
<p>Tim Hensen, January 2025</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.runMars">
<span class="sig-name descname"><span class="pre">runMars</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">readsTablePath</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/runMars.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.runMars" title="Permalink to this definition">¶</a></dt>
<dd><p>This function processes microbiome taxonomy and read abundance
data and maps microbial species on a microbial reconstruction database,
such as AGORA2 and APOLLO.</p>
<dl class="field-list simple">
<dt class="field-odd">INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>readsTablePath</strong> – String; path to the reads abundance file. If
taxonomic assignment is not present in this
file, provide taxonomy in taxaTablePath.</p></li>
<li><p><strong>cutoffMars</strong> – Numeric; value under which individual taxa
relative abundances are considered to be zero.
Optional, defaults to 1e-6</p></li>
<li><p><strong>flagLongSpecies</strong> – Boolean; indicates if the genus name is in the
name of the species. E.g., if the species name
is Prevotella copri, set to false. If the
species name is copri set to true. Optional,
defaults to true.</p></li>
<li><p><strong>taxaDelimiter</strong> – String; delimiter used to separate taxonomic
levels. Optional, defaults to ;</p></li>
<li><p><strong>removeClade</strong> – Boolean; specifies to remove clade name
extensions from all taxonomic levels of
microbiome taxa. If set to false, MARS might
find significantly less models in
AGORA2 and APOLLO databases, as clade
extensions are not included there. Optional,
defaults to true.</p></li>
<li><p><strong>reconstructionDb</strong> – String; defining if AGORA2, APOLLO, a
combination of both (full) or a user-defined database
should be used as the model database to check
presence in. Allowed Input (case-insensitive):
“AGORA2”, “APOLLO”, “full_db”, “user_db”.
Optional, defaults to full_db.</p></li>
<li><p><strong>userDbPath</strong> – String; The path to the user database if
reconstructionDb is set to user_db. Optional,
defaults to ‘’.</p></li>
<li><p><strong>sampleReadCountCutoff</strong> – Numeric; value for total read counts per sample
under which samples are excluded from analysis.
If the reads table is already normalised, set
this value to 0 or 0.1 to ensure that samples
are not removed.</p></li>
<li><p><strong>taxaTablePath</strong> – String; path to the file where taxonomies are
matched to taxonomic unit (OTU/ASV etc.) This
requires that the taxonomic unit is present in
the reads table as well in order to match the
taxonomy with the reads. Ensure that the column
with the taxonomic unit has the same header in
both the reads table and taxonomy table and
that the column with the taxonomies in the
taxonomy table is called ‘taxon’. Optional,
defaults to ‘’.</p></li>
<li><p><strong>outputPathMars</strong> – String; path to the directory where the output
of MARS is stored. Optional, defaults to
[pwd, filesep, ‘resultMars’].</p></li>
<li><p><strong>calculateBrayCurtis</strong> – Boolean; Specifies if the Bray-Curtis
dissimilarity index is calculated. Caution!
Putting this to true can greatly increase the
time MARS needs to run. Defaults to false.</p></li>
<li><p><strong>compoundedDatabase</strong> – Boolean, indicates if the database reads are
compounded or not. E.g., if the total number of
reads for an order o__x is the same as the
genus g__x. This means the reads assigned to
g__x are in the number of reads of o__x. This
requires a different data handling as compared
where o__x only has reads that could be
assocated up to that level. Defaults to false.</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>The function does not return variables but writes processed results</strong></p></li>
<li><p><strong>to the specified output directory in the MARS pipeline.</strong></p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt>AUTHOR:   Tim Hensen, July 2025</dt><dd><p>Bram Nap, August 2025</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.runNonparametricTests">
<span class="sig-name descname"><span class="pre">runNonparametricTests</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">data</span></em>, <em class="sig-param"><span class="pre">metadata</span></em>, <em class="sig-param"><span class="pre">predictor</span></em>, <em class="sig-param"><span class="pre">response</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/runNonparametricTests.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.runNonparametricTests" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs non-parametric statistical tests (Wilcoxon or Kruskal-Wallis) on metabolic data</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>resultTable = runNonparametricTests</strong> (<em>data, metadata, predictor, response</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>data</strong> – (table) m x n table containing:
* m samples
* n reactions/taxa with flux or abundance values
* First column must contain sample IDs</p></li>
<li><p><strong>metadata</strong> – (table) Sample metadata containing:
* Sample IDs in first column
* Response variable in separate column</p></li>
<li><p><strong>predictor</strong> – (char) Name of predictor variable (‘Flux’ or ‘relative_abundance’)</p></li>
<li><p><strong>response</strong> – (char) Name of response variable in metadata</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>resultTable</strong> – (table) Statistical results with columns:
* name:      Reaction/Taxa identifier
* groups:    Cell array of group names
* n:         Array of group sizes
* statistic: Test statistic (Wilcoxon or Kruskal-Wallis)
* p-value:   Unadjusted p-value
* FDR:       Benjamini-Hochberg adjusted p-value
* effectSize: Effect size (r for Wilcoxon, η² for Kruskal-Wallis)
* confidence: 95% confidence interval (only for binary comparisons)</p>
</dd>
</dl>
<p class="rubric">Example</p>
<p>% Binary comparison (Wilcoxon test)
resultTable = runNonparametricTests(fluxData, metadata, ‘Flux’, ‘Disease’)</p>
<p>% Multiple group comparison (Kruskal-Wallis test)
resultTable = runNonparametricTests(abundanceData, metadata, ‘relative_abundance’, ‘Treatment’)</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ol class="arabic simple">
<li><p>Automatically selects appropriate test based on number of groups</p></li>
<li><p>Data is automatically normalized before testing</p></li>
<li><p>Missing values (NaN) are removed before analysis</p></li>
<li><p>Multiple testing correction uses Benjamini-Hochberg FDR</p></li>
<li><p>Minimum group size of 3 samples is required for valid testing</p></li>
</ol>
</div>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.runPersephone">
<span class="sig-name descname"><span class="pre">runPersephone</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">configPath</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/runPersephone.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.runPersephone" title="Permalink to this definition">¶</a></dt>
<dd><p>This pipeline orchestrates the end-to-end process of constructing and
analysing human–microbiome whole-body models (WBMs). It integrates
sequencing data processing, metagenomic mapping, microbiome model
generation, WBM personalisation, host–microbiome model creation, flux
balance analysis (FBA), and statistical evaluation.</p>
<p>Updated: 2025.07.01 wbarton
Updated: 2025.09.09 asheehy
Updated: 2025.09.19 bnap</p>
<dl class="field-list simple">
<dt class="field-odd">Usage<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>runPersephone</strong> (<em>configPath</em>)</p>
</dd>
</dl>
<p>Overview of steps:
1. SeqC:                  Pre-process sequencing data and prepare input for MARS.
2. MARS:                  Map taxonomy-assigned reads to AGORA2/APOLLO reconstructions</p>
<blockquote>
<div><p>to generate relative abundance files.</p>
</div></blockquote>
<ol class="arabic simple" start="3">
<li><p>MgPipe:                Build microbiome community models using relative abundances.</p></li>
<li><dl class="simple">
<dt>WBM personalisation:   Personalise WBMs with physiological and/or</dt><dd><p>metabolomic data.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>mWBM Creation:         Combine WBMs (personalised or base) with microbiome models</dt><dd><p>to create host–microbiome WBMs (miWBMs).</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>FBA:                   Perform flux balance analysis on mWBMs using user-specified</dt><dd><p>reaction objectives, followed by flux post-processing.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>Statistics:            Run statistical analyses (e.g., mixed-effects linear</dt><dd><p>regression) on flux and abundance results,
accounting for metadata-specified predictors and confounders.</p>
</dd>
</dl>
</li>
</ol>
<p>Inputs %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</p>
<dl class="simple">
<dt>configPath<span class="classifier">Path to a .mat file which should contain all of the</span></dt><dd><p>necessary inputs needed to run specific or all sections of
this pipeline. A template of this config file can be found
at: <a class="reference external" href="https://github.com/opencobra/cobratoolbox/tree/master/src/analysis/persephone">https://github.com/opencobra/cobratoolbox/tree/master/src/analysis/persephone</a>
or you can fill in an online version and download at: <a class="reference external" href="https://vmh2.life/persephone">https://vmh2.life/persephone</a></p>
</dd>
</dl>
<p>Outputs %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%</p>
<p>All results will be saved to your computer in the directories you can
choose in the config file. The output ‘progress’ will also be produced
and saved to your workspace. It will detail all sections that have been
run and how long they took.</p>
<p>%%%%%%%%%%%%%%%%%%% PERSEPHONE INITIALISATION %%%%%%%%%%%%%%%%%%%%%%%%%
Start timer</p>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.runSeqC">
<span class="sig-name descname"><span class="pre">runSeqC</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">repoPathSeqC</span></em>, <em class="sig-param"><span class="pre">outputPathSeqC</span></em>, <em class="sig-param"><span class="pre">fileIDSeqC</span></em>, <em class="sig-param"><span class="pre">procKeepSeqC</span></em>, <em class="sig-param"><span class="pre">maxMemSeqC</span></em>, <em class="sig-param"><span class="pre">maxCpuSeqC</span></em>, <em class="sig-param"><span class="pre">maxProcSeqC</span></em>, <em class="sig-param"><span class="pre">debugSeqC</span></em>, <em class="sig-param"><span class="pre">runApptainer</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/runSeqC.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.runSeqC" title="Permalink to this definition">¶</a></dt>
<dd><dl>
<dt>======================================================================================================#</dt><dd><p>Title: SeqC as Flux Pipeline MATLAB Wrapper
Author: Wiley Barton
Modified code sources:</p>
<blockquote>
<div><p>matlab script structure: Tim Hensen, runMars.m, 2025.01
assistance and reference from a generative AI model [ChatGPT](<a class="reference external" href="https://chatgpt.com/">https://chatgpt.com/</a>)</p>
<blockquote>
<div><p>clean-up and improved readability</p>
</div></blockquote>
</div></blockquote>
<p>Last Modified: 2025.10.02 - wbarton
Part of: Persephone Pipeline</p>
<dl>
<dt>Description:</dt><dd><p>This function builds and runs the SeqC Docker image from a MATLAB environment.
It ensures necessary databases exist and coordinates execution with the MARS pipeline.</p>
</dd>
<dt>Inputs:</dt><dd><ul class="simple">
<li><p>repoPathSeqC (char) : Path to the SeqC repository</p></li>
<li><p>outputPathSeqC (char) : Path for SeqC output</p></li>
<li><p>fileIDSeqC (char) : Unique identifier for file processing</p></li>
<li><p>procKeepSeqC (logical) : Keep all files (true/false)</p></li>
<li><p>maxMemSeqC (int) : Maximum memory allocation for SeqC</p></li>
<li><p>maxCpuSeqC (int) : Maximum CPU allocation for SeqC</p></li>
<li><p>maxProcSeqC (int) : Maximum processes for SeqC</p></li>
<li><p>debugSeqC (logical) : Enable debug mode (true/false)</p></li>
<li><p>runApptainer (logical) : Enable apptainer wrapping (true/false)</p></li>
</ul>
<p>…</p>
</dd>
<dt>Dependencies:</dt><dd><ul class="simple">
<li><p>MATLAB</p></li>
<li><p>Docker installed and accessible in the system path</p></li>
</ul>
</dd>
<dt>Optional:</dt><dd><ul class="simple">
<li><p>Apptainer (formerly Singularity) version 1.3.4</p></li>
</ul>
</dd>
</dl>
</dd>
<dt>======================================================================================================#</dt><dd><p>Determine Operating System</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.persephone.validatePersephoneInputs">
<span class="sig-name descname"><span class="pre">validatePersephoneInputs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">paths</span></em>, <em class="sig-param"><span class="pre">resultPath</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/persephone/validatePersephoneInputs.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.persephone.validatePersephoneInputs" title="Permalink to this definition">¶</a></dt>
<dd><p>Validates the inputs provided in the <cite>paths</cite> structure for the Persephone
pipeline. This function ensures that all required fields, flags, and
parameters are present, appropriately formatted, and meet the expected
constraints. Validation is performed across multiple pipeline components,
including MARS, mgPipe, WBM personalization, HM creation, FBA, and statistics.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>validated = validatePersephoneInput</strong> (<em>paths</em>)</p>
</dd>
<dt class="field-even">INPUT<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>paths       Structure containing all required input fields for the Persephone</strong> – pipeline. Fields include:
- General flags and paths (e.g., <cite>flagSeqC</cite>, <cite>resultPath</cite>)
- Component-specific flags and parameters (e.g., <cite>flagMars</cite>,
<cite>marsRepoPath</cite>, <cite>flagMgPipe</cite>, <cite>metadataPath</cite>, etc.)</p>
</dd>
<dt class="field-odd">OUTPUT<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>validated   Logical scalar indicating whether the validation succeeded</strong> – (true) or failed (false).</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>Ensure all required flags and fields in the <cite>paths</cite> structure are defined
prior to invoking this function. Undefined or mismatched fields will
result in validation failure.</p></li>
<li><p>This function is designed to be modular and extendable for future
additions to the Persephone pipeline.</p></li>
</ul>
<dl class="simple">
<dt>AUTHOR:</dt><dd><p>Tim Hensen, January 2025</p>
</dd>
</dl>
<p>See also: validateattributes, validatestring, COBRA Toolbox</p>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="additionalFunctions/index.html" class="btn btn-default float-right" title="Additionalfunctions" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../thermo/molFiles/old/index.html" class="btn btn-default" title="Old" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2025, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>