

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Pairwiseinteractionmodeling &mdash; The COBRA Toolbox</title>
  

  
  
  
    <link rel="apple-touch-icon" sizes="180x180" href="../../../../../_static/img/apple-touch-icon.png">
  
  
    <link rel="icon" type="image/png" sizes="32x32" href="../../../../../_static/img/favicon-32x32.png">
  
  
    <link rel="icon" type="image/png" sizes="16x16" href="../../../../../_static/img/favicon-16x16.png">
  
  
    <link rel="mask-icon" href="../../../../../_static/img/safari-pinned-tab.svg" color="#5bbad5">
  
  
    <link rel="manifest" href="../../../../../_static/json/manifest.json">
  
  
    <meta name="msapplication-config" content="../../../../../_static/browserconfig.xml">
  
  <meta name="theme-color" content="#ffffff">


  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
  <script src="https://code.jquery.com/jquery-1.11.2.min.js"></script>
  <script src="../../../../../_static/js/jquery.easy-autocomplete.min.js" type="text/javascript" ></script>

    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>
  <link href="../../../../../_static/css/easy-autocomplete.min.css" rel="stylesheet" type="text/css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../../../_static/css/cobra.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../../../_static/pygments.css" type="text/css" />
  
    <link rel="stylesheet" href="../../../../../_static/css/cobra.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../../../../../genindex.html"/>
        <link rel="search" title="Search" href="../../../../../search.html"/>
    <link rel="top" title="The COBRA Toolbox" href="../../../../../index.html"/>
        <link rel="up" title="Microbiomemodelingtoolbox" href="../index.html"/>
        <link rel="next" title="Additionalanalysis" href="../additionalAnalysis/index.html"/>
        <link rel="prev" title="Mgpipe" href="../mgPipe/index.html"/> 

  
  <script src="../../../../../_static/js/modernizr.min.js"></script>
</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../../../../../index.html" class="icon icon-home"> The COBRA Toolbox
          

          
            <img src="https://king.nuigalway.ie/cobratoolbox/img/cobratoolbox_logo_transparent.png" class="logo" alt="cobratoolbox logo" />
          
          </a>

              
          <h1>The COBRA Toolbox</h1>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" id="simple"/>
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
    <script>
    	var options = {

		url: "../../../../../_static/json/functions.json",

		getValue: "name",

		list: {
            onKeyEnterEvent: function(item) {
                    return "_static/../" + item.website_url
            },
            maxNumberOfElements: 25,
			match: {
					enabled: true,
			},
            sort: {
			    enabled: true
		    }
		},

    	template: {
	    	type: "custom",
		    method: function(value, item) {
                return "<span><a href='../../../../../search.html?q=" + item.name + "&check_keywords=yes&area=default'>" + value + "</a>" + "<a href='../../../../../_static/../" + item.website_url + "' ><img src='../../../../../_static/link.svg'/></a></span>";
		    }
	    },

		theme: "square"

	};

	$("#simple").easyAutocomplete(options);


    </script>
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../../../../../index.html">Home</a></li>
              </ul>
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../../installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../../index.html">Functions</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../../index.html">Analysis</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../../../QFCA/index.html">Qfca</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rMTA/index.html">Rmta</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../MOMA/index.html">Moma</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../percentileCalc/index.html">Percentilecalc</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../geometricFBA/index.html">Geometricfba</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="../../index.html">Multispecies</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="../index.html">Microbiomemodelingtoolbox</a><ul class="current">
<li class="toctree-l5"><a class="reference internal" href="../mgPipe/index.html">Mgpipe</a></li>
<li class="toctree-l5 current"><a class="current reference internal" href="#">Pairwiseinteractionmodeling</a></li>
<li class="toctree-l5"><a class="reference internal" href="../additionalAnalysis/index.html">Additionalanalysis</a></li>
</ul>
</li>
<li class="toctree-l4"><a class="reference internal" href="../../SteadyCom/index.html">Steadycom</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../robustnessAnalysis/index.html">Robustnessanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../KLdistance/index.html">Kldistance</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../ICONGEMs/index.html">Icongems</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rFBA/index.html">Rfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../nutritionAlgorithm/index.html">Nutritionalgorithm</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../topology/index.html">Topology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../parsimoniousFBA/index.html">Parsimoniousfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FVA/index.html">Fva</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../exploration/index.html">Exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../coupling/index.html">Coupling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../fluxEnrichmentAnalysis/index.html">Fluxenrichmentanalysis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../rumba/index.html">Rumba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../deletion/index.html">Deletion</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../igemRNA/index.html">Igemrna</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../fastForest/index.html">Fastforest</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../FBA/index.html">Fba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../findMIIS/index.html">Findmiis</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../persephone/index.html">Persephone</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../thermo/index.html">Thermo</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sampling/index.html">Sampling</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../wholeBody/index.html">Wholebody</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../mdFBA/index.html">Mdfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../transport/index.html">Transport</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../dynamicFBA/index.html">Dynamicfba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../uFBA/index.html">Ufba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../sparseFBA/index.html">Sparsefba</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../ROOM/index.html">Room</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../gMCS/index.html">Gmcs</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../subspaces/index.html">Subspaces</a></li>
<li class="toctree-l3"><a class="reference internal" href="../../../relaxedFBA/index.html">Relaxedfba</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../../base/index.html">Base</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../dataIntegration/index.html">Dataintegration</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../design/index.html">Design</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../reconstruction/index.html">Reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../../visualization/index.html">Visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../tutorials/index.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributing.html">How to contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../cite.html">How to cite</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../support.html">Support</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contributors.html">Contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../funding.html">Funding</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../plan.html">Development Plan</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../contact.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../../citations.html">Citations</a></li>
</ul>

          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../../index.html">The COBRA Toolbox</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../../../contents.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../../../index.html">Functions</a></li>
          <li class="breadcrumb-item"><a href="../../../index.html">Analysis</a></li>
          <li class="breadcrumb-item"><a href="../../index.html">Multispecies</a></li>
          <li class="breadcrumb-item"><a href="../index.html">Microbiomemodelingtoolbox</a></li>
      <li class="breadcrumb-item active">Pairwiseinteractionmodeling</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="https://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="pairwiseinteractionmodeling">
<span id="id1"></span><h1>Pairwiseinteractionmodeling<a class="headerlink" href="#pairwiseinteractionmodeling" title="Permalink to this heading">¶</a></h1>
<span class="target" id="module-src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling"></span><dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.calculateInteractionsByTaxon">
<span class="sig-name descname"><span class="pre">calculateInteractionsByTaxon</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">pairwiseInteractions</span></em>, <em class="sig-param"><span class="pre">taxonInformation</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/calculateInteractionsByTaxon.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.calculateInteractionsByTaxon" title="Permalink to this definition">¶</a></dt>
<dd><p>Part of the Microbiome Modeling Toolbox. This script analyzes computed pairwise
interactions on different taxonomical levels to yield the interactions
that each taxon (genus, family, order, class, phylum) displayed in total.
The computed pairwise interactions and the taxnomical information for each
analyzed strain are required as inputs.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[interactionsByTaxon]=calculateinteractionsByTaxon</strong> (<em>pairwiseInteractions,taxonInfo</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>pairwiseInteractions</strong> – a table with pairwise interactions computed
for a number of analyzed microbe models (output of the function
simulatePairwiseInteractions)</p></li>
<li><p><strong>taxonInformation</strong> – a table with taxonomical information on the
analyzed microbes. Needs to contain at least six columns:
1. The names of the analyzed model structures
in the first column
2-6. A column with the appropriate information
and the header
‘Genus’,’Family’,’Order’,’Class’,’Phylum’
respectively.</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUT<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>interactionsByTaxon</strong> – a structure with the outcomes predicted for
all taxa on each taxonomical level.</p>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeParetoOptimality">
<span class="sig-name descname"><span class="pre">computeParetoOptimality</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">rxn1</span></em>, <em class="sig-param"><span class="pre">rxn2</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/computeParetoOptimality.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeParetoOptimality" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs Pareto optimality analysis for two objective functions by
simultaneously optimizing two reactions (e.g., the biomass objective
functions of two joined organisms). The result is a depiction of the
tradeoff between the two competing objectives.</p>
<p>Multiobjective analysis of two reactions is performed by using the
approach described in Oberhardt, M. A., J. B. Goldberg, et al. (2010).
“Metabolic network analysis of Pseudomonas aeruginosa during chronic
cystic fibrosis lung infection.” J Bacteriol 192(20): 5534-5548. One
reaction is fixed at different intervals and the other reaction is
optimized.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[ParetoFrontier,fluxSolutions,minFluxes,maxFluxes] = computeParetoOptimality</strong> (<em>model,rxn1,rxn2,dinc,FVAflag</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>model</strong> – COBRA metabolic reconstruction</p></li>
<li><p><strong>rxn1</strong> – Reaction ID of the first reaction to be optimized</p></li>
<li><p><strong>rxn2</strong> – Reaction ID of the second reaction to be optimized</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dinc</strong> – An index which indicates the distance between steps at
which the flux of each reaction is fixed (default=0.001).</p></li>
<li><p><strong>FVAflag</strong> – If true, flux variability analysis is performed at each
step.</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>ParetoFrontier</strong> – Lists the objective values for both reactions next to
the interval step. Column 1:the interval step. Column 2
and 3: the flux values of rxn1 in and rxn2,
respectively, at the interval step.</p>
</dd>
</dl>
<dl class="simple">
<dt>OPTIONAL OUTPUTS:</dt><dd><dl class="simple">
<dt>fluxSolutions:    Contains the flux solutions every interval step as a</dt><dd><p>matrix of structures</p>
</dd>
<dt>minFluxes:        Reports fastFVA results for every step (minFlux) with</dt><dd><p>one column for each interval step</p>
</dd>
<dt>maxFluxes:        Reports fastFVA results for every step (maxFlux) with</dt><dd><p>one column for each interval step</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeRescuedGenes">
<span class="sig-name descname"><span class="pre">computeRescuedGenes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/computeRescuedGenes.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.computeRescuedGenes" title="Permalink to this definition">¶</a></dt>
<dd><p>Part of the Microbiome Modeling Toolbox. This function determines the
effect of the presence of another species on gene deletions in a species.
A joint model consisting of two species is entered. For each gene
deletion that has an effect in each single model, the function conputes
whether the presence of the other species can rescue the flux through the
objective function (e.g., biomass).</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[OptSolKO,OptSolWT,OptSolRatio,fluxesKO]=computeRescuedGenes</strong> (<em>varargin</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>modelJoint</strong> – Joint model structure consisting of two COBRA models</p></li>
<li><p><strong>Rxn1</strong> – Reaction in the first joined COBRA model for which the
effect of gene deletions is calculated</p></li>
<li><p><strong>Rxn2</strong> – Reaction in the second joined COBRA model for the effect
of gene deletions is calculated</p></li>
<li><p><strong>nameTag1</strong> – Species identifier for reactions of the first model</p></li>
<li><p><strong>nameTag2</strong> – Species identifier for reactions of the second model</p></li>
<li><p><strong>OriModel1</strong> – Original COBRA model structure of first model
(needed to determine reactions to be constrained)</p></li>
<li><p><strong>OriModel2</strong> – Original COBRA model structure of second model
(needed to determine reactions to be constrained)</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>OptSolKO</strong> – Matlab structure containing the computed optimal solutions
for the gene deletions that had an effect on the two
reactions</p></li>
<li><p><strong>OptSolRatio</strong> – Matlab structure containing all knockout to wildtype
optimal solution ratios for the gene deletions that had
an effect on the two reactions</p></li>
<li><p><strong>OptSolWT</strong> – Matlab structure containing wildtype flux values for both
reactions in the two models</p></li>
<li><p><strong>RescuedGenes</strong> – List of genes for which deletion is lethal or causes &gt;50%
growth reduction in single but not pairwise model</p></li>
<li><p><strong>fluxesKO</strong> – Matlab structure containing all solutions for all gene
deletions and for both reactions in the two models. Can be
used to identify mechanisms for rescued genes.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.createMultipleSpeciesModel">
<span class="sig-name descname"><span class="pre">createMultipleSpeciesModel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">models</span></em>, <em class="sig-param"><span class="pre">biomasses</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/createMultipleSpeciesModel.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.createMultipleSpeciesModel" title="Permalink to this definition">¶</a></dt>
<dd><p>Based on the implementation from <em>Klitgord and Segre 2010, PMID 21124952</em>.
The present implementation has been used in <em>PMID 23022739</em>, <em>PMID 25841013</em>,
<em>PMID 25901891</em>, <em>PMID 27893703</em>.</p>
<p>Joins one or more COBRA models with or without another COBRA model
representing the host. The created setup when a host is entered is
depicted schematically in Figures 1 and 2 in <em>PMID 27893703</em>.</p>
<p>Creates a common space u (lumen) through which all cells can feed and exchange metabolites,
and separate extracellular spaces for all joined models.
If a host model is entered, a separate compartment <cite>b</cite> (body fluids) which has no
connection to extracellular space is created for the host. Metabolites can be transported from
the lumen to <cite>e</cite>, from <cite>e</cite> to the cytosol and from the cytosol to <cite>b</cite>,
but not from body fluids to the lumen.</p>
<dl class="field-list">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[modelJoint] = createMultipleSpeciesModel</strong> (<em>models, varargin</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul>
<li><p><strong>models</strong> – cell array of COBRA models(at least one).
Format</p>
<blockquote>
<div><ul class="simple">
<li><p>models{1,1} = model 1</p></li>
<li><p>models{2,1} = model 2…</p></li>
</ul>
</div></blockquote>
</li>
<li><p><strong>biomasses</strong> – Cell array containing names of biomass objective
functions of models to join. Needs to be the same
length as models.</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul>
<li><p><strong>nameTagsModels</strong> – cell array of tags for reaction/metabolite abbreviation
corresponding to each model.
Format</p>
<blockquote>
<div><ul class="simple">
<li><p>nameTagsModels{1,1} = ‘name tag 1’</p></li>
<li><p>nameTagsModels{2,1} = ‘name tag 2’…</p></li>
</ul>
</div></blockquote>
</li>
<li><p><strong>modelHost</strong> – COBRA model for host</p></li>
<li><p><strong>nameTagHost</strong> – string of tag for reaction/metabolite abbreviation of host model</p></li>
<li><p><strong>mergeGenesFlag</strong> – If true, the gene associations in both models are
included in the joined model. If false, empty fields are created
instead (default:false). Note: merging genes is time-consuming
and may crash certain models.</p></li>
<li><p><strong>remCyclesFlag</strong> – If true, the function will attempt to remove futile
cycles that appear after merging the models (default: false).</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUT<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>modelJoint</strong> – model structure for joint model</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This function assumes, that exchange reactions are identified by
containing ‘EX’ in the reaction name and that no other reactions do have this property!</p>
</div>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.getMultiSpeciesModelId">
<span class="sig-name descname"><span class="pre">getMultiSpeciesModelId</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modelJoint</span></em>, <em class="sig-param"><span class="pre">nameTagsModels</span></em>, <em class="sig-param"><span class="pre">nameTagHost</span></em>, <em class="sig-param"><span class="pre">metTagRe</span></em>, <em class="sig-param"><span class="pre">rxnTagRe</span></em>, <em class="sig-param"><span class="pre">compCom</span></em>, <em class="sig-param"><span class="pre">compHost</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/getMultiSpeciesModelId.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.getMultiSpeciesModelId" title="Permalink to this definition">¶</a></dt>
<dd><p>Get names and IDs for metabolites and exchange reactions in the [u] and [b] space</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[names, ids] = getMultiSpeciesModelId</strong> (<em>modelJoint, nameTagsModels, nameTagHost, metTagRe, rxnTagRe, compCom, compHost</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>modelJoint</strong> – COBRA multi-organism model</p></li>
<li><p><strong>nameTagsModels</strong> – cell array of tags for species to identify the respective
reactions and metabolites from <cite>modelJoint.rxns</cite> and <cite>.mets</cite></p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>nameTagHost</strong> – string of tag for the host model if exist. Input [] if no host is present</p></li>
<li><p><strong>metTagRe</strong> – a regular expression to identify the tag in nameTagsModels from <cite>modelJoint.mets</cite>.
Use ‘%s’ for the tag in <cite>nameTagsModels</cite> for each species.
Default ‘^%s’, the beginning of the string. In this case, if the tag for species 1 is
‘SP1’, then mets with id ‘SP1glc-D[e]’ will be identified as belonging to species 1.
E.g., ‘met[compartment_SP1]’ where ‘SP1’ is the tag for species 1, input ‘[[^_]+_%s]$’.</p></li>
<li><p><strong>rxnTagRe</strong> – a regular expression to idenify the tag in <cite>nameTagsModels</cite> from <cite>modelJoint.rxns</cite>.
Default ‘^%s’, the beginning of the string</p></li>
<li><p><strong>compCom</strong> – compartment Id for the community exchange compartment. Default ‘u’</p></li>
<li><p><strong>compHost</strong> – compartment Id for the exchange compartment specific to the host. Default ‘b’</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>names</strong> – structure of reaction/metabolite name (.rxns/.mets) with the following fields:</p>
<ul>
<li><p>spAbbr - species abbreviation (= <cite>nameTags</cite>). Host put at the end</p></li>
<li><p>EXcom - #met[u]-by-1 cell. Exchange reactions for community metabolites <cite>met[u]</cite></p></li>
<li><p>EXhost - #met[b]-by-1 cell. Exchange reactions for host-specific exchange metabolites <cite>met[b]</cite></p></li>
<li><p>EXsp - #met[u]-by-#species cell. <cite>EXsp(i,j)</cite> is the species-community exchange reactions
for the i-th met[u] and the j-th species</p></li>
<li><p>Mcom - #met[u]-by-1 cell. All community metabolites <cite>met[u]</cite></p></li>
<li><p>Mhost - #met[u]-by-1 cell. Host-specific exchange metabolites <cite>met[b]</cite></p></li>
<li><p>Msp - #met[u]-by-#species cell. <cite>Msp(i,j)</cite> is the metabolite in <cite>met[e]</cite> of the <cite>j</cite>-th species
participating in reaction <cite>EXsp(i,j)</cite></p></li>
<li><p>rxnSps - #rxns-by-1 cell. <cite>rxnSps(i)</cite> would be <cite>spAbbr(j)</cite> if the <cite>i</cite>-th reaction belongs to the <cite>j</cite>-th species</p></li>
<li><p>metSps - #mets-by-1 cell. <cite>metSps(i)</cite> would be <cite>spAbbr(j)</cite> if the i-th metabolite belongs to the <cite>j</cite>-th species</p></li>
</ul>
</li>
<li><p><strong>ids</strong> – structure of reaction/metabolite index with the same fields as names except without <cite>spAbbr</cite></p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.joinModelsPairwiseFromList">
<span class="sig-name descname"><span class="pre">joinModelsPairwiseFromList</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modelList</span></em>, <em class="sig-param"><span class="pre">modelFolder</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/joinModelsPairwiseFromList.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.joinModelsPairwiseFromList" title="Permalink to this definition">¶</a></dt>
<dd><p>This function joins a list of microbial genome-scale reconstructions in
all combinations. Models are not paired with themselves and pairs are
only created once (model1+model2 = model2+model1). The reactions in each
joined reconstruction structure receive a suffix (entered as input
parameter model list).
Coupling constraints are created in which the flux through all reactions
of each individual microbe is coupled to the flux through its own biomass
objective function. By default, coupling constraints are implemented with
a coupling factor c of 400 and a threshold u of 0. The coupling
parameters can be changed by entering the input parameters c and u.
By default, the gene-protein-reaction associations of the joined models
are not merged as the gene associations are not needed for the simulation
of pairwise interactions and significantly slow down pairwise model
creation.  If merging of genes is desired, please set the input parameter
mergeGenes to true. If you are not using AGORA/AGORA2 reconstructions,
you may need to define the biomass objective functions manually.</p>
<p>Please note: the function takes a cell array with the names of the
COBRA models to join as well as the name of the folder where they are
located at as the inputs, e.g.:
joinModelsPairwiseFromList(modelList,’/Users/almut.heinken/Documents/AGORA’);</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>joinModelsPairwiseFromList</strong> (<em>modelList,  modelFolder, varargin</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>modelList</strong> – Cell array with names of reconstruction structures to be
joined</p></li>
<li><p><strong>modelFolder</strong> – Path to folder with COBRA model structures to be joined</p></li>
</ul>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>biomasses</strong> – Cell array containing names of biomass objective
functions of models to join. Needs to be the same
length as modelList.</p></li>
<li><p><strong>c</strong> – Coupling factor by which reactions in each joined model are
coupled to its biomass reaction (default: 400)</p></li>
<li><p><strong>u</strong> – Threshold representing minimal flux allowed when flux
through biomass reaction in zero (default: 0)</p></li>
<li><p><strong>mergeGenes</strong> – If true, genes in the joined models are merged and included
in the joined model structure (default: false)</p></li>
<li><p><strong>remCyclesFlag</strong> – If true, futile cycles that may arise from joining the
models are removed (default: false). Recommended
if AGORA/AGORA2 reconstructions are used, does not work
with other namespaces.</p></li>
<li><p><strong>numWorkers</strong> – Number of workers in parallel pool if desired</p></li>
<li><p><strong>pairwiseModelFolder</strong> – Folder where pairwise models will be saved</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.printUptakeBoundCom">
<span class="sig-name descname"><span class="pre">printUptakeBoundCom</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">model</span></em>, <em class="sig-param"><span class="pre">SpFlag</span></em>, <em class="sig-param"><span class="pre">metFlag</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/printUptakeBoundCom.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.printUptakeBoundCom" title="Permalink to this definition">¶</a></dt>
<dd><p>Prints the uptake bounds of the whole community and individual species for a community COBRA model</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>rxnIDs = printUptakeBoundCom</strong> (<em>model, SpFlag, metFlag</em>)</p>
</dd>
<dt class="field-even">INPUT<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>model</strong> – the community model with field <cite>.infoCom</cite> or <cite>.indCom</cite> indicating the indicies of
community exchange reactions/metabolites. Can be obtained from <cite>getMultiSpeciesModelId.m</cite></p>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>SpFlag</strong> – true to show individual uptake rates though community uptake is not allowed (default false)</p></li>
<li><p><strong>metFlag</strong> – true to print with <cite>model.metNames</cite> (default false)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>printMatrix</strong> – matrix of the uptake bounds being printed</p></li>
<li><p><strong>printMet</strong> – column of metabolites whose uptake bounds are printed</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.simulatePairwiseInteractions">
<span class="sig-name descname"><span class="pre">simulatePairwiseInteractions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">pairwiseModelFolder</span></em>, <em class="sig-param"><span class="pre">varargin</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/simulatePairwiseInteractions.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.simulatePairwiseInteractions" title="Permalink to this definition">¶</a></dt>
<dd><p>This function predicts the outcome of pairwise simulations in every
combination from a given list of pairwise models. The pairwise models
need to be created first with the function joinModelsPairwiseFromList.
This script requires the COBRA Toolbox function solveCobraLP. Due to the
coupling constraints on the pairwise models, the simulations cannot
currently be run with optimizeCbModel.</p>
<p>Below is a description of all possible consequences for the two joined
organisms. Please note that the outcomes depend on the two genome-scale
reconstructions joined and are highly dependent on the applied
constraints.</p>
<ul class="simple">
<li><p>Competition: both organisms grow slower in co-growth than separately
(same outcome for both).</p></li>
<li><p>Parasitism: one organism grows faster in co-growth than separately, while
the other grows slower in co-growth than separately. Outcome for
faster-growing organism: Parasitism_Taker, for slower-growing organism:
Parasitism_Giver</p></li>
<li><p>Amensalism: one organism’s growth is unaffected by co-growth,  while
the other grows slower in co-growth than separately. Outcome for
unaffected organism: Amensalism_Unaffected, for slower-growing organism:
Amensalism_Affected</p></li>
<li><p>Neutralism: both organisms’ growths are unaffected by co-growth (same
outcome for both)</p></li>
<li><p>Commensalism: one organism’s growth is unaffected by co-growth,  while
the other grows fatser in co-growth than separately. Outcome for
unaffected organism: Commensalism_Giver, for slower-growing organism:
Commensalism_Taker</p></li>
<li><p>Mutualism: both organisms growth faster in co-growth than separately
(same outcome for both)</p></li>
</ul>
<p>Please note: the function takes the name of the folder containing
previously created pairwise models as the input, e.g.:
pairwiseInteractions, pairwiseSolutions] =
simulatePairwiseInteractions(‘/Users/almut.heinken/Documents/PairwiseModels’);</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[pairwiseInteractions, pairwiseSolutions] = simulatePairwiseInteractions</strong> (<em>pairwiseModelFolder, varargin</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pairwiseModelFolder</strong> – Folder where pairwise models and pairwise model info file are located</p>
</dd>
<dt class="field-odd">OPTIONAL INPUTS<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>inputDiet</strong> – Cell array of strings with three columns containing
exchange reaction model abbreviations, lower bounds,
and upper bounds.
If no diet is input then the input pairwise models
will be used with unchanged constraints.</p></li>
<li><p><strong>saveSolutionsFlag</strong> – If true, flux solutions are stored (may result in
large output files)</p></li>
<li><p><strong>numWorkers</strong> – Number of workers in parallel pool if desired</p></li>
<li><p><strong>sigD</strong> – Difference in growth rate that counts as significant
change (by default: 10%)</p></li>
</ul>
</dd>
<dt class="field-even">OUTPUTS<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pairwiseInteractions</strong> – Table with computed pairwise and single growth
rates for all entered microbe-microbe models</p>
</dd>
</dl>
<dl class="simple">
<dt>OPTIONAL OUTPUT:</dt><dd><dl class="simple">
<dt>pairwiseSolutions:       Table with all computed flux solutions saved</dt><dd><p>(may result in large output files)</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="mat function">
<dt class="sig sig-object mat" id="src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.useDiet">
<span class="sig-name descname"><span class="pre">useDiet</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">modelIn</span></em>, <em class="sig-param"><span class="pre">dietConstraints</span></em>, <em class="sig-param"><span class="pre">printLevel</span></em><span class="sig-paren">)</span><a class="reference external" href="https://github.com/opencobra/cobratoolbox/blob/master/src/analysis/multiSpecies/microbiomeModelingToolbox/pairwiseInteractionModeling/useDiet.m"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#src.analysis.multiSpecies.microbiomeModelingToolbox.pairwiseInteractionModeling.useDiet" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements diet constraints in a COBRA model structure.</p>
<dl class="field-list simple">
<dt class="field-odd">USAGE<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>[modelOut] = useDiet</strong> (<em>modelIn, dietConstraints</em>)</p>
</dd>
<dt class="field-even">INPUTS<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>modelIn</strong> – original model</p></li>
<li><p><strong>dietConstraints</strong> – cell array of three columns containing
exchanges, lower bounds, and  upper bounds respectively</p></li>
<li><p><strong>printLevel</strong> – Verbose level (default: printLevel = 1)</p></li>
</ul>
</dd>
<dt class="field-odd">OUTPUT<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>modelOut</strong> – model with applied constraints</p>
</dd>
</dl>
</dd></dl>

</section>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../additionalAnalysis/index.html" class="btn btn-default float-right" title="Additionalanalysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../mgPipe/index.html" class="btn btn-default" title="Mgpipe" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017-2021, The COBRA Toolbox developers.

    </p>
  </div>
  Built with Sphinx using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../../../',
            VERSION:'2.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
    <script type="text/javascript" src="../../../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../../_static/sphinx_highlight.js"></script>

  


  
  
    <script type="text/javascript" src="../../../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TRCMZL1FKK"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-TRCMZL1FKK');
  </script>
    
</body>
</html>